{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Name: Umad ul hassan Rai\n",
    "# Exercise 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pyspark.sql import SparkSession\n",
    "from pyspark.sql import SQLContext\n",
    "from pyspark.sql.types import *\n",
    "from pyspark.sql.types import StringType\n",
    "from pyspark.sql.types import StructType\n",
    "from pyspark.sql.types import StructField\n",
    "from pyspark.sql import Row\n",
    "from pyspark.sql.functions import concat, col, lit, split, udf, size\n",
    "from pyspark.sql import functions as sf\n",
    "from operator import add\n",
    "import csv\n",
    "import time\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# initialise Spark Session\n",
    "sparkSession = SparkSession.builder.appName(\"Exercise1\").getOrCreate()\n",
    "sc = sparkSession.sparkContext\n",
    "sqlC = SQLContext(sc)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exercise 1.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('f05bcffe7951de9e5a32fff4a42eb088',\n",
       "  ['1158654',\n",
       "   '478707',\n",
       "   '12054725',\n",
       "   '6670515',\n",
       "   '781057',\n",
       "   '13329754',\n",
       "   '781055',\n",
       "   '2825',\n",
       "   '553840',\n",
       "   '988447',\n",
       "   '12938139',\n",
       "   '12037219',\n",
       "   '12788583',\n",
       "   '6595566',\n",
       "   '4027225',\n",
       "   '920055',\n",
       "   '3129258',\n",
       "   '2242776',\n",
       "   '3112352',\n",
       "   '144287',\n",
       "   '706033',\n",
       "   '525396',\n",
       "   '622633',\n",
       "   '9172127',\n",
       "   '7357993',\n",
       "   '230211',\n",
       "   '12790816',\n",
       "   '503161',\n",
       "   '12937120',\n",
       "   '942241',\n",
       "   '166220',\n",
       "   '8493138',\n",
       "   '2945819',\n",
       "   '227173',\n",
       "   '11191048',\n",
       "   '949352',\n",
       "   '227096',\n",
       "   '11597926',\n",
       "   '921623',\n",
       "   '833638',\n",
       "   '882809',\n",
       "   '12738996',\n",
       "   '99',\n",
       "   '7154210',\n",
       "   '11852474',\n",
       "   '72879',\n",
       "   '10723139',\n",
       "   '11923443',\n",
       "   '2862386',\n",
       "   '802634',\n",
       "   '1959297',\n",
       "   '4488840',\n",
       "   '12735311',\n",
       "   '5102465',\n",
       "   '5177766',\n",
       "   '129',\n",
       "   '1467354',\n",
       "   '9396734',\n",
       "   '494336',\n",
       "   '1218954',\n",
       "   '2783692',\n",
       "   '3733678',\n",
       "   '8423325',\n",
       "   '1160828',\n",
       "   '507529',\n",
       "   '7804574',\n",
       "   '5201306',\n",
       "   '407124',\n",
       "   '504896',\n",
       "   '3281478',\n",
       "   '13687046',\n",
       "   '6486855',\n",
       "   '13207613',\n",
       "   '11847579',\n",
       "   '556147',\n",
       "   '10362464',\n",
       "   '6531882',\n",
       "   '714977',\n",
       "   '9946377',\n",
       "   '742892',\n",
       "   '3044720',\n",
       "   '1119267',\n",
       "   '7538620',\n",
       "   '1426283',\n",
       "   '2705980',\n",
       "   '9001253',\n",
       "   '2121165',\n",
       "   '12901647',\n",
       "   '10462916',\n",
       "   '261290',\n",
       "   '921399',\n",
       "   '6196237',\n",
       "   '5664256',\n",
       "   '213068',\n",
       "   '1769635',\n",
       "   '1055600',\n",
       "   '876703',\n",
       "   '7623038',\n",
       "   '1405472',\n",
       "   '1320137',\n",
       "   '4492925',\n",
       "   '1825619',\n",
       "   '13408130',\n",
       "   '1320157',\n",
       "   '682116',\n",
       "   '405672',\n",
       "   '1814546',\n",
       "   '12732269',\n",
       "   '12394780',\n",
       "   '1835552',\n",
       "   '9445526',\n",
       "   '80543',\n",
       "   '5413529',\n",
       "   '1153421',\n",
       "   '12925513',\n",
       "   '1784935',\n",
       "   '10802130',\n",
       "   '6305290',\n",
       "   '3856997',\n",
       "   '2097471',\n",
       "   '3748163',\n",
       "   '12796779',\n",
       "   '878326',\n",
       "   '10098153',\n",
       "   '3459771',\n",
       "   '827938',\n",
       "   '4448813',\n",
       "   '239',\n",
       "   '9478785',\n",
       "   '579614',\n",
       "   '7355647',\n",
       "   '423698',\n",
       "   '556149',\n",
       "   '13329593',\n",
       "   '9081047',\n",
       "   '151946',\n",
       "   '1453872',\n",
       "   '671699',\n",
       "   '3578102',\n",
       "   '137486',\n",
       "   '703464',\n",
       "   '9069826',\n",
       "   '622635',\n",
       "   '6792520',\n",
       "   '247',\n",
       "   '12716731',\n",
       "   '10045309',\n",
       "   '255030',\n",
       "   '968854',\n",
       "   '3726797',\n",
       "   '8217163',\n",
       "   '2945839',\n",
       "   '11839662',\n",
       "   '10685085',\n",
       "   '2308256',\n",
       "   '9983047',\n",
       "   '10138849',\n",
       "   '11536389',\n",
       "   '945310',\n",
       "   '1036492',\n",
       "   '6982661',\n",
       "   '1649749',\n",
       "   '1325105',\n",
       "   '6475030',\n",
       "   '488726',\n",
       "   '100186',\n",
       "   '3317637',\n",
       "   '336911',\n",
       "   '1321106',\n",
       "   '3956160'])]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# I have measure time for exercise 1.3 which i have put in comments\n",
    "# Top 10 word count took around 46 mintues to complete \n",
    "\n",
    "# Creating userRating RDD from users_libraries.\n",
    "rdd1 = sparkSession.sparkContext.textFile(\"users_libraries.txt\")\n",
    "userRatingsRDD = rdd1.map(lambda x: (x.split(';')[0], x.split(';')[1].split(',')))\n",
    "\n",
    "# Extracting key/value pair from key/values(list) so i can join\n",
    "exploded = userRatingsRDD.flatMap(lambda l: [(l[0], value) for value in l[1]])\n",
    "\n",
    "#Reversing rdd to paper_id/User to perform join on papersTermRdd\n",
    "reversed_ = exploded.map(lambda x: (x[1],x[0]))\n",
    "\n",
    "userRatingsRDD.take(1)\n",
    "#reversed_.take(10)\n",
    "#exploded.take(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('80546',\n",
       "  \"the genetic code has been regarded as arbitrary in the sense that the codon-amino acid assignments could be different than they actually are. this general idea has been spelled out differently by previous, often rather implicit accounts of arbitrariness. they have drawn on the frozen accident theory, on evolutionary contingency, on alternative causal pathways, and on the absence of direct stereochemical interactions between codons and amino acids. it has also been suggested that the arbitrariness of the genetic code justifies attributing semantic information to macromolecules, notably to {dna}. i argue that these accounts of arbitrariness are unsatisfactory. i propose that the code is arbitrary in the sense of jacques monod's concept of chemical arbitrariness: the genetic code is arbitrary in that any codon requires certain chemical and structural properties to specify a particular amino acid, but these properties are not required in virtue of a principle of chemistry. this notion of arbitrariness is compatible with several recent hypotheses about code evolution. i maintain that the code's chemical arbitrariness is neither sufficient nor necessary for attributing semantic information to nucleic acids.\")]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Using csv to parse the papers.CSV file to map paper id and abstract as Key/Value\n",
    "# Also allows to replace null bytes in csv file.\n",
    "# Creating paperTermsRDD\n",
    "rdd2 = sparkSession.sparkContext.textFile(\"papers.csv\")\n",
    "paperTermsRDD = rdd2.map(lambda line: next(csv.reader(x.replace(\"\\x00\", \"\") for x in [line]))) \\\n",
    "                    .map(lambda fields: (fields[0], fields[14]))\n",
    "paperTermsRDD.take(1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exercise 1.3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# BroadCasting StopWords\n",
    "# Timer\n",
    "start = time.time()\n",
    "data = [word.rstrip() for word in open(\"stopwords_en.txt\", 'r').readlines()]\n",
    "broadcastVar = sc.broadcast(data)\n",
    "# broadcastVar.value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Joined UserRatingsRDD(Reversed) with PaperTermsRDD\n",
    "joinedRDD = reversed_.join(paperTermsRDD)\n",
    "\n",
    "# joinedRDD.take(20)\n",
    "\n",
    "user_abs = joinedRDD.map(lambda x: (x[1][0], x[1][1]))\n",
    "\n",
    "# user_abs.take(20)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('8123c9baf50aa777ed3211b5b6306bd0',\n",
       "  \"{meme} is a tool for discovering motifs in sets of protein or {dna} sequences. this paper describes several extensions to {meme} which increase its ability to find motifs in a totally unsupervised fashion, but which also allow it to benefit when prior knowledge is available. when no background knowledge is asserted. {meme} obtains increased robustness from a method for determining motif widths automatically, and from probabilistic models that allow motifs to be absent in some input sequences. on the other hand, {meme} can exploit prior knowledge about a motif being present in all input sequences, about the length of a motif and whether it is a palindrome, and (using dirichlet mixtures) about expected patterns in individual motif positions. extensive experiments are reported which support the claim that {meme} benefits from, but does not require, background knowledge. the experiments use seven previously studied {dna} and protein sequence families and 75 of the protein families documented in the prosite database of sites and patterns, release 11.1.systematic mapping of protein–protein interactions, or 'interactome' mapping, was initiated in model organisms, starting with defined biological processes1, 2 and then expanding to the scale of the proteome3, 4, 5, 6, 7. although far from complete, such maps have revealed global topological and dynamic features of interactome networks that relate to known biological properties8, 9, suggesting that a human interactome map will provide insight into development and disease mechanisms at a systems level. here we describe an initial version of a proteome-scale map of human binary protein–protein interactions. using a stringent, high-throughput yeast two-hybrid system, we tested pairwise interactions among the products of 8,100 currently available gateway-cloned open reading frames and detected 2,800 interactions. this data set, called {ccsb}-{hi1}, has a verification rate of 78\\\\% as revealed by an independent co-affinity purification assay, and correlates significantly with other biological attributes. the {ccsb}-{hi1} data set increases by 70\\\\% the set of available binary interactions within the tested space and reveals more than 300 new connections to over 100 disease-associated proteins. this work represents an important step towards a systematic and comprehensive human interactome project.oligonucleotide and complementary {dna} microarrays are being used to subclassify histologically similar tumours, monitor disease progress, and individualize treatment regimens1, 2, 3, 4, 5. however, extracting new biological insight from high-throughput genomic studies of human diseases is a challenge, limited by difficulties in recognizing and evaluating relevant biological processes from huge quantities of experimental data. here we present a structured network knowledge-base approach to analyse genome-wide transcriptional responses in the context of known functional interrelationships among proteins, small molecules and phenotypes. this approach was used to analyse changes in blood leukocyte gene expression patterns in human subjects receiving an inflammatory stimulus (bacterial endotoxin). we explore the known genome-wide interaction network to identify significant functional modules perturbed in response to this stimulus. our analysis reveals that the human blood leukocyte response to acute systemic inflammation includes the transient dysregulation of leukocyte bioenergetics and modulation of translational machinery. these findings provide insight into the regulation of global leukocyte activities as they relate to innate immune system tolerance and increased susceptibility to infection in humans.protein complexes are key molecular entities that integrate multiple gene products to perform cellular functions. here we report the first genome-wide screen for complexes in an organism, budding yeast, using affinity purification and mass spectrometry. through systematic tagging of open reading frames ({orfs}), the majority of complexes were purified several times, suggesting screen saturation. the richness of the data set enabled a de novo characterization of the composition and organization of the cellular machinery. the ensemble of cellular proteins partitions into 491 complexes, of which 257 are novel, that differentially combine with additional attachment proteins or protein modules to enable a diversification of potential functions. support for this modular organization of the proteome comes from integration with available data on expression, localization, function, evolutionary conservation, protein structure and binary interactions. this study provides the largest collection of physically determined eukaryotic cellular machines so far and a platform for biological data integration and modelling.the role of small {rnas} as key regulators of {mrna} turnover and translation has been well established. recent advances indicate that the small {rnas} termed {micrornas} play important roles in animal development and physiology. cellular activities such as proliferation, morphogenesis, apoptosis and differentiation are regulated by {micrornas}. the expression of various genes are regulated by {micrornas}, and several {micrornas} act in reciprocal negative feedback loops with protein factors to control cell fate decisions that are triggered by signal transduction activity. these observations implicate small {rnas} as important mediators of gene regulation in response to cell-cell signaling. the mechanism by which {micrornas} silence gene expression is post-transcriptional, possibly influencing the stability, compartmentalization and translation of {mrnas}. this mechanism is an efficient means to regulate production of a diverse range of proteins.we describe a systematic genome-wide approach for learning the complex combinatorial code underlying gene expression. our probabilistic approach identifies local {dna}-sequence elements and the positional and combinatorial constraints that determine their context-dependent role in transcriptional regulation. the inferred regulatory rules correctly predict expression patterns for 73\\\\% of genes in saccharomyces cerevisiae, utilizing microarray expression data and sequences in the 800 bp upstream of genes. application to caenorhabditis elegans identifies predictive regulatory elements and combinatorial rules that control the phased temporal expression of transcription factors, histones, and germline specific genes. successful prediction requires diverse and complex rules utilizing {and}, {or}, and {not} logic, with significant constraints on motif strength, orientation, and relative position. this system generates a large number of mechanistic hypotheses for focused experimental validation, and establishes a predictive dynamical framework for understanding cellular behavior from genomic sequence.the exploration of quantitative variation in human populations has become one of the major priorities for medical genetics. the successful identification of variants that contribute to complex traits is highly dependent on reliable assays and genetic maps. we have performed a genome-wide quantitative trait analysis of 630 genes in 60 unrelated utah residents with ancestry from northern and western europe using the publicly available phase i data of the international {hapmap} project. the genes are located in regions of the human genome with elevated functional annotation and disease interest including the {encode} regions spanning 1\\\\% of the genome, chromosome 21 and chromosome 20q12–13.2. we apply three different methods of multiple test correction, including bonferroni, false discovery rate, and permutations. for the 374 expressed genes, we find many regions with statistically significant association of single nucleotide polymorphisms ({snps}) with expression variation in lymphoblastoid cell lines after correcting for multiple tests. based on our analyses, the signal proximal (cis-) to the genes of interest is more abundant and more stable than distal and trans across statistical methodologies. our results suggest that regulatory polymorphism is widespread in the human genome and show that the 5-kb (phase i) {hapmap} has sufficient density to enable linkage disequilibrium mapping in humans. such studies will significantly enhance our ability to annotate the non-coding part of the genome and interpret functional variation. in addition, we demonstrate that the {hapmap} cell lines themselves may serve as a useful resource for quantitative measurements at the cellular level. with the finished reference sequence of the human genome now available, focus has shifted towards trying to identify all of the functional elements within the sequence. although quite a lot of progress has been made towards identifying some classes of genomic elements, in particular protein-coding sequences, the characterization of regulatory elements remains a challenge. the authors describe the genetic mapping of regions of the genome that have functional effects on quantitative levels of gene expression. gene expression of 630 genes was measured in cell lines derived from 60 unrelated human individuals, the same utah residents of northern and western european ancestry that have been genetically well-characterized by the international {hapmap} project. this paper reports significant variation among individuals with respect to levels of gene expression, and demonstrates that this quantitative trait has a genetic basis. for some genes, the genetic signal was localized to specific locations in the human genome sequence; in most cases the genomic region associated with expression variation was physically close to the gene whose expression it regulated. the authors demonstrate the feasibility of performing whole-genome association scans to map quantitative traits, and highlight statistical issues that are increasingly important for whole-genome disease mapping studies.the synthesis of gene expression data and cis -regulatory analysis permits the elucidation of genomic regulatory networks. these networks provide a direct visualization of the functional interconnections among the regulatory genes and signaling components leading to cell-specific patterns of gene activity. complex developmental processes are thereby illuminated in ways not revealed by the conventional analysis of individual genes. in this review, we describe emerging networks in several different model systems, and compare them with the gene regulatory network that controls dorsoventral patterning of the drosophila embryo.we propose a model-driven approach for analyzing genomic expression data that permits genetic regulatory networks to be represented in a biologically interpretable computational form. our models permit latent variables capturing unobserved factors, describe arbitrarily complex (more than pair-wise) relationships at varying levels of refinement, and can be scored rigorously against observational data. the models that we use are based on bayesian networks and their extensions. as a demonstration of this approach, we utilize 52 genomes worth of affymetrix {genechip} expression data to correctly differentiate between alternative hypotheses of the galactose regulatory network in s. cerevisiae. when we extend the graph semantics to permit annotated edges, we are able to score models describing relationships at a finer degree of specification.we develop a new framework for inferring models of transcriptional regulation. the models, which we call physical network models, are annotated molecular interaction graphs. the attributes in the model correspond to verifiable properties of the underlying biological system such as the existence of protein-protein and {protein-dna} interactions, the directionality of signal transduction in protein-protein interactions, as well as signs of the immediate effects of these interactions. possible configurations of these variables are constrained by the available data sources. some of the data sources, such as factor-binding data, involve measurements that are directly tied to the variables in the model. other sources, such as gene knock-outs, are functional in nature and provide only indirect evidence about the variables. we associate each observed knock-out effect in the deletion mutant data with a set of causal paths (molecular cascades) that could in principle explain the effect, resulting in aggregate constraints about the physical variables in the model. the most likely settings of all the variables, specifying the most likely graph annotations, are found by a recursive application of the max-product algorithm. by testing our approach on datasets related to the pheromone response pathway in s. cerevisiae, we demonstrate that the resulting model is consistent with previous studies about the pathway. moreover, we successfully predict gene knock-out effects with a high degree of accuracy in a cross-validation setting. when applying this approach genome-wide, we extract submodels consistent with previous studies. the approach can be readily extended to other data sources or to facilitate automated experimental design.as genome-scale measurements lead to increasingly complex models of gene regulation, systematic approaches are needed to validate and refine these models. towards this goal, we describe an automated procedure for prioritizing genetic perturbations in order to discriminate optimally between alternative models of a gene-regulatory network. using this procedure, we evaluate 38 candidate regulatory networks in yeast and perform four high-priority gene knockout experiments. the refined networks support previously unknown regulatory mechanisms downstream of {sok2} and {swi4}.{background}: the study of complex biological networks and prediction of gene function has been enabled by high-throughput ({htp}) methods for detection of genetic and protein interactions. sparse coverage in {htp} datasets may, however, distort network properties and confound predictions. although a vast number of well substantiated interactions are recorded in the scientific literature, these data have not yet been distilled into networks that enable system-level inference. {results}: we describe here a comprehensive database of genetic and protein interactions, and associated experimental evidence, for the budding yeast saccharomyces cerevisiae, as manually curated from over 31,793 abstracts and online publications. this literature-curated ({lc}) dataset contains 33,311 interactions, on the order of all extant {htp} datasets combined. surprisingly, {htp} protein-interaction datasets currently achieve only around 14\\\\% coverage of the interactions in the literature. the {lc} network nevertheless shares attributes with {htp} networks, including scale-free connectivity and correlations between interactions, abundance, localization, and expression. we find that essential genes or proteins are enriched for interactions with other essential genes or proteins, suggesting that the global network may be functionally unified. this interconnectivity is supported by a substantial overlap of protein and genetic interactions in the {lc} dataset. we show that the {lc} dataset considerably improves the predictive power of network-analysis approaches. the full {lc} dataset is available at the {biogrid} (http://www.thebiogrid.org) and {sgd} (http://www.yeastgenome.org/) databases. {conclusion}: comprehensive datasets of biological interactions derived from the primary literature provide critical benchmarks for {htp} methods, augment functional prediction, and reveal system-level attributes of biological networks.in this paper, we describe an approach for identifying' pathways' from gene expression and protein interaction data. our approach is based on the assumption that many pathways exhibit two properties: their genes exhibit a similar gene expression profile, and the protein products of the genes often interact.  our approach is based on a unified probabilistic model, which is learned from the data using the {em} algorithm. we present results on two saccharomyces cerevisiae gene expression data sets, combined with a binary protein interaction data set. our results show that our approach is much more successful than other approaches at discovering both coherent functional groups and entire protein {complexes.contact}: {eran@cs.stanford.edukeywords}: probabilistic models, protein interaction, gene expression.we propose a statistical method to estimate gene networks from {dna} microarray data and protein-protein interactions. because physical interactions between proteins or multiprotein complexes are likely to regulate biological processes, using only {mrna} expression data is not sufficient for estimating a gene network accurately. our method adds knowledge about protein-protein interactions to the estimation method of gene networks under a bayesian statistical framework. in the estimated gene network, a protein complex is modeled as a virtual node based on principal component analysis. we show the effectiveness of the proposed method through the analysis of saccharomyces cerevisiae cell cycle data. the proposed method improves the accuracy of the estimated gene networks, and successfully identifies some biological facts.the prediction of regulatory elements is a problem where computational methods offer great hope. over the past few years, numerous tools have become available for this task. the purpose of the current assessment is twofold: to provide some guidance to users regarding the accuracy of currently available tools in various settings, and to provide a benchmark of data sets for assessing future tools.we develop principled methods for the automatic induction (discovery) of genetic regulatory network models from multiple data sources and data modalities. models of regulatory networks are represented as bayesian networks, allowing the models to compactly and robustly capture probabilistic multivariate statistical dependencies between the various cellular factors in these networks. we build on previous bayesian network validation results by extending the validation framework to the context of model induction, leveraging heuristic simulated annealing search algorithms and posterior model averaging. using expression data in isolation yields results inconsistent with location data so we incorporate genomic location data to guide the model induction process. we combine these two data modalities by allowing location data to influence the model prior and expression data to influence the model likelihood. we demonstrate the utility of this approach by discovering genetic regulatory models of thirty-three variables involved in s. cerevisiae pheromone response. the models we automatically generate are consistent with the current understanding regarding this regulatory network, but also suggest new directions for future experimental investigation.ciona is an emerging model system for elucidating gene networks in development. comprehensive in situ hybridization assays have identified 76 regulatory genes with localized expression patterns in the early embryo, at the time when naive blastomeres are determined to follow specific cell fates. systematic gene disruption assays provided more than 3000 combinations of gene expression profiles in mutant backgrounds. deduced gene circuit diagrams describing the formation of larval tissues were computationally visualized. these diagrams constitute a blueprint for the ciona embryo and provide a foundation for understanding the evolutionary origins of the chordate body plan. 10.1126/science.1123404genome-wide transcript profiling was used to monitor signal transduction during yeast pheromone response. genetic manipulations allowed analysis of changes in gene expression underlying pheromone signaling, cell cycle control, and polarized morphogenesis. a two-dimensional hierarchical clustered matrix, covering 383 of the most highly regulated genes, was constructed from 46 diverse experimental conditions. diagnostic subsets of coexpressed genes reflected signaling activity, cross talk, and overlap of multiple mitogen-activated protein kinase ({mapk}) pathways. analysis of the profiles specified by two different {mapks}-fus3p and kss1p-revealed functional overlap of the filamentous growth and mating responses. global transcript analysis reflects biological responses associated with the activation and perturbation of signal transduction pathways.we present a probabilistic framework that models the process by which transcriptional binding explains the {mrna} expression of different genes. our joint probabilistic model unifies the two key components of this process: the prediction of gene regulation events from sequence motifs in the gene's promoter region, and the prediction of {mrna} expression from combinations of gene regulation events in different settings. our approach has several advantages. by learning promoter sequence motifs that...we propose a probabilistic model for cellular processes, and an algorithm for discovering them from gene expression data. a process is associated with a set of genes that participate in it; unlike clustering techniques, our model allows genes to participate in multiple processes. each process may be active to a different degree in each experiment. the expression measurement for gene g in array a is a sum, over all processes in which g participates, of the activity levels of these processes in array a. we describe an iterative procedure, based on the {em} algorithm, for decomposing the expression matrix into a given number of processes. we present results on yeast gene expression data, which indicate that our approach identifies real biological processes.motivation: in model organisms such as yeast, large databases of protein–protein and {protein-dna} interactions have become an extremely important resource for the study of protein function, evolution, and gene regulatory dynamics. in this paper we demonstrate that by integrating these interactions with widely-available {mrna} expression data, it is possible to generate concrete hypotheses for the underlying mechanisms governing the observed changes in gene expression. to perform this integration systematically and at large scale, we introduce an approach for screening a molecular interaction network to identify active subnetworks, i.e., connected regions of the network that show significant changes in expression over particular subsets of conditions. the method we present here combines a rigorous statistical measure for scoring subnetworks with a search algorithm for identifying subnetworks with high score.the modeling of cellular signaling pathways is an emerging field. sachs et al. illustrate the application of bayesian networks to an example cellular pathway involving the activation of focal adhesion kinase ({fak}) and extracellular signal-regulated kinase ({erk}) in response to fibronectin binding to an integrin. they describe how to use the analysis to select from among proposed models, formulate hypotheses regarding component interactions, and uncover potential dynamic changes in the interactions between these components. although the data sets currently available for this example problem are too small to definitively point to a particular model, the approach and results provide a glimpse into the power that these methods will achieve once the technology for obtaining the necessary data becomes readily available.{motivation}: biological processes in cells are properly performed by gene regulations, signal transductions and interactions between proteins. to understand such molecular networks, we propose a statistical method to estimate gene regulatory networks and protein-protein interaction networks simultaneously from {dna} microarray data, protein-protein interaction data and other genome-wide data. {results}: we unify bayesian networks and markov networks for estimating gene regulatory networks and protein-protein interaction networks according to the reliability of each biological information source. through the simultaneous construction of gene regulatory networks and protein-protein interaction networks of saccharomyces cerevisiae cell cycle, we predict the role of several genes whose functions are currently unknown. by using our probabilistic model, we can detect false positives of high-throughput data, such as yeast two-hybrid data. in a genome-wide experiment, we find possible gene regulatory relationships and protein-protein interactions between large protein complexes that underlie complex regulatory mechanisms of biological processes. {contact}: nariai@ims.u-tokyo.ac.jp.genetic loci that regulate inherited traits are routinely identified using quantitative trait locus ({qtl}) mapping methods. however, the genotype-phenotype associations do not provide information on the gene expression program through which the genetic loci regulate the traits. transcription modules are 'self-consistent regulatory units' and are closely related to the modular components of gene regulatory network [ihmels, j., friedlander, g., bergmann, s., sarig, o., ziv, y. and barkai, n. (2002) revealing modular organization in the yeast transcriptional network. nat. genet., 31, 370-377; segal, e., shapira, m., regev, a., pe'er, d., botstein, d., koller, d. and friedman, n. (2003) module networks: identifying regulatory modules and their condition-specific regulators from gene expression data. nat. genet., 34, 166-176]. we used genome-wide genotype and gene expression data of a genetic reference population that consists of mice of 32 recombinant inbred strains to identify the transcription modules and the genetic loci regulating them. twenty-nine transcription modules defined by genetic variations were identified. statistically significant associations between the transcription modules and 18 classical physiological and behavioral traits were found. genome-wide interval mapping showed that major {qtls} regulating the transcription modules are often co-localized with the {qtls} regulating the associated classical traits. the association and the possible co-regulation of the classical trait and transcription module indicate that the transcription module may be involved in the gene pathways connecting the {qtl} and the classical trait. our results show that a transcription module may associate with multiple seemingly unrelated classical traits and a classical trait may associate with different modules. literature mining results provided strong independent evidences for the relations among genes of the transcription modules, genes in the regions of the {qtls} regulating the transcription modules and the keywords representing the classical traits.a central problem in the bioinformatics of gene regulation is to find the binding sites for regulatory proteins. one of the most promising approaches toward identifying these short and fuzzy sequence patterns is the comparative analysis of orthologous intergenic regions of related species. this analysis is complicated by various factors. first, one needs to take the phylogenetic relationship between the species into account in order to distinguish conservation that is due to the occurrence of functional sites from spurious conservation that is due to evolutionary proximity. second, one has to deal with the complexities of multiple alignments of orthologous intergenic regions, and one has to consider the possibility that functional sites may occur outside of conserved segments. here we present a new motif sampling algorithm, {phylogibbs}, that runs on arbitrary collections of multiple local sequence alignments of orthologous sequences. the algorithm searches over all ways in which an arbitrary number of binding sites for an arbitrary number of transcription factors ({tfs}) can be assigned to the multiple sequence alignments. these binding site configurations are scored by a bayesian probabilistic model that treats aligned sequences by a model for the evolution of binding sites and  ” background” intergenic {dna}. this model takes the phylogenetic relationship between the species in the alignment explicitly into account. the algorithm uses simulated annealing and monte carlo markov-chain sampling to rigorously assign posterior probabilities to all the binding sites that it reports. in tests on synthetic data and real data from five saccharomyces species our algorithm performs significantly better than four other motif-finding algorithms, including algorithms that also take phylogeny into account. our results also show that, in contrast to the other algorithms, {phylogibbs} can make realistic estimates of the reliability of its predictions. our tests suggest that, running on the five-species multiple alignment of a single gene's upstream region, {phylogibbs} on average recovers over 50\\\\% of all binding sites in s. cerevisiae at a specificity of about 50\\\\%, and 33\\\\% of all binding sites at a specificity of about 85\\\\%. we also tested {phylogibbs} on collections of multiple alignments of intergenic regions that were recently annotated, based on {chip}-on-chip data, to contain binding sites for the same {tf}. we compared {phylogibbs}'s results with the previous analysis of these data using six other motif-finding algorithms. for 16 of 21 {tfs} for which all other motif-finding methods failed to find a significant motif, {phylogibbs} did recover a motif that matches the literature consensus. in 11 cases where there was disagreement in the results we compiled lists of known target genes from the literature, and found that running {phylogibbs} on their regulatory regions yielded a binding motif matching the literature consensus in all but one of the cases. interestingly, these literature gene lists had little overlap with the targets annotated based on the {chip}-on-chip data. the {phylogibbs} code can be downloaded from http://www.biozentrum.unibas.ch/\\\\~{}nimwege\\u200bn/cgi-bin/phylogibbs.cgi or http://www.imsc.res.in/\\\\~{}rsidd/phylogibbs. the full set of predicted sites from our tests on yeast are available at http://www.swissregulon.unibas.ch. computational discovery of regulatory sites in intergenic {dna} is one of the central problems in bioinformatics. up until recently motif finders would typically take one of the following two general approaches. given a known set of co-regulated genes, one searches their promoter regions for significantly overrepresented sequence motifs. alternatively, in a  ” phylogenetic footprinting” approach one searches multiple alignments of orthologous intergenic regions for short segments that are significantly more conserved than expected based on the phylogeny of the species. in this work the authors present an algorithm, {phylogibbs}, that combines these two approaches into one integrated bayesian framework. the algorithm searches over all ways in which an arbitrary number of binding sites for an arbitrary number of transcription factors can be assigned to arbitrary collections of multiple sequence alignments while taking into account the phylogenetic relations between the sequences. the authors perform a number of tests on synthetic data and real data from saccharomyces genomes in which {phylogibbs} significantly outperforms other existing methods. finally, a novel anneal-and-track strategy allows {phylogibbs} to make accurate estimates of the reliability of its predictions.machine learning was applied for the automated derivation of causal influences in cellular signaling networks. this derivation relied on the simultaneous measurement of multiple phosphorylated protein and phospholipid components in thousands of individual primary human immune system cells. perturbing these cells with molecular interventions drove the ordering of connections between pathway components, wherein bayesian network computational methods automatically elucidated most of the traditionally reported signaling relationships and predicted novel interpathway network causalities, which we verified experimentally. reconstruction of network models from physiologically relevant primary single cells might be applied to understanding native-state tissue signaling biology, complex drug actions, and dysfunctional signaling in diseased cells.{motivation}: {micrornas} ({mirnas}) are small endogenous {rnas} that can play important regulatory roles via the {rna}-interference pathway by targeting {mrnas} for cleavage or translational repression. we propose a computational method to predict {mirna} regulatory modules ({mrms}) or groups of {mirnas} and target genes that are believed to participate cooperatively in post-transcriptional gene regulation. {results}: we tested our method with the human genes and {mirnas}, predicting 431 {mrms}. we analyze a module with genes: {btg2}, {wt1}, {ppm1d}, {pak7} and {rab9b}, and {mirnas}: {mir}-15a and {mir}-16. review of the literature and annotation with gene ontology terms reveal that the roles of these genes can indeed be closely related in specific biological processes, such as gene regulation involved in breast, renal and prostate cancers. furthermore, it has been reported that {mir}-15a and {mir}-16 are deleted together in certain types of cancer, suggesting a possible connection between these {mirnas} and cancers. given that most known functionalities of {mirnas} are related to negative gene regulation, extending our approach and exploiting the insight thus obtained may provide clues to achieving practical accuracy in the reverse-engineering of gene regulatory networks. {availability}: a list of predicted modules is available from the authors upon request.thousands of mammalian messenger {rnas} are under selective pressure to maintain 7-nucleotide sites matching {micrornas} ({mirnas}). we found that these conserved targets are often highly expressed at developmental stages before {mirna} expression and that their levels tend to fall as the {mirna} that targets them begins to accumulate. nonconserved sites, which outnumber the conserved sites 10 to 1, also mediate repression. as a consequence, genes preferentially expressed at the same time and place as a {mirna} have evolved to selectively avoid sites matching the {mirna}. this phenomenon of selective avoidance extends to thousands of genes and enables spatial and temporal specificities of {mirnas} to be revealed by finding tissues and developmental stages in which messages with corresponding sites are expressed at lower levels.gene expression variation has been the focus of many studies in the past few years. the relevance of gene regulation and gene expression to disease and the development of the technologies used to screen large numbers of genes simultaneously have allowed this rapid development. in this review we discuss issues relating to the biological information one obtains from such studies and the biological significance and use of signals from mapping of gene expression variation.\"),\n",
       " ('1507e0abf427bfc9bc3d017d63644092',\n",
       "  'in the hierarchy of data, information and knowledge, computational methods play a major role in the initial processing of data to extract information, but they alone become less effective to compile knowledge from information. the kyoto encyclopedia of genes and genomes ({kegg}) resource (http://www.kegg.jp/ or http://www.genome.jp/kegg/) has been developed as a reference knowledge base to assist this latter process. in particular, the {kegg} pathway maps are widely used for biological interpretation of genome sequences and other high-throughput data. the link from genomes to pathways is made through the {kegg} orthology system, a collection of manually defined ortholog groups identified by k numbers. to better automate this interpretation process the {kegg} modules defined by boolean expressions of k numbers have been expanded and improved. once genes in a genome are annotated with k numbers, the {kegg} modules can be computationally evaluated revealing metabolic capacities and other phenotypic features. the reaction modules, which represent chemical units of reactions, have been used to analyze design principles of metabolic networks and also to improve the definition of k numbers and associated annotations. for translational bioinformatics, the {kegg} {medicus} resource has been developed by integrating drug labels (package inserts) used in society.many areas of science depend on exploratory data analysis and visualization. the need to analyze large amounts of multivariate data raises the fundamental problem of dimensionality reduction: how to discover compact representations of high-dimensional data. here, we introduce locally linear embedding ({lle}), an unsupervised learning algorithm that computes low-dimensional, neighborhood-preserving embeddings of high-dimensional inputs. unlike clustering methods for local dimensionality reduction, {lle} maps its inputs into a single global coordinate system of lower dimensionality, and its optimizations do not involve local minima. by exploiting the local symmetries of linear reconstructions, {lle} is able to learn the global structure of nonlinear manifolds, such as those generated by images of faces or documents of text.characterizing the genome-wide dynamic regulation of gene expression is important and will be of much interest in the future. however, there is currently no established method for identifying differentially expressed genes in a time course study. here we propose a significance method for analyzing time course microarray studies that can be applied to the typical types of comparisons and sampling schemes. this method is applied to two studies on humans. in one study, genes are identified that show differential expression over time in response to in vivo endotoxin administration. by using our method, 7,409 genes are called significant at a 1\\\\% false-discovery rate level, whereas several existing approaches fail to identify any genes. in another study, 417 genes are identified at a 10\\\\% false-discovery rate level that show expression changing with age in the kidney cortex. here it is also shown that as many as 47\\\\% of the genes change with age in a manner more complex than simple exponential growth or decay. the methodology proposed here has been implemented in the freely distributed and open-source edge software package.the genomes of individuals from the same species vary in sequence as a result of different evolutionary processes. to examine the patterns of, and the forces shaping, sequence variation in arabidopsis thaliana, we performed high-density array resequencing of 20 diverse strains (accessions). more than 1 million nonredundant single-nucleotide polymorphisms ({snps}) were identified at moderate false discovery rates ({fdrs}), and ∼4\\\\% of the genome was identified as being highly dissimilar or deleted relative to the reference genome sequence. patterns of polymorphism are highly nonrandom among gene families, with genes mediating interaction with the biotic environment having exceptional polymorphism levels. at the chromosomal scale, regional variation in polymorphism was readily apparent. a scan for recent selective sweeps revealed several candidate regions, including a notable example in which almost all variation was removed in a 500-kilobase window. analyzing the polymorphisms we describe in larger sets of accessions will enable a detailed understanding of forces shaping population-wide sequence variation in a. thaliana.the homeostatic framework has dominated our understanding of cellular physiology. we question whether homeostasis alone adequately explains microbial responses to environmental stimuli, and explore the capacity of intracellular networks for predictive behavior in a fashion similar to metazoan nervous systems. we show that in silico biochemical networks, evolving randomly under precisely defined complex habitats, capture the dynamical, multidimensional structure of diverse environments by forming internal representations that allow prediction of environmental change. we provide evidence for such anticipatory behavior by revealing striking correlations of escherichia coli transcriptional responses to temperature and oxygen perturbations--precisely mirroring the covariation of these parameters upon transitions between the outside world and the mammalian gastrointestinal tract. we further show that these internal correlations reflect a true associative learning paradigm, because they show rapid decoupling upon exposure to novel environments.the fine detail provided by sequencing-based transcriptome surveys suggests that {rna}-seq is likely to become the platform of choice for interrogating steady state {rna}. in order to discover biologically important changes in expression, we show that normalization continues to be an essential step in the analysis. we outline a simple and effective method for performing normalization and show dramatically improved results for inferring differential expression in simulated and publicly available data sets.motivation: understanding the association between genetic diseases and their causal genes is an important problem concerning human health. with the recent influx of high-throughput data describing interactions between gene products, scientists have been provided a new avenue through which these associations can be inferred. despite the recent interest in this problem, however, there is little understanding of the relative benefits and drawbacks underlying the proposed techniques.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsmany bacterial systems rely on dynamic genetic circuits to control crucial biological processes. a major goal of systems biology is to understand these behaviours in terms of individual genes and their interactions. however, traditional techniques based on population averages \\'wash out\\' crucial dynamics that are either unsynchronized between cells or are driven by fluctuations, or \\'noise\\', in cellular components. recently, the combination of time-lapse microscopy, quantitative image analysis and fluorescent protein reporters has enabled direct observation of multiple cellular components over time in individual cells. in conjunction with mathematical modelling, these techniques are now providing powerful insights into genetic circuit behaviour in diverse microbial systems.{rna} polymerases are highly regulated molecular machines. we present a method (global run-on sequencing, {gro}-seq) that maps the position, amount, and orientation of transcriptionally engaged {rna} polymerases genome-wide. in this method, nuclear run-on {rna} molecules are subjected to large-scale parallel sequencing and mapped to the genome. we show that peaks of promoter-proximal polymerase reside on approximately 30\\\\% of human genes, transcription extends beyond pre-messenger {rna} 3\\' cleavage, and antisense transcription is prevalent. additionally, most promoters have an engaged polymerase upstream and in an orientation opposite to the annotated gene. this divergent polymerase is associated with active genes but does not elongate effectively beyond the promoter. these results imply that the interplay between polymerases and regulators over broad promoter regions dictates the orientation and efficiency of productive transcription.high-throughput {rna} sequencing ({rna}-seq) promises a comprehensive picture of the transcriptome, allowing for the complete annotation and quantification of all genes and their isoforms across samples. realizing this promise requires increasingly complex computational methods. these computational challenges fall into three main categories: (i) read mapping, (ii) transcriptome reconstruction and (iii) expression quantification. here we explain the major conceptual and practical challenges, and the general classes of solutions for each category. finally, we highlight the interdependence between these categories and discuss the benefits for different biological applications.summary.\\u2002 we consider the problem of selecting grouped variables (factors) for accurate prediction in regression. such a problem arises naturally in many practical situations with the multifactor analysis-of-variance problem as the most important and well-known example. instead of selecting factors by stepwise backward elimination, we focus on the accuracy of estimation and consider extensions of the lasso, the {lars} algorithm and the non-negative garrotte for factor selection. the lasso, the {lars} algorithm and the non-negative garrotte are recently proposed regression methods that can be used to select individual variables. we study and propose efficient algorithms for the extensions of these methods for factor selection and show that these extensions give superior performance to the traditional stepwise backward elimination method in factor selection problems. we study the similarities and the differences between these methods. simulations and real examples are used to illustrate the methods.a system of cluster analysis for genome-wide expression data from {dna} microarray hybridization is described that uses standard statistical algorithms to arrange genes according to similarity in pattern of gene expression. the output is displayed graphically, conveying the clustering and the underlying expression data simultaneously in a form intuitive for biologists. we have found in the budding yeast saccharomyces cerevisiae that clustering gene expression data groups together efficiently genes of known similar function, and we find a similar tendency in human data. thus patterns seen in genome-wide expression experiments can be interpreted as indications of the status of cellular processes. also, coexpression of genes of known function with poorly characterized or novel genes may provide a simple means of gaining leads to the functions of many genes for which information is not available currently.th17 cells have critical roles in mucosal defense and are major contributors to inflammatory disease. their differentiation requires the nuclear hormone receptor {rorγt} working with multiple other essential transcription factors ({tfs}). we have used an iterative systems approach, combining genome-wide {tf} occupancy, expression profiling of {tf} mutants, and expression time series to delineate the th17 global transcriptional regulatory network. we find that cooperatively bound {batf} and {irf4} contribute to initial chromatin accessibility and, with {stat3}, initiate a transcriptional program that is then globally tuned by the lineage-specifying {tf} {rorγt}, which plays a focal deterministic role at key loci. integration of multiple data sets allowed inference of an accurate predictive model that we computationally and experimentally validated, identifying multiple new th17 regulators, including fosl2, a key determinant of cellular plasticity. this interconnected network can be used to investigate new therapeutic approaches to manipulate th17 functions in the setting of inflammatory disease. copyright {\\\\copyright} 2012 elsevier inc. all rights reserved.attempts to correlate protein abundance with {mrna} expression levels have had variable success. we review the results of these comparisons, focusing on yeast. in the process, we survey experimental techniques for determining protein abundance, principally two-dimensional gel electrophoresis and mass-spectrometry. we also merge many of the available yeast protein-abundance datasets, using the resulting larger \\'meta-dataset\\' to find correlations between protein and {mrna} expression, both globally and within smaller categories.comparative developmental evidence indicates that reorganizations in developmental gene regulatory networks ({grns}) underlie evolutionary changes in animal morphology, including body plans. we argue here that the nature of the evolutionary alterations that arise from regulatory changes depends on the hierarchical position of the change within a {grn}. this concept cannot be accomodated by microevolutionary nor macroevolutionary theory. it will soon be possible to investigate these ideas experimentally, by assessing the effects of {grn} changes on morphological evolution.machine learning approaches offer the potential to systematically identify transcriptional regulatory interactions from a compendium of microarray expression profiles. however, experimental validation of the performance of these methods at the genome scale has remained elusive. here we assess the global performance of four existing classes of inference algorithms using 445 escherichia coli affymetrix arrays and 3,216 known e. coli regulatory interactions from {regulondb}. we also developed and applied the context likelihood of relatedness ({clr}) algorithm, a novel extension of the relevance networks class of algorithms. {clr} demonstrates an average precision gain of 36\\\\% relative to the next-best performing algorithm. at a 60\\\\% true positive rate, {clr} identifies 1,079 regulatory interactions, of which 338 were in the previously known network and 741 were novel predictions. we tested the predicted interactions for three transcription factors with chromatin immunoprecipitation, confirming 21 novel interactions and verifying our {regulondb}-based performance estimates. {clr} also identified a regulatory link providing central metabolic control of iron transport, which we confirmed with real-time quantitative {pcr}. the compendium of expression data compiled in this study, coupled with {regulondb}, provides a valuable model system for further improvement of network inference algorithms using experimental data.{dna} hybridization arrays simultaneously measure the expression level for thousands of genes. these measurements provide a \"snapshot\" of transcription levels within the cell. a major challenge in computational biology is to uncover, from such measurements, gene/protein interactions and key biological features of cellular systems. in this paper, we propose a new framework for discovering interactions between genes based on multiple expression measurements. this framework builds on the use of bayesian networks for representing statistical dependencies. a bayesian network is a graph-based model of joint multivariate probability distributions that captures properties of conditional independence between variables. such models are attractive for their ability to describe complex stochastic processes and because they provide a clear methodology for learning from (noisy) observations. we start by showing how bayesian networks can describe interactions between genes. we then describe a method for recovering gene interactions from microarray data using tools for learning bayesian networks. finally, we demonstrate this method on the s. cerevisiae cell-cycle measurements of spellman et al. (1998).just as reference genome sequences revolutionized human genetics, reference maps of interactome networks will be critical to fully understand genotype-phenotype relationships. here, we describe a systematic map of ?14,000 high-quality human binary protein-protein interactions. at equal quality, this map is ?30\\\\% larger than what is available from small-scale studies published in the literature in the last few decades. while currently available information is highly biased and only covers a relatively small portion of the proteome, our systematic map appears strikingly more homogeneous, revealing a ?broader? human interactome network than currently appreciated. the map also uncovers significant interconnectivity between known and candidate cancer gene products, providing unbiased evidence for an expanded functional cancer landscape, while demonstrating how high-quality interactome models will help ?connect the dots? of the genomic revolution.the merging of network theory and microarray data analysis techniques has spawned a new field: gene coexpression network analysis. while network methods are increasingly used in biology, the network vocabulary of computational biologists tends to be far more limited than that of, say, social network theorists. here we review and propose several potentially useful network concepts. we take advantage of the relationship between network theory and the field of microarray data analysis to clarify the meaning of and the relationship among network concepts in gene coexpression networks. network theory offers a wealth of intuitive concepts for describing the pairwise relationships among genes, which are depicted in cluster trees and heat maps. conversely, microarray data analysis techniques (singular value decomposition, tests of differential expression) can also be used to address difficult problems in network theory. we describe conditions when a close relationship exists between network analysis and microarray data analysis techniques, and provide a rough dictionary for translating between the two fields. using the angular interpretation of correlations, we provide a geometric interpretation of network theoretic concepts and derive unexpected relationships among them. we use the singular value decomposition of module expression data to characterize approximately factorizable gene coexpression networks, i.e., adjacency matrices that factor into node specific contributions. high and low level views of coexpression networks allow us to study the relationships among modules and among module genes, respectively. we characterize coexpression networks where hub genes are significant with respect to a microarray sample trait and show that the network concept of intramodular connectivity can be interpreted as a fuzzy measure of module membership. we illustrate our results using human, mouse, and yeast microarray gene expression data. the unification of coexpression network methods with traditional data mining methods can inform the application and development of systems biologic methods. similar to natural languages, network language is ever evolving. while some network terms (concepts) are widely used in gene coexpression network analysis, others still need to be developed to meet the ever increasing demand for describing the system of gene transcripts. there is a need to provide an intuitive geometric explanation of network concepts and to study their relationships. for example, we show that certain seemingly disparate network concepts turn out to be synonyms in the context of coexpression modules. we show how coexpression network language affects our understanding of biology. for example, there are geometric reasons why highly connected hub genes in important coexpression modules tend to be important, and why hub genes in one module cannot be hubs in another distinct module. we provide a short dictionary for translating between microarray data analysis language and network theory language to facilitate communication between the two fields. we describe several examples that illustrate how the two data analysis fields can inform each other.we used the yeast genome sequences of gene families, microarray profiles and regulatory motif data to test the current wisdom that there is a strong correlation between regulatory motif structure and gene expression profile. our results suggest that duplicate genes tend to be co-expressed but the correlation between motif content and expression similarity is generally poor, only approximately 2-3\\\\% of expression variation can be explained by the motif divergence. our observations suggest that, in addition to the cis-regulatory motif structure in the upstream region of the gene, multiple trans-acting factors in the gene network can influence the pattern of gene expression significantly.development requires the establishment of precise patterns of gene expression, which are primarily controlled by transcription factors binding to cis-regulatory modules. although transcription factor occupancy can now be identified at genome-wide scales, decoding this regulatory landscape remains a daunting challenge. here we used a novel approach to predict spatio-temporal cis-regulatory activity based only on in vivo transcription factor binding and enhancer activity data. we generated a high-resolution atlas of cis-regulatory modules describing their temporal and combinatorial occupancy during drosophila mesoderm development. the binding profiles of cis-regulatory modules with characterized expression were used to train support vector machines to predict five spatio-temporal expression patterns. in vivo transgenic reporter assays demonstrate the high accuracy of these predictions and reveal an unanticipated plasticity in transcription factor binding leading to similar expression. this data-driven approach does not require previous knowledge of transcription factor sequence affinity, function or expression, making it widely applicable.prostate cancer is the second most common cause of male cancer deaths in the united states. however, the full range of prostate cancer genomic alterations is incompletely characterized. here we present the complete sequence of seven primary human prostate cancers and their paired normal counterparts. several tumours contained complex chains of balanced (that is, \\'copy-neutral\\') rearrangements that occurred within or adjacent to known cancer genes. rearrangement breakpoints were enriched near open chromatin, androgen receptor and {erg} {dna} binding sites in the setting of the {ets} gene fusion {tmprss2}-{erg}, but inversely correlated with these regions in tumours lacking {ets} fusions. this observation suggests a link between chromatin or transcriptional regulation and the genesis of genomic aberrations. three tumours contained rearrangements that disrupted {cadm2}, and four harboured events disrupting either {pten} (unbalanced events), a prostate tumour suppressor, or {magi2} (balanced events), a {pten} interacting protein not previously implicated in prostate tumorigenesis. thus, genomic rearrangements may arise from transcriptional or chromatin aberrancies and engage prostate tumorigenic mechanisms.a large fraction of eukaryotic genomes consists of {dna} that is not translated into protein sequence, and little is known about its functional significance. here i show that several classes of non-coding {dna} in drosophila are evolving considerably slower than synonymous sites, and yet show an excess of between-species divergence relative to polymorphism when compared with synonymous sites. the former is a hallmark of selective constraint, but the latter is a signature of adaptive evolution, resembling general patterns of protein evolution in drosophila. i estimate that about 40-70\\\\% of nucleotides in intergenic regions, untranslated portions of mature {mrnas} ({utrs}) and most intronic {dna} are evolutionarily constrained relative to synonymous sites. however, i also use an extension to the {mcdonald}-kreitman test to show that a substantial fraction of the nucleotide divergence in these regions was driven to fixation by positive selection (about 20\\\\% for most intronic and intergenic {dna}, and 60\\\\% for {utrs}). on the basis of these observations, i suggest that a large fraction of the non-translated genome is functionally important and subject to both purifying selection and adaptive evolution. these results imply that, although positive selection is clearly an important facet of protein evolution, adaptive changes to non-coding {dna} might have been considerably more common in the evolution of d. melanogaster.progress in systems biology depends on accurate descriptions of biological networks. connections in a regulatory network are identified as correlations of gene expression across a set of environmental or genetic perturbations. to use this information to predict system behavior, we must test how the nature of perturbations affects topologies of networks they reveal. to probe this question, we focused on metabolism of drosophila melanogaster. our source of perturbations is a set of crosses among 92 wild-derived lines from five populations, replicated in a manner permitting separate assessment of the effects of genetic variation and environmental fluctuation. we directly assayed activities of enzymes and levels of metabolites. using a multivariate bayesian model, we estimated covariance among metabolic parameters and built fine-grained probabilistic models of network topology. the environmental and genetic co-regulation networks are substantially the same among five populations. however, genetic and environmental perturbations reveal qualitative differences in metabolic regulation, suggesting that environmental shifts, such as diet modifications, produce different systemic effects than genetic changes, even if the primary targets are the same.the three-dimensional molecular structure of {dna}, specifically the shape of the backbone and grooves of genomic {dna}, can be dramatically affected by nucleotide changes, which can cause differences in protein-binding affinity and phenotype. we developed an algorithm to measure constraint on the basis of similarity of {dna} topography among multiple species, using hydroxyl radical cleavage patterns to interrogate the solvent-accessible surface area of {dna}. this algorithm found that 12\\\\% of bases in the human genome are evolutionarily constrained—double the number detected by nucleotide sequence–based algorithms. topography-informed constrained regions correlated with functional noncoding elements, including enhancers, better than did regions identified solely on the basis of nucleotide sequence. these results support the idea that the molecular shape of {dna} is under selection and can identify evolutionary history.understanding the relationship between genetic variation and gene expression is a central question in genetics. with the availability of data from high-throughput technologies such as {chip}-chip, expression, and genotyping arrays, we can begin to not only identify associations but to understand how genetic variations perturb the underlying transcription regulatory networks to induce differential gene expression. in this study, we describe a simple model of transcription regulation where the expression of a gene is completely characterized by two properties: the concentrations and promoter affinities of active transcription factors. we devise a method that extends network component analysis ({nca}) to determine how genetic variations in the form of single nucleotide polymorphisms ({snps}) perturb these two properties. applying our method to a segregating population of saccharomyces cerevisiae, we found statistically significant examples of trans-acting {snps} located in regulatory hotspots that perturb transcription factor concentrations and affinities for target promoters to cause global differential expression and cis-acting genetic variations that perturb the promoter affinities of transcription factors on a single gene to cause local differential expression. although many genetic variations linked to gene expressions have been identified, it is not clear how they perturb the underlying regulatory networks that govern gene expression. our work begins to fill this void by showing that many genetic variations affect the concentrations of active transcription factors in a cell and their affinities for target promoters. understanding the effects of these perturbations can help us to paint a more complete picture of the complex landscape of transcription regulation. the software package implementing the algorithms discussed in this work is available as a {matlab} package upon request. one of the fundamental challenges in biology in the post-genomics era is understanding the complex regulatory mechanisms that govern how genes are turned on and off. in a single organism where the functions of individual genes in a population do not differ much, many of the differences between individuals including physical phenotypes, susceptibility to disease, and response to drugs can be attributed to how genes are regulated. previous studies have largely focused on identifying regulator and target genes whose expressions are linked to genetic variations in a population. we present work that focuses on considering a specific set of regulators called transcription factors whose targets can be verified from experiments and whose interactions with those targets have been well studied and modeled. in this setting, we can begin to understand how genetic variations perturb the concentrations and promoter affinities of active transcription factors to induce differential expression of the targets. understanding the effects of these perturbations is important to understanding the fundamental biology of gene regulation and can help us to design and assess therapeutics and treatments for complex diseases.the sequence specificity of {dna}-binding proteins is the primary mechanism by which the cell recognizes genomic features. here, we describe systematic determination of yeast transcription factor {dna}-binding specificities. we obtained binding specificities for 112 {dna}-binding proteins representing 19 distinct structural classes. one-third of the binding specificities have not been previously reported. several binding sequences have striking genomic distributions relative to transcription start sites, supporting their biological relevance and suggesting a role in promoter architecture. among these are rsc3 binding sequences, containing the core {cgcg}, which are found preferentially <100 bp upstream of transcription start sites. mutation of {rsc3} results in a dramatic increase in nucleosome occupancy in hundreds of proximal promoters containing a rsc3 binding element, but has little impact on promoters lacking rsc3 binding sequences, indicating that rsc3 plays a broad role in targeting nucleosome exclusion at yeast promoters.the interpretation of data-driven experiments in genomics often involves a search for biological categories that are enriched for the responder genes identified by the experiments. however, knowledge bases such as the gene ontology ({go}) contain hundreds or thousands of categories with very high overlap between categories. thus, enrichment analysis performed on one category at a time frequently returns large numbers of correlated categories, leaving the choice of the most relevant ones to the user\\'s; interpretation.although there is only one human genome sequence, different genes are expressed in many different cell types and tissues, as well as in different developmental stages or diseases. the structure of this \\'expression space\\' is still largely unknown, as most transcriptomics experiments focus on sampling small regions.genome-scale studies have revealed extensive, cell type-specific colocalization of transcription factors, but the mechanisms underlying this phenomenon remain poorly understood. here, we demonstrate in macrophages and b cells that collaborative interactions of the common factor {pu}.1 with small sets of macrophage- or b cell lineage-determining transcription factors establish cell-specific binding sites that are associated with the majority of promoter-distal {h3k4me1}-marked genomic regions. {pu}.1 binding initiates nucleosome remodeling, followed by {h3k4} monomethylation at large numbers of genomic regions associated with both broadly and specifically expressed genes. these locations serve as beacons for additional factors, exemplified by liver x receptors, which drive both cell-specific gene expression and signal-dependent responses. together with analyses of transcription factor binding and {h3k4me1} patterns in other cell types, these studies suggest that simple combinations of lineage-determining transcription factors can specify the genomic sites ultimately responsible for both cell identity and cell type-specific responses to diverse signaling inputs. \\\\^{a}\\x96¸ lineage-determining transcription factors collaboratively define their cistromes \\\\^{a}\\x96¸ {pu}.1 binding causes nucleosome repositioning and subsequent {h3k4} monomethylation \\\\^{a}\\x96¸ {h3k4me1}+ regions are enriched for motifs for lineage-determining factors \\\\^{a}\\x96¸ second-tier transcription factors bind to regions opened by these master regulatorswhen prioritizing hits from a high-throughput experiment, it is important to correct for random events that falsely appear significant. how is this done and what methods should be used?an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsthe observation that animal morphology tends to be conserved during the embryonic phylotypic period (a period of maximal similarity between the species within each animal phylum) led to the proposition that embryogenesis diverges more extensively early and late than in the middle, known as the hourglass model. this pattern of conservation is thought to reflect a major constraint on the evolution of animal body plans. despite a wealth of morphological data confirming that there is often remarkable divergence in the early and late embryos of species from the same phylum, it is not yet known to what extent gene expression evolution, which has a central role in the elaboration of different animal forms, underpins the morphological hourglass pattern. here we address this question using species-specific microarrays designed from six sequenced drosophila species separated by up to 40 million years. we quantify divergence at different times during embryogenesis, and show that expression is maximally conserved during the arthropod phylotypic period. by fitting different evolutionary models to each gene, we show that at each time point more than 80\\\\% of genes fit best to models incorporating stabilizing selection, and that for genes whose evolutionarily optimal expression level is the same across all species, selective constraint is maximized during the phylotypic period. the genes that conform most to the hourglass pattern are involved in key developmental processes. these results indicate that natural selection acts to conserve patterns of gene expression during mid-embryogenesis, and provide a genome-wide insight into the molecular basis of the hourglass pattern of developmental evolution.deciphering the impact of genetic variants on gene regulation is fundamental to understanding human disease. although gene regulation often involves long-range interactions, it is unknown to what extent non-coding genetic variants influence distal molecular phenotypes. here, we integrate chromatin profiling for three histone marks in lymphoblastoid cell lines ({lcls}) from 75 sequenced individuals with {lcl}-specific {hi-c} and {chia}-{pet}-based chromatin contact maps to uncover one of the largest collections of local and distal histone quantitative trait loci ({hqtls}). distal {qtls} are enriched within topologically associated domains and exhibit largely concordant variation of\\xa0chromatin state coordinated by proximal and distal\\xa0non-coding genetic variants. histone {qtls} are enriched for common variants associated with autoimmune diseases and enable identification of putative target genes of disease-associated variants from genome-wide association studies. these analyses provide insights into how genetic variation can affect human disease phenotypes by coordinated changes in chromatin at interacting regulatory elements.recently, a number of advanced screening technologies have allowed for the comprehensive quantification of aggravating and alleviating genetic interactions among gene pairs. in parallel, {tap}-{ms} studies (tandem affinity purification followed by mass spectroscopy) have been successful at identifying physical protein interactions that can indicate proteins participating in the same molecular complex. here, we propose a method for the joint learning of protein complexes and their functional relationships by integration of quantitative genetic interactions and {tap}-{ms} data. using 3 independent benchmark datasets, we demonstrate that this method is >50\\\\% more accurate at identifying functionally related protein pairs than previous approaches. application to genes involved in yeast chromosome organization identifies a functional map of 91 multimeric complexes, a number of which are novel or have been substantially expanded by addition of new subunits. interestingly, we find that complexes that are enriched for aggravating genetic interactions (i.e., synthetic lethality) are more likely to contain essential genes, linking each of these interactions to an underlying mechanism. these results demonstrate the importance of both large-scale genetic and physical interaction data in mapping pathway architecture and function. biologists are currently producing large amounts of data focused on physical and genetic protein interactions. physical interactions dictate the architecture of the cell in terms of how direct associations between molecules constitute protein complexes, while genetic interactions define functional relationships through cause-and-effect relationships between genes. both of these types of interactions can indicate shared protein functions; however, these two types of interactions are commonly non-overlapping, making their interpretation difficult. along these lines, it has been noted that genetic interactions commonly occur between members of the same protein complex as well as between functionally related complexes. here, we present an integrated framework that incorporates both types of interactions to generate large maps of protein complexes as well as highlight connections between related complexes. the ability to rapidly integrate these two types of data in an automated fashion can accelerate the discovery of new members of protein complexes as well as identify functionally related cellular components.chromatin is composed of {dna} and a variety of modified histones and non-histone proteins, which have an impact on cell differentiation, gene regulation and other key cellular processes. here we present a genome-wide chromatin landscape for drosophila melanogaster based on eighteen histone modifications, summarized by nine prevalent combinatorial patterns. integrative analysis with other data (non-histone chromatin proteins, {dnase} i hypersensitivity, {gro}-seq reads produced by engaged polymerase, short/long {rna} products) reveals discrete characteristics of chromosomes, genes, regulatory elements and other functional domains. we find that active genes display distinct chromatin signatures that are correlated with disparate gene lengths, exon patterns, regulatory functions and genomic contexts. we also demonstrate a diversity of signatures among polycomb targets that include a subset with paused polymerase. this systematic profiling and integrative analysis of chromatin signatures provides insights into how genomic elements are regulated, and will serve as a resource for future experimental investigations of genome structure and function.we present a method for bayesian model-based hierarchical coclustering of gene expression data and use it to study the temporal transcription responses of an anopheles gambiae cell line upon challenge with multiple microbial elicitors. the method fits statistical regression models to the gene expression time series for each experiment and performs coclustering on the genes by optimizing a joint probability model, characterizing gene coregulation between multiple experiments. we compute the model using a two-stage {expectation-maximization}-type algorithm, first fixing the cross-experiment covariance structure and using efficient bayesian hierarchical clustering to obtain a locally optimal clustering of the gene expression profiles and then, conditional on that clustering, carrying out bayesian inference on the cross-experiment covariance using markov chain monte carlo simulation to obtain an expectation. for the problem of model choice, we use a cross-validatory approach to decide between individual experiment modeling and varying levels of coclustering. our method successfully generates tightly coregulated clusters of genes that are implicated in related processes and therefore can be used for analysis of global transcript responses to various stimuli and prediction of gene functions.strong positive selection has resulted in changes to the regulation of the human prodynorphin gene, with evidence for increased expression as compared with chimp. additionally, recent selection has led to different alleles in different parts of the world.large-scale integrated cancer genome characterization efforts including the cancer genome atlas and the cancer cell line encyclopedia have created unprecedented opportunities to study cancer biology in the context of knowing the entire catalog of genetic alterations. a clinically important challenge is to discover cancer subtypes and their molecular drivers in a comprehensive genetic context. curtis et al. [nature (2012) 486(7403):346–352] has recently shown that integrative clustering of copy number and gene expression in 2,000 breast tumors reveals novel subgroups beyond the classic expression subtypes that show distinct clinical outcomes. to extend the scope of integrative analysis for the inclusion of somatic mutation data by massively parallel sequencing, we propose a framework for joint modeling of discrete and continuous variables that arise from integrated genomic, epigenomic, and transcriptomic profiling. the core idea is motivated by the hypothesis that diverse molecular phenotypes can be predicted by a set of orthogonal latent variables that represent distinct molecular drivers, and thus can reveal tumor subgroups of biological and clinical importance. using the cancer cell line encyclopedia dataset, we demonstrate our method can accurately group cell lines by their cell-of-origin for several cancer types, and precisely pinpoint their known and potential cancer driver genes. our integrative analysis also demonstrates the power for revealing subgroups that are not lineage-dependent, but consist of different cancer types driven by a common genetic alteration. application of the cancer genome atlas colorectal cancer data reveals distinct integrated tumor subtypes, suggesting different genetic pathways in colon cancer progression.the diversity of life is one of the most striking aspects of our planet; hence knowing how many species inhabit earth is among the most fundamental questions in science. yet the answer to this question remains enigmatic, as efforts to sample the world\\'s biodiversity to date have been limited and thus have precluded direct quantification of global species richness, and because indirect estimates rely on assumptions that have proven highly controversial. here we show that the higher taxonomic classification of species (i.e., the assignment of species to phylum, class, order, family, and genus) follows a consistent and predictable pattern from which the total number of species in a taxonomic group can be estimated. this approach was validated against well-known taxa, and when applied to all domains of life, it predicts \\\\~{}8.7 million (± 1.3 million {se}) eukaryotic species globally, of which \\\\~{}2.2 million (± 0.18 million {se}) are marine. in spite of 250 years of taxonomic classification and over 1.2 million species already catalogued in a central database, our results suggest that some 86\\\\% of existing species on earth and 91\\\\% of species in the ocean still await description. renewed interest in further exploration and taxonomy is required if this significant gap in our knowledge of life on earth is to be closed.following the identification of several disease-associated polymorphisms by genome-wide association ({gwa}) analysis, interest is now focusing on the detection of effects that, owing to their interaction with other genetic or environmental factors, might not be identified by using standard single-locus tests. in addition to increasing the power to detect associations, it is hoped that detecting interactions between loci will allow us to elucidate the biological and biochemical pathways that underpin disease. here i provide a critical survey of the methods and related software packages currently used to detect the interactions between genetic loci that contribute to human genetic disease. i also discuss the difficulties in determining the biological relevance of statistical interactions.an association between enrichment and depletion of {microrna} ({mirna}) binding sites, 3\\\\^{a}€² {utr} length, and {mrna} expression has been demonstrated in various developing tissues and tissues from different mature organs; but functional, context-dependent {mirna} regulations have yet to be elucidated. towards that goal, we examined {mirna\\\\^{a}}€ ” {mrna} interactions by measuring {mirna} and {mrna} in the same tissue during development and also in malignant conditions. we identified significant {mirna}-mediated biological process categories in developing mouse cerebellum and lung using non-targeted {mrna} expression as the negative control. although {mirnas} in general suppress target {mrna} messages, many predicted {mirna} targets demonstrate a significantly higher level of co-expression than non-target genes in developing cerebellum. this phenomenon is tissue specific since it is not observed in developing lungs. comparison of mouse cerebellar development and medulloblastoma demonstrates a shared {mirna\\\\^{a}}€ ” {mrna} co-expression program for brain-specific neurologic processes such as synaptic transmission and exocytosis , in which {mirna} target expression increases with the accumulation of multiple {mirnas} in developing cerebellum and decreases with the loss of these {mirnas} in brain tumors. these findings demonstrate the context-dependence of {mirna\\\\^{a}}€ ” {mrna} co-expression.an ultimate goal of genetic research is to understand the connection between genotype and phenotype in order to improve the diagnosis and treatment of diseases. the quantitative genetics field has developed a suite of statistical methods to associate genetic loci with diseases and phenotypes, including quantitative trait loci ({qtl}) linkage mapping and genome-wide association studies ({gwas}). however, each of these approaches have technical and biological shortcomings. for example, the amount of heritable variation explained by {gwas} is often surprisingly small and the resolution of many {qtl} linkage mapping studies is poor. the predictive power and interpretation of {qtl} and {gwas} results are consequently limited. in this study, we propose a complementary approach to quantitative genetics by interrogating the vast amount of high-throughput genomic data in model organisms to functionally associate genes with phenotypes and diseases. our algorithm combines the genome-wide functional relationship network for the laboratory mouse and a state-of-the-art machine learning method. we demonstrate the superior accuracy of this algorithm through predicting genes associated with each of 1157 diverse phenotype ontology terms. comparison between our prediction results and a meta-analysis of quantitative genetic studies reveals both overlapping candidates and distinct, accurate predictions uniquely identified by our approach. focusing on bone mineral density ({bmd}), a phenotype related to osteoporotic fracture, we experimentally validated two of our novel predictions (not observed in any previous {gwas}/{qtl} studies) and found significant bone density defects for both timp2 and abcg8 deficient mice. our results suggest that the integration of functional genomics data into networks, which itself is informative of protein function and interactions, can successfully be utilized as a complementary approach to quantitative genetics to predict disease risks. all supplementary material is available at http://cbfg.jax.org/phenotype. many recent efforts to understand the genetic origins of complex diseases utilize statistical approaches to analyze phenotypic traits measured in genetically well-characterized populations. while these quantitative genetics methods are powerful, their success is limited by sampling biases and other confounding factors, and the biological interpretation of results can be challenging since these methods are not based on any functional information for candidate loci. on the other hand, the functional genomics field has greatly expanded in past years, both in terms of experimental approaches and analytical algorithms. however, functional approaches have been applied to understanding phenotypes in only the most basic ways. in this study, we demonstrate that functional genomics can complement traditional quantitative genetics by analytically extracting protein function information from large collections of high throughput data, which can then be used to predict genotype-phenotype associations. we applied our prediction methodology to the laboratory mouse, and we experimentally confirmed a role in osteoporosis for two of our predictions that were not candidates from any previous quantitative genetics study. the ability of our approach to produce accurate and unique predictions implies that functional genomics can complement quantitative genetics and can help address previous limitations in identifying disease genes.motivation: recently developed network component analysis ({nca}) approach is promising for gene regulatory network reconstruction from microarray data. the existing {nca} algorithm is an iterative method which has two potential limitations: computational instability and multiple local solutions. the subsequently developed {nca}-r algorithm with tikhonov regularization can help solve the first issue but cannot completely handle the second one. here we develop a novel fast network component analysis ({fastnca}) algorithm which has an analytical solution that is much faster and does not have the above limitations.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismschanges in gene regulation may be important in evolution. however, the evolutionary properties of regulatory mutations are currently poorly understood. this is partly the result of an incomplete annotation of functional regulatory {dna} in many species. for example, transcription factor binding sites ({tfbss}), a major component of eukaryotic regulatory architecture, are typically short, degenerate, and therefore difficult to differentiate from randomly occurring, nonfunctional sequences. furthermore, although sites such as {tfbss} can be computationally predicted using evolutionary conservation as a criterion, estimates of the true level of selective constraint (defined as the fraction of strongly deleterious mutations occurring at a locus) in regulatory regions will, by definition, be upwardly biased in datasets that are a priori evolutionarily conserved. here we investigate the fitness effects of regulatory mutations using two complementary datasets of human {tfbss} that are likely to be relatively free of ascertainment bias with respect to evolutionary conservation but, importantly, are supported by experimental data. the first is a collection of almost >2,100 human {tfbss} drawn from the literature in the {transfac} database, and the second is derived from several recent high-throughput chromatin immunoprecipitation coupled with genomic microarray ({chip}-chip) analyses. we also define a set of putative cis-regulatory modules ({pcrms}) by spatially clustering multiple {tfbss} that regulate the same gene. we find that a relatively high proportion ( approximately 37\\\\%) of mutations at {tfbss} are strongly deleterious, similar to that at a 2-fold degenerate protein-coding site. however, constraint is significantly reduced in human and chimpanzee {pcrms} and {chip}-chip sequences, relative to macaques. we estimate that the fraction of regulatory mutations that have been driven to fixation by positive selection in humans is not significantly different from zero. we also find that the level of selective constraint in our {tfbss}, {pcrms}, and {chip}-chip sequences is negatively correlated with the expression breadth of the regulated gene, whereas the opposite relationship holds at that gene\\'s nonsynonymous and synonymous sites. finally, we find that the rate of protein evolution in a transcription factor appears to be positively correlated with the breadth of expression of the gene it regulates. our study suggests that strongly deleterious regulatory mutations are considerably more likely (1.6-fold) to occur in tissue-specific than in housekeeping genes, implying that there is a fitness cost to increasing \"complexity\" of gene expression.forward genetic screens in model organisms have provided important insights into numerous aspects of development, physiology and pathology. with the availability of complete genome sequences and the introduction of {rna}-mediated gene interference ({rnai}), systematic reverse genetic screens are now also possible. until now, such genome-wide {rnai} screens have mostly been restricted to cultured cells and ubiquitous gene inactivation in caenorhabditis elegans. this powerful approach has not yet been applied in a tissue-specific manner. here we report the generation and validation of a genome-wide library of drosophila melanogaster {rnai} transgenes, enabling the conditional inactivation of gene function in specific tissues of the intact organism. our {rnai} transgenes consist of short gene fragments cloned as inverted repeats and expressed using the binary {gal4}/{uas} system. we generated 22,270 transgenic lines, covering 88\\\\% of the predicted protein-coding genes in the drosophila genome. molecular and phenotypic assays indicate that the majority of these transgenes are functional. our transgenic {rnai} library thus opens up the prospect of systematically analysing gene functions in any tissue and at any stage of the drosophila lifespan.borrowing ideas that were originally developed to study electronic circuits, two reports decipher how yeast reacts to changes in its environment by analysing the organism\\'s responses to oscillating input signals. systems biology has certainly caught people\\'s attention. but when pressed to define precisely what it is, let alone how it will complement classic reductionist approaches, concrete answers can be hard to find.transcription and regulation of genes originate from transcription pre-initiation complexes ({pics}). their structural and positional organization across eukaryotic genomes is unknown. here we applied lambda exonuclease to chromatin immunoprecipitates (termed {chip}-exo) to examine the precise location of 6,045 {pics} in saccharomyces. {pics}, including {rna} polymerase {ii} and protein complexes {tfiia}, {tfiib}, {tfiid} (or {tbp}), {tfiie}, {tfiif}, {tfiih} and {tfiik} were positioned within promoters and excluded from coding regions. exonuclease patterns were in agreement with crystallographic models of the {pic}, and were sufficiently precise to identify {tata}-like elements at so-called {tata}-less promoters. these {pics} and their transcription start sites were positionally constrained at {tfiid}-engaged downstream +1 nucleosomes. at {tata}-box-containing promoters, which are depleted of {tfiid}, a +1 nucleosome was positioned to be in competition with the {pic}, which may allow greater latitude in start-site selection. our genomic localization of messenger {rna} and non-coding {rna} {pics} reveals that two {pics}, in inverted orientation, may occupy the flanking borders of nucleosome-free regions. their unambiguous detection may help distinguish bona fide genes from transcriptional noise.a large number of cis-regulatory sequences have been annotated in the human genome, but defining their target genes remains a challenge. one strategy is to identify the long-range looping interactions at these elements with the use of chromosome conformation capture ({3c})-based techniques. however, previous studies lack either the resolution or coverage to permit a whole-genome, unbiased view of chromatin interactions. here we report a comprehensive chromatin interaction map generated in human fibroblasts using a genome-wide {3c} analysis method ({hi-c}). we determined over one million long-range chromatin interactions at 5\\\\&\\\\#19;10-kb resolution, and uncovered general principles of chromatin organization at different types of genomic features. we also characterized the dynamics of promoter\\\\&\\\\#19;enhancer contacts after {tnf}-\\\\&\\\\#177; signalling in these cells. unexpectedly, we found that {tnf}-\\\\&\\\\#177;-responsive enhancers are already in contact with their target promoters before signalling. such pre-existing chromatin looping, which also exists in other cell types with different extracellular signalling, is a strong predictor of gene induction. our observations suggest that the three-dimensional chromatin landscape, once established in a particular cell type, is relatively stable and could influence the selection or activation of target genes by a ubiquitous transcription activator in a cell-specific manner.changes in {dna} and messenger {rna} ({mrna}) expression levels have been used to estimate evolutionary changes between species. however protein expression levels may better reflect selection on divergent and constrained phenotypes. khan et al. (p. 1100, published online 17 october; see the perspective by vogel) measured the differences among and within species between {mrna} expression and protein levels in humans, chimpanzees, and rhesus macaques, identifying protein transcripts that seem to be under lineage-specific constraint between humans and chimpanzees.the genetics of complex disease produce alterations in the molecular interactions of cellular pathways whose collective effect may become clear through the organized structure of molecular networks. to characterize molecular systems associated with late-onset alzheimer\\'s disease ({load}), we constructed gene-regulatory networks in 1,647 postmortem brain tissues from {load} patients and nondemented subjects, and we demonstrate that {load} reconfigures specific portions of the molecular interaction structure. through an integrative network-based approach, we rank-ordered these network structures for relevance to {load} pathology, highlighting an immune- and microglia-specific module that is dominated by genes involved in pathogen phagocytosis, contains {tyrobp} as a key regulator, and is upregulated in {load}. mouse microglia cells overexpressing intact or truncated {tyrobp} revealed expression changes that significantly overlapped the human brain {tyrobp} network. thus the causal network structure is a useful predictor of response to gene perturbations and presents a framework to test models of disease mechanisms underlying {load}. copyright {\\\\copyright} 2013 elsevier inc. all rights reserved.recently, there has been an increasing interest in stoichiometric analysis of metabolic flux distributions. flux balance methods only require information about metabolic reaction stoichiometry, metabolic requirements for growth, and the measurement of a few strain–specific parameters. this information determines the domain of stoichiometrically allowable flux distributions that may be taken to define a strain\\'s \"metabolic genotype\". within this domain a single flux distribution is sought based on assumed behavior, such as maximal growth rates. the optimal flux distributions are calculated using linear optimization and may be taken to represent the strain\\'s \"metabolic phenotype\" under the particular conditions. this flux balance methodology allows the quantitative interpretation of metabolic physiology, gives an interpretation of experimental data, provides a guide to metabolic engineering, enables optimal medium formulation, and provides a method for bioprocess optimization. this spectrum of applications, and its ease of use, makes the metabolic flux balance model a potentially valuable approach for the design and optimization of bioprocesses.intracellular signalling systems are highly complex. this complexity makes handling, analysis and visualisation of available knowledge a major challenge in current signalling research. here, we present a novel framework for mapping signal-transduction networks that avoids the combinatorial explosion by breaking down the network in reaction and contingency information. it provides two new visualisation methods and automatic export to mathematical models. we use this framework to compile the presently most comprehensive map of the yeast {map} kinase network. our method improves previous strategies by combining (i) more concise mapping adapted to empirical data, ({ii}) individual referencing for each piece of information, ({iii}) visualisation without simplifications or added uncertainty, ({iv}) automatic visualisation in multiple formats, (v) automatic export to mathematical models and ({vi}) compatibility with established formats. the framework is supported by an open source software tool that facilitates integration of the three levels of network analysis: definition, visualisation and mathematical modelling. the framework is species independent and we expect that it will have wider impact in signalling research on any system.biological networks have evolved to be highly functional within uncertain environments while remaining extremely adaptable. one of the main contributors to the robustness and evolvability of biological networks is believed to be their modularity of function, with modules defined as sets of genes that are strongly interconnected but whose function is separable from those of other modules. here, we investigate the in silico evolution of modularity and robustness in complex artificial metabolic networks that encode an increasing amount of information about their environment while acquiring ubiquitous features of biological, social, and engineering networks, such as scale-free edge distribution, small-world property, and fault-tolerance. these networks evolve in environments that differ in their predictability, and allow us to study modularity from topological, information-theoretic, and gene-epistatic points of view using new tools that do not depend on any preconceived notion of modularity. we find that for our evolved complex networks as well as for the yeast protein–protein interaction network, synthetic lethal gene pairs consist mostly of redundant genes that lie close to each other and therefore within modules, while knockdown suppressor gene pairs are farther apart and often straddle modules, suggesting that knockdown rescue is mediated by alternative pathways or modules. the combination of network modularity tools together with genetic interaction data constitutes a powerful approach to study and dissect the role of modularity in the evolution and function of biological networks. the modular organization of cells is not immediately obvious from the network of interacting genes, proteins, and molecules. a new window into cellular modularity is opened up by genetic data that identifies pairs of genes that interact either directly or indirectly to provide robustness to cellular function. such pairs can map out the modular nature of a network if we understand how they relate to established mathematical clustering methods applied to networks to identify putative modules. we can test the relationship between genetically interacting pairs and modules on artificial data: large networks of interacting proteins and molecules that were evolved within an artificial chemistry and genetics, and that pass the standard tests for biological networks. modularity evolves in these networks in order to deal with a multitude of functional goals, with a degree depending on environmental variability. relationships between genetically interacting pairs and modules similar to those displayed by the artificial gene networks are found in the protein–protein interaction network of baker\\'s yeast. the evolution of complex functional biological networks in silico provides an opportunity to develop and test new methods and tools to understand the complexity of biological systems at the network level.genome-wide association studies ({gwas}) for quantitative traits and disease in humans and other species have shown that there are many loci that contribute to the observed resemblance between relatives. {gwas} to date have mostly focussed on discovery of genes or regulatory regions habouring causative polymorphisms, using single {snp} analyses and setting stringent {type-i} error rates. genome-wide marker data can also be used to predict genetic values and therefore predict phenotypes. here, we propose a bayesian method that utilises all marker data simultaneously to predict phenotypes. we apply the method to three traits: coat colour, \\\\%{cd8} cells, and mean cell haemoglobin, measured in a heterogeneous stock mouse population. we find that a model that contains both additive and dominance effects, estimated from genome-wide marker data, is successful in predicting unobserved phenotypes and is significantly better than a prediction based upon the phenotypes of close relatives. correlations between predicted and actual phenotypes were in the range of 0.4 to 0.9 when half of the number of families was used to estimate effects and the other half for prediction. posterior probabilities of {snps} being associated with coat colour were high for regions that are known to contain loci for this trait. the prediction of phenotypes using large samples, high-density {snp} data, and appropriate statistical methodology is feasible and can be applied in human medicine, forensics, or artificial selection programs. results from recent genome-wide association studies indicate that for most complex traits, there are many loci that contribute to variation in observed phenotype and that the effect of a single variant (single nucleotide polymorphism, {snp}) on a phenotype is small. here, we propose a method that combines the effects of multiple {snps} to make a prediction of a phenotype that has not been observed. we apply the method to data on mice, using phenotypic and genomic data from some individuals to predict phenotypes in other, either related or unrelated, individuals. we find that correlations between predicted and actual phenotypes are in the range of 0.4 to 0.9. the method also shows that the {snps} used in the prediction appear in regions that are known to contain genes associated with the traits studied. the prediction of unobserved phenotypes from high-density {snp} data and appropriate statistical methodology is feasible and can be applied in human medicine, forensics, or artificial breeding programs.the brisk discovery of novel inherited disease markers by genome-wide association ({gwa}) studies has raised expectations for predicting disease risk by analysing multiple common alleles. however, the statistics used during the discovery phase of research (such as odds ratios or p values for association) are not the most appropriate measures for evaluating the predictive value of genetic profiles. we argue that other measures--such as sensitivity, specificity, and positive and negative predictive values--are more useful when proposing a genetic profile for risk prediction.background: mixture model on graphs ({mmg}) is a probabilistic model that integrates network topology with (gene, protein) expression data to predict the regulation state of genes and proteins. it is remarkably robust to missing data, a feature particularly important for its use in quantitative proteomics. a new implementation in c and interfaced with r makes {mmg} extremely fast and easy to use and to extend.transcription factors ({tfs}) and epigenetic modifications play crucial roles in the regulation of gene expression, and correlations between the two types of factors have been discovered. however, methods for quantitatively studying the correlations remain limited. here, we present a computational approach to systematically investigating how epigenetic changes in chromatin architectures or {dna} sequences relate to {tf} binding. we implemented statistical analyses to illustrate that epigenetic modifications are predictive of {tf} binding affinities, without the need of sequence information. intriguingly, by considering genome locations relative to transcription start sites ({tsss}) or enhancer midpoints, our analyses show that different locations display various relationship patterns. for instance, {h3k4me3}, h3k9ac and h3k27ac contribute more in the regions near {tsss}, whereas {h3k4me1} and h3k79me2 dominate in the regions far from {tsss}. {dna} methylation plays relatively important roles when close to {tsss} than in other regions. in addition, the results show that epigenetic modification models for the predictions of {tf} binding affinities are cell line-specific. taken together, our study elucidates highly coordinated, but location- and cell type-specific relationships between epigenetic modifications and binding affinities of {tfs}.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismschanges in gene regulation are thought to have contributed to the evolution of human development. however, in vivo evidence for uniquely human developmental regulatory function has remained elusive. in transgenic mice, a conserved noncoding sequence ({hacns1}) that evolved extremely rapidly in humans acted as an enhancer of gene expression that has gained a strong limb expression domain relative to the orthologous elements from chimpanzee and rhesus macaque. this gain of function was consistent across two developmental stages in the mouse and included the presumptive anterior wrist and proximal thumb. in vivo analyses with synthetic enhancers, in which human-specific substitutions were introduced into the chimpanzee enhancer sequence or reverted in the human enhancer to the ancestral state, indicated that 13 substitutions clustered in an 81-base pair module otherwise highly constrained among terrestrial vertebrates were sufficient to confer the human-specific limb expression domain.recently developed methods for learning sparse classifiers are among the state-of-the-art in supervised learning. these methods learn classifiers that incorporate weighted sums of basis functions with sparsity-promoting priors encouraging the weight estimates to be either significantly large or exactly zero. from a learning-theoretic perspective, these methods control the capacity of the learned classifier by minimizing the number of basis functions used, resulting in better generalization. this paper presents three contributions related to learning sparse classifiers. first, we introduce a true multiclass formulation based on multinomial logistic regression. second, by combining a bound optimization approach with a component-wise update procedure, we derive fast exact algorithms for learning sparse multiclass classifiers that scale favorably in both the number of training samples and the feature dimensionality, making them applicable even to large data sets in high-dimensional feature spaces. to the best of our knowledge, these are the first algorithms to perform exact multinomial logistic regression with a sparsity-promoting prior. third, we show how nontrivial generalization bounds can be derived for our classifier in the binary case. experimental results on standard benchmark data sets attest to the accuracy, sparsity, and efficiency of the proposed methods.determining how transcriptional regulatory signals are encoded in vertebrate genomes is essential for understanding the origins of multicellular complexity; yet the genetic code of vertebrate gene regulation remains poorly understood. in an attempt to elucidate this code, we synergistically combined genome-wide gene-expression profiling, vertebrate genome comparisons, and transcription factor binding-site analysis to define sequence signatures characteristic of candidate tissue-specific enhancers in the human genome. we applied this strategy to microarray-based gene expression profiles from 79 human tissues and identified 7187 candidate enhancers that defined their flanking gene expression, the majority of which were located outside of known promoters. we cross-validated this method for its ability to de novo predict tissue-specific gene expression and confirmed its reliability in 57 of the 79 available human tissues, with an average precision in enhancer recognition ranging from 32\\\\% to 63\\\\% and a sensitivity of 47\\\\%. we used the sequence signatures identified by this approach to successfully assign tissue-specific predictions to ∼328,000 human–mouse conserved noncoding elements in the human genome. by overlapping these genome-wide predictions with a data set of enhancers validated in vivo, in transgenic mice, we were able to confirm our results with a 28\\\\% sensitivity and 50\\\\% precision. these results indicate the power of combining complementary genomic data sets as an initial computational foray into a global view of tissue-specific gene regulation in vertebrates.inferring gene regulatory networks from high-throughput genomic data is one of the central problems in computational biology. in this paper, we describe a predictive modeling approach for studying regulatory networks, based on a machine learning algorithm called {medusa}. {medusa} integrates promoter sequence, {mrna} expression, and transcription factor occupancy data to learn gene regulatory programs that predict the differential expression of target genes. instead of using clustering or correlation of expression profiles to infer regulatory relationships, {medusa} determines condition-specific regulators and discovers regulatory motifs that mediate the regulation of target genes. in this way, {medusa} meaningfully models biological mechanisms of transcriptional regulation. {medusa} solves the problem of predicting the differential (up/down) expression of target genes by using boosting, a technique from statistical learning, which helps to avoid overfitting as the algorithm searches through the high-dimensional space of potential regulators and sequence motifs. experimental results demonstrate that {medusa} achieves high prediction accuracy on held-out experiments (test data), that is, data not seen in training. we also present context-specific analysis of {medusa} regulatory programs for {dna} damage and hypoxia, demonstrating that {medusa} identifies key regulators and motifs in these processes. a central challenge in the field is the difficulty of validating reverse-engineered networks in the absence of a gold standard. our approach of learning regulatory programs provides at least a partial solution for the problem: {medusa}\\'s prediction accuracy on held-out data gives a concrete and statistically sound way to validate how well the algorithm performs. with {medusa}, statistical validation becomes a prerequisite for hypothesis generation and network building rather than a secondary consideration.genome-wide expression profiles of genetic mutants provide a wide variety of measurements of cellular responses to perturbations. typical analysis of such data identifies genes affected by perturbation and uses clustering to group genes of similar function. in this paper we discover a finer structure of interactions between genes, such as causality, mediation, activation, and inhibition by using a bayesian network framework. we extend this framework to correctly handle perturbations, and to identify significant subnetworks of interacting genes. we apply this method to expression data of s. cerevisiae mutants and uncover a variety of structured metabolic, signaling and regulatory {pathways.contact}: danab@cs.huji.ac.ilmotivation: gene regulatory networks underlying temporal processes, such as the cell cycle or the life cycle of an organism, can exhibit significant topological changes to facilitate the underlying dynamic regulatory functions. thus, it is essential to develop methods that capture the temporal evolution of the regulatory networks. these methods will be an enabling first step for studying the driving forces underlying the dynamic gene regulation circuitry and predicting the future network structures in response to internal and external stimuli.we investigated the organization of interacting proteins and protein complexes into networks of modules. a network-clustering method was developed to identify modules. this method of network-structure determination was validated by clustering known signaling-protein modules and by identifying module rudiments in exclusively high-throughput protein-interaction data with high error frequencies and low coverage. the signaling network controlling the yeast developmental transition to a filamentous form was clustered. abstraction of a modular network-structure model identified module-organizer proteins and module-connector proteins. the functions of these proteins suggest that they are important for module function and intermodule communication.motivation: liquid chromatography tandem mass spectrometry ({lc}-{ms}/{ms}) is the predominant method to comprehensively characterize complex protein mixtures such as samples from prefractionated or complete proteomes. in order to maximize proteome coverage for the studied sample, i.e. identify as many traceable proteins as possible, {lc}-{ms}/{ms} experiments are typically repeated extensively and the results combined. proteome coverage prediction is the task of estimating the number of peptide discoveries of future {lc}-{ms}/{ms} experiments. proteome coverage prediction is important to enhance the design of efficient proteomics studies. to date, there does not exist any method to reliably estimate the increase of proteome coverage at an early stage.  results: we propose an extended infinite markov model {dirisim} to extrapolate the progression of proteome coverage based on a small number of already performed {lc}-{ms}/{ms} experiments. the method explicitly accounts for the uncertainty of peptide identifications. we tested {dirisim} on a set of 37 {lc}-{ms}/{ms} experiments of a complete proteome sample and demonstrated that {dirisim} correctly predicts the coverage progression already from a small subset of experiments. the predicted progression enabled us to specify maximal coverage for the test sample. we demonstrated that quality requirements on the final proteome map impose an upper bound on the number of useful experiment repetitions and limit the achievable proteome coverage.  contact: manfredc@inf.ethz.ch; jbuhmann@inf.ethz.ch 10.1093/bioinformatics/btp233{background}:a tremendous amount of efforts have been devoted to identifying genes for diagnosis and prognosis of diseases using microarray gene expression data. it has been demonstrated that gene expression data have cluster structure, where the clusters consist of co-regulated genes which tend to have coordinated functions. however, most available statistical methods for gene selection do not take into consideration the cluster {structure.results}:we propose a supervised group lasso approach that takes into account the cluster structure in gene expression data for gene selection and predictive model building. for gene expression data without biological cluster information, we first divide genes into clusters using the k-means approach and determine the optimal number of clusters using the gap method. the supervised group lasso consists of two steps. in the first step, we identify important genes within each cluster using the lasso method. in the second step, we select important clusters using the group lasso. tuning parameters are determined using v-fold cross validation at both steps to allow for further flexibility. prediction performance is evaluated using leave-one-out cross validation. we apply the proposed method to disease classification and survival analysis with microarray {data.conclusion}:we analyze four microarray data sets using the proposed approach: two cancer data sets with binary cancer occurrence as outcomes and two lymphoma data sets with survival outcomes. the results show that the proposed approach is capable of identifying a small number of influential gene clusters and important genes within those clusters, and has better prediction performance than existing methods.much recent work in bioinformatics has focused on the inference of various types of biological networks, representing gene regulation, metabolic processes, protein-protein interactions, etc. a common setting involves inferring network edges in a supervised fashion from a set of high-confidence edges, possibly characterized by multiple, heterogeneous data sets (protein sequence, gene expression, etc.). here, we distinguish between two modes of inference in this setting: direct inference based upon similarities between nodes joined by an edge, and indirect inference based upon similarities between one pair of nodes and another pair of nodes. we propose a supervised approach for the direct case by translating it into a distance metric learning problem. a relaxation of the resulting convex optimization problem leads to the support vector machine ({svm}) algorithm with a particular kernel for pairs, which we call the metric learning pairwise kernel. this new kernel for pairs can easily be used by most {svm} implementations to solve problems of supervised classification and inference of pairwise relationships from heterogeneous data. we demonstrate, using several real biological networks and genomic datasets, that this approach often improves upon the state-of-the-art {svm} for indirect inference with another pairwise kernel, and that the combination of both kernels always improves upon each individual kernel. the metric learning pairwise kernel is a new formulation to infer pairwise relationships with {svm}, which provides state-of-the-art results for the inference of several biological networks from heterogeneous genomic data.gene expression measurements are influenced by a wide range of factors, such as the state of the cell, experimental conditions and variants in the sequence of regulatory regions. to understand the effect of a variable of interest, such as the genotype of a locus, it is important to account for variation that is due to confounding causes. here, we present {vbqtl}, a probabilistic approach for mapping expression quantitative trait loci ({eqtls}) that jointly models contributions from genotype as well as known and hidden confounding factors. {vbqtl} is implemented within an efficient and flexible inference framework, making it fast and tractable on large-scale problems. we compare the performance of {vbqtl} with alternative methods for dealing with confounding variability on {eqtl} mapping datasets from simulations, yeast, mouse, and human. employing bayesian complexity control and joint modelling is shown to result in more precise estimates of the contribution of different confounding factors resulting in additional associations to measured transcript levels compared to alternative approaches. we present a threefold larger collection of cis {eqtls} than previously found in a whole-genome {eqtl} scan of an outbred human population. altogether, 27\\\\% of the tested probes show a significant genetic association in cis, and we validate that the additional {eqtls} are likely to be real by replicating them in different sets of individuals. our method is the next step in the analysis of high-dimensional phenotype data, and its application has revealed insights into genetic regulation of gene expression by demonstrating more abundant cis-acting {eqtls} in human than previously shown. our software is freely available online at http://www.sanger.ac.uk/resources/softwa\\u200bre/peer/. gene expression is a complex phenotype. the measured expression level in an experiment can be affected by a wide range of factors—state of the cell, experimental conditions, variants in the sequence of regulatory regions, and others. to understand genotype-to-phenotype relationships, we need to be able to distinguish the variation that is due to the genetic state from all the confounding causes. we present {vbqtl}, a probabilistic method for dissecting gene expression variation by jointly modelling the underlying global causes of variability and the genetic effect. our method is implemented in a flexible framework that allows for quick model adaptation and comparison with alternative models. the probabilistic approach yields more accurate estimates of the contributions from different sources of variation. applying {vbqtl}, we find that common genetic variation controlling gene expression levels in human is more abundant than previously shown, which has implications for a wide range of studies relating genotype to phenotype.understanding host response to influenza virus infection will facilitate development of better diagnoses and therapeutic interventions. several different experimental models have been used as a proxy for human infection, including cell cultures derived from human cells, mice, and non-human primates. each of these systems has been studied extensively in isolation, but little effort has been directed toward systematically characterizing the conservation of host response on a global level beyond known immune signaling cascades.motivation: the existing supervised methods for biological network inference work on each of the networks individually based only on intra-species information such as gene expression data. we believe that it will be more effective to use genomic data and cross-species evolutionary information from different species simultaneously, rather than to use the genomic data {alone.results}: we created a new semi-supervised learning method called link propagation for inferring biological networks of multiple species based on genome-wide data and evolutionary information. the new method was applied to simultaneous reconstruction of three metabolic networks of caenorhabditis elegans, helicobacter pylori and saccharomyces cerevisiae, based on gene expression similarities and amino acid sequence similarities. the experimental results proved that the new simultaneous network inference method consistently improves the predictive performance over the individual network inferences, and it also outperforms in accuracy and speed other established methods such as the pairwise support vector {machine.availability}: the software and data are available at {http://cbio.ensmp.fr/∼yyamanishi/linkpropagation}/.contact: {kashima@mist.i.u-tokyo.ac.jpsupplementary} information: supplementary data are available at bioinformatics online.motivation: inferring a gene regulatory network exclusively from microarray expression profiles is a difficult but important task. the aim of this work is to compare the predictive power of some of the most popular algorithms in different conditions (like data taken at equilibrium or time courses) and on both synthetic and real microarray data. we are in particular interested in comparing similarity measures both of linear type (like correlations and partial correlations) and of non-linear type (mutual information and conditional mutual information), and in investigating the underdetermined case (less samples than {genes).results}: in our simulations we see that all network inference algorithms obtain better performances from data produced with \\'structural\\' perturbations, like gene knockouts at steady state, than with any dynamical perturbation. the predictive power of all algorithms is confirmed on a reverse engineering problem from escherichia coli gene profiling data: the edges of the \\'physical\\' network of transcription factor–binding sites are significantly overrepresented among the highest weighting edges of the graph that we infer directly from the data without any structure supervision. comparing synthetic and in vivo data on the same network graph allows us to give an indication of how much more complex a real transcriptional regulation program is with respect to an artificial {model.availability}: software is freely available at the {url} {http://people.sissa.it/\\\\~{}altafini/papers/sobial07}/contact: {altafini@sissa.itsupplementary} information: supplementary data are available at bioinformatics online.motivation: recent advances in high-throughput omics technologies have enabled biomedical researchers to collect large-scale genomic data. as a consequence, there has been growing interest in developing methods to integrate such data to obtain deeper insights regarding the underlying biological system. a key challenge for integrative studies is the heterogeneity present in the different omics data sources, which makes it difficult to discern the coordinated signal of interest from source-specific noise or extraneous effects.gene expression data generated systematically in a given system over multiple time points provides a source of perturbation that can be leveraged to infer causal relationships among genes explaining network changes. previously, we showed that food intake has a large impact on blood gene expression patterns and that these responses, either in terms of gene expression level or gene-gene connectivity, are strongly associated with metabolic diseases. in this study, we explored which genes drive the changes of gene expression patterns in response to time and food intake. we applied the granger causality test and the dynamic bayesian network to gene expression data generated from blood samples collected at multiple time points during the course of a day. the simulation result shows that combining many short time series together is as powerful to infer granger causality as using a single long time series. using the granger causality test, we identified genes that were supported as the most likely causal candidates for the coordinated temporal changes in the network. these results show that {per1} is a key regulator of the blood transcriptional network, in which multiple biological processes are under circadian rhythm regulation. the fasted and fed dynamic bayesian networks showed that over 72\\\\% of dynamic connections are self links. finally, we show that different processes such as inflammation and lipid metabolism, which are disconnected in the static network, become dynamically linked in response to food intake, which would suggest that increasing nutritional load leads to coordinate regulation of these biological processes. in conclusion, our results suggest that food intake has a profound impact on the dynamic co-regulation of multiple biological processes, such as metabolism, immune response, apoptosis and circadian rhythm. the results could have broader implications for the design of studies of disease association and drug response in clinical trials. peripheral blood is the most readily accessible human tissue for clinical studies and experimental research more generally. large-scale molecular profiling technologies have enabled measurements of {mrna} expression on the scale of whole genomes. understanding the relationships between human blood gene expression profiles and clinical traits is extremely useful for inferring causal factors for human disease and for studying drug response. biological pathways and the complex behaviors they induce are not static, but change dynamically in response to external factors such as intake/uptake of nutrients and administration of drugs. we employed a randomized, two-arm cross-over design to assess the effects of fasting and feeding on the dynamic changes of blood transcriptional network. our work has convincingly shown that feeding or increasing nutritional load affects the human circadian rhythm system which connects to other biological processes including metabolic and immune responses. we believe this is a first step towards a more comprehensive population-based study that seeks to connect changes in the blood transcriptome to drug response, and to disease and biology more generally.finding the components of cellular circuits and determining their functions systematically remains a major\\xa0challenge in mammalian cells. here, we introduced genome-wide pooled {crispr}-cas9 libraries into dendritic cells ({dcs}) to identify genes that control the induction of tumor necrosis factor (tnf) by bacterial lipopolysaccharide ({lps}), a key process in the host response to pathogens, mediated by the tlr4 pathway. we found many of the known regulators of tlr4 signaling, as well as dozens of previously unknown candidates that we validated. by measuring protein markers and {mrna} profiles in {dcs} that are deficient in known or candidate genes, we classified\\xa0the genes into three functional modules with distinct effects on the canonical responses to {lps} and highlighted functions for the {paf} complex and oligosaccharyltransferase ({ost}) complex. our findings uncover new facets of innate immune circuits in primary cells and provide a genetic approach for dissection of mammalian cell circuits.modeling of gene expression data from time course experiments often involves the use of linear models such as those obtained from principal component analysis ({pca}), independent component analysis ({ica}), or other methods. such methods do not generally yield factors with a clear biological interpretation. moreover, implicit assumptions about the measurement errors often limit the application of these methods to log-transformed data, destroying linear structure in the untransformed expression data. in this work, a method for the linear decomposition of gene expression data by multivariate curve resolution ({mcr}) is introduced. the {mcr} method is based on an alternating least-squares ({als}) algorithm implemented with a weighted least squares approach. the new method, {mcr}-{wals}, extracts a small number of basis functions from untransformed microarray data using only non-negativity constraints. measurement error information can be incorporated into the modeling process and missing data can be imputed. the utility of the method is demonstrated through its application to yeast cell cycle data. profiles extracted by {mcr}-{wals} exhibit a strong correlation with cell cycle-associated genes, but also suggest new insights into the regulation of those genes. the unique features of the {mcr}-{wals} algorithm are its freedom from assumptions about the underlying linear model other than the non-negativity of gene expression, its ability to analyze non-log-transformed data, and its use of measurement error information to obtain a weighted model and accommodate missing measurements.recent high-throughput efforts such as {encode} have generated a large body of genome-scale transcriptional data in multiple conditions (e.g., cell-types and disease states). leveraging these data is especially important for network-based approaches to human disease, for instance to identify coherent transcriptional modules (subnetworks) that can inform functional disease mechanisms and pathological pathways. yet, genome-scale network analysis across conditions is significantly hampered by the paucity of robust and computationally-efficient methods. building on the {higher-order} generalized singular value decomposition, we introduce a new algorithmic approach for efficient, parameter-free and reproducible identification of network-modules simultaneously across multiple conditions. our method can accommodate weighted (and unweighted) networks of any size and can similarly use co-expression or raw gene expression input data, without hinging upon the definition and stability of the correlation used to assess gene co-expression. in simulation studies, we demonstrated distinctive advantages of our method over existing methods, which was able to recover accurately both common and condition-specific network-modules without entailing ad-hoc input parameters as required by other approaches. we applied our method to genome-scale and multi-tissue transcriptomic datasets from rats (microarray-based) and humans ({mrna}-sequencing-based) and identified several common and tissue-specific subnetworks with functional significance, which were not detected by other methods. in humans we recapitulated the crosstalk between cell-cycle progression and cell-extracellular matrix interactions processes in ventricular zones during neocortex expansion and further, we uncovered pathways related to development of later cognitive functions in the cortical plate of the developing brain which were previously unappreciated. analyses of seven rat tissues identified a multi-tissue subnetwork of co-expressed heat shock protein (hsp) and cardiomyopathy genes (bag3, cryab, kras, emd, plec), which was significantly replicated using separate failing heart and liver gene expression datasets in humans, thus revealing a conserved functional role for hsp genes in cardiovascular disease.in transcriptional regulatory networks, the coincident binding of a combination of factors to regulate a gene implies the existence of complex mechanisms to control both the gene expression profile and specificity of the response. unraveling this complexity is a major challenge to biologists. here, a novel network topology-based clustering approach was applied to condition-specific genome-wide chromatin localization and expression data to characterize a dynamic transcriptional regulatory network responsive to the fatty acid oleate. a network of four (predicted) regulators of the response (oaf1p, pip2p, adr1p and oaf3p) was investigated. by analyzing trends in the network structure, we found that two groups of multi-input motifs form in response to oleate, each controlling distinct functional classes of genes. this functionality is contributed in part by oaf1p, which is a component of both types of multi-input motifs and has two different regulatory activities depending on its binding context. the dynamic cooperation between oaf1p and pip2p appears to temporally synchronize the two different responses. together, these data suggest a network mechanism involving dynamic combinatorial control for coordinating transcriptional responses.identifying conserved and divergent response patterns in gene networks is becoming increasingly important. a common approach is integrating expression information with gene association networks in order to find groups of connected genes that are activated or repressed. in many cases, researchers are also interested in comparisons across species (or conditions). finding an active sub-network is a hard problem and applying it across species requires further considerations (e.g. orthology information, expression data and networks from different sources). to address these challenges we devised {moduleblast}, which uses both expression and network topology to search for highly relevant sub-networks. we have applied {moduleblast} to expression and interaction data from mouse, macaque and human to study immune response and aging. the immune response analysis identified several relevant modules, consistent with recent findings on apoptosis and {nfκb} activation following infection. temporal analysis of these data revealed cascades of modules that are dynamically activated within and across species. we have experimentally validated some of the novel hypotheses resulting from the analysis of the {moduleblast} results leading to new insights into the mechanisms used by a key mammalian aging protein.treating messenger {rna} transcript abundances as quantitative traits and mapping gene expression quantitative trait loci for these traits has been pursued in gene-specific ways. transcript abundances often serve as a surrogate for classical quantitative traits in that the levels of expression are significantly correlated with the classical traits across members of a segregating population. the correlation structure between transcript abundances and classical traits has been used to identify susceptibility loci for complex diseases such as diabetes and allergic asthma. one study recently completed the first comprehensive dissection of transcriptional regulation in budding yeast, giving a detailed glimpse of a genome-wide survey of the genetics of gene expression. unlike classical quantitative traits, which often represent gross clinical measurements that may be far removed from the biological processes giving rise to them, the genetic linkages associated with transcript abundance affords a closer look at cellular biochemical processes. here we describe comprehensive genetic screens of mouse, plant and human transcriptomes by considering gene expression values as quantitative traits. we identify a gene expression pattern strongly associated with obesity in a murine cross, and observe two distinct obesity subtypes. furthermore, we find that these obesity subtypes are under the control of different loci.time-series analysis of whole-genome expression data during drosophila melanogaster development indicates that up to 86\\\\% of its genes change their relative transcript level during embryogenesis. by applying conservative filtering criteria and requiring \\'sharp\\' transcript changes, we identified 1534 maternal genes, 792 transient zygotic genes, and 1053 genes whose transcript levels increase during embryogenesis. each of these three categories is dominated by groups of genes where all transcript levels increase and/or decrease at similar times, suggesting a common mode of regulation. for example, 34\\\\% of the transiently expressed genes fall into three groups, with increased transcript levels between 2.5–12, 11–20, and 15–20 h of development, respectively. we highlight common and distinctive functional features of these expression groups and identify a coupling between downregulation of transcript levels and targeted protein degradation. by mapping the groups to the protein network, we also predict and experimentally confirm new functional associations.the intestinal microflora, typically equated with bacteria, influences diseases such as obesity and inflammatory bowel disease. here, we show that the mammalian gut contains a rich fungal community that interacts with the immune system through the innate immune receptor dectin-1. mice lacking dectin-1 exhibited increased susceptibility to chemically induced colitis, which was the result of altered responses to indigenous fungi. in humans, we identified a polymorphism in the gene for dectin-1 ({clec7a}) that is strongly linked to a severe form of ulcerative colitis. together, our findings reveal a eukaryotic fungal community in the gut (the  ” mycobiome”) that coexists with bacteria and substantially expands the repertoire of organisms interacting with the intestinal immune system to influence health and disease.a key problem in understanding transcriptional regulatory networks is deciphering what cis regulatory logic is encoded in gene promoter sequences and how this sequence information maps to expression. a typical computational approach to this problem involves clustering genes by their expression profiles and then searching for overrepresented motifs in the promoter sequences of genes in a cluster. however, genes with similar expression profiles may be controlled by distinct regulatory programs. moreover, if many gene expression profiles in a data set are highly correlated, as in the case of whole organism developmental time series, it may be difficult to resolve fine-grained clusters in the first place. we present a predictive framework for modeling the natural flow of information, from promoter sequence to expression, to learn cis regulatory motifs and characterize gene expression patterns in developmental time courses. we introduce a cluster-free algorithm based on a graph-regularized version of partial least squares ({pls}) regression to learn sequence patterns\\\\^{a}\\x80\\x94represented by graphs of k-mers, or \\\\^{a}\\x80\\x9cgraph-mers\\\\^{a}\\x80\\x9d\\\\^{a}\\x80\\x94that predict gene expression trajectories. applying the approach to wildtype germline development in caenorhabditis elegans, we found that the first and second latent {pls} factors mapped to expression profiles for oocyte and sperm genes, respectively. we extracted both known and novel motifs from the graph-mers associated to these germline-specific patterns, including novel {cg}-rich motifs specific to oocyte genes. we found evidence supporting the functional relevance of these putative regulatory elements through analysis of positional bias, motif conservation and in situ gene expression. this study demonstrates that our regression model can learn biologically meaningful latent structure and identify potentially functional motifs from subtle developmental time course expression data. a major challenge in functional genomics is to decipher the gene regulatory networks operating in multi-cellular organisms, such as the nematode c. elegans. the expression level of a gene is controlled, to a great extent, by regulatory proteins called transcription factors that bind short motifs in the gene\\'s promoter (regulatory region in the non-coding {dna}). in a temporal regulatory process, for example in development, the \\\\^{a}\\x80\\x9cregulatory logic\\\\^{a}\\x80\\x9d of {dna} motifs in the promoter largely determines the gene\\'s expression trajectory, as the gene responds over time to changing concentrations of the transcription factors that control it. this study addresses the problem of learning {dna} motifs that predict temporal expression profiles, using genomewide expression data from developmental time series in c. elegans. we developed a novel algorithm based on techniques from multivariate regression that sets up a correspondence between sequence patterns and expression trajectories. sequence motifs are represented as graphs of sequence-similar k-length subsequences called \\\\^{a}\\x80\\x9cgraph-mers\\\\^{a}\\x80\\x9d. by applying the method to germline development in c. elegans, we found both known and novel {dna} motifs associated with oocyte and sperm genes.a biweekly scientific journal publishing high-quality research in molecular biology and genetics, cancer biology, biochemistry, and related fieldsthe falling cost of genome sequencing is having a marked impact on the research community with respect to which genomes are sequenced and how and where they are annotated. genome annotation projects have generally become small-scale affairs that are often carried out by an individual laboratory. although annotating a eukaryotic genome assembly is now within the reach of non-experts, it remains a challenging task. here we provide an overview of the genome annotation process and the available tools and describe some best-practice approaches.the efficiency of parameter estimation using a pseudolikelihood technique is assessed for some simple spatial gaussian schemes.transposable elements ({tes}) are abundant in the human genome, and some are capable of generating new insertions through {rna} intermediates. in cancer, the disruption of cellular mechanisms that normally suppress {te} activity may facilitate mutagenic retrotranspositions. we performed single-nucleotide resolution analysis of {te} insertions in 43 high-coverage whole-genome sequencing data sets from five cancer types. we identified 194 high-confidence somatic {te} insertions, as well as thousands of polymorphic {te} insertions in matched normal genomes. somatic insertions were present in epithelial tumors but not in blood or brain cancers. somatic l1 insertions tend to occur in genes that are commonly mutated in cancer, disrupt the expression of the target genes, and are biased toward regions of cancer-specific {dna} hypomethylation, highlighting their potential impact in tumorigenesis.gene duplication provides raw material for functional innovation. recent advances have shed light on two fundamental questions regarding gene duplication: which genes tend to undergo duplication? and how does natural selection subsequently act on them? genomic data suggest that different gene classes tend to be retained after single-gene and whole-genome duplications. we also know that functional differences between duplicate genes can originate in several different ways, including mutations that directly impart new functions, subdivision of ancestral functions and selection for changes in gene dosage. interestingly, in many cases the \\'new\\' function of one copy is a secondary property that was always present, but that has been co-opted to a primary role after the duplication.we now have a wealth of information about the molecular signals that act on cells in embryos, but how do the control systems based on these signals generate pattern and govern the timing of developmental events? here, i discuss four examples to show how mathematical modeling and quantitative experimentation can give some useful answers. the examples concern the bicoid gradient in the early drosophila embryo, the dorsoventral patterning of a frog embryo by bone morphogenetic protein signals, the auxin-mediated patterning of plant meristems, and the notch-dependent somite segmentation clock.protein complexes are dynamic. a new analysis of two quantitative proteomic datasets reveals cell type-specific changes in the stoichiometry of complexes, which often involve paralog {switching.please} see related research article: www.dx.doi.org/10.1186/s13059-016-0912-5.despite considerable genetic heterogeneity underlying neurodevelopmental diseases, there is compelling evidence that many disease genes will map to a much smaller number of biological subnetworks. we developed a computational method, termed {magi} (merging affected genes into integrated networks), that simultaneously integrates protein-protein interactions and {rna}-seq expression profiles during brain development to discover \"modules\" enriched for de novo mutations in probands. we applied this method to recent exome sequencing of 1116 patients with autism and intellectual disability, discovering two distinct modules that differ in their properties and associated phenotypes. the first module consists of 80 genes associated with wnt, notch, {swi}/{snf}, and {ncor} complexes and shows the highest expression early during embryonic development (8-16 post-conception weeks [pcw]). the second module consists of 24 genes associated with synaptic function, including long-term potentiation and calcium signaling with higher levels of postnatal expression. patients with de novo mutations in these modules are more significantly intellectually impaired and carry more severe missense mutations when compared to probands with de novo mutations outside of these modules. we used our approach to define subsets of the network associated with higher functioning autism as well as greater severity with respect to {iq}. finally, we applied {magi} independently to epilepsy and schizophrenia exome sequencing cohorts and found significant overlap as well as expansion of these modules, suggesting a core set of integrated neurodevelopmental networks common to seemingly diverse human diseases. {\\\\copyright} 2015 hormozdiari et al.; published by cold spring harbor laboratory press.the graph orientation problem calls for orienting the edges of a graph so as to maximize the number of pre-specified source–target vertex pairs that admit a directed path from the source to the target. most algorithmic approaches to this problem share a common preprocessing step, in which the input graph is reduced to a tree by repeatedly contracting its cycles. although this reduction is valid from an algorithmic perspective, the assignment of directions to the edges of the contracted cycles becomes arbitrary, and the connecting source–target paths may be arbitrarily long. in the context of biological networks, the connection of vertex pairs via shortest paths is highly motivated, leading to the following problem variant: given a graph and a collection of source–target vertex pairs, assign directions to the edges so as to maximize the number of pairs that are connected by a shortest (in the original graph) directed path. this problem is {np}-complete and hard to approximate to within sub-polynomial factors. here we provide a first polynomial-size integer linear program formulation for this problem, which allows its exact solution in seconds on current networks. we apply our algorithm to orient protein–protein interaction networks in yeast and compare it with two state-of-the-art algorithms. we find that our algorithm outperforms previous approaches and can orient considerable parts of the network, thus revealing its structure and function.we previously reported the analysis of genome-wide expression profiles and various diabetes-related traits in a segregating cross between inbred mouse strains {c57bl}/{6j} (b6) and {dba}/{2j} ({dba}). by considering transcript levels as quantitative traits, we identified several thousand expression quantitative trait loci ({eqtl}) with {lod} score >4.3. we now experimentally address the problem of multiple comparisons by estimating the fraction of false-positive {eqtl} that are under cis-acting regulation. for this, we have utilized a classic cis–trans test with (b6 × {dba})f1 mice to determine the relative levels of transcripts from the b6 and {dba} alleles. the results suggest that at least 64\\\\% of cis-acting {eqtl} with {lod} >4.3 are true positives, while the remaining 36\\\\% could not be confirmed as truly cis-acting. moreover, we find that >96\\\\% of apparent cis-acting {eqtl} occur in regions that do not share {snp} haplotypes. cis-acting {eqtl} serve as an important new resource for the identification of positional candidates in {qtl} studies in mice. also, we use the analysis of the correlation structures between genotypes, gene expression traits, and phenotypic traits to further characterize genes expressed in liver that are under cis-acting control, and highlight the advantages and disadvantages of integrating genetics and gene expression data in segregating populations.10.1093/bioinformatics/bti794 summary: bioinformatics studies of transcriptional regulation in the metazoa are significantly hindered by the absence of readily available data on large numbers of transcriptional cis-regulatory modules ({crms}). even the richly annotated drosophila melanogaster genome lacks extensive {crm} information. we therefore present here a database of drosophila {crms} curated from the literature complete with both {dna} sequence and a searchable description of the gene expression pattern regulated by each {crm}. this resource should greatly facilitate the development of computational approaches to {crm} discovery as well as bioinformatics analyses of regulatory sequence properties and {evolution.availability}:{http://redfly.ccr.buffalo.educontact}:mshalfon@buffalo.eduone of the obstacles hindering a better understanding of cancer is its heterogeneity. however, computational approaches to model cancer heterogeneity have lagged behind. to bridge this gap, we have developed a new probabilistic approach that models individual cancer cases as mixtures of subtypes. our approach can be seen as a meta-model that summarizes the results of a large number of alternative models. it does not assume predefined subtypes nor does it assume that such subtypes have to be sharply defined. instead given a measure of phenotypic similarity between patients and a list of potential explanatory features, such as mutations, copy number variation, {microrna} levels, etc., it explains phenotypic similarities with the help of these features. we applied our approach to glioblastoma multiforme ({gbm}). the resulting model {prob\\\\_gbm}, not only correctly inferred known relationships but also identified new properties underlining phenotypic similarities. the proposed probabilistic framework can be applied to model relations between similarity of gene expression and a broad spectrum of potential genetic causes.genome-scale metabolic network models can be reconstructed for well-characterized organisms using genomic annotation and literature information. however, there are many instances in which model predictions of metabolic fluxes are not entirely consistent with experimental data, indicating that the reactions in the model do not match the active reactions in the in vivo system. we introduce a method for determining the active reactions in a genome-scale metabolic network based on a limited number of experimentally measured fluxes. this method, called optimal metabolic network identification ({omni}), allows efficient identification of the set of reactions that results in the best agreement between in silico predicted and experimentally measured flux distributions. we applied the method to intracellular flux data for evolved escherichia coli mutant strains with lower than predicted growth rates in order to identify reactions that act as flux bottlenecks in these strains. the expression of the genes corresponding to these bottleneck reactions was often found to be downregulated in the evolved strains relative to the wild-type strain. we also demonstrate the ability of the {omni} method to diagnose problems in e. coli strains engineered for metabolite overproduction that have not reached their predicted production potential. the {omni} method applied to flux data for evolved strains can be used to provide insights into mechanisms that limit the ability of microbial strains to evolve towards their predicted optimal growth phenotypes. when applied to industrial production strains, the {omni} method can also be used to suggest metabolic engineering strategies to improve byproduct secretion. in addition to these applications, the method should prove to be useful in general for reconstructing metabolic networks of ill-characterized microbial organisms based on limited amounts of experimental data.various efforts have been made to elucidate the cooperating proteins involved in maintaining chromatin interactions; however, many are still unknown. here, we present {3cpet}, a tool based on a non-parametric bayesian approach, to infer the set of the most probable protein complexes involved in maintaining chromatin interactions and the regions that they may control, making it a valuable downstream analysis tool in chromatin conformation studies. {3cpet} does so by combining data from {chia}-{pet}, transcription factor binding sites, and protein interactions. {3cpet} results show biologically significant and accurate predictions when validated against experimental and simulation data.a key challenge of functional genomics today is to generate well-annotated data sets that can be interpreted across different platforms and technologies. large-scale functional genomics data often fail to connect to standard experimental approaches of gene characterization in individual laboratories. furthermore, a lack of universal annotation standards for phenotypic data sets makes it difficult to compare different screening approaches. here we address this problem in a screen designed to identify all genes required for the first two rounds of cell division in the caenorhabditis elegans embryo. we used {rna}-mediated interference to target 98\\\\% of all genes predicted in the c. elegans genome in combination with differential interference contrast time-lapse microscopy. through systematic annotation of the resulting movies, we developed a phenotypic profiling system, which shows high correlation with cellular processes and biochemical pathways, thus enabling us to predict new functions for previously uncharacterized genes.network models are a fundamental tool for the visualization and analysis of molecular interactions occurring in biological systems. while broadly illuminating the molecular machinery of the cell, graphical representations of protein interaction networks mask complex patterns of interaction that depend on temporal, spatial, or condition-specific contexts. in this paper, we introduce a novel graph construct called a biological context network that explicitly captures these changing patterns of interaction from one biological context to another. we consider known gene ontology biological process and cellular component annotations as a proxy for context, and show that aggregating small process-specific protein interaction sub-networks leads to the emergence of observed scale-free properties. the biological context model also provides the basis for characterizing proteins in terms of several context-specific measures, including \\'interactive promiscuity,\\' which identifies proteins whose interacting partners vary from one context to another. we show that such context-sensitive measures are significantly better predictors of knockout lethality than node degree, reaching better than 70\\\\% accuracy among the top scoring proteins.despite advances in high-throughput methods for discovering protein-protein interactions, the interaction networks of even well-studied model organisms are sketchy at best, highlighting the continued need for computational methods to help direct experimentalists in the search for novel interactions. we present a kernel method for predicting protein-protein interactions using a combination of data sources, including protein sequences, gene ontology annotations, local properties of the network, and homologous interactions in other species. whereas protein kernels proposed in the literature provide a similarity between single proteins, prediction of interactions requires a kernel between pairs of proteins. we propose a pairwise kernel that converts a kernel between single proteins into a kernel between pairs of proteins, and we illustrate the kernel\\'s effectiveness in conjunction with a support vector machine classifier. furthermore, we obtain improved performance by combining several sequence-based kernels based on k-mer frequency, motif and domain content and by further augmenting the pairwise sequence kernel with features that are based on other sources of data. we apply our method to predict physical interactions in yeast using data from the {bind} database. at a false positive rate of 1\\\\% the classifier retrieves close to 80\\\\% of a set of trusted interactions. we thus demonstrate the ability of our method to make accurate predictions despite the sizeable fraction of false positives that are known to exist in interaction databases. the classification experiments were performed using {pyml} available at http://pyml.sourceforge.net. data are available at: http://noble.gs.washington.edu/proj/sppi.much of the inter-individual variation in gene expression is triggered via perturbations of signaling networks by {dna} variants. we present a novel probabilistic approach for identifying the particular pathways by which {dna} variants perturb the signaling network. our procedure, called {pine}, relies on a systematic integration of established biological knowledge of signaling networks with data on transcriptional responses to various experimental conditions. unlike previous approaches, {pine} provides statistical aspects that are critical for prioritizing hypotheses for followup experiments. using simulated data, we show that higher accuracy is attained with {pine} than with existing methods. we used {pine} to analyze transcriptional responses of immune dendritic cells to several pathogenic stimulations. {pine} identified statistically significant genetic perturbations in the pathogen-sensing signaling network, suggesting previously uncharacterized regulatory mechanisms for functional {dna} variants. {\\\\copyright} the author(s) 2015. published by oxford university press on behalf of nucleic acids research.the hierarchical levels of genome architecture exert transcriptional control by tuning the accessibility and proximity of genes and regulatory elements. here, we review current insights into the trans-acting factors that enable the genome to flexibly adopt different functionally relevant conformations.genome wide maps of nucleosome occupancy in yeast have recently been produced through deep sequencing of nuclease-protected {dna}. these maps have been obtained from both crosslinked and uncrosslinked chromatin in vivo, and from chromatin assembled from genomic {dna} and nucleosomes in vitro. here, we analyze these maps in combination with existing {chip}-chip data, and with new {chip}-{qpcr} experiments reported here. we show that the apparent nucleosome density in crosslinked chromatin, when compared to uncrosslinked chromatin, is preferentially increased at transcription factor ({tf}) binding sites, suggesting a strategy for mapping generic transcription factor binding sites that would not require immunoprecipitation of a particular factor. we also confirm previous conclusions that the intrinsic, sequence dependent binding of nucleosomes helps determine the localization of {tf} binding sites. however, we find that the association between low nucleosome occupancy and {tf} binding is typically greater if occupancy at a site is averaged over a 600bp window, rather than using the occupancy at the binding site itself. we have also incorporated intrinsic nucleosome binding occupancies as weights in a computational model for {tf} binding, and by this measure as well we find better prediction if the high resolution nucleosome occupancy data is averaged over 600bp. we suggest that the intrinsic {dna} binding specificity of nucleosomes plays a role in {tf} binding site selection not so much through the specification of precise nucleosome positions that permit or occlude binding, but rather through the creation of low occupancy regions that can accommodate competition from {tfs} through rearrangement of nucleosomes.summary.\\u2002 we propose covariance-regularized regression, a family of methods for prediction in high dimensional settings that uses a shrunken estimate of the inverse covariance matrix of the features to achieve superior prediction. an estimate of the inverse covariance matrix is obtained by maximizing the log-likelihood of the data, under a multivariate normal model, subject to a penalty; it is then used to estimate coefficients for the regression of the response onto the features. we show that ridge regression, the lasso and the elastic net are special cases of covariance-regularized regression, and we demonstrate that certain previously unexplored forms of covariance-regularized regression can outperform existing methods in a range of situations. the covariance-regularized regression framework is extended to generalized linear models and linear discriminant analysis, and is used to analyse gene expression data sets with multiple class and survival outcomes.attribution of biological robustness to the specific structural properties of a regulatory network is an important yet unsolved problem in systems biology. it is widely believed that the topological characteristics of a biological control network largely determine its dynamic behavior, yet the actual mechanism is still poorly understood. here, we define a novel structural feature of biological networks, termed \\'regulation entropy\\', to quantitatively assess the influence of network topology on the robustness of the systems. using the cell-cycle control networks of the budding yeast (saccharomyces cerevisiae) and the fission yeast (schizosaccharomyces pombe) as examples, we first demonstrate the correlation of this quantity with the dynamic stability of biological control networks, and then we establish a significant association between this quantity and the structural stability of the networks. and we further substantiate the generality of this approach with a broad spectrum of biological and random networks. we conclude that the regulation entropy is an effective order parameter in evaluating the robustness of biological control networks. our work suggests a novel connection between the topological feature and the dynamic property of biological regulatory networks.peptide recognition modules mediate many protein-protein interactions critical for the assembly of macromolecular complexes. complete genome sequences have revealed thousands of these domains, requiring improved methods for identifying their physiologically relevant binding partners. we have developed a strategy combining computational prediction of interactions from phage-display ligand consensus sequences with large-scale two-hybrid physical interaction tests. application to yeast {sh3} domains generated a phage-display network containing 394 interactions among 206 proteins and a two-hybrid network containing 233 interactions among 145 proteins. graph theoretic analysis identified 59 highly likely interactions common to both networks. las17 (bee1), a member of the {wiskott-aldrich} syndrome protein ({wasp}) family of actin-assembly proteins, showed multiple {sh3} interactions, many of which were confirmed in vivo by coimmunoprecipitation.trained immunity refers to the memory characteristics of the innate immune system. memory traits of innate immunity have been reported in plants and invertebrates, as well as in mice lacking functional t and b cells that are protected against secondary infections after exposure to certain infections or vaccinations. the underlying mechanism of trained immunity is represented by epigenetic programming through histone modifications, leading to stronger gene transcription upon restimulation. however, the specific cellular processes that mediate trained immunity in monocytes or macrophages are poorly understood.º genome-wide {rna}, promoter, and distal chromatin changes dissect t\\xa0cell commitment º multipotency to commitment: alternative fates excluded by distinct repression modes º dynamic and reversible epigenetic marking responds to transcriptional regulators º distinct regimes of binding by dose and epigenetic interaction for {pu}.1 and {gata}-3 examining the epigenetic changes that define the stepwise maturation of t cells from multipotent precursors reveals the complexity in the mechanisms that silence genes used in alternative lineages and shows that, although permissive histone marks anticipate changes in transcription, repressive marks often lag behind.mapping {dnase} i hypersensitive ({hs}) sites is an accurate method of identifying the location of genetic regulatory elements, including promoters, enhancers, silencers, insulators, and locus control regions. we employed high-throughput sequencing and whole-genome tiled array strategies to identify {dnase} i {hs} sites within human primary {cd4}+ t cells. combining these two technologies, we have created a comprehensive and accurate genome-wide open chromatin map. surprisingly, only 16\\\\%-21\\\\% of the identified 94,925 {dnase} i {hs} sites are found in promoters or first exons of known genes, but nearly half of the most open sites are in these regions. in conjunction with expression, motif, and chromatin immunoprecipitation data, we find evidence of cell-type-specific characteristics, including the ability to identify transcription start sites and locations of different chromatin marks utilized in these cells. in addition, and unexpectedly, our analyses have uncovered detailed features of nucleosome structure.the average individual is expected to harbor thousands of variants within non-coding genomic regions involved in gene regulation. however, it is currently not possible to interpret reliably the functional consequences of genetic variation within any given transcription factor recognition sequence. to address this, we comprehensively analyzed heritable genome-wide binding patterns of a major sequence-specific regulator ({ctcf}) in relation to genetic variability in binding site sequences across a multi-generational pedigree. we localized and quantified {ctcf} occupancy by {chip}-seq in 12 related and unrelated individuals spanning three generations, followed by comprehensive targeted resequencing of the entire {ctcf}–binding landscape across all individuals. we identified hundreds of variants with reproducible quantitative effects on {ctcf} occupancy (both positive and negative). while these effects paralleled {protein–dna} recognition energetics when averaged, they were extensively buffered by striking local context dependencies. in the significant majority of cases buffering was complete, resulting in silent variants spanning every position within the {dna} recognition interface irrespective of level of binding energy or evolutionary constraint. the prevalence of complex partial or complete buffering effects severely constrained the ability to predict reliably the impact of variation within any given binding site instance. surprisingly, 40\\\\% of variants that increased {ctcf} occupancy occurred at positions of human–chimp divergence, challenging the expectation that the vast majority of functional regulatory variants should be deleterious. our results suggest that, even in the presence of  ” perfect” genetic information afforded by resequencing and parallel studies in multiple related individuals, genomic site-specific prediction of the consequences of individual variation in regulatory {dna} will require systematic coupling with empirical functional genomic measurements. a comprehensive understanding of the contribution of individual genome sequences to disease and quantitative traits will require the general ability to predict consequences of genetic variation in non-protein-coding regions, particularly those involved in gene regulation. here we tested the power to predict such consequences when presented with  ” complete” information encompassing the genomic {dna} binding site patterns of a well-studied regulatory protein across multiple related individuals, coupled with all individual genome sequences at the binding positions. we find that, while there is reasonable ability to predict the average effects of variation within the consensus recognition sequence of a transcriptional regulator, it is not possible to determine reliably the consequences of variation at any given genomic instance. this suggests that the interpretation of individual genome sequences will require comprehensive complementation with functional genomic studies.understanding the molecular basis for phenotypic differences between humans and other primates remains an outstanding challenge. mutations in non-coding regulatory {dna} that alter gene expression have been hypothesized as a key driver of these phenotypic differences. this has been supported by differential gene expression analyses in general, but not by the identification of specific regulatory elements responsible for changes in transcription and phenotype. to identify the genetic source of regulatory differences, we mapped {dnasei} hypersensitive ({dhs}) sites, which mark all types of active gene regulatory elements, genome-wide in the same cell type isolated from human, chimpanzee, and macaque. most {dhs} sites were conserved among all three species, as expected based on their central role in regulating transcription. however, we found evidence that several hundred {dhs} sites were gained or lost on the lineages leading to modern human and chimpanzee. species-specific {dhs} site gains are enriched near differentially expressed genes, are positively correlated with increased transcription, show evidence of branch-specific positive selection, and overlap with active chromatin marks. species-specific sequence differences in transcription factor motifs found within these {dhs} sites are linked with species-specific changes in chromatin accessibility. together, these indicate that the regulatory elements identified here are genetic contributors to transcriptional and phenotypic differences among primate species. the human genome shares a remarkable amount of genomic sequence with our closest living primate relatives. researchers have long sought to understand what regions of the genome are responsible for unique species-specific traits. previous studies have shown that many genes are differentially expressed between species, but the regulatory elements contributing to these differences are largely unknown. here we report a genome-wide comparison of active gene regulatory elements in human, chimpanzee, and macaque, and we identify hundreds of regulatory elements that have been gained or lost in the human or chimpanzee genomes since their evolutionary divergence. these elements contain evidence of natural selection and correlate with species-specific changes in gene expression. polymorphic {dna} bases in transcription factor motifs that we found in these regulatory elements may be responsible for the varied biological functions across species. this study directly links phenotypic and transcriptional differences between species with changes in chromatin structure.both solid tumours and leukaemias show considerable histological and functional heterogeneity. it is widely accepted that genetic lesions have a major role in determining tumour phenotype, but evidence is also accumulating that cancers of distinct subtypes within an organ may derive from different \\'cells of origin\\'. these cells acquire the first genetic hit or hits that culminate in the initiation of cancer. the identification of these crucial target cell populations may allow earlier detection of malignancies and better prediction of tumour behaviour, and ultimately may lead to preventive therapies for individuals at high risk of developing cancer.regulatory regions with similar transcriptional output often have little overt sequence similarity, both within and between genomes. although cis- and trans-regulatory changes can contribute to sequence divergence without dramatically altering gene expression outputs, heterologous {dna} often functions similarly in organisms that share little regulatory sequence similarities (e.g. human {dna} in fish), indicating that trans-regulatory mechanisms tend to diverge more slowly and can accommodate a variety of cis-regulatory configurations. this capacity to \\'tinker\\' with regulatory {dna} probably relates to the complexity, robustness and evolvability of regulatory systems, but cause-and-effect relationships among evolutionary processes and properties of regulatory systems remain a topic of debate. the challenge of understanding the concrete mechanisms underlying cis-regulatory evolution – including the conservation of function without the conservation of sequence – relates to the challenge of understanding the function of regulatory systems in general. currently, we are largely unable to recognize functionally similar regulatory {dna}.proper regulation of protein levels is essential for health, and abnormal levels of proteins are hallmarks of many diseases. a number of studies have recently shown that messenger {rna} levels vary among individuals of a species and that genetic linkage analysis can be used to identify quantitative trait loci that influence these levels. by contrast, little is known about the genetic basis of variation in protein levels in genetically diverse populations, in large part because techniques for large-scale measurements of protein abundance lag far behind those for measuring transcript abundance. here we describe a label-free, mass spectrometry-based approach to measuring protein levels in total unfractionated cellular proteins, and we apply this approach to elucidate the genetic basis of variation in protein abundance in a cross between two diverse strains of yeast. loci that influenced protein abundance differed from those that influenced transcript levels, emphasizing the importance of direct analysis of the proteome.we describe an open-source {kpal} package that facilitates an alignment-free assessment of the quality and comparability of sequencing datasets by analyzing k-mer frequencies. we show that {kpal} can detect technical artefacts such as high duplication rates, library chimeras, contamination and differences in library preparation protocols. {kpal} also successfully captures the complexity and diversity of microbiomes and provides a powerful means to study changes in microbial communities. together, these features make {kpal} an attractive and broadly applicable tool to determine the quality and comparability of sequence libraries even in the absence of a reference sequence. {kpal} is freely available at {https://github.com/lumc}/{kpal}.systems biology attempts to describe complex systems behaviors in terms of dynamic operations of biological networks. however, there is lack of tools that can effectively decode complex network dynamics over multiple conditions. we present principal network analysis ({pna}) that can automatically capture major dynamic activation patterns over multiple conditions and then generate protein and metabolic subnetworks for the captured patterns. we first demonstrated the utility of this method by applying it to a synthetic dataset. the results showed that {pna} correctly captured the subnetworks representing dynamics in the data. we further applied {pna} to two time-course gene expression profiles collected from (i) {mcf7} cells after treatments of {hrg} at multiple doses and (ii) brain samples of four strains of mice infected with two prion strains. the resulting subnetworks and their interactions revealed network dynamics associated with {hrg} dose-dependent regulation of cell proliferation and differentiation and early {prpsc} accumulation during prion infection. the web-based software is available at: http://sbm.postech.ac.kr/pna.motivation: multi-series time-course microarray experiments are useful approaches for exploring biological processes. in this type of experiments, the researcher is frequently interested in studying gene expression changes along time and in evaluating trend differences between the various experimental groups. the large amount of data, multiplicity of experimental conditions and the dynamic nature of the experiments poses great challenges to data analysis.this paper reviews recent computational approaches to the understanding of gene regulation in eukaryotes. cis-regulation of gene expression by the binding of transcription factors is a critical component of cellular physiology. in eukaryotes, a number of transcription factors often work together in a combinatorial fashion to enable cells to respond to a wide spectrum of environmental and developmental signals. integration of genome sequences and/or chromatin immunoprecipitation on chip data with gene-expression data has facilitated in silico discovery of how the combinatorics and positioning of transcription factors binding sites underlie gene activation in a variety of cellular {processes.the} process of gene regulation is extremely complex and intriguing, therefore all possible points of view and related links should be carefully considered. here we attempt to collect an inventory, not claiming it to be comprehensive and complete, of related computational biological topics covering gene regulation, which may en-lighten the process, and briefly review what is currently occurring in these {areas.we} will consider the following computational areas:o gene regulatory network construction;o evolution of regulatory {dna};o studies of its structural and statistical informational properties;o and finally, regulatory {rna}.antisense transcription (transcription from the opposite strand to a protein-coding or sense strand) has been ascribed roles in gene regulation involving degradation of the corresponding sense transcripts ({rna} interference), as well as gene silencing at the chromatin level. global transcriptome analysis provides evidence that a large proportion of the genome can produce transcripts from both strands, and that antisense transcripts commonly link neighboring  ” genes” in complex loci into chains of linked transcriptional units. expression profiling reveals frequent concordant regulation of sense/antisense pairs. we present experimental evidence that perturbation of an antisense {rna} can alter the expression of sense messenger {rnas}, suggesting that antisense transcription contributes to control of transcriptional outputs in mammals.we profiled 162 lines of arabidopsis for variation in transcript, protein and metabolite abundance using {mrna} microarrays, two-dimensional polyacrylamide gel electrophoresis, gas chromatography time-of-flight mass spectrometry, liquid chromatography quadrupole time-of-flight mass spectrometry, and proton nuclear magnetic resonance. we added all publicly available phenotypic data from the same lines and mapped quantitative trait loci ({qtl}) for 40,580 molecular and 139 phenotypic traits. we found six {qtl} hot spots with major, system-wide effects, suggesting there are six breakpoints in a system otherwise buffered against many of the 500,000 {snps}.motivation: living cells are the product of gene expression programs that involve the regulated transcription of thousands of genes. the elucidation of transcriptional regulatory networks is thus needed to understand the cell\\'s working mechanism, and can for example, be useful for the discovery of novel therapeutic targets. although several methods have been proposed to infer gene regulatory networks from gene expression data, a recent comparison on a large-scale benchmark experiment revealed that most current methods only predict a limited number of known regulations at a reasonable precision level.inferring the topology of a gene-regulatory network ({grn}) from genome-scale time-series measurements of transcriptional change has proved useful for disentangling complex biological processes. to address the challenges associated with this inference, a number of competing approaches have previously been used, including examples from information theory, bayesian and dynamic bayesian networks ({dbns}), and ordinary differential equation ({ode}) or stochastic differential equation. the performance of these competing approaches have previously been assessed using a variety of in silico and in vivo datasets. here, we revisit this work by assessing the performance of more recent network inference algorithms, including a novel non-parametric learning approach based upon nonlinear dynamical systems. for larger {grns}, containing hundreds of genes, these non-parametric approaches more accurately infer network structures than do traditional approaches, but at significant computational cost. for smaller systems, {dbns} are competitive with the non-parametric approaches with respect to computational time and accuracy, and both of these approaches appear to be more accurate than granger causality-based methods and those using simple {odes} models.recent advancements in molecular methods have made it possible to capture physical contacts between multiple chromatin fragments. the resulting association matrices provide a noisy estimate for average spatial proximity that can be used to gain insights into the genome organization inside the nucleus. however, extracting topological information from these data is challenging and their integration across resolutions is still poorly addressed. recent findings suggest that a hierarchical approach could be advantageous for addressing these challenges. we present an algorithmic framework, which is based on hierarchical block matrices ({hbms}), for topological analysis and integration of chromosome conformation capture ({3c}) data. we first describe {chromohbm}, an algorithm that compresses high-throughput {3c} ({hit}-{3c}) data into topological features that are efficiently summarized with an {hbm} representation. we suggest that instead of directly combining {hit}-{3c} datasets across resolutions, which is a difficult task, we can integrate their {hbm} representations, and describe {chromohbm}-{3c}, an algorithm which merges {hbms}. since three-dimensional ({3d}) reconstruction can also benefit from topological information, we further present {chromohbm}-{3d}, an algorithm which exploits the {hbm} representation in order to gradually introduce topological constraints to the reconstruction process. we evaluate our approach in light of previous image microscopy findings and epigenetic data, and show that it can relate multiple spatial scales and provide a more complete view of the {3d} genome architecture. the presented algorithms are available from: https://github.com/yolish/hbm ys388@cam.ac.uk or pl219@cam.ac.uk supplementary data are available at bioinformatics online. {\\\\copyright} the author 2015. published by oxford university press. all rights reserved. for permissions, please e-mail: journals.permissions@oup.com.mammalian interphase chromosomes interact with the nuclear lamina ({nl}) through hundreds of large lamina-associated domains ({lads}). we report a method to map {nl} contacts genome-wide in single human cells. analysis of nearly 400 maps reveals a core architecture consisting of gene-poor {lads} that contact the {nl} with high cell-to-cell consistency, interspersed by {lads} with more variable {nl} interactions. the variable contacts tend to be cell-type specific and are more sensitive to changes in genome ploidy than the consistent contacts. single-cell maps indicate that {nl} contacts involve multivalent interactions over hundreds of kilobases. moreover, we observe extensive intra-chromosomal coordination of {nl} contacts, even over tens of megabases. such coordinated loci exhibit preferential interactions as detected by {hi-c}. finally, the consistency of {nl} contacts is inversely linked to gene activity in single cells and correlates positively with the heterochromatic histone modification {h3k9me3}. these results highlight fundamental principles of single-cell chromatin organization.models derived from human pluripotent stem cells that accurately recapitulate neural development in vitro and allow for the generation of specific neuronal subtypes are of major interest to the stem cell and biomedical community. notch signalling, particularly through the notch effector {hes5}, is a major pathway critical for the onset and maintenance of neural progenitor cells in the embryonic and adult nervous system. here we report the transcriptional and epigenomic analysis of six consecutive neural progenitor cell stages derived from a {hes5}::{egfp} reporter human embryonic stem cell line. using this system, we aimed to model cell-fate decisions including specification, expansion and patterning during the ontogeny of cortical neural stem and progenitor cells. in order to dissect regulatory mechanisms that orchestrate the stage-specific differentiation process, we developed a computational framework to infer key regulators of each cell-state transition based on the progressive remodelling of the epigenetic landscape and then validated these through a pooled short hairpin {rna} screen. we were also able to refine our previous observations on epigenetic priming at transcription factor binding sites and suggest here that they are mediated by combinations of core and stage-specific factors. taken together, we demonstrate the utility of our system and outline a general framework, not limited to the context of the neural lineage, to dissect regulatory circuits of differentiation.{handling inherent uncertainty and exploiting compositional structure are fundamental to understanding and designing large-scale systems. statistical relational learning builds on ideas from probability theory and statistics to address uncertainty while incorporating tools from logic, databases, and programming languages to represent structure. in <i>introduction to statistical relational learning,</i> leading researchers in this emerging area of machine learning describe current formalisms, models, and algorithms that enable effective and robust reasoning about richly structured systems and data.<br /> <br /> the early chapters provide tutorials for material used in later chapters, offering introductions to representation, inference and learning in graphical models, and logic. the book then describes object-oriented approaches, including probabilistic relational models, relational markov networks, and probabilistic entity-relationship models as well as logic-based formalisms including bayesian logic programs, markov logic, and stochastic logic programs. later chapters discuss such topics as probabilistic models with unknown objects, relational dependency networks, reinforcement learning in relational domains, and information extraction.<br /> <br /> by presenting a variety of approaches, the book highlights commonalities and clarifies important differences among proposed approaches and, along the way, identifies important representational and algorithmic issues. numerous applications are provided throughout.}dendritic cells ({dc}) are professional antigen presenting cells that develop from hematopoietic stem cells through successive steps of lineage commitment and differentiation. multipotent progenitors ({mpp}) are committed to {dc} restricted common {dc} progenitors ({cdp}), which differentiate into specific {dc} subsets, classical {dc} ({cdc}) and plasmacytoid {dc} ({pdc}). to determine epigenetic states and regulatory circuitries during {dc} differentiation, we measured consecutive changes of genome-wide gene expression, histone modification and transcription factor occupancy during the sequel {mpp}-{cdp}-{cdc}/{pdc}. specific histone marks in {cdp} reveal a {dc}-primed epigenetic signature, which is maintained and reinforced during {dc} differentiation. epigenetic marks and transcription factor {pu}.1 occupancy increasingly coincide upon {dc} differentiation. by integrating {pu}.1 occupancy and gene expression we devised a transcription factor regulatory circuitry for {dc} commitment and subset specification. the circuitry provides the transcription factor hierarchy that drives the sequel {mpp}-{cdp}-{cdc}/{pdc}, including irf4, irf8, tcf4, spib and stat factors. the circuitry also includes feedback loops inferred for individual or multiple factors, which stabilize distinct stages of {dc} development and {dc} subsets. in summary, here we describe the basic regulatory circuitry of transcription factors that drives {dc} development.there has been a recent surge in the use of genome-wide methodologies to identify and annotate the transcriptional regulatory elements in the human genome. here we review some of these methodologies and the conceptual insights about transcription regulation that have been gained from the use of genome-wide studies. it has become clear that the binding of transcription factors is itself a highly regulated process, and binding does not always appear to have functional consequences. numerous properties have now been associated with regulatory elements that may be useful in their identification. several aspects of enhancer function have been shown to be more widespread than was previously appreciated, including the highly combinatorial nature of transcription factor binding, the postinitiation regulation of many target genes, and the binding of enhancers at early stages to maintain their competence during development. going forward, the integration of multiple genome-wide data sets should become a standard approach to elucidate higher-order regulatory interactions.mouse studies have been instrumental in forming our current understanding of early cell-lineage decisions; however, similar insights into the early human development are severely limited. here, we present a comprehensive transcriptional map of human embryo development, including the sequenced transcriptomes of 1,529 individual cells from 88 human preimplantation embryos. these data show that cells undergo an intermediate state of co-expression of lineage-specific genes, followed by a concurrent establishment of the trophectoderm, epiblast, and primitive endoderm lineages, which coincide with blastocyst formation. female cells of all three lineages achieve dosage compensation of x chromosome {rna} levels prior to implantation. however, in contrast to the mouse, {xist} is transcribed from both alleles throughout the progression of this expression dampening, and x chromosome genes maintain biallelic expression while dosage compensation proceeds. we envision broad utility of this transcriptional atlas in future studies on human development as well as in stem cell research.is evolution predictable at the molecular level? the ambitious goal to answer this question requires an understanding of the mutational effects that govern the complex relationship between genotype and phenotype. in practice, it involves integrating systems-biology modelling, microbial laboratory evolution experiments and large-scale mutational analyses — a feat that is made possible by the recent availability of the necessary computational tools and experimental techniques. this review investigates recent progresses in mapping evolutionary trajectories and discusses the degree to which these predictions are realistic.evolutionary rewiring of regulatory networks is an important source of diversity among species. previous evidence suggested substantial divergence of regulatory networks across species. however, systematically assessing the extent of this plasticity and its functional implications has been challenging due to limited experimental data and the noisy nature of computational predictions. here, we introduce a novel approach to study cis-regulatory evolution, and use it to trace the regulatory history of 88 {dna} motifs of transcription factors across 23 ascomycota fungi. while motifs are conserved, we find a pervasive gain and loss in the regulation of their target genes. despite this turnover, the biological processes associated with a motif are generally conserved. we explain these trends using a model with a strong selection to conserve the overall function of a transcription factor, and a much weaker selection over the specific genes it targets. the model also accounts for the turnover of bound targets measured experimentally across species in yeasts and mammals. thus, selective pressures on regulatory networks mostly tolerate local rewiring, and may allow for subtle fine-tuning of gene regulation during evolution.the early drosophila embryo provides a unique system for the analysis of transcriptional repression since a broad spectrum of repressors are distributed in spatially distinct patterns. kr\\\\\"{u}ppel (kr) and snail (sna), two zinc finger repressors, are essential for segmentation and for the establishment of the mesoderm/neuroectoderm boundary, respectively. both repressors were examined in the context of synthetic gene complexes containing modular promoters and divergently transcribed reporter genes. these studies indicate that kr and sna function as short-range repressors, which can mediate either quenching or direct repression of the transcription complex, depending on the location of repressor sites. when located within an upstream enhancer, the repressor locally quenches nearby activators and permits other enhancers to interact with the transcription complex (enhancer autonomy). in contrast, when bound to promoter-proximal regions the repressor functions in a dominant fashion and blocks multiple enhancers. local quenching and dominant repression require close linkage (<100 bp) of the repressor with either upstream activators or the transcription complex. these studies establish short-range repression as a flexible form of gene regulation and suggest that the key distinction among repressors is their range of action.genome architecture is associated with cellular identity, but how this organization changes during reprogramming is not well understood. now in cell stem cell, krijger et\\xa0al. (2016) and beagan et\\xa0al. (2016) report {3d} chromatin interaction maps before and after reprogramming, providing evidence for topological memory in induced pluripotent stem cells.regulation of gene expression by transcription factors ({tfs}) is highly dependent on genetic background and interactions with cofactors. identifying specific context factors is a major challenge that requires new approaches. here we show that exploiting natural variation is a potent strategy for probing functional interactions within gene regulatory networks. we developed an algorithm to identify genetic polymorphisms that modulate the regulatory connectivity between specific transcription factors and their target genes in vivo. as a proof of principle, we mapped connectivity quantitative trait loci ({cqtls}) using parallel genotype and gene expression data for segregants from a cross between two strains of the yeast saccharomyces cerevisiae. we identified a nonsynonymous mutation in the {dig2} gene as a {cqtl} for the transcription factor ste12p and confirmed this prediction empirically. we also identified three polymorphisms in {taf13} as putative modulators of regulation by gcn4p. our method has potential for revealing how genetic differences among individuals influence gene regulatory networks in any organism for which gene expression and genotype data are available along with information on binding preferences for transcription factors.motivation: transcriptional regulatory networks controlling cell fate decisions in mammalian embryonic development remain elusive despite a long time of research. the recent emergence of single-cell {rna} profiling technology raises hope for new discovery. although experimental works have obtained intriguing insights into the mouse early development, a holistic and systematic view is still missing. mathematical models of cell fates tend to be concept-based, not designed to learn from real data. to elucidate the regulatory mechanisms behind cell fate decisions, it is highly desirable to synthesize the data-driven and knowledge-driven modeling approaches.in order to understand even the simplest cellular processes, we need to integrate proteomic, gene expression and other biomolecular data. to date, most computational approaches aimed at integrating proteomics and gene expression data use direct gene/protein correlation measures. however, due to post-transcriptional and translational regulations, the correspondence between the expression of a gene and its protein is complicated. we apply a multivariate statistical method, co-inertia analysis ({cia}), to visualise gene and proteomic expression data stemming from the same biological samples. principal components analysis or correspondence analysis can be used for data exploration on single datasets. {cia} is then used to explore the relationships between two or more datasets. we further explore the data by projecting gene ontology ({go}) information onto these plots to describe the cellular processes in action. we apply these techniques to gene expression and protein abundance data from studies of the human malarial parasite life cycle and the {nci}-60 cancer cell lines. in each case, we visualise gene expression, protein abundance and {go} classes in the same low dimensional projections and identify {go} classes that are likely to be of biological importance.motivation: by collecting multiple samples per subject, researchers can characterize intra-subject variation using physiologically relevant measurements such as gene expression profiling. this can yield important insights into fundamental biological questions ranging from cell type identity to tumour development. for each subject, the data measurements can be written as a matrix with the different subsamples (e.g. multiple tissues) indexing the columns and the genes indexing the rows. in this context, neither the genes nor the tissues are expected to be independent and straightforward application of traditional statistical methods that ignore this two-way dependence might lead to erroneous conclusions. herein, we present a suite of tools embedded within the {r/bioconductor} package {hdtd} for robustly estimating and performing hypothesis tests about the mean relationship and the covariance structure within the rows and columns. we illustrate the utility of {hdtd} by applying it to analyze data generated by the {genotype-tissue} expression consortium.networks are ubiquitous in biology, and computational approaches have been largely investigated for their inference. in particular, supervised machine learning methods can be used to complete a partially known network by integrating various measurements. two main supervised frameworks have been proposed: the local approach, which trains a separate model for each network node, and the global approach, which trains a single model over pairs of nodes. here, we systematically investigate, theoretically and empirically, the exploitation of tree-based ensemble methods in the context of these two approaches for biological network inference. we first formalize the problem of network inference as a classification of pairs, unifying in the process homogeneous and bipartite graphs and discussing two main sampling schemes. we then present the global and the local approaches, extending the latter for the prediction of interactions between two unseen network nodes, and discuss their specializations to tree-based ensemble methods, highlighting their interpretability and drawing links with clustering techniques. extensive computational experiments are carried out with these methods on various biological networks that clearly highlight that these methods are competitive with existing methods.transcriptional regulation is the most committed type of regulation in living cells where transcription factors ({tfs}) control the expression of their target genes and {tf} expression is controlled by other {tfs} forming complex transcriptional regulatory networks that can be highly interconnected. here we analyze the topology and organization of nine transcriptional regulatory networks for e. coli, yeast, mouse and human, and we evaluate how the structure of these networks influences two of their key properties, namely controllability and stability. we calculate the controllability for each network as a measure of the organization and interconnectivity of the network. we find that the number of driver nodes {nd} needed to control the whole network is 64\\\\% of the {tfs} in the e. coli transcriptional regulatory network in contrast to only 17\\\\% for the yeast network, 4\\\\% for the mouse network and 8\\\\% for the human network. the high controllability (low number of drivers needed to control the system) in yeast, mouse and human is due to the presence of internal loops in their regulatory networks where the {tfs} regulate each other in a circular fashion. we refer to these internal loops as circular control motifs ({ccm}). the e. coli transcriptional regulatory network, which does not have any {ccms}, shows a hierarchical structure of the transcriptional regulatory network in contrast to the eukaryal networks. the presence of {ccms} also has influence on the stability of these networks, as the presence of cycles can be associated with potential unstable steady-states where even small changes in binding affinities can cause dramatic rearrangements of the state of the network.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsthe precise regulation of gene transcription during metazoan development is controlled by a complex system of interactions between transcription factors, histone modifications and modifying enzymes and chromatin conformation. developments in chromosome conformation capture technologies have revealed that interactions between regions of chromatin are pervasive and highly cell-type specific. the movement of enhancers and promoters in and out of higher-order chromatin structures within the nucleus are associated with changes in expression and histone modifications. however, the factors responsible for mediating these changes and determining enhancer:promoter specificity are still not completely known. in this review, we summarize what is known about the patterns of epigenetic and chromatin features characteristic of elements involved in long-range interactions. in addition, we review the insights into both local and global patterns of chromatin interactions that have been revealed by the latest experimental and computational methods.we present a method for jointly learning dynamic models of transcriptional regulatory networks from gene expression data and transcription factor binding location data. models are automatically learned using dynamic bayesian network inference algorithms; joint learning is accomplished by incorporating evidence from gene expression data through the likelihood, and from transcription factor binding location data through the prior. we propose a new informative structure prior with two advantages. first, the prior incorporates evidence from location data probabilistically, allowing it to be weighed against evidence from expression data. second, the prior takes on a factorable form that is computationally efficient when learning dynamic regulatory networks. results obtained from both simulated and experimental data from the yeast cell cycle demonstrate that this joint learning algorithm can recover dynamic regulatory networks from multiple types of data that are more accurate than those recovered from each type of data in isolation.competitive interactions between cells are the basis of many homeostatic processes in biology. some of the best-described cases of competition between cells occur in drosophila: cell competition, whereby somatic cells within a growing epithelium compete with one another for contribution to the adult, and stem cell competition, in which germline or somatic stem cells vie for residency in the niche. both types of competition are conserved physiological processes, with much to tell us about how cellular neighborhoods influence cell behavior, and have importance to stem cell biology, regeneration and transplantation, and cancer. 10.1126/science.1163862regulatory and signaling networks coordinate the enormously complex interactions and processes that control cellular processes (such as metabolism and cell division), coordinate response to the environment, and carry out multiple cell decisions (such as development and quorum sensing). regulatory network inference is the process of inferring these networks, traditionally from microarray data but increasingly incorporating other measurement types such as proteomics, {chip}-seq, metabolomics, and mass cytometry. we discuss existing techniques for network inference. we review in detail our pipeline, which consists of an initial biclustering step, designed to estimate co-regulated groups; a network inference step, designed to select and parameterize likely regulatory models for the control of the co-regulated groups from the biclustering step; and a visualization and analysis step, designed to find and communicate key features of the network. learning biological networks from even the most complete data sets is challenging; we argue that integrating new data types into the inference pipeline produces networks of increased accuracy, validity, and biological relevance.forward genetics is a common approach to dissecting complex traits like common human diseases. the ultimate aim of this approach was the identification of genes that are causal for disease or other phenotypes of interest. however, the forward genetics approach is by definition restricted to the identification of genes that have incurred mutations over the course of evolution or that incurred mutations as a result of chemical mutagenesis, and that as a result lead to disease or to variations in other phenotypes of interest. genes that harbour no such mutations, but that play key roles in parts of the biological network that lead to disease, are systematically missed by this class of approaches. recently, a class of novel integrative genomics approaches has been devised to elucidate the complexity of common human diseases by intersecting genotypic, molecular profiling, and clinical data in segregating populations. these novel approaches take a more holistic view of biological systems and leverage the vast network of gene-gene interactions, in combination with {dna} variation data, to establish causal relationships among molecular profiling traits and between molecular profiling and disease (or other classic phenotypes). a number of novel genes for disease phenotypes have been identified as a result of these approaches, highlighting the utility of integrating orthogonal sources of data to get at the underlying causes of disease.a key aim of postgenomic biomedical research is to systematically catalogue all molecules and their interactions within a living cell. there is a clear need to understand how these molecules and the interactions between them determine the function of this enormously complex machinery, both in isolation and when surrounded by other cells. rapid advances in network biology indicate that cellular networks are governed by universal laws and offer a new conceptual framework that could potentially revolutionize our view of biology and disease pathologies in the twenty-first century.{background}:a myriad of methods to reverse-engineer transcriptional regulatory networks have been developed in recent years. direct methods directly reconstruct a network of pairwise regulatory interactions while module-based methods predict a set of regulators for modules of coexpressed genes treated as a single unit. to date, there has been no systematic comparison of the relative strengths and weaknesses of both types of {methods.results}:we have compared a recently developed module-based algorithm, {lemone} (learning module networks), to a mutual information based direct algorithm, {clr} (context likelihood of relatedness), using benchmark expression data and databases of known transcriptional regulatory interactions for escherichia coli and saccharomyces cerevisiae. a global comparison using recall versus precision curves hides the topologically distinct nature of the inferred networks and is not informative about the specific subtasks for which each method is most suited. analysis of the degree distributions and a regulator specific comparison show that {clr} is \\'regulator-centric\\', making true predictions for a higher number of regulators, while {lemone} is \\'target-centric\\', recovering a higher number of known targets for fewer regulators, with limited overlap in the predicted interactions between both methods. detailed biological examples in e. coli and s. cerevisiae are used to illustrate these differences and to prove that each method is able to infer parts of the network where the other fails. biological validation of the inferred networks cautions against over-interpreting recall and precision values computed using incomplete reference {networks.conclusion}:our results indicate that module-based and direct methods retrieve largely distinct parts of the underlying transcriptional regulatory networks. the choice of algorithm should therefore be based on the particular biological problem of interest and not on global metrics which cannot be transferred between organisms. the development of sound statistical methods for integrating the predictions of different reverse-engineering strategies emerges as an important challenge for future research.{rna}-seq is a recently developed approach to transcriptome profiling that uses deep-sequencing technologies. studies using this method have already altered our view of the extent and complexity of eukaryotic transcriptomes. {rna}-seq also provides a far more precise measurement of levels of transcripts and their isoforms than other methods. this article describes the {rna}-seq approach, the challenges associated with its application, and the advances made so far in characterizing several eukaryote transcriptomes.a fundamental challenge in human health is the identification of disease-causing genes. recently, several studies have tackled this challenge via a network-based approach, motivated by the observation that genes causing the same or similar diseases tend to lie close to one another in a network of protein-protein or functional interactions. however, most of these approaches use only local network information in the inference process and are restricted to inferring single gene associations. here, we provide a global, network-based method for prioritizing disease genes and inferring protein complex associations, which we call {prince}. the method is based on formulating constraints on the prioritization function that relate to its smoothness over the network and usage of prior information. we exploit this function to predict not only genes but also protein complex associations with a disease of interest. we test our method on gene-disease association data, evaluating both the prioritization achieved and the protein complexes inferred. we show that our method outperforms extant approaches in both tasks. using data on 1,369 diseases from the {omim} knowledgebase, our method is able (in a cross validation setting) to rank the true causal gene first for 34\\\\% of the diseases, and infer 139 disease-related complexes that are highly coherent in terms of the function, expression and conservation of their member proteins. importantly, we apply our method to study three multi-factorial diseases for which some causal genes have been found already: prostate cancer, alzheimer and type 2 diabetes mellitus. {prince}\\'s predictions for these diseases highly match the known literature, suggesting several novel causal genes and protein complexes for further investigation. understanding the genetic background of diseases is crucial to medical research, with implications in diagnosis, treatment and drug development. as molecular approaches to this challenge are time consuming and costly, computational approaches offer an efficient alternative. such approaches aim at prioritizing genes in a genomic interval of interest according to their predicted strength-of-association with a given disease. state-of-the-art prioritization problems are based on the observation that genes causing similar diseases tend to lie close to one another in a network of protein-protein interactions. here we develop a novel prioritization approach that uses the network data in a global manner and can tie not only single genes but also whole protein machineries with a given disease. our method, {prince}, is shown to outperform previous methods in both the gene prioritization task and the protein complex task. applying {prince} to prostate cancer, alzheimer\\'s disease and type 2 diabetes, we are able to infer new causal genes and related protein complexes with high confidence.{general state-space markov chain theory has evolved to make it both more  accessible and more powerful. markov chain monte carlo in practice  introduces mcmc methods and their applications while also providing some  theoretical background. considering the broad audience, the editors  emphasize practice rather than theory and keep the technical content to a  minimum. they offer step-by-step instructions for using the methods  presented and show the importance of mcmc in real applications with  examples ranging from the simple to the more complex in fields such as  archaeology, astronomy, biostatistics, genetics, epidemiology, and image  analysis.  }a number of supervised learning methods have been introduced in the last decade. unfortunately, the last comprehensive empirical evaluation of supervised learning was the statlog project in the early 90\\'s. we present a large-scale empirical comparison between ten supervised learning methods: {svms}, neural nets, logistic regression, naive bayes, memory-based learning, random forests, decision trees, bagged trees, boosted trees, and boosted stumps. we also examine the effect that calibrating the models via platt scaling and isotonic regression has on their performance. an important aspect of our study is the use of a variety of performance criteria to evaluate the learning methods.much of a cell\\'s activity is organized as a network of interacting modules: sets of genes coregulated to respond to different conditions. we present a probabilistic method for identifying regulatory modules from gene expression data. our procedure identifies modules of coregulated genes, their regulators and the conditions under which regulation occurs, generating testable hypotheses in the form \\'regulator x regulates module y under conditions w\\'. we applied the method to a saccharomyces cerevisiae expression data set, showing its ability to identify functionally coherent modules and their correct regulators. we present microarray experiments supporting three novel predictions, suggesting regulatory roles for previously uncharacterized proteins.many cellular components are present in such low numbers per cell that random births and deaths of individual molecules can cause substantial  ” noise” in concentrations. but biochemical events do not necessarily occur in single steps of individual molecules. some processes are greatly randomized when synthesis or degradation occurs in large bursts of many molecules during a short time interval. conversely, each birth or death of a macromolecule could involve several small steps, creating a memory between individual events. we present a generalized theory for stochastic gene expression, formulating the variance in protein abundance in terms of the randomness of the individual gene expression events. we show that common types of molecular mechanisms can produce gestation and senescence periods that reduce noise without requiring higher abundances, shorter lifetimes, or any concentration-dependent control loops. we also show that most single-cell experimental methods cannot distinguish between qualitatively different stochastic principles, although this in turn makes such methods better suited for identifying which components introduce fluctuations. characterizing the random events that give rise to noise in concentrations instead requires dynamic measurements with single-molecule resolution.understanding the genetic and molecular mechanisms governing the evolution of morphology is a major challenge in biology. because most animals share a conserved repertoire of body-building and -patterning genes, morphological diversity appears to evolve primarily through changes in the deployment of these genes during development. the complex expression patterns of developmentally regulated genes are typically controlled by numerous independent cis-regulatory elements ({cres}). it has been proposed that morphological evolution relies predominantly on changes in the architecture of gene regulatory networks and in particular on functional changes within {cres}. here, we discuss recent experimental studies that support this hypothesis and reveal some unanticipated features of how regulatory evolution occurs. from this growing body of evidence, we identify three key operating principles underlying regulatory evolution, that is, how regulatory evolution: (i) uses available genetic components in the form of preexisting and active transcription factors and {cres} to generate novelty; (ii) minimizes the penalty to overall fitness by introducing discrete changes in gene expression; and (iii) allows interactions to arise among any transcription factor and downstream {cre}. these principles endow regulatory evolution with a vast creative potential that accounts for both relatively modest morphological differences among closely related species and more profound anatomical divergences among groups at higher taxonomical levels.variation in transcriptional regulation is thought to be a major cause of phenotypic diversity1, 2. although widespread differences in gene expression among individuals of a species have been observed3, 4, 5, 6, 7, 8, studies to examine the variability of transcription factor binding on a global scale have not been performed, and thus the extent and underlying genetic basis of transcription factor binding diversity is unknown. by mapping differences in transcription factor binding among individuals, here we present the genetic basis of such variation on a genome-wide scale. whole-genome ste12-binding profiles were determined using chromatin immunoprecipitation coupled with {dna} sequencing in pheromone-treated cells of 43 segregants of a cross between two highly diverged yeast strains and their parental lines. we identified extensive ste12-binding variation among individuals, and mapped underlying cis- and trans-acting loci responsible for such variation. we showed that most transcription factor binding variation is cis-linked, and that many variations are associated with polymorphisms residing in the binding motifs of ste12 as well as those of several proposed ste12 cofactors. we also identified two trans-factors, {amn1} and {flo8}, that modulate ste12 binding to promoters of more than ten genes under α-factor treatment. neither of these two genes was previously known to regulate ste12, and we suggest that they may be mediators of gene activity and phenotypic diversity. ste12 binding strongly correlates with gene expression for more than 200 genes, indicating that binding variation is functional. many of the variable-bound genes are involved in cell wall organization and biogenesis. overall, these studies identified genetic regulators of molecular diversity among individuals and provide new insights into mechanisms of gene regulation.complex networks are studied across many fields of science. to uncover their structural design principles, we defined  ” network motifs,” patterns of interconnections occurring in complex networks at numbers that are significantly higher than those in randomized networks. we found such motifs in networks from biochemistry, neurobiology, ecology, and engineering. the motifs shared by ecological food webs were distinct from the motifs shared by the genetic networks of escherichia coli and saccharomyces cerevisiae or from those found in the world wide web. similar motifs were found in networks that perform information processing, even though they describe elements as different as biomolecules within a cell and synaptic connections between neurons in caenorhabditis elegans. motifs may thus define universal classes of networks. this approach may uncover the basic building blocks of most networks.we propose a simple approach to combining first-order logic and probabilistic graphical models in a single representation. a markov logic network ({mln}) is a first-order knowledge base with a weight attached to each formula (or clause). together with a set of constants representing objects in the domain, it specifies a ground markov network containing one feature for each possible grounding of a first-order formula in the {kb}, with the corresponding weight. inference in {mlns} is performed by {mcmc} over the minimal subset of the ground network required for answering the query. weights are efficiently learned from relational databases by iteratively optimizing a pseudo-likelihood measure. optionally, additional clauses are learned using inductive logic programming techniques. experiments with a real-world database and knowledge base in a university domain illustrate the promise of this approach.in this survey we overview the definitions and methods for graph clustering, that is, finding sets of  ” related” vertices in graphs. we review the many definitions for what is a cluster in a graph and measures of cluster quality. then we present global algorithms for producing a clustering for the entire vertex set of an input graph, after which we discuss the task of identifying a cluster for a specific seed vertex by local computation. some ideas on the application areas of graph clustering algorithms are given. we also address the problematics of evaluating clusterings and benchmarking cluster algorithms.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismswe describe an algorithm for discovering regulatory networks of gene modules, {gram} (genetic regulatory modules), that combines information from genome-wide location and expression data sets. a gene module is defined as a set of coexpressed genes to which the same set of transcription factors binds. unlike previous approaches1, 2, 3, 4, 5 that relied primarily on functional information from expression data, the {gram} algorithm explicitly links genes to the factors that regulate them by incorporating {dna} binding data, which provide direct physical evidence of regulatory interactions. we use the {gram} algorithm to describe a genome-wide regulatory network in saccharomyces cerevisiae using binding information for 106 transcription factors profiled in rich medium conditions data* from over 500 expression experiments. we also present a genome-wide location analysis data set for regulators in yeast cells treated with rapamycin, and use the {gram} algorithm to provide biological insights into this regulatory network. *note: in the version of this article initially published online, the word \"and\" was omitted from the fourth sentence of the abstract, altering the meaning. the sentence should read: \"we use the {gram} algorithm to describe a genome-wide regulatory network in saccharomyces cerevisiae using binding information for 106 transcription factors profiled in rich medium conditions and data from over 500 expression experiments.\" this mistake has been corrected for the {html} and print versions of the article.the establishment of complex expression patterns at precise times and locations is key to metazoan development, yet a mechanistic understanding of the underlying transcription control networks is still missing. here we describe a novel thermodynamic model that computes expression patterns as a function of cis-regulatory sequence and of the binding-site preferences and expression of participating transcription factors. we apply this model to the segmentation gene network of drosophila melanogaster and find that it predicts expression patterns of cis-regulatory modules with remarkable accuracy, demonstrating that positional information is encoded in the regulatory sequence and input factor distribution. our analysis reveals that both strong and weaker binding sites contribute, leading to high occupancy of the module {dna}, and conferring robustness against mutation; short-range homotypic clustering of weaker sites facilitates cooperative binding, which is necessary to sharpen the patterns. our computational framework is generally applicable to most {protein-dna} interaction systems.due to the selection pressure imposed by highly variable environmental conditions, stress sensing and regulatory response mechanisms in plants are expected to evolve rapidly. one potential source of innovation in plant stress response mechanisms is gene duplication. in this study, we examined the evolution of stress-regulated gene expression among duplicated genes in the model plant arabidopsis thaliana . key to this analysis was reconstructing the putative ancestral stress regulation pattern. by comparing the expression patterns of duplicated genes with the patterns of their ancestors, duplicated genes likely lost and gained stress responses at a rapid rate initially, but the rate is close to zero when the synonymous substitution rate (a proxy for time) is >\\\\~{}0.8. when considering duplicated gene pairs, we found that partitioning of putative ancestral stress responses occurred more frequently compared to cases of parallel retention and loss. furthermore, the pattern of stress response partitioning was extremely asymmetric. an analysis of putative cis -acting {dna} regulatory elements in the promoters of the duplicated stress-regulated genes indicated that the asymmetric partitioning of ancestral stress responses are likely due, at least in part, to differential loss of {dna} regulatory elements; the duplicated genes losing most of their stress responses were those that had lost more of the putative cis -acting elements. finally, duplicate genes that lost most or all of the ancestral responses are more likely to have gained responses to other stresses. therefore, the retention of duplicates that inherit few or no functions seems to be coupled to neofunctionalization. taken together, our findings provide new insight into the patterns of evolutionary changes in gene stress responses after duplication and lay the foundation for testing the adaptive significance of stress regulatory changes under highly variable biotic and abiotic environments.we demonstrate that the binding sites for highly conserved transcription factors vary extensively between human and mouse. we mapped the binding of four tissue-specific transcription factors ({foxa2}, {hnf1a}, {hnf4a} and {hnf6}) to 4,000 orthologous gene pairs in hepatocytes purified from human and mouse livers. despite the conserved function of these factors, from 41\\\\% to 89\\\\% of their binding events seem to be species specific. when the same protein binds the promoters of orthologous genes, approximately two-thirds of the binding sites do not align.this year, 2011, marks the forty-year anniversary of the statistical analysis of retinoblastoma that provided the first evidence that tumorigenesis can be initiated by as few as two mutations. this work provided the foundation for the two-hit hypothesis that explained the role of recessive tumour suppressor genes ({tsgs}) in dominantly inherited cancer susceptibility syndromes. however, four decades later, it is now known that even partial inactivation of tumour suppressors can critically contribute to tumorigenesis. here we analyse this evidence and propose a continuum model of {tsg} function to explain the full range of {tsg} mutations found in cancer.we detected and mapped a dynamically spreading wave of gray matter loss in the brains of patients with alzheimer\\'s disease ({ad}). the loss pattern was visualized in four dimensions as it spread over time from temporal and limbic cortices into frontal and occipital brain regions, sparing sensorimotor cortices. the shifting deficits were asymmetric (left hemisphere > right hemisphere) and correlated with progressively declining cognitive status (p < 0.0006). novel brain mapping methods allowed us to visualize dynamic patterns of atrophy in 52 high-resolution magnetic resonance image scans of 12 patients with {ad} (age 68.4 +/- 1.9 years) and 14 elderly matched controls (age 71.4 +/- 0.9 years) scanned longitudinally (two scans; interscan interval 2.1 +/- 0.4 years). a cortical pattern matching technique encoded changes in brain shape and tissue distribution across subjects and time. cortical atrophy occurred in a well defined sequence as the disease progressed, mirroring the sequence of neurofibrillary tangle accumulation observed in cross sections at autopsy. advancing deficits were visualized as dynamic maps that change over time. frontal regions, spared early in the disease, showed pervasive deficits later (>15\\\\% loss). the maps distinguished different phases of {ad} and differentiated {ad} from normal aging. local gray matter loss rates (5.3 +/- 2.3\\\\% per year in {ad} v 0.9 +/- 0.9\\\\% per year in controls) were faster in the left hemisphere (p < 0.029) than the right. transient barriers to disease progression appeared at limbic/frontal boundaries. this degenerative sequence, observed in vivo as it developed, provides the first quantitative, dynamic visualization of cortical atrophic rates in normal elderly populations and in those with dementia.we demonstrate an integrated approach to build, test, and refine a model of a cellular pathway, in which perturbations to critical pathway components are analyzed using {dna} microarrays, quantitative proteomics, and databases of known physical interactions. using this approach, we identify 997 messenger {rnas} responding to 20 systematic perturbations of the yeast galactose-utilization pathway, provide evidence that approximately 15 of 289 detected proteins are regulated posttranscriptionally, and identify explicit physical interactions governing the cellular response to each perturbation. we refine the model through further iterations of perturbation and global measurements, suggesting hypotheses about the regulation of galactose utilization and physical interactions between this and a variety of other metabolic pathways.summary: a motif is a short {dna} or protein sequence that contributes to the biological function of the sequence in which it resides. over the past several decades, many computational methods have been described for identifying, characterizing and searching with sequence motifs. critical to nearly any motif-based sequence analysis pipeline is the ability to scan a sequence database for occurrences of a given motif described by a position-specific frequency {matrix.results}: we describe find individual motif occurrences ({fimo}), a software tool for scanning {dna} or protein sequences with motifs described as position-specific scoring matrices. the program computes a log-likelihood ratio score for each position in a given sequence database, uses established dynamic programming methods to convert this score to a p-value and then applies false discovery rate analysis to estimate a q-value for each position in the given sequence. {fimo} provides output in a variety of formats, including {html}, {xml} and several santa cruz genome browser formats. the program is efficient, allowing for the scanning of {dna} sequences at a rate of 3.5 mb/s on a single {cpu}.availability and implementation: {fimo} is part of the {meme} suite software toolkit. a web server and source code are available at {http://meme.sdsc.edu.contact}: t.bailey@imb.uq.edu.au; {t.bailey@imb.uq.edu.ausupplementary} information: supplementary data are available at bioinformatics online.10.1101/gr.094144.109 transcriptional regulation is largely enacted by transcription factors ({tfs}) binding {dna}. large numbers of {tf} binding motifs have been revealed by {chip}-chip experiments followed by computational {dna} motif discovery. however, the success of motif discovery algorithms has been limited when applied to sequences bound in vivo (such as those identified by {chip}-chip) because the observed {tf\\\\^{a}dna} interactions are not necessarily direct: some {tfs} predominantly associate with {dna} indirectly through protein partners, while others exhibit both direct and indirect binding. here, we present the first method for distinguishing between direct and indirect {tf\\\\^{a}dna} interactions, integrating in vivo {tf} binding data, in vivo nucleosome occupancy data, and motifs from in vitro protein binding microarray experiments. when applied to yeast {chip}-chip data, our method reveals that only 48\\\\% of the data sets can be readily explained by direct binding of the profiled {tf}, while 16\\\\% can be explained by indirect {dna} binding. in the remaining 36\\\\%, none of the motifs used in our analysis was able to explain the {chip}-chip data, either because the data were too noisy or because the set of motifs was incomplete. as more in vitro {tf} {dna} binding motifs become available, our method could be used to build a complete catalog of direct and indirect {tf\\\\^{a}dna} interactions. our method is not restricted to yeast or to {chip}-chip data, but can be applied in any system for which both in vivo binding data and in vitro {dna} binding motifs are available.promoters are crucial for gene regulation. they vary greatly in terms of associated regulatory elements, sequence motifs, the choice of transcription start sites and other features. several technologies that harness next-generation sequencing have enabled recent advances in identifying promoters and their features, helping researchers who are investigating functional categories of promoters and their modes of regulation. additional features of promoters that are being characterized include types of histone modifications, nucleosome positioning, {rna} polymerase pausing and novel small {rnas}. in this review, we discuss recent findings relating to metazoan promoters and how these findings are leading to a revised picture of what a gene promoter is and how it works.we implement a strategy for aligning two protein–protein interaction networks that combines interaction topology and protein sequence similarity to identify conserved interaction pathways and complexes. using this approach we show that the protein–protein interaction networks of two distantly related species, saccharomyces cerevisiae and helicobacter pylori, harbor a large complement of evolutionarily conserved pathways, and that a large number of pathways appears to have duplicated and specialized within yeast. analysis of these findings reveals many well characterized interaction pathways as well as many unanticipated pathways, the significance of which is reinforced by their presence in the networks of both species.direct in vivo investigation of mammalian metabolism is complicated by the distinct metabolic functions of different tissues. we present a computational method that successfully describes the tissue specificity of human metabolism on a large scale. by integrating tissue-specific gene- and protein-expression data with an existing comprehensive reconstruction of the global human metabolic network, we predict tissue-specific metabolic activity in ten human tissues. this reveals a central role for post-transcriptional regulation in shaping tissue-specific metabolic activity profiles. the predicted tissue specificity of genes responsible for metabolic diseases and tissue-specific differences in metabolite exchange with biofluids extend markedly beyond tissue-specific differences manifest in enzyme-expression data, and are validated by large-scale mining of tissue-specificity data. our results establish a computational basis for the genome-wide study of normal and abnormal human metabolism in a tissue-specific manner.whole-genome approaches to identify genetic and epigenetic alterations in cancer genomes have begun to provide new insights into the range of molecular events that occurs in human tumours. although in some cases this knowledge immediately illuminates a path towards diagnostic or therapeutic implementation, the bewildering lists of mutations in each tumour make it clear that systematic functional approaches are also necessary to obtain a comprehensive molecular understanding of cancer. here we review the current range of methods, assays and approaches for genome-scale interrogation of gene function in cancer. we also discuss the integration of functional-genomics approaches with the outputs from cancer genome sequencing efforts.a common assumption in comparative genomics is that orthologous genes share greater functional similarity than do paralogous genes (the \"ortholog conjecture\"). many methods used to computationally predict protein function are based on this assumption, even though it is largely untested. here we present the first large-scale test of the ortholog conjecture using comparative functional genomic data from human and mouse. we use the experimentally derived functions of more than 8,900 genes, as well as an independent microarray dataset, to directly assess our ability to predict function using both orthologs and paralogs. both datasets show that paralogs are often a much better predictor of function than are orthologs, even at lower sequence identities. among paralogs, those found within the same species are consistently more functionally similar than those found in a different species. we also find that paralogous pairs residing on the same chromosome are more functionally similar than those on different chromosomes, perhaps due to higher levels of interlocus gene conversion between these pairs. in addition to offering implications for the computational prediction of protein function, our results shed light on the relationship between sequence divergence and functional divergence. we conclude that the most important factor in the evolution of function is not amino acid sequence, but rather the cellular context in which proteins act. {\\\\copyright} 2011 nehrt et al.the elucidation of breast cancer subgroups and their molecular drivers requires integrated views of the genome and transcriptome from representative numbers of patients. we present an integrated analysis of copy number and gene expression in a discovery and validation set of 997 and 995 primary breast tumours, respectively, with long-term clinical follow-up. inherited variants (copy number variants and single nucleotide polymorphisms) and acquired somatic copy number aberrations ({cnas}) were associated with expression in \\\\~{}40\\\\% of genes, with the landscape dominated by cis- and trans-acting {cnas}. by delineating expression outlier genes driven in cis by {cnas}, we identified putative cancer genes, including deletions in {ppp2r2a}, {mtap} and {map2k4}. unsupervised analysis of paired {dna}–{rna} profiles revealed novel subgroups with distinct clinical outcomes, which reproduced in the validation cohort. these include a high-risk, oestrogen-receptor-positive 11q13/14 cis-acting subgroup and a favourable prognosis subgroup devoid of {cnas}. trans-acting aberration hotspots were found to modulate subgroup-specific gene networks, including a {tcr} deletion-mediated adaptive immune response in the \\'{cna}-devoid\\' subgroup and a basal-specific chromosome 5 deletion-associated mitotic network. our results provide a novel molecular stratification of the breast cancer population, derived from the impact of somatic {cnas} on the transcriptome.phenotypic diversity is generated through changes in gene structure or gene regulation. the availability of full genomic sequences allows for the analysis of gene sequence evolution. in contrast, little is known about the principles driving the evolution of gene expression. here we describe the differential transcriptional response of four closely related yeast species to a variety of environmental stresses. genes containing a {tata} box in their promoters show an increased interspecies variability in expression, independent of their functional association. examining additional data sets, we find that this enhanced expression divergence of {tata}-containing genes is consistent across all eukaryotes studied to date, including nematodes, fruit flies, plants and mammals. {tata}-dependent regulation may enhance the sensitivity of gene expression to genetic perturbations, thus facilitating expression divergence at particular genetic loci.we have catalogued the protein kinase complement of the human genome (the  ” kinome”) using public and proprietary genomic, complementary {dna}, and expressed sequence tag ({est}) sequences. this provides a starting point for comprehensive analysis of protein phosphorylation in normal and disease states, as well as a detailed view of the current state of human genome analysis through a focus on one large gene family. we identify 518 putative protein kinase genes, of which 71 have not previously been reported or described as kinases, and we extend or correct the protein sequences of 56 more kinases. new genes include members of well-studied families as well as previously unidentified families, some of which are conserved in model organisms. classification and comparison with model organism kinomes identified orthologous groups and highlighted expansions specific to human and other lineages. we also identified 106 protein kinase pseudogenes. chromosomal mapping revealed several small clusters of kinase genes and revealed that 244 kinases map to disease loci or cancer amplicons.obesity is a particularly complex disease that at least partially involves genetic and environmental perturbations to gene-networks connecting the hypothalamus and several metabolic tissues, resulting in an energy imbalance at the systems level. to provide an inter-tissue view of obesity with respect to molecular states that are associated with physiological states, we developed a framework for constructing tissue-to-tissue coexpression networks between genes in the hypothalamus, liver or adipose tissue. these networks have a scale-free architecture and are strikingly independent of gene-gene coexpression networks that are constructed from more standard analyses of single tissues. this is the first systematic effort to study inter-tissue relationships and highlights genes in the hypothalamus that act as information relays in the control of peripheral tissues in obese mice. the subnetworks identified as specific to tissue-to-tissue interactions are enriched in genes that have obesity-relevant biological functions such as circadian rhythm, energy balance, stress response, or immune response. tissue-to-tissue networks enable the identification of disease-specific genes that respond to changes induced by different tissues and they also provide unique details regarding candidate genes for obesity that are identified in genome-wide association studies. identifying such genes from single tissue analyses would be difficult or impossible.surveys of protein-coding sequences for evidence of positive selection in humans or chimpanzees have flagged only a few genes known to function in neural or nutritional processes1, 2, 3, 4, 5, despite pronounced differences between humans and chimpanzees in behavior, cognition and diet6, 7, 8. it may be that most such differences are due to changes in gene regulation rather than protein structure9. here, we present the first survey of promoter (5\\'-flanking) regions, which are rich in cis-regulatory sequences, for evidence of positive selection in humans. our results indicate that positive selection has targeted the regulation of many genes known to be involved in neural development and function, both in the brain and elsewhere in the nervous system, and in nutrition, particularly in glucose metabolism.complex diseases are caused by a combination of genetic and environmental factors. uncovering the molecular pathways through which genetic factors affect a phenotype is always difficult, but in the case of complex diseases this is further complicated since genetic factors in affected individuals might be different. in recent years, systems biology approaches and, more specifically, network based approaches emerged as powerful tools for studying complex diseases. these approaches are often built on the knowledge of physical or functional interactions between molecules which are usually represented as an interaction network. an interaction network not only reports the binary relationships between individual nodes but also encodes hidden higher level organization of cellular communication. computational biologists were challenged with the task of uncovering this organization and utilizing it for the understanding of disease complexity, which prompted rich and diverse algorithmic approaches to be proposed. we start this chapter with a description of the general characteristics of complex diseases followed by a brief introduction to physical and functional networks. next we will show how these networks are used to leverage genotype, gene expression, and other types of data to identify dysregulated pathways, infer the relationships between genotype and phenotype, and explain disease heterogeneity. we group the methods by common underlying principles and first provide a high level description of the principles followed by more specific examples. we hope that this chapter will give readers an appreciation for the wealth of algorithmic techniques that have been developed for the purpose of studying complex diseases as well as insight into their strengths and limitations.elucidating gene regulatory networks is crucial for understanding normal cell physiology and complex pathologic phenotypes. existing computational methods for the genome-wide \"reverse engineering\" of such networks have been successful only for lower eukaryotes with simple genomes. here we present {aracne}, a novel algorithm, using microarray expression profiles, specifically designed to scale up to the complexity of regulatory networks in mammalian cells, yet general enough to address a wider range of network deconvolution problems. this method uses an information theoretic approach to eliminate the majority of indirect interactions inferred by co-expression methods.the genetic variation that occurs naturally in a population is a powerful resource for studying how genotype affects phenotype. each allele is a perturbation of the biological system, and genetic crosses, through the processes of recombination and segregation, randomize the distribution of these alleles among the progeny of a cross. the randomized genetic perturbations affect traits directly and indirectly, and the similarities and differences between traits in their responses to common perturbations allow inferences about whether variation in a trait is a cause of a phenotype (such as disease) or whether the trait variation is, instead, an effect of that phenotype. it is then possible to use this information about causes and effects to build models of probabilistic \\'causal networks\\'. these networks are beginning to define the outlines of the \\'genotype–phenotype map\\'.a conceptual framework for integrating diverse functional genomics data was developed by reinterpreting experiments to provide numerical likelihoods that genes are functionally linked. this allows direct comparison and integration of different classes of data. the resulting probabilistic gene network estimates the functional coupling between genes. within this framework, we reconstructed an extensive, high-quality functional gene network for saccharomyces cerevisiae, consisting of 4681 (∼81\\\\%) of the known yeast genes linked by ∼34,000 probabilistic linkages comparable in accuracy to small-scale interaction assays. the integrated linkages distinguish true from false-positive interactions in earlier data sets; new interactions emerge from genes\\' network contexts, as shown for genes in chromatin modification and ribosome biogenesis.signaling cascades are triggered by environmental stimulation and propagate the signal to regulate transcription. systematic reconstruction of the underlying regulatory mechanisms requires pathway-targeted, informative experimental data. however, practical experimental design approaches are still in their infancy. here, we propose a framework that iterates design of experiments and identification of regulatory relationships downstream of a given pathway. the experimental design component, called {meed}, aims to minimize the amount of laboratory effort required in this process. to avoid ambiguity in the identification of regulatory relationships, the choice of experiments maximizes diversity between expression profiles of genes regulated through different mechanisms. the framework takes advantage of expert knowledge about the pathways under study, formalized in a predictive logical model. by considering model-predicted dependencies between experiments, {meed} is able to suggest a whole set of experiments that can be carried out simultaneously. our framework was applied to investigate interconnected signaling pathways in yeast. in comparison with other approaches, {meed} suggested the most informative experiments for unambiguous identification of transcriptional regulation in this system.a first step in identifying the content of a document is determining which topics that document addresses. we describe a generative model for documents, introduced by blei, ng, and jordan [blei, d. m., ng, a. y. \\\\& jordan, m. i. (2003) j. machine learn. res. 3, 993-1022], in which each document is generated by choosing a distribution over topics and then choosing each word in the document from a topic selected according to this distribution. we then present a markov chain monte carlo algorithm for inference in this model. we use this algorithm to analyze abstracts from {pnas} by using bayesian model selection to establish the number of topics. we show that the extracted topics capture meaningful structure in the data, consistent with the class designations provided by the authors of the articles, and outline further applications of this analysis, including identifying  ” hot topics” by examining temporal dynamics and tagging abstracts to illustrate semantic content.author summary in network applications, one is often interested in studying whether modules are preserved across multiple networks. for example, to determine whether a pathway of genes is perturbed in a certain condition, one can study whether its connectivity pattern is no longer preserved. non-preserved modules can either be biologically uninteresting (e.g., reflecting data outliers) or interesting (e.g., reflecting sex specific modules). an intuitive approach for studying module preservation is to cross-tabulate module membership. but this approach often cannot address questions about the preservation of connectivity patterns between nodes. thus, cross-tabulation based approaches often fail to recognize that important aspects of a network module are preserved. cross-tabulation methods make it difficult to argue that a module is not preserved. the weak statement (\"the reference module does not overlap with any of the identified test set modules\") is less relevant in practice than the strong statement (\"the module cannot be found in the test network irrespective of the parameter settings of the module detection procedure\"). module preservation statistics have important applications, e.g. we show that the wiring of apoptosis genes in a human cortical network differs from that in chimpanzees.epigenetic signals are responsible for the establishment, maintenance, and reversal of metastable transcriptional states that are fundamental for the cell\\'s ability to  ” remember” past events, such as changes in the external environment or developmental cues. complex epigenetic states are orchestrated by several converging and reinforcing signals, including transcription factors, noncoding {rnas}, {dna} methylation, and histone modifications. although all of these pathways modulate transcription from chromatin in vivo, the mechanisms by which epigenetic information is transmitted through cell division remain unclear. because epigenetic states are metastable and change in response to the appropriate signals, a deeper understanding of their molecular framework will allow us to tackle the dysregulation of epigenetics in disease.molecular genetic studies of drosophila melanogaster have led to profound advances in understanding the regulation of development. here we report gene expression patterns for nearly one-third of all drosophila genes during a complete time course of development. mutations that eliminate eye or germline tissue were used to further analyze tissue-specific gene expression programs. these studies define major characteristics of the transcriptional programs that underlie the life cycle, compare development in males and females, and show that large-scale gene expression data collected from whole animals can be used to identify genes expressed in particular tissues and organs or genes involved in specific biological and biochemical processes.gene expression over time is, biologically, a continuous process and can thus be represented by a continuous function, i.e. a curve. individual genes often share similar expression patterns (functional forms). however, the shape of each function, the number of such functions, and the genes that share similar functional forms are typically unknown. here we introduce an approach that allows direct discovery of related patterns of gene expression and their underlying functions (curves) from data without a priori specification of either cluster number or functional form. smoothing spline clustering ({ssc}) models natural properties of gene expression over time, taking into account natural differences in gene expression within a cluster of similarly expressed genes, the effects of experimental measurement error, and missing data. furthermore, {ssc} provides a visual summary of each cluster\\'s gene expression function and goodness-of-fit by way of a \\'mean curve\\' construct and its associated confidence bands. we apply this method to gene expression data over the life-cycle of drosophila melanogaster and caenorhabditis elegans to discover 17 and 16 unique patterns of gene expression in each species, respectively. new and previously described expression patterns in both species are discovered, the majority of which are biologically meaningful and exhibit statistically significant gene function enrichment. software and source code implementing the algorithm, {ssclust}, is freely available ({http://genemerge.bioteam.net/ssclust}.html).mutations often have consequences that vary across individuals. here, we show that the stimulation of a stress response can reduce mutation penetrance in caenorhabditis elegans. moreover, this induced mutation buffering varies across isogenic individuals because of interindividual differences in stress signaling. this variation has important consequences in wild-type animals, producing some individuals with higher stress resistance but lower reproductive fitness and other individuals with lower stress resistance and higher reproductive fitness. this may be beneficial in an unpredictable environment, acting as a  ” bet-hedging” strategy to diversify risk. these results illustrate how transient environmental stimuli can induce protection against mutations, how environmental responses can underlie variable mutation buffering, and how a fitness trade-off may make variation in stress signaling advantageous.transcriptomics studies often rely on partial reference transcriptomes that fail to capture the full catalogue of transcripts and their variations. recent advances in sequencing technologies and assembly algorithms have facilitated the reconstruction of the entire transcriptome by deep {rna} sequencing ({rna}-seq), even without a reference genome. however, transcriptome assembly from billions of {rna}-seq reads, which are often very short, poses a significant informatics challenge. this review summarizes the recent developments in transcriptome assembly approaches — reference-based, de novo and combined strategies — along with some perspectives on transcriptome assembly in the near future.we develop an approach utilizing randomized genotypes to rigorously infer causal regulatory relationships among genes at the transcriptional level, based on experiments in which genotyping and expression profiling are performed. this approach can be used to build transcriptional regulatory networks and to identify putative regulators of genes. we apply the method to an experiment in yeast, in which genes known to be in the same processes and functions are recovered in the resulting transcriptional regulatory network.phenotypically identical cells can dramatically vary with respect to behavior during their lifespan and this variation is reflected in their molecular composition such as the transcriptomic landscape. single-cell transcriptomics using next-generation transcript sequencing ({rna}-seq) is now emerging as a powerful tool to profile cell-to-cell variability on a genomic scale. its application has already greatly impacted our conceptual understanding of diverse biological processes with broad implications for both basic and clinical research. different single-cell {rna}-seq protocols have been introduced and are reviewed here—each one with its own strengths and current limitations. we further provide an overview of the biological questions single-cell {rna}-seq has been used to address, the major findings obtained from such studies, and current challenges and expected future developments in this booming field.meiosis is a complex developmental process that generates haploid cells from diploid progenitors. we measured messenger {rna} ({mrna}) abundance and protein production through the yeast meiotic sporulation program and found strong, stage-specific expression for most genes, achieved through control of both {mrna} levels and translational efficiency. monitoring of protein production timing revealed uncharacterized recombination factors and extensive organellar remodeling. meiotic translation is also shifted toward noncanonical sites, including short open reading frames ({orfs}) on unannnotated transcripts and upstream regions of known transcripts ({uorfs}). ribosome occupancy at near-cognate {uorfs} was associated with more efficient {orf} translation; by contrast, some {aug} {uorfs}, often exposed by regulated 5′ leader extensions, acted competitively. this work reveals pervasive translational control in meiosis and helps to illuminate the molecular basis of the broad restructuring of meiotic cells.receiver operator characteristic ({roc}) curves are commonly used to present results for binary decision problems in machine learning. however, when dealing with highly skewed datasets, {precision-recall} ({pr}) curves give a more informative picture of an algorithm\\'s performance. we show that a deep connection exists between {roc} space and {pr} space, such that a curve dominates in {roc} space if and only if it dominates in {pr} space. a corollary is the notion of an achievable {pr} curve, which has properties much like the convex hull in {roc} space; we show an efficient algorithm for computing this curve. finally, we also note differences in the two types of curves are significant for algorithm design. for example, in {pr} space it is incorrect to linearly interpolate between points. furthermore, algorithms that optimize the area under the {roc} curve are not guaranteed to optimize the area under the {pr} curve.understanding complex networks of protein-protein interactions ({ppis}) is one of the foremost challenges of the post-genomic era. due to the recent advances in experimental bio-technology, including yeast-2-hybrid ({y2h}), tandem affinity purification ({tap}) and other high-throughput methods for protein-protein interaction ({ppi}) detection, huge amounts of {ppi} network data are becoming available. of major concern, however, are the levels of noise and incompleteness. for example, for {y2h} screens, it is thought that the false positive rate could be as high as 64\\\\%, and the false negative rate may range from 43\\\\% to 71\\\\%. {tap} experiments are believed to have comparable levels of noise. we present a novel technique to assess the confidence levels of interactions in {ppi} networks obtained from experimental studies. we use it for predicting new interactions and thus for guiding future biological experiments. this technique is the first to utilize currently the best fitting network model for {ppi} networks, geometric graphs. our approach achieves specificity of 85\\\\% and sensitivity of 90\\\\%. we use it to assign confidence scores to physical protein-protein interactions in the human {ppi} network downloaded from {biogrid}. using our approach, we predict 251 interactions in the human {ppi} network, a statistically significant fraction of which correspond to protein pairs sharing common {go} terms. moreover, we validate a statistically significant portion of our predicted interactions in the {hprd} database and the newer release of {biogrid}. the data and matlab code implementing the methods are freely available from the web site: {http://www.kuchaev.com/denoising}. proteins are responsible for much of the biological \\'heavy lifting\\' that keeps our cells functioning. however, proteins don\\'t usually work alone; instead they typically bind together to form geometrically and chemically complex structures that are tailored for a specific task. experimental techniques allow us to detect whether two types of proteins are capable of binding together, or \\'interacting\\'. this creates a network where two proteins are connected if they have been seen to interact, just as we could regard two people as being connected if they are linked on facebook. such protein-protein interaction networks have been developed for several organisms, using a range of methods, all of which are subject to experimental errors. these network data reveal a fascinating and intricate pattern of connections. in particular, it is known that proteins can be arranged into a low-dimensional space, such as a three-dimensional cube, so that interacting proteins are close together. our work shows that this structure can be exploited to assign confidence levels to recorded protein-protein interactions and predict new interactions that were overlooked experimentally. in tests, we predicted 251 new human protein-protein interactions, and through literature curation we independently validated a statistically significant number of them.the ability of an organism to survive depends on its capability to adapt to external conditions. in addition to metabolic versatility and efficient replication, reliable signal transduction is essential. as signaling systems are under permanent evolutionary pressure one may assume that their structure reflects certain functional properties. however, despite promising theoretical studies in recent years, the selective forces which shape signaling network topologies in general remain unclear. here, we propose prevention of autoactivation as one possible evolutionary design principle. a generic framework for continuous kinetic models is used to derive topological implications of demanding a dynamically stable ground state in signaling systems. to this end graph theoretical methods are applied. the index of the underlying digraph is shown to be a key topological property which determines the so-called kinetic ground state (or off-state) robustness. the kinetic robustness depends solely on the composition of the subdigraph with the strongly connected components, which comprise all positive feedbacks in the network. the component with the highest index in the feedback family is shown to dominate the kinetic robustness of the whole network, whereas relative size and girth of these motifs are emphasized as important determinants of the component index. moreover, depending on topological features, the maintenance of robustness differs when networks are faced with structural perturbations. this structural off-state robustness, defined as the average kinetic robustness of a network\\'s neighborhood, turns out to be useful since some structural features are neutral towards kinetic robustness, but show up to be supporting against structural perturbations. among these are a low connectivity, a high divergence and a low path sum. all results are tested against real signaling networks obtained from databases. the analysis suggests that ground state robustness may serve as a rationale for some structural peculiarities found in intracellular signaling networks.reconstructing gene regulatory networks from high-throughput data is a long-standing challenge. through the dialogue on reverse engineering assessment and methods ({dream}) project, we performed a comprehensive blind assessment of over 30 network inference methods on escherichia coli, staphylococcus aureus, saccharomyces cerevisiae and in silico microarray data. we characterize the performance, data requirements and inherent biases of different inference approaches, and we provide guidelines for algorithm application and development. we observed that no single inference method performs optimally across all data sets. in contrast, integration of predictions from multiple inference methods shows robust and high performance across diverse data sets. we thereby constructed high-confidence networks for e. coli and s. aureus, each comprising \\\\~{}1,700 transcriptional interactions at a precision of \\\\~{}50\\\\%. we experimentally tested 53 previously unobserved regulatory interactions in e. coli, of which 23 (43\\\\%) were supported. our results establish community-based methods as a powerful and robust tool for the inference of transcriptional gene regulatory networks.{background}:high throughput gene expression experiments yield large amounts of data that can augment our understanding of disease processes, in addition to classifying samples. here we present new paradigms of data separation based on construction of transcriptional regulatory networks for normal and abnormal cells using sequence predictions, literature based data and gene expression studies. we analyzed expression datasets from a number of diseased and normal cells, including different types of acute leukemia, and breast cancer with variable clinical {outcome.results}:we constructed sample-specific regulatory networks to identify links between transcription factors ({tfs}) and regulated genes that differentiate between healthy and diseased states. this approach carries the advantage of identifying key transcription factor-gene pairs with differential activity between healthy and diseased states rather than merely using gene expression profiles, thus alluding to processes that may be involved in gene deregulation. we then generalized this approach by studying simultaneous changes in functionality of multiple regulatory links pointing to a regulated gene or emanating from one {tf} (or changes in gene centrality defined by its in-degree or out-degree measures, respectively). we found that samples can often be separated based on these measures of gene centrality more robustly than using individual {links.we} examined distributions of distances (the number of links needed to traverse the path between each pair of genes) in the transcriptional networks for gene subsets whose collective expression profiles could best separate each dataset into predefined groups. we found that genes that optimally classify samples are concentrated in neighborhoods in the gene regulatory networks. this suggests that genes that are deregulated in diseased states exhibit a remarkable degree of {connectivity.conclusion}:transcription factor-regulated gene links and centrality of genes on transcriptional networks can be used to differentiate between cell types. transcriptional network blueprints can be used as a basis for further research into gene deregulation in diseased states.eukaryotic transcription occurs within a chromatin environment, whose organization has an important regulatory function and is partly encoded in cis by the {dna} sequence itself. here, we examine whether evolutionary changes in gene expression are linked to changes in the {dna}-encoded nucleosome organization of promoters. we find that in aerobic yeast species, where cellular respiration genes are active under typical growth conditions, the promoter sequences of these genes encode a relatively open (nucleosome-depleted) chromatin organization. this nucleosome-depleted organization requires only {dna} sequence information, is independent of any cofactors and of transcription, and is a general property of growth-related genes. in contrast, in anaerobic yeast species, where cellular respiration genes are relatively inactive under typical growth conditions, respiration gene promoters encode relatively closed (nucleosome-occupied) chromatin organizations. our results suggest a previously unidentified genetic mechanism underlying phenotypic diversity, consisting of {dna} sequence changes that directly alter the {dna}-encoded nucleosome organization of promoters.transcription factors ({tfs}) and their specific interactions with targets are crucial for specifying gene-expression programs. to gain insights into the transcriptional regulatory networks in embryonic stem ({es}) cells, we use chromatin immunoprecipitation coupled with ultra-high-throughput {dna} sequencing ({chip}-seq) to map the locations of 13 sequence-specific {tfs} (nanog, oct4, {stat3}, smad1, sox2, zfx, {c-myc}, {n-myc}, klf4, esrrb, tcfcp2l1, e2f1, and {ctcf}) and 2 transcription regulators (p300 and suz12). these factors are known to play different roles in {es}-cell biology as components of the {lif} and {bmp} signaling pathways, self-renewal regulators, and key reprogramming factors. our study provides insights into the integration of the signaling pathways into the {es}-cell-specific transcription circuitries. intriguingly, we find specific genomic regions extensively targeted by different {tfs}. collectively, the comprehensive mapping of {tf}-binding sites identifies important features of the transcriptional regulatory networks that define {es}-cell identity.discovering novel disease genes is still challenging for diseases for which no prior knowledge - such as known disease genes or disease-related pathways - is available. performing genetic studies frequently results in large lists of candidate genes of which only few can be followed up for further investigation. we have recently developed a computational method for constitutional genetic disorders that identifies the most promising candidate genes by replacing prior knowledge by experimental data of differential gene expression between affected and healthy individuals.the functional consequences of genetic variation in mammalian regulatory elements are poorly understood. we report the in vivo dissection of three mammalian enhancers at single-nucleotide resolution through a massively parallel reporter assay. for each enhancer, we synthesized a library of >100,000 mutant haplotypes with 2-3\\\\% divergence from the wild-type sequence. each haplotype was linked to a unique sequence tag embedded within a transcriptional cassette. we introduced each enhancer library into mouse liver and measured the relative activities of individual haplotypes en masse by sequencing the transcribed tags. linear regression analysis yielded highly reproducible estimates of the effect of every possible single-nucleotide change on enhancer activity. the functional consequence of most mutations was modest, with ∼22\\\\% affecting activity by >1.2-fold and ∼3\\\\% by >2-fold. several, but not all, positions with higher effects showed evidence for purifying selection, or co-localized with known liver-associated transcription factor binding sites, demonstrating the value of empirical high-resolution functional analysis.comparative functional genomics studies the evolution of biological processes by analyzing functional data, such as gene expression profiles, across species. a major challenge is to compare profiles collected in a complex phylogeny. here, we present arboretum, a novel scalable computational algorithm that integrates expression data from multiple species with species and gene phylogenies to infer modules of coexpressed genes in extant species and their evolutionary histories. we also develop new, generally applicable measures of conservation and divergence in gene regulatory modules to assess the impact of changes in gene content and expression on module evolution. we used arboretum to study the evolution of the transcriptional response to heat shock in eight species of ascomycota fungi and to reconstruct modules of the ancestral environmental stress response ({esr}). we found substantial conservation in the stress response across species and in the reconstructed components of the ancestral {esr} modules. the greatest divergence was in the most induced stress, primarily through module expansion. the divergence of the heat stress response exceeds that observed in the response to glucose depletion in the same species. arboretum and its associated analyses provide a comprehensive framework to systematically study regulatory evolution of condition-specific responses.gene regulation differs greatly between related species, constituting a major source of phenotypic diversity. recent studies characterized extensive differences in the gene expression programs of closely related species. in contrast, virtually nothing is known about the evolution of chromatin structure and how it influences the divergence of gene expression. here, we compare the genome-wide nucleosome positioning of two closely related yeast species and, by profiling their inter-specific hybrid, trace the genetic basis of the observed differences into mutations affecting the local {dna} sequences (cis effects) or the upstream regulators (trans effects). the majority (\\\\~{}70\\\\%) of inter-species differences is due to cis effects, leaving a significant contribution (30\\\\%) for trans factors. we show that cis effects are well explained by mutations in nucleosome-disfavoring {at}-rich sequences, but are not associated with divergence of nucleosome-favoring sequences. differences in nucleosome positioning propagate to multiple adjacent nucleosomes, supporting the statistical positioning hypothesis, and we provide evidence that nucleosome-free regions, but not the +1 nucleosome, serve as stable border elements. surprisingly, although we find that differential nucleosome positioning among cell types is strongly correlated with differential expression, this does not seem to be the case for evolutionary changes: divergence of nucleosome positioning is excluded from regulatory elements and is not correlated with gene expression divergence, suggesting a primarily neutral mode of evolution. our results provide evolutionary insights to the genetic determinants and regulatory function of nucleosome positioning.motivation: further advancement of {rna}-seq technology and its application call for the development of effective normalization methods for {rna}-seq data. currently, different normalization methods are compared and validated by their correlations with a certain gold standard. gene expression measurements generated by a different technology or platform such as real-time reverse transcription polymerase chain reaction ({qrt}–{pcr}) or microarray are usually used as the gold standard. although the current approach is intuitive and easy to implement, it becomes statistically inadequate when the gold standard is also subject to measurement error ({me}). furthermore, the current approach is not informative, because the correlation of a normalization method with a certain gold standard does not provide much information about the exact quality of the normalized {rna}-seq measurements.protein-protein interactions are essential in every aspect of cellular activity. multiprotein complexes form and dissociate constantly in a specifically tuned manner, often by conserved mechanisms. protein domains that bind proline-rich motifs ({prms}) are frequently involved in signaling events. the unique properties of proline provide a mechanism for highly discriminatory recognition without requiring high affinities. we present herein a detailed, quantitative assessment of the structural features that define the interfaces between {prm}-binding domains and their target {prms}, and investigate the specificity of {prm} recognition. together with the analysis of peptide-library screens, this approach has allowed the identification of several highly conserved key interactions found in all complexes of {prm}-binding domains. the inhibition of protein-protein interactions by using small-molecule agents is very challenging. therefore, it is important to first pinpoint the critical interactions that must be considered in the design of inhibitors of {prm}-binding domains.reverse engineering methods are typically first tested on simulated data from in silico networks, for systematic and efficient performance assessment, before an application to real biological networks. in this paper, we present a method for generating biologically plausible in silico networks, which allow realistic performance assessment of network inference algorithms. instead of using random graph models, which are known to only partly capture the structural properties of biological networks, we generate network structures by extracting modules from known biological interaction networks. using the yeast transcriptional regulatory network as a test case, we show that extracted modules have a biologically plausible connectivity because they preserve functional and structural properties of the original network. our method was selected to generate the ?gold standard? networks for the gene network reverse engineering challenge of the third {dream} conference (dialogue on reverse engineering assessment and methods 2008, cambridge, {ma}).genome-wide comparisons between enteric bacteria yield large sets of conserved putative regulatory sites on a gene-by-gene basis that need to be clustered into regulons. using the assumption that regulatory sites can be represented as samples from weight matrices ({wms}), we derive a unique probability distribution for assignments of sites into clusters. our algorithm, \"{procse}\" (probabilistic clustering of sequences), uses monte carlo sampling of this distribution to partition and align thousands of short {dna} sequences into clusters. the algorithm internally determines the number of clusters from the data and assigns significance to the resulting clusters. we place theoretical limits on the ability of any algorithm to correctly cluster sequences drawn from {wms} when these {wms} are unknown. our analysis suggests that the set of all putative sites for a single genome (e.g., escherichia coli) is largely inadequate for clustering. when sites from different genomes are combined and all the homologous sites from the various species are used as a block, clustering becomes feasible. we predict 50-100 new regulons as well as many new members of existing regulons, potentially doubling the number of known regulatory sites in e. coli.assembling the tree of life is a major goal of biology, but progress has been hindered by the difficulty and expense of obtaining the orthologous {dna} required for accurate and fully resolved phylogenies. next-generation {dna} sequencing technologies promise to accelerate progress, but sequencing the genomes of hundreds of thousands of eukaryotic species remains impractical. eukaryotic transcriptomes, which are smaller than genomes and biased toward highly expressed genes that tend to be conserved, could potentially provide a rich set of phylogenetic characters. we sampled the transcriptomes of 10 mosquito species by assembling 36-bp sequence reads into phylogenomic data matrices containing hundreds of thousands of orthologous nucleotides from hundreds of genes. analysis of these data matrices yielded robust phylogenetic inferences, even with data matrices constructed from surprisingly few sequence reads. this approach is more efficient, data-rich, and economical than traditional {pcr}-based and {est}-based methods and provides a scalable strategy for generating phylogenomic data matrices to infer the branches and twigs of the tree of life.in complex diseases, various combinations of genomic perturbations often lead to the same phenotype. on a molecular level, combinations of genomic perturbations are assumed to dys-regulate the same cellular pathways. such a pathway-centric perspective is fundamental to understanding the mechanisms of complex diseases and the identification of potential drug targets. in order to provide an integrated perspective on complex disease mechanisms, we developed a novel computational method to simultaneously identify causal genes and dys-regulated pathways. first, we identified a representative set of genes that are differentially expressed in cancer compared to non-tumor control cases. assuming that disease-associated gene expression changes are caused by genomic alterations, we determined potential paths from such genomic causes to target genes through a network of molecular interactions. applying our method to sets of genomic alterations and gene expression profiles of 158 glioblastoma multiforme ({gbm}) patients we uncovered candidate causal genes and causal paths that are potentially responsible for the altered expression of disease genes. we discovered a set of putative causal genes that potentially play a role in the disease. combining an expression quantitative trait loci ({eqtl}) analysis with pathway information, our approach allowed us not only to identify potential causal genes but also to find intermediate nodes and pathways mediating the information flow between causal and target genes. our results indicate that different genomic perturbations indeed dys-regulate the same functional pathways, supporting a pathway-centric perspective of cancer. while copy number alterations and gene expression data of glioblastoma patients provided opportunities to test our approach, our method can be applied to any disease system where genetic variations play a fundamental causal role. it is now being recognized that complex diseases should be studied from the perspective of dys-regulated pathways and processes rather than individual genes. indeed, various combinations of molecular perturbations might lead to the same disease. in such cases, responses to these perturbations are expected to converge to common pathways. in addition, signals that are associated with each individual perturbation might be weak, rendering studies of complex diseases particularly challenging. aiming to provide an integrated perspective on complex disease mechanisms we developed a novel computational method to simultaneously identify causal genes and dys-regulated pathways. starting with an identification of a disease-associated set of genes and their statistical associations with genomic alterations, we utilized graph-theoretical techniques and combinatorial algorithms to determine potential paths from the genomic causes through a network of molecular interactions. we applied our method to sets of genomic alterations and gene expression profiles of glioblastoma multiforme ({gbm}) patients, uncovering candidate causal genes and causal paths that are potentially responsible for the altered expression of disease associated target genes. while copy number alterations and gene expression data of {gbm} patients provided opportunities to test our approach, our method can be applied to any disease system where genetic alterations play a fundamental causal role, and provides an important step toward the understanding of complex diseases.the use of correlation networks is widespread in the analysis of gene expression and proteomics data, even though it is known that correlations not only confound direct and indirect associations but also provide no means to distinguish between cause and effect. for \"causal\" analysis typically the inference of a directed graphical model is required. however, this is rather difficult due to the curse of dimensionality. we propose a simple heuristic for the statistical learning of a high-dimensional \"causal\" network. the method first converts a correlation network into a partial correlation graph. subsequently, a partial ordering of the nodes is established by multiple testing of the log-ratio of standardized partial variances. this allows identifying a directed acyclic causal network as a subgraph of the partial correlation network. we illustrate the approach by analyzing a large arabidopsis thaliana expression data set. the proposed approach is a heuristic algorithm that is based on a number of approximations, such as substituting lower order partial correlations by full order partial correlations. nevertheless, for small samples and for sparse networks the algorithm not only yield sensible first order approximations of the causal structure in high-dimensional genomic data but is also computationally highly efficient. {availability} {and} {requirements}: the method is implemented in the \"{genenet}\" r package (version 1.2.0), available from {cran} and from http://strimmerlab.org/software/genets/. the software includes an r script for reproducing the network analysis of the arabidopsis thaliana data.reconstruction of the molecular pathways controlling organ development has been hampered by a lack of methods to resolve embryonic progenitor cells. here we describe a strategy to address this problem that combines gene expression profiling of large numbers of single cells with data analysis based on diffusion maps for dimensionality reduction and network synthesis from state transition graphs. applying the approach to hematopoietic development in the mouse embryo, we map the progression of mesoderm toward blood using single-cell gene expression analysis of 3,934 cells with blood-forming potential captured at four time points between e7.0 and e8.5. transitions between individual cellular states are then used as input to develop a single-cell network synthesis toolkit to generate a computationally executable transcriptional regulatory network model of blood development. several model predictions concerning the roles of sox and hox factors are validated experimentally. our results demonstrate that single-cell analysis of a developing organ coupled with computational approaches can reveal the transcriptional programs that underpin organogenesis.for decades, it has been hypothesized that gene regulation has had a central role in human evolution, yet much remains unknown about the genome-wide impact of regulatory mutations. here we use whole-genome sequences and genome-wide chromatin immunoprecipitation and sequencing data to demonstrate that natural selection has profoundly influenced human transcription factor binding sites since the divergence of humans from chimpanzees 4–6 million years ago. our analysis uses a new probabilistic method, called {insight}, for measuring the influence of selection on collections of short, interspersed noncoding elements. we find that, on average, transcription factor binding sites have experienced somewhat weaker selection than protein-coding genes. however, the binding sites of several transcription factors show clear evidence of adaptation. several measures of selection are strongly correlated with predicted binding affinity. overall, regulatory elements seem to contribute substantially to both adaptive substitutions and deleterious polymorphisms with key implications for human evolution and disease.quantifying the differential expression of genes in various human organs, tissues, and cell types is vital to understand human physiology and disease. recently, several large-scale transcriptomics studies have analyzed the expression of protein-coding genes across tissues. these datasets provide a framework for defining the molecular constituents of the human body as well as for generating comprehensive lists of proteins expressed across tissues or in a tissue-restricted manner. here, we review publicly available human transcriptome resources and discuss body-wide data from independent genome-wide transcriptome analyses of different tissues. gene expression measurements from these independent datasets, generated using samples from fresh frozen surgical specimens and postmortem tissues, are consistent. overall, the different genome-wide analyses support a distribution in which many proteins are found in all tissues and relatively few in a tissue-restricted manner. moreover, we discuss the applications of publicly available omics data for building genome-scale metabolic models, used for analyzing cell and tissue functions both in physiological and in disease contexts. {\\\\copyright} 2016 the authors. published under the terms of the {cc} {by} 4.0 license.motivation: over the past decade, the prospect of inferring networks of gene regulation from high-throughput experimental data has received a great deal of attention. in contrast to the massive effort that has gone into automated deconvolution of biological networks, relatively little effort has been invested in benchmarking the proposed algorithms. the rate at which new network inference methods are being proposed far outpaces our ability to objectively evaluate and compare them. this is largely due to a lack of fully understood biological networks to use as gold standards.transcriptomic studies routinely measure expression levels across numerous conditions. these datasets allow identification of genes that are specifically expressed in a small number of conditions. however, there are currently no statistically robust methods for identifying such genes. here we present {specond}, a method to detect condition-specific genes that outperforms alternative approaches. we apply the method to a dataset of 32 human tissues to determine 2,673 specifically expressed genes. an implementation of {specond} is freely available as a bioconductor package at {http://www.bioconductor.org/packages/release/bioc/html/specond}.html.regulatory networks that control gene expression are important in diverse biological contexts including stress response and development. each gene\\'s regulatory program is determined by module-level regulation (e.g. co-regulation via the same signaling system), as well as gene-specific determinants that can fine-tune expression. we present a novel approach, modular regulatory network learning with per gene information ({merlin}), that infers regulatory programs for individual genes while probabilistically constraining these programs to reveal module-level organization of regulatory networks. using edge-, regulator- and module-based comparisons of simulated networks of known ground truth, we find {merlin} reconstructs regulatory programs of individual genes as well or better than existing approaches of network reconstruction, while additionally identifying modular organization of the regulatory networks. we use {merlin} to dissect global transcriptional behavior in two biological contexts: yeast stress response and human embryonic stem cell differentiation. regulatory modules inferred by {merlin} capture co-regulatory relationships between signaling proteins and downstream transcription factors thereby revealing the upstream signaling systems controlling transcriptional responses. the inferred networks are enriched for regulators with genetic or physical interactions, supporting the inference, and identify modules of functionally related genes bound by the same transcriptional regulators. our method combines the strengths of per-gene and per-module methods to reveal new insights into transcriptional regulation in stress and development. the state of a cell is largely determined by the genes the cell expresses. transcriptional control of gene expression is exerted by transcription factor proteins that bind to regulatory regions of genes and affect their expression. transcriptional programs have a modular organization enabling multiple genes to be coordinately regulated, and at the same time are fine-tuned for each gene through interactions of transcription factors with a gene\\'s regulatory region. transcription factors are themselves controlled by upstream signaling proteins, that in turn can be transcriptionally controlled. this complex process of gene expression control is described by a regulatory network that captures who regulates whom. a key challenge in systems biology is to reconstruct regulatory networks that capture precise gene-specific regulatory information, as well as the modular organization of transcriptional programs. we developed a novel regulatory network inference approach, {merlin}, modular regulatory network learning with per gene information. when applied to examine transcriptional responses in two distinct processes, stress response and cellular differentiation, {merlin} accurately reconstructed regulatory programs of individual genes while revealing regulatory module organization and predicted upstream signaling proteins for regulatory modules. {merlin} is applicable to different environmental, developmental and disease contexts to dissect regulatory programs and ultimately build network-based predictive models of cellular states.genome-wide {rna} expression data provide a detailed view of an organism\\'s biological state; hence, a dataset measuring expression variation between genetically diverse individuals ({eqtl} data) may provide important insights into the genetics of complex traits. however, with data from a relatively small number of individuals, it is difficult to distinguish true causal polymorphisms from the large number of possibilities. the problem is particularly challenging in populations with significant linkage disequilibrium, where traits are often linked to large chromosomal regions containing many genes. here, we present a novel method, lirnet, that automatically learns a regulatory potential for each sequence polymorphism, estimating how likely it is to have a significant effect on gene expression. this regulatory potential is defined in terms of  ” regulatory features”—including the function of the gene and the conservation, type, and position of genetic polymorphisms—that are available for any organism. the extent to which the different features influence the regulatory potential is learned automatically, making lirnet readily applicable to different datasets, organisms, and feature sets. we apply lirnet both to the human {hapmap} {eqtl} dataset and to a yeast {eqtl} dataset and provide statistical and biological results demonstrating that lirnet produces significantly better regulatory programs than other recent approaches. we demonstrate in the yeast data that lirnet can correctly suggest a specific causal sequence variation within a large, linked chromosomal region. in one example, lirnet uncovered a novel, experimentally validated connection between puf3—a sequence-specific {rna} binding protein—and p-bodies—cytoplasmic structures that regulate translation and {rna} stability—as well as the particular causative polymorphism, a {snp} in mkt1, that induces the variation in the pathway. gene expression data of genetically diverse individuals ({eqtl} data) provide a unique perspective on the effect of genetic variation on cellular pathways. however, the burden of multiple hypotheses, combined with the challenges of linkage disequilibrium, makes it difficult to correctly identify causal polymorphisms. researchers traditionally apply heuristics for selecting among plausible hypotheses, favoring polymorphisms that are more conserved, that lead to significant amino acid change, or that reside in genes whose function is related to that of the targets. but how do we know how much weight to attribute to different regulatory features? we describe lirnet, which learns from {eqtl} data how to weight regulatory features and induce a regulatory potential for sequence variations. lirnet assesses these weights simultaneously to learning a regulatory network, finding weights that lead to a more predictive network. we show that lirnet constructs high-accuracy regulatory programs and demonstrate its ability to correctly identify causative polymorphisms. lirnet can flexibly use any regulatory features, including sequence features that are available for any sequenced organism, and automatically learn their weights in a dataset-specific way. this feature makes it especially advantageous for mammalian systems, where many forms of prior knowledge used in simple model organisms are incomplete or unavailable.in the simplest view of transcriptional regulation, the expression of a gene is turned on or off by changes in the concentration of a transcription factor ({tf}). we use recent data on noise levels in gene expression to show that it should be possible to transmit much more than just one regulatory bit. realizing this optimal information capacity would require that the dynamic range of {tf} concentrations used by the cell, the input/output relation of the regulatory module, and the noise in gene expression satisfy certain matching relations, which we derive. these results provide parameter-free, quantitative predictions connecting independently measurable quantities. although we have considered only the simplified problem of a single gene responding to a single {tf}, we find that these predictions are in surprisingly good agreement with recent experiments on the {bicoid/hunchback} system in the early drosophila embryo and that this system achieves ∼90\\\\% of its theoretical maximum information transmission.nonrandom mating induces correlations in allelic states within and among loci that can be exploited to understand the genetic structure of natural populations ({wright} 1965). for many species, it is of considerable interest to quantify the contribution of two forms of nonrandom mating to patterns of standing genetic variation: inbreeding (mating among relatives) and population substructure (limited dispersal of gametes). here, we extend the popular bayesian clustering approach {structure} ({pritchard} et al. 2000) for simultaneous inference of inbreeding or selfing rates and population-of-origin classification using multilocus genetic markers. this is accomplished by eliminating the assumption of {hardyweinberg} equilibrium within clusters and, instead, calculating expected genotype frequencies on the basis of inbreeding or selfing rates. we demonstrate the need for such an extension by showing that selfing leads to spurious signals of population substructure using the standard {structure} algorithm with a bias toward spurious signals of admixture. we gauge the performance of our method using extensive coalescent simulations and demonstrate that our approach can correct for this bias. we also apply our approach to understanding the population structure of the wild relative of domesticated rice, oryza rufipogon, an important partially selfing grass species. using a sample of n = 16 individuals sequenced at 111 random loci, we find strong evidence for existence of two subpopulations, which correlates well with geographic location of sampling, and estimate selfing rates for both groups that are consistent with estimates from experimental data (s approx 0.480.70). 10.1534/genetics.107.072371{oscar} is a web platform for cluster and cross-species analysis of microarray data. it provides a comprehensive but friendly environment to both users and algorithm developers. for users, {oscar} provides cluster tools for both single and multiple species data, together with interactive analysis features. for single species data, {oscar} currently provides hierarchical clustering, k-means, partition around medoids ({pam}), {self-organizing} map ({som}), tight clustering and a novel algorithm called \\'consensus tight-clustering\\'. the new consensus tight-clustering algorithm delivers robust gene clusters and its result is more resistant to false positives than other state-of-the-art algorithms. for cross-species data analysis, {oscar} provides two novel computational tools: \\'{coherentcluster}\\', \\'{coherentsubset}\\' and a novel visualization tool: \\'comparative heatmap\\'. applying the {coherentcluster} algorithm to human and fly aging data, we identified several coherent clusters of genes, which share co-regulation patterns that are highly correlated with the aging process in both of the two species. one coherent cluster suggests insulin receptor ({insr}) may regulate pax6 in both species and across different tissues. further analysis with human brain expression and pathological data suggests an {insr}->pax6->quinone oxidoreductase ({nqo1})->detoxification neuro-protective pathway might be present in aging or diseased brain. for algorithm developers, {oscar} is a plug-and-play platform. with little effort, developers can plug their own algorithms into the {oscar} server without revealing the source codes, which will equip their command line executables with user-friendly interface and interactive analysis capability. in summary, {oscar} initiates an open platform for development and application of clustering and cross-species analysis programs. {oscar} stands for an open system for cluster analysis of microarray data. it is available at: http://biocomp.bioen.uiuc.edu/oscar.gene regulatory networks, in which edges between nodes describe interactions between transcriptional regulators and their target genes, determine the coordinated spatiotemporal expression of genes. especially in higher organisms, context-specific combinatorial regulation by transcription factors ({tfs}) is believed to determine cellular states and fates. {tf}-target gene interactions can be studied using high-throughput techniques such as {chip}-chip or {chip}-seq. these experiments are time and cost intensive, and further limited by, for instance, availability of high affinity {tf} antibodies. hence, there is a practical need for methods that can predict {tf}-{tf} and {tf}-target gene interactions in silico, i.e. from gene expression and {dna} sequence data alone. we propose {gemula}, a novel approach based on linear models to predict {tf}-gene expression associations and {tf}-{tf} interactions from experimental data. {gemula} is based on linear models, fast and considers a wide range of biologically plausible models that describe gene expression data as a function of predicted {tf} binding to gene promoters. we show that models inferred with {gemula} are able to explain roughly 70\\\\% of the observed variation in gene expression in the yeast heat shock response. the functional relevance of the inferred {tf}-{tf} interactions in these models are validated by different sources of independent experimental evidence. we also have applied {gemula} to an in vitro model of neuronal outgrowth. our findings confirm existing knowledge on gene regulatory interactions underlying neuronal outgrowth, but importantly also generate new insights into the temporal dynamics of this gene regulatory network that can now be addressed experimentally. the {gemula} r-package is available from http://www.few.vu.nl/\\\\~{}degunst/gemula\\\\_1.0.tar.gz.the regulatory information for a eukaryotic gene is encoded in cis-regulatory modules. the binding sites for a set of interacting transcription factors have the tendency to colocalize to the same modules. current de novo motif discovery methods do not take advantage of this knowledge. we propose a hierarchical mixture approach to model the cis-regulatory module structure. based on the model, a new de novo motif-module discovery algorithm, {cismodule}, is developed for the bayesian inference of module locations and within-module motif sites. dynamic programming-like recursions are developed to reduce the computational complexity from exponential to linear in sequence length. by using both simulated and real data sets, we demonstrate that {cismodule} is not only accurate in predicting modules but also more sensitive in detecting motif patterns and binding sites than standard motif discovery methods are.motivation: a knowledge of the dynamics of transcription factors is fundamental to understand the transcriptional regulation mechanism. nowadays, an experimental measure of transcription factor activities in vivo represents a challenge. several methods have been developed to infer these activities from easily measurable quantities such as {mrna} expression of target genes. a limitation of these methods is represented by the fact that they rely on very simple single-layer structures, typically consisting of one or more transcription factors regulating a number of target genes.sleep disorders are common in humans, and sleep loss increases the risk of obesity and diabetes. studies in drosophila have revealed molecular pathways and neural tissues regulating sleep; however, genes that maintain genetic variation for sleep in natural populations are unknown. here, we characterized sleep in 40 wild-derived drosophila lines and observed abundant genetic variation in sleep architecture. we associated sleep with genome-wide variation in gene expression to identify candidate genes. we independently confirmed that molecular polymorphisms in catsup (catecholamines up) are associated with variation in sleep and that p-element mutations in four candidate genes affect sleep and gene expression. transcripts associated with sleep grouped into biologically plausible genetically correlated transcriptional modules. we confirmed co-regulated gene expression using p-element mutants. quantitative genetic analysis of natural phenotypic variation is an efficient method for revealing candidate genes and pathways.embryogenesis is controlled by large gene-regulatory networks, which generate spatially and temporally refined patterns of gene expression. here, we report the characteristics of the regulatory network orchestrating early mesodermal development in the fruitfly drosophila, where the transcription factor twist is both necessary and sufficient to drive development. through the integration of chromatin immunoprecipitation followed by microarray analysis ({chip}-on-chip) experiments during discrete time periods with computational approaches, we identified >2000 twist-bound cis-regulatory modules ({crms}) and almost 500 direct target genes. unexpectedly, twist regulates an almost complete cassette of genes required for cell proliferation in addition to genes essential for morophogenesis and cell migration. twist targets almost 25\\\\% of all annotated drosophila transcription factors, which may represent the entire set of regulators necessary for the early development of this system. by combining in vivo binding data from twist, mef2, tinman, and dorsal we have constructed an initial transcriptional network of early mesoderm development. the network topology reveals extensive combinatorial binding, feed-forward regulation, and complex logical outputs as prevalent features. in addition to binary activation and repression, we suggest that twist binds to almost all mesodermal {crms} to provide the competence to integrate inputs from more specialized transcription factors.network analysis is currently used in a myriad of contexts, from identifying potential drug targets to predicting the spread of epidemics and designing vaccination strategies and from finding friends to uncovering criminal activity. despite the promise of the network approach, the reliability of network data is a source of great concern in all fields where complex networks are studied. here, we present a general mathematical and computational framework to deal with the problem of data reliability in complex networks. in particular, we are able to reliably identify both missing and spurious interactions in noisy network observations. remarkably, our approach also enables us to obtain, from those noisy observations, network reconstructions that yield estimates of the true network properties that are more accurate than those provided by the observations themselves. our approach has the potential to guide experiments, to better characterize network data sets, and to drive new discoveries.high-quality, well-annotated genome sequences and standardized laboratory strains fuel experimental and evolutionary research. we present improved genome sequences of three species of saccharomyces sensu stricto yeasts: s. bayanus var. uvarum ({cbs} 7001), s. kudriavzevii ({ifo} {1802t} and {zp} 591), and s. mikatae ({ifo} {1815t}), and describe their comparison to the genomes of s. cerevisiae and s. paradoxus. the new sequences, derived by assembling millions of short {dna} sequence reads together with previously published sanger shotgun reads, have vastly greater long-range continuity and far fewer gaps than the previously available genome sequences. new gene predictions defined a set of 5261 protein-coding orthologs across the five most commonly studied saccharomyces yeasts, enabling a re-examination of the tempo and mode of yeast gene evolution and improved inferences of species-specific gains and losses. to facilitate experimental investigations, we generated genetically marked, stable haploid strains for all three of these saccharomyces species. these nearly complete genome sequences and the collection of genetically marked strains provide a valuable toolset for comparative studies of gene function, metabolism, and evolution, and render saccharomyces sensu stricto the most experimentally tractable model genus. these resources are freely available and accessible through {www.saccharomycessensustricto}.org.motivation: although many gene set analysis methods have been proposed to explore associations between a phenotype and a group of genes sharing common biological functions or involved in the same biological process, the underlying biological mechanisms of identified gene sets are typically {unexplained.results}: we propose a method called differential regulation-based enrichment analysis for {gene} sets ({dragen}) to identify gene sets in which a significant proportion of genes have their transcriptional regulatory patterns changed in a perturbed phenotype. we conduct comprehensive simulation studies to demonstrate the capability of our method in identifying differentially regulated gene sets. we further apply our method to three human microarray expression datasets, two with hormone treated and control samples and one concerning different cell cycle phases. results indicate that the capability of {dragen} in identifying phenotype-associated gene sets is significantly superior to those of four existing methods for analyzing differentially expressed gene sets. we conclude that the proposed differential regulation enrichment analysis method, though exploratory in nature, complements the existing gene set analysis methods and provides a promising new direction for the interpretation of gene expression {data.availability} and implementation: the program of {dragen} is freely available at {http://bioinfo.au.tsinghua.edu.cn/dragen/.contact}: ruijiang@tsinghua.edu.cn or {jiang@cs.ucr.edusupplementary} information: supplementary data are available at bioinformatics online.the differentiation of hematopoietic stem cells into cells of the immune system has been studied extensively in mammals, but the transcriptional circuitry that controls it is still only partially understood. here, the immunological genome project gene-expression profiles across mouse immune lineages allowed us to systematically analyze these circuits. to analyze this data set we developed ontogenet, an algorithm for reconstructing lineage-specific regulation from gene-expression profiles across lineages. using ontogenet, we found differentiation stage-specific regulators of mouse hematopoiesis and identified many known hematopoietic regulators and 175 previously unknown candidate regulators, as well as their target genes and the cell types in which they act. among the previously unknown regulators, we emphasize the role of {etv5} in the differentiation of γδ t cells. as the transcriptional programs of human and mouse cells are highly conserved, it is likely that many lessons learned from the mouse model apply to humans.{background}: correlations between polymorphic markers and observed phenotypes provide the basis for mapping traits in quantitative genetics. when the phenotype is gene expression, then loci involved in regulatory control can theoretically be implicated. recent efforts to construct gene regulatory networks from genotype and gene expression data have shown that biologically relevant networks can be achieved from an integrative approach. in this paper, we consider the problem of identifying individual pairs of genes in a direct or indirect, causal, trans-acting relationship. {results}: inspired by epistatic models of multi-locus quantitative trait ({qtl}) mapping, we propose a unified model of expression and genotype to identify quantitative trait genes ({qtg}) by extending the conventional linear model to include both genotype and expression of regulator genes and their interactions. the model provides mapping of specific genes in contrast to standard linkage approaches that implicate large {qtl} intervals typically containing tens of genes. in simulations, we found that the method can often detect weak trans-acting regulators amid the background noise of thousands of traits and is robust to transcription models containing multiple regulator genes. we reanalyze several pleiotropic loci derived from a large set of yeast matings and identify a likely alternative regulator not previously published. however, we also found that many regulators can not be so easily mapped due to the presence of cis-acting {qtls} on the regulators, which induce close linkage among small neighborhoods of genes. {qtg} mapped regulator-target pairs linked to {arn1} were combined to form a regulatory module, which we observed to be highly enriched in iron homeostasis related genes and contained several causally directed links that had not been identified in other automatic reconstructions of that regulatory module. finally, we also confirm the surprising, previously published results that regulators controlling gene expression are not enriched for transcription factors, but we do show that our more precise mapping model reveals functional enrichment for several other biological processes related to the regulation of the cell. {conclusion}: by incorporating interacting expression and genotype, our {qtg} mapping method can identify specific regulator genes in contrast to standard {qtl} interval mapping. we have shown that the method can recover biologically significant regulator-target pairs and the approach leads to a general framework for inducing a regulatory module network topology of directed and undirected edges that can be used to identify leads in pathway analysis.accurate and efficient splicing of eukaryotic {pre-mrnas} requires recognition by trans-acting factors of a complex array of cis-acting {rna} elements. here, we developed a generalized bayesian network to model the coevolution of splicing cis elements in diverse eukaryotic taxa. cross-exon but not cross-intron compensatory interactions between the 5′ splice site (5′ss) and 3′ splice site (3′ss) were observed in human/mouse, indicating that the exon is the primary evolutionary unit in mammals. studied plants, fungi, and invertebrates exhibited exclusively cross-intron interactions, suggesting that intron definition drives evolution in these organisms. in mammals, 5′ss strength and the strength of several classes of exonic splicing silencers ({esss}) evolved in a correlated way, whereas specific exonic splicing enhancers ({eses}), including motifs associated with {htra2}, {srp55}, and {srp20}, evolved in a compensatory manner relative to the 5′ss and 3′ss. interactions between specific {ess} or {ese} motifs were not observed, suggesting that elements bound by different factors are not commonly interchangeable. thus, the splicing elements defining exons coevolve in a way that preserves overall exon strength, allowing specific elements to substitute for loss or weakening of others.to understand the genetic heterogeneity underlying developmental delay, we compared copy number variants ({cnvs}) in 15,767 children with intellectual disability and various congenital defects (cases) to {cnvs} in 8,329 unaffected adult controls. we estimate that ∼14.2\\\\% of disease in these children is caused by {cnvs} >400 kb. we observed a greater enrichment of {cnvs} in individuals with craniofacial anomalies and cardiovascular defects compared to those with epilepsy or autism. we identified 59 pathogenic {cnvs}, including 14 new or previously weakly supported candidates, refined the critical interval for several genomic disorders, such as the 17q21.31 microdeletion syndrome, and identified 940 candidate dosage-sensitive genes. we also developed methods to opportunistically discover small, disruptive {cnvs} within the large and growing diagnostic array datasets. this evolving {cnv} morbidity map, combined with exome and genome sequencing, will be critical for deciphering the genetic basis of developmental delay, intellectual disability and autism spectrum disorders.transcriptional and post-transcriptional regulation of gene expression is of fundamental importance to numerous biological processes. nowadays, an increasing amount of gene regulatory relationships have been documented in various databases and literature. however, to more efficiently exploit such knowledge for biomedical research and applications, it is necessary to construct a genome-wide regulatory network database to integrate the information on gene regulatory relationships that are widely scattered in many different places. therefore, in this work, we build a knowledge-based database, named \\'{regnetwork}\\', of gene regulatory networks for human and mouse by collecting and integrating the documented regulatory interactions among transcription factors ({tfs}), {micrornas} ({mirnas}) and target genes from 25 selected databases. moreover, we also inferred and incorporated potential regulatory relationships based on transcription factor binding site ({tfbs}) motifs into {regnetwork}. as a result, {regnetwork} contains a comprehensive set of experimentally observed or predicted transcriptional and post-transcriptional regulatory relationships, and the database framework is flexibly designed for potential extensions to include gene regulatory networks for other organisms in the future. based on {regnetwork}, we characterized the statistical and topological properties of genome-wide regulatory networks for human and mouse, we also extracted and interpreted simple yet important network motifs that involve the interplays between {tf}-{mirna} and their targets. in summary, {regnetwork} provides an integrated resource on the prior information for gene regulatory relationships, and it enables us to further investigate context-specific transcriptional and post-transcriptional regulatory interactions based on domain-specific experimental data.the synthesis of gene expression data and cis -regulatory analysis permits the elucidation of genomic regulatory networks. these networks provide a direct visualization of the functional interconnections among the regulatory genes and signaling components leading to cell-specific patterns of gene activity. complex developmental processes are thereby illuminated in ways not revealed by the conventional analysis of individual genes. in this review, we describe emerging networks in several different model systems, and compare them with the gene regulatory network that controls dorsoventral patterning of the drosophila embryo.a biweekly scientific journal publishing high-quality research in molecular biology and genetics, cancer biology, biochemistry, and related fieldsgene expression is regulated pre- and post-transcriptionally via cis-regulatory {dna} and {rna} motifs. identification of individual functional instances of such motifs in genome sequences is a major goal for inferring regulatory networks yet has been hampered due to the motifs\\' short lengths that lead to many chance matches and poor signal-to-noise ratios. in this paper, we develop a general methodology for the comparative identification of functional motif instances across many related species, using a phylogenetic framework that accounts for the evolutionary relationships between species, allows for motif movements, and is robust against missing data due to artifacts in sequencing, assembly, or alignment. we also provide a robust statistical framework for evaluating motif confidence, which enables us to translate evolutionary conservation into a confidence measure for each motif instance, correcting for varying motif length, composition, and background conservation of the target regions. we predict targets of fly transcription factors and {mirnas} in alignments of 12 recently sequenced drosophila species. when compared to extensive genome-wide experimental data, predicted targets are of high quality, matching and surpassing {chip}-chip microarrays and recovering {mirna} targets with high sensitivity. the resulting regulatory network suggests significant redundancy between pre- and post-transcriptional regulation of gene expression.we present here a new computational method for discovering cis-regulatory elements that circumvents the need to cluster genes based on their expression profiles. based on a model in which upstream motifs contribute additively to the log-expression level of a gene, this method requires a single genome-wide set of expression ratios and the upstream sequence for each gene, and outputs statistically significant motifs. analysis of publicly available expression data for saccharomyces cerevisiae reveals several new putative regulatory elements, some of which plausibly control the early, transient induction of genes during sporulation. known motifs generally have high statistical significance.understanding how sets of genes are coordinately regulated in space and time to generate the diversity of cell types that characterise complex metazoans is a major challenge in modern biology. the use of high-throughput approaches, such as large-scale in situ hybridisation and genome-wide expression profiling via {dna} microarrays, is beginning to provide insights into the complexities of development. however, in many organisms the collection and annotation of comprehensive in situ localisation data is a difficult and time-consuming task. here, we present a widely applicable computational approach, integrating developmental time-course microarray data with annotated in situ hybridisation studies, that facilitates the de novo prediction of tissue-specific expression for genes that have no in vivo gene expression localisation data available. using a classification approach, trained with data from microarray and in situ hybridisation studies of gene expression during drosophila embryonic development, we made a set of predictions on the tissue-specific expression of drosophila genes that have not been systematically characterised by in situ hybridisation experiments. the reliability of our predictions is confirmed by literature-derived annotations in {flybase}, by overrepresentation of gene ontology biological process annotations, and, in a selected set, by detailed gene-specific studies from the literature. our novel organism-independent method will be of considerable utility in enriching the annotation of gene function and expression in complex multicellular organisms.we present a method for learning sparse representations shared across multiple tasks. this method is a generalization of the well-known single-task 1-norm regularization. it is based on a novel non-convex regularizer which controls the number of learned features common across the tasks. we prove that the method is equivalent to solving a convex optimization problem for which there is an iterative algorithm which converges to an optimal solution. the algorithm has a simple interpretation: it alternately performs a supervised and an unsupervised step, where in the former step it learns task-specific functions and in the latter step it learns common-across-tasks sparse representations for these functions. we also provide an extension of the algorithm which learns sparse nonlinear representations using kernels. we report experiments on simulated and real data sets which demonstrate that the proposed method can both improve the performance relative to learning each task independently and lead to a few learned features common across related tasks. our algorithm can also be used, as a special case, to simply select--not learn--a few common variables across the tasks.the human genome contains hundreds of regions whose patterns of genetic variation indicate recent positive natural selection, yet for most the underlying gene and the advantageous mutation remain unknown. we developed a method, composite of multiple signals ({cms}), that combines tests for multiple signals of selection and increases resolution by up to 100-fold. by applying {cms} to candidate regions from the international haplotype map, we localized population-specific selective signals to 55 kilobases (median), identifying known and novel causal variants. {cms} can not just identify individual loci but implicates precise variants selected by evolution.dedicated to professor l. l. campbell, in celebration of his longstanding career in mathematics and statistics and in tribute to his many scholarly contributions to information theory. an overview is given of the several methods in use for the nonparametric estimation of the differential entropy of a continuous random variable. the properties of various methods are compared. several applications are given such as tests for goodness-of-fit, parameter estimation, quantization theory and spectral estimation. imotivation: microarray gene expression data has increasingly become the common data source that can provide insights into biological processes at a system-wide level. one of the major problems with microarrays is that a dataset consists of relatively few time points with respect to a large number of genes, which makes the problem of inferring gene regulatory network an ill-posed one. on the other hand, gene expression data generated by different groups worldwide are increasingly accumulated on many species and can be accessed from public databases or individual websites, although each experiment has only a limited number of {time-points.results}: this paper proposes a novel method to combine multiple time-course microarray datasets from different conditions for inferring gene regulatory networks. the proposed method is called {gnr} (gene network reconstruction tool) which is based on linear programming and a decomposition procedure. the method theoretically ensures the derivation of the most consistent network structure with respect to all of the datasets, thereby not only significantly alleviating the problem of data scarcity but also remarkably improving the prediction reliability. we tested {gnr} using both simulated data and experimental data in yeast and arabidopsis. the result demonstrates the effectiveness of {gnr} in terms of predicting new gene regulatory relationship in yeast and {arabidopsis.availability}: the software is available from http://zhangorup.aporc.org/bioinfo/grninfer/, http://digbio.missouri.edu/grninfer/ and http://intelligent.eic.osaka-sandai.ac.jp or upon request from the {authors.contact}:chen@eic.osaka-sandai.ac.jp, xudong@missouri.edu, {zxs@amt.ac.cnsupplementary} information: supplementary data are available at bioinformatics online.the human body contains several hundred cell types, all of which share the same genome. in metazoans, much of the regulatory code that drives cell type-specific gene expression is located in distal elements called enhancers. although mammalian genomes contain millions of potential enhancers, only a small subset of them is active in a given cell type. cell type-specific enhancer selection involves the binding of lineage-determining transcription factors that prime enhancers. signal-dependent transcription factors bind to primed enhancers, which enables these broadly expressed factors to regulate gene expression in a cell type-specific manner. the expression of genes that specify cell type identity and function is associated with densely spaced clusters of active enhancers known as super-enhancers. the functions of enhancers and super-enhancers are influenced by, and affect, higher-order genomic organization.motivation: time series expression experiments are used to study a wide range of biological systems. more than 80\\\\% of all time series expression datasets are short (8 time points or fewer). these datasets present unique challenges. on account of the large number of genes profiled (often tens of thousands) and the small number of time points many patterns are expected to arise at random. most clustering algorithms are unable to distinguish between real and random patterns.we present a method to find motifs by simultaneously using the overrepresentation property and the evolutionary conservation property of motifs. this method is applicable to divergent species where alignment is unreliable, which overcomes a major limitation of the current methods. the method has been applied to search regulatory motifs in four yeast species based on {chip}-chip data in saccharomyces cerevisiae and obtained 20\\\\% higher accuracy than the best current methods. we also discovered cis-regulatory elements that govern the tight regulation of ribosomal protein genes in two distantly related insects by using this method. these results demonstrate that our method will be useful for the extraction of regulatory signals in multiple genomes.in this study, a reverse-engineering strategy was used to infer and analyze the structure and function of an aging and glucose repressed gene regulatory network in the budding yeast saccharomyces cerevisiae. the method uses transcriptional perturbations to model the functional interactions between genes as a system of first-order ordinary differential equations. the resulting network model correctly identified the known interactions of key regulators in a 10-gene network from the snf1 signaling pathway, which is required for expression of glucose-repressed genes upon calorie restriction. the majority of interactions predicted by the network model were confirmed using promoter-reporter gene fusions in gene-deletion mutants and chromatin immunoprecipitation experiments, revealing a more complex network architecture than previously appreciated. the reverse-engineered network model also predicted an unexpected role for transcriptional regulation of the {snf1} gene by hexose kinase enzyme/transcriptional repressor hxk2, mediator subunit med8, and transcriptional repressor mig1. these interactions were validated experimentally and used to design new experiments demonstrating snf1 and its transcriptional regulators hxk2 and mig1 as modulators of chronological lifespan. this work demonstrates the value of using network inference methods to identify and characterize the regulators of complex phenotypes, such as aging.{background}:as carbon sources are exhausted, saccharomyces cerevisiae cells exhibit reduced metabolic activity and cultures enter the stationary phase. we asked whether cells in stationary phase cultures respond to additional stress at the level of transcript {abundance.results}:microarrays were used to quantify changes in transcript abundance in cells from stationary phase cultures in response to stress. more than 800 {mrnas} increased in abundance by one minute after oxidative stress. a significant number of these {mrnas} encode proteins involved in stress responses. we tested whether {mrna} increases were due to new transcription, rapid poly-adenylation of message (which would not be detected by microarrays), or potential release of mature {mrna} present in the cell but resistant to extraction during {rna} isolation. examination of the response to oxidative stress in an {rna} polymerase {ii} mutant, rpb1-1, suggested that new transcription was not required. quantitative {rt}-{pcr} analysis of a subset of these transcripts further suggested that the transcripts present in isolated total {rna} from stationary phase cultures were polyadenylated. in contrast, over 2,000 transcripts increased after protease treatment of cell-free lysates from stationary phase but not exponentially growing cultures. different subsets of transcripts were released by oxidative stress and temperature upshift, suggesting that {mrna} release is {stress-specific.conclusions}:cells in stationary phase cultures contain a large number of extraction-resistant {mrnas} in a protease-labile, rapidly releasable form. the transcript release appears to be stress-specific. we hypothesize that these transcripts are associated with p-bodies.the pluripotent state of embryonic stem cells ({escs}) is produced by active transcription of genes that control cell identity and repression of genes encoding lineage-specifying developmental regulators. here, we use {esc} cohesin {chia}-{pet} data to identify the local chromosomal structures at both active and repressed genes across the genome. the results produce a map of enhancer-promoter interactions and reveal that super-enhancer-driven genes generally occur within chromosome structures that are formed by the looping of two interacting {ctcf} sites co-occupied by cohesin. these looped structures form insulated neighborhoods whose integrity is important for proper expression of local genes. we also find that repressed genes encoding lineage-specifying developmental regulators occur within insulated neighborhoods. these results provide insights into the relationship between transcriptional control of cell identity genes and control of local chromosome structure.eukaryotic chromosomes replicate in a temporal order known as the replication-timing program. in mammals, replication timing is cell-type-specific with at least half the genome switching replication timing during development, primarily in units of 400\\\\&\\\\#19;800 kilobases (\\\\&\\\\#24;replication domains\\\\&\\\\#25;), whose positions are preserved in different cell types, conserved between species, and appear to confine long-range effects of chromosome rearrangements. early and late replication correlate, respectively, with open and closed three-dimensional chromatin compartments identified by high-resolution chromosome conformation capture ({hi-c}), and, to a lesser extent, late replication correlates with lamina-associated domains ({lads}). recent {hi-c} mapping has unveiled substructure within chromatin compartments called topologically associating domains ({tads}) that are largely conserved in their positions between cell types and are similar in size to replication domains. however, {tads} can be further sub-stratified into smaller domains, challenging the significance of structures at any particular scale. moreover, attempts to reconcile {tads} and {lads} to replication-timing data have not revealed a common, underlying domain structure. here we localize boundaries of replication domains to the early-replicating border of replication-timing transitions and map their positions in 18 human and 13 mouse cell types. we demonstrate that, collectively, replication domain boundaries share a near one-to-one correlation with {tad} boundaries, whereas within a cell type, adjacent {tads} that replicate at similar times obscure replication domain boundaries, largely accounting for the previously reported lack of alignment. moreover, cell-type-specific replication timing of {tads} partitions the genome into two large-scale sub-nuclear compartments revealing that replication-timing transitions are indistinguishable from late-replicating regions in chromatin composition and lamina association and accounting for the reduced correlation of replication timing to {lads} and heterochromatin. our results reconcile cell-type-specific sub-nuclear compartmentalization and replication timing with developmentally stable structural domains and offer a unified model for large-scale chromosome structure and function.the flood of genome-wide data generated by high-throughput technologies currently provides biologists with an unprecedented opportunity: to manipulate, query, and reconstruct functional molecular networks of cells. here, we outline three underlying principles and six strategies to infer network models from genomic data. then, using cancer as an example, we describe experimental and computational approaches to infer \"differential\" networks that can identify genes and processes driving disease phenotypes. in conclusion, we discuss how a network-level understanding of cancer can be used to predict drug response and guide therapeutics. copyright {\\\\copyright} 2011 elsevier inc. all rights reserved.quiescence is the most common and, arguably, most poorly understood cell cycle state. this is in part because pure populations of quiescent cells are typically difficult to isolate. we report the isolation and characterization of quiescent and nonquiescent cells from stationary-phase ({sp}) yeast cultures by density-gradient centrifugation. quiescent cells are dense, unbudded daughter cells formed after glucose exhaustion. they synchronously reenter the mitotic cell cycle, suggesting that they are in a g(0) state. nonquiescent cells are less dense, heterogeneous, and composed of replicatively older, asynchronous cells that rapidly lose the ability to reproduce. microscopic and flow cytometric analysis revealed that nonquiescent cells accumulate more reactive oxygen species than quiescent cells, and over 21 d, about half exhibit signs of apoptosis and necrosis. the ability to isolate both quiescent and nonquiescent yeast cells from {sp} cultures provides a novel, tractable experimental system for studies of quiescence, chronological and replicative aging, apoptosis, and the cell cycle.cells rapidly alter gene expression in response to environmental stimuli such as nutrients, hormones, and drugs. during the imposed  ” remodeling” of gene expression, changes in the levels of particular {mrnas} do not necessarily correlate with those of the encoded proteins, which could in part rely on the differential recruitment of {mrnas} to translating ribosomes. to systematically address this issue, we have established an approach to rapidly access the translational status of each {mrna} in the yeast saccharomyces cerevisiae by affinity purification of endogenously formed ribosomes and the analysis of associated {mrnas} with {dna} microarrays. using this method, we compared changes in total {mrna} levels (transcriptome) with ribosome associations (translatome) after the application of different conditions of cellular stress. severe stresses, induced by amino acid depletion or osmotic shock, stimulated highly correlated responses affecting about 15\\\\% of both total {rna} levels and translatome. many of the regulated messages code for functionally related proteins, thus reflecting logical responses to the particular stress. in contrast, mild stress provoked by addition of calcofluor-white and menadione altered the translatome of approximately 1\\\\% of messages with only marginal effects on total {mrna}, suggesting largely uncorrelated responses of transcriptome and translatome. among these putative translationally regulated messages were most components of the mitochondrial {atpase}. increased polysome associations of corresponding messages and higher mitochondrial {atpase} activities upon treatment confirmed the relevance for regulation of this macromolecular complex. our results suggest the presence of highly sensitive translational regulatory networks that coordinate functionally related messages. these networks are preferentially activated for rapid adaptation of cells to minor environmental perturbations. organisms respond to environmental or physiological changes by altering the amounts and activities of specific proteins that are necessary for their adaptation and survival. importantly, protein levels can be modulated by changing either the rate of synthesis or the stability of the messenger {rna} ({mrna} or transcript), or the synthesis or stability of the protein itself. scientists often measure global {mrna} levels upon changing conditions to identify transcripts that are differentially regulated, and often the assumption is made that changes in transcript levels lead to corresponding changes in protein levels. here, we systematically compared global transcript levels (transcriptome) with global alterations in the levels of ribosome association of transcripts (translatome) when yeast cells are exposed to different stresses to determine how significant the discrepancy between transcript and protein levels can be. we found that changes in the transcriptome correlate well with those in the translatome after application of harsh stresses that arrest cell growth. however, this correlation is generally lost under more mild stresses that do not affect cell growth. in this case, remodeling of gene expression is mainly executed at the translational level by modulating {mrna} association with ribosomes. as one example, we show that expression for many components of the mitochondrial {atpase}, the major energy production machinery in cells, is translationally but not transcriptionally activated under a specific mild stress condition. our results therefore show that alteration of protein synthesis can be the dominant mediator of changes of gene expression during adaptation to minor changes in cellular needs.the state of the transcriptome reflects a balance between {mrna} production and degradation. yet how these two regulatory arms interact in shaping the kinetics of the transcriptome in response to environmental changes is not known. we subjected yeast to two stresses, one that induces a fast and transient response, and another that triggers a slow enduring response. we then used microarrays following transcriptional arrest to measure genome-wide decay profiles under each condition. we found condition-specific changes in {mrna} decay rates and coordination between {mrna} production and degradation. in the transient response, most induced genes were surprisingly destabilized, whereas repressed genes were somewhat stabilized, exhibiting counteraction between production and degradation. this strategy can reconcile high steady-state level with short response time among induced genes. in contrast, the stress that induces the slow response displays the more expected behavior, whereby most induced genes are stabilized, and repressed genes are destabilized. our results show genome-wide interplay between {mrna} production and degradation, and that alternative modes of such interplay determine the kinetics of the transcriptome in response to stress.the activation of {cd4}+ t lymphocytes by antigen initiates adaptive immune responses, amplifying rare antigen-specific clones and leading to functional specialization of effector t cells, in particular through the spectrum of cytokines they produce. pathogens have exerted selective pressure during recent human evolution and migrations, but selection for a type of response that is optimal against one microbe class could carry a price in less-effective responses to other microbes or in impaired self-tolerance and autoimmune disease. little is known about interindividual variation in the responsiveness of human {cd4}+ t cells and how genetic variation affects the tone and intensity of t cell responses in human individuals.motivation: comparing the genomes from closely related species provides a powerful tool to identify functional elements in a reference genome. many methods have been developed to identify conserved sequences across species; however, existing methods only model conservation as a decrease in the rate of mutation and have ignored selection acting on the pattern of mutations.noncoding sequence contains pathogenic mutations. yet, compared with mutations in protein-coding sequence, pathogenic regulatory mutations are notoriously difficult to recognize. most fundamentally, we are not yet adept at recognizing the sequence stretches in the human genome that are most important in regulating the expression of genes. for this reason, it is difficult to apply to the regulatory regions the same kinds of analytical paradigms that are being successfully applied to identify mutations among protein-coding regions that influence risk. to determine whether dosage sensitive genes have distinct patterns among their noncoding sequence, we present two primary approaches that focus solely on a gene\\'s proximal noncoding regulatory sequence. the first approach is a regulatory sequence analogue of the recently introduced residual variation intolerance score ({rvis}), termed noncoding {rvis}, or {ncrvis}. the {ncrvis} compares observed and predicted levels of standing variation in the regulatory sequence of human genes. the second approach, termed {ncgerp}, reflects the phylogenetic conservation of a gene\\'s regulatory sequence using {gerp}++. we assess how well these two approaches correlate with four gene lists that use different ways to identify genes known or likely to cause disease through changes in expression: 1) genes that are known to cause disease through haploinsufficiency, 2) genes curated as dosage sensitive in {clingen}\\'s genome dosage map, 3) genes judged likely to be under purifying selection for mutations that change expression levels because they are statistically depleted of loss-of-function variants in the general population, and 4) genes judged unlikely to cause disease based on the presence of copy number variants in the general population. we find that both noncoding scores are highly predictive of dosage sensitivity using any of these criteria. in a similar way to {ncgerp}, we assess two ensemble-based predictors of regional noncoding importance, {nccadd} and {ncgwava}, and find both scores are significantly predictive of human dosage sensitive genes and appear to carry information beyond conservation, as assessed by {ncgerp}. these results highlight that the intolerance of noncoding sequence stretches in the human genome can provide a critical complementary tool to other genome annotation approaches to help identify the parts of the human genome increasingly likely to harbor mutations that influence risk of disease. mutations in noncoding sequence can cause disease but are very difficult to recognize. here, we present two approaches intended to help identify noncoding regions of the genome that may carry mutations influencing disease. the first approach is based on comparing observed and predicted levels of standing human variation in the noncoding exome sequence of a gene. the second approach is based on the phylogenetic conservation of a gene\\'s noncoding exome sequence using {gerp}++. we find that both approaches can predict genes known to cause disease through changes in expression level, genes depleted of loss-of-function alleles in the general population, and genes permissive of copy number variants in the general population. we find that both scores aid in interpreting loss-of-function mutations and in defining regions of noncoding sequence that are more likely to harbor mutations that influence risk of disease.according to the disease module hypothesis, the cellular components associated with a disease segregate in the same neighborhood of the human interactome, the map of biologically relevant molecular interactions. yet, given the incompleteness of the interactome and the limited knowledge of disease-associated genes, it is not obvious if the available data have sufficient coverage to map out modules associated with each disease. here we derive mathematical conditions for the identifiability of disease modules and show that the network-based location of each disease module determines its pathobiological relationship to other diseases. for example, diseases with overlapping network modules show significant coexpression patterns, symptom similarity, and comorbidity, whereas diseases residing in separated network neighborhoods are phenotypically distinct. these tools represent an interactome-based platform to predict molecular commonalities between phenotypically related diseases, even if they do not share primary disease genes. copyright {\\\\copyright} 2015, american association for the advancement of science.based on large-scale data for the yeast saccharomyces cerevisiae (protein and {mrna} abundance, translational status, transcript length), we investigate the relation of transcription, translation, and protein turnover on a genome-wide scale. we elucidate variations between different spatial cell compartments and functional modules by comparing {protein-to-mrna} ratios, translational activity, and a novel descriptor for protein-specific degradation (protein half-life descriptor). this analysis helps to understand the cell\\'s strategy to use transcriptional and post-transcriptional regulation mechanisms for managing protein levels. for instance, it is possible to identify modules that are subject to suppressed translation under normal conditions ( ” translation on demand”). in order to reduce inconsistencies between the datasets, we compiled a new reference {mrna} abundance dataset and we present a novel approach to correct large microarray signals for a saturation bias. accounting for ribosome density based on transcript length rather than {orf} length improves the correlation of observed protein levels to translational activity. we discuss potential causes for the deviations of these correlations. finally, we introduce a quantitative descriptor for protein degradation (protein half-life descriptor) and compare it to measured half-lives. the study demonstrates significant post-transcriptional control of protein levels for a number of different compartments and functional modules, which is missed when exclusively focusing on transcript levels.regulation of gene expression is usually separated into cis and trans components. the separation may become artificial if much of the variation in expression is under multigenic and epistatic (e.g., cis-by-trans) control. there is hence a need to quantify the relative contribution of cis, trans, and cis-by-trans effects on expression divergence at different levels of evolution. to do so across the whole genome, we analyzed the full set of chromosome-substitution lines between the two behavioral races of drosophila melanogaster. our observations: (i) only approximately 3\\\\% of the genes with an expression difference are purely cis regulated. in fact, relatively few genes are governed by simple genetics because nearly 80\\\\% of expression differences are controlled by at least two chromosomes. (ii) for 14\\\\% of the genes, cis regulation does play a role but usually in conjunction with trans regulation. this joint action of cis and trans effects, either additive or epistatic, is referred to as inclusive cis effect. (iii) the percentage of genes with inclusive cis effect increases to 32\\\\% among genes that are strongly differentiated between the two races. (iv) we observed a nonrandom distribution of trans-acting factors, with a substantial deficit on the second chromosome. between drosophila racial groups, trans regulation of expression difference is extensive, and cis regulation often evolves in conjunction with trans effects.although the locations of promoters and enhancers have been identified in several cell types, we still have limited information on their connectivity. we developed {hicap}, which combines a 4-cutter restriction enzyme {hi-c} with sequence capture of promoter regions. applying the method to mouse embryonic stem cells, we identified promoter-anchored interactions involving 15,905 promoters and 71,984 distal regions. the distal regions were enriched for enhancer marks and transcription, and had a mean fragment size of only 699 bp — close to single-enhancer resolution. high-resolution maps of promoter-anchored interactions with {hicap} will be important for detailed characterizations of chromatin interaction landscapes.differences in expression, protein interactions, and {dna} binding of paralogous transcription factors (\"{tf} parameters\") are thought to be important determinants of regulatory and biological specificity. however, both the extent of {tf} divergence and the relative contribution of individual {tf} parameters remain undetermined. we comprehensively identify dimerization partners, spatiotemporal expression patterns, and {dna}-binding specificities for the c. elegans {bhlh} family of {tfs}, and model these data into an integrated network. this network displays both specificity and promiscuity, as some {bhlh} proteins, {dna} sequences, and tissues are highly connected, whereas others are not. by comparing all {bhlh} {tfs}, we find extensive divergence and that all three parameters contribute equally to {bhlh} divergence. our approach provides a framework for examining divergence for other protein families in c. elegans and in other complex multicellular organisms, including humans. cross-species comparisons of integrated networks may provide further insights into molecular features underlying protein family evolution. for a video summary of this article, see the {paperflick} file available with the online supplemental data.deciphering gene regulatory mechanisms through the analysis of high-throughput expression data is a challenging computational problem. previous computational studies have used large expression datasets in order to resolve fine patterns of coexpression, producing clusters or modules of potentially coregulated genes. these methods typically examine promoter sequence information, such as {dna} motifs or transcription factor occupancy data, in a separate step after clustering. we needed an alternative and more integrative approach to study the oxygen regulatory network in saccharomyces cerevisiae using a small dataset of perturbation experiments. mechanisms of oxygen sensing and regulation underlie many physiological and pathological processes, and only a handful of oxygen regulators have been identified in previous studies. we used a new machine learning algorithm called {medusa} to uncover detailed information about the oxygen regulatory network using genome-wide expression changes in response to perturbations in the levels of oxygen, heme, hap1, and co2+. {medusa} integrates {mrna} expression, promoter sequence, and {chip}-chip occupancy data to learn a model that accurately predicts the differential expression of target genes in held-out data. we used a novel margin-based score to extract significant condition-specific regulators and assemble a global map of the oxygen sensing and regulatory network. this network includes both known oxygen and heme regulators, such as hap1, mga2, hap4, and upc2, as well as many new candidate regulators. {medusa} also identified many {dna} motifs that are consistent with previous experimentally identified transcription factor binding sites. because {medusa}\\'s regulatory program associates regulators to target genes through their promoter sequences, we directly tested the predicted regulators for {ole1}, a gene specifically induced under hypoxia, by experimental analysis of the activity of its promoter. in each case, deletion of the candidate regulator resulted in the predicted effect on promoter activity, confirming that several novel regulators identified by {medusa} are indeed involved in oxygen regulation. {medusa} can reveal important information from a small dataset and generate testable hypotheses for further experimental analysis. supplemental data are included. the cell uses complex regulatory networks to modulate the expression of genes in response to changes in cellular and environmental conditions. the transcript level of a gene is directly affected by the binding of transcriptional regulators to {dna} motifs in its promoter sequence. therefore, both expression levels of transcription factors and other regulatory proteins as well as sequence information in the promoters contribute to transcriptional gene regulation. in this study, we describe a new computational strategy for learning gene regulatory programs from gene expression data based on the {medusa} algorithm. we learn a model that predicts differential expression of target genes from the expression levels of regulators, the presence of {dna} motifs in promoter sequences, and binding data for transcription factors. unlike many previous approaches, we do not assume that genes are regulated in clusters, and we learn {dna} motifs de novo from promoter sequences as an integrated part of our algorithm. we use {medusa} to produce a global map of the yeast oxygen and heme regulatory network. to demonstrate that {medusa} can reveal detailed information about regulatory mechanisms, we perform biochemical experiments to confirm the predicted regulators for an important hypoxia gene.motivation: finding suitable models of dynamic biochemical systems is an important task in systems biology approaches to the biosciences. on the one hand, a correct model helps to understand the underlying mechanisms and on the other hand, one can use the model to predict the behavior of a biological system under various circumstances. typically, before the correct model of a biochemical system is found, different hypothetical models might be reasonable and consistent with previous knowledge and available data. the main goal now is to find the best suited model out of different hypotheses. the process of falsifying inappropriate candidate models is called model discrimination.significant cell-to-cell heterogeneity is ubiquitously observed in isogenic cell populations. consequently, parameters of models of intracellular processes, usually fitted to population-averaged data, should rather be fitted to individual cells to obtain a population of models of similar but non-identical individuals. here, we propose a quantitative modeling framework that attributes specific parameter values to single cells for a standard model of gene expression. we combine high quality single-cell measurements of the response of yeast cells to repeated hyperosmotic shocks and state-of-the-art statistical inference approaches for mixed-effects models to infer multidimensional parameter distributions describing the population, and then derive specific parameters for individual cells. the analysis of single-cell parameters shows that single-cell identity (e.g. gene expression dynamics, cell size, growth rate, mother-daughter relationships) is, at least partially, captured by the parameter values of gene expression models (e.g. rates of transcription, translation and degradation). our approach shows how to use the rich information contained into longitudinal single-cell data to infer parameters that can faithfully represent single-cell identity. because of non-genetic variability, cells in an isogenic population respond differently to a same stimulation. therefore, the mean behavior of a cell population does not generally correspond to the behavior of the mean cell, and more generally, neglecting cell-to-cell differences biases our quantitative representation and understanding of the functioning of cellular systems. here we introduce a statistical inference approach allowing for the calibration of (a population of) single cell models, differing by their parameter values. it enables to view time-lapse microscopy data as many experiments performed on one cell rather than one experiment performed on many cells. by harnessing existing cell-to-cell differences, one can then learn how environmental cues affect (non-observed) intracellular processes. our approach is generic and enables to exploit in unprecedented manner the high informative content of single-cell longitudinal data.changes in cis-regulatory element composition that result in novel patterns of gene expression are thought to be a major contributor to the evolution of lineage-specific traits. although transcription factor binding events show substantial variation across species, most computational approaches to study regulatory elements focus primarily upon highly conserved sites, and rely heavily upon multiple sequence alignments. however, sequence conservation based approaches have limited ability to detect lineage-specific elements that could contribute to species-specific traits. in this paper, we describe a novel framework that utilizes a birth-death model to trace the evolution of lineage-specific binding sites without relying on detailed base-by-base cross-species alignments. our model was applied to analyze the evolution of binding sites based on the {chip}-seq data for six transcription factors ({gata1}, {sox2}, {ctcf}, {myc}, {max}, {ets1}) along the lineage toward human after human-mouse common ancestor. we estimate that a substantial fraction of binding sites (∼58-79\\\\% for each factor) in humans have origins since the divergence with mouse. over 15\\\\% of all binding sites are unique to hominids. such elements are often enriched near genes associated with specific pathways, and harbor more common {snps} than older binding sites in the human genome. these results support the ability of our method to identify lineage-specific regulatory elements and help understand their roles in shaping variation in gene regulation across species.most organisms reproduce through outcrossing, even though it comes with substantial costs. the red queen hypothesis proposes that selection from coevolving pathogens facilitates the persistence of outcrossing despite these costs. we used experimental coevolution to test the red queen hypothesis and found that coevolution with a bacterial pathogen (serratia marcescens) resulted in significantly more outcrossing in mixed mating experimental populations of the nematode caenorhabditis elegans. furthermore, we found that coevolution with the pathogen rapidly drove obligately selfing populations to extinction, whereas outcrossing populations persisted through reciprocal coevolution. thus, consistent with the red queen hypothesis, coevolving pathogens can select for biparental sex.numerically, microbial species dominate the oceans, yet their population dynamics, metabolic complexity and synergistic interactions remain largely uncharted. a full understanding of life in the ocean requires more than knowledge of marine microbial taxa and their genome sequences. the latest experimental techniques and analytical approaches can provide a fresh perspective on the biological interactions within marine ecosystems, aiding in the construction of predictive models that can interrelate microbial dynamics with the biogeochemical matter and energy fluxes that make up the ocean ecosystem.to celebrate the first 10 years of nature reviews genetics, we asked eight leading researchers for their views on the key developments in genetics and genomics in the past decade and the prospects for the future. their responses highlight the incredible changes that the field has seen, from the explosion of genomic data and the many possibilities it has opened up to the ability to reprogramme adult cells to pluripotency. the way ahead looks similarly exciting as we address questions such as how cells function as systems and how complex interactions among genetics, epigenetics and the environment combine to shape phenotypes.gene prioritization aims at identifying the most promising candidate genes among a large pool of candidates-so as to maximize the yield and biological relevance of further downstream validation experiments and functional studies. during the past few years, several gene prioritization tools have been defined, and some of them have been implemented and made available through freely available web tools. in this study, we aim at comparing the predictive performance of eight publicly available prioritization tools on novel data. we have performed an analysis in which 42 recently reported disease-gene associations from literature are used to benchmark these tools before the underlying databases are updated. cross-validation on retrospective data provides performance estimate likely to be overoptimistic because some of the data sources are contaminated with knowledge from disease-gene association. our approach mimics a novel discovery more closely and thus provides more realistic performance estimates. there are, however, marked differences, and tools that rely on more advanced data integration schemes appear more powerful. yves.moreau@esat.kuleuven.be supplementary data are available at bioinformatics online.the kinetics of label uptake and dilution in dividing stem cells, e.g., using bromodeoxyuridine ({brdu}) as a labeling substance, are a common way to assess the cellular turnover of all hematopoietic stem cells ({hscs}) in vivo. the assumption that {hscs} form a homogeneous population of cells which regularly undergo cell division has recently been challenged by new experimental results. for a consistent functional explanation of heterogeneity among {hscs}, we propose a concept in which stem cells flexibly and reversibly adapt their cycling state according to systemic needs. applying a mathematical model analysis, we demonstrate that different experimentally observed label dilution kinetics are consistently explained by the proposed model. the dynamically stabilized equilibrium between quiescent and activated cells leads to a biphasic label dilution kinetic in which an initial and pronounced decline of label retaining cells is attributed to faster turnover of activated cells, whereas a secondary, decelerated decline results from the slow turnover of quiescent cells. these results, which support our previous model prediction of a reversible activation/deactivation of {hscs}, are also consistent with recent findings that use {gfp}-conjugated histones as a label instead of {brdu}. based on our findings we interpret {hsc} organization as an adaptive and regulated process in which the slow activation of quiescent cells and their possible return into quiescence after division are sufficient to explain the simultaneous occurrence of self-renewal and differentiation. furthermore, we suggest an experimental strategy which is suited to demonstrate that the repopulation ability among the population of label retaining cells changes during the course of dilution.the abundance and identity of functional variation segregating in natural populations is paramount to dissecting the molecular basis of quantitative traits as well as human genetic diseases. genome sequencing of multiple organisms of the same species provides an efficient means of cataloging rearrangements, insertion, or deletion polymorphisms ({indels}) and single-nucleotide polymorphisms ({snps}). while inbreeding depression and heterosis imply that a substantial amount of polymorphism is deleterious, distinguishing deleterious from neutral polymorphism remains a significant challenge. to identify deleterious and neutral {dna} sequence variation within saccharomyces cerevisiae, we sequenced the genome of a vineyard and oak tree strain and compared them to a reference genome. among these three strains, 6\\\\% of the genome is variable, mostly attributable to variation in genome content that results from large {indels}. out of the 88,000 polymorphisms identified, 93\\\\% are {snps} and a small but significant fraction can be attributed to recent interspecific introgression and ectopic gene conversion. in comparison to the reference genome, there is substantial evidence for functional variation in gene content and structure that results from large {indels}, frame-shifts, and polymorphic start and stop codons. comparison of polymorphism to divergence reveals scant evidence for positive selection but an abundance of evidence for deleterious {snps}. we estimate that 12\\\\% of coding and 7\\\\% of noncoding {snps} are deleterious. based on divergence among 11 yeast species, we identified 1,666 nonsynonymous {snps} that disrupt conserved amino acids and 1,863 noncoding {snps} that disrupt conserved noncoding motifs. the deleterious coding {snps} include those known to affect quantitative traits, and a subset of the deleterious noncoding {snps} occurs in the promoters of genes that show allele-specific expression, implying that some cis-regulatory {snps} are deleterious. our results show that the genome sequences of both closely and distantly related species provide a means of identifying deleterious polymorphisms that disrupt functionally conserved coding and noncoding sequences. {dna} sequence variation makes an important contribution to most traits that vary in natural populations. however, mapping mutations that underlie a trait of interest is a significant challenge. genome sequencing of multiple organisms provides a complete list of {dna} sequence differences responsible for any trait that differs among the organisms. yet, distinguishing those {dna} sequence variants that contribute to a trait from all other variants is not easy. here, we sequence the genomes of two strains of yeast and, through comparisons with a reference genome, we catalog multiple types of {dna} sequence variation among the three strains. using a variety of comparative genomics methods, we show that a substantial fraction of {dna} sequence variations has deleterious effects on fitness. finally, we show that a subset of deleterious mutations is associated with changes in gene expression levels. our results imply that comparative genomics methods will be a valuable approach to identifying {dna} sequence changes underlying numerous traits of interest.understanding the molecular details of the sequence of events in multistep evolutionary pathways can reveal the extent to which natural selection exploits regulatory mutations affecting expression, amino acid replacements affecting the active site, amino acid replacements affecting protein folding or stability, or variations affecting gene copy number. in experimentally exploring the adaptive landscape of the evolution of resistance to β-lactam antibiotics in enteric bacteria, we noted that a regulatory mutation that increases β-lactamase expression by about 2-fold has a very strong tendency to be fixed at or near the end of the evolutionary pathway. this pattern contrasts with previous experiments selecting for the utilization of novel substrates, in which regulatory mutations that increase expression are often fixed early in the process. to understand the basis of the difference, we carried out experiments in which the expression of β-lactamase was under the control of a tunable arabinose promoter. we find that the fitness effect of an increase in gene expression is highly dependent on the catalytic activity of the coding sequence. an increase in expression of an inefficient enzyme has a negligible effect on drug resistance; however, the effect of an increase in expression of an efficient enzyme is very large. the contrast in the temporal incorporation of regulatory mutants between antibiotic resistance and the utilization of novel substrates is related to the nature of the function that relates enzyme activity to fitness. a mathematical model of β-lactam resistance is examined in detail and shown to be consistent with the observed results.a pooling-based approach maps {qtls} for molecular-level traits with reduced cost thousands of cis-acting {qtls} affect transcription factor binding in humans {ctcf} anchors binding of multiple transcription factors binding {qtls} link genetic variation to {3d} genome architecture and complex traits cis-regulatory elements such as transcription factor ({tf}) binding sites can be identified genome-wide, but it remains far more challenging to pinpoint genetic variants affecting {tf} binding. here, we introduce a pooling-based approach to mapping quantitative trait loci ({qtls}) for molecular-level traits. applying this to five {tfs} and a histone modification, we mapped thousands of cis-acting {qtls}, with over 25-fold lower cost compared to standard {qtl} mapping. we\\xa0found that single genetic variants frequently affect binding of multiple {tfs}, and {ctcf} can recruit all five {tfs} to its binding sites. these {qtls} often affect local chromatin and transcription but can also\\xa0influence long-range chromosomal contacts, demonstrating a\\xa0role for natural genetic variation in\\xa0chromosomal architecture. thousands of these {qtls} have been implicated in genome-wide association studies, providing candidate molecular mechanisms for many disease risk loci and suggesting that {tf} binding variation may underlie a large fraction of human phenotypic variation.accurately translating genotype to phenotype requires accounting for the functional impact of genetic variation at many biological scales. here, we\\xa0present a strategy for genotype-phenotype reasoning based on existing knowledge of cellular subsystems. these subsystems and their hierarchical organization are defined by the gene ontology or a complementary ontology inferred directly from previously published datasets. guided by the ontology?s hierarchical structure, we organize genotype data into an ?ontotype,? that is, a hierarchy of perturbations representing the effects of genetic variation at multiple cellular scales. the ontotype is then interpreted using logical rules generated by machine learning to predict phenotype. this approach substantially outperforms previous non-hierarchical methods for translating yeast genotype to cell growth phenotype, and it accurately predicts the growth outcomes of two new screens of 2,503 double gene knockouts affecting {dna} repair or nuclear lumen. ontotypes also generalize to larger knockout combinations, setting the stage for interpreting the complex genetics of disease.genetic variants identified by genome-wide association studies explain only a modest proportion of heritability, suggesting that meaningful associations lie \\'hidden\\' below current thresholds. here, we integrate information from association studies with epigenomic maps to demonstrate that enhancers significantly overlap known loci associated with the cardiac {qt} interval and {qrs} duration. we apply functional criteria to identify loci associated with {qt} interval that do not meet genome-wide significance and are missed by existing studies. we demonstrate that these \\'sub-threshold\\' signals represent novel loci, and that epigenomic maps are effective at discriminating true biological signals from noise. we experimentally validate the molecular, gene-regulatory, cellular and organismal phenotypes of these sub-threshold loci, demonstrating that most sub-threshold loci have regulatory consequences and that genetic perturbation of nearby genes causes cardiac phenotypes in mouse. our work provides a general approach for improving the detection of novel loci associated with complex human traits.motivation: cluster analysis (of gene-expression data) is a useful tool for identifying biologically relevant groups of genes that show similar expression patterns under multiple experimental conditions. various methods have been proposed for clustering gene-expression data. however most of these algorithms have several shortcomings for gene-expression data clustering. in the present article, we focus on several shortcomings of conventional clustering algorithms and propose a new one that is able to produce better clustering solution than that produced by some {others.results}: we present the divisive correlation clustering algorithm ({dcca}) that is suitable for finding a group of genes having similar pattern of variation in their expression values. to detect clusters with high correlation and biological significance, we use the correlation clustering concept introduced by bansal et al. our proposed algorithm {dcca} produces a clustering solution without taking number of clusters to be created as an input. {dcca} uses the correlation matrix in such a way that all genes in a cluster have highest average correlation with genes in that cluster. to test the performance of the {dcca}, we have applied {dcca} and some well-known conventional methods to an artificial dataset, and nine gene-expression datasets, and compared the performance of the algorithms. the clustering results of the {dcca} are found to be more significantly relevant to the biological annotations than those of the other methods. all these facts show the superiority of the {dcca} over some others for the clustering of gene-expression {data.availability}: the software has been developed using c and visual basic languages, and can be executed on the microsoft windows platforms. the software may be downloaded as a zip file from http://www.isical.ac.in/\\\\~{}rajat. then it needs to be installed. two word files (included in the zip file) need to be consulted before installation and execution of the {software.contact}: {rajat@isical.ac.insupplementary} information: supplementary data are available at bioinformatics online.motivation: the identification of active transcriptional regulatory elements is crucial to understand regulatory networks driving cellular processes such as cell development and the onset of diseases. it has recently been shown that chromatin structure information, such as {dnase} i hypersensitivity ({dhs}) or histone modifications, significantly improves cell-specific predictions of transcription factor binding sites. however, no method has so far successfully combined both {dhs} and histone modification data to perform active binding site {prediction.results}: we propose here a method based on hidden markov models to integrate {dhs} and histone modifications occupancy for the detection of open chromatin regions and active binding sites. we have created a framework that includes treatment of genomic signals, model training and genome-wide application. in a comparative analysis, our method obtained a good trade-off between sensitivity versus specificity and superior area under the curve statistics than competing methods. moreover, our technique does not require further training or sequence information to generate binding location predictions. therefore, the method can be easily applied on new cell types and allow flexible downstream analysis such as de novo motif {finding.availability} and implementation: our framework is available as part of the regulatory genomics toolbox. the software information and all benchmarking data are available at {http://costalab.org/wp/dh-hmm.contact}: ivan.costa@rwth-aachen.de or {eduardo.gusmao@rwth-aachen.desupplementary} information: supplementary data are available at bioinformatics online.the establishment and maintenance of differential patterns of gene expression lie at the heart of development. how the precision of developmental gene regulation is achieved, despite the highly repetitive and complex nature of the mammalian genome, remains an important question. it is becoming increasingly clear that genetic regulation must be considered not only in the context of short- and long-range regulatory sequences and local chromatin structure, but also at the level of position within the nucleus. recent studies have addressed the extent to which the location of a gene relative to its interphase chromosome territory affects its regulation or its capacity to be expressed. two model systems have emphasized the role of this level of nuclear organization during development. hox gene clusters have provided important insights into the dynamic repositioning of a locus relative to its chromosome territory during spatial and temporal patterning of gene expression. the inactive x chromosome has also become a useful paradigm for studying the differential chromatin status and chromosomal organization of the two x\\'s within the same nucleus. recent work suggests that chromosome territory reorganisation can be an important step in the gene silencing process.statistical network modeling techniques are increasingly important tools to analyze cancer genomics data. however, current tools and resources are not designed to work across multiple diagnoses and technical platforms, thus limiting their applicability to comprehensive pan-cancer datasets such as the cancer genome atlas ({tcga}). to address this, we describe a new data driven modeling method, based on generalized sparse inverse covariance selection ({sics}). the method integrates genetic, epigenetic and transcriptional data from multiple cancers, to define links that are present in multiple cancers, a subset of cancers, or a single cancer. it is shown to be statistically robust and effective at detecting direct pathway links in data from {tcga}. to facilitate interpretation of the results, we introduce a publicly accessible tool (cancerlandscapes.org), in which the derived networks are explored as interactive web content, linked to several pathway and pharmacological databases. to evaluate the performance of the method, we constructed a model for eight {tcga} cancers, using data from 3900 patients. the model rediscovered known mechanisms and contained interesting predictions. possible applications include prediction of regulatory relationships, comparison of network modules across multiple forms of cancer and identification of drug targets.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsmodel-based analysis of regulation of gene expression ({marge}) is a framework for interpreting the relationship between the {h3k27ac} chromatin environment and differentially expressed gene sets. the framework has three main functions: {marge}-potential, {marge}-express, and {marge}-cistrome. {marge}-potential defines a regulatory potential ({rp}) for each gene as the sum of {h3k27ac} {chip}-seq signals weighted by a function of genomic distance from the transcription start site. the {marge} framework includes a compendium of {rps} derived from 365 human and 267 mouse {h3k27ac} {chip}-seq data sets. relative {rps}, scaled using this compendium, are superior to superenhancers in predicting {bet} (bromodomain and extraterminal domain) -inhibitor repressed genes. {marge}-express, which uses logistic regression to retrieve relevant {h3k27ac} profiles from the compendium to accurately model a query set of differentially expressed genes, was tested on 671 diverse gene sets from {msigdb}. {marge}-cistrome adopts a novel semisupervised learning approach to identify cis-regulatory elements regulating a gene set. {marge}-cistrome exploits information from {h3k27ac} signal at {dnase} i hypersensitive sites identified from published human and mouse {dnase}-seq data. we tested the framework on newly generated {rna}-seq and {h3k27ac} {chip}-seq profiles upon {sirna} silencing of multiple transcriptional and epigenetic regulators in a prostate cancer cell line, {lncap}-abl. {marge}-cistrome can predict the binding sites of silenced transcription factors without matched {h3k27ac} {chip}-seq data. even when the matching {h3k27ac} {chip}-seq profiles are available, {marge} leverages public {h3k27ac} profiles to enhance these data. this study demonstrates the advantage of integrating a large compendium of historical epigenetic data for genomic studies of transcriptional regulation.{interproscan} [e. m. zdobnov and r. apweiler (2001) bioinformatics, 17, 847-848] is a tool that combines different protein signature recognition methods from the {interpro} [n. j. mulder, r. apweiler, t. k. attwood, a. bairoch, a. bateman, d. binns, p. bradley, p. bork, p. bucher, l. cerutti et al. (2005) nucleic acids res., 33, {d201-d205}] consortium member databases into one resource. at the time of writing there are 10 distinct publicly available databases in the application. protein as well as {dna} sequences can be analysed. a web-based version is accessible for academic and commercial organizations from the {ebi} ({http://www.ebi.ac.uk/interproscan}/). in addition, a standalone perl version and a {soap} web service [j. snell, d. tidwell and p. kulchenko (2001) programming web services with {soap}, 1st edn. {o\\'reilly} publishers, sebastopol, {ca}, {http://www.w3.org/tr}/soap/] are also available to the users. various output formats are supported and include text tables, {xml} documents, as well as various graphs to help interpret the results.a defining characteristic of all eukaryotes is the presence of three {rna} polymerases, each of which transcribes a particular subset of nuclear genes. {rna} polymerase i transcribes {rrna} genes; {rna} polymerase {ii} transcribes {mrna}, {mirna}, {snrna}, and {snorna} genes; and {rna} polymerase {iii} transcribes {5s} {rrna} and {trna} genes. here, we use the sequences of up to 25 ascomycete species to show that the type of genes transcribed by each {rna} polymerase affects their evolutionary rates and those of their transcription factors ({tfs}). the {rna} polymerase subunits and {tfs} of genes whose promoters experience higher levels of concerted evolution evolve significantly faster than those experiencing lower levels of concerted evolution. the rates of evolution of {rna} polymerase genes and their {tfs} are therefore not only the result of diverse selective constraints but are also influenced by the level of concerted evolution of the genes they transcribe.the widespread adoption of {rna}-seq to quantitatively measure gene expression has increased the scope of sequencing experimental designs to include time-course experiments. {masigpro} is an r package specifically suited for the analysis of time-course gene expression data, which was developed originally for microarrays and hence was limited in its application to count data. we have updated {masigpro} to support {rna}-seq time series analysis by introducing generalized linear models in the algorithm to support the modeling of count data while maintaining the traditional functionalities of the package. we show a good performance of the {masigpro}-{glm} method in several simulated time-course scenarios and in a real experimental dataset. the package is freely available under the {lgpl} license from the bioconductor web site (http://bioconductor.org). {\\\\copyright} the author 2014. published by oxford university press.motivation: epistasis analysis is an essential tool of classical genetics for inferring the order of function of genes in a common pathway. typically, it considers single and double mutant phenotypes and for a pair of genes observes whether a change in the first gene masks the effects of the mutation in the second gene. despite the recent emergence of biotechnology techniques that can provide gene interaction data on a large, possibly genomic scale, few methods are available for quantitative epistasis analysis and epistasis-based network reconstruction.the immune system is a highly complex and dynamic system. historically, the most common scientific and clinical practice has been to evaluate its individual components. this kind of approach cannot always expose the interconnecting pathways that control immune-system responses and does not reveal how the immune system works across multiple biological systems and scales. high-throughput technologies can be used to measure thousands of parameters of the immune system at a genome-wide scale. these system-wide surveys yield massive amounts of quantitative data that provide a means to monitor and probe immune-system function. new integrative analyses can help synthesize and transform these data into valuable biological insight. here we review some of the computational analysis tools for high-dimensional data and how they can be applied to immunology.current human pluripotent stem cells lack the transcription factor circuitry that governs the ground state of mouse embryonic stem cells ({esc}). here, we report that short-term expression of two components, {nanog} and {klf2}, is sufficient to ignite other elements of the network and reset the human pluripotent state. inhibition of {erk} and protein kinase c sustains a transgene-independent rewired state. reset cells self-renew continuously without {erk} signaling, are phenotypically stable, and are karyotypically intact. they differentiate in\\xa0vitro and form teratomas in\\xa0vivo. metabolism is reprogrammed with activation of mitochondrial respiration as in {esc}. {dna} methylation is dramatically reduced and transcriptome state is globally realigned across multiple cell lines. depletion of ground-state transcription factors, {tfcp2l1} or {klf4}, has marginal impact on conventional human pluripotent stem cells but collapses the reset state. these findings demonstrate feasibility of installing and propagating functional control circuitry for ground-state pluripotency in human cells. copyright {\\\\copyright} 2014 the authors. published by elsevier inc. all rights reserved.a genetic interaction network containing ∼1000 genes and ∼4000 interactions was mapped by crossing mutations in 132 different query genes into a set of ∼4700 viable gene yeast deletion mutants and scoring the double mutant progeny for fitness defects. network connectivity was predictive of function because interactions often occurred among functionally related genes, and similar patterns of interactions tended to identify components of the same pathway. the genetic network exhibited dense local neighborhoods; therefore, the position of a gene on a partially mapped network is predictive of other genetic interactions. because digenic interactions are common in yeast, similar networks may underlie the complex genetics associated with inherited phenotypes in other organisms.although controlled biochemical or biological vocabularies, such as gene ontology ({go}) (http://www.geneontology.org), address the need for consistent descriptions of genes in different data sources, there is still no effective method to determine the functional similarities of genes based on gene annotation information from heterogeneous data sources. to address this critical need, we proposed a novel method to encode a {go} term\\'s semantics (biological meanings) into a numeric value by aggregating the semantic contributions of their ancestor terms (including this specific term) in the {go} graph and, in turn, designed an algorithm to measure the semantic similarity of {go} terms. based on the semantic similarities of {go} terms used for gene annotation, we designed a new algorithm to measure the functional similarity of genes. the results of using our algorithm to measure the functional similarities of genes in pathways retrieved from the saccharomyces genome database ({sgd}), and the outcomes of clustering these genes based on the similarity values obtained by our algorithm are shown to be consistent with human perspectives. furthermore, we developed a set of online tools for gene similarity measurement and knowledge discovery. the online tools are available at: {http://bioinformatics.clemson.edu/g}-{sesame}. {http://bioinformatics.clemson.edu/publication}/supplement/gsp.htm.we describe cell type-specific significance analysis of microarrays ({cssam}) for analyzing differential gene expression for each cell type in a biological sample from microarray data and relative cell-type frequencies. first, we validated {cssam} with predesigned mixtures and then applied it to whole-blood gene expression datasets from stable post-transplant kidney transplant recipients and those experiencing acute transplant rejection, which revealed hundreds of differentially expressed genes that were otherwise undetectable.type 2 diabetes mellitus is a complex disorder associated with multiple genetic, epigenetic, developmental, and environmental factors. animal models of type 2 diabetes differ based on diet, drug treatment, and gene knockouts, and yet all display the clinical hallmarks of hyperglycemia and insulin resistance in peripheral tissue. the recent advances in gene-expression microarray technologies present an unprecedented opportunity to study type 2 diabetes mellitus at a genome-wide scale and across different models. to date, a key challenge has been to identify the biological processes or signaling pathways that play significant roles in the disorder. here, using a network-based analysis methodology, we identified two sets of genes, associated with insulin signaling and a network of nuclear receptors, which are recurrent in a statistically significant number of diabetes and insulin resistance models and transcriptionally altered across diverse tissue types. we additionally identified a network of protein–protein interactions between members from the two gene sets that may facilitate signaling between them. taken together, the results illustrate the benefits of integrating high-throughput microarray studies, together with protein–protein interaction networks, in elucidating the underlying biological processes associated with a complex disorder. type 2 diabetes mellitus currently affects millions of people. it is clinically characterized by insulin resistance in addition to an impaired glucose response and associated with numerous complications including heart disease, stroke, neuropathy, and kidney failure, among others. accurate identification of the underlying molecular mechanisms of the disease or its complications is an important research problem that could lead to novel diagnostics and therapy. the main challenge stems from the fact that insulin resistance is a complex disorder and affects a multitude of biological processes, metabolic networks, and signaling pathways. in this report, the authors develop a network-based methodology that appears to be more sensitive than previous approaches in detecting deregulated molecular processes in a disease state. the methodology revealed that both insulin signaling and nuclear receptor networks are consistently and differentially expressed in many models of insulin resistance. the positive results suggest such network-based diagnostic technologies hold promise as potentially useful clinical and research tools in the future.mapping the pathways that give rise to metastasis is one of the key challenges of breast cancer research. recently, several large-scale studies have shed light on this problem through analysis of gene expression profiles to identify markers correlated with metastasis. here, we apply a protein-network-based approach that identifies markers not as individual genes but as subnetworks extracted from protein interaction databases. the resulting subnetworks provide novel hypotheses for pathways involved in tumor progression. although genes with known breast cancer mutations are typically not detected through analysis of differential expression, they play a central role in the protein network by interconnecting many differentially expressed genes. we find that the subnetwork markers are more reproducible than individual marker genes selected without network information, and that they achieve higher accuracy in the classification of metastatic versus non-metastatic tumors.to make a case for or against a trend in the evolution of complexity in biological evolution, complexity needs to be both rigorously defined and measurable. a recent information-theoretic (but intuitively evident) definition identifies genomic complexity with the amount of information a sequence stores about its environment. we investigate the evolution of genomic complexity in populations of digital organisms and monitor in detail the evolutionary transitions that increase complexity. we show that, because natural selection forces genomes to behave as a natural  ” maxwell demon,” within a fixed environment, genomic complexity is forced to increase.what is the lineage relation among the cells of an organism? the answer is sought by developmental biology, immunology, stem cell research, brain research, and cancer research, yet complete cell lineage trees have been reconstructed only for simple organisms such as caenorhabditis elegans. we discovered that somatic mutations accumulated during normal development of a higher organism implicitly encode its entire cell lineage tree with very high precision. our mathematical analysis of known mutation rates in microsatellites ({mss}) shows that the entire cell lineage tree of a human embryo, or a mouse, in which no cell is a descendent of more than 40 divisions, can be reconstructed from information on somatic {ms} mutations alone with no errors, with probability greater than 99.95\\\\%. analyzing all approximately 1.5 million {mss} of each cell of an organism may not be practical at present, but we also show that in a genetically unstable organism, analyzing only a few hundred {mss} may suffice to reconstruct portions of its cell lineage tree. we demonstrate the utility of the approach by reconstructing cell lineage trees from {dna} samples of a human cell line displaying {ms} instability. our discovery and its associated procedure, which we have automated, may point the way to a future \"human cell lineage project\" that would aim to resolve fundamental open questions in biology and medicine by reconstructing ever larger portions of the human cell lineage tree.related organisms typically respond to a given cue by altering the level or activity of orthologous transcription factors, which, paradoxically, often regulate expression of distinct gene sets. although promoter rewiring of shared genes is primarily responsible for regulatory differences among related eukaryotic species, in bacteria, species-specific genes are often controlled by ancestral transcription factors, and regulatory circuit evolution has been further shaped by horizontal gene transfer. modifications in transcription factors and in promoter structure also contribute to divergence in bacterial regulatory circuits.identifying the functional elements encoded in a genome is one of the principal challenges in modern biology. comparative genomics should offer a powerful, general approach. here, we present a comparative analysis of the yeast saccharomyces cerevisiae based on high-quality draft sequences of three related species (s. paradoxus, s. mikatae and s. bayanus). we first aligned the genomes and characterized their evolution, defining the regions and mechanisms of change. we then developed methods for direct identification of genes and regulatory motifs. the gene analysis yielded a major revision to the yeast gene catalogue, affecting approximately 15\\\\% of all genes and reducing the total count by about 500 genes. the motif analysis automatically identified 72 genome-wide elements, including most known regulatory motifs and numerous new motifs. we inferred a putative function for most of these motifs, and provided insights into their combinatorial interactions. the results have implications for genome analysis of diverse organisms, including the human.high-throughput genome-wide molecular assays, which probe cellular networks from different perspectives, have become central to molecular biology. probabilistic graphical models are useful for extracting meaningful biological insights from the resulting data sets. these models provide a concise representation of complex cellular networks by composing simpler submodels. procedures based on well-understood principles for inferring such models from data facilitate a model-based methodology for analysis and discovery. this methodology and its capabilities are illustrated by several recent applications to gene expression data.many networks of interest in the sciences, including social networks, computer networks, and metabolic and regulatory networks, are found to divide naturally into communities or modules. the problem of detecting and characterizing this community structure is one of the outstanding issues in the study of networked systems. one highly effective approach is the optimization of the quality function known as  ” modularity” over the possible divisions of a network. here i show that the modularity can be expressed in terms of the eigenvectors of a characteristic matrix for the network, which i call the modularity matrix, and that this expression leads to a spectral algorithm for community detection that returns results of demonstrably higher quality than competing methods in shorter running times. i illustrate the method with applications to several published network data sets.while escherichia coli has one of the most comprehensive datasets of experimentally verified transcriptional regulatory interactions of any organism, it is still far from complete. this presents a problem when trying to combine gene expression and regulatory interactions to model transcriptional regulatory networks. using the available regulatory interactions to predict new interactions may lead to better coverage and more accurate models. here, we develop {serend} ({semi}-supervised {regulatory} network discoverer), a semi-supervised learning method that uses a curated database of verified transcriptional factor\\\\^{a}\\x80\\x93gene interactions, {dna} sequence binding motifs, and a compendium of gene expression data in order to make thousands of new predictions about transcription factor\\\\^{a}\\x80\\x93gene interactions, including whether the transcription factor activates or represses the gene. using genome-wide binding datasets for several transcription factors, we demonstrate that our semi-supervised classification strategy improves the prediction of targets for a given transcription factor. to further demonstrate the utility of our inferred interactions, we generated a new microarray gene expression dataset for the aerobic to anaerobic shift response in e. coli. we used our inferred interactions with the verified interactions to reconstruct a dynamic regulatory network for this response. the network reconstructed when using our inferred interactions was better able to correctly identify known regulators and suggested additional activators and repressors as having important roles during the aerobic\\\\^{a}\\x80\\x93anaerobic shift interface. the proper functioning of transcriptional gene regulation is essential for all living organisms. several diseases are associated with loss of appropriate transcriptional regulation. even in relatively simple organisms, such as the bacterium e. coli, response to environmental stress is a complex and highly regulated process. this process is controlled by a set of transcription factors that causes an increase or decrease in the expression levels of their target\\'s gene. however, identifying the set of targets regulated by each of these factors remains a challenge. even after decades of experimental research on e. coli, only a quarter of all gene products have a known regulator. here, we develop a method that extends the known set of regulator\\\\^{a}\\x80\\x93target relationships with additional predictions. our method utilizes the {dna} sequence control code and expression levels of known targets in a variety of conditions, as well as genes for which it is not known if they are targets of a specific regulator. we show that our method more accurately identifies true targets of known regulators than previous methods suggested for this task. we then applied our predictions to identify active regulators involved in the dynamic response that occurs in e. coli when it is deprived of oxygen.after the completion of the human and other genome projects it emerged that the number of genes in organisms as diverse as fruit flies, nematodes, and humans does not reflect our perception of their relative complexity. here, we provide reliable evidence that the size of protein interaction networks in different organisms appears to correlate much better with their apparent biological complexity. we develop a stable and powerful, yet simple, statistical procedure to estimate the size of the whole network from subnet data. this approach is then applied to a range of eukaryotic organisms for which extensive protein interaction data have been collected and we estimate the number of interactions in humans to be ≈650,000. we find that the human interaction network is one order of magnitude bigger than the drosophila melanogaster interactome and ≈3 times bigger than in caenorhabditis elegans.systems biology studies biological systems by systematically perturbing them (biologically, genetically, or chemically); monitoring the gene, protein, and informational pathway responses; integrating these data; and ultimately, formulating mathematical models that describe the structure of the system and its response to individual perturbations. the emergence of systems biology is described, as are several examples of specific systems approaches.in contrast to protein-coding sequences, the significance of variation in non-coding {dna} in human disease has been minimally explored. a great number of recent genome-wide association studies suggest that non-coding variation is a significant risk factor for common disorders, but the mechanisms by which this variation contributes to disease remain largely obscure. distant-acting transcriptional enhancers--a major category of functional non-coding {dna}--are involved in many developmental and disease-relevant processes. genome-wide approaches to their discovery and functional characterization are now available and provide a growing knowledge base for the systematic exploration of their role in human biology and disease susceptibility.{rna}-seq is revolutionizing the way transcript abundances are measured. a key challenge in transcript quantification from {rna}-seq data is the handling of reads that map to multiple genes or isoforms. this issue is particularly important for quantification with de novo transcriptome assemblies in the absence of sequenced genomes, as it is difficult to determine which transcripts are isoforms of the same gene. a second significant issue is the design of {rna}-seq experiments, in terms of the number of reads, read length, and whether reads come from one or both ends of {cdna} fragments. we present {rsem}, an user-friendly software package for quantifying gene and isoform abundances from single-end or paired-end {rna}-seq data. {rsem} outputs abundance estimates, 95\\\\% credibility intervals, and visualization files and can also simulate {rna}-seq data. in contrast to other existing tools, the software does not require a reference genome. thus, in combination with a de novo transcriptome assembler, {rsem} enables accurate transcript quantification for species without sequenced genomes. on simulated and real data sets, {rsem} has superior or comparable performance to quantification methods that rely on a reference genome. taking advantage of {rsem}\\'s ability to effectively use ambiguously-mapping reads, we show that accurate gene-level abundance estimates are best obtained with large numbers of short single-end reads. on the other hand, estimates of the relative frequencies of isoforms within single genes may be improved through the use of paired-end reads, depending on the number of possible splice forms for each gene. {rsem} is an accurate and user-friendly software tool for quantifying transcript abundances from {rna}-seq data. as it does not rely on the existence of a reference genome, it is particularly useful for quantification with de novo transcriptome assemblies. in addition, {rsem} has enabled valuable guidance for cost-efficient design of quantification experiments with {rna}-seq, which is currently relatively expensive.methods for learning bayesian networks can discover dependency structure between observed variables. although these methods are useful in many applications, they run into computational and statistical problems in domains that involve a large number of variables. in this paper, we consider a solution that is applicable when many variables have similar behavior. we introduce a new class of models, module networks, that explicitly partition the variables into modules, so that the variables in each module share the same parents in the network and the same conditional probability distribution. we define the semantics of module networks, and describe an algorithm that learns the modules\\' composition and their dependency structure from data. evaluation on real data in the domains of gene expression and the stock market shows that module networks generalize better than bayesian networks, and that the learned module network structure reveals regularities that are obscured in learned bayesian networks.{background}:transcription factors function by binding different classes of regulatory elements. the encyclopedia of {dna} elements ({encode}) project has recently produced binding data for more than 100 transcription factors from about 500 {chip}-seq experiments in multiple cell types. while this large amount of data creates a valuable resource, it is nonetheless overwhelmingly complex and simultaneously incomplete since it covers only a small fraction of all human transcription {factors.results}:as part of the consortium effort in providing a concise abstraction of the data for facilitating various types of downstream analyses, we constructed statistical models that capture the genomic features of three paired types of regions by machine-learning methods: firstly, regions with active or inactive binding; secondly, those with extremely high or low degrees of co-binding, termed {hot} and {lot} regions; and finally, regulatory modules proximal or distal to genes. from the distal regulatory modules, we developed computational pipelines to identify potential enhancers, many of which were validated experimentally. we further associated the predicted enhancers with potential target transcripts and the transcription factors involved. for {hot} regions, we found a significant fraction of transcription factor binding without clear sequence motifs and showed that this observation could be related to strong {dna} accessibility of these {regions.conclusions}:overall, the three pairs of regions exhibit intricate differences in chromosomal locations, chromatin features, factors that bind them, and cell-type specificity. our machine learning approach enables us to identify features potentially general to all transcription factors, including those not included in the data.tissue-specific gene expression plays a fundamental role in metazoan biology and is an important aspect of many complex diseases. nevertheless, an organism-wide map of tissue-specific expression remains elusive due to difficulty in obtaining these data experimentally. here, we leveraged existing whole-animal caenorhabditis elegans microarray data representing diverse conditions and developmental stages to generate accurate predictions of tissue-specific gene expression and experimentally validated these predictions. these patterns of tissue-specific expression are more accurate than existing high-throughput experimental studies for nearly all tissues; they also complement existing experiments by addressing tissue-specific expression present at particular developmental stages and in small tissues. we used these predictions to address several experimentally challenging questions, including the identification of tissue-specific transcriptional motifs and the discovery of potential {mirna} regulation specific to particular tissues. we also investigate the role of tissue context in gene function through tissue-specific functional interaction networks. to our knowledge, this is the first study producing high-accuracy predictions of tissue-specific expression and interactions for a metazoan organism based on whole-animal data. in animals, a crucial facet of any gene\\'s function is the tissue or cell type in which that gene is expressed and the proteins that it interacts with in that cell. however, genome-wide identification of expression across the multitude of tissues of varying size and complexity is difficult to achieve experimentally. in this paper, we show that microararray data collected from whole animals can be analyzed to yield high-quality predictions of tissue-specific expression. these predictions are of better or comparable accuracy to tissue-specific expression determined from high-throughput experiments. our results provide a global view of tissue-specific expression in caenorhabditis elegans, allowing us to address the question of how expression patterns are regulated and to analyze how the functions of genes that are expressed in several tissues are influenced by the cellular context.comparative analysis is a fundamental tool in biology. conservation among species greatly assists the detection and characterization of functional elements, whereas inter-species differences are probably the best indicators of biological adaptation. traditionally, comparative approaches were applied to the analysis of genomic sequences. with the growing availability of functional genomic data, comparative paradigms are now being extended also to the study of other functional attributes, most notably the gene expression. here we review recent works applying comparative analysis to large-scale gene expression datasets and discuss the central principles and challenges of such approaches.comparing the genomes of two different species allow the exploration of a host of intriguing evolutionary and genetic questions.since the advent of gene expression microarray technology more than 10 years ago, many computational approaches have been developed aimed at using statistical associations between {mrna} abundance profiles to predict transcriptional regulatory interactions. the ultimate goal is to develop causal network models describing the transcriptional influences that genes exert on each other (via their protein products), which can be used to predict network disruptions (e.g., mutations) leading to a disease phenotype, as well as the appropriate therapeutic intervention. however, microarray data measure only a small component of the interacting variables in a genetic regulatory network, as cells are known to regulate gene expression via many diverse mechanisms. although many researchers have acknowledged the questionable interpretation of statistical dependencies between {mrna} profiles, very little work has been done on theoretically characterizing the nature of inferred dependencies using models that account for unobserved interacting variables. in this work, we review the theory behind reverse engineering algorithms derived from three separate disciplines-system control theory, graphical models, and information theory-and highlight several mathematical relationships between the various methods. we then apply recent theoretical work on constructing graphical models with latent variables to the context of reverse engineering genetic networks. we demonstrate that even the addition of simple latent variables induces statistical dependencies between non-directly interacting (e.g., co-regulated) genes that cannot be eliminated by conditioning on any observed variables.{background}: site-specific transcription factors ({tfs}) are coordinators of developmental and physiological gene expression programs. their binding to cis-regulatory modules of target genes mediates the precise cell- and context-specific activation and repression of genes. the expression of {tfs} should therefore reflect the core expression program of each cell. {results}: we studied the expression dynamics of about 750 {tfs} using the available genomics resources in drosophila melanogaster. we find that 95\\\\% of these {tfs} are expressed at some point during embryonic development, with a peak roughly between 10 and 12 hours after egg laying, the core stages of organogenesis. we address the differential utilization of {dna}-binding domains in different developmental programs systematically in a spatio-temporal context, and show that the zinc finger class of {tfs} is predominantly early expressed, while homeobox {tfs} exhibit later expression in embryogenesis. {conclusions}: previous work, dissecting cis-regulatory modules during drosophila development, suggests that {tfs} are deployed in groups acting in a cooperative manner. in contrast, we find that there is rapid exchange of co-expressed partners amongst the fly {tfs}, at rates similar to the genome-wide dynamics of co-expression clusters. this suggests there may also be a high level of combinatorial complexity of {tfs} at cis-regulatory modules.complex interactions between genes or proteins contribute substantially to phenotypic evolution. we present a probabilistic model and a maximum likelihood approach for cross-species clustering analysis and for identification of conserved as well as species-specific co-expression modules. this model enables a  ” soft” cross-species clustering ({scsc}) approach by encouraging but not enforcing orthologous genes to be grouped into the same cluster. {scsc} is therefore robust to obscure orthologous relationships and can reflect different functional roles of orthologous genes in different species. we generated a time-course gene expression dataset for differentiating mouse embryonic stem ({es}) cells, and compiled a dataset of published gene expression data on differentiating human {es} cells. applying {scsc} to analyze these datasets, we identified conserved and species-specific gene regulatory modules. together with {protein-dna} binding data, an {scsc} cluster specifically induced in murine {es} cells indicated that the {klf2}/4/5 transcription factors, although critical to maintaining the pluripotent phenotype in mouse {es} cells, were decoupled from the {oct4}/{sox2}/{nanog} regulatory module in human {es} cells. two of the target genes of murine {klf2}/4/5, {lin28} and {nodal}, were rewired to be targets of {oct4}/{sox2}/{nanog} in human {es} cells. moreover, by mapping {scsc} clusters onto {kegg} signaling pathways, we identified the signal transduction components that were induced in pluripotent {es} cells in either a conserved or a species-specific manner. these results suggest that the pluripotent cell identity can be established and maintained through more than one gene regulatory network. a major goal in biology is to understand the evolution of complex traits, such as the development of multicellular body plans. to a certain extent, complex traits are governed by regulated gene expression. the comparison expression data between species requires extra considerations than sequence comparison, because gene expression is not static and the level of expression is influenced by external conditions. considering that co-expression patterns are often comparable across species, we developed a statistical model for cross-species clustering analysis. the model allows each species to create its own clusters of the genes but also encourages the species to borrow strength from each others\\' clusters of orthologous genes. the result is a pairing of clusters, one from each species, where the paired clusters share many but not necessarily all orthologous genes. the model-based approach not only reduces subjective influence but also enables effective use of evolutionary dependence. applying this model to analyze human and mouse embryonic stem ({es}) cell data, we identified the transcription factors and the signaling proteins that are specifically expressed in either human or mouse {es} cells. these results suggest that the pluripotent cell identity can be established and maintained through more than one gene regulatory network.to begin to understand the genetic architecture of natural variation in gene expression, we carried out genetic linkage analysis of genomewide expression patterns in a cross between a laboratory strain and a wild strain of saccharomyces cerevisiae. over 1500 genes were differentially expressed between the parent strains. expression levels of 570 genes were linked to one or more different loci, with most expression levels showing complex inheritance patterns. the loci detected by linkage fell largely into two categories: cis-acting modulators of single genes and trans-acting modulators of many genes. we found eight such trans-acting loci, each affecting the expression of a group of 7 to 94 genes of related function.an abstract is not available.second-generation sequencing technology can now be used to sequence an entire human genome in a matter of days and at low cost. sequence read lengths, initially very short, have rapidly increased since the technology first appeared, and we now are seeing a growing number of efforts to sequence large genomes de novo from these short reads. in this perspective, we describe the issues associated with short-read assembly, the different types of data produced by second-gen sequencers, and the latest assembly algorithms designed for these data. we also review the genomes that have been assembled recently from short reads and make recommendations for sequencing strategies that will yield a high-quality assembly.a major yet unresolved quest in decoding the human genome is the identification of the regulatory sequences that control the spatial and temporal expression of genes. distant-acting transcriptional enhancers are particularly challenging to uncover because they are scattered among the vast non-coding portion of the genome. evolutionary sequence constraint can facilitate the discovery of enhancers, but fails to predict when and where they are active in vivo. here we present the results of chromatin immunoprecipitation with the enhancer-associated protein p300 followed by massively parallel sequencing, and map several thousand in vivo binding sites of p300 in mouse embryonic forebrain, midbrain and limb tissue. we tested 86 of these sequences in a transgenic mouse assay, which in nearly all cases demonstrated reproducible enhancer activity in the tissues that were predicted by p300 binding. our results indicate that in vivo mapping of p300 binding is a highly accurate means for identifying enhancers and their associated activities, and suggest that such data sets will be useful to study the role of tissue-specific enhancers in human biology and disease on a genome-wide scale.we describe {hi-c}, a method that probes the three-dimensional architecture of whole genomes by coupling proximity-based ligation with massively parallel sequencing. we constructed spatial proximity maps of the human genome with {hi-c} at a resolution of 1 megabase. these maps confirm the presence of chromosome territories and the spatial proximity of small, gene-rich chromosomes. we identified an additional level of genome organization that is characterized by the spatial segregation of open and closed chromatin to form two genome-wide compartments. at the megabase scale, the chromatin conformation is consistent with a fractal globule, a knot-free, polymer conformation that enables maximally dense packing while preserving the ability to easily fold and unfold any genomic locus. the fractal globule is distinct from the more commonly used globular equilibrium model. our results demonstrate the power of {hi-c} to map the dynamic conformations of whole genomes.despite the successes of genomics, little is known about how genetic information produces complex organisms. a look at the crucial functional elements of fly and worm genomes could change that.identifying driver genes in cancer remains a crucial bottleneck in therapeutic development and basic understanding of the disease. we developed helios, an algorithm that integrates genomic data from primary tumors with data from functional {rnai} screens to pinpoint driver genes within large recurrently amplified regions of {dna}. applying helios to breast cancer data identified a set of candidate drivers highly enriched with known drivers (p\\xa0< 10?14). nine of ten top-scoring helios genes are known drivers of breast cancer, and in\\xa0vitro validation of 12 candidates predicted by helios found ten conferred enhanced anchorage-independent growth, demonstrating helios?s exquisite sensitivity and specificity. we extensively characterized {rsf}-1, a driver identified by helios whose amplification correlates with poor prognosis, and found increased tumorigenesis and metastasis in mouse models. we have demonstrated a powerful approach for identifying driver genes and how it can yield important insights into cancer.we present a network framework for analyzing multi-level regulation in higher eukaryotes based on systematic integration of various high-throughput datasets. the network, namely the integrated regulatory network, consists of three major types of regulation: {tf}→gene, {tf}→{mirna} and {mirna}→gene. we identified the target genes and target {mirnas} for a set of {tfs} based on the {chip}-seq binding profiles, the predicted targets of {mirnas} using annotated {3′utr} sequences and conservation information. making use of the system-wide {rna}-seq profiles, we classified transcription factors into positive and negative regulators and assigned a sign for each regulatory interaction. other types of edges such as protein-protein interactions and potential intra-regulations between {mirnas} based on the embedding of {mirnas} in their host genes were further incorporated. we examined the topological structures of the network, including its hierarchical organization and motif enrichment. we found that transcription factors downstream of the hierarchy distinguish themselves by expressing more uniformly at various tissues, have more interacting partners, and are more likely to be essential. we found an over-representation of notable network motifs, including a {ffl} in which a {mirna} cost-effectively shuts down a transcription factor and its target. we used data of c. elegans from the {modencode} project as a primary model to illustrate our framework, but further verified the results using other two data sets. as more and more genome-wide {chip}-seq and {rna}-seq data becomes available in the near future, our methods of data integration have various potential applications. the precise control of gene expression lies at the heart of many biological processes. in eukaryotes, the regulation is performed at multiple levels, mediated by different regulators such as transcription factors and {mirnas}, each distinguished by different spatial and temporal characteristics. these regulators are further integrated to form a complex regulatory network responsible for the orchestration. the construction and analysis of such networks is essential for understanding the general design principles. recent advances in high-throughput techniques like {chip}-seq and {rna}-seq provide an opportunity by offering a huge amount of binding and expression data. we present a general framework to combine these types of data into an integrated network and perform various topological analyses, including its hierarchical organization and motif enrichment. we find that the integrated network possesses an intrinsic hierarchical organization and is enriched in several network motifs that include both transcription factors and {mirnas}. we further demonstrate that the framework can be easily applied to other species like human and mouse. as more and more genome-wide {chip}-seq and {rna}-seq data are going to be generated in the near future, our methods of data integration have various potential applications.large-scale cancer genome sequencing has uncovered thousands of gene mutations, but distinguishing tumor driver genes from functionally neutral passenger mutations is a major challenge. we analyzed 800 cancer genomes of eight types to find single-nucleotide variants ({snvs}) that precisely target phosphorylation machinery, important in cancer development and drug targeting. assuming that cancer-related biological systems involve unexpectedly frequent mutations, we used novel algorithms to identify genes with significant phosphorylation-associated {snvs} ({psnvs}), phospho-mutated pathways, kinase networks, drug targets, and clinically correlated signaling modules. we highlight increased survival of patients with {tp53} {psnvs}, hierarchically organized cancer kinase modules, a novel {psnv} in {egfr}, and an immune-related network of {psnvs} that correlates with prolonged survival in ovarian cancer. our findings include multiple actionable cancer gene candidates ({flnb}, {grm1}, {pou2f1}), protein complexes ({hcf1}, {asf1}), and kinases ({prkcz}). this study demonstrates new ways of interpreting cancer genomes and presents new leads for cancer research.motivation: transcription factors ({tfs}) are crucial during the lifetime of the cell. their functional roles are defined by the genes they regulate. uncovering these roles not only sheds light on the {tf} at hand but puts it into the context of the complete regulatory {network.results}: here, we present an alignment- and threshold-free comparative genomics approach for assigning functional roles to {dna} regulatory motifs. we incorporate our approach into the gomo algorithm, a computational tool for detecting associations between a user-specified {dna} regulatory motif [expressed as a position weight matrix ({pwm})] and gene ontology ({go}) terms. incorporating multiple species into the analysis significantly improves gomo\\'s ability to identify {go} terms associated with the regulatory targets of {tfs}. including three comparative species in the process of predicting {tf} roles in saccharomyces cerevisiae and homo sapiens increases the number of significant predictions by 75 and 200\\\\%, respectively. the predicted {go} terms are also more specific, yielding deeper biological insight into the role of the {tf}. adjusting motif (binding) affinity scores for individual sequence composition proves to be essential for avoiding false positive associations. we describe a novel {dna} sequence-scoring algorithm that compensates a thermodynamic measure of {dna}-binding affinity for individual sequence base composition. gomo\\'s prediction accuracy proves to be relatively insensitive to how promoters are defined. because gomo uses a threshold-free form of gene set analysis, there are no free parameters to tune. biologists can investigate the potential roles of {dna} regulatory motifs of interest using gomo via the web ({http://meme.nbcr.net).contact}: {t.bailey@uq.edu.ausupplementary} information: supplementary data are available at bioinformatics online.over the past decade, a number of researchers in systems biology have sought to relate the function of biological systems to their network-level descriptions—lists of the most important players and the pairwise interactions between them. both for large networks (in which statistical analysis is often framed in terms of the abundance of repeated small subgraphs) and for small networks which can be analyzed in greater detail (or even synthesized in vivo and subjected to experiment), revealing the relationship between the topology of small subgraphs and their biological function has been a central goal. we here seek to pose this revelation as a statistical task, illustrated using a particular setup which has been constructed experimentally and for which parameterized models of transcriptional regulation have been studied extensively. the question  ” how does function follow form” is here mathematized by identifying which topological attributes correlate with the diverse possible information-processing tasks which a transcriptional regulatory network can realize. the resulting method reveals one form-function relationship which had earlier been predicted based on analytic results, and reveals a second for which we can provide an analytic interpretation. resulting source code is distributed via http://formfunction.sourceforge.net.our goal is to cluster genes into transcriptional modules--sets of genes where similarity in expression is explained by common regulatory mechanisms at the transcriptional level. we want to learn modules from both time series gene expression data and genome-wide motif data that are now readily available for organisms such as s. cereviseae as a result of prior computational studies or experimental results. we present a generative probabilistic model for combining regulatory sequence and time series expression data to cluster genes into coherent transcriptional modules. starting with a set of motifs representing known or putative regulatory elements (transcription factor binding sites) and the counts of occurrences of these motifs in each gene\\'s promoter region, together with a time series expression profile for each gene, the learning algorithm uses expectation maximization to learn module assignments based on both types of data. we also present a technique based on the {jensen-shannon} entropy contributions of motifs in the learned model for associating the most significant motifs to each module. thus, the algorithm gives a global approach for associating sets of regulatory elements to \"modules\" of genes with similar time series expression profiles. the model for expression data exploits our prior belief of smooth dependence on time by using statistical splines and is suitable for typical time course data sets with relatively few experiments. moreover, the model is sufficiently interpretable that we can understand how both sequence data and expression data contribute to the cluster assignments, and how to interpolate between the two data sources. we present experimental results on the yeast cell cycle to validate our method and find that our combined expression and motif clustering algorithm discovers modules with both coherent expression and similar motif patterns, including binding motifs associated to known cell cycle transcription factors.there has been much progress in genomics in the ten years since a draft sequence of the human genome was published. opportunities for understanding health and disease are now unprecedented, as advances in genomics are harnessed to obtain robust foundational knowledge about the structure and function of the human genome and about the genetic contributions to human health and disease. here we articulate a 2011 vision for the future of genomics research and describe the path towards an era of genomic medicine.changes in the patterns of gene expression are widely believed to underlie many of the phenotypic differences within and between species. although much emphasis has been placed on changes in transcriptional regulation, gene expression is regulated at many levels, all of which must ultimately be studied together to obtain a complete picture of the evolution of gene expression. here we compare the evolution of transcriptional regulation and post-transcriptional regulation that is mediated by {micrornas}, a large class of small, non-coding {rnas} in plants and animals, focusing on the evolution of the individual regulators and their binding sites. as an initial step towards integrating these mechanisms into a unified framework, we propose a simple model that describes the transcriptional regulation of new {microrna} genes.as genome-scale measurements lead to increasingly complex models of gene regulation, systematic approaches are needed to validate and refine these models. towards this goal, we describe an automated procedure for prioritizing genetic perturbations in order to discriminate optimally between alternative models of a gene-regulatory network. using this procedure, we evaluate 38 candidate regulatory networks in yeast and perform four high-priority gene knockout experiments. the refined networks support previously unknown regulatory mechanisms downstream of {sok2} and {swi4}.genetic variants that are associated with common human diseases do not lead directly to disease, but instead act on intermediate, molecular phenotypes that in turn induce changes in higher-order disease traits. therefore, identifying the molecular phenotypes that vary in response to changes in {dna} and that also associate with changes in disease traits has the potential to provide the functional information required to not only identify and validate the susceptibility genes that are directly affected by changes in {dna}, but also to understand the molecular networks in which such genes operate and how changes in these networks lead to changes in disease traits. toward that end, we profiled more than 39,000 transcripts and we genotyped 782,476 unique single nucleotide polymorphisms ({snps}) in more than 400 human liver samples to characterize the genetic architecture of gene expression in the human liver, a metabolically active tissue that is important in a number of common human diseases, including obesity, diabetes, and atherosclerosis. this genome-wide association study of gene expression resulted in the detection of more than 6,000 associations between {snp} genotypes and liver gene expression traits, where many of the corresponding genes identified have already been implicated in a number of human diseases. the utility of these data for elucidating the causes of common human diseases is demonstrated by integrating them with genotypic and expression data from other human and mouse populations. this provides much-needed functional support for the candidate susceptibility genes being identified at a growing number of genetic loci that have been identified as key drivers of disease from genome-wide association studies of disease. by using an integrative genomics approach, we highlight how the gene {rps26} and not {erbb3} is supported by our data as the most likely susceptibility gene for a novel type 1 diabetes locus recently identified in a large-scale, genome-wide association study. we also identify {sort1} and {celsr2} as candidate susceptibility genes for a locus recently associated with coronary artery disease and plasma low-density lipoprotein cholesterol levels in the process. genome-wide association studies seek to identify regions of the genome in which changes in {dna} in a given population are correlated with disease, drug response, or other phenotypes of interest. however, changes in {dna} that associate with traits like common human diseases do not lead directly to disease, but instead act on intermediate, molecular phenotypes that in turn induce changes in the higher-order disease traits. therefore, identifying molecular phenotypes that vary in response to changes in {dna} that also associate with changes in disease traits can provide the functional information necessary to not only identify and validate the susceptibility genes directly affected by changes in {dna}, but to understand as well the molecular networks in which such genes operate and how changes in these networks lead to changes in disease traits. to enable this type of approach we profiled the expression levels of 39,280 transcripts and genotyped 782,476 {snps} in 427 human liver samples, identifying thousands of {dna} variants that strongly associated with liver gene expression. these relationships were then leveraged by integrating them with genotypic and expression data from other human and mouse populations, leading to the direct identification of candidate susceptibility genes corresponding to genetic loci identified as key drivers of disease. our analysis is able to provide much needed functional support for these candidate susceptibility genes.functional annotation of proteins is a fundamental problem in the post-genomic era. the recent availability of protein interaction networks for many model species has spurred on the development of computational methods for interpreting such data in order to elucidate protein function. in this review, we describe the current computational approaches for the task, including direct methods, which propagate functional information through the network, and module-assisted methods, which infer functional modules within the network and use those for the annotation task. although a broad variety of interesting approaches has been developed, further progress in the field will depend on systematic evaluation of the methods and their dissemination in the biological community.evolutionary developmental biology, often called evo-devo, seeks to understand the ancestral relationship among organisms by comparing their developmental strategies and ultimately reconstructing the pathways that led to the extraordinary variety of biological forms. the insights from this synthesis of developmental biology and evolutionary principles are useful for understanding the development of the nervous system. the pervasive and crucial roles of {micrornas} in nervous system development suggest that these short non-coding transcripts deserve a chapter in the unfolding evo-devo story. the structure of {micrornas}, their physical proximity to other genes and their network effects on targets make this class of transcripts tractable genetic material for evolutionary change.in the last decade, advances in high-throughput technologies such as {dna} microarrays have made it possible to simultaneously measure the expression levels of tens of thousands of genes and proteins. this has resulted in large amounts of biological data requiring analysis and interpretation. nonnegative matrix factorization ({nmf}) was introduced as an unsupervised, parts-based learning paradigm involving the decomposition of a nonnegative matrix v into two nonnegative matrices, w and h, via a multiplicative updates algorithm. in the context of a p×n gene expression matrix v consisting of observations on p genes from n samples, each column of w defines a metagene, and each column of h represents the metagene expression pattern of the corresponding sample. {nmf} has been primarily applied in an unsupervised setting in image and natural language processing. more recently, it has been successfully utilized in a variety of applications in computational biology. examples include molecular pattern discovery, class comparison and prediction, cross-platform and cross-species analysis, functional characterization of genes and biomedical informatics. in this paper, we review this method as a data analytical and interpretive tool in computational biology with an emphasis on these applications.inferring, or \\'reverse-engineering\\', gene networks can be defined as the process of identifying gene interactions from experimental data through computational analysis. gene expression data from microarrays are typically used for this purpose. here we compared different reverse-engineering algorithms for which ready-to-use software was available and that had been tested on experimental data sets. we show that reverse-engineering algorithms are indeed able to correctly infer regulatory interactions among genes, at least when one performs perturbation experiments complying with the algorithm requirements. these algorithms are superior to classic clustering algorithms for the purpose of finding regulatory interactions among genes, and, although further improvements are needed, have reached a discreet performance for being practically useful.the boundaries between prokaryotes, unicellular eukaryotes and multicellular eukaryotes are accompanied by orders-of-magnitude reductions in effective population size, with concurrent amplifications of the effects of random genetic drift and mutation. the resultant decline in the efficiency of selection seems to be sufficient to influence a wide range of attributes at the genomic level in a non-adaptive manner. a key remaining question concerns the extent to which variation in the power of random genetic drift is capable of influencing phylogenetic diversity at the subcellular and cellular levels. should this be the case, population size would have to be considered as a potential determinant of the mechanistic pathways underlying long-term phenotypic evolution. here we demonstrate a phylogenetically broad inverse relation between the power of drift and the structural integrity of protein subunits. this leads to the hypothesis that the accumulation of mildly deleterious mutations in populations of small size induces secondary selection for protein-protein interactions that stabilize key gene functions. by this means, the complex protein architectures and interactions essential to the genesis of phenotypic diversity may initially emerge by non-adaptive mechanisms.an epistasis map ({e-map}) was constructed in the fission yeast, schizosaccharomyces pombe, by systematically measuring the phenotypes associated with pairs of mutations. this high-density, quantitative genetic interaction map focused on various aspects of chromosome function, including transcription regulation and {dna} repair/replication. the {e-map} uncovered a previously unidentified component of the {rna} interference ({rnai}) machinery (rsh1) and linked the {rnai} pathway to several other biological processes. comparison of the s. pombe {e-map} to an analogous genetic map from the budding yeast revealed that, whereas negative interactions were conserved between genes involved in similar biological processes, positive interactions and overall genetic profiles between pairs of genes coding for physically associated proteins were even more conserved. hence, conservation occurs at the level of the functional module (protein complex), but the genetic cross talk between modules can differ substantially.our goal is gene network inference in genetical genomics or systems genetics experiments. for species where sequence information is available, we first perform expression quantitative trait locus ({eqtl}) mapping by jointly utilizing cis-, cis–trans-, and trans-regulation. after using local structural models to identify regulator–target pairs for each {eqtl}, we construct an encompassing directed network ({edn}) by assembling all retained regulator–target relationships. the {edn} has nodes corresponding to expressed genes and {eqtl} and directed edges from {eqtl} to cis-regulated target genes, from cis-regulated genes to cis–trans-regulated target genes, from trans-regulator genes to target genes, and from {trans-eqtl} to target genes. for network inference within the strongly constrained search space defined by the {edn}, we propose structural equation modeling ({sem}), because it can model cyclic networks and the {edn} indeed contains feedback relationships. on the basis of a factorization of the likelihood and the constrained search space, our {sem} algorithm infers networks involving several hundred genes and {eqtl}. structure inference is based on a penalized likelihood ratio and an adaptation of occam\\'s window model selection. the {sem} algorithm was evaluated using data simulated with nonlinear ordinary differential equations and known cyclic network topologies and was applied to a real yeast data set.recent efforts in systematically profiling embryonic stem ({es}) cells have yielded a wealth of high-throughput data. complementarily, emerging databases and computational tools facilitate {es} cell studies and further pave the way toward the in silico reconstruction of regulatory networks encompassing multiple molecular layers. here, we briefly survey databases, algorithms, and software tools used to organize and analyze high-throughput experimental data collected to study mammalian cellular systems with a focus on {es} cells. the vision of using heterogeneous data to reconstruct a complete multi-layered {es} cell regulatory network is discussed. this review also provides an accompanying manually extracted dataset of different types of regulatory interactions from low-throughput experimental {es} cell studies available at http://amp.pharm.mssm.edu/iscmid/literature. {\\\\copyright} 2010 john wiley \\\\& sons, inc.transcriptional regulation plays vital roles in many fundamental biological processes. reverse engineering of genome-wide regulatory networks from high-throughput transcriptomic data provides a promising way to characterize the global scenario of regulatory relationships between regulators and their targets. in this review, we summarize and categorize the main frameworks and methods currently available for inferring transcriptional regulatory networks from microarray gene expression profiling data. we overview each of strategies and introduce representative methods respectively. their assumptions, advantages, shortcomings, and possible improvements and extensions are also clarified and commented.genetic interactions occur when a combination of mutations results in a surprising phenotype. these interactions capture functional redundancy, and thus are important for predicting function, dissecting protein complexes into functional pathways, and exploring the mechanistic underpinnings of common human diseases. synthetic sickness and lethality are the most studied types of genetic interactions in yeast. however, even in yeast, only a small proportion of gene pairs have been tested for genetic interactions due to the large number of possible combinations of gene pairs. to expand the set of known synthetic lethal ({sl}) interactions, we have devised an integrative, multi-network approach for predicting these interactions that significantly improves upon the existing approaches. first, we defined a large number of features for characterizing the relationships between pairs of genes from various data sources. in particular, these features are independent of the known {sl} interactions, in contrast to some previous approaches. using these features, we developed a non-parametric multi-classifier system for predicting {sl} interactions that enabled the simultaneous use of multiple classification procedures. several comprehensive experiments demonstrated that the {sl}-independent features in conjunction with the advanced classification scheme led to an improved performance when compared to the current state of the art method. using this approach, we derived the first yeast transcription factor genetic interaction network, part of which was well supported by literature. we also used this approach to predict {sl} interactions between all non-essential gene pairs in yeast (http://sage.fhcrc.org/downloads/download\\u200bs/predicted\\\\_yeast\\\\_genetic\\\\_interactions.z\\u200bip). this integrative approach is expected to be more effective and robust in uncovering new genetic interactions from the tens of millions of unknown gene pairs in yeast and from the hundreds of millions of gene pairs in higher organisms like mouse and human, in which very few genetic interactions have been identified to date. genetic interactions occur when a combination of mutations results in a surprising phenotype. these interactions capture functional redundancy, and thus are important for predicting function, dissecting protein complexes into functional pathways, and exploring the mechanistic underpinnings of common human diseases. due to the large number of possible combinations of genes, only a small portion of gene pairs in yeast and only a few pairs in higher organisms like mouse and human have been tested. therefore, predicting genetic interactions has received significant attention in the past several years. the existing methods primarily rely on the known genetic interactions, and thus are far less effective in classifying most gene pairs not well connected with known genetic interactions. here we developed a non-parametric multi-classifier system for predicting genetic interactions based on a large number of novel features independent of the known genetic interactions. this approach led to an improved performance when compared to the current state-of-the-art method. using this approach, we derived the first yeast transcription factor genetic interaction network, part of which was well supported by literature. this integrative approach is expected to be more effective and robust in uncovering new genetic interactions in yeast and other species.quantifying the impact of heritable epigenetic variation on complex traits is an emerging challenge in population genetics. here, we analyze a population of isogenic arabidopsis lines that segregate experimentally induced {dna} methylation changes at hundreds of regions across the genome. we demonstrate that several of these differentially methylated regions ({dmrs}) act as bona fide epigenetic quantitative trait loci ({qtl}(epi)), accounting for 60 to 90\\\\% of the heritability for two complex traits, flowering time and primary root length. these {qtl}(epi) are reproducible and can be subjected to artificial selection. many of the experimentally induced {dmrs} are also variable in natural populations of this species and may thus provide an epigenetic basis for darwinian evolution independently of {dna} sequence changes.it is estimated that the human genome contains hundreds of thousands of enhancers, so understanding these gene-regulatory elements is a crucial goal. several fundamental questions need to be addressed about enhancers, such as how do we identify them all, how do they work, and how do they contribute to disease and evolution? five prominent researchers in this field look at how much we know already and what needs to be done to answer these questions.developmental fate decisions are dictated by master transcription factors ({tfs}) that interact with cis-regulatory elements to direct transcriptional programs. certain malignant tumors may also depend on cellular hierarchies reminiscent of normal development but superimposed on underlying genetic aberrations. in glioblastoma ({gbm}), a subset of stem-like tumor-propagating cells ({tpcs}) appears to drive tumor progression and underlie therapeutic resistance yet remain poorly understood. here, we identify a core set of neurodevelopmental {tfs} ({pou3f2}, {sox2}, {sall2}, and {olig2}) essential for {gbm} propagation. these {tfs} coordinately bind and activate {tpc}-specific regulatory elements and are sufficient to fully reprogram differentiated {gbm} cells to \"induced\" {tpcs}, recapitulating the epigenetic landscape and phenotype of native {tpcs}. we reconstruct a network model that highlights critical interactions and identifies candidate therapeutic targets for eliminating {tpcs}. our study establishes the epigenetic basis of a developmental hierarchy in {gbm}, provides detailed insight into underlying gene regulatory programs, and suggests attendant therapeutic strategies. {paperclip}: copyright {\\\\copyright} 2014 elsevier inc. all rights reserved.many genetic variants have been implicated in disease but their effects in function across tissues and cell-types remain to be resolved. raj et al. (p. 519) present an analysis of expression quantative trait loci ({eqtl}) measuring messenger {rna} levels and examined correlations between genotypes and gene expression in purified monocytes and t cells in healthy individuals of european, african, and asian descent. most, but not all, of the {eqtls} and their effects on expression were shared between the populations, as well as a substantial proportion between the cell types. links were found with disease-associated variants and loci that previous genome-wide analyses have implicated in neurodegenerative and autoimmune diseases.obtaining {rna}-seq measurements involves a complex data analytical process with a large number of competing algorithms as options. there is much debate about which of these methods provides the best approach. unfortunately, it is currently difficult to evaluate their performance due in part to a lack of sensitive assessment metrics. we present a series of statistical summaries and plots to evaluate the performance in terms of specificity and sensitivity, available as a {r/bioconductor} package ( http://bioconductor.org/packages/rnaseqcomp ). using two independent datasets, we assessed seven competing pipelines. performance was generally poor, with two methods clearly underperforming and {rsem} slightly outperforming the rest.motivation: identifying the cellular wiring that connects genomic perturbations to transcriptional changes in cancer is essential to gain a mechanistic understanding of disease initiation, progression and ultimately to predict drug response. we have developed a method called tied diffusion through interacting events ({tiedie}) that uses a network diffusion approach to connect genomic perturbations to gene expression changes characteristic of cancer subtypes. the method computes a subnetwork of protein–protein interactions, predicted transcription factor-to-target connections and curated interactions from literature that connects genomic and transcriptomic perturbations.high-throughput experimental technologies are generating increasingly massive and complex genomic data sets. the sheer enormity and heterogeneity of these data threaten to make the arising problems computationally infeasible. fortunately, powerful algorithmic techniques lead to software that can answer important biomedical questions in practice. in this review, we sample the algorithmic landscape, focusing on state-of-the-art techniques, the understanding of which will aid the bench biologist in analysing omics data. we spotlight specific examples that have facilitated and enriched analyses of sequence, transcriptomic and network data sets.comparing the gene-expression profiles of sick and healthy individuals can help in understanding disease. such differential expression analysis is a well-established way to find gene sets whose expression is altered in the disease. recent approaches to gene-expression analysis go a step further and seek differential co-expression patterns, wherein the level of co-expression of a set of genes differs markedly between disease and control samples. such patterns can arise from a disease-related change in the regulatory mechanism governing that set of genes, and pinpoint dysfunctional regulatory networks. here we present {dicer}, a new method for detecting differentially co-expressed gene sets using a novel probabilistic score for differential correlation. {dicer} goes beyond standard differential co-expression and detects pairs of modules showing differential co-expression. the expression profiles of genes within each module of the pair are correlated across all samples. the correlation between the two modules, however, differs markedly between the disease and normal samples. we show that {dicer} outperforms the state of the art in terms of significance and interpretability of the detected gene sets. moreover, the gene sets discovered by {dicer} manifest regulation by disease-specific {microrna} families. in a case study on alzheimer\\'s disease, {dicer} dissected biological processes and protein complexes into functional subunits that are differentially co-expressed, thereby revealing inner structures in disease regulatory networks. the most fundamental and popular gene-expression experiments measure genome-wide transcription levels in two populations: perturbed and wild type, or cases and controls. the genes that show significantly different expression between the two populations (the differentially expressed genes) are useful for understanding the biology underlying the phenotype difference, and can sometimes also serve as biomarkers for classification. in contrast, genes that have similar expression to each other across all profiles (co-expressed genes) can yield clues about the functional commonality of the two populations. differential co-expression has recently been proposed as a way to combine the benefits of these two approaches: it seeks gene groups that are co-expressed in one phenotype much more than in the other. here we develop a new method for detecting differential co-expression and test it on case-control expression profiles of several diseases. our algorithm improves upon the state of the art in the strength of the detected patterns and in agreement with current biological knowledge. we show that our method can predict gene regulators that are associated with the disease of interest and demonstrate that it can dissect known biological pathways into subcomponents that are not detected using standard analyses.maintaining balanced growth in a changing environment is a fundamental systems-level challenge for cellular physiology, particularly in microorganisms. while the complete set of regulatory and functional pathways supporting growth and cellular proliferation are not yet known, portions of them are well understood. in particular, cellular proliferation is governed by mechanisms that are highly conserved from unicellular to multicellular organisms, and the disruption of these processes in metazoans is a major factor in the development of cancer. in this paper, we develop statistical methodology to identify quantitative aspects of the regulatory mechanisms underlying cellular proliferation in saccharomyces cerevisiae. we find that the expression levels of a small set of genes can be exploited to predict the instantaneous growth rate of any cellular culture with high accuracy. the predictions obtained in this fashion are robust to changing biological conditions, experimental methods, and technological platforms. the proposed model is also effective in predicting growth rates for the related yeast saccharomyces bayanus and the highly diverged yeast schizosaccharomyces pombe, suggesting that the underlying regulatory signature is conserved across a wide range of unicellular evolution. we investigate the biological significance of the gene expression signature that the predictions are based upon from multiple perspectives: by perturbing the regulatory network through the {ras/pka} pathway, observing strong upregulation of growth rate even in the absence of appropriate nutrients, and discovering putative transcription factor binding sites, observing enrichment in growth-correlated genes. more broadly, the proposed methodology enables biological insights about growth at an instantaneous time scale, inaccessible by direct experimental methods. data and tools enabling others to apply our methods are available at http://function.princeton.edu/growthrate. a major challenge for living organisms is the regulation of cellular growth in a fluctuating environment. sudden changes in nutrient availability or the presence of stress factors typically require rapid adjustments of cellular growth. the misregulation of growth control in higher organisms is a major factor in the development of cancer. a statistical characterization of cellular growth based on gene expression levels provides a quantitative perspective to understand the regulatory network that controls growth. we develop a model of cellular growth in the yeast saccharomyces cerevisiae, grounded in the expression levels of a small set of genes. the model is able to predict the growth rate of new cellular cultures from expression data and is robust to changing biological conditions, experimental methods, and technological platforms. the predictions are informative about changes in growth at very short time scales, which direct experimental methods cannot generally access. the model also predicts growth rates in saccharomyces bayanus and in schizosaccharomyces pombe, a yeast diverged by approximately a billion years of evolution. our findings suggest that the model describes fundamental characteristics of the unicellular eukaryotic growth regulatory program. a case study explores the role of nutrient sensing in the yeast growth regulatory system.motivation: {rna}-seq technology has been widely adopted as an attractive alternative to microarray-based methods to study global gene expression. however, robust statistical tools to analyze these complex datasets are still lacking. by grouping genes with similar expression profiles across treatments, cluster analysis provides insight into gene functions and networks, and hence is an important technique for {rna}-seq data analysis.when the human genome project started, the major challenge was how to sequence a 3 billion letter code in an organized and cost-effective manner. when completed, the project had laid the foundation for a huge variety of biomedical fields through the production of a complete human genome sequence, but also had driven the development of laboratory and analytical methods that could produce large amounts of sequencing data cheaply. these technological developments made possible the sequencing of many more vertebrate genomes, which have been necessary for the interpretation of the human genome. they have also enabled large-scale studies of vertebrate genome evolution, as well as comparative and human medicine. in this review, we give examples of evolutionary analysis using a wide variety of time frames-from the comparison of populations within a species to the comparison of species separated by at least 300 million years. furthermore, we anticipate discoveries related to evolutionary mechanisms, adaptation, and disease to quickly accelerate in the coming years.for decades, evolutionary biologists have argued that changes in cis-regulatory sequences constitute an important part of the genetic basis for adaptation. although originally based on first principles, this claim is now empirically well supported: numerous studies have identified cis-regulatory mutations with functionally significant consequences for morphology, physiology and behaviour. the focus has now shifted to considering whether cis-regulatory and coding mutations make qualitatively different contributions to phenotypic evolution. cases in which parallel mutations have produced parallel trait modifications in particular suggest that some phenotypic changes are more likely to result from cis-regulatory mutations than from coding mutations.one goal of human genetics is to understand how the information for precise and dynamic gene expression programs is encoded in the genome. the interactions of transcription factors ({tfs}) with {dna} regulatory elements clearly play an important role in determining gene expression outputs, yet the regulatory logic underlying functional transcription factor binding is poorly understood. many studies have focused on characterizing the genomic locations of {tf} binding, yet it is unclear to what extent {tf} binding at any specific locus has functional consequences with respect to gene expression output. to evaluate the context of functional {tf} binding we knocked down 59 {tfs} and chromatin modifiers in one {hapmap} lymphoblastoid cell line. we then identified genes whose expression was affected by the knockdowns. we intersected the gene expression data with transcription factor binding data (based on {chip}-seq and {dnase}-seq) within 10 kb of the transcription start sites of expressed genes. this combination of data allowed us to infer functional {tf} binding. using this approach, we found that only a small subset of genes bound by a factor were differentially expressed following the knockdown of that factor, suggesting that most interactions between {tf} and chromatin do not result in measurable changes in gene expression levels of putative target genes. we found that functional {tf} binding is enriched in regulatory elements that harbor a large number of {tf} binding sites, at sites with predicted higher binding affinity, and at sites that are enriched in genomic regions annotated as  ” active enhancers.” an important question in genomics is to understand how a class of proteins called  ” transcription factors” controls the expression level of other genes in the genome in a cell-type-specific manner – a process that is essential to human development. one major approach to this problem is to study where these transcription factors bind in the genome, but this does not tell us about the effect of that binding on gene expression levels and it is generally accepted that much of the binding does not strongly influence gene expression. to address this issue, we artificially reduced the concentration of 59 different transcription factors in the cell and then examined which genes were impacted by the reduced transcription factor level. our results implicate some attributes that might influence what binding is functional, but they also suggest that a simple model of functional vs. non-functional binding may not suffice.various characteristics of complex gene regulatory networks ({grns}) have been discovered during the last decade, e.g., redundancy, exponential indegree distributions, scale-free outdegree distributions, mutational robustness, and evolvability. although progress has been made in this field, it is not well understood whether these characteristics are the direct products of selection or those of other evolutionary forces such as mutational biases and biophysical constraints. to elucidate the causal factors that promoted the evolution of complex {grns}, we examined the effect of fluctuating environmental selection and some intrinsic constraining factors on {grn} evolution by using an individual-based model. we found that the evolution of complex {grns} is remarkably promoted by fixation of beneficial gene duplications under unpredictably fluctuating environmental conditions and that some internal factors inherent in organisms, such as mutational bias, gene expression costs, and constraints on expression dynamics, are also important for the evolution of {grns}. the results indicate that various biological properties observed in {grns} could evolve as a result of not only adaptation to unpredictable environmental changes but also non-adaptive processes owing to the properties of the organisms themselves. our study emphasizes that evolutionary models considering such intrinsic constraining factors should be used as null models to analyze the effect of selection on {grn} evolution. various organismal traits, including the morphology of multicellular species and metabolism in unicellular species, are determined by the amount and combinations of proteins in the cell. the complex regulatory network plays an important role in controlling the protein profiles in a cell. recent studies have revealed that gene regulatory networks have many interesting structural and mutational features such as their scale-free structure, mutational robustness, and evolvability. however, why and how these features have emerged from evolution is unknown. in this paper, we constructed an evolutionary model of gene regulatory networks and simulated its evolution under various environmental conditions. the results show that most features of known gene regulatory networks evolve as a result of adaptation to unpredictable environmental fluctuations. in addition, some internal organismal factors, such as mutational bias, gene expression costs, and constraints on expression dynamics, are also important for {grn} evolution observed in real organisms. thus, these {grn} features appear to evolve as a result of not only adaptation to unpredictable environmental changes but also non-adaptive processes owing to the properties of the organisms themselves.gene regulatory networks are perhaps the most important organizational level in the cell where signals from the cell state and the outside environment are integrated in terms of activation and inhibition of genes. for the last decade, the study of such networks has been fueled by large-scale experiments and renewed attention from the theoretical field. different models have been proposed to, for instance, investigate expression dynamics, explain the network topology we observe in bacteria and yeast, and for the analysis of evolvability and robustness of such networks. yet how these gene regulatory networks evolve and become evolvable remains an open question. an individual-oriented evolutionary model is used to shed light on this matter. each individual has a genome from which its gene regulatory network is derived. mutations, such as gene duplications and deletions, alter the genome, while the resulting network determines the gene expression pattern and hence fitness. with this protocol we let a population of individuals evolve under darwinian selection in an environment that changes through time. our work demonstrates that long-term evolution of complex gene regulatory networks in a changing environment can lead to a striking increase in the efficiency of generating beneficial mutations. we show that the population evolves towards genotype-phenotype mappings that allow for an orchestrated network-wide change in the gene expression pattern, requiring only a few specific gene indels. the genes involved are hubs of the networks, or directly influencing the hubs. moreover, throughout the evolutionary trajectory the networks maintain their mutational robustness. in other words, evolution in an alternating environment leads to a network that is sensitive to a small class of beneficial mutations, while the majority of mutations remain neutral: an example of evolution of evolvability. a cell receives signals both from its internal and external environment and responds by changing the expression of genes. in this manner the cell adjusts to heat, osmotic pressures and other circumstances during its lifetime. over long timescales, the network of interacting genes and its regulatory actions also undergo evolutionary adaptation. yet how do such networks evolve and become adapted? in this paper we describe the study of a simple model of gene regulatory networks, focusing solely on evolutionary adaptation. we let a population of individuals evolve, while the external environment changes through time. to ensure evolution is the only source of adaptation, we do not provide the individuals with a sensor to the environment. we show that the interplay between the long-term process of evolution and short-term gene regulation dynamics leads to a striking increase in the efficiency of creating well-adapted offspring. beneficial mutations become more frequent, nevertheless robustness to the majority of mutations is maintained. thus we demonstrate a clear example of the evolution of evolvability.cellular functions are mediated through complex systems of macromolecules and metabolites linked through biochemical and physical interactions, represented in interactome models as \\'nodes\\' and \\'edges\\', respectively. better understanding of genotype-to-phenotype relationships in human disease will require modeling of how disease-causing mutations affect systems or interactome properties. here we investigate how perturbations of interactome networks may differ between complete loss of gene products (\\'node removal\\') and interaction-specific or edge-specific (\\'edgetic\\') alterations. global computational analyses of approximately 50,000 known causative mutations in human mendelian disorders revealed clear separations of mutations probably corresponding to those of node removal versus edgetic perturbations. experimental characterization of mutant alleles in various disorders identified diverse edgetic interaction profiles of mutant proteins, which correlated with distinct structural properties of disease proteins and disease mechanisms. edgetic perturbations seem to confer distinct functional consequences from node removal because a large fraction of cases in which a single gene is linked to multiple disorders can be modeled by distinguishing edgetic network perturbations. edgetic network perturbation models might improve both the understanding of dissemination of disease alleles in human populations and the development of molecular therapeutic strategies.a central challenge in genetics is to understand when and why mutations alter the phenotype of an organism. the consequences of gene inhibition have been systematically studied and can be predicted reasonably well across a genome. however, many sequence variants important for disease and evolution may alter gene regulation rather than gene function. the consequences of altering a regulatory interaction (or  ” edge”) rather than a gene (or  ” node”) in a network have not been as extensively studied. here we use an integrative analysis and evolutionary conservation to identify features that predict when the loss of a regulatory interaction is detrimental in the extensively mapped transcription network of budding yeast. properties such as the strength of an interaction, location and context in a promoter, regulator and target gene importance, and the potential for compensation (redundancy) associate to some extent with interaction importance. combined, however, these features predict quite well whether the loss of a regulatory interaction is detrimental across many promoters and for many different transcription factors. thus, despite the potential for regulatory diversity, common principles can be used to understand and predict when changes in regulation are most harmful to an organism. the genomes of individuals differ in sequence at thousands of base pairs. some of these polymorphisms affect the sequence of proteins, but many are likely to alter how genes are regulated. when are changes in gene regulation detrimental to an organism? we have used an integrative analysis of transcription factor binding site conservation in budding yeast to address the extent to which different features predict when potential changes in gene regulation are detrimental. we found that, despite the diversity of transcription factors and regulatory regions in a genome, a few simple properties can be used to predict and understand when changes in regulation are most harmful.the cancer stem cell ({csc}) concept is a highly debated topic in cancer research. while experimental evidence in favor of the cancer stem cell theory is apparently abundant, the results are often criticized as being difficult to interpret. an important reason for this is that most experimental data that support this model rely on transplantation studies. in this study we use a novel cellular potts model to elucidate the dynamics of established malignancies that are driven by a small subset of {cscs}. our results demonstrate that epigenetic mutations that occur during mitosis display highly altered dynamics in {csc}-driven malignancies compared to a classical, non-hierarchical model of growth. in particular, the heterogeneity observed in {csc}-driven tumors is considerably higher. we speculate that this feature could be used in combination with epigenetic (methylation) sequencing studies of human malignancies to prove or refute the {csc} hypothesis in established tumors without the need for transplantation. moreover our tumor growth simulations indicate that {csc}-driven tumors display evolutionary features that can be considered beneficial during tumor progression. besides an increased heterogeneity they also exhibit properties that allow the escape of clones from local fitness peaks. this leads to more aggressive phenotypes in the long run and makes the neoplasm more adaptable to stringent selective forces such as cancer treatment. indeed when therapy is applied the clone landscape of the regrown tumor is more aggressive with respect to the primary tumor, whereas the classical model demonstrated similar patterns before and after therapy. understanding these often counter-intuitive fundamental properties of (non-)hierarchically organized malignancies is a crucial step in validating the {csc} concept as well as providing insight into the therapeutical consequences of this model. cancer is in essence a genetic disease that leads to uncontrolled cell proliferation, invasion and metastasis. the cancer stem cell ({csc}) hypothesis states that tumors are not just a mass of uniform malignant cells but they are hierarchically organized, like normal tissues. at the top of such a hierarchy are cancer stem cells that fuel tumor growth in the long run, whereas the majority of other cells are able to divide only a few times. the experiments that support the {csc} hypothesis are often criticized as being difficult to interpret. a novel approach to test the {csc} paradigm is to integrate mathematical modeling with {dna} variation data that carry the phylogenetic history of cells. we have developed a model that simulates the occurrence of such changes under both the {csc} hypothesis and the classical, purely stochastic scenario. we found that although a {csc}-driven tumor has a smaller number of tumorigenic cells, it triggers more malignant properties such as invasive growth, heterogeneity and evolutionary escape from peaks in the fitness landscape. these properties, that are unique to the {csc} model, are enhanced even further when a treatment is applied to the tumor.cis-regulatory sequences, such as enhancers and promoters, control development and physiology by regulating gene expression. mutations that affect the function of these sequences contribute to phenotypic diversity within and between species. with many case studies implicating divergent cis-regulatory activity in phenotypic evolution, researchers have recently begun to elucidate the genetic and molecular mechanisms that are responsible for cis-regulatory divergence. approaches include detailed functional analysis of individual cis-regulatory elements and comparing mechanisms of gene regulation among species using the latest genomic tools. despite the limited number of mechanistic studies published to date, this work shows how cis-regulatory activity can diverge and how studies of cis-regulatory divergence can address long-standing questions about the genetic mechanisms of phenotypic evolution.divergence in gene regulation can play a major role in evolution. here, we used a phylogenetic framework to measure {mrna} profiles in 15 yeast species from the phylum ascomycota and reconstruct the evolution of their modular regulatory programs along a time course of growth on glucose over 300 million years. we found that modules have diverged proportionally to phylogenetic distance, with prominent changes in gene regulation accompanying changes in lifestyle and ploidy, especially in carbon metabolism. paralogs have significantly contributed to regulatory divergence, typically within a very short window from their duplication. paralogs from a whole genome duplication ({wgd}) event have a uniquely substantial contribution that extends over a longer span. similar patterns occur when considering the evolution of the heat shock regulatory program measured in eight of the species, suggesting that these are general evolutionary principles. {doi}: {http://dx.doi.org/10.7554/elife}.00603.001reversible protein phosphorylation is a signaling mechanism involved in all cellular processes. to create a systems view of the signaling apparatus in budding yeast, we generated an epistatic miniarray profile ({e-map}) comprised of 100,000 pairwise, quantitative genetic interactions, including virtually all protein and small-molecule kinases and phosphatases as well as key cellular regulators. quantitative genetic interaction mapping reveals factors working in compensatory pathways (negative genetic interactions) or those operating in linear pathways (positive genetic interactions). we found an enrichment of positive genetic interactions between kinases, phosphatases, and their substrates. in addition, we assembled a higher-order map from sets of three genes that display strong interactions with one another: triplets enriched for functional connectivity. the resulting network view provides insights into signaling pathway regulation and reveals a link between the cell-cycle kinase, cak1, the fus3 {map} kinase, and a pathway that regulates chromatin integrity during transcription by {rna} polymerase {ii}.motivation: assigning functions for unknown genes based on diverse large-scale data is a key task in functional genomics. previous work on gene function prediction has addressed this problem using independent classifiers for each function. however, such an approach ignores the structure of functional class taxonomies, such as the gene ontology ({go}). over a hierarchy of functional classes, a group of independent classifiers where each one predicts gene membership to a particular class can produce a hierarchically inconsistent set of predictions, where for a given gene a specific class may be predicted positive while its inclusive parent class is predicted negative. taking the hierarchical structure into account resolves such inconsistencies and provides an opportunity for leveraging all classifiers in the hierarchy to achieve higher specificity of predictions.variation in gene expression is an important mechanism underlying susceptibility to complex disease. the simultaneous genome-wide assay of gene expression and genetic variation allows the mapping of the genetic factors that underpin individual differences in quantitative levels of expression (expression {qtls}; {eqtls}). the availability of systematically generated {eqtl} information could provide immediate insight into a biological basis for disease associations identified through genome-wide association ({gwa}) studies, and can help to identify networks of genes involved in disease pathogenesis. although there are limitations to current {eqtl} maps, understanding of disease will be enhanced with novel technologies and international efforts that extend to a wide range of new samples and tissues.a new field of genetic analysis of global gene expression has emerged in recent years, driven by the realization that traditional techniques of linkage and association analysis can be applied to thousands of transcript levels measured by microarrays. genetic dissection of transcript abundance has shed light on the architecture of quantitative traits, provided a new approach for connecting {dna} sequence variation with phenotypic variation, and improved our understanding of transcriptional regulation and regulatory variation.the genome has often been called the operating system ({os}) for a living organism. a computer {os} is described by a regulatory control network termed the call graph, which is analogous to the transcriptional regulatory network in a cell. to apply our firsthand knowledge of the architecture of software systems to understand cellular design principles, we present a comparison between the transcriptional regulatory network of a well-studied bacterium (escherichia coli) and the call graph of a canonical {os} (linux) in terms of topology and evolution. we show that both networks have a fundamentally hierarchical layout, but there is a key difference: the transcriptional regulatory network possesses a few global regulators at the top and many targets at the bottom; conversely, the call graph has many regulators controlling a small set of generic functions. this top-heavy organization leads to highly overlapping functional modules in the call graph, in contrast to the relatively independent modules in the regulatory network. we further develop a way to measure evolutionary rates comparably between the two networks and explain this difference in terms of network evolution. the process of biological evolution via random mutation and subsequent selection tightly constrains the evolution of regulatory network hubs. the call graph, however, exhibits rapid evolution of its highly connected generic components, made possible by designers\\' continual fine-tuning. these findings stem from the design principles of the two systems: robustness for biological systems and cost effectiveness (reuse) for software systems.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsenhancer mapping has been greatly facilitated by various genomic marks associated with it. however, little is available in our toolbox to link enhancers with their target promoters, hampering mechanistic understanding of enhancer-promoter ({ep}) interaction. we develop and characterize multiple genomic features for distinguishing true {ep} pairs from noninteracting pairs. we integrate these features into a probabilistic predictor for {ep} interactions. multiple validation experiments demonstrate a significant improvement over state-of-the-art approaches. systematic analyses of {ep} interactions across 12 cell types reveal several global features of {ep} interactions: (i) a larger fraction of {ep} interactions are cell type specific than enhancers; (ii) promoters controlled by multiple enhancers have higher tissue specificity, but the regulating enhancers are less conserved; (iii) cohesin plays a role in mediating tissue-specific {ep} interactions via chromatin looping in a {ctcf}-independent manner. our approach presents a systematic and effective strategy to decipher the mechanisms underlying {ep} communication.emerging technologies have enabled the acquisition of large genomics and proteomics data sets. however, current methodologies for analysis do not permit interpretation of the data in ways that unravel cellular networking. we propose a quantitative method for determining functional interactions in cellular signaling and gene networks. it can be used to explore cell systems at a mechanistic level or applied within a  ” modular” framework, which dramatically decreases the number of variables to be assayed. this method is based on a mathematical derivation that demonstrates how the topology and strength of network connections can be retrieved from experimentally measured network responses to successive perturbations of all modules. importantly, our analysis can reveal functional interactions even when the components of the system are not all known. under these circumstances, some connections retrieved by the analysis will not be direct but correspond to the interaction routes through unidentified elements. the method is tested and illustrated by using computer-generated responses of a modeled mitogen-activated protein kinase cascade and gene network.somatic cell reprogramming, directed differentiation of pluripotent stem cells, and direct conversions between differentiated cell lineages represent powerful\\xa0approaches to engineer cells for research and regenerative medicine. we have developed {cellnet}, a network biology platform that more accurately assesses the fidelity of cellular engineering than existing methodologies and generates hypotheses for improving cell derivations. analyzing expression data from 56 published reports, we found that cells derived via directed differentiation more closely resemble their in\\xa0vivo counterparts than products of direct conversion, as reflected by the establishment of target cell-type gene regulatory networks ({grns}). furthermore, we discovered that directly converted cells fail to adequately silence expression programs of the starting population and that the establishment of unintended {grns} is common to virtually every cellular engineering paradigm. {cellnet} provides a platform for quantifying how closely engineered cell populations resemble their target cell type and a rational strategy to guide enhanced cellular engineering.summary.\\u2002 estimation of structure, such as in variable selection, graphical modelling or cluster analysis, is notoriously difficult, especially for high dimensional data. we introduce stability selection. it is based on subsampling in combination with (high dimensional) selection algorithms. as such, the method is extremely general and has a very wide range of applicability. stability selection provides finite sample control for some error rates of false discoveries and hence a transparent principle to choose a proper amount of regularization for structure estimation. variable selection and structure estimation improve markedly for a range of selection methods if stability selection is applied. we prove for the randomized lasso that stability selection will be variable selection consistent even if the necessary conditions for consistency of the original lasso method are violated. we demonstrate stability selection for variable selection and gaussian graphical modelling, using real and simulated data.{background}:previous work has demonstrated that chromatin feature levels correlate with gene expression. the {encode} project enables us to further explore this relationship using an unprecedented volume of data. expression levels from more than 100,000 promoters were measured using a variety of high-throughput techniques applied to {rna} extracted by different protocols from different cellular compartments of several human cell lines. {encode} also generated the genome-wide mapping of eleven histone marks, one histone variant, and {dnase} i hypersensitivity sites in seven cell {lines.results}:we built a novel quantitative model to study the relationship between chromatin features and expression levels. our study not only confirms that the general relationships found in previous studies hold across various cell lines, but also makes new suggestions about the relationship between chromatin features and gene expression levels. we found that expression status and expression levels can be predicted by different groups of chromatin features, both with high accuracy. we also found that expression levels measured by {cage} are better predicted than by {rna}-{pet} or {rna}-seq, and different categories of chromatin features are the most predictive of expression for different {rna} measurement methods. additionally, {polya}+ {rna} is overall more predictable than {polya}- {rna} among different cell compartments, and {polya}+ cytosolic {rna} measured with {rna}-seq is more predictable than {polya}+ nuclear {rna}, while the opposite is true for {polya}- {rna}.{conclusions}:our study provides new insights into transcriptional regulation by analyzing chromatin features in different cellular contexts.the variation in genetic sequences determining the binding of transcription factors is believed to be an important facet of evolution. however, the degree to which a genome is robust, that is, able to withstand changes and how robustness affects evolution is unclear. payne and wagner (p. 875) investigated the empirical support for mutational robustness by examining transcription factor ({tf}) binding in the mouse and yeast genomes. a network analysis of the degree of variation revealed that the sites with the highest affinity for {tf} binding exhibit the greatest tolerance for nucleotide substitutions, whereas low-affinity sites exhibit greater sensitivity to mutation. thus, while mutational robustness and evolvability are antagonistic at the genotypic level, they are synergistic at the phenotypic level.{dna} sequence polymorphism in a regulatory protein can have a widespread transcriptional effect. here we present a computational approach for analyzing modules of genes with a common regulation that are affected by specific {dna} polymorphisms. we identify such regulatory-linkage modules by integrating genotypic and expression data for individuals in a segregating population with complementary expression data of strains mutated in a variety of regulatory proteins. our procedure searches simultaneously for groups of co-expressed genes, for their common underlying linkage interval, and for their shared regulatory proteins. we applied the method to a cross between laboratory and wild strains of s. cerevisiae , demonstrating its ability to correctly suggest modules and to outperform extant approaches. our results suggest that middle sporulation genes are under the control of polymorphism in the sporulation-specific tertiary complex {sum1p/rfm1p}/hst1p. in another example, our analysis reveals novel inter-relations between swi3 and two mitochondrial inner membrane proteins underlying variation in a module of aerobic cellular respiration genes. overall, our findings demonstrate that this approach provides a useful framework for the systematic mapping of quantitative trait loci and their role in gene expression variation.microarray technologies allow the identification of large numbers of expression differences within and between species. although environmental and physiological stimuli are clearly responsible for changes in the expression levels of many genes, it is not known whether the majority of changes of gene expression fixed during evolution between species and between various tissues within a species are caused by darwinian selection or by stochastic processes. we find the following: (1) expression differences between species accumulate approximately linearly with time; (2) gene expression variation among individuals within a species correlates positively with expression divergence between species; (3) rates of expression divergence between species do not differ significantly between intact genes and expressed pseudogenes; (4) expression differences between brain regions within a species have accumulated approximately linearly with time since these regions emerged during evolution. these results suggest that the majority of expression differences observed between species are selectively neutral or nearly neutral and likely to be of little or no functional significance. therefore, the identification of gene expression differences between species fixed by selection should be based on null hypotheses assuming functional neutrality. furthermore, it may be possible to apply a molecular clock based on expression differences to infer the evolutionary history of tissues.over the past years, tremendous efforts have been made to elucidate the molecular basis of the initiation and progression of ovarian cancer. however, most existing studies have been focused on individual genes or a single type of data, which may lack the power to detect the complex mechanisms of cancer formation by overlooking the interactions of different genetic and epigenetic factors. we propose an integrative framework to identify genetic and epigenetic features related to ovarian cancer and to quantify the causal relationships among these features using a probabilistic graphical model based on the cancer genome atlas ({tcga}) data. in the feature selection, we first defined a set of seed genes by including 48 candidate tumor suppressors or oncogenes and an additional 20 ovarian cancer related genes reported in the literature. the seed genes were then fed into a stepwise correlation-based selector to identify 271 additional features including 177 genes, 82 copy number variation sites, 11 methylation sites and 1 somatic mutation (at gene {tp53}). we built a bayesian network model with a logit link function to quantify the causal relationships among these features and discovered a set of 13 hub genes including {arid1a}, c19orf53, {cskn2a1} and {col5a2}. the directed graph revealed many potential genetic pathways, some of which confirmed the existing results in the literature. clustering analysis further suggested four gene clusters, three of which correspond to well-defined cellular processes including cell division, tumor invasion and mitochondrial system. in addition, two genes related to glycoprotein synthesis, {psg11} and {galnt10}, were found highly predictive for the overall survival time of ovarian cancer patients. the proposed framework is effective in identifying possible important genetic and epigenetic features that are related to complex cancer diseases. the constructed bayesian network has identified some new genetic/epigenetic pathways, which may shed new light into the molecular mechanisms of ovarian cancer.microarray experiments have yielded massive amounts of expression information measured under various conditions for the model species arabidopsis (arabidopsis thaliana) and rice (oryza sativa). expression compendia grouping multiple experiments make it possible to define correlated gene expression patterns within one species and to study how expression has evolved between species. we developed a robust framework to measure expression context conservation ({ecc}) and found, by analyzing 4,630 pairs of orthologous arabidopsis and rice genes, that 77\\\\% showed conserved coexpression. examples of nonconserved {ecc} categories suggested a link between regulatory evolution and environmental adaptations and included genes involved in signal transduction, response to different abiotic stresses, and hormone stimuli. to identify genomic features that influence expression evolution, we analyzed the relationship between {ecc}, tissue specificity, and protein evolution. tissue-specific genes showed higher expression conservation compared with broadly expressed genes but were fast evolving at the protein level. no significant correlation was found between protein and expression evolution, implying that both modes of gene evolution are not strongly coupled in plants. by integration of cis-regulatory elements, many {ecc} conserved genes were significantly enriched for shared {dna} motifs, hinting at the conservation of ancestral regulatory interactions in both model species. surprisingly, for several tissue-specific genes, patterns of concerted network evolution were observed, unveiling conserved coexpression in the absence of conservation of tissue specificity. these findings demonstrate that orthologs inferred through sequence similarity in many cases do not share similar biological functions and highlight the importance of incorporating expression information when comparing genes across species.network inference, which is the reconstruction of biological networks from high-throughput data, can provide valuable information about the regulation of gene expression in cells. however, it is an underdetermined problem, as the number of interactions that can be inferred exceeds the number of independent measurements. different state-of-the-art tools for network inference use specific assumptions and simplifications to deal with underdetermination, and these influence the inferences. the outcome of network inference therefore varies between tools and can be highly complementary. here we categorize the available tools according to the strategies that they use to deal with the problem of underdetermination. such categorization allows an insight into why a certain tool is more appropriate for the specific research question or data set at hand.the hypothesis that differences in gene regulation have an important role in speciation and adaptation is more than 40 years old. with the advent of new sequencing technologies, we are able to characterize and study gene expression levels and associated regulatory mechanisms in a large number of individuals and species at an unprecedented resolution and scale. we have thus gained new insights into the evolutionary pressures that shape gene expression levels and have developed an appreciation for the relative importance of evolutionary changes in different regulatory genetic and epigenetic mechanisms. the current challenge is to link gene regulatory changes to adaptive evolution of complex phenotypes. here we mainly focus on comparative studies in primates and how they are complemented by studies in model organisms.interaction data for a given protein may be spread across multiple databases. we set out to create a unifying index that would facilitate searching for these data and that would group together redundant interaction data while recording the methods used to perform this grouping. we present a method to generate a key for a protein interaction record and a key for each participant protein. these keys may be generated by anyone using only the primary sequence of the proteins, their taxonomy identifiers and the secure hash algorithm. two interaction records will have identical keys if they refer to the same set of identical protein sequences and taxonomy identifiers. we define records with identical keys as a redundant group. our method required that we map protein database references found in interaction records to current protein sequence records. operations performed during this mapping are described by a mapping score that may provide valuable feedback to source interaction databases on problematic references that are malformed, deprecated, ambiguous or unfound. keys for protein participants allow for retrieval of interaction information independent of the protein references used in the original records. we have applied our method to protein interaction records from {bind}, {biogrid}, {dip}, {hprd}, {intact}, {mint}, {mpact}, {mppi} and {ophid}. the resulting interaction reference index is provided in {psi}-{mitab} 2.5 format at http://irefindex.uio.no. this index may form the basis of alternative redundant groupings based on gene identifiers or near sequence identity groupings.computer science and biology have enjoyed a long and fruitful relationship for decades. biologists rely on computational methods to analyze and integrate large data sets, while several computational methods were inspired by the high-level design principles of biological systems. recently, these two directions have been converging. in this review, we argue that thinking computationally about biological processes may lead to more accurate models, which in turn can be used to improve the design of algorithms. we discuss the similar mechanisms and requirements shared by computational and biological processes and then present several recent studies that apply this joint analysis strategy to problems related to coordination, network analysis, and tracking and vision. we also discuss additional biological processes that can be studied in a similar manner and link them to potential computational problems. with the rapid accumulation of data detailing the inner workings of biological systems, we expect this direction of coupling biological and computational studies to greatly expand in the future.mathematical models of natural systems are abstractions of much more complicated processes. developing informative and realistic models of such systems typically involves suitable statistical inference methods, domain expertise, and a modicum of luck. except for cases where physical principles provide sufficient guidance, it will also be generally possible to come up with a large number of potential models that are compatible with a given natural system and any finite amount of data generated from experiments on that system. here we develop a computational framework to systematically evaluate potentially vast sets of candidate differential equation models in light of experimental and prior knowledge about biological systems. this topological sensitivity analysis enables us to evaluate quantitatively the dependence of model inferences and predictions on the assumed model structures. failure to consider the impact of structural uncertainty introduces biases into the analysis and potentially gives rise to misleading conclusions.buffered qualitative stability explains the robustness and evolvability of transcriptional networks | the theory of buffered qualitative stability uses the importance of biological robustness to explain many features of gene regulatory networks in a wide range of organisms, and has implications for diverse biological phenomena including the ability of bacteria and cancer cells to \\'loosen\\' their robustness and hence evade treatment.genetic screens are powerful methods for the discovery of gene–phenotype associations. however, a systems biology approach to genetics must leverage the massive amount of  ” omics” data to enhance the power and speed of functional gene discovery in vivo. thus far, few computational methods for gene function prediction have been rigorously tested for their performance on a genome-wide scale in vivo. in this work, we demonstrate that integrating genome-wide computational gene prioritization with large-scale genetic screening is a powerful tool for functional gene discovery. to discover genes involved in neural development in drosophila, we extend our strategy for the prioritization of human candidate disease genes to functional prioritization in drosophila. we then integrate this prioritization strategy with a large-scale genetic screen for interactors of the proneural transcription factor atonal using genomic deficiencies and mutant and {rnai} collections. using the prioritized genes validated in our genetic screen, we describe a novel genetic interaction network for atonal. lastly, we prioritize the whole drosophila genome and identify candidate gene associations for ten receptor-signaling pathways. this novel database of prioritized pathway candidates, as well as a web application for functional prioritization in drosophila, called {endeavour}-{highfly}, and the atonal network, are publicly available resources. a systems genetics approach that combines the power of computational predictions with in vivo genetic screens strongly enhances the process of gene function and gene–gene association discovery. genome sequencing and annotation, combined with large-scale molecular experiments to query gene expression and molecular interactions, collectively known as systems biology, have resulted in an enormous wealth in biological databases. yet, it remains a daunting task to use these data to decipher the rules that govern biological systems. one of the most trusted approaches in biology is genetic analysis because of its emphasis on gene function in living organisms. genetics, however, proceeds slowly and unravels small-scale interactions. turning genetics into an effective tool of systems biology requires harnessing the large-scale molecular data for the design and execution of genetic screens. in this work, we test the idea of exploiting a computational approach known as gene prioritization to pre-rank genes for the likelihood of their involvement in a process of interest. by carrying out a gene prioritization–supported genetic screen, we greatly enhance the speed and output of in vivo genetic screens without compromising their sensitivity. these results mean that future genetic screens can be custom-catered for any process of interest and carried out with a speed and efficiency that is comparable to other large-scale molecular experiments. we refer to this combined approach as systems genetics.one of the most time-consuming tasks after performing a gene expression experiment is the biological interpretation of the results by identifying physiologically important associations between the differentially expressed genes. a large part of the relevant functional evidence can be represented in the form of graphs, e.g. metabolic and signaling pathways, protein interaction maps, shared {geneontology} annotations, or literature co-citation relations. such graphs are easily constructed from available genome annotation data. the problem of biological interpretation can then be described as identifying the subgraphs showing the most significant patterns of gene expression. we applied a graph-based extension of our iterative group analysis ({iga}) approach to obtain a statistically rigorous identification of the subgraphs of interest in any evidence graph. we validated the graph-based iterative group analysis ({giga}) by applying it to the classic yeast diauxic shift experiment of {derisi} et al., using {geneontology} and metabolic network information. {giga} reliably identified and summarized all the biological processes discussed in the original publication. visualization of the detected subgraphs allowed the convenient exploration of the results. the method also identified several processes that were not presented in the original paper but are of obvious relevance to the yeast starvation response. {giga} provides a fast and flexible delimitation of the most interesting areas in a microarray experiment, and leads to a considerable speed-up and improvement of the interpretation process.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismswe present a computational framework, called {discern} ({differential} {sparse} regulatory network), to identify informative topological changes in gene-regulator dependence networks inferred on the basis of {mrna} expression datasets within distinct biological states. {discern} takes two expression datasets as input: an expression dataset of diseased tissues from patients with a disease of interest and another expression dataset from matching normal tissues. {discern} estimates the extent to which each gene is perturbed—having distinct regulator connectivity in the inferred gene-regulator dependencies between the disease and normal conditions. this approach has distinct advantages over existing methods. first, {discern} infers conditional dependencies between candidate regulators and genes, where conditional dependence relationships discriminate the evidence for direct interactions from indirect interactions more precisely than pairwise correlation. second, {discern} uses a new likelihood-based scoring function to alleviate concerns about accuracy of the specific edges inferred in a particular network. {discern} identifies perturbed genes more accurately in synthetic data than existing methods to identify perturbed genes between distinct states. in expression datasets from patients with acute myeloid leukemia ({aml}), breast cancer and lung cancer, genes with high {discern} scores in each cancer are enriched for known tumor drivers, genes associated with the biological processes known to be important in the disease, and genes associated with patient prognosis, in the respective cancer. finally, we show that {discern} can uncover potential mechanisms underlying network perturbation by explaining observed epigenomic activity patterns in cancer and normal tissue types more accurately than alternative methods, based on the available epigenomic data from the {encode} project. certain genes can regulate other genes\\' expression and activity levels to perform key biological processes in a cell. understanding how genes affect expression levels among one another is a fundamental goal in molecular biology. new statistical techniques that analyze genome-wide {mrna} expression data obtained from different individuals can improve our understanding of gene interactions. in this paper, we focus on identifying the genes involved in changes in the topology of gene networks between two distinct biological status, for example, between diseased tissue and normal tissue. our method, called {discern}, takes two expression datasets, each from a different condition, for example, cancer and normal tissue, and identifies the genes that are likely to change their neighbors in the estimated gene networks between disease and normal conditions. most analysis methods that compare gene expression datasets from two conditions address the question of which genes are significantly differentially expressed between conditions. the {discern} method addresses a distinct question concerning which genes are significantly rewired in the inferred gene-regulator network in disease tissues. we show that {discern} successfully identifies known disease specific genes, for example, genes whose expression levels are significantly associated with survival time in cancer, genes that have known disease specific driver mutations, as well as genes that have epigenomic evidence of regulatory rewiring between cancer and normal tissue types.a major assumption in many machine learning and data mining algorithms is that the training and future data must be in the same feature space and have the same distribution. however, in many real-world applications, this assumption may not hold. for example, we sometimes have a classification task in one domain of interest, but we only have sufficient training data in another domain of interest, where the latter data may be in a different feature space or follow a different data distribution. in such cases, knowledge transfer, if done successfully, would greatly improve the performance of learning by avoiding much expensive data-labeling efforts. in recent years, transfer learning has emerged as a new learning framework to address this problem. this survey focuses on categorizing and reviewing the current progress on transfer learning for classification, regression, and clustering problems. in this survey, we discuss the relationship between transfer learning and other related machine learning techniques such as domain adaptation, multitask learning and sample selection bias, as well as covariate shift. we also explore some potential future issues in transfer learning research.comparisons between human and rodent {dna} sequences are widely used for the identification of regulatory regions (phylogenetic footprinting), and the importance of such intergenomic comparisons for promoter annotation is expanding. the efficacy of such comparisons for the identification of functional regulatory elements hinges on the evolutionary dynamics of promoter sequences. although it is widely appreciated that conservation of sequence motifs may provide a suggestion of function, it is not known as to what proportion of the functional binding sites in humans is conserved in distant species. in this report, we present an analysis of the evolutionary dynamics of transcription factor binding sites whose function had been experimentally verified in promoters of 51 human genes and compare their sequence to homologous sequences in other primate species and rodents. our results show that there is extensive divergence within the nucleotide sequence of transcription factor binding sites. using direct experimental data from functional studies in both human and rodents for 20 of the regulatory regions, we estimate that 32\\\\%-40\\\\% of the human functional sites are not functional in rodents. this is evidence that there is widespread turnover of transcription factor binding sites. these results have important implications for the efficacy of phylogenetic footprinting and the interpretation of the pattern of evolution in regulatory sequences.the human brain represents a unique evolutionary trajectory. to better understand how the human brain came to be, reilly et al. sought to identify changes in gene expression between mice, macaques, and humans. they compared epigenetic marks in the embryonic cortex, which revealed changes in gene regulation in biological pathways associated with cortical development.motivation: an important problem in systems biology is the inference of biochemical pathways and regulatory networks from postgenomic data. various reverse engineering methods have been proposed in the literature, and it is important to understand their relative merits and shortcomings. in the present paper, we compare the accuracy of reconstructing gene regulatory networks with three different modelling and inference paradigms: (1) relevance networks ({rns}): pairwise association scores independent of the remaining network; (2) graphical gaussian models ({ggms}): undirected graphical models with constraint-based inference, and (3) bayesian networks ({bns}): directed graphical models with score-based inference. the evaluation is carried out on the raf pathway, a cellular signalling network describing the interaction of 11 phosphorylated proteins and phospholipids in human immune system cells. we use both laboratory data from cytometry experiments as well as data simulated from the gold-standard network. we also compare passive observations with active interventions.  results: on gaussian observational data, {bns} and {ggms} were found to outperform {rns}. the difference in performance was not significant for the non-linear simulated data and the cytoflow data, though. also, we did not observe a significant difference between {bns} and {ggms} on observational data in general. however, for interventional data, {bns} outperform {ggms} and {rns}, especially when taking the edge directions rather than just the skeletons of the graphs into account. this suggests that the higher computational costs of inference with {bns} over {ggms} and {rns} are not justified when using only passive observations, but that active interventions in the form of gene knockouts and over-expressions are required to exploit the full potential of {bns}.  availability: data, software and supplementary material are available from http://www.bioss.sari.ac.uk/staff/adriano/research.html.  contact: adriano@bioss.ac.uk, dirk@bioss.ac.uk, grzegorc@statistik.uni-dortmund.de 10.1093/bioinformatics/btl391{background}:in vertebrates, a large part of gene transcriptional regulation is operated by cis-regulatory modules. these modules are believed to be regulating much of the tissue-specificity of gene {expression.results}:we develop a bayesian network approach for identifying cis-regulatory modules likely to regulate tissue-specific expression. the network integrates predicted transcription factor binding site information, transcription factor expression data, and target gene expression data. at its core is a regression tree modeling the effect of combinations of transcription factors bound to a module. a new unsupervised {em}-like algorithm is developed to learn the parameters of the network, including the regression tree {structure.conclusion}:our approach is shown to accurately identify known human liver and erythroid-specific modules. when applied to the prediction of tissue-specific modules in 10 different tissues, the network predicts a number of important transcription factor combinations whose concerted binding is associated to specific expression.transcription factors ({tfs}) play a central role in regulating gene expression by interacting with cis-regulatory {dna} elements associated with their target genes. recent surveys have examined the {dna} binding specificities of most saccharomyces cerevisiae {tfs}, but a comprehensive evaluation of their data has been lacking.a quantitative data set of {rna}, proteins, and metabolites provides an unprecedented starting point to understand, at a systems level, the effects of perturbations on a cell.the cells of organisms as diverse as bacteria and humans can enter stable, nonproliferating quiescent states. quiescent cells of eukaryotic and prokaryotic microorganisms can survive for long periods without nutrients. this alternative state of cells is still poorly understood, yet much benefit is to be gained by understanding it both scientifically and with reference to human health. here, we review our knowledge of one  ” model” quiescent cell population, in cultures of yeast grown to stationary phase in rich media. we outline the importance of understanding quiescence, summarize the properties of quiescent yeast cells, and clarify some definitions of the state. we propose that the processes by which a cell enters into, maintains viability in, and exits from quiescence are best viewed as an environmentally triggered cycle: the cell quiescence cycle. we synthesize what is known about the mechanisms by which yeast cells enter into quiescence, including the possible roles of the protein kinase a, {tor}, protein kinase c, and snf1p pathways. we also discuss selected mechanisms by which quiescent cells maintain viability, including metabolism, protein modification, and redox homeostasis. finally, we outline what is known about the process by which cells exit from quiescence when nutrients again become available.over the past few years, genome sequences have become available from an increasing range of yeast species, which has led to notable advances in our understanding of evolutionary mechanisms in eukaryotes. yeasts offer us a unique opportunity to examine how molecular and reproductive mechanisms combine to affect genome architectures and drive evolutionary changes over a broad range of species. this review summarizes recent progress in understanding the molecular mechanisms \\\\^{a}\\x80\\x94 such as gene duplication, mutation and acquisition of novel genetic material \\\\^{a}\\x80\\x94 that underlie yeast evolutionary genomics. i also discuss how results from yeasts can be extended to other eukaryotes.the protein-protein interaction networks of even well-studied model organisms are sketchy at best, highlighting the continued need for computational methods to help direct experimentalists in the search for novel interactions. this need has prompted the development of a number of methods for predicting protein-protein interactions based on various sources of data and methodologies. the common method for choosing negative examples for training a predictor of protein-protein interactions is based on annotations of cellular localization, and the observation that pairs of proteins that have different localization patterns are unlikely to interact. while this method leads to high quality sets of non-interacting proteins, we find that this choice can lead to biased estimates of prediction accuracy, because the constraints placed on the distribution of the negative examples makes the task easier. the effects of this bias are demonstrated in the context of both sequence-based and non-sequence based features used for predicting protein-protein interactions.we present a genomewide cross-species analysis of regulation for broad-acting transcription factors in yeast. our model for binding site evolution is founded on biophysics: the binding energy between transcription factor and site is a quantitative phenotype of regulatory function, and selection is given by a fitness landscape that depends on this phenotype. the model quantifies conservation, as well as loss and gain, of functional binding sites in a coherent way. its predictions are supported by direct cross-species comparison between four yeast species. we find ubiquitous compensatory mutations within functional sites, such that the energy phenotype and the function of a site evolve in a significantly more constrained way than does its sequence. we also find evidence for substantial evolution of regulatory function involving point mutations as well as sequence insertions and deletions within binding sites. genes lose their regulatory link to a given transcription factor at a rate similar to the neutral point mutation rate, from which we infer a moderate average fitness advantage of functional over nonfunctional sites. in a wider context, this study provides an example of inference of selection acting on a quantitative molecular trait.organisms respond to changes in their environment, and many such responses are initiated at the level of gene transcription. here, we provide evidence for a previously undiscovered mechanism for directing transcriptional regulators to new binding targets in response to an environmental change. we show that repressor-activator protein 1 (rap1), a master regulator of yeast metabolism, binds to an expanded target set after glucose depletion despite decreasing protein levels and no evidence of posttranslational modification. computational analysis predicts that proteins capable of recruiting the chromatin regulator tup1 act to restrict the binding distribution of rap1 in the presence of glucose. deletion of the gene(s) encoding tup1, recruiters of tup1 or chromatin regulators recruited by tup1 cause rap1 to bind specifically and inappropriately to low-glucose targets. these data, combined with whole-genome measurements of nucleosome occupancy and tup1 distribution, provide evidence for a mechanism of dynamic target specification that coordinates the genome-wide distribution of intermediate-affinity {dna} sequence motifs with chromatin-mediated regulation of accessibility to those sites.motivation: the {protein–dna} interactions between transcription factors ({tfs}) and transcription factor binding sites ({tfbss}, also known as {dna} motifs) are critical activities in gene transcription. the identification of the {dna} motifs is a vital task for downstream analysis. unfortunately, the long-range coupling information between different {dna} motifs is still lacking. to fill the void, as the first-of-its-kind study, we have identified the coupling {dna} motif pairs on long-range chromatin interactions in human.biochemical networks are characterized by recurrent patterns and motifs, but the design principles underlying the dynamics of the mammalian intracellular signalling network remain unclear. we systematically analysed decay rates of 134 signalling proteins and investigated their gene expression profiles in response to stimulation to get insights into transcriptional feedback regulation. we found a clear separation of the signalling pathways into flexible and static parts: for each pathway a subgroup of unstable signal inhibitors is transcriptionally induced upon stimulation, while the other constitutively expressed signalling proteins are long-lived. kinetic modelling suggests that this design principle allows for swift feedback regulation and establishes latency phases after signalling, and that it might be an optimal design due to a trade-off between energy efficiency and flexibility.molecular chaperones are known to be involved in many cellular functions, however, a detailed and comprehensive overview of the interactions between chaperones and their cofactors and substrates is still absent. systematic analysis of physical {tap}-tag based protein–protein interactions of all known 63 chaperones in saccharomyces cerevisiae has been carried out. these chaperones include seven small heat-shock proteins, three members of the {aaa}+ family, eight members of the {cct}/{tric} complex, six members of the {prefoldin/gimc} complex, 22 hsp40s, 1 hsp60, 14 hsp70s, and 2 hsp90s. our analysis provides a clear distinction between chaperones that are functionally promiscuous and chaperones that are functionally specific. we found that a given protein can interact with up to 25 different chaperones during its lifetime in the cell. the number of interacting chaperones was found to increase with the average number of hydrophobic stretches of length between one and five in a given protein. importantly, cellular hot spots of chaperone interactions are elucidated. our data suggest the presence of endogenous multicomponent chaperone modules in the cell.we present an approach for the analysis of genome-wide expression data. our method is designed to overcome the limitations of traditional techniques, when applied to large-scale data. rather than alloting each gene to a single cluster, we assign both genes and conditions to context-dependent and potentially overlapping transcription modules. we provide a rigorous definition of a transcription module as the object to be retrieved from the expression data. an efficient algorithm, which searches for the modules encoded in the data by iteratively refining sets of genes and conditions until they match this definition, is established. each iteration involves a linear map, induced by the normalized expression matrix, followed by the application of a threshold function. we argue that our method is in fact a generalization of singular value decomposition, which corresponds to the special case where no threshold is applied. we show analytically that for noisy expression data our approach leads to better classification due to the implementation of the threshold. this result is confirmed by numerical analyses based on in silico expression data. we discuss briefly results obtained by applying our algorithm to expression data from the yeast saccharomyces cerevisiae.we constructed a large-scale functional network model in drosophila melanogaster built around two key transcription factors involved in the process of embryonic segmentation. analysis of the model allowed the identification of a new role for the ubiquitin e3 ligase complex factor {spop}. in drosophila, the gene encoding {spop} is a target of segmentation transcription factors. drosophila {spop} mediates degradation of the jun kinase phosphatase puckered, thereby inducing tumor necrosis factor ({tnf})/eiger–dependent apoptosis. in humans, we found that {spop} plays a conserved role in {tnf}-mediated {jnk} signaling and was highly expressed in 99\\\\% of clear cell renal cell carcinomas ({rccs}), the most prevalent form of kidney cancer. {spop} expression distinguished histological subtypes of {rcc} and facilitated identification of clear cell {rcc} as the primary tumor for metastatic lesions.the eukaryotic genome is packaged as chromatin with nucleosomes comprising its basic structural unit, but the detailed structure of chromatin and its dynamic remodeling in terms of individual nucleosome positions has not been completely defined experimentally for any genome. we used ultra-high–throughput sequencing to map the remodeling of individual nucleosomes throughout the yeast genome before and after a physiological perturbation that causes genome-wide transcriptional changes. nearly 80\\\\% of the genome is covered by positioned nucleosomes occurring in a limited number of stereotypical patterns in relation to transcribed regions and transcription factor binding sites. chromatin remodeling in response to physiological perturbation was typically associated with the eviction, appearance, or repositioning of one or two nucleosomes in the promoter, rather than broader region-wide changes. dynamic nucleosome remodeling tends to increase the accessibility of binding sites for transcription factors that mediate transcriptional changes. however, specific nucleosomal rearrangements were also evident at promoters even when there was no apparent transcriptional change, indicating that there is no simple, globally applicable relationship between chromatin remodeling and transcriptional activity. our study provides a detailed, high-resolution, dynamic map of single-nucleosome remodeling across the yeast genome and its relation to global transcriptional changes. the eukaryotic genome is packed in a systematic hierarchy to accommodate it within the confines of the cell\\'s nucleus. this packing, however, presents an impediment to the transcription machinery when it must access genomic {dna} to regulate gene expression. a fundamental aspect of genome packing is the spooling of {dna} around nucleosomes—structures formed from histone proteins—which must be dislodged during transcription. in this study, we identified all the nucleosome displacements associated with a physiological perturbation causing genome-wide transcriptional changes in the eukaryote saccharomyces cerevisiae. we isolated nucleosomal {dna} before and after subjecting cells to heat shock, then identified the ends of these {dna} fragments and, thereby, the location of nucleosomes along the genome, using ultra-high–throughput sequencing. we identified localized patterns of nucleosome displacement at gene promoters in response to heat shock, and found that nucleosome eviction was generally associated with activation and their appearance with gene repression. nucleosome remodeling generally improved the accessibility of {dna} to transcriptional regulators mediating the response to stresses like heat shock. however, not all nucleosomal remodeling was associated with transcriptional changes, indicating that the relationship between nucleosome repositioning and transcriptional activity is not merely a reflection of competing access to {dna}.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsrecent analyses indicate that differences in protein concentrations are only 20\\\\%-40\\\\% attributable to variable {mrna} levels, underlining the importance of posttranscriptional regulation. generally, protein concentrations depend on the translation rate (which is proportional to the translational activity, {ta}) and the degradation rate. by integrating 12 publicly available large-scale datasets and additional database information of the yeast saccharomyces cerevisiae, we systematically analyzed five factors contributing to {ta}: {mrna} concentration, ribosome density, ribosome occupancy, the codon adaptation index, and a newly developed \"{trna} adaptation index.\" our analysis of the functional relationship between the {ta} and measured protein concentrations suggests that the {ta} follows {michaelis-menten} kinetics. the calculated {ta}, together with measured protein concentrations, allowed us to estimate degradation rates for 4,125 proteins under standard conditions. a significant correlation to recently published degradation rates supports our approach. moreover, based on a newly developed scoring system, we identified and analyzed genes subjected to the posttranscriptional regulation mechanism, translation on demand. next we applied these findings to publicly available data of protein and {mrna} concentrations under four stress conditions. the integration of these measurements allowed us to compare the condition-specific responses at the posttranscriptional level. our analysis of all 62 proteins that have been measured under all four conditions revealed proteins with very specific posttranscriptional stress response, in contrast to more generic responders, which were nonspecifically regulated under several conditions. the concept of specific and generic responders is known for transcriptional regulation. here we show that it also holds true at the posttranscriptional level.exome sequencing studies in complex diseases are challenged by the allelic heterogeneity, large number and modest effect sizes of associated variants on disease risk and the presence of large numbers of neutral variants, even in phenotypically relevant genes. isolated populations with recent bottlenecks offer advantages for studying rare variants in complex diseases as they have deleterious variants that are present at higher frequencies as well as a substantial reduction in rare neutral variation. to explore the potential of the finnish founder population for studying low-frequency (0.5–5\\\\%) variants in complex diseases, we compared exome sequence data on 3,000 finns to the same number of {non-finnish} europeans and discovered that, despite having fewer variable sites overall, the average finn has more low-frequency loss-of-function variants and complete gene knockouts. we then used several well-characterized finnish population cohorts to study the phenotypic effects of 83 enriched loss-of-function variants across 60 phenotypes in 36,262 finns. using a deep set of quantitative traits collected on these cohorts, we show 5 associations (p<5×10−8) including splice variants in {lpa} that lowered plasma lipoprotein(a) levels (p = 1.5×10−117). through accessing the national medical records of these participants, we evaluate the {lpa} finding via mendelian randomization and confirm that these splice variants confer protection from cardiovascular disease ({or} = 0.84, p = 3×10−4), demonstrating for the first time the correlation between very low levels of {lpa} in humans with potential therapeutic implications for cardiovascular diseases. more generally, this study articulates substantial advantages for studying the role of rare variation in complex phenotypes in founder populations like the finns and by combining a unique population genetic history with data from large population cohorts and centralized research access to national health registers. we explored the coding regions of 3,000 finnish individuals with 3,000 {non-finnish} europeans ({nfes}) using whole-exome sequence data, in order to understand how an individual from a bottlenecked population might differ from an individual from an out-bred population. we provide empirical evidence that there are more rare and low-frequency deleterious alleles in finns compared to {nfes}, such that an average finn has almost twice as many low-frequency complete knockouts of a gene. as such, we hypothesized that some of these low-frequency loss-of-function variants might have important medical consequences in humans and genotyped 83 of these variants in 36,000 finns. in doing so, we discovered that completely knocking out the {tsfm} gene might result in inviability or a very severe phenotype in humans and that knocking out the {lpa} gene might confer protection against coronary heart diseases, suggesting that {lpa} is likely to be a good potential therapeutic target.the landscape of human phosphorylation networks has not been systematically explored, representing vast, unchartered territories within cellular signaling networks. although a large number of in vivo phosphorylated residues have been identified by mass spectrometry ({ms})-based approaches, assigning the upstream kinases to these residues requires biochemical analysis of kinase-substrate relationships ({ksrs}). here, we developed a new strategy, called {ceasar}, based on functional protein microarrays and bioinformatics to experimentally identify substrates for 289 unique kinases, resulting in 3656 high-quality {ksrs}. we then generated consensus phosphorylation motifs for each of the kinases and integrated this information, along with information about in vivo phosphorylation sites determined by {ms}, to construct a high-resolution map of phosphorylation networks that connects 230 kinases to 2591 in vivo phosphorylation sites in 652 substrates. the value of this data set is demonstrated through the discovery of a new role for {pka} downstream of btk (bruton\\'s tyrosine kinase) during b-cell receptor signaling. overall, these studies provide global insights into kinase-mediated signaling pathways and promise to advance our understanding of cellular signaling processes in humans.online mendelian inheritance in man ({omim}) is a comprehensive, authoritative and timely knowledgebase of human genes and genetic disorders compiled to support human genetics research and education and the practice of clinical genetics. started by dr victor a. {mckusick} as the definitive reference mendelian inheritance in man, {omim} (http://www.ncbi.nlm.nih.gov/omim/) is now distributed electronically by the national center for biotechnology information, where it is integrated with the entrez suite of databases. derived from the biomedical literature, {omim} is written and edited at johns hopkins university with input from scientists and physicians around the world. each {omim} entry has a full-text summary of a genetically determined phenotype and/or gene and has numerous links to other genetic databases such as {dna} and protein sequence, {pubmed} references, general and locus-specific mutation databases, {hugo} nomenclature, {mapviewer}, {genetests}, patient support groups and many others. {omim} is an easy and straightforward portal to the burgeoning information in human genetics.motivation: in high-dimensional phenotyping screens, a large number of cellular features is observed after perturbing genes by knockouts or {rna} interference. comprehensive analysis of perturbation effects is one of the most powerful techniques for attributing functions to genes, but not much work has been done so far to adapt statistical and computational methodology to the specific needs of large-scale and high-dimensional phenotyping screens.  results: we introduce and compare probabilistic methods to efficiently infer a genetic hierarchy from the nested structure of observed perturbation effects. these hierarchies elucidate the structures of signaling pathways and regulatory networks. our methods achieve two goals: (1) they reveal clusters of genes with highly similar phenotypic profiles, and (2) they order (clusters of) genes according to subset relationships between phenotypes. we evaluate our algorithms in the controlled setting of simulation studies and show their practical use in two experimental scenarios: (1) a data set investigating the response to microbial challenge in drosophila melanogaster, and (2) a compendium of expression profiles of saccharomyces cerevisiae knockout strains. we show that our methods identify biologically justified genetic hierarchies of perturbation effects.  availability: the software used in our analysis is freely available in the r package  nem\\' from www.bioconductor.org  contact: ogt@cs.princeton.edu 10.1093/bioinformatics/btm178various types of genetic modification and selective forces have been implicated in the process of adaptation to novel or adverse environments. however, the underlying molecular mechanisms are not well understood in most natural populations. here we report that a set of yeast strains collected from evolution canyon ({ec}), israel, exhibit an extremely high tolerance to the heavy metal cadmium. we found that cadmium resistance is primarily caused by an enhanced function of a metal efflux pump, {pca1}. molecular analyses demonstrate that this enhancement can be largely attributed to mutations in the promoter sequence, while mutations in the coding region have a minor effect. reconstruction experiments show that three single nucleotide substitutions in the {pca1} promoter quantitatively increase its activity and thus enhance the cells\\' cadmium resistance. comparison among different yeast species shows that the critical nucleotides found in {ec} strains are conserved and functionally important for cadmium resistance in other species, suggesting that they represent an ancestral type. however, these nucleotides had diverged in most saccharomyces cerevisiae populations, which gave cells growth advantages under conditions where cadmium is low or absent. our results provide a rare example of a selective sweep in yeast populations driven by a tradeoff in metal resistance.targeted gene perturbations have become a major tool to gain insight into complex cellular processes. in combination with the measurement of downstream effects via {dna} microarrays, this approach can be used to gain insight into signaling pathways. nested effects models were first introduced by markowetz et al. as a probabilistic method to reverse engineer signaling cascades based on the nested structure of downstream perturbation effects. the basic framework was substantially extended later on by fr\\\\\"{o}hlich et al., markowetz et al., and tresch and markowetz. in this paper, we present a review of the complete methodology with a detailed comparison of so far proposed algorithms on a qualitative and quantitative level. as an application, we present results on estimating the signaling network between 13 genes in the {er}-alpha pathway of human {mcf}-7 breast cancer cells. comparison with the literature shows a substantial overlap.the effects of genetic variants on phenotypic traits often depend on environmental and physiological conditions, but such gene\\\\^{a}\\x80\\x93environment interactions are poorly understood. recently developed approaches that treat transcript abundances of thousands of genes as quantitative traits offer the opportunity to broadly characterize the architecture of gene\\\\^{a}\\x80\\x93environment interactions. we examined the genetic and molecular basis of variation in gene expression between two yeast strains ({by} and {rm}) grown in two different conditions (glucose and ethanol as carbon sources). we observed that most transcripts vary by strain and condition, with 2,996, 3,448, and 2,037 transcripts showing significant strain, condition, and strain\\\\^{a}\\x80\\x93condition interaction effects, respectively. we expression profiled over 100 segregants derived from a cross between {by} and {rm} in both growth conditions, and identified 1,555 linkages for 1,382 transcripts that show significant gene\\\\^{a}\\x80\\x93environment interaction. at the locus level, local linkages, which usually correspond to polymorphisms in cis-regulatory elements, tend to be more stable across conditions, such that they are more likely to show the same effect or the same direction of effect across conditions. distant linkages, which usually correspond to polymorphisms influencing trans-acting factors, are more condition-dependent, and often show effects in different directions in the two conditions. we characterized a locus that influences expression of many growth-related transcripts, and showed that the majority of the variation is explained by polymorphism in the gene {ira2}. the {rm} allele of {ira2} appears to inhibit {ras/pka} signaling more strongly than the {by} allele, and has undergone a change in selective pressure. our results provide a broad overview of the genetic architecture of gene\\\\^{a}\\x80\\x93environment interactions, as well as a detailed molecular example, and lead to key insights into how the effects of different classes of regulatory variants are modulated by the environment. these observations will guide the design of studies aimed at understanding the genetic basis of complex traits. individuals frequently encounter different environmental conditions, and the physiological and behavioral responses to these conditions can depend on an individual\\'s genetic makeup. this phenomenon is known as gene\\\\^{a}\\x80\\x93environment interaction. for example, individuals who are infected with the plasmodium falciparum parasite are susceptible to malaria, but not if they carry the sickle-cell allele of hemoglobin. the general properties of gene\\\\^{a}\\x80\\x93environment interaction are poorly understood, and a better understanding is essential if individuals are to make informed health choices guided by their genomic information. we have investigated gene\\\\^{a}\\x80\\x93environment interaction on a genomic level, characterizing its role in over 4,000 traits at once by investigating natural variation in yeast gene expression. we compared lab and vineyard strains of yeast growing in two conditions (glucose and ethanol as carbon sources) in which they adopt two different metabolic states: fermentation and aerobic respiration, respectively. we show that gene\\\\^{a}\\x80\\x93environment interaction is a common phenomenon, describe how different classes of genetic variants affect the nature of the interactions, and provide detailed molecular examples of interactions.living organisms are robust to a great variety of genetic changes. gene regulation networks and metabolic pathways self-organize and reaccommodate to make the organism perform with stability and reliability under many point mutations, gene duplications and gene deletions. at the same time, living organisms are evolvable, which means that these kind of genetic perturbations can eventually make the organism acquire new functions and adapt to new environments. it is still an open problem to determine how robustness and evolvability blend together at the genetic level to produce stable organisms that yet can change and evolve. here we address this problem by studying the robustness and evolvability of the attractor landscape of genetic regulatory network models under the process of gene duplication followed by divergence. we show that an intrinsic property of this kind of networks is that, after the divergence of the parent and duplicate genes, with a high probability the previous phenotypes, encoded in the attractor landscape of the network, are preserved and new ones might appear. the above is true in a variety of network topologies and even for the case of extreme divergence in which the duplicate gene bears almost no relation with its parent. our results indicate that networks operating close to the so-called  ” critical regime” exhibit the maximum robustness and evolvability simultaneously.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsmotivation: understanding the mechanisms that govern nucleosome positioning over genomes in vivo is essential for unraveling the role of chromatin organization in transcriptional regulation. until now, models for predicting genome-wide nucleosome occupancy have assumed that the {dna} associations of neighboring nucleosomes on the genome are independent. we present a new model that relaxes this independence assumption by modeling interactions between adjacent nucleosomes.  results: we show that modeling interactions between adjacent nucleosomes improves genome-wide nucleosome occupancy predictions in an in vitro system that includes only nucleosomes and purified {dna}, where the resulting model has a preference for short spacings (linkers) of less than 20 bp in length between neighboring nucleosomes. since nucleosome occupancy in vitro depends only on properties intrinsic to nucleosomes, these results suggest that the interactions we find are intrinsic to nucleosomes and do not depend on other factors, such as transcription factors and chromatin remodelers. we also show that modeling these intrinsic interactions significantly improves genome-wide predictions of nucleosome occupancy in vivo.  contact: eran.segal@weizmann.ac.il  supplementary information: supplementary data are available at bioinformatics online. 10.1093/bioinformatics/btp216the large volume of genome-scale data that is being produced and made available in databases on the world wide web is demanding the development of integrated mathematical models of cellular processes. the analysis of reconstructed metabolic networks as systems leads to the development of an  in silico  or computer representation of collections of cellular metabolic constituents, their interactions and their integrated function as a whole. the use of quantitative analysis methods to generate testable hypotheses and drive experimentation at a whole-genome level signals the advent of a systemic modeling approach to cellular and molecular biology.a three-dimensional chromatin state underpins the structural and functional basis of the genome by bringing regulatory elements and genes into close spatial proximity to ensure proper, cell-type-specific gene expression profiles. here, we performed {hi-c} chromosome conformation capture sequencing to investigate how three-dimensional chromatin organization is disrupted in the context of copy-number variation, long-range epigenetic remodeling, and atypical gene expression programs in prostate cancer. we find that cancer cells retain the ability to segment their genomes into megabase-sized topologically associated domains ({tads}); however, these domains are generally smaller due to establishment of additional domain boundaries. interestingly, a large proportion of the new cancer-specific domain boundaries occur at regions that display copy-number variation. notably, a common deletion on 17p13.1 in prostate cancer spanning the {tp53} tumor suppressor locus results in bifurcation of a single {tad} into two distinct smaller {tads}. change in domain structure is also accompanied by novel cancer-specific chromatin interactions within the {tads} that are enriched at regulatory elements such as enhancers, promoters, and insulators, and associated with alterations in gene expression. we also show that differential chromatin interactions across regulatory regions occur within long-range epigenetically activated or silenced regions of concordant gene activation or repression in prostate cancer. finally, we present a novel visualization tool that enables integrated exploration of {hi-c} interaction data, the transcriptome, and epigenome. this study provides new insights into the relationship between long-range epigenetic and genomic dysregulation and changes in higher-order chromatin interactions in cancer. {\\\\copyright} 2016 taberlay et al.; published by cold spring harbor laboratory press.multiple characteristics of individual cells define cell types and their physiological states. spitzer et al. quantitated the abundance of 39 different cell surface proteins or transcription factors on individual cells of the mouse immune system. they used these measurements to create a map that clustered similar individual cells into groups corresponding to cell type and function. their extensible experimental platform will allow the inclusion of other data types and data from independent laboratories.we integrate heterogeneous molecular data of 11,289 tumors and 1,001 cell lines we measure the response of 1,001 cancer cell lines to 265 anti-cancer drugs we uncover numerous oncogenic aberrations that sensitize to an anti-cancer drug our study forms a resource to identify therapeutic options for cancer sub-populations systematic studies of cancer genomes have provided unprecedented insights into the molecular nature of cancer. using this information to guide the development and application of therapies in the clinic is challenging. here, we report how cancer-driven alterations identified in 11,289 tumors from 29 tissues (integrating somatic mutations, copy number alterations, {dna} methylation, and gene expression) can be mapped onto 1,001 molecularly annotated human cancer cell lines and correlated with sensitivity to 265 drugs. we find that cell lines faithfully recapitulate oncogenic alterations identified in tumors, find that many of these associate with drug sensitivity/resistance, and highlight the importance of tissue lineage in mediating drug response. logic-based modeling uncovers combinations of alterations that sensitize to drugs, while machine learning demonstrates the relative importance of different data types in predicting drug response. our analysis and datasets are rich resources to link genotypes with cellular phenotypes and to identify therapeutic options for selected cancer sub-populations.multicellular organismal development is controlled by a complex network of transcription factors, promoters and enhancers. although reliable computational and experimental methods exist for enhancer detection, prediction of their target genes remains a major challenge. on the basis of available literature and {chip}-seq and {chip}-chip data for enhanceosome factor p300 and the transcriptional regulator gli3, we found that genomic proximity and conserved synteny predict target genes with a relatively low recall of 12–27\\\\% within 2\\u2009mb intervals centered at the enhancers. here, we show that functional similarities between enhancer binding proteins and their transcriptional targets and proximity in the protein–protein interactome improve prediction of target genes. we used all four features to train random forest classifiers that predict target genes with a recall of 58\\\\% in 2\\u2009mb intervals that may contain dozens of genes, representing a better than two-fold improvement over the performance of prediction based on single features alone. genome-wide {chip} data is still relatively poorly understood, and it remains difficult to assign biological significance to binding events. our study represents a first step in integrating various genomic features in order to elucidate the genomic network of long-range regulatory interactions.motivation: cellular signaling pathways, which are not modulated on a transcriptional level, cannot be directly deduced from expression profiling experiments. the situation changes, when external interventions such as {rna} interference or gene knock-outs come into play. even if the expression of the signaling genes is not changed, secondary effects in downstream genes shed light on the pathway, and allow partial reconstruction of its {topology.results}: we introduce an algorithm to infer non-transcriptional pathway features based on differential gene expression in silencing assays. we demonstrate the power of our algorithm in the controlled setting of simulation studies, and explain its practical use in the context of an {rna} interference dataset investigating the response to microbial challenge in drosophila {melanogaster.contact}: florian.markowetz@molgen.mpg.dethe tissues of a developing embryo are simultaneously patterned, moved and differentiated according to an exchange of information between their constituent cells. we argue that these complex self-organizing phenomena can only be fully understood with quantitative mathematical frameworks that allow specific hypotheses to be formulated and tested. the quantitative and dynamic imaging of growing embryos at the molecular, cellular and tissue level is the key experimental advance required to achieve this interaction between theory and experiment. here we describe how mathematical modelling has become an invaluable method to integrate quantitative biological information across temporal and spatial scales, serving to connect the activity of regulatory molecules with the morphological development of organisms.the genomes of bacterial species show enormous plasticity in the function of individual genes, in genome organization and in regulatory organization. over millions of years, both bacterial genes and their genomes have been extensively reorganized and adapted so that bacteria occupy virtually every environmental niche on the earth. in addition, changes have occurred in the regulatory circuitry that controls cell operations, cell-cycle progression and responses to environmental signals. the mechanisms that underlie the adaptation of the bacterial regulatory circuitry are crucial for understanding the bacterial biosphere and have important roles in the emergence of antibiotic resistance.summary: motifs are small connected subnetworks that a network displays in significantly higher frequencies than would be expected for a random network. they have recently gathered much attention as a concept to uncover structural design principles of complex biological networks. {fanmod} is a tool for fast network motif detection; it relies on recently developed algorithms to improve the efficiency of network motif detection by some orders of magnitude over existing tools. this facilitates the detection of larger motifs in bigger networks than previously possible. additional benefits of {fanmod} are the ability to analyze colored networks, a graphical user interface and the ability to export results to a variety of machine- and human-readable file formats including comma-separated values and {html}.in recent years, the advent of high-throughput assays, coupled with their diminishing cost, has facilitated a systems approach to biology. as a consequence, massive amounts of data are currently being generated, requiring efficient methodology aimed at the reduction of scale. whole-genome transcriptional profiling is a standard component of systems-level analyses, and to reduce scale and improve inference clustering genes is common. since clustering is often the first step toward generating hypotheses, cluster quality is critical. conversely, because the validation of cluster-driven hypotheses is indirect, it is critical that quality clusters not be obtained by subjective means. in this paper, we present a new objective-based clustering method and demonstrate that it yields high-quality results. our method, modulated modularity clustering ({mmc}), seeks community structure in graphical data. {mmc} modulates the connection strengths of edges in a weighted graph to maximize an objective function (called modularity) that quantifies community structure. the result of this maximization is a clustering through which tightly-connected groups of vertices emerge. our application is to systems genetics, and we quantitatively compare {mmc} both to the hierarchical clustering method most commonly employed and to three popular spectral clustering approaches. we further validate {mmc} through analyses of human and drosophila melanogaster expression data, demonstrating that the clusters we obtain are biologically meaningful. we show {mmc} to be effective and suitable to applications of large scale. in light of these features, we advocate {mmc} as a standard tool for exploration and hypothesis generation.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsinferring global regulatory networks ({grns}) from genome-wide data is a computational challenge central to the field of systems biology. although the primary data currently used to infer {grns} consist of gene expression and proteomics measurements, there is a growing abundance of alternate data types that can reveal regulatory interactions, e.g. {chip}-chip, literature-derived interactions, protein-protein interactions. {grn} inference requires the development of integrative methods capable of using these alternate data as priors on the {grn} structure. each source of structure priors has its unique biases and inherent potential errors; thus, {grn} methods using these data must be robust to noisy inputs. we developed two methods for incorporating structure priors into {grn} inference. both methods [modified elastic net ({men}) and bayesian best subset regression ({bbsr})] extend the previously described inferelator framework, enabling the use of prior information. we test our methods on one synthetic and two bacterial datasets, and show that both {men} and {bbsr} infer accurate {grns} even when the structure prior used has significant amounts of error (>90\\\\% erroneous interactions). we find that {bbsr} outperforms {men} at inferring {grns} from expression data and noisy structure priors. code, datasets and networks presented in this article are available at http://bonneaulab.bio.nyu.edu/software.html. supplementary data are available at bioinformatics online.cross-species comparison has emerged as a powerful paradigm for predicting cis -regulatory modules ({crms}) and understanding their evolution. the comparison requires reliable sequence alignment, which remains a challenging task for less conserved noncoding sequences. furthermore, the existing models of {dna} sequence evolution generally do not explicitly treat the special properties of {crm} sequences. to address these limitations, we propose a model of {crm} evolution that captures different modes of evolution of functional transcription factor binding sites ({tfbss}) and the background sequences. a particularly novel aspect of our work is a probabilistic model of gains and losses of {tfbss}, a process being recognized as an important part of regulatory sequence evolution. we present a computational framework that uses this model to solve the problems of {crm} alignment and prediction. our alignment method is similar to existing methods of statistical alignment but uses the conserved binding sites to improve alignment. our {crm} prediction method deals with the inherent uncertainties of binding site annotations and sequence alignment in a probabilistic framework. in simulated as well as real data, we demonstrate that our program is able to improve both alignment and prediction of {crm} sequences over several state-of-the-art methods. finally, we used alignments produced by our program to study binding site conservation in genome-wide binding data of key transcription factors in the drosophila blastoderm, with two intriguing results: (i) the factor-bound sequences are under strong evolutionary constraints even if their neighboring genes are not expressed in the blastoderm and (ii) binding sites in distal bound sequences (relative to transcription start sites) tend to be more conserved than those in proximal regions. our approach is implemented as software, {emma} (evolutionary model-based cis -regulatory module analysis), ready to be applied in a broad biological context.in the last few years we have witnessed tremendous progress in detecting associations between genetic variations and complex traits. while genome-wide association studies have been able to discover genomic regions that may influence many common human diseases, these discoveries created an urgent need for methods that extend the knowledge of genotype-phenotype relationships to the level of the molecular mechanisms behind them. to address this emerging need, computational approaches increasingly utilize a pathway-centric perspective. these new methods often utilize known or predicted interactions between genes and/or gene products. in this review, we survey recently developed network based methods that attempt to bridge the genotype-phenotype gap. we note that although these methods help narrow the gap between genotype and phenotype relationships, these approaches alone cannot provide the precise details of underlying mechanisms and current research is still far from closing the gap.a major challenge in systems biology is the ability to model complex regulatory interactions, such as gene regulatory networks, and a number of computational approaches have been developed over recent years to address this challenge. this paper reviews a number of these approaches, with a focus on probabilistic graphical models and the integration of diverse data sets, such as gene expression and transcription factor binding site location and activity.{imp} (integrative multi-species prediction), originally released in 2012, is an interactive web server that enables molecular biologists to interpret experimental results and to generate hypotheses in the context of a large cross-organism compendium of functional predictions and networks. the system provides biologists with a framework to analyze their candidate gene sets in the context of functional networks, expanding or refining their sets using functional relationships predicted from integrated high-throughput data. {imp} 2.0 integrates updated prior knowledge and data collections from the last three years in the seven supported organisms (homo sapiens, mus musculus, rattus norvegicus, drosophila melanogaster, danio rerio, caenorhabditis elegans, and saccharomyces cerevisiae) and extends function prediction coverage to include human disease. {imp} identifies homologs with conserved functional roles for disease knowledge transfer, allowing biologists to analyze disease contexts and predictions across all organisms. additionally, {imp} 2.0 implements a new flexible platform for experts to generate custom hypotheses about biological processes or diseases, making sophisticated data-driven methods easily accessible to researchers. {imp} does not require any registration or installation and is freely available for use at http://imp.princeton.edu. {\\\\copyright} the author(s) 2015. published by oxford university press on behalf of nucleic acids research.a major mode of gene expression evolution is based on changes in cis-regulatory elements ({cres}) whose function critically depends on the presence of transcription factor\\\\^{a}\\x80\\x93binding sites ({tfbs}). because {cres} experience extensive {tfbs} turnover even with conserved function, alignment-based studies of {cre} sequence evolution are limited to very closely related species. here, we propose an alternative approach based on a stochastic model of {tfbs} turnover. we implemented a maximum likelihood model that permits variable turnover rates in different parts of the species tree. this model can be used to detect changes in turnover rate as a proxy for differences in the selective pressures acting on {tfbs} in different clades. we applied this method to five {tfbs} in the fungi methionine biosynthesis pathway and three {tfbs} in the {hoxa} clusters of vertebrates. we find that the estimated turnover rate is generally high, with half-life ranging between \\\\^{a}\\x88¼5 and 150 my and a mode around tens of millions of years. this rate is consistent with the finding that even functionally conserved enhancers can show very low sequence similarity. we also detect statistically significant differences in the equilibrium densities of estrogen- and progesterone-response elements in the {hoxa} clusters between mammal and nonmammal vertebrates. even more extreme clade-specific differences were found in the fungal data. we conclude that stochastic models of {tfbs} turnover enable the detection of shifts in the selective pressures acting on {cres} in different {organisms.the} analysis tool, called {creto} ({cis-regulatory} element {turn-over}) can be downloaded from {http://www.bioinf.uni-leipzig.de/software}/creto/.we present single-cell clustering using bifurcation analysis ({scuba}), a novel computational method for extracting lineage relationships from single-cell gene expression data and modeling the dynamic changes associated with cell differentiation. {scuba} draws techniques from nonlinear dynamics and stochastic differential equation theories, providing a systematic framework for modeling complex processes involving multilineage specifications. by applying {scuba} to analyze two complementary, publicly available datasets we successfully reconstructed the cellular hierarchy during early development of mouse embryos, modeled the dynamic changes in gene expression patterns, and predicted the effects of perturbing key transcriptional regulators on inducing lineage biases. the results were robust with respect to experimental platform differences between {rt}-{pcr} and {rna} sequencing. we selectively tested our predictions in nanog mutants and found good agreement between {scuba} predictions and the experimental data. we further extended the utility of {scuba} by developing a method to reconstruct missing temporal-order information from a typical single-cell dataset. analysis of a hematopoietic dataset suggests that our method is effective for reconstructing gene expression dynamics during human b-cell development. in summary, {scuba} provides a useful single-cell data analysis tool that is well-suited for the investigation of developmental processes.with the development of various high throughput technologies and analysis methods, researchers can study different aspects of a biological phenomenon simultaneously or one aspect repeatedly with different experimental techniques and analysis methods. the output from each study is a rank list of components of interest. aggregation of the rank lists of components, such as proteins, genes and single nucleotide variants ({snv}), produced by these experiments has been proven to be helpful in both filtering the noise and bringing forth a more complete understanding of the biological problems. current available rank aggregation methods do not consider the network information that has been observed to provide vital contributions in many data integration studies. we developed network tuned rank aggregation methods incorporating network information and demonstrated its superior performance over aggregation methods without network information.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsmotivation: recently, a range of new statistics have become available for the alignment-free comparison of two sequences based on k-tuple word content. here, we extend these statistics to the simultaneous comparison of more than two sequences. our suite of statistics contains, first,  and , extensions of statistics for pairwise comparison of the joint k-tuple content of all the sequences, and second, ,  and , averages of sums of pairwise comparison statistics. the two tasks we consider are, first, to identify sequences that are similar to a set of target sequences, and, second, to measure the similarity within a set of sequences.circuit diagrams and unified modeling language diagrams are just two examples of standard visual languages that help accelerate work by promoting regularity, removing ambiguity and enabling software tool support for communication of complex information. ironically, despite having one of the highest ratios of graphical to textual information, biology still lacks standard graphical notations. the recent deluge of biological knowledge makes addressing this deficit a pressing concern. toward this goal, we present the systems biology graphical notation ({sbgn}), a visual language developed by a community of biochemists, modelers and computer scientists. {sbgn} consists of three complementary languages: process diagram, entity relationship diagram and activity flow diagram. together they enable scientists to represent networks of biochemical interactions in a standard, unambiguous way. we believe that {sbgn} will foster efficient and accurate representation, visualization, storage, exchange and reuse of information on all kinds of biological knowledge, from gene regulation, to metabolism, to cellular signaling.genomic sequencing has made it clear that a large fraction of the genes specifying the core biological functions are shared by all eukaryotes. knowledge of the biological role of such shared proteins in one organism can often be transferred to other organisms. the goal of the gene ontology consortium is to produce a dynamic, controlled vocabulary that can be applied to all eukaryotes even as knowledge of gene and protein roles in cells is accumulating and changing. to this end, three independent ontologies accessible on the {world-wide} web (http://www.geneontology.org) are being constructed: biological process, molecular function and cellular component.a broadly applicable algorithm for computing maximum likelihood estimates from incomplete data is presented at various levels of generality. theory showing the monotone behaviour of the likelihood and convergence of the algorithm is derived. many examples are sketched, including missing value situations, applications to grouped, censored or truncated data, finite mixture models, variance component estimation, hyperparameter estimation, iteratively reweighted least squares and factor analysis.given the functional interdependencies between the molecular components in a human cell, a disease is rarely a consequence of an abnormality in a single gene, but reflects the perturbations of the complex intracellular and intercellular network that links tissue and organ systems. the emerging tools of network medicine offer a platform to explore systematically not only the molecular complexity of a particular disease, leading to the identification of disease modules and pathways, but also the molecular relationships among apparently distinct (patho)phenotypes. advances in this direction are essential for identifying new disease genes, for uncovering the biological significance of disease-associated mutations identified by genome-wide association studies and full-genome sequencing, and for identifying drug targets and biomarkers for complex diseases.to comprehend the multipartite organization of large-scale biological and social systems, we introduce an information theoretic approach that reveals community structure in weighted and directed networks. we use the probability flow of random walks on a network as a proxy for information flows in the real system and decompose the network into modules by compressing a description of the probability flow. the result is a map that both simplifies and highlights the regularities in the structure and their relationships. we illustrate the method by making a map of scientific communication as captured in the citation patterns of >6,000 journals. we discover a multicentric organization with fields that vary dramatically in size and degree of integration into the network of science. along the backbone of the network—including physics, chemistry, molecular biology, and medicine—information flows bidirectionally, but the map reveals a directional pattern of citation from the applied fields to the basic sciences.the identification of genes associated with hereditary disorders has contributed to improving medical care and to a better understanding of gene functions, interactions, and pathways. however, there are well over 1500 mendelian disorders whose molecular basis remains unknown. at present, methods such as linkage analysis can identify the chromosomal region in which unknown disease genes are located, but the regions could contain up to hundreds of candidate genes. in this work, we present a method for prioritization of candidate genes by use of a global network distance measure, random walk analysis, for definition of similarities in protein-protein interaction networks. we tested our method on 110 disease-gene families with a total of 783 genes and achieved an area under the {roc} curve of up to 98\\\\% on simulated linkage intervals of 100 genes surrounding the disease gene, significantly outperforming previous methods based on local distance measures. our results not only provide an improved tool for positional-cloning projects but also add weight to the assumption that phenotypically similar diseases are associated with disturbances of subnetworks within the larger protein interactome that extend beyond the disease proteins themselves.{background}:high-throughput sequencing technologies, such as the illumina genome analyzer, are powerful new tools for investigating a wide range of biological and medical questions. statistical and computational methods are key for drawing meaningful and accurate conclusions from the massive and complex datasets generated by the sequencers. we provide a detailed evaluation of statistical methods for normalization and differential expression ({de}) analysis of illumina transcriptome sequencing ({mrna}-seq) {data.results}:we compare statistical methods for detecting genes that are significantly {de} between two types of biological samples and find that there are substantial differences in how the test statistics handle low-count genes. we evaluate how {de} results are affected by features of the sequencing platform, such as, varying gene lengths, base-calling calibration method (with and without phi x control lane), and flow-cell/library preparation effects. we investigate the impact of the read count normalization method on {de} results and show that the standard approach of scaling by total lane counts (e.g., {rpkm}) can bias estimates of {de}. we propose more general quantile-based normalization procedures and demonstrate an improvement in {de} {detection.conclusions}:our results have significant practical and methodological implications for the design and analysis of {mrna}-seq experiments. they highlight the importance of appropriate statistical methods for normalization and {de} inference, to account for features of the sequencing platform that could impact the accuracy of results. they also reveal the need for further research in the development of statistical and computational methods for {mrna}-seq.with the increase in genomewide experiments and the sequencing of multiple genomes, the analysis of large data sets has become commonplace in biology. it is often the case that thousands of features in a genomewide data set are tested against some null hypothesis, where a number of features are expected to be significant. here we propose an approach to measuring statistical significance in these genomewide studies based on the concept of the false discovery rate. this approach offers a sensible balance between the number of true and false positives that is automatically calibrated and easily interpreted. in doing so, a measure of statistical significance called the q value is associated with each tested feature. the q value is similar to the well known p value, except it is a measure of significance in terms of the false discovery rate rather than the false positive rate. our approach avoids a flood of false positive results, while offering a more liberal criterion than what has been used in genome scans for linkage.nucleosome organization is critical for gene regulation. in living cells this organization is determined by multiple factors, including the action of chromatin remodellers, competition with site-specific {dna}-binding proteins, and the {dna} sequence preferences of the nucleosomes themselves. however, it has been difficult to estimate the relative importance of each of these mechanisms in vivo, because in vivo nucleosome maps reflect the combined action of all influencing factors. here we determine the importance of nucleosome {dna} sequence preferences experimentally by measuring the genome-wide occupancy of nucleosomes assembled on purified yeast genomic {dna}. the resulting map, in which nucleosome occupancy is governed only by the intrinsic sequence preferences of nucleosomes, is similar to in vivo nucleosome maps generated in three different growth conditions. in vitro, nucleosome depletion is evident at many transcription factor binding sites and around gene start and end sites, indicating that nucleosome depletion at these sites in vivo is partly encoded in the genome. we confirm these results with a micrococcal nuclease-independent experiment that measures the relative affinity of nucleosomes for approximately 40,000 double-stranded 150-base-pair oligonucleotides. using our in vitro data, we devise a computational model of nucleosome sequence preferences that is significantly correlated with in vivo nucleosome occupancy in caenorhabditis elegans. our results indicate that the intrinsic {dna} sequence preferences of nucleosomes have a central role in determining the organization of nucleosomes in vivo.stochastic mechanisms are ubiquitous in biological systems. biochemical reactions that involve small numbers of molecules are intrinsically noisy, being dominated by large concentration fluctuations1, 2, 3. this intrinsic noise has been implicated in the random lysis/lysogeny decision of bacteriophage-4, in the loss of synchrony of circadian clocks5, 6 and in the decrease of precision of cell signals7. we sought to quantitatively investigate the extent to which the occurrence of molecular fluctuations within single cells (biochemical noise) could explain the variation of gene expression levels between cells in a genetically identical population (phenotypic noise). we have isolated the biochemical contribution to phenotypic noise from that of other noise sources by carrying out a series of differential measurements. we varied independently the rates of transcription and translation of a single fluorescent reporter gene in the chromosome of bacillus subtilis, and we quantitatively measured the resulting changes in the phenotypic noise characteristics. we report that of these two parameters, increased translational efficiency is the predominant source of increased phenotypic noise. this effect is consistent with a stochastic model of gene expression in which proteins are produced in random and sharp bursts. our results thus provide the first direct experimental evidence of the biochemical origin of phenotypic noise, demonstrating that the level of phenotypic variation in an isogenic population can be regulated by genetic parameters.histone modifications are important markers of function and chromatin state, yet the {dna} sequence elements that direct them to specific genomic locations are poorly understood. here, we identify hundreds of quantitative trait loci, genome-wide, that affect histone modification or {rna} polymerase {ii} (pol {ii}) occupancy in yoruba lymphoblastoid cell lines ({lcls}). in many cases, the same variant is associated with quantitative changes in multiple histone marks and pol {ii}, as well as in deoxyribonuclease i sensitivity and nucleosome positioning. transcription factor binding site polymorphisms are correlated overall with differences in local histone modification, and we identify specific transcription factors whose binding leads to histone modification in {lcls}. furthermore, variants that affect chromatin at distal regulatory sites frequently also direct changes in chromatin and gene expression at associated promoters.though many individual transcription factors are known to regulate hematopoietic differentiation, major aspects of the global architecture of hematopoiesis remain unknown. here, we profiled gene expression in 38 distinct purified populations of human hematopoietic cells and used probabilistic models of gene expression and analysis of cis-elements in gene promoters to decipher the general organization of their regulatory circuitry. we identified modules of highly coexpressed genes, some of which are restricted to a single lineage but most of which are expressed at variable levels across multiple lineages. we found densely interconnected cis-regulatory circuits and a large number of transcription factors that are differentially expressed across hematopoietic states. these findings suggest a more complex regulatory system for hematopoiesis than previously assumed. copyright {\\\\copyright} 2011 elsevier inc. all rights reserved.epistasis analysis, which reports on the extent to which the function of one gene depends on the presence of a second, is a powerful tool for studying the functional organization of the cell. systematic genome-wide studies of epistasis, however, have been limited, with the majority of data being collected in the budding yeast, saccharomyces cerevisiae. here we present two \\'pombe epistasis mapper\\' strategies, {pem}-1 and {pem}-2, which allow for high-throughput double mutant generation in the fission yeast, s. pombe. these approaches take advantage of a previously undescribed, recessive, cycloheximide-resistance mutation. both systems can be used for genome-wide screens or for the generation of high-density, quantitative epistatic miniarray profiles ({e-maps}). since s. cerevisiae and s. pombe are evolutionary distant, this methodology will provide insight into conserved biological pathways that are present in s. pombe, but not s. cerevisiae, and will enable a comprehensive analysis of the conservation of genetic interaction networks.transcription regulation networks control the expression of genes. the transcription networks of well-studied microorganisms appear to be made up of a small set of recurring regulation patterns, called network motifs. the same network motifs have recently been found in diverse organisms from bacteria to humans, suggesting that they serve as basic building blocks of transcription networks. here i review network motifs and their functions, with an emphasis on experimental studies. network motifs in other biological networks are also mentioned, including signalling and neuronal networks.motivation: the enormous amount of short reads generated by the new {dna} sequencing technologies call for the development of fast and accurate read alignment programs. a first generation of hash table-based methods has been developed, including {maq}, which is accurate, feature rich and fast enough to align short reads from a single individual. however, {maq} does not support gapped alignment for single-end reads, which makes it unsuitable for alignment of longer reads where indels may occur frequently. the speed of {maq} is also a concern when the alignment is scaled up to the resequencing of hundreds of individuals.a key goal of biology is to construct networks that predict complex system behavior. we combine multiple types of molecular data, including genotypic, expression, transcription factor binding site ({tfbs}), and protein-protein interaction ({ppi}) data previously generated from a number of yeast experiments, in order to reconstruct causal gene networks. networks based on different types of data are compared using metrics devised to assess the predictive power of a network. we show that a network reconstructed by integrating genotypic, {tfbs} and {ppi} data is the most predictive. this network is used to predict causal regulators responsible for hot spots of gene expression activity in a segregating yeast population. we also show that the network can elucidate the mechanisms by which causal regulators give rise to larger-scale changes in gene expression activity. we then prospectively validate predictions, providing direct experimental evidence that predictive networks can be constructed by integrating multiple, appropriate data types.this paper presents a bayesian method for constructing probabilistic networks from databases. in particular, we focus on constructing bayesian belief networks. potential applications include computer-assisted hypothesis testing, automated scientific discovery, and automated construction of probabilistic expert systems. we extend the basic method to handle missing data and hidden (latent) variables. we show how to perform probabilistic inference by averaging over the inferences of multiple belief networks. results are presented of a preliminary evaluation of an algorithm for constructing a belief network from a database of cases. finally, we relate the methods in this paper to previous work, and we discuss open problems.biologists have long used model organisms to study human diseases, particularly when the model bears a close resemblance to the disease. we present a method that quantitatively and systematically identifies nonobvious equivalences between mutant phenotypes in different species, based on overlapping sets of orthologous genes from human, mouse, yeast, worm, and plant (212,542 gene-phenotype associations). these orthologous phenotypes, or phenologs, predict unique genes associated with diseases. our method suggests a yeast model for angiogenesis defects, a worm model for breast cancer, mouse models of autism, and a plant model for the neural crest defects associated with waardenburg syndrome, among others. using these models, we show that {sox13} regulates angiogenesis, and that {sec23ip} is a likely waardenburg gene. phenologs reveal functionally coherent, evolutionarily conserved gene networks—many predating the plant-animal divergence—capable of identifying candidate disease genes.there are currently few therapeutic options for patients with pancreatic cancer, and new insights into the pathogenesis of this lethal disease are urgently needed. toward this end, we performed a comprehensive genetic analysis of 24 pancreatic cancers. we first determined the sequences of 23,219 transcripts, representing 20,661 protein-coding genes, in these samples. then, we searched for homozygous deletions and amplifications in the tumor {dna} by using microarrays containing probes for approximately 10(6) single-nucleotide polymorphisms. we found that pancreatic cancers contain an average of 63 genetic alterations, the majority of which are point mutations. these alterations defined a core set of 12 cellular signaling pathways and processes that were each genetically altered in 67 to 100\\\\% of the tumors. analysis of these tumors\\' transcriptomes with next-generation sequencing-by-synthesis technologies provided independent evidence for the importance of these pathways and processes. our data indicate that genetically altered core pathways and regulatory processes only become evident once the coding regions of the genome are analyzed in depth. dysregulation of these core pathways and processes through mutation can explain the major features of pancreatic tumorigenesis.this work shows that {ephrin-b2} from astrocytes provides a critical niche signal for cell fate determination in adult mouse hippocampus, in part by directing neuronal differentiation of adult neural stem cells through {ephb4}-dependent juxtacrine signaling.drosophila melanogaster is one of the most well studied genetic model organisms; nonetheless, its genome still contains unannotated coding and non-coding genes, transcripts, exons and {rna} editing sites. full discovery and annotation are pre-requisites for understanding how the regulation of transcription, splicing and {rna} editing directs the development of this complex organism. here we used {rna}-seq, tiling microarrays and {cdna} sequencing to explore the transcriptome in 30 distinct developmental stages. we identified 111,195 new elements, including thousands of genes, coding and non-coding transcripts, exons, splicing and editing events, and inferred protein isoforms that previously eluded discovery using established experimental, prediction and conservation-based approaches. these data substantially expand the number of known transcribed elements in the drosophila genome and provide a high-resolution view of transcriptome dynamics throughout development.motivation: deciphering the complex mechanisms by which regulatory networks control gene expression remains a major challenge. while some studies infer regulation from dependencies between the expression levels of putative regulators and their targets, others focus on measured physical interactions.chromatin modifiers and histone modifications are components of a chromatin-signaling network involved in transcription and its regulation. the interactions between chromatin modifiers and histone modifications are often unknown, are based on the analysis of few genes or are studied in vitro. here, we apply computational methods to recover interactions between chromatin modifiers and histone modifications from genome-wide {chip}-seq data. these interactions provide a high-confidence backbone of the chromatin-signaling network. many recovered interactions have literature support; others provide hypotheses about yet unknown interactions. we experimentally verified two of these predicted interactions, leading to a link between {h4k20me1} and members of the polycomb repressive complexes 1 and 2. our results suggest that our computationally derived interactions are likely to lead to novel biological insights required to establish the connectivity of the chromatin-signaling network involved in transcription and its regulation.comparing genomic properties of different organisms is of fundamental importance in the study of biological and evolutionary principles. although differences among organisms are often attributed to differential gene expression, genome-wide comparative analysis thus far has been based primarily on genomic sequence information. we present a comparative study of large datasets of expression profiles from six evolutionarily distant organisms: s. cerevisiae, c. elegans, e. coli, a. thaliana, d. melanogaster, and h. sapiens. we use genomic sequence information to connect these data and compare global and modular properties of the transcription programs. linking genes whose expression profiles are similar, we find that for all organisms the connectivity distribution follows a power-law, highly connected genes tend to be essential and conserved, and the expression program is highly modular. we reveal the modular structure by decomposing each set of expression data into coexpressed modules. functionally related sets of genes are frequently coexpressed in multiple organisms. yet their relative importance to the transcription program and their regulatory relationships vary among organisms. our results demonstrate the potential of combining sequence and expression data for improving functional gene annotation and expanding our understanding of how gene expression and diversity evolved.the complementarity of gene expression and {protein-dna} interaction data led to several successful models of biological systems. however, recent studies in multiple species raise doubts about the relationship between these two datasets. these studies show that the overwhelming majority of genes bound by a particular transcription factor ({tf}) are not affected when that factor is knocked out. here, we show that this surprising result can be partially explained by considering the broader cellular context in which {tfs} operate. factors whose functions are not backed up by redundant paralogs show a fourfold increase in the agreement between their bound targets and the expression levels of those targets. in addition, we show that incorporating protein interaction networks provides physical explanations for knockout effects. new double knockout experiments support our conclusions. our results highlight the robustness provided by redundant {tfs} and indicate that in the context of diverse cellular systems, binding is still largely functional.histones are frequently decorated with covalent modifications. these histone modifications are thought to be involved in various chromatin-dependent processes including transcription. to elucidate the relationship between histone modifications and transcription, we derived quantitative models to predict the expression level of genes from histone modification levels. we found that histone modification levels and gene expression are very well correlated. moreover, we show that only a small number of histone modifications are necessary to accurately predict gene expression. we show that different sets of histone modifications are necessary to predict gene expression driven by high {cpg} content promoters ({hcps}) or low {cpg} content promoters ({lcps}). quantitative models involving {h3k4me3} and {h3k79me1} are the most predictive of the expression levels in {lcps}, whereas {hcps} require {h3k27ac} and {h4k20me1}. finally, we show that the connections between histone modifications and gene expression seem to be general, as we were able to predict gene expression levels of one cell type using a model trained on another one.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismstranscriptional network evolution is central to the development of complex biological systems. networks can evolve through variation of master regulators and/or by changes in regulation of genes within networks. to gain insight into meaningful evolutionary differences in large networks, it is essential to address the functional consequences of sequence differences in response elements ({res}) targeted by transcription factors. using a combination of custom bioinformatics and multispecies alignment of promoter regions, we investigated the functional evolution of {res} in terms of responsiveness to the sequence-specific transcription factor p53, a tumor suppressor and master regulator of stress responses. we identified {res} orthologous to known p53 targets in human and rodent cells or alternatively {res} related to the established p53 consensus. the orthologous {res} were assigned p53 transactivation capabilities based on rules determined from model systems, and a functional heat map was developed to visually summarize conservation of sequence and relative level of responsiveness to p53 for 47 {res} in 14 species. individual {res} exhibited marked differences in transactivation potentials and widespread evolutionary turnover. functional differences were often not predicted from consensus sequence evaluations. of the established human p53 {res} analyzed, 91\\\\% had sequence conservation in at least one nonprimate species compared with 67.5\\\\% for functional conservation. surprisingly, there was almost no conservation of functional {res} for genes involved in {dna} metabolism or repair between humans and rodents, suggesting important differences in p53 stress responses and cancer development.motif discovery has been one of the most widely studied problems in bioinformatics ever since genomic and protein sequences have been available. in particular, its application to the de novo prediction of putative over-represented transcription factor binding sites in nucleotide sequences has been, and still is, one of the most challenging flavors of the problem. recently, novel experimental techniques like chromatin immunoprecipitation ({chip}) have been introduced, permitting the genome-wide identification of {protein–dna} interactions. {chip}, applied to transcription factors and coupled with genome tiling arrays ({chip} on chip) or next-generation sequencing technologies ({chip}-seq) has opened new avenues in research, as well as posed new challenges to bioinformaticians developing algorithms and methods for motif discovery.recent advances in technology have led to a dramatic increase in the number of available transcription factor {chip}-seq and {chip}-chip data sets. understanding the motif content of these data sets is an important step in understanding the underlying mechanisms of regulation. here we provide a systematic motif analysis for 427 human {chip}-seq data sets using motifs curated from the literature and also discovered de novo using five established motif discovery tools. we use a systematic pipeline for calculating motif enrichment in each data set, providing a principled way for choosing between motif variants found in the literature and for flagging potentially problematic data sets. our analysis confirms the known specificity of 41 of the 56 analyzed factor groups and reveals motifs of potential cofactors. we also use cell type-specific binding to find factors active in specific conditions. the resource we provide is accessible both for browsing a small number of factors and for performing large-scale systematic analyses. we provide motif matrices, instances and enrichments in each of the {encode} data sets. the motifs discovered here have been used in parallel studies to validate the specificity of antibodies, understand cooperativity between data sets and measure the variation of motif binding across individuals and species.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsan international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismssequencing technologies have placed a wide range of genomic analyses within the capabilities of many laboratories. however, sequencing costs often set limits to the amount of sequences that can be generated and, consequently, the biological outcomes that can be achieved from an experimental design. in this review, we discuss the issue of sequencing depth in the design of next-generation sequencing experiments. we review current guidelines and precedents on the issue of coverage, as well as their underlying considerations, for four major study designs, which include de novo genome sequencing, genome resequencing, transcriptome sequencing and genomic location analyses (for example, chromatin immunoprecipitation followed by sequencing ({chip}-seq) and chromosome conformation capture ({3c})).the generation of mathematical models of biological processes, the simulation of these processes under different conditions, and the comparison and integration of multiple data sets are explicit goals of systems biology that require the knowledge of the absolute quantity of the system\\'s components. to date, systematic estimates of cellular protein concentrations have been exceptionally scarce. here, we provide a quantitative description of the proteome of a commonly used human cell line in two functional states, interphase and mitosis. we show that these human cultured cells express at least \\\\~{}10 000 proteins and that the quantified proteins span a concentration range of seven orders of magnitude up to 20 000 000 copies per cell. we discuss how protein abundance is linked to function and evolution.dynamic probabilistic networks are a compact representation of complex stochastic processes. in this paper we examine how to learn the structure of a {dpn} from data. we extend structure scoring rules for standard probabilistic networks to the dynamic case, and show how to search for structure when some of the variables are hidden. finally, we examine two applications where such a technology might be useful: predicting and classifying dynamic behaviors, and learning causal orderings in biological processes. we provide empirical results that demonstrate the applicability of our methods in both domains.the past year has witnessed substantial advances in understanding the genetic basis of many common phenotypes of biomedical importance. these advances have been the result of systematic, well-powered, genome-wide surveys exploring the relationships between common sequence variation and disease predisposition. this approach has revealed over 50 disease-susceptibility loci and has provided insights into the allelic architecture of multifactorial traits. at the same time, much has been learned about the successful prosecution of association studies on such a scale. this review highlights the knowledge gained, defines areas of emerging consensus, and describes the challenges that remain as researchers seek to obtain more complete descriptions of the susceptibility architecture of biomedical traits of interest and to translate the information gathered into improvements in clinical management.it has been suggested that evolutionary changes in gene expression account for most phenotypic differences between species, in particular between humans and apes. what general rules can be described governing expression evolution? we find that a neutral model where negative selection and divergence time are the major factors is a useful null hypothesis for both transcriptome and genome evolution. two tissues that stand out with regard to gene expression are the testes, where positive selection has exerted a substantial influence in both humans and chimpanzees, and the brain, where gene expression has changed less than in other organs but acceleration might have occurred in human ancestors.changes in genetic regulation contribute to adaptations in natural populations and influence susceptibility to human diseases. despite their potential phenotypic importance, the selective pressures acting on regulatory processes in general and gene expression levels in particular are largely unknown. studies in model organisms suggest that the expression levels of most genes evolve under stabilizing selection, although a few are consistent with adaptive evolution. however, it has been proposed that gene expression levels in primates evolve largely in the absence of selective constraints. in this article, we discuss the microarray-based observations that led to these disparate interpretations. we conclude that in both primates and model organisms, stabilizing selection is likely to be the dominant mode of gene expression evolution. an important implication is that mutations affecting gene expression will often be deleterious and might underlie many human diseases.motivation: microarray-based gene expression data have been generated widely to study different biological processes and systems. gene co-expression networks are often used to extract information about groups of genes that are \\'functionally\\' related or co-regulated. however, the structural properties of such co-expression networks have not been rigorously studied and fully compared with known biological networks. in this article, we aim at investigating the structural properties of co-expression networks inferred for the species saccharomyces cerevisiae and comparing them with the topological properties of the known, well-established transcriptional network, {mips} physical network and protein–protein interaction ({ppi}) network of {yeast.results}: these topological comparisons indicate that co-expression networks are not distinctly related with either the {ppi} or the {mips} physical interaction networks, showing important structural differences between them. when focusing on a more literal comparison, vertex by vertex and edge by edge, the conclusion is the same: the fact that two genes exhibit a high gene expression correlation degree does not seem to obviously correlate with the existence of a physical binding between the proteins produced by these genes or the existence of a {mips} physical interaction between the genes. the comparison of the yeast regulatory network with inferred yeast co-expression networks would suggest, however, that they could somehow be {related.conclusions}: we conclude that the gene expression-based co-expression networks reflect more on the gene regulatory networks but less on the {ppi} or {mips} physical interaction {networks.contact}: {hongzhe@mail.med.upenn.edusupplementary} information: supplementary data are available at bioinformatics online.the application of established drug compounds to new therapeutic indications, known as drug repositioning, offers several advantages over traditional drug development, including reduced development costs and shorter paths to approval. recent approaches to drug repositioning use high-throughput experimental approaches to assess a compound\\'s potential therapeutic qualities. here, we present a systematic computational approach to predict novel therapeutic indications on the basis of comprehensive testing of molecular signatures in drug-disease pairs. we integrated gene expression measurements from 100 diseases and gene expression measurements on 164 drug compounds, yielding predicted therapeutic potentials for these drugs. we recovered many known drug and disease relationships using computationally derived therapeutic potentials and also predict many new indications for these 164 drugs. we experimentally validated a prediction for the antiulcer drug cimetidine as a candidate therapeutic in the treatment of lung adenocarcinoma, and demonstrate its efficacy both in vitro and in vivo using mouse xenograft models. this computational method provides a systematic approach for repositioning established drugs to treat a wide range of human diseases.10.1101/gad.1484207 a large portion of the eukaryotic genome is transcribed as noncoding {rnas} ({ncrnas}). while once thought of primarily as  ” junk,” recent studies indicate that a large number of these {rnas} play central roles in regulating gene expression at multiple levels. the increasing diversity of {ncrnas} identified in the eukaryotic genome suggests a critical nexus between the regulatory potential of {ncrnas} and the complexity of genome organization. we provide an overview of recent advances in the identification and function of eukaryotic {ncrnas} and the roles played by these {rnas} in chromatin organization, gene expression, and disease etiology.messenger {rna} decay measurements are typically performed on a population of cells. however, this approach cannot reveal sufficient complexity to provide\\xa0information on mechanisms that may regulate {mrna}\\xa0degradation, possibly on short timescales. to address this deficiency, we measured cell cycle-regulated decay in single yeast cells using single-molecule {fish}. we found that two genes responsible for mitotic progression, {swi5} and {clb2}, exhibit\\xa0a mitosis-dependent {mrna} stability switch. their transcripts are stable until mitosis, when a precipitous decay eliminates the {mrna} complement, preventing carryover into the next cycle. remarkably, the specificity and timing of decay is entirely regulated by their promoter, independent of specific cis {mrna} sequences. the mitotic exit network protein dbf2p binds to {swi5} and {clb2} {mrnas} cotranscriptionally and regulates their decay. this work reveals the promoter-dependent control of {mrna} stability,\\xa0a regulatory mechanism that could be employed by a variety of {mrnas} and organisms. º {mrna} turnover was quantified in unperturbed yeast cells using single-molecule {fish} º {swi5} and {clb2} exhibit a mitosis-dependent switch in their {mrna} stability º their decay is promoter regulated and independent of cis {mrna} sequences º dbf2p binds to {swi5} and {clb2} cotranscriptionally and regulates their stability control of promoter-mediated {mrna} decay extends beyond transcription factors to include kinases previously associated with the cell cycle.recent technologies have made it cost-effective to collect diverse types of genome-wide data. computational methods are needed to combine these data to create a comprehensive view of a given disease or a biological process. similarity network fusion ({snf}) solves this problem by constructing networks of samples (e.g., patients) for each available data type and then efficiently fusing these into one network that represents the full spectrum of underlying data. for example, to create a comprehensive view of a disease given a cohort of patients, {snf} computes and fuses patient similarity networks obtained from each of their data types separately, taking advantage of the complementarity in the data. we used {snf} to combine {mrna} expression, {dna} methylation and {microrna} ({mirna}) expression data for five cancer data sets. {snf} substantially outperforms single data type analysis and established integrative approaches when identifying cancer subtypes and is effective for predicting survival.high-throughput yeast two-hybrid screening is used to generate the largest c. elegans interactome resource available thus far. using an empirical quality control framework presented in venkatesan et al., also online, the data set is evaluated for quality and is used to estimate the total size of the worm interactome.limma is an {r/bioconductor} software package that provides an integrated solution for analysing data from gene expression experiments. it contains rich features for handling complex experimental designs and for information borrowing to overcome the problem of small sample sizes. over the past decade, limma has been a popular choice for gene discovery through differential expression analyses of microarray and high-throughput {pcr} data. the package contains particularly strong facilities for reading, normalizing and exploring such data. recently, the capabilities of limma have been significantly expanded in two important directions. first, the package can now perform both differential expression and differential splicing analyses of {rna} sequencing ({rna}-seq) data. all the downstream analysis tools previously restricted to microarray data are now available for {rna}-seq as well. these capabilities allow users to analyse both {rna}-seq and microarray data with very similar pipelines. second, the package is now able to go past the traditional gene-wise expression analyses in a variety of ways, analysing expression profiles in terms of co-regulated sets of genes or in terms of higher-order expression signatures. this provides enhanced possibilities for biological interpretation of gene expression differences. this article reviews the philosophy and design of the limma package, summarizing both new and historical features, with an emphasis on recent enhancements and features that have not been previously described. {\\\\copyright} the author(s) 2015. published by oxford university press on behalf of nucleic acids research.recent studies have emphasized the importance of pathway-specific interpretations for understanding the functional relevance of gene alterations in human cancers. although signaling activities are often conceptualized as linear events, in reality, they reflect the activity of complex functional networks assembled from modules that each respond to input signals. to acquire a deeper understanding of this network structure, we developed an approach to deconstruct pathways into modules represented by gene expression signatures. our studies confirm that they represent units of underlying biological activity linked to known biochemical pathway structures. importantly, we show that these signaling modules provide tools to dissect the complexity of oncogenic states that define disease outcomes as well as response to pathway-specific therapeutics. we propose that this model of pathway structure constitutes a framework to study the processes by which information propogates through cellular networks and to elucidate the relationships of fundamental modules to cellular and clinical phenotypes.defining the protein profiles of tissues and organs is critical to understanding the unique characteristics of the various cell types in the human body. in this study, we report on an anatomically comprehensive analysis of 4842 protein profiles in 48 human tissues and 45 human cell lines. a detailed analysis of over 2 million manually annotated, high-resolution, immunohistochemistry-based images showed a high fraction (>65\\\\%) of expressed proteins in most cells and tissues, with very few proteins (<2\\\\%) detected in any single cell type. similarly, confocal microscopy in three human cell lines detected expression of more than 70\\\\% of the analyzed proteins. despite this ubiquitous expression, hierarchical clustering analysis, based on global protein expression patterns, shows that the analyzed cells can be still subdivided into groups according to the current concepts of histology and cellular differentiation. this study suggests that tissue specificity is achieved by precise regulation of protein levels in space and time, and that different tissues in the body acquire their unique characteristics by controlling not which proteins are expressed but how much of each is produced.enhancers are distal cis-regulatory {dna} elements that increase the expression of target genes. various experimental and computational approaches including chromatin signature profiling have been developed to predict enhancers on a genome-wide scale, although each method has its advantages and disadvantages. here we overview an emerging method to identify transcribed enhancers at exceedingly high nucleotide resolution based on enhancer {rna} transcripts captured by cap analysis of gene expression ({cage}) technology. we further argue that disease-causative regulatory mutations at enhancers are increasingly recognized, emphasizing the importance of enhancer identification in functional and clinical genomics including, but not limited to, genome-wide association studies ({gwass}) and cancer genomics studies.epigenetic mechanisms have been proposed to play crucial roles in mammalian development, but their precise functions are only partially understood. to investigate epigenetic regulation of embryonic development, we differentiated human embryonic stem cells into mesendoderm, neural progenitor cells, trophoblast-like cells, and mesenchymal stem cells and systematically characterized {dna} methylation, chromatin modifications, and the transcriptome in each lineage. we found that promoters that are active in early developmental stages tend to be {cg} rich and mainly engage {h3k27me3} upon silencing in nonexpressing lineages. by contrast, promoters for genes expressed preferentially at later stages are often {cg} poor and primarily employ {dna} methylation upon repression. interestingly, the early developmental regulatory genes are often located in large genomic domains that are generally devoid of {dna} methylation in most lineages, which we termed {dna} methylation valleys ({dmvs}). our results suggest that distinct epigenetic mechanisms regulate early and late stages of {es} cell differentiation. copyright {\\\\copyright} 2013 elsevier inc. all rights reserved.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsaccurately modeling the {dna} sequence preferences of transcription factors ({tfs}), and using these models to predict in vivo genomic binding sites for {tfs}, are key pieces in deciphering the regulatory code. these efforts have been frustrated by the limited availability and accuracy of {tf} binding site motifs, usually represented as position-specific scoring matrices ({pssms}), which may match large numbers of sites and produce an unreliable list of target genes. recently, protein binding microarray ({pbm}) experiments have emerged as a new source of high resolution data on in vitro {tf} binding specificities. {pbm} data has been analyzed either by estimating {pssms} or via rank statistics on probe intensities, so that individual sequence patterns are assigned enrichment scores (e-scores). this representation is informative but unwieldy because every {tf} is assigned a list of thousands of scored sequence patterns. meanwhile, high-resolution in vivo {tf} occupancy data from {chip}-seq experiments is also increasingly available. we have developed a flexible discriminative framework for learning {tf} binding preferences from high resolution in vitro and in vivo data. we first trained support vector regression ({svr}) models on {pbm} data to learn the mapping from probe sequences to binding intensities. we used a novel -mer based string kernel called the di-mismatch kernel to represent probe sequence similarities. the {svr} models are more compact than e-scores, more expressive than {pssms}, and can be readily used to scan genomics regions to predict in vivo occupancy. using a large data set of yeast and mouse {tfs}, we found that our {svr} models can better predict probe intensity than the e-score method or {pbm}-derived {pssms}. moreover, by using {svrs} to score yeast, mouse, and human genomic regions, we were better able to predict genomic occupancy as measured by {chip}-chip and {chip}-seq experiments. finally, we found that by training kernel-based models directly on {chip}-seq data, we greatly improved in vivo occupancy prediction, and by comparing a {tf}\\'s in vitro and in vivo models, we could identify cofactors and disambiguate direct and indirect binding. transcription factors ({tfs}) are proteins that bind sites in the non-coding {dna} and regulate the expression of targeted genes. being able to predict the genome-wide binding locations of {tfs} is an important step in deciphering gene regulatory networks. historically, there was very limited experimental data on the {dna}-binding preferences of most {tfs}. computational biologists used known sites to estimate simple binding site motifs, called position-specific scoring matrices, and scan the genome for additional potential binding locations, but this approach often led to many false positive predictions. here we introduce a machine learning approach to leverage new high resolution data on the binding preferences of {tfs}, namely, protein binding microarray ({pbm}) experiments which measure the in vitro binding affinities of {tfs} with respect to an array of double-stranded {dna} probes, and chromatin immunoprecipitation experiments followed by next generation sequencing ({chip}-seq) which measure in vivo genome-wide binding of {tfs} in a given cell type. we show that by training statistical models on high resolution {pbm} and {chip}-seq data, we can more accurately represent the subtle {dna} binding preferences of {tfs} and predict their genome-wide binding locations. these results will enable advances in the computational analysis of transcriptional regulation in mammalian genomes.despite the availability of dozens of animal genome sequences, two key questions remain unanswered: first, what fraction of any species\\' genome confers biological function, and second, are apparent differences in organismal complexity reflected in an objective measure of genomic complexity? here, we address both questions by applying, across the mammalian phylogeny, an evolutionary model that estimates the amount of functional {dna} that is shared between two species\\' genomes. our main findings are, first, that as the divergence between mammalian species increases, the predicted amount of pairwise shared functional sequence drops off dramatically. we show by simulations that this is not an artifact of the method, but rather indicates that functional (and mostly noncoding) sequence is turning over at a very high rate. we estimate that between 200 and 300 mb (∼6.5\\\\%–10\\\\%) of the human genome is under functional constraint, which includes five to eight times as many constrained noncoding bases than bases that code for protein. in contrast, in d. melanogaster we estimate only 56–66 mb to be constrained, implying a ratio of noncoding to coding constrained bases of about 2. this suggests that, rather than genome size or protein-coding gene complement, it is the number of functional bases that might best mirror our na\\\\\"{i}ve preconceptions of organismal complexity.the increasing availability of polymorphism data has allowed more gene association studies to be carried out and the number of published genetic association studies is growing rapidly. studies done secondarily to successful linkage studies over the last decade have also fueled the increase in published association studies. although there are single-nucleotide polymorphism and human variation databases1, 2, there is currently no public repository for genetic association data. it is difficult to query association data in a systematic manner or to integrate association data with other molecular databases. {omim3}, the main repository of genetic information for mendelian disorders, is largely text based and is of a historical narrative design, making it difficult to compare large sets of molecular data. moreover, {omim} archives mature, high-quality data of high significance, the standard in rare mendelian disorders. although this data is useful, {omim} does not routinely collect findings of lower significance or negative findings. the study of nonmendelian, common complex disorders is often a struggle to find disease relevance with lower significance values, and often conflicting evidence. negative data are often not reported or are marginalized into obscure and less accessible scientific journals, resulting in a publication bias favoring positive genetic associations4. here, we describe the development of a genetic association database ({gad}; http://geneticassociationdb.nih.gov) that aims to collect, standardize and archive genetic association study data and to make it easily accessible to the scientific community.accurate functional annotation of regulatory elements is essential for understanding global gene regulation. here, we report a genome-wide map of 827,000 transcription factor binding sites in human lymphoblastoid cell lines, which is comprised of sites corresponding to 239 position weight matrices of known transcription factor binding motifs, and 49 novel sequence motifs. to generate this map, we developed a probabilistic framework that integrates cell- or tissue-specific experimental data such as histone modifications and {dnase} i cleavage patterns with genomic information such as gene annotation and evolutionary conservation. comparison to empirical {chip}-seq data suggests that our method is highly accurate yet has the advantage of targeting many factors in a single assay. we anticipate that this approach will be a valuable tool for genome-wide studies of gene regulation in a wide variety of cell types or tissues under diverse conditions.we define a  ” threaded blockset,” which is a novel generalization of the classic notion of a multiple alignment. a new computer program called {tba} (for  ” threaded blockset aligner”) builds a threaded blockset under the assumption that all matching segments occur in the same order and orientation in the given sequences; inversions and duplications are not addressed. {tba} is designed to be appropriate for aligning many, but by no means all, megabase-sized regions of multiple mammalian genomes. the output of {tba} can be projected onto any genome chosen as a reference, thus guaranteeing that different projections present consistent predictions of which genomic positions are orthologous. this capability is illustrated using a new visualization tool to view {tba}-generated alignments of vertebrate hox clusters from both the mammalian and fish perspectives. experimental evaluation of alignment quality, using a program that simulates evolutionary change in genomic sequences, indicates that {tba} is more accurate than earlier programs. to perform the dynamic-programming alignment step, {tba} runs a stand-alone program called {multiz}, which can be used to align highly rearranged or incompletely sequenced genomes. we describe our use of {multiz} to produce the whole-genome multiple alignments at the santa cruz genome browser.models of mammalian regulatory networks controlling gene expression have been inferred from genomic data but have largely not been validated. we present an unbiased strategy to systematically perturb candidate regulators and monitor cellular transcriptional responses. we applied this approach to derive regulatory networks that control the transcriptional response of mouse primary dendritic cells to pathogens. our approach revealed the regulatory functions of 125 transcription factors, chromatin modifiers, and {rna} binding proteins, which enabled the construction of a network model consisting of 24 core regulators and 76 fine-tuners that help to explain how pathogen-sensing pathways achieve specificity. this study establishes a broadly applicable, comprehensive, and unbiased approach to reveal the wiring and functions of a regulatory network controlling a major transcriptional response in primary mammalian cells.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organisms{background}:large-scale quantitative analysis of transcriptional co-expression has been used to dissect regulatory networks and to predict the functions of new genes discovered by genome sequencing in model organisms such as yeast. although the idea that tissue-specific expression is indicative of gene function in mammals is widely accepted, it has not been objectively tested nor compared with the related but distinct strategy of correlating gene co-expression as a means to predict gene {function.results}:we generated microarray expression data for nearly 40,000 known and predicted {mrnas} in 55 mouse tissues, using custom-built oligonucleotide arrays. we show that quantitative transcriptional co-expression is a powerful predictor of gene function. hundreds of functional categories, as defined by gene ontology \\'biological processes\\', are associated with characteristic expression patterns across all tissues, including categories that bear no overt relationship to the tissue of origin. in contrast, simple tissue-specific restriction of expression is a poor predictor of which genes are in which functional categories. as an example, the highly conserved mouse gene {pwp1} is widely expressed across different tissues but is co-expressed with many {rna}-processing genes; we show that the uncharacterized yeast homolog of {pwp1} is required for {rrna} {biogenesis.conclusions}:we conclude that \\'functional genomics\\' strategies based on quantitative transcriptional co-expression will be as fruitful in mammals as they have been in simpler organisms, and that transcriptional control of mammalian physiology is more modular than is generally appreciated. our data and analyses provide a public resource for mammalian functional genomics.network analysis has been applied widely, providing a unifying language to describe disparate systems ranging from social interactions to power grids. it has recently been used in molecular biology, but so far the resulting networks have only been analysed statically1, 2, 3, 4, 5, 6, 7, 8. here we present the dynamics of a biological network on a genomic scale, by integrating transcriptional regulatory information9, 10, 11 and gene-expression data12, 13, 14, 15, 16 for multiple conditions in saccharomyces cerevisiae. we develop an approach for the statistical analysis of network dynamics, called {sandy}, combining well-known global topological measures, local motifs and newly derived statistics. we uncover large changes in underlying network architecture that are unexpected given current viewpoints and random simulations. in response to diverse stimuli, transcription factors alter their interactions to varying degrees, thereby rewiring the network. a few transcription factors serve as permanent hubs, but most act transiently only during certain conditions. by studying sub-network structures, we show that environmental responses facilitate fast signal propagation (for example, with short regulatory cascades), whereas the cell cycle and sporulation direct temporal progression through multiple stages (for example, with highly inter-connected transcription factors). indeed, to drive the latter processes forward, phase-specific transcription factors inter-regulate serially, and ubiquitously active transcription factors layer above them in a two-tiered hierarchy. we anticipate that many of the concepts presented here—particularly the large-scale topological changes and hub transience—will apply to other biological networks, including complex sub-systems in higher eukaryotes.a key goal of biomedical research is to elucidate the complex network of gene interactions underlying complex traits such as common human diseases. here we detail a multistep procedure for identifying potential key drivers of complex traits that integrates {dna}-variation and gene-expression data with other complex trait data in segregating mouse populations. ordering gene expression traits relative to one another and relative to other complex traits is achieved by systematically testing whether variations in {dna} that lead to variations in relative transcript abundances statistically support an independent, causative or reactive function relative to the complex traits under consideration. we show that this approach can predict transcriptional responses to single gene-perturbation experiments using gene-expression data in the context of a segregating mouse population. we also demonstrate the utility of this approach by identifying and experimentally validating the involvement of three new genes in susceptibility to obesity.10.1073/pnas.0904863106 next-generation sequencing has greatly increased the scope and the resolution of transcriptional regulation study. {rna} sequencing ({rna}-seq) and {chip}-seq experiments are now generating comprehensive data on transcript abundance and on {regulator\\\\^{a}dna} interactions. we propose an approach for an integrated analysis of these data based on feature extraction of {chip}-seq signals, principal component analysis, and regression-based component selection. compared with traditional methods, our approach not only offers higher power in predicting gene expression from {chip}-seq data but also provides a way to capture cooperation among regulators. in mouse embryonic stem cells ({escs}), we find that a remarkably high proportion of variation in gene expression (65\\\\%) can be explained by the binding signals of 12 transcription factors ({tfs}). two groups of {tfs} are identified. whereas the first group (, , , and ) act as activators in general, the second group (, , , , , , and ) may serve as either activator or repressor depending on the target. the two groups of {tfs} cooperate tightly to activate genes that are differentially up-regulated in {escs}. in the absence of binding by the first group, the binding of the second group is associated with genes that are repressed in {escs} and derepressed upon early differentiation.binding of transcription factors ({tfs}) to regulatory sequences is a pivotal step in the control of gene expression. despite many advances in the characterization of sequence motifs recognized by {tfs}, our ability to quantitatively predict {tf} binding to different regulatory sequences is still limited. here, we present a novel experimental assay termed {bundle}-seq that provides quantitative measurements of {tf} binding to thousands of fully designed sequences of 200 bp in length within a single experiment. applying this binding assay to two yeast {tfs}, we demonstrate that sequences outside the core {tf} binding site profoundly affect {tf} binding. we show that {tf}-specific models based on the sequence or {dna} shape of the regions flanking the core binding site are highly predictive of the measured differential {tf} binding. we further characterize the dependence of {tf} binding, accounting for measurements of single and co-occurring binding events, on the number and location of binding sites and on the {tf} concentration. finally, by coupling our in vitro {tf} binding measurements, and another application of our method probing nucleosome formation, to in vivo expression measurements carried out with the same template sequences serving as promoters, we offer insights into mechanisms that may determine the different expression outcomes observed. our assay thus paves the way to a more comprehensive understanding of {tf} binding to regulatory sequences and allows the characterization of {tf} binding determinants within and outside of core binding sites. {\\\\copyright} 2015 levo et al.; published by cold spring harbor laboratory press.recruitment of cofactors to specific {dna} sites is integral for specificity in gene regulation. as a model system, we examined how targeting and transcriptional control of the sulfur metabolism genes in saccharomyces cerevisiae is governed by recruitment of the transcriptional co-activator met4. we developed genome-scale approaches to measure transcription factor ({tf}) {dna}-binding affinities and cofactor recruitment to >1300 genomic binding site sequences. we report that genes responding to the {tf} cbf1 and cofactor met28 contain a novel \\'recruitment motif\\' ({ryaat}), adjacent to cbf1 binding sites, which enhances the binding of a {met4-met28}-cbf1 regulatory complex, and that abrogation of this motif significantly reduces gene induction under low-sulfur conditions. furthermore, we show that correct recognition of this composite motif requires both {non-dna}-binding cofactors met4 and met28. finally, we demonstrate that the presence of an {ryaat} motif next to a cbf1 site, rather than cbf1 binding affinity, specifies cbf1-dependent sulfur metabolism genes. our results highlight the need to examine {tf}/cofactor complexes, as novel specificity can result from cofactors that lack intrinsic {dna}-binding specificity.motivation: blood cell development is thought to be controlled by a circuit of transcription factors ({tfs}) and chromatin modifications that determine the cell fate through activating cell type-specific expression programs. to shed light on the interplay between histone marks and {tfs} during blood cell development, we model gene expression from regulatory signals by means of combinations of sparse linear regression models.cells employ multiple levels of regulation, including transcriptional and translational regulation, that drive core biological processes and enable cells to respond to genetic and environmental changes. small-molecule metabolites are one category of critical cellular intermediates that can influence as well as be a target of cellular regulations. because metabolites represent the direct output of protein-mediated cellular processes, endogenous metabolite concentrations can closely reflect cellular physiological states, especially when integrated with other molecular-profiling data. here we develop and apply a network reconstruction approach that simultaneously integrates six different types of data: endogenous metabolite concentration, {rna} expression, {dna} variation, {dna}–protein binding, protein–metabolite interaction, and protein–protein interaction data, to construct probabilistic causal networks that elucidate the complexity of cell regulation in a segregating yeast population. because many of the metabolites are found to be under strong genetic control, we were able to employ a causal regulator detection algorithm to identify causal regulators of the resulting network that elucidated the mechanisms by which variations in their sequence affect gene expression and metabolite concentrations. we examined all four expression quantitative trait loci ({eqtl}) hot spots with colocalized metabolite {qtls}, two of which recapitulated known biological processes, while the other two elucidated novel putative biological mechanisms for the {eqtl} hot spots. it is now possible to score variations in {dna} across whole genomes, {rna} levels and alternative isoforms, metabolite levels, protein levels and protein state information, protein–protein interactions, and {protein–dna} interactions, in a comprehensive fashion in populations of individuals. interactions among these molecular entities define the complex web of biological processes that give rise to all higher order phenotypes, including disease. the development of analytical approaches that simultaneously integrate different dimensions of data is essential if we are to extract the meaning from large-scale data to elucidate the complexity of living systems. here, we use a novel bayesian network reconstruction algorithm that simultaneously integrates {dna} variation, {rna} levels, metabolite levels, protein–protein interaction data, {protein–dna} binding data, and protein–small-molecule interaction data to construct molecular networks in yeast. we demonstrate that these networks can be used to infer causal relationships among genes, enabling the identification of novel genes that modulate cellular regulation. we show that our network predictions either recapitulate known biology or can be prospectively validated, demonstrating a high degree of accuracy in the predicted network.motivation: over the last decade, numerous methods have been developed for inference of regulatory networks from gene expression data. however, accurate and systematic evaluation of these methods is hampered by the difficulty of constructing adequate benchmarks and the lack of tools for a differentiated analysis of network predictions on such benchmarks.establishing a functional network is invaluable to our understanding of gene function, pathways, and systems-level properties of an organism and can be a powerful resource in directing targeted experiments. in this study, we present a functional network for the laboratory mouse based on a bayesian integration of diverse genetic and functional genomic data. the resulting network includes probabilistic functional linkages among 20,581 protein-coding genes. we show that this network can accurately predict novel functional assignments and network components and present experimental evidence for predictions related to nanog homeobox (nanog), a critical gene in mouse embryonic stem cell pluripotency. an analysis of the global topology of the mouse functional network reveals multiple biologically relevant systems-level features of the mouse proteome. specifically, we identify the clustering coefficient as a critical characteristic of central modulators that affect diverse pathways as well as genes associated with different phenotype traits and diseases. in addition, a cross-species comparison of functional interactomes on a genomic scale revealed distinct functional characteristics of conserved neighborhoods as compared to subnetworks specific to higher organisms. thus, our global functional network for the laboratory mouse provides the community with a key resource for discovering protein functions and novel pathway components as well as a tool for exploring systems-level topological and evolutionary features of cellular interactomes. to facilitate exploration of this network by the biomedical research community, we illustrate its application in function and disease gene discovery through an interactive, web-based, publicly available interface at {http://mousenet}.princeton.edu. functionally related proteins interact in diverse ways to carry out biological processes, and each protein often participates in multiple pathways. proteins are therefore organized into a complex network through which different functions of the cell are carried out. an accurate description of such a network is invaluable to our understanding of both the system-level features of a cell and those of an individual biological process. in this study, we used a probabilistic model to combine information from diverse genome-scale studies as well as individual investigations to generate a global functional network for mouse. our analysis of the global topology of this network reveals biologically relevant systems-level characteristics of the mouse proteome, including conservation of functional neighborhoods and network features characteristic of known disease genes and key transcriptional regulators. we have made this network publicly available for search and dynamic exploration by researchers in the community. our web interface enables users to easily generate hypotheses regarding potential functional roles of uncharacterized proteins, investigate possible links between their proteins of interest and disease, and identify new players in specific biological processes.cellular signal transduction generally involves cascades of post-translational protein modifications that rapidly catalyze changes in {protein-dna} interactions and gene expression. high-throughput measurements are improving our ability to study each of these stages individually, but do not capture the connections between them. here we present an approach for building a network of physical links among these data that can be used to prioritize targets for pharmacological intervention. our method recovers the critical missing links between proteomic and transcriptional data by relating changes in chromatin accessibility to changes in expression and then uses these links to connect proteomic and transcriptome data. we applied our approach to integrate epigenomic, phosphoproteomic and transcriptome changes induced by the variant {iii} mutation of the epidermal growth factor receptor ({egfrviii}) in a cell line model of glioblastoma multiforme ({gbm}). to test the relevance of the network, we used small molecules to target highly connected nodes implicated by the network model that were not detected by the experimental data in isolation and we found that a large fraction of these agents alter cell viability. among these are two compounds, {icg}-001, targeting {creb} binding protein ({crebbp}), and {pkf118}–310, targeting β-catenin ({ctnnb1}), which have not been tested previously for effectiveness against {gbm}. at the level of transcriptional regulation, we used chromatin immunoprecipitation sequencing ({chip}-seq) to experimentally determine the genome-wide binding locations of p300, a transcriptional co-regulator highly connected in the network. analysis of p300 target genes suggested its role in tumorigenesis. we propose that this general method, in which experimental measurements are used as constraints for building regulatory networks from the interactome while taking into account noise and missing data, should be applicable to a wide range of high-throughput datasets. the ways in which cells respond to changes in their environment are controlled by networks of physical links among the proteins and genes. the initial signal of a change in conditions rapidly passes through these networks from the cytoplasm to the nucleus, where it can lead to long-term alterations in cellular behavior by controlling the expression of genes. these cascades of signaling events underlie many normal biological processes. as a result, being able to map out how these networks change in disease can provide critical insights for new approaches to treatment. we present a computational method for reconstructing these networks by finding links between the rapid short-term changes in proteins and the longer-term changes in gene regulation. this method brings together systematic measurements of protein signaling, genome organization and transcription in the context of protein-protein and {protein-dna} interactions. when used to analyze datasets from an oncogene expressing cell line model of human glioblastoma, our approach identifies key nodes that affect cell survival and functional transcriptional regulators.{background}:identifying large gene regulatory networks is an important task, while the acquisition of data through perturbation experiments (e.g., gene switches, {rnai}, heterozygotes) is expensive. it is thus desirable to use an identification method that effectively incorporates available prior knowledge - such as sparse connectivity - and that allows to design experiments such that maximal information is gained from each {one.results}:our main contributions are twofold: a method for consistent inference of network structure is provided, incorporating prior knowledge about sparse connectivity. the algorithm is time efficient and robust to violations of model assumptions. moreover, we show how to use it for optimal experimental design, reducing the number of required experiments substantially. we employ sparse linear models, and show how to perform full bayesian inference for these. we not only estimate a single maximum likelihood network, but compute a posterior distribution over networks, using a novel variant of the expectation propagation method. the representation of uncertainty enables us to do effective experimental design in a standard statistical setting: experiments are selected such that the experiments are maximally {informative.conclusion}:few methods have addressed the design issue so far. compared to the most well-known one, our method is more transparent, and is shown to perform qualitatively superior. in the former, hard and unrealistic constraints have to be placed on the network structure for mere computational tractability, while such are not required in our method. we demonstrate reconstruction and optimal experimental design capabilities on tasks generated from realistic non-linear network {simulators.the} methods described in the paper are available as a matlab package athttp://www.kyb.tuebingen.mpg.de/sparselinearmodel webcite.ascertaining the impact of uncharacterized perturbations on the cell is a fundamental problem in biology. here, we describe how a single assay can be used to monitor hundreds of different cellular functions simultaneously. we constructed a reference database or \"compendium\" of expression profiles corresponding to 300 diverse mutations and chemical treatments in s. cerevisiae, and we show that the cellular pathways affected can be determined by pattern matching, even among very subtle profiles. the utility of this approach is validated by examining profiles caused by deletions of uncharacterized genes: we identify and experimentally confirm that eight uncharacterized open reading frames encode proteins required for sterol metabolism, cell wall function, mitochondrial respiration, or protein synthesis. we also show that the compendium can be used to characterize pharmacological perturbations by identifying a novel target of the commonly used drug dyclonine.we have determined how most of the transcriptional regulators encoded in the eukaryote saccharomyces cerevisiaeassociate with genes across the genome in living cells. just as maps of metabolic networks describe the potential pathways that may be used by a cell to accomplish metabolic processes, this network of regulator-gene interactions describes potential pathways yeast cells can use to regulate global gene expression programs. we use this information to identify network motifs, the simplest units of network architecture, and demonstrate that an automated process can use motifs to assemble a transcriptional regulatory network structure. our results reveal that eukaryotic cellular functions are highly connected through networks of transcriptional regulators that regulate other transcriptional regulators.{background}: cell-fate specification and tissue differentiation during development are largely achieved by the regulation of gene transcription. {results}: as a first step to creating a comprehensive atlas of gene-expression patterns during drosophila embryogenesis, we examined 2,179 genes by in situ hybridization to fixed drosophila embryos. of the genes assayed, 63.7\\\\% displayed dynamic expression patterns that were documented with 25,690 digital photomicrographs of individual embryos. the photomicrographs were annotated using controlled vocabularies for anatomical structures that are organized into a developmental hierarchy. we also generated a detailed time course of gene expression during embryogenesis using microarrays to provide an independent corroboration of the in situ hybridization results. all image, annotation and microarray data are stored in publicly available database. we found that the {rna} transcripts of about 1\\\\% of genes show clear subcellular localization. nearly all the annotated expression patterns are distinct. we present an approach for organizing the data by hierarchical clustering of annotation terms that allows us to group tissues that express similar sets of genes as well as genes displaying similar expression patterns. {conclusions}: analyzing gene-expression patterns by in situ hybridization to whole-mount embryos provides an extremely rich dataset that can be used to identify genes involved in developmental processes that have been missed by traditional genetic analysis. systematic analysis of rigorously annotated patterns of gene expression will complement and extend the types of analyses carried out using expression microarrays.in this review we give an overview of computational and statistical methods to reconstruct cellular networks. although this area of research is vast and fast developing, we show that most currently used methods can be organized by a few key concepts. the first part of the review deals with conditional independence models including gaussian graphical models and bayesian networks. the second part discusses probabilistic and graph-based methods for data from experimental interventions and perturbations.motivation: discovering patterns in networks of protein–protein interactions ({ppis}) is a central problem in systems biology. alignments between these networks aid functional understanding as they uncover important information, such as evolutionary conserved pathways, protein complexes and functional orthologs. however, the complexity of the multiple network alignment problem grows exponentially with the number of networks being aligned and designing a multiple network aligner that is both scalable and that produces biologically relevant alignments is a challenging task that has not been fully addressed. the objective of multiple network alignment is to create clusters of nodes that are evolutionarily and functionally conserved across all networks. unfortunately, the alignment methods proposed thus far do not meet this objective as they are guided by pairwise scores that do not utilize the entire functional and evolutionary information across all networks.monocytes circulate in the bloodstream for up to 3 to 5 days. concomitantly, immunological imprinting of either tolerance (immunosuppression) or trained immunity (innate immune memory) determines the functional fate of monocytes and monocyte-derived macrophages, as observed after infection or vaccination.motivation: inferring the genetic interaction mechanism using bayesian networks has recently drawn increasing attention due to its well-established theoretical foundation and statistical robustness. however, the relative insufficiency of experiments with respect to the number of genes leads to many false positive {inferences.results}: we propose a novel method to infer genetic networks by alleviating the shortage of available {mrna} expression data with prior knowledge. we call the proposed method \\'modularized network learning\\' ({monet}). firstly, the proposed method divides a whole gene set to overlapped modules considering biological annotations and expression data together. secondly, it infers a bayesian network for each module, and integrates the learned subnetworks to a global network. an algorithm that measures a similarity between genes based on hierarchy, specificity and multiplicity of biological annotations is presented. the proposed method draws a global picture of inter-module relationships as well as a detailed look of intra-module interactions. we applied the proposed method to analyze saccharomyces cerevisiae stress data, and found several hypotheses to suggest putative functions of unclassified genes. we also compared the proposed method with a whole-set-based approach and two expression-based clustering {approaches.availability}: {java} programs for the {monet} framework are available from the corresponding author upon request. web supplementary data is accessible at {http://biosoft.kaist.ac.kr/\\\\~{}dhlee/monet/index.htmlcontact}: doheon@kaist.ac.krdissecting components of key transcriptional networks is essential for understanding complex developmental processes and phenotypes. genetic studies have highlighted the role of members of the mef2 family of transcription factors as essential regulators in myogenesis from flies to man. to understand how these transcription factors control diverse processes in muscle development, we have combined chromatin immunoprecipitation analysis with gene expression profiling to obtain a temporal map of mef2 activity during drosophila embryonic development. this global approach revealed three temporal patterns of mef2 enhancer binding, providing a glimpse of dynamic enhancer use within the context of a developing embryo. our results provide mechanistic insight into the regulation of mef2\\'s activity at the level of {dna} binding and suggest cooperativity with the {bhlh} protein twist. the number and diversity of new direct target genes indicates a much broader role for mef2, at all stages of myogenesis, than previously anticipated.motivation: increasing quantity and quality of data in transcriptomics and interactomics create the need for integrative approaches to network analysis. here, we present a comprehensive r-package for the analysis of biological networks including an exact and a heuristic approach to identify functional modules.the advancements in \"omics\" (proteomics, genomics, lipidomics, and metabolomics) technologies have yielded large inventories of genes, transcripts, proteins, and metabolites. the challenge is to find out how these entities work together to regulate the processes by which cells respond to external and internal signals. mathematical and computational modeling of signaling networks has a key role in this task, and network analysis provides insights into biological systems and has applications for medicine. here, we review experimental and theoretical progress and future challenges toward this goal. we focus on how networks are reconstructed from data, how these networks are structured to control the flow of biological information, and how the design features of the networks specify biological decisions.motivation: reconstructing the topology of gene regulatory networks ({grns}) from time series of gene expression data remains an important open problem in computational systems biology. existing {grn} inference algorithms face one of two limitations: model-free methods are scalable but suffer from a lack of interpretability and cannot in general be used for out of sample predictions. on the other hand, model-based methods focus on identifying a dynamical model of the system. these are clearly interpretable and can be used for predictions; however, they rely on strong assumptions and are typically very demanding {computationally.results}: here, we propose a new hybrid approach for {grn} inference, called jump3, exploiting time series of expression data. jump3 is based on a formal on/off model of gene expression but uses a non-parametric procedure based on decision trees (called \\'jump trees\\') to reconstruct the {grn} topology, allowing the inference of networks of hundreds of genes. we show the good performance of jump3 on in silico and synthetic networks and applied the approach to identify regulatory interactions activated in the presence of interferon {gamma.availability} and implementation: our {matlab} implementation of jump3 is available at {http://homepages.inf.ed.ac.uk/vhuynht/software.html.contact}: vhuynht@inf.ed.ac.uk or {g.sanguinetti}@{ed.ac.uksupplementary} information: supplementary data are available at bioinformatics online.many genomic techniques have been developed to study gene essentiality genome-wide, such as {crispr} and {shrna} screens. our analyses of public {crispr} screens suggest protein interaction networks, when integrated with gene expression or histone marks, are highly predictive of gene essentiality. meanwhile, the quality of {crispr} and {shrna} screen results can be significantly enhanced through network neighbor information. we also found network neighbor information to be very informative on prioritizing {chip}-seq target genes and survival indicator genes from tumor profiling. thus, our study provides a general method for gene essentiality analysis in functional genomic experiments (http://nest.dfci.harvard.edu).gene duplication with subsequent interaction divergence is one of the primary driving forces in the evolution of genetic systems. yet little is known about the precise mechanisms and the role of duplication divergence in the evolution of protein networks from the prokaryote and eukaryote domains. we developed a novel, model-based approach for bayesian inference on biological network data that centres on approximate bayesian computation, or likelihood-free inference. instead of computing the intractable likelihood of the protein network topology, our method summarizes key features of the network and, based on these, uses a {mcmc} algorithm to approximate the posterior distribution of the model parameters. this allowed us to reliably fit a flexible mixture model that captures hallmarks of evolution by gene duplication and subfunctionalization to protein interaction network data of helicobacter pylori and plasmodium falciparum. the 80\\\\% credible intervals for the duplication-divergence component are [0.64, 0.98] for h. pylori and [0.87, 0.99] for p. falciparum. the remaining parameter estimates are not inconsistent with sequence data. an extensive sensitivity analysis showed that incompleteness of {pin} data does not largely affect the analysis of models of protein network evolution, and that the degree sequence alone barely captures the evolutionary footprints of protein networks relative to other statistics. our likelihood-free inference approach enables a fully bayesian analysis of a complex and highly stochastic system that is otherwise intractable at present. modelling the evolutionary history of {pin} data, it transpires that only the simultaneous analysis of several global aspects of protein networks enables credible and consistent inference to be made from available datasets. our results indicate that gene duplication has played a larger part in the network evolution of the eukaryote than in the prokaryote, and suggests that single gene duplications with immediate divergence alone may explain more than 60\\\\% of biological network data in both domains.{jclust} is a user-friendly application which provides access to a set of widely used clustering and clique finding algorithms. the toolbox allows a range of filtering procedures to be applied and is combined with an advanced implementation of the medusa interactive visualization module. these implemented algorithms are {k-means}, affinity propagation, {bron\\\\^{a}\\x80\\x93kerbosch}, {mulic}, restricted neighborhood search cluster algorithm, markov clustering and spectral clustering, while the supported filtering procedures are haircut, outside\\\\^{a}\\x80\\x93inside, best neighbors and density control operations. the combination of a simple input file format, a set of clustering and filtering algorithms linked together with the visualization tool provides a powerful tool for data analysis and information {extraction.availability}: {http://jclust.embl.de/contact}: pavlopou@embl.de; rschneid@embl.de; {skossida@bioacademy.grsupplementary} information: supplementary data are available at bioinformatics online.organisms are constantly exposed to a wide range of environmental changes, including both short-term changes during their lifetime and longer-term changes across generations. stress-related gene expression programmes, characterized by distinct transcriptional mechanisms and high levels of noise in their expression patterns, need to be balanced with growth-related gene expression programmes. a range of recent studies give fascinating insight into cellular strategies for keeping gene expression in tune with physiological needs dictated by the environment, promoting adaptation to both short- and long-term environmental changes. not only do organisms show great resilience to external challenges, but emerging data suggest that they also exploit these challenges to fuel phenotypic variation and evolutionary innovation.surveys of gene expression reveal extensive variability both within and between a wide range of species. compelling cases have been made for adaptive changes in gene regulation, but the proportion of expression divergence attributable to natural selection remains unclear. distinguishing adaptive changes driven by positive selection from neutral divergence resulting from mutation and genetic drift is critical for understanding the evolution of gene expression. here, we review the various methods that have been used to test for signs of selection in genomic expression data. we also discuss properties of regulatory systems relevant to neutral models of gene expression. despite some potential caveats, published studies provide considerable evidence for adaptive changes in gene expression. future challenges for studies of regulatory evolution will be to quantify the frequency of adaptive changes, identify the genetic basis of expression divergence and associate changes in gene expression with specific organismal phenotypes.we study the evolution of transcription factor-binding sites in prokaryotes, using an empirically grounded model with point mutations and genetic drift. selection acts on the site sequence via its binding affinity to the corresponding transcription factor. calibrating the model with populations of functional binding sites, we verify this form of selection and show that typical sites are under substantial selection pressure for functionality: for {camp} response protein sites in escherichia coli, the product of fitness difference and effective population size takes values {2nδf} of order 10. we apply this model to cross-species comparisons of binding sites in bacteria and obtain a prediction method for binding sites that uses evolutionary information in a quantitative way. at the same time, this method predicts the functional histories of orthologous sites in a phylogeny, evaluating the likelihood for conservation or loss or gain of function during evolution. we have performed, as an example, a cross-species analysis of e. coli, salmonella typhimurium, and yersinia pseudotuberculosis. detailed lists of predicted sites and their functional phylogenies are available.biological networks change dynamically as protein components are synthesized and degraded. understanding the time-dependence and, in a multicellular organism, tissue-dependence of a network leads to insight beyond a view that collapses time-varying interactions into a single static map. conventional algorithms are limited to analyzing evolving networks by reducing them to a series of unrelated snapshots. here we introduce an approach that groups proteins according to shared interaction patterns through a dynamical hierarchical stochastic block model. protein membership in a block is permitted to evolve as interaction patterns shift over time and space, representing the spatial organization of cell types in a multicellular organism. the spatiotemporal evolution of the protein components are inferred from transcript profiles, using arabidopsis root development (5 tissues, 3 temporal stages) as an example. the new model requires essentially no parameter tuning, out-performs existing snapshot-based methods, identifies protein modules recruited to specific cell types and developmental stages, and could have broad application to social networks and other similar dynamic systems.in analyzing and mathematical modeling of complex (bio)chemical reaction networks, formal methods that connect network structure and dynamic behavior are needed because often, quantitative knowledge of the networks is very limited. this applies to many important processes in cell biology. chemical reaction network theory allows for the classification of the potential network behavior-for instance, with respect to the existence of multiple steady states-but is computationally limited to small systems. here, we show that by analyzing subnetworks termed elementary flux modes, the applicability of the theory can be extended to more complex networks. for an example network inspired by cell cycle control in budding yeast, the approach allows for model discrimination, identification of key mechanisms for multistationarity, and robustness analysis. the presented methods will be helpful in modeling and analyzing other complex reaction networks.systems biology relies on data sets in which the same group of proteins is consistently identified and precisely quantified across multiple samples, a requirement that is only partially achieved by current proteomics approaches. selected reaction monitoring ({srm})-also called multiple reaction monitoring-is emerging as a technology that ideally complements the discovery capabilities of shotgun strategies by its unique potential for reliable quantification of analytes of low abundance in complex mixtures. in an {srm} experiment, a predefined precursor ion and one of its fragments are selected by the two mass filters of a triple quadrupole instrument and monitored over time for precise quantification. a series of transitions (precursor/fragment ion pairs) in combination with the retention time of the targeted peptide can constitute a definitive assay. typically, a large number of peptides are quantified during a single {lc}-{ms} experiment. this tutorial explains the application of {srm} for quantitative proteomics, including the selection of proteotypic peptides and the optimization and validation of transitions. furthermore, normalization and various factors affecting sensitivity and accuracy are discussed.the escherichia coli transcription network has an essentially feedforward structure, with abundant feedback at the level of self-regulations. here, we investigate how these properties emerged during evolution. an assessment of the role of gene duplication based on protein domain architecture shows that (i) transcriptional autoregulators have mostly arisen through duplication, whereas (ii) the expected feedback loops stemming from their initial cross-regulation are strongly selected against. this requires a divergent coevolution of the transcription factor {dna}-binding sites and their respective {dna} cis-regulatory regions. moreover, we find that the network tends to grow by expansion of the existing hierarchical layers of computation, rather than by addition of new layers. we also argue that rewiring of regulatory links due to mutation/selection of novel transcription {factor/dna} binding interactions appears not to significantly affect the network global hierarchy, and that horizontally transferred genes are mainly added at the bottom, as new target nodes. these findings highlight the important evolutionary roles of both duplication and selective deletion of cross-talks between autoregulators in the emergence of the hierarchical transcription network of e. coli.one of the important goals of microarray research is the identification of genes whose expression is considerably higher or lower in some tissues than in others. we would like to have ways of identifying such tissue-specific genes. we describe a method, {roku}, which selects tissue-specific patterns from gene expression data for many tissues and thousands of genes. {roku} ranks genes according to their overall tissue specificity using shannon entropy and detects tissues specific to each gene if any exist using an outlier detection method. we evaluated the capacity for the detection of various specific expression patterns using synthetic and real data. we observed that {roku} was superior to a conventional entropy-based method in its ability to rank genes according to overall tissue specificity and to detect genes whose expression pattern are specific only to objective tissues. {roku} is useful for the detection of various tissue-specific expression patterns. the framework is also directly applicable to the selection of diagnostic markers for molecular classification of multiple classes.one of the most important recent findings in cancer genomics is the identification of novel driver mutations which often target genes that regulate genome-wide chromatin and {dna} methylation marks. little is known, however, as to whether these genes exhibit patterns of epigenomic deregulation that transcend cancer types.10.1073/pnas.0307554100 how does a network of documents grow without centralized control? this question is becoming crucial as we try to explain the emergent scale-free topology of the world wide web and use link analysis to identify important information resources. existing models of growing information networks have focused on the structure of links but neglected the content of nodes. here i show that the current models fail to reproduce a critical characteristic of information networks, namely the distribution of textual similarity among linked documents. i propose a more realistic model that generates links by using both popularity and content. this model yields remarkably accurate predictions of both degree and similarity distributions in networks of web pages and scientific literature.the past decade has seen great advances in our understanding of the role of noise in gene regulation and the physical limits to signaling in biological networks. here, we introduce the spectral method for computation of the joint probability distribution over all species in a biological network. the spectral method exploits the natural eigenfunctions of the master equation of birth – death processes to solve for the joint distribution of modules within the network, which then inform each other and facilitate calculation of the entire joint distribution. we illustrate the method on a ubiquitous case in nature: linear regulatory cascades. the efficiency of the method makes possible numerical optimization of the input and regulatory parameters, revealing design properties of, e.g., the most informative cascades. we find, for threshold regulation, that a cascade of strong regulations converts a unimodal input to a bimodal output, that multimodal inputs are no more informative than bimodal inputs, and that a chain of up-regulations outperforms a chain of down-regulations. we anticipate that this numerical approach may be useful for modeling noise in a variety of small network topologies in biology.changes in gene expression underlie phenotypic plasticity, variation within species, and phenotypic divergence between species. these expression differences arise from modulation of regulatory networks. to understand the source of expression differences, networks of interactions among genes and gene products that orchestrate gene expression must be considered. here i review the basic structure of eukaryotic regulatory networks and discuss selected case studies that provide insight into how these networks are altered to create expression differences within and between species.motivation: the regulation of proliferation and differentiation of embryonic and adult stem cells into mature cells is central to developmental biology. gene expression measured in distinguishable developmental stages helps to elucidate underlying molecular processes. in previous work we showed that functional gene modules, which act distinctly in the course of development, can be represented by a mixture of trees. in general, the similarities in the gene expression programs of cell populations reflect the similarities in the differentiation path.it is widely suspected that gene regulatory networks are highly plastic. the rapid turnover of transcription factor binding sites has been predicted on theoretical grounds and has been experimentally demonstrated in closely related species. we combined experimental approaches with comparative genomics to focus on the role of combinatorial control in the evolution of a large transcriptional circuit in the fungal lineage. our study centers on mcm1, a transcriptional regulator that, in combination with five cofactors, binds roughly 4\\\\% of the genes in saccharomyces cerevisiae and regulates processes ranging from the cell-cycle to mating. in kluyveromyces lactis and candida albicans , two other hemiascomycetes, we find that the mcm1 combinatorial circuits are substantially different. this massive rewiring of the mcm1 circuitry has involved both substantial gain and loss of targets in ancient combinatorial circuits as well as the formation of new combinatorial interactions. we have dissected the gains and losses on the global level into subsets of functionally and temporally related changes. one particularly dramatic change is the acquisition of mcm1 binding sites in close proximity to rap1 binding sites at 70 ribosomal protein genes in the k. lactis lineage. another intriguing and very recent gain occurs in the c. albicans lineage, where mcm1 is found to bind in combination with the regulator wor1 at many genes that function in processes associated with adaptation to the human host, including the white-opaque epigenetic switch. the large turnover of mcm1 binding sites and the evolution of new mcm1\\\\^{a}€ ” cofactor interactions illuminate in sharp detail the rapid evolution of combinatorial transcription networks.external perturbations, by forcing cells to adapt to a new environment, often elicit large-scale changes in gene expression resulting in an altered proteome that improves the cell\\'s fitness in the new conditions. steady-state levels of a proteome depend on transcription, the levels of transcripts, translation and protein degradation but system-level contribution that each of these processes make to the final protein expression change has yet to be explored. we therefore applied a systems biology approach to characterize the regulation of protein expression during cellular differentiation using quantitative proteomics. as a general rule, it seems that protein expression during cellular differentiation is largely controlled by changes in the relative synthesis rate, whereas the relative degradation rate of the majority of proteins stays constant. in these data, we also observe that the proteins in defined sub-structures of larger protein complexes tend to have highly correlated synthesis and degradation rates but that this does not necessarily extend to the holo-complex. finally, we provide strong evidence that the generally poor correlation observed between transcript and protein levels can fully be explained once the protein synthesis and degradation rates are taken into account.motivation: methods available for the inference of genetic regulatory networks strive to produce a single network, usually by optimizing some quantity to fit the experimental observations. in this article we investigate the possibility that multiple networks can be inferred, all resulting in similar dynamics. this idea is motivated by theoretical work which suggests that biological networks are robust and adaptable to change, and that the overall behavior of a genetic regulatory network might be captured in terms of dynamical basins of attraction.cell types are fundamental units of multicellular life but their evolution is obscure. how did the first cell types emerge and become distinct in animal evolution? what were the sets of cell types that existed at important evolutionary nodes that represent eumetazoan or bilaterian ancestors? how did these ancient cell types diversify further during the evolution of organ systems in the descending evolutionary lines? the recent advent of cell type molecular fingerprinting has yielded initial insights into the evolutionary interrelationships of cell types between remote animal phyla and has allowed us to define some first principles of cell type diversification in animal evolution.understanding the functional consequences of genetic variation, and how it affects complex human disease and quantitative traits, remains a critical challenge for biomedicine. we present an analysis of {rna} sequencing data from 1641 samples across 43 tissues from 175 individuals, generated as part of the pilot phase of the {genotype-tissue} expression ({gtex}) project. we describe the landscape of gene expression across tissues, catalog thousands of tissue-specific and shared regulatory expression quantitative trait loci ({eqtl}) variants, describe complex network relationships, and identify signals from genome-wide association studies explained by {eqtls}. these findings provide a systematic understanding of the cellular and biological consequences of human genetic variation and of the heterogeneity of such effects among a diverse set of human tissues. copyright {\\\\copyright} 2015, american association for the advancement of science.characterizing the {dna}-binding specificities of transcription factors is a key problem in computational biology that has been addressed by multiple algorithms. these usually take as input sequences that are putatively bound by the same factor and output one or more {dna} motifs. a common practice is to apply several such algorithms simultaneously to improve coverage at the price of redundancy. in interpreting such results, two tasks are crucial: clustering of redundant motifs, and attributing the motifs to transcription factors by retrieval of similar motifs from previously characterized motif libraries. both tasks inherently involve motif comparison. here we present a novel method for comparing and merging motifs, based on bayesian probabilistic principles. this method takes into account both the similarity in positional nucleotide distributions of the two motifs and their dissimilarity to the background distribution. we demonstrate the use of the new comparison method as a basis for motif clustering and retrieval procedures, and compare it to several commonly used alternatives. our results show that the new method outperforms other available methods in accuracy and sensitivity. we incorporated the resulting motif clustering and retrieval procedures in a large-scale automated pipeline for analyzing {dna} motifs. this pipeline integrates the results of various {dna} motif discovery algorithms and automatically merges redundant motifs from multiple training sets into a coherent annotated library of motifs. application of this pipeline to recent genome-wide transcription factor location data in s. cerevisiae successfully identified {dna} motifs in a manner that is as good as semi-automated analysis reported in the literature. moreover, we show how this analysis elucidates the mechanisms of condition-specific preferences of transcription factors.this primer discusses a study from bradley et al. in which changes in transcription factor binding between two drosophila species are revealed by chromatin immunoprecipitation and high-throughput sequencing.epigenomics is the study of the physical modifications, associations and conformations of genomic {dna} sequences, with the aim of linking these with epigenetic memory, cellular identity and tissue-specific functions. while current techniques in the field are characterizing the average epigenomic features across large cell ensembles,an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organisms{background}: recent technical and methodological advances have placed microbial models at the forefront of evolutionary and environmental genomics. to better understand the logic of genetic network evolution, we combined comparative transcriptomics, a differential clustering algorithm and promoter analyses in a study of the evolution of transcriptional networks responding to an antifungal agent in two yeast species: the free-living model organism saccharomyces cerevisiae and the human pathogen candida glabrata. {results}: we found that although the gene expression patterns characterizing the response to drugs were remarkably conserved between the two species, part of the underlying regulatory networks differed. in particular, the roles of the oxidative stress response transcription factors {scyap1p} (in s. cerevisiae) and cgap1p (in c. glabrata) had diverged. the sets of genes whose benomyl response depends on these factors are significantly different. also, the {dna} motifs targeted by {scyap1p} and cgap1p are differently represented in the promoters of these genes, suggesting that the {dna} binding properties of the two proteins are slightly different. experimental assays of {scyap1p} and cgap1p activities in vivo were in accordance with this last observation. {conclusions}: based on these results and recently published data, we suggest that the robustness of environmental stress responses among related species contrasts with the rapid evolution of regulatory sequences, and depends on both the coevolution of transcription factor binding properties and the versatility of regulatory associations within transcriptional networks.proteins do not carry out their functions alone. instead, they often act by participating in macromolecular complexes and play different functional roles depending on the other members of the complex. it is therefore interesting to identify co-complex relationships. although protein complexes can be identified in a high-throughput manner by experimental technologies such as affinity purification coupled with mass spectrometry ({apms}), these large-scale datasets often suffer from high false positive and false negative rates. here, we present a computational method that predicts co-complexed protein pair ({ccpp}) relationships using kernel methods from heterogeneous data sources. we show that a diffusion kernel based on random walks on the full network topology yields good performance in predicting {ccpps} from protein interaction networks. in the setting of direct ranking, a diffusion kernel performs much better than the mutual clustering coefficient. in the setting of {svm} classifiers, a diffusion kernel performs much better than a linear kernel. we also show that combination of complementary information improves the performance of our {ccpp} recognizer. a summation of three diffusion kernels based on two-hybrid, {apms}, and genetic interaction networks and three sequence kernels achieves better performance than the sequence kernels or diffusion kernels alone. inclusion of additional features achieves a still better {roc} 50 of 0.937. assuming a negative-to-positive ratio of 600:1, the final classifier achieves 89.3\\\\% coverage at an estimated false discovery rate of 10\\\\%. finally, we applied our prediction method to two recently described {apms} datasets. we find that our predicted positives are highly enriched with {ccpps} that are identified by both datasets, suggesting that our method successfully identifies true {ccpps}. an {svm} classifier trained from heterogeneous data sources provides accurate predictions of {ccpps} in yeast. this computational method thereby provides an inexpensive method for identifying protein complexes that extends and complements high-throughput experimental data.to study the evolutionary dynamics of regulatory {dna}, we mapped >1.3 million deoxyribonuclease i-hypersensitive sites ({dhss}) in 45 mouse cell and tissue types, and systematically compared these with human {dhs} maps from orthologous compartments. we found that the mouse and human genomes have undergone extensive cis-regulatory rewiring that combines branch-specific evolutionary innovation and loss with widespread repurposing of conserved {dhss} to alternative cell fates, and that this process is mediated by turnover of transcription factor ({tf}) recognition elements. despite pervasive evolutionary remodeling of the location and content of individual cis-regulatory regions, within orthologous mouse and human cell types the global fraction of regulatory {dna} bases encoding recognition sites for each {tf} has been strictly conserved. our findings provide new insights into the evolutionary forces shaping mammalian regulatory {dna} landscapes. copyright {\\\\copyright} 2014, american association for the advancement of science.10.1126/scisignal.2000037the abundance of genome polymorphism and divergence data has provided unprecedented insight into how mutation, drift and natural selection shape genome evolution. application of the {mcdonald}–kreitman ({mk}) test to such data indicates a pervasive influence of positive selection, particularly in drosophila species. however, evidence for positive selection in other species ranging from yeast to humans is often weak or absent. although evidence for positive selection could be obscured in some species, there is also reason to believe that the frequency of adaptive substitutions could be overestimated as a result of epistatic fitness effects or hitchhiking of deleterious mutations. based on these considerations it is argued that the common assumption of independence among sites must be relaxed before abandoning the neutral theory of molecular evolution.{background}:graphs provide a natural framework for visualizing and analyzing networks of many types, including biological networks. network clustering is a valuable approach for summarizing the structure in large networks, for predicting unobserved interactions, and for predicting functional annotations. many current clustering algorithms suffer from a common set of limitations: poor resolution of top-level clusters; over-splitting of bottom-level clusters; requirements to pre-define the number of clusters prior to analysis; and an inability to jointly cluster over multiple interaction {types.results}:a new algorithm, hierarchical agglomerative clustering ({hac}), is developed for fast clustering of heterogeneous interaction networks. this algorithm uses maximum likelihood to drive the inference of a hierarchical stochastic block model for network structure. bayesian model selection provides a principled method for collapsing the fine-structure within the smallest groups, and for identifying the top-level groups within a network. model scores are additive over independent interaction types, providing a direct route for simultaneous analysis of multiple interaction types. in addition to inferring network structure, this algorithm generates link predictions that with cross-validation provide a quantitative assessment of performance for real-world {examples.conclusions}:when applied to genome-scale data sets representing several organisms and interaction types, {hac} provides the overall best performance in link prediction when compared with other clustering methods and with model-free graph diffusion kernels. investigation of performance on genome-scale yeast protein interactions reveals roughly 100 top-level clusters, with a long-tailed distribution of cluster sizes. these are in turn partitioned into 1000 fine-level clusters containing 5 proteins on average, again with a long-tailed size distribution. top-level clusters correspond to broad biological processes, whereas fine-level clusters correspond to discrete complexes. surprisingly, link prediction based on joint clustering of physical and genetic interactions performs worse than predictions based on individual data sets, suggesting a lack of synergy in current high-throughput data.{background}:microbes must sense environmental stresses, transduce these signals and mount protective responses to survive in hostile environments. in this study we have tested the hypothesis that fungal stress signalling pathways have evolved rapidly in a niche-specific fashion that is independent of phylogeny. to test this hypothesis we have compared the conservation of stress signalling molecules in diverse fungal species with their stress resistance. these fungi, which include ascomycetes, basidiomycetes and microsporidia, occupy highly divergent niches from saline environments to plant or mammalian {hosts.results}:the fungi displayed significant variation in their resistance to osmotic ({nacl} and sorbitol), oxidative ({h2o2} and menadione) and cell wall stresses (calcofluor white and congo red). there was no strict correlation between fungal phylogeny and stress resistance. rather, the human pathogens tended to be more resistant to all three types of stress, an exception being the sensitivity of candida albicans to the cell wall stress, calcofluor white. in contrast, the plant pathogens were relatively sensitive to oxidative stress. the degree of conservation of osmotic, oxidative and cell wall stress signalling pathways amongst the eighteen fungal species was examined. putative orthologues of functionally defined signalling components in saccharomyces cerevisiae were identified by performing reciprocal {blastp} searches, and the percent amino acid identities of these orthologues recorded. this revealed that in general, central components of the osmotic, oxidative and cell wall stress signalling pathways are relatively well conserved, whereas the sensors lying upstream and transcriptional regulators lying downstream of these modules have diverged significantly. there was no obvious correlation between the degree of conservation of stress signalling pathways and the resistance of a particular fungus to the corresponding {stress.conclusion}:our data are consistent with the hypothesis that fungal stress signalling components have undergone rapid recent evolution to tune the stress responses in a niche-specific fashion.motivation: many methods have been described to identify regulatory motifs in the transcription control regions of genes that exhibit similar patterns of gene expression across a variety of experimental conditions. here we focus on a single experimental condition, and utilize gene expression data to identify sequence motifs associated with genes that are activated under this experimental condition. we use a linear model with two-way interactions to model gene expression as a function of sequence features (words) present in presumptive transcription control regions. the most relevant features are selected by a feature selection method called stepwise selection with monte carlo cross validation. we apply this method to a publicly available dataset of the yeast saccharomyces cerevisiae, focussing on the 800 basepairs immediately upstream of each gene\\'s translation start site (the upstream control region ({ucr})).extracting biologically meaningful information from chromosomal interactions obtained with genome-wide chromosome conformation capture ({3c}) analyses requires the elimination of systematic biases. we present a computational pipeline that integrates a strategy to map sequencing reads with a data-driven method for iterative correction of biases, yielding genome-wide maps of relative contact probabilities. we validate this {ice} (iterative correction and eigenvector decomposition) technique on published data obtained by the high-throughput {3c} method {hi-c}, and we demonstrate that eigenvector decomposition of the obtained maps provides insights into local chromatin states, global patterns of chromosomal interactions, and the conserved organization of human and mouse chromosomes.although it was originally believed that enhancers activate only the nearest promoter, recent global analyses enabled by high-throughput technology suggest that the network of enhancer-promoter interactions is far more complex. the mechanisms that determine the specificity of enhancer-promoter interactions are still poorly understood, but they are thought to include biochemical compatibility, constraints imposed by the three-dimensional architecture of chromosomes, insulator elements, and possibly the effects of local chromatin composition. in this review, we assess the current insights into these determinants, and highlight the functional genomic approaches that will lead the way towards better mechanistic understanding. copyright {\\\\copyright} 2014 elsevier ltd. all rights reserved.complex traits are the product of multiple genes with effects that depend on both the genetic and environmental background. although this complexity makes a comprehensive genetic analysis difficult, identification of even a single gene provides insight into the biochemical and/or signaling pathway underlying a trait. however, it is unknown whether multiple pathways, and consequently multiple genes, must be identified to adequately understand a trait\\'s molecular basis. using crosses between three natural isolates of saccharomyces cerevisiae, we mapped sensitivity to a number of pharmacologically active compounds to a single nonsynonymous polymorphism in cystathione-β-synthase ({cys4}), which is required for the first committed step in the cysteine biosynthesis pathway. drug sensitivity is mediated by a deficiency in cysteine and consequently glutathione production, because drug sensitivity is abrogated by cysteine or glutathione supplementation. within a diverse panel of 60 natural yeast isolates, the drug-sensitive {cys4} allele is rare, and glutathione supplementation failed to alleviate drug-dependent growth defects in two other drug-sensitive strains. these results implicate the cysteine/glutathione biosynthesis pathway as a significant, but not the sole contributor to pharmacological variation in yeast.the epigenome is established and maintained by the site-specific recruitment of chromatin-modifying enzymes and their cofactors. identifying the cis elements that regulate epigenomic modification is critical for understanding the regulatory mechanisms that control gene expression patterns. we present epigram, an analysis pipeline that predicts histone modification and {dna} methylation patterns from {dna} motifs. the identified cis elements represent interactions with the site-specific {dna}-binding factors that establish and maintain epigenomic modifications. we cataloged the cis elements in embryonic stem cells and four derived lineages and found numerous motifs that have location preference, such as at the center of {h3k27ac} or at the edges of {h3k4me3} and {h3k9me3}, which provides mechanistic insight about the shaping of the epigenome. the epigram pipeline and predictive motifs are at http://wanglab.ucsd.edu/star/epigram/.conditional maximum entropy ({me}) models provide a general purpose machine learning technique which has been successfully applied to fields as diverse as computer vision and econometrics, and which is used for a wide variety of classification problems in natural language processing. however, the flexibility of {me} models is not without cost. while parameter estimation for {me} models is conceptually straightforward, in practice {me} models for typical natural language tasks are very large, and may...spatially distinct gene expression profiles in neural stem cells ({nscs}) are a prerequisite to the formation of neuronal diversity, but how these arise from the regulatory interactions between chromatin accessibility and transcription factor activity has remained unclear. here, we demonstrate that, despite their distinct gene expression profiles, {nscs} of the mouse cortex and spinal cord share the majority of their {dnase} i hypersensitive sites ({dhss}). regardless of this similarity, domain-specific gene expression is highly correlated with the relative accessibility of associated {dhss}, as determined by sequence read density. notably, the binding pattern of the general {nsc} transcription factor {sox2} is also largely cell type specific and coincides with an enrichment of {lhx2} motifs in the cortex and {hoxa9} motifs in the spinal cord. interestingly, in a zebrafish reporter gene system, these motifs were critical determinants of patterned gene expression along the rostral-caudal axis. our findings establish a predictive model for patterned {nsc} gene expression, whereby domain-specific expression of {lhx2} and {hox} proteins act on their target motifs within commonly accessible cis-regulatory regions to specify {sox2} binding. in turn, this binding correlates strongly with these {dhss} relative accessibility—a robust predictor of neighboring gene expression.motivation: large amount of research efforts have been focused on estimating gene networks based on gene expression data to understand the functional basis of a living organism. such networks are often obtained by considering pairwise correlations between genes, thus may not reflect the true connectivity between genes. by treating gene expressions as quantitative traits while considering genetic markers, genetical genomics analysis has shown its power in enhancing the understanding of gene regulations. previous works have shown the improved performance on estimating the undirected network graphical structure by incorporating genetic markers as covariates. knowing that gene expressions are often due to directed regulations, it is more meaningful to estimate the directed graphical network.precise regulatory control of genes, particularly in eukaryotes, frequently requires the joint action of multiple sequence-specific transcription factors. a cis-regulatory module ({crm}) is a genomic locus that is responsible for gene regulation and that contains multiple transcription factor binding sites in close proximity. given a collection of known transcription factor binding motifs, many bioinformatics methods have been proposed over the past 15 years for identifying within a genomic sequence candidate {crms} consisting of clusters of those motifs. the {mcast} algorithm uses a hidden markov model with a p-value-based scoring scheme to identify candidate {crms}. here, we introduce a new version of {mcast} that offers improved graphical output, a dynamic background model, statistical confidence estimates based on false discovery rate estimation and, most significantly, the ability to predict {crms} while taking into account epigenomic data such as {dnase} i sensitivity or histone modification data. we demonstrate the validity of {mcast}\\'s statistical confidence estimates and the utility of epigenomic priors in identifying {crms}. {mcast} is part of the {meme} suite software toolkit. a web server and source code are available at http://meme-suite.org and http://alternate.meme-suite.org t.bailey@imb.uq.edu.au or william-noble@uw.edu supplementary data are available at bioinformatics online. {\\\\copyright} the author 2015. published by oxford university press. all rights reserved. for permissions, please e-mail: journals.permissions@oup.com.multipotent differentiation, where cells adopt one of several possible fates, occurs in diverse systems ranging from bacteria to mammals. this decision-making process is driven by multiple differentiation programs that operate simultaneously in the cell. how these programs interact to govern cell fate choice is poorly understood. to investigate this issue, we simultaneously measured activities of the competing sporulation and competence programs in single bacillus subtilis cells. this approach revealed that these competing differentiation programs progress independently without cross-regulation before the decision point. cells seem to arrive at a fate choice through differences in the relative timing between the two programs. to test this proposed dynamic mechanism, we altered the relative timing by engineering artificial cross-regulation between the sporulation and competence circuits. results suggest a simple model that does not require a checkpoint or intricate cross-regulation before cellular decision-making. rather, cell fate choice appears to be the outcome of a \\'molecular race\\' between differentiation programs that compete in time, providing a simple dynamic mechanism for decision-making.several lines of evidence point to a key role for dynamic epigenetic changes during brain development, maturation, and learning. {dna} methylation ({mc}) is a stable covalent modification that persists in post-mitotic cells throughout their lifetime, defining their cellular identity. however, the methylation status at each of the \\\\~{}1 billion cytosines in the genome is potentially an information-rich and flexible substrate for epigenetic modification that can be altered by cellular activity. indeed, changes in {dna} methylation have been implicated in learning and memory, as well as in age-related cognitive decline. however, little is known about the cell type–specific patterning of {dna} methylation and its dynamics during mammalian brain development.we consider the problem of multi-task learning, that is, learning multiple related functions. our approach is based on a hierarchical bayesian framework, that exploits the equivalence between parametric linear models and nonparametric gaussian processes ({gps}). the resulting models can be learned easily via an {em}-algorithm. empirical studies on multi-label text categorization suggest that the presented models allow accurate solutions of these multi-task problems.many essential transcription factors have conserved roles in regulating biological programs, yet their genomic occupancy can diverge significantly. a new study demonstrates that such variations are primarily due to cis-regulatory sequences, rather than differences between the regulators or nuclear environments.10.1073/pnas.1732547100 we present a general algorithm to detect genes differentially expressed  between two nonhomogeneous time-series data sets. as increasing amounts of  high-throughput biological data become available, a major challenge in genomic  and computational biology is to develop methods for comparing data from  different experimental sources. time-series whole-genome expression data are a  particularly valuable source of information because they can describe an  unfolding biological process such as the cell cycle or immune response.  however, comparisons of time-series expression data sets are hindered by  biological and experimental inconsistencies such as differences in sampling  rate, variations in the timing of biological processes, and the lack of  repeats. our algorithm overcomes these difficulties by using a continuous  representation for time-series data and combining a noise model for individual  samples with a global difference measure. we introduce a corresponding  statistical method for computing the significance of this differential  expression measure. we used our algorithm to compare cell-cycle-dependent gene  expression in wild-type and knockout yeast strains. our algorithm identified a  set of 56 differentially expressed genes, and these results were validated by  using independent {protein-dna}-binding data. unlike previous methods, our  algorithm was also able to identify 22 non-cell-cycle-regulated genes as  differentially expressed. this set of genes is significantly correlated in a  set of independent expression experiments, suggesting additional roles for the  transcription factors fkh1 and fkh2 in controlling cellular activity in  yeast.we consider the problem of estimating the parameters of a gaussian or binary distribution in such a way that the resulting undirected graphical model is sparse. our approach is to solve a maximum likelihood problem with an added l1-norm penalty term. the problem as formulated is convex but the memory requirements and complexity of existing interior point methods are prohibitive for problems with more than tens of nodes. we present two new algorithms for solving problems with at least a thousand nodes in the gaussian case. our first algorithm uses block coordinate descent, and can be interpreted as recursive l1-norm penalized regression. our second algorithm, based on nesterov\\'s first order method, yields a complexity estimate with a better dependence on problem size than existing interior point methods. using a log determinant relaxation of the log partition function (wainwright and jordan, 2006), we show that these same algorithms can be used to solve an approximate sparse maximum likelihood problem for the binary case. we test our algorithms on synthetic data, as well as on gene expression and senate voting records data.we consider the problem of discovering gene regulatory networks from time-series microarray data. recently, graphical granger modeling has gained considerable attention as a promising direction for addressing this problem. these methods apply graphical modeling methods on time-series data and invoke the notion of \\'granger causality\\' to make assertions on causality through inference on time-lagged effects. existing algorithms, however, have neglected an important aspect of the problem—the group structure among the lagged temporal variables naturally imposed by the time series they belong to. specifically, existing methods in computational biology share this shortcoming, as well as additional computational limitations, prohibiting their effective applications to the large datasets including a large number of genes and many data points. in the present article, we propose a novel methodology which we term \\'grouped graphical granger modeling method\\', which overcomes the limitations mentioned above by applying a regression method suited for high-dimensional and large data, and by leveraging the group structure among the lagged temporal variables according to the time series they belong to. we demonstrate the effectiveness of the proposed methodology on both simulated and actual gene expression data, specifically the human cancer cell ({hela} s3) cycle data. the simulation results show that the proposed methodology generally exhibits higher accuracy in recovering the underlying causal structure. those on the gene expression data demonstrate that it leads to improved accuracy with respect to prediction of known links, and also uncovers additional causal relationships uncaptured by earlier {works.contact}: aclozano@us.ibm.comchromatin immunoprecipitation coupled with deep sequencing ({chip}-seq) has great potential for elucidating transcriptional networks, by measuring genome-wide binding of transcription factors ({tfs}) at high resolution. despite the precision of these experiments, identification of genes directly regulated by a {tf} (target genes) is not trivial. numerous target gene scoring methods have been used in the past. however, their suitability for the task and their performance remain unclear, because a thorough comparative assessment of these methods is still lacking. here we present a systematic evaluation of computational methods for defining {tf} targets based on {chip}-seq data. we validated predictions based on 68 {chip}-seq studies using a wide range of genomic expression data and functional information. we demonstrate that peak-to-gene assignment is the most crucial step for correct target gene prediction and propose a parameter-free method performing most consistently across the evaluation tests. transcription factors ({tfs}) are the main regulators of gene transcription. thus, knowing the genes that are targeted by a specific {tf} is of utmost importance for understanding developmental processes, cellular stress response, or disease etiology. chromatin immunoprecipitation coupled with deep sequencing ({chip}-seq) allows for measuring the genome-wide binding of {tfs}. several computational methods have been used for inferring the genes that are targeted by {tfs} using this binding information, but a thorough evaluation of their performance has not been performed so far. here we present an assessment of a range of {tf}-target-calling methods using 68 {chip}-seq datasets. it turns out that the first step of the scoring, the assignment of binding events to genes, is the most important for correctly calling target genes. our evaluation revealed important performance differences between the target-calling methods, with some simplistic methods exhibiting a particularly poor performance compared to more elaborate scorings. one of the methods is particularly attractive, because it does not require the a priori definition of any parameter — all parameters are \\'learned\\' from the data. this and other methods tested were implemented in a freely available software package for future testing and application to other {chip}-seq datasets.transcription factors ({tfs}) direct gene expression by binding to {dna} regulatory regions. to explore the evolution of gene regulation, we used chromatin immunoprecipitation with high-throughput sequencing ({chip}-seq) to determine experimentally the genome-wide occupancy of two {tfs}, {ccaat}/enhancer-binding protein alpha and hepatocyte nuclear factor 4 alpha, in the livers of five vertebrates. although each {tf} displays highly conserved {dna} binding preferences, most binding is species-specific, and aligned binding events present in all five species are rare. regions near genes with expression levels that are dependent on a {tf} are often bound by the {tf} in multiple species yet show no enhanced {dna} sequence constraint. binding divergence between species can be largely explained by sequence changes to the bound motifs. among the binding events lost in one lineage, only half are recovered by another binding event within 10 kilobases. our results reveal large interspecies differences in transcriptional regulation and provide insight into regulatory evolution.sequence preferences of {dna} binding proteins are a primary mechanism by which cells interpret the genome. despite the central importance of these proteins in physiology, development, and evolution, comprehensive {dna} binding specificities have been determined experimentally for only a few proteins. here, we used microarrays containing all 10–base pair sequences to examine the binding specificities of 104 distinct mouse {dna} binding proteins representing 22 structural classes. our results reveal a complex landscape of binding, with virtually every protein analyzed possessing unique preferences. roughly half of the proteins each recognized multiple distinctly different sequence motifs, challenging our molecular understanding of how proteins interact with their {dna} binding sites. this complexity in {dna} recognition may be important in gene regulation and in the evolution of transcriptional regulatory networks.choosing good problems is essential for being a good scientist. but what is a good problem, and how do you choose one? the subject is not usually discussed explicitly within our profession. scientists are expected to be smart enough to figure it out on their own and through the observation of their teachers. this lack of explicit discussion leaves a vacuum that can lead to approaches such as choosing problems that can give results that merit publication in valued journals, resulting in a job and tenure.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsa network of disorders and disease genes linked by known disorder-gene associations offers a platform to explore in a single graph-theoretic framework all known phenotype and disease gene associations, indicating the common genetic origin of many diseases. genes associated with similar disorders show both higher likelihood of physical interactions between their products and higher expression profiling similarity for their transcripts, supporting the existence of distinct disease-specific functional modules. we find that essential human genes are likely to encode hub proteins and are expressed widely in most tissues. this suggests that disease genes also would play a central role in the human interactome. in contrast, we find that the vast majority of disease genes are nonessential and show no tendency to encode hub proteins, and their expression pattern indicates that they are localized in the functional periphery of the network. a selection-based model explains the observed difference between essential and disease genes and also suggests that diseases caused by somatic mutations should not be peripheral, a prediction we confirm for cancer genes.which transcription factors control the distribution of metabolic fluxes under a given condition? we address this question by systematically quantifying metabolic fluxes in 119 transcription factor deletion mutants of saccharomyces cerevisiae under five growth conditions. while most knockouts did not affect fluxes, we identified 42 condition-dependent interactions that were mediated by a total of 23 transcription factors that control almost exclusively the cellular decision between respiration and fermentation. this relatively sparse, condition-specific network of active metabolic control contrasts with the much larger gene regulation network inferred from expression and {dna} binding data. based on protein and transcript analyses in key mutants, we identified three enzymes in the tricarboxylic acid cycle as the key targets of this transcriptional control. for the transcription factor gcn4, we demonstrate that this control is mediated through the {pka} and snf1 signaling cascade. the discrepancy between flux response predictions, based on the known regulatory network architecture and our functional {13c}-data, demonstrates the importance of identifying and quantifying the extent to which regulatory effectors alter cellular functions.although genomewide {rna} expression analysis has become a routine tool in biomedical research, extracting biological insight from such information remains a major challenge. here, we describe a powerful analytical method called gene set enrichment analysis ({gsea}) for interpreting gene expression data. the method derives its power by focusing on gene sets, that is, groups of genes that share common biological function, chromosomal location, or regulation. we demonstrate how {gsea} yields insights into several cancer-related data sets, including leukemia and lung cancer. notably, where single-gene analysis finds little similarity between two independent studies of patient survival in lung cancer, {gsea} reveals many biological pathways in common. the {gsea} method is embodied in a freely available software package, together with an initial database of 1,325 biologically defined gene sets.the human body is composed of diverse cell types with distinct functions. although it is known that lineage specification depends on cell-specific gene expression, which in turn is driven by promoters, enhancers, insulators and other cis-regulatory {dna} sequences for each gene, the relative roles of these regulatory elements in this process are not clear. we have previously developed a chromatin-immunoprecipitation-based microarray method ({chip}-chip) to locate promoters, enhancers and insulators in the human genome. here we use the same approach to identify these elements in multiple cell types and investigate their roles in cell-type-specific gene expression. we observed that the chromatin state at promoters and {ctcf}-binding at insulators is largely invariant across diverse cell types. in contrast, enhancers are marked with highly cell-type-specific histone modification patterns, strongly correlate to cell-type-specific gene expression programs on a global scale, and are functionally active in a cell-type-specific manner. our results define over 55,000 potential transcriptional enhancers in the human genome, significantly expanding the current catalogue of human enhancers and highlighting the role of these elements in cell-type-specific gene expression.interacting or functionally related protein families tend to have similar phylogenetic trees. based on this observation, techniques have been developed to predict interaction partners. the observed degree of similarity between the phylogenetic trees of two proteins is the result of many different factors besides the actual interaction or functional relationship between them. such factors influence the performance of interaction predictions. one aspect that can influence this similarity is related to the fact that a given protein interacts with many others, and hence it must adapt to all of them. accordingly, the interaction or coadaptation signal within its tree is a composite of the influence of all of the interactors. here, we introduce a new estimator of coevolution to overcome this and other problems. instead of relying on the individual value of tree similarity between two proteins, we use the whole network of similarities between all of the pairs of proteins within a genome to reassess the similarity of that pair, thereby taking into account its coevolutionary context. we show that this approach offers a substantial improvement in interaction prediction performance, providing a degree of accuracy/coverage comparable with, or in some cases better than, that of experimental techniques. moreover, important information on the structure, function, and evolution of macromolecular complexes can be inferred with this methodology.the study of starvation-resistant biological programs has elucidated numerous mechanisms influencing aging. here we present the discovery and characterization of starvation-induced adult reproductive diapause ({ard}) in caenorhabditis elegans. {ard} differs from the c. elegans dauer diapause in that it enables sexually mature adults to delay reproductive onset 15-fold and extend total adult life span at least threefold. the effectiveness of {ard} requires apoptotic death of the entire germ line, except for a small population of protected germline stem cells ({gscs}). when feeding is resumed, surviving {gscs} regenerate a new germ line capable of offspring production near the level of nonstarved animals. the starvation-sensing nuclear receptor {nhr}-49 is required for {ard} entry and recovery. our findings establish mechanisms for preserving stem cell potency and reproductive potential during prolonged starvation.to elucidate gene function on a global scale, we identified pairs of genes that are coexpressed over 3182 {dna} microarrays from humans, flies, worms, and yeast. we found 22,163 such coexpression relationships, each of which has been conserved across evolution. this conservation implies that the coexpression of these gene pairs confers a selective advantage and therefore that these genes are functionally related. manyof these relationships provide strong evidence for the involvement of new genes in core biological functions such as the cell cycle, secretion, and protein expression. we experimentallyconfirmed the predictions implied bysome of these links and identified cell proliferation functions for several genes. by assembling these links into a gene-coexpression network, we found several components that were animal-specific as well as interrelationships between newly evolved and ancient modules.this tutorial provides an overview of the basic theory of hidden markov models ({hmms}) as originated by {l.e}. baum and t. petrie (1966) and gives practical details on methods of implementation of the theory along with a description of selected applications of the theory to distinct problems in speech recognition. results from a number of original sources are combined to provide a single source of acquiring the background required to pursue further this area of research. the author first reviews the theory of discrete markov chains and shows how the concept of hidden states, where the observation is a probabilistic function of the state, can be used effectively. the theory is illustrated with two simple examples, namely coin-tossing, and the classic balls-in-urns system. three fundamental problems of {hmms} are noted and several practical techniques for solving these problems are given. the various types of {hmms} that have been studied, including ergodic as well as left-right models, are describedrecent technological advances have expanded the breadth of available omic data, from whole-genome sequencing data, to extensive transcriptomic, methylomic and metabolomic data. a key goal of analyses of these data is the identification of effective models that predict phenotypic traits and outcomes, elucidating important biomarkers and generating important insights into the genetic underpinnings of the heritability of complex traits. there is still a need for powerful and advanced analysis strategies to fully harness the utility of these comprehensive high-throughput data, identifying true associations and reducing the number of false associations. in this review, we explore the emerging approaches for data integration - including meta-dimensional and multi-staged analyses - which aim to deepen our understanding of the role of genetics and genomics in complex outcomes. with the use and further development of these approaches, an improved understanding of the relationship between genomic variation and human phenotypes may be revealed.gene expression array data provide shadows of intricate cellular processes. learning how to make the most of the information present in expression arrays has become a discipline in itself. in recent years, there has been an explosion of methods that analyze gene expression arrays to produce long lists of genes that express differentially in distinct cellular states. these lists will have to be organized, and the algorithms that produced them combined, if we wish to piece together the rich cellular structures probed by this high-throughput technology. researchers will have to understand the benefits and limitations of the many existing methods to produce the combination of algorithms that best suits their gene expression experiments.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsthe ageing of the human brain is a cause of cognitive decline in the elderly and the major risk factor for alzheimer\\'s disease. the time in life when brain ageing begins is undefined. here we show that transcriptional profiling of the human frontal cortex from individuals ranging from 26 to 106 years of age defines a set of genes with reduced expression after age 40. these genes play central roles in synaptic plasticity, vesicular transport and mitochondrial function. this is followed by induction of stress response, antioxidant and {dna} repair genes. {dna} damage is markedly increased in the promoters of genes with reduced expression in the aged cortex. moreover, these gene promoters are selectively damaged by oxidative stress in cultured human neurons, and show reduced base-excision {dna} repair. thus, {dna} damage may reduce the expression of selectively vulnerable genes involved in learning, memory and neuronal survival, initiating a programme of brain ageing that starts early in adult life.we provide here a molecular movie that captures key aspects of {rna} polymerase {ii} initiation and elongation. to create the movie, we combined structural snapshots of the initiation-elongation transition and of elongation, including nucleotide addition, translocation, pausing, proofreading, backtracking, arrest, reactivation, and inhibition. the movie reveals open questions about the mechanism of transcription and provides a useful teaching tool. copyright {\\\\copyright} 2012 elsevier inc. all rights reserved.gene duplication and loss is a powerful source of functional innovation. however, the general principles that govern this process are still largely unknown. with the growing number of sequenced genomes, it is now possible to examine these events in a comprehensive and unbiased manner. here, we develop a procedure that resolves the evolutionary history of all genes in a large group of species. we apply our procedure to seventeen fungal genomes to create a genome-wide catalogue of gene trees that determine precise orthology and paralogy relations across these species. we show that gene duplication and loss is highly constrained by the functional properties and interacting partners of genes. in particular, stress-related genes exhibit many duplications and losses, whereas growth-related genes show selection against such changes. whole-genome duplication circumvents this constraint and relaxes the dichotomy, resulting in an expanded functional scope of gene duplication. by characterizing the functional fate of duplicate genes we show that duplicated genes rarely diverge with respect to biochemical function, but typically diverge with respect to regulatory control. surprisingly, paralogous modules of genes rarely arise, even after whole-genome duplication. rather, gene duplication may drive the modularization of functional networks through specialization, thereby disentangling cellular systems.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsidentifying interaction sites in proteins provides important clues to the function of a protein and is becoming increasingly relevant in topics such as systems biology and drug discovery. although there are numerous papers on the prediction of interaction sites using information derived from structure, there are only a few case reports on the prediction of interaction residues based solely on protein sequence. here, a sliding window approach is combined with the random forests method to predict protein interaction sites using (i) a combination of sequence- and structure-derived parameters and (ii) sequence information alone. for sequence-based prediction we achieved a precision of 84\\\\% with a 26\\\\% recall and an f-measure of 40\\\\%. when combined with structural information, the prediction performance increases to a precision of 76\\\\% and a recall of 38\\\\% with an f-measure of 51\\\\%. we also present an attempt to rationalize the sliding window size and demonstrate that a nine-residue window is the most suitable for predictor construction. finally, we demonstrate the applicability of our prediction methods by modeling the {ras-raf} complex using predicted interaction sites as target binding interfaces. our results suggest that it is possible to predict protein interaction sites with quite a high accuracy using only sequence information.transcriptional modules of coregulated genes play a key role in regulatory networks. comparative studies show that modules of coexpressed genes are conserved across taxa. however, little is known about the mechanisms underlying the evolution of module regulation. here, we explore the evolution of cis-regulatory programs associated with conserved modules by integrating expression profiles for two yeast species and sequence data for a total of 17 fungal genomes. we show that although the cis-elements accompanying certain conserved modules are strictly conserved, those of other conserved modules are remarkably diverged. in particular, we infer the evolutionary history of the regulatory program governing ribosomal modules. we show how a cis-element emerged concurrently in dozens of promoters of ribosomal protein genes, followed by the loss of a more ancient cis-element. we suggest that this formation of an intermediate redundant regulatory program allows conserved transcriptional modules to gradually switch from one regulatory mechanism to another while maintaining their functionality. our work provides a general framework for the study of the dynamics of promoter evolution at the level of transcriptional modules and may help in understanding the evolvability and increased redundancy of transcriptional regulation in higher organisms.synonymous mutations - sometimes called \\'silent\\' mutations - are now widely acknowledged to be able to cause changes in protein expression, conformation and function. the recent increase in knowledge about the association of genetic variants with disease, particularly through genome-wide association studies, has revealed a substantial contribution of synonymous {snps} to human disease risk and other complex traits. here we review current understanding of the extent to which synonymous mutations influence disease, the various molecular mechanisms that underlie these effects and the implications for future research and biomedical applications.a catalogue of molecular aberrations that cause ovarian cancer is critical for developing and deploying therapies that will improve patients\\' lives. the cancer genome atlas project has analysed messenger {rna} expression, {microrna} expression, promoter methylation and {dna} copy number in 489 high-grade serous ovarian adenocarcinomas and the {dna} sequences of exons from coding genes in 316 of these tumours. here we report that high-grade serous ovarian cancer is characterized by {tp53} mutations in almost all tumours (96\\\\%); low prevalence but statistically recurrent somatic mutations in nine further genes including {nf1}, {brca1}, {brca2}, {rb1} and {cdk12}; 113 significant focal {dna} copy number aberrations; and promoter methylation events involving 168 genes. analyses delineated four ovarian cancer transcriptional subtypes, three {microrna} subtypes, four promoter methylation subtypes and a transcriptional signature associated with survival duration, and shed new light on the impact that tumours with {brca1}/2 ({brca1} or {brca2}) and {ccne1} aberrations have on survival. pathway analyses suggested that homologous recombination is defective in about half of the tumours analysed, and that {notch} and {foxm1} signalling are involved in serous ovarian cancer pathophysiology.recent genome sequencing studies have shown that the somatic mutations that drive cancer development are distributed across a large number of genes. this mutational heterogeneity complicates efforts to distinguish functional mutations from sporadic, passenger mutations. since cancer mutations are hypothesized to target a relatively small number of cellular signaling and regulatory pathways, a common practice is to assess whether known pathways are enriched for mutated genes. we introduce an alternative approach that examines mutated genes in the context of a genome-scale gene interaction network. we present a computationally efficient strategy for de novo identification of subnetworks in an interaction network that are mutated in a statistically significant number of patients. this framework includes two major components. first, we use a diffusion process on the interaction network to define a local neighborhood of \"influence\" for each mutated gene in the network. second, we derive a two-stage multiple hypothesis test to bound the false discovery rate ({fdr}) associated with the identified subnetworks. we test these algorithms on a large human protein-protein interaction network using somatic mutation data from glioblastoma and lung adenocarcinoma samples. we successfully recover pathways that are known to be important in these cancers and also identify additional pathways that have been implicated in other cancers but not previously reported as mutated in these samples. we anticipate that our approach will find increasing use as cancer genome studies increase in size and scope.the dissection of complex biological systems is a challenging task, made difficult by the size of the underlying molecular network and the heterogeneous nature of the control mechanisms involved. novel high-throughput techniques are generating massive data sets on various aspects of such systems. here, we perform analysis of a highly diverse collection of genomewide data sets, including gene expression, protein interactions, growth phenotype data, and transcription factor binding, to reveal the modular organization of the yeast system. by integrating experimental data of heterogeneous sources and types, we are able to perform analysis on a much broader scope than previous studies. at the core of our methodology is the ability to identify modules, namely, groups of genes with statistically significant correlated behavior across diverse data sources. numerous biological processes are revealed through these modules, which also obey global hierarchical organization. we use the identified modules to study the yeast transcriptional network and predict the function of >800 uncharacterized genes. our analysis framework, samba ({statistical-algorithmic} method for bicluster analysis), enables the processing of current and future sources of biological information and is readily extendable to experimental techniques and higher organisms.tandem repeats of {dna} that contain transcription factor ({tf}) binding sites could serve as decoys, competitively binding to {tfs} and affecting target gene expression. using a synthetic system in budding yeast, we demonstrate that repeated decoy sites inhibit gene expression by sequestering a transcriptional activator and converting the graded dose–response of target promoters to a sharper, sigmoidal-like response. on the basis of both modeling and chromatin immunoprecipitation measurements, we attribute the altered response to {tf} binding decoy sites more tightly than promoter binding sites. tight {tf} binding to arrays of contiguous repeated decoy sites only occurs when the arrays are mostly unoccupied. finally, we show that the altered sigmoidal-like response can convert the graded response of a transcriptional positive-feedback loop to a bimodal response. together, these results show how changing numbers of repeated {tf} binding sites lead to qualitative changes in behavior and raise new questions about the stability of {tf}/promoter binding.the {chip}-seq technique enables genome-wide mapping of in vivo {protein-dna} interactions and chromatin states. current analytical approaches for {chip}-seq analysis are largely geared towards single-sample investigations, and have limited applicability in comparative settings that aim to identify combinatorial patterns of enrichment across multiple datasets. we describe a novel probabilistic method, {jmosaics}, for jointly analyzing multiple {chip}-seq datasets. we demonstrate its usefulness with a wide range of data-driven computational experiments and with a case study of histone modifications on {gata1}-occupied segments during erythroid differentiation. {jmosaics} is open source software and can be downloaded from bioconductor 1.a crucial question in the field of gene regulation is whether the location at which a transcription factor binds influences its effectiveness or the mechanism by which it regulates transcription. comprehensive transcription factor binding maps are needed to address these issues, and genome-wide mapping is now possible thanks to the technological advances of {chip}–chip and {chip}–seq. this review discusses how recent genomic profiling of transcription factors gives insight into how binding specificity is achieved and what features of chromatin influence the ability of transcription factors to interact with the genome. it also suggests future experiments that may further our understanding of the causes and consequences of transcription factor–genome interactions.{background}:the regulatory map of a genome consists of the binding sites for proteins that determine the transcription of nearby genes. an initial regulatory map for s. cerevisiae was recently published using six motif discovery programs to analyze genome-wide chromatin immunoprecipitation data for 203 transcription factors. the programs were used to identify sequence motifs that were likely to correspond to the {dna}-binding specificity of the immunoprecipitated proteins. we report improved versions of two conservation-based motif discovery algorithms, {phylocon} and converge. using these programs, we create a refined regulatory map for s. cerevisiae by reanalyzing the same chromatin immunoprecipitation {data.results}:applying the same conservative criteria that were applied in the original study, we find that {phylocon} and converge each separately discover more known specificities than the combination of all six programs in the previous study. combining the results of {phylocon} and converge, we discover significant sequence motifs for 36 transcription factors that were previously missed. the new set of motifs identifies 636 more regulatory interactions than the previous one. the new network contains 28\\\\% more regulatory interactions among transcription factors, evidence of greater cross-talk between {regulators.conclusion}:combining two complementary computational strategies for conservation-based motif discovery improves the ability to identify the specificity of transcriptional regulators from genome-wide chromatin immunoprecipitation data. the increased sensitivity of these methods significantly expands the map of yeast regulatory sites without the need to alter any of the thresholds for statistical significance. the new map of regulatory sites reveals a more elaborate and complex view of the yeast genetic regulatory network than was observed previously.chromatin immunoprecipitation ({chip}) allows specific {protein–dna} interactions to be isolated. combining {chip} with high-throughput sequencing reveals the {dna} sequence involved in these interactions. here, we describe how to perform {chip}-seq starting with whole tissues or cell lines, and ending with millions of short sequencing tags that can be aligned to the reference genome of the species under investigation. we also outline additional procedures to recover {chip}-chip libraries for {chip}-seq and discuss contemporary issues in data analysis.genome-wide measurements of {protein-dna} interactions and transcriptomes are increasingly done by deep {dna} sequencing methods ({chip}-seq and {rna}-seq). the power and richness of these counting-based measurements comes at the cost of routinely handling tens to hundreds of millions of reads. whereas early adopters necessarily developed their own custom computer code to analyze the first {chip}-seq and {rna}-seq datasets, a new generation of more sophisticated algorithms and software tools are emerging to assist in the analysis phase of these projects. here we describe the multilayered analyses of {chip}-seq and {rna}-seq datasets, discuss the software packages currently available to perform tasks at each layer and describe some upcoming challenges and features for future analysis tools. we also discuss how software choices and uses are affected by specific aspects of the underlying biology and data structure, including genome size, positional clustering of transcription factor binding sites, transcript discovery and expression quantification.cluster analysis is used in many disciplines to group objects according to a defined measure of distance. numerous algorithms exist, some based on the analysis of the local density of data points, and others on predefined probability distributions. rodriguez and laio devised a method in which the cluster centers are recognized as local density maxima that are far away from any points of higher density. the algorithm depends only on the relative densities rather than their absolute values. the authors tested the method on a series of data sets, and its performance compared favorably to that of established techniques.motivation: a fundamental task in systems biology is the identification of groups of genes that are involved in the cellular response to particular signals. at its simplest level, this often reduces to identifying biological quantities ({mrna} abundance, enzyme concentrations, etc.) which are differentially expressed in two different conditions. popular approaches involve using t-test statistics, based on modelling the data as arising from a mixture distribution. a common assumption of these approaches is that the data are independent and identically distributed; however, biological quantities are usually related through a complex (weighted) network of interactions, and often the more pertinent question is which subnetworks are differentially expressed, rather than which genes. furthermore, in many interesting cases (such as high-throughput proteomics and metabolomics), only very partial observations are available, resulting in the need for efficient imputation techniques.  results: we introduce mixture model on graphs ({mmg}), a novel probabilistic model to identify differentially expressed submodules of biological networks and pathways. the method can easily incorporate information about weights in the network, is robust against missing data and can be easily generalized to directed networks. we propose an efficient sampling strategy to infer posterior probabilities of differential expression, as well as posterior probabilities over the model parameters. we assess our method on artificial data demonstrating significant improvements over standard mixture model clustering. analysis of our model results on quantitative high-throughput proteomic data leads to the identification of biologically significant subnetworks, as well as the prediction of the expression level of a number of enzymes, some of which are then verified experimentally.  availability: {matlab} code is available from http://www.dcs.shef.ac.uk/\\\\~{}guido/software.html  contact: guido@dcs.shef.ac.uk  supplementary information: supplementary data are available at bioinformatics online. 10.1093/bioinformatics/btn066genomic information on tumors from 50 cancer types cataloged by the international cancer genome consortium ({icgc}) shows that only a few well-studied driver genes are frequently mutated, in contrast to many infrequently mutated genes that may also contribute to tumor biology. hence there has been large interest in developing pathway and network analysis methods that group genes and illuminate the processes involved. we provide an overview of these analysis techniques and show where they guide mechanistic and translational investigations.chromatin state variation at gene regulatory elements is abundant across individuals, yet we understand little about the genetic basis of this variability. here, we profiled several histone modifications, the transcription factor ({tf}) {pu}.1, {rna} polymerase {ii}, and gene expression in lymphoblastoid cell lines from 47 whole-genome sequenced individuals. we observed that distinct cis-regulatory elements exhibit coordinated chromatin variation across individuals in\\xa0the form of variable chromatin modules ({vcms}) at\\xa0{sub-mb} scale. {vcms} were associated with thousands of genes and preferentially cluster within chromosomal contact domains. we mapped strong proximal and weak, yet more ubiquitous, distal-acting chromatin quantitative trait loci ({cqtl}) that frequently explain this variation. {cqtls} were associated with molecular activity at clusters of cis-regulatory elements and mapped preferentially within {tf}-bound regions. we propose that local, sequence-independent chromatin variation emerges as a result of genetic perturbations in cooperative interactions between cis-regulatory elements that are located within the same genomic domain.cancer cells within individual tumors often exist in distinct phenotypic states that differ in functional attributes. while cancer cell populations typically display distinctive equilibria in the proportion of cells in various states, the mechanisms by which this occurs are poorly understood. here, we study the dynamics of phenotypic proportions in human breast cancer cell lines. we show that subpopulations of cells purified for a given phenotypic state return towards equilibrium proportions over time. these observations can be explained by a markov model in which cells transition stochastically between states. a prediction of this model is that, given certain conditions, any subpopulation of cells will return to\\xa0equilibrium phenotypic proportions over time. a second prediction is that breast cancer stem-like cells arise de novo from non-stem-like cells. these findings contribute to our understanding of cancer heterogeneity and reveal how stochasticity in single-cell behaviors promotes phenotypic equilibrium in populations of cancer cells. º cancer cell populations interconvert between phenotypic states º cell-state transitions can be described with a stochastic markov model º markov model predicts convergence to equilibrium phenotypic proportions º cancer stem cells can arise de novo from noncancer stem cellsan international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsrecent experiments revealed large-scale differences in the transcription programs of related species, yet little is known about the genetic basis underlying the evolution of gene expression and its contribution to phenotypic diversity. here we describe a large-scale modulation of the yeast transcription program that is connected to the emergence of the capacity for rapid anaerobic growth. genes coding for mitochondrial and cytoplasmic ribosomal proteins display a strongly correlated expression pattern in candida albicans, but this correlation is lost in the fermentative yeast saccharomyces cerevisiae. we provide evidence that this change in gene expression is connected to the loss of a specific cis-regulatory element from dozens of genes following the apparent whole-genome duplication event. our results shed new light on the genetic mechanisms underlying the large-scale evolution of transcriptional networks.regulatory circuits controlling gene expression constantly rewire to adapt to environmental stimuli, differentiation cues, and disease. we review our current understanding of the temporal dynamics of gene expression in eukaryotes and prokaryotes and the molecular mechanisms that shape them. we delineate several prototypical temporal patterns, including  ” impulse” (or single-pulse) patterns in response to transient environmental stimuli, sustained (or state-transitioning) patterns in response to developmental cues, and oscillating patterns. we focus on impulse responses and their higher-order temporal organization in regulons and cascades and describe how core protein circuits and cis-regulatory sequences in promoters integrate with chromatin architecture to generate these responses.gene duplication is an important mechanism in the evolution of protein interaction networks. duplications are followed by the gain and loss of interactions, rewiring the network at some unknown rate. because rewiring is likely to change the distribution of network motifs within the duplicated interaction set, it should be possible to study network rewiring by tracking the evolution of these motifs. we have developed a mathematical framework that, together with duplication data from comparative genomic and proteomic studies, allows us to infer the connectivity of the preduplication network and the changes in connectivity over time. we focused on the whole-genome duplication ({wgd}) event in saccharomyces cerevisiae. the model allowed us to predict the frequency of intergene interaction before {wgd} and the post duplication probabilities of interaction gain and loss. we find that the predicted frequency of self-interactions in the preduplication network is significantly higher than that observed in today\\'s network. this could suggest a structural difference between the modern and ancestral networks, preferential addition or retention of interactions between ohnologs, or selective pressure to preserve duplicates of self-interacting proteins.although experimental and theoretical efforts have been applied to globally map genetic interactions, we still do not understand how gene-gene interactions arise from the operation of biomolecular networks. to bridge the gap between empirical and computational studies, we i, quantitatively measured genetic interactions between ∼185,000 metabolic gene pairs in saccharomyces cerevisiae, ii, superposed the data on a detailed systems biology model of metabolism and iii, introduced a machine-learning method to reconcile empirical interaction data with model predictions. we systematically investigated the relative impacts of functional modularity and metabolic flux coupling on the distribution of negative and positive genetic interactions. we also provide a mechanistic explanation for the link between the degree of genetic interaction, pleiotropy and gene dispensability. last, we show the feasibility of automated metabolic model refinement by correcting misannotations in {nad} biosynthesis and confirming them by in vivo experiments.the embryonic stem cell-specific cell cycle-regulating ({escc}) family of {micrornas} ({mirnas}) enhances reprogramming of mouse embryonic fibroblasts to induced pluripotent stem cells. here we show that the human {escc} {mirna} orthologs {hsa-mir}-302b and {hsa-mir}-372 promote human somatic cell reprogramming. furthermore, these {mirnas} repress multiple target genes, with downregulation of individual targets only partially recapitulating the total {mirna} effects. these targets regulate various cellular processes, including cell cycle, epithelial-mesenchymal transition ({emt}), epigenetic regulation and vesicular transport. {escc} {mirnas} have a known role in regulating the unique embryonic stem cell cycle. we show that they also increase the kinetics of mesenchymal-epithelial transition during reprogramming and block {tgfβ}-induced {emt} of human epithelial cells. these results demonstrate that the {escc} {mirnas} promote dedifferentiation by acting on multiple downstream pathways. we propose that individual {mirnas} generally act through numerous pathways that synergize to regulate and enforce cell fate decisions.we present an integrated approach to identify genetic mechanisms that control self-renewal in mouse embryonic stem cells. we use short hairpin {rna} ({shrna}) loss-of-function techniques to downregulate a set of gene products whose expression patterns suggest self-renewal regulatory functions. we focus on transcriptional regulators and identify seven genes for which {shrna}-mediated depletion negatively affects self-renewal, including four genes with previously unrecognized roles in self-renewal. perturbations of these gene products are combined with dynamic, global analyses of gene expression. our studies suggest specific biological roles for these molecules and reveal the complexity of cell fate regulation in embryonic stem cells.characterization of interspecies differences in gene regulation is crucial for understanding the molecular basis of both phenotypic diversity and evolution. by means of chromatin immunoprecipitation and {dna} microarray analysis, the divergence in the binding sites of the pseudohyphal regulators ste12 and tec1 was determined in the yeasts saccharomyces cerevisiae, s. mikatae, and s. bayanus under pseudohyphal conditions. we have shown that most of these sites have diverged across these species, far exceeding the interspecies variation in orthologous genes. a group of ste12 targets was shown to be bound only in s. mikatae and s. bayanus under pseudohyphal conditions. many of these genes are targets of ste12 during mating in s. cerevisiae, indicating that specialization between the two pathways has occurred in this species. transcription factor binding sites have therefore diverged substantially faster than ortholog content. thus, gene regulation resulting from transcription factor binding is likely to be a major cause of divergence between related species.covalent post-translational modifications ({ptms}) provide vast indexing potential and expanded protein use. the \\'histone code\\' hypothesis has inspired rapid advances throughout chromatin biology, and has recently been tapped for its relevance to non-histone proteins. comprehensive analyses suggest that rather than constituting a general code, the covalent modifications of proteins (including histones) provide surfaces that are recognized by effectors that can give rise to intricate interactions and downstream events. these are reminiscent of other regulatory cascades in transcription and cell signalling.{dna} methylation is a critical epigenetic regulator in mammalian development. here, we present a whole-genome comparative view of {dna} methylation using bisulfite sequencing of three cultured cell types representing progressive stages of differentiation: human embryonic stem cells ({hescs}), a fibroblastic differentiated derivative of the {hescs}, and neonatal fibroblasts. as a reference, we compared our maps with a methylome map of a fully differentiated adult cell type, mature peripheral blood mononuclear cells (monocytes). we observed many notable common and cell-type-specific features among all cell types. promoter hypomethylation (both {cg} and {ca}) and higher levels of gene body methylation were positively correlated with transcription in all cell types. exons were more highly methylated than introns, and sharp transitions of methylation occurred at exon-intron boundaries, suggesting a role for differential methylation in transcript splicing. developmental stage was reflected in both the level of global methylation and extent of {non-cpg} methylation, with {hesc} highest, fibroblasts intermediate, and monocytes lowest. differentiation-associated differential methylation profiles were observed for developmentally regulated genes, including the {hox} clusters, other homeobox transcription factors, and pluripotence-associated genes such as {pou5f1}, {tcf3}, and {klf4}. our results highlight the value of high-resolution methylation maps, in conjunction with other systems-level analyses, for investigation of previously undetectable developmental regulatory mechanisms.identifying variations in {dna} that increase susceptibility to disease is one of the primary aims of genetic studies using a forward genetics approach. however, identification of disease-susceptibility genes by means of such studies provides limited functional information on how genes lead to disease. in fact, in most cases there is an absence of functional information altogether, preventing a definitive identification of the susceptibility gene or genes. here we develop an alternative to the classic forward genetics approach for dissecting complex disease traits where, instead of identifying susceptibility genes directly affected by variations in {dna}, we identify gene networks that are perturbed by susceptibility loci and that in turn lead to disease. application of this method to liver and adipose gene expression data generated from a segregating mouse population results in the identification of a macrophage-enriched network supported as having a causal relationship with disease traits associated with metabolic syndrome. three genes in this network, lipoprotein lipase (lpl), lactamase beta (lactb) and protein phosphatase 1-like (ppm1l), are validated as previously unknown obesity genes, strengthening the association between this network and metabolic disease traits. our analysis provides direct experimental support that complex traits such as obesity are emergent properties of molecular networks that are modulated by complex genetic loci and environmental factors.yeast cells sense the amount and quality of external nutrients through multiple interconnected signaling networks, which allow them to adjust their metabolism, transcriptional profile and developmental program to adapt readily and appropriately to changing nutritional states. we present our current understanding of the nutritional sensing networks yeast cells rely on for perceiving the nutritional landscape, with particular emphasis on those sensitive to carbon and nitrogen sources. we describe the means by which these networks inform the cell\\'s decision among the different developmental programs available to them—growth, quiescence, filamentous development, or meiosis/sporulation. we conclude that the highly interconnected signaling networks provide the cell with a highly nuanced view of the environment and that the cell can interpret that information through a sophisticated calculus to achieve optimum responses to any nutritional condition.the pluripotency control regions ({plucrs}) are defined as genomic regions that are bound by {pou5f1}, {sox2}, and {nanog} in vivo. we utilized a high-throughput binding assay to record more than 270,000 different {dna}/protein binding measurements along incrementally tiled windows of {dna} within these {plucrs}. this high-resolution binding map is then used to systematically define the context of {pou} factor binding, and reveals patterns of cooperativity and competition in the pluripotency network. the most prominent pattern is a pervasive binding competition between {pou5f1} and the forkhead transcription factors. like many transcription factors, {pou5f1} is co-expressed with a paralog, {pou2f1}, that shares an apparently identical binding specificity. by analyzing thousands of binding measurements, we discover context effects that discriminate {pou2f1} from {pou5f1} binding. proximal {nanog} binding promotes {pou5f1} binding, whereas nearby {sox2} binding favors {pou2f1}. we demonstrate by cross-species comparison and by chromatin immunoprecipitation ({chip}) that the contextual sequence determinants learned in vitro are sufficient to predict {pou2f1} binding in vivo.\\'{remodiscovery}\\' is an intuitive algorithm to correlate regulatory programs with regulators and corresponding motifs to a set of co-expressed genes. it exploits in a concurrent way three independent data sources: {chip}-chip data, motif information and gene expression profiles. when compared to published module discovery algorithms, {remodiscovery} is fast and easily tunable. we evaluated our method on yeast data, where it was shown to generate biologically meaningful findings and allowed the prediction of potential novel roles of transcriptional regulators.near the 5′ end of most eukaryotic genes, nucleosomes form highly regular arrays that begin at canonical distances from the transcriptional start site. determinants of this and other aspects of genomic nucleosome organization have been ascribed to statistical positioning, intrinsically {dna}-encoded positioning, or some aspect of transcription initiation. here, we provide evidence for a different explanation. biochemical reconstitution of proper nucleosome positioning, spacing, and occupancy levels was achieved across the 5′ ends of most yeast genes by adenosine triphosphate–dependent trans-acting factors. these transcription-independent activities override {dna}-intrinsic positioning and maintain uniform spacing at the 5′ ends of genes even at low nucleosome densities. thus, an active, nonstatistical nucleosome packing mechanism creates chromatin organizing centers at the 5′ ends of genes where important regulatory elements reside.deciphering the signaling networks that underlie normal and disease processes remains a major challenge. here, we report the discovery of signaling components involved in the toll-like receptor ({tlr}) response of immune dendritic cells ({dcs}), including a previously unkown pathway shared across mammalian antiviral responses. by combining transcriptional profiling, genetic and small-molecule perturbations, and phosphoproteomics, we uncover 35 signaling regulators, including 16 known regulators, involved in {tlr} signaling. in particular, we find that polo-like kinases (plk) 2 and 4 are essential components of antiviral pathways in vitro and in vivo and activate a signaling branch involving a dozen proteins, among which is tnfaip2, a gene associated with autoimmune diseases but whose role was unknown. our study illustrates the power of combining systematic measurements and perturbations to elucidate complex signaling circuits and discover potential therapeutic targets. copyright {\\\\copyright} 2011 elsevier inc. all rights reserved.in contrast to batch cultivation, chemostat cultivation allows the identification of carbon source responses without interference by carbon-catabolite repression, accumulation of toxic products, and differences in specific growth rate. this study focuses on the yeast saccharomyces cerevisiae, grown in aerobic, carbon-limited chemostat cultures. genome-wide transcript levels and in vivo fluxes were compared for growth on two sugars, glucose and maltose, and for two c2-compounds, ethanol and acetate. in contrast to previous reports on batch cultures, few genes (180 genes) responded to changes of the carbon source by a changed transcript level. very few transcript levels were changed when glucose as the growth-limiting nutrient was compared with maltose (33 transcripts), or when acetate was compared with ethanol (16 transcripts). although metabolic flux analysis using a stoichiometric model revealed major changes in the central carbon metabolism, only 117 genes exhibited a significantly different transcript level when sugars and c2-compounds were provided as the growth-limiting nutrient. despite the extensive knowledge on carbon source regulation in yeast, many of the carbon source-responsive genes encoded proteins with unknown or incompletely characterized biological functions. in silico promoter analysis of carbon source-responsive genes confirmed the involvement of several known transcriptional regulators and suggested the involvement of additional regulators. transcripts involved in the glyoxylate cycle and gluconeogenesis showed a good correlation with in vivo fluxes. this correlation was, however, not observed for other important pathways, including the pentose-phosphate pathway, tricarboxylic acid cycle, and, in particular, glycolysis. these results indicate that in vivo fluxes in the central carbon metabolism of s. cerevisiae grown in steadystate, carbon-limited chemostat cultures are controlled to a large extent via post-transcriptional mechanisms.the simultaneous utilization of efficient respiration and inefficient fermentation even in the presence of abundant oxygen is a puzzling phenomenon commonly observed in bacteria, yeasts, and cancer cells. despite extensive research, the biochemical basis for this phenomenon remains obscure. we hypothesize that the outcome of a competition for membrane space between glucose transporters and respiratory chain (which we refer to as economics of membrane occupancy) proteins influences respiration and fermentation. by incorporating a sole constraint based on this concept in the genome-scale metabolic model of escherichia coli, we were able to simulate respiro-fermentation. further analysis of the impact of this constraint revealed differential utilization of the cytochromes and faster glucose uptake under anaerobic conditions than under aerobic conditions. based on these simulations, we propose that bacterial cells manage the composition of their cytoplasmic membrane to maintain optimal {atp} production by switching between oxidative and substrate-level phosphorylation. these results suggest that the membrane occupancy constraint may be a fundamental governing constraint of cellular metabolism and physiology, and establishes a direct link between cell morphology and physiology.predicting the metabolic state of an organism after a gene knockout is a challenging task, because the regulatory system governs a series of transient metabolic changes that converge to a steady-state condition. regulatory on/off minimization ({room}) is a constraint-based algorithm for predicting the metabolic steady state after gene knockouts. it aims to minimize the number of significant flux changes (hence on/off) with respect to the wild type. {room} is shown to accurately predict steady-state metabolic fluxes that maintain flux linearity, in agreement with experimental flux measurements, and to correctly identify short alternative pathways used for rerouting metabolic flux in response to gene knockouts. {room}\\'s growth rate and flux predictions are compared with previously suggested algorithms, minimization of metabolic adjustment, and flux balance analysis ({fba}). we find that minimization of metabolic adjustment provides accurate predictions for the initial transient growth rates observed during the early postperturbation state, whereas {room} and {fba} more successfully predict final higher steady-state growth rates. although {fba} explicitly maximizes the growth rate, {room} does not, and only implicitly favors flux distributions having high growth rates. this indicates that, even though the cell has not evolved to cope with specific mutations, regulatory mechanisms aiming to minimize flux changes after genetic perturbations may indeed work to this effect. further work is needed to identify metrics that characterize the complete trajectory from the initial to the final metabolic steady states after genetic perturbations.systems analysis of metabolic and growth functions in microbial organisms is rapidly developing and maturing. such studies are enabled by reconstruction, at the genomic scale, of the biochemical reaction networks that underlie cellular processes. the network reconstruction process is organism specific and is based on an annotated genome sequence, high-throughput network-wide data sets and bibliomic data on the detailed properties of individual network components. here we describe the process that is currently used to achieve comprehensive network reconstructions and discuss how these reconstructions are curated and validated. this review should aid the growing number of researchers who are carrying out reconstructions for particular target organisms.summary: protein–protein interaction networks are an important component of modern systems biology. yet, comparatively few efforts have been made to tailor their topology to the actual cellular condition being studied. here, we present a network construction method that exploits expression data at the transcript-level and thus reveals alterations in protein connectivity not only caused by differential gene expression but also by alternative splicing. we achieved this by establishing a direct correspondence between individual protein interactions and underlying domain interactions in a complete but condition-unspecific protein interaction network. this knowledge was then used to infer the condition-specific presence of interactions from the dominant protein isoforms. when we compared contextualized interaction networks of matched normal and tumor samples in breast cancer, our transcript-based construction identified more significant alterations that affected proteins associated with cancerogenesis than a method that only uses gene expression data. the approach is provided as the user-friendly tool {ppixpress}.the relationships between the levels of transcripts and the levels of the proteins they encode have not been examined comprehensively in mammals, although previous work in plants and yeast suggest a surprisingly modest correlation. we have examined this issue using a genetic approach in which natural variations were used to perturb both transcript levels and protein levels among inbred strains of mice. we quantified over 5,000 peptides and over 22,000 transcripts in livers of 97 inbred and recombinant inbred strains and focused on the 7,185 most heritable transcripts and 486 most reliable proteins. the transcript levels were quantified by microarray analysis in three replicates and the proteins were quantified by liquid {chromatography–mass} spectrometry using o(18)-reference-based isotope labeling approach. we show that the levels of transcripts and proteins correlate significantly for only about half of the genes tested, with an average correlation of 0.27, and the correlations of transcripts and proteins varied depending on the cellular location and biological function of the gene. we examined technical and biological factors that could contribute to the modest correlation. for example, differential splicing clearly affects the analyses for certain genes; but, based on deep sequencing, this does not substantially contribute to the overall estimate of the correlation. we also employed genome-wide association analyses to map loci controlling both transcript and protein levels. surprisingly, little overlap was observed between the protein- and transcript-mapped loci. we have typed numerous clinically relevant traits among the strains, including adiposity, lipoprotein levels, and tissue parameters. using correlation analysis, we found that a low number of clinical trait relationships are preserved between the protein and {mrna} gene products and that the majority of such relationships are specific to either the protein levels or transcript levels. surprisingly, transcript levels were more strongly correlated with clinical traits than protein levels. in light of the widespread use of high-throughput technologies in both clinical and basic research, the results presented have practical as well as basic implications. an old dogma in biology states that, in every cell, the flow of biological information is from {dna} to {rna} to proteins and that the latter act as a working force to determine the organism\\'s phenotype. this model predicts that changes in {dna} that affect the clinical phenotype should also similarly change the cellular levels of {rna} and protein levels. in this report, we test this prediction by looking at the concordance between {dna} variation in population of mouse inbred strains, the {rna} and protein variation in the liver tissue of these mice, and variation in metabolic phenotypes. we show that the relationship between various biological traits is not simple and that there is relatively little concordance of {rna} levels and the corresponding protein levels in response to {dna} perturbations. in addition, we also find that, surprisingly, metabolic traits correlate better to {rna} levels than to protein levels. in light of current efforts in searching for the molecular bases of disease susceptibility in humans, our findings highlight the complexity of information flow that underlies clinical outcomes.understanding the effect of genetic sequence variation on phenotype is a major challenge that lies at the heart of genetics. we developed {golph} ({genomic} linkage to {phenotype}), a statistical method to identify genetic interactions, and used it to characterize the landscape of genetic interactions between gene expression quantitative trait loci. our results reveal that allele-specific interactions, in which a gene only exerts an influence on the phenotype in the presence of a particular allele at the primary locus, are widespread and that genetic interactions are predominantly nonadditive. the data portray a complex picture in which interacting loci influence the expression of modules of coexpressed genes involved in coherent biological processes and pathways. we show that genetic variation at a single gene can have a major impact on the global transcriptional response, altering interactions between genes through shutdown or activation of pathways. thus, different cellular states occur not only in response to the external environment but also result from intrinsic genetic variation.transcription factor ({tf}) regulation is often post-translational. {tf} modifications such as reversible phosphorylation and missense mutations, which can act independent of {tf} expression level, are overlooked by differential expression analysis. using bovine piedmontese myostatin mutants as proof-of-concept, we propose a new algorithm that correctly identifies the gene containing the causal mutation from microarray data alone. the myostatin mutation releases the brakes on piedmontese muscle growth by translating a dysfunctional protein. compared to a less muscular non-mutant breed we find that myostatin is not differentially expressed at any of ten developmental time points. despite this challenge, the algorithm identifies the myostatin \\'smoking gun\\' through a coordinated, simultaneous, weighted integration of three sources of microarray information: transcript abundance, differential expression, and differential wiring. by asking the novel question \"which regulator is cumulatively most differentially wired to the abundant most differentially expressed genes?\" it yields the correct answer, \"myostatin\". our new approach identifies causal regulatory changes by globally contrasting co-expression network dynamics. the entirely data-driven \\'weighting\\' procedure emphasises regulatory movement relative to the phenotypically relevant part of the network. in contrast to other published methods that compare co-expression networks, significance testing is not used to eliminate connections.by analyzing, in parallel, large literature-derived and high-throughput experimental datasets we investigate genes harboring human inherited disease mutations in the context of molecular interaction networks. our results demonstrate that network properties influence the likelihood and phenotypic consequences of disease mutations. genes with intermediate connectivities have the highest probability of harboring germ-line disease mutations, suggesting that disease genes tend to occupy an intermediate niche in terms of their physiological and cellular importance. our analysis of tissue expression profiles supports this view. we show that disease mutations are less likely to occur in essential genes compared with all human genes. disease genes display significant functional clustering in the analyzed molecular network. for about one-third of known disorders with two or more associated genes we find physical clusters of genes with the same phenotype. these clusters are likely to represent disorder-specific functional modules and suggest a framework for identifying yet-undiscovered disease genes.motivation: the power of a microarray experiment derives from the identification of genes differentially regulated across biological conditions. to date, differential regulation is most often taken to mean differential expression, and a number of useful methods for identifying differentially expressed ({de}) genes or gene sets are available. however, such methods are not able to identify many relevant classes of differentially regulated genes. one important example concerns differentially co-expressed ({dc}) genes.eukaryotic organisms age, yet detrimental age-associated traits are not passed on to progeny. how life span is reset from one generation to the next is not known. we show that in budding yeast resetting of life span occurs during gametogenesis. gametes (spores) generated by aged cells show the same replicative potential as gametes generated by young cells. age-associated damage is no longer detectable in mature gametes. furthermore, transient induction of a transcription factor essential for later stages of gametogenesis extends the replicative life span of aged cells. our results indicate that gamete formation brings about rejuvenation by eliminating age-induced cellular damage.the availability of genomes of many closely related bacteria with diverse metabolic capabilities offers the possibility of tracing metabolic evolution on a phylogeny relating the genomes to understand the evolutionary processes and constraints that affect the evolution of metabolic networks. using simple (independent loss/gain of reactions) or complex (incorporating dependencies among reactions) stochastic models of metabolic evolution, it is possible to study how metabolic networks evolve over time. here, we describe a model that takes the reaction neighborhood into account when modeling metabolic evolution. the model also allows estimation of the strength of the neighborhood effect during the course of evolution. we present gibbs samplers for sampling networks at the internal node of a phylogeny and for estimating the parameters of evolution over a phylogeny without exploring the whole search space by iteratively sampling from the conditional distributions of the internal networks and parameters. the samplers are used to estimate the parameters of evolution of metabolic networks of bacteria in the genus pseudomonas and to infer the metabolic networks of the ancestral pseudomonads. the results suggest that pathway maps that are conserved across the pseudomonas phylogeny have a stronger neighborhood structure than those which have a variable distribution of reactions across the phylogeny, and that some pseudomonas lineages are going through genome reduction resulting in the loss of a number of reactions from their metabolic networks.differentiation of human embryonic stem cells ({hescs}) provides a unique opportunity to study the regulatory mechanisms that facilitate cellular transitions in a human context. to that end, we performed comprehensive transcriptional and epigenetic profiling of populations derived through directed differentiation of {hescs} representing each of the three embryonic germ layers. integration of whole-genome bisulfite sequencing, chromatin immunoprecipitation sequencing, and {rna} sequencing reveals unique events associated with specification toward each lineage. lineage-specific dynamic alterations in {dna} methylation and {h3k4me1} are evident at putative distal regulatory elements that are frequently bound by pluripotency factors in the undifferentiated {hescs}. in addition, we identified germ-layer-specific {h3k27me3} enrichment at sites exhibiting high {dna} methylation in the undifferentiated state. a better understanding of these initial specification events will facilitate identification of deficiencies in current approaches, leading to more faithful differentiation strategies as well as providing insights into the rewiring of human regulatory programs during cellular transitions. \"epigenetic and transcriptional dynamics in {hescs} and {hesc}-derived {populations\"lineage}-specific remodeling at regions bound by {oct4}, {sox2}, and {nanog} in {hescs}\"germ-layer-specific switch to {h3k4me1} or {h3k27me3} at sites of high {dna} {methylation\"epigenetic} dynamics frequently precede transcriptional activation the epigenetic and transcriptional landscapes of three cell types representing each embryonic lineage derived from human embryonic stem cells are profiled, revealing distinct histone modification and {dna} methylation dynamics that accompany lineage specification.computational methods attempting to identify instances of cis-regulatory modules ({crms}) in the genome face a challenging problem of searching for potentially interacting transcription factor binding sites while knowledge of the specific interactions involved remains limited. without a comprehensive comparison of their performance, the reliability and accuracy of these tools remains unclear. faced with a large number of different tools that address this problem, we summarized and categorized them based on search strategy and input data requirements. twelve representative methods were chosen and applied to predict {crms} from the drosophila {crm} database {redfly}, and across the human {encode} regions. our results show that the optimal choice of method varies depending on species and composition of the sequences in question. when discriminating {crms} from non-coding regions, those methods considering evolutionary conservation have a stronger predictive power than methods designed to be run on a single genome. different {crm} representations and search strategies rely on different {crm} properties, and different methods can complement one another. for example, some favour homotypical clusters of binding sites, while others perform best on short {crms}. furthermore, most methods appear to be sensitive to the composition and structure of the genome to which they are applied. we analyze the principal features that distinguish the methods that performed well, identify weaknesses leading to poor performance, and provide a guide for users. we also propose key considerations for the development and evaluation of future {crm}-prediction methods. transcriptional regulation involves multiple transcription factors binding to {dna} sequences. a limited repertoire of transcription factors performs this complex regulatory step through various spatial and temporal interactions between themselves and their binding sites. these transcription factor binding interactions are clustered as distinct modules: cis-regulatory modules ({crms}). computational methods attempting to identify instances of {crms} in the genome face a challenging problem because a majority of these interactions between transcription factors remain unknown. to investigate the reliability and accuracy of these methods, we chose twelve representative methods and applied them to predict {crms} on both the fly and human genomes. our results show that the optimal choice of method varies depending on species and composition of the sequences in question. different {crm} representations and search strategies rely on different {crm} properties, and different methods can complement one another. we provide a guide for users and key considerations for developers. we also expect that, along with new technology generating new types of genomic data, future {crm} prediction methods will be able to reveal transcription binding interactions in three-dimensional space.we present a cross-species chemogenomic screening platform using libraries of haploid deletion mutants from two yeast species, saccharomyces cerevisiae and schizosaccharomyces pombe. we screened a set of compounds of known and unknown mode of action ({moa}) and derived quantitative drug scores (or d-scores), identifying mutants that are either sensitive or resistant to particular compounds. we found that compound–functional module relationships are more conserved than individual compound–gene interactions between these two species. furthermore, we observed that combining data from both species allows for more accurate prediction of {moa}. finally, using this platform, we identified a novel small molecule that acts as a {dna} damaging agent and demonstrate that its {moa} is conserved in human cells.cryptic variation is caused by the robustness of phenotypes to mutations. cryptic variation has no effect on phenotypes in a given genetic or environmental background, but it can have effects after mutations or environmental change. because evolutionary adaptation by natural selection requires phenotypic variation, phenotypically revealed cryptic genetic variation may facilitate evolutionary adaptation. this is possible if the cryptic variation happens to be pre-adapted, or \"exapted\", to a new environment, and is thus advantageous once revealed. however, this facilitating role for cryptic variation has not been proven, partly because most pertinent work focuses on complex phenotypes of whole organisms whose genetic basis is incompletely understood. here we show that populations of {rna} enzymes with accumulated cryptic variation adapt more rapidly to a new substrate than a population without cryptic variation. a detailed analysis of our evolving {rna} populations in genotype space shows that cryptic variation allows a population to explore new genotypes that become adaptive only in a new environment. our observations show that cryptic variation contains new genotypes pre-adapted to a changed environment. our results highlight the positive role that robustness and epistasis can have in adaptive evolution.motivation: biological experiments give insight into networks of processes inside a cell, but are subject to error and uncertainty. however, due to the overlap between the large number of experiments reported in public databases it is possible to assess the chances of individual observations being correct. in order to do so, existing methods rely on high-quality \\'gold standard\\' reference networks, but such reference networks are not always available.reversible phosphorylation is one of the major mechanisms of signal transduction, and signaling networks are critical regulators of cell growth and development. however, few of these networks have been delineated completely. towards this end, quantitative phosphoproteomics is emerging as a useful tool enabling large-scale determination of relative phosphorylation levels. however, phosphoproteomics differs from classical proteomics by a more extensive sampling limitation due to the limited number of detectable sites per protein. here, we propose a comprehensive quantitative analysis pipeline customized for phosphoproteome data from interventional experiments for identifying key proteins in specific pathways, discovering the protein-protein interactions and inferring the signaling network. we also made an effort to partially compensate for the missing value problem, a chronic issue for proteomics studies. the dataset used for this study was generated using {silac} (stable isotope labeling with amino acids in cell culture) technique with interventional experiments (kinase-dead mutations). the major components of the pipeline include phosphopeptide meta-analysis, correlation network analysis and causal relationship discovery. we have successfully applied our pipeline to interventional experiments identifying phosphorylation events underlying the transition to a filamentous growth form in saccharomyces cerevisiae. we identified 5 high-confidence proteins from meta-analysis, and 19 hub proteins from correlation analysis (pbi2p and hsp42p were identified by both analyses). all these proteins are involved in stress responses. nine of them have direct or indirect evidence of involvement in filamentous growth. in addition, we tested four of our predicted proteins, nth1p, pbi2p, pdr12p and rcn2p, by interventional phenotypic experiments and all of them present differential invasive growth, providing prospective validation of our approach. this comprehensive pipeline presents a systematic way for discovering signaling networks using interventional phosphoproteome data and can suggest candidate proteins for further investigation. we anticipate the methodology to be applicable as well to other interventional studies via different experimental platforms. signal transduction is a ubiquitous and essential mechanism regulating cellular functions, including responses to environmental stress. dysfunction of signaling pathways results in a variety of diseases, including cancer, diabetes, and cardiovascular disease. phosphorylation regulates the activity of signaling and target proteins at different cellular locations and controls activation and inactivation of signal pathways. here, we provide an analysis of phosphoproteome datasets from yeast, utilizing kinase mutants versus wild type strains. in order to provide an objective approach to identify candidate proteins involved in the transition to a filamentous growth form, we proposed and applied a comprehensive pipeline incorporating statistical and mathematical methods to investigate the phosphoproteome data from multiple perspectives. this included phosphorylation variation in response to a single mutant, phosphorylation variation patterns over multiple mutants, and the relationships represented by these patterns. we make an effort to discover the components and targets of the signaling network, infer the network structure, and to find the relationships of changes of protein phosphorylation to cellular functions, specifically in response to stress in the context of filamentous growth.phenotypic variation for quantitative traits results from the segregation of alleles at multiple quantitative trait loci ({qtl}) with effects that are sensitive to the genetic, sexual, and external environments. major challenges for biology in the post-genome era are to map the molecular polymorphisms responsible for variation in medically, agriculturally, and evolutionarily important complex traits; and to determine their gene frequencies and their homozygous, heterozygous, epistatic, and pleiotropic effects in multiple environments. the ease with which {qtl} can be mapped to genomic intervals bounded by molecular markers belies the difficulty in matching the {qtl} to a genetic locus. the latter requires high-resolution recombination or linkage disequilibrium mapping to nominate putative candidate genes, followed by genetic and/or functional complementation and gene expression analyses. complete genome sequences and improved technologies for polymorphism detection will greatly advance the genetic dissection of quantitative traits in model organisms, which will open avenues for exploration of homologous {qtl} in related taxa.transcriptional regulatory circuits govern how cis and trans factors transform signals into messenger {rna} ({mrna}) expression levels. with advances in quantitative and high-throughput technologies that allow measurement of gene expression state in different conditions, data that can be used to build and test models of transcriptional regulation is being generated at a rapid pace. here, we review experimental and computational methods used to derive detailed quantitative circuit models on a small scale and cruder, genome-wide models on a large scale. we discuss the potential of combining small- and large-scale approaches to understand the working and wiring of transcriptional regulatory circuits.{background}:in computational biology, one often faces the problem of deriving the causal relationship among different elements such as genes, proteins, metabolites, neurons and so on, based upon multi-dimensional temporal data. currently, there are two common approaches used to explore the network structure among elements. one is the granger causality approach, and the other is the dynamic bayesian network inference approach. both have at least a few thousand publications reported in the literature. a key issue is to choose which approach is used to tackle the data, in particular when they give rise to contradictory {results.results}:in this paper, we provide an answer by focusing on a systematic and computationally intensive comparison between the two approaches on both synthesized and experimental data. for synthesized data, a critical point of the data length is found: the dynamic bayesian network outperforms the granger causality approach when the data length is short, and vice versa. we then test our results in experimental data of short length which is a common scenario in current biological experiments: it is again confirmed that the dynamic bayesian network works {better.conclusion}:when the data size is short, the dynamic bayesian network inference performs better than the granger causality approach; otherwise the granger causality approach is better.one of the pressing open problems of computational systems biology is the elucidation of the topology of genetic regulatory networks ({grns}) using high throughput genomic data, in particular microarray gene expression data. the dialogue for reverse engineering assessments and methods ({dream}) challenge aims to evaluate the success of {grn} inference algorithms on benchmarks of simulated data. in this article, we present {genie3}, a new algorithm for the inference of {grns} that was best performer in the {dream4} in silico multifactorial challenge. {genie3} decomposes the prediction of a regulatory network between p genes into p different regression problems. in each of the regression problems, the expression pattern of one of the genes (target gene) is predicted from the expression patterns of all the other genes (input genes), using tree-based ensemble methods random forests or {extra-trees}. the importance of an input gene in the prediction of the target gene expression pattern is taken as an indication of a putative regulatory link. putative regulatory links are then aggregated over all genes to provide a ranking of interactions from which the whole network is reconstructed. in addition to performing well on the {dream4} in silico multifactorial challenge simulated data, we show that {genie3} compares favorably with existing algorithms to decipher the genetic regulatory network of escherichia coli. it doesn\\'t make any assumption about the nature of gene regulation, can deal with combinatorial and non-linear interactions, produces directed {grns}, and is fast and scalable. in conclusion, we propose a new algorithm for {grn} inference that performs well on both synthetic and real gene expression data. the algorithm, based on feature selection with tree-based ensemble methods, is simple and generic, making it adaptable to other types of genomic data and interactions.the biotechnological advances of the last decade have confronted us with an explosion of genetics, genomics, transcriptomics, proteomics, and metabolomics data. these data need to be organized and structured before they may provide a coherent biological picture. to accomplish this formidable task, the availability of an accurate map of the physical interactions in the cell that are responsible for cellular behavior and function would be exceedingly helpful, as these data are ultimately the result of such molecular interactions. however, all we have at this time is, at best, a fragmentary and only partially correct representation of the interactions between genes, their byproducts, and other cellular entities. if we want to succeed in our quest for understanding the biological whole as more than the sum of the individual parts, we need to build more comprehensive and cell-context-specific maps of the biological interaction networks. {dream}, the dialogue on reverse engineering assessment and methods, is fostering a concerted effort by computational and experimental biologists to understand the limitations and to enhance the strengths of the efforts to reverse engineer cellular networks from high-throughput data. in this chapter we will discuss the salient arguments of the first {dream} conference. we will highlight both the state of the art in the field of reverse engineering as well as some of its challenges and opportunities.{background}:gene regulation and metabolic reactions are two primary activities of life. although many works have been dedicated to study each system, the coupling between them is less well understood. to bridge this gap, we propose a joint model of gene regulation and metabolic {reactions.results}:we integrate regulatory and metabolic networks by adding links specifying the feedback control from the substrates of metabolic reactions to enzyme gene expressions. we adopt two alternative approaches to build those links: inferring the links between metabolites and transcription factors to fit the data or explicitly encoding the general hypotheses of feedback control as links between metabolites and enzyme expressions. a perturbation data is explained by paths in the joint network if the predicted response along the paths is consistent with the observed response. the consistency requirement for explaining the perturbation data imposes constraints on the attributes in the network such as the functions of links and the activities of paths. we build a probabilistic graphical model over the attributes to specify these constraints, and apply an inference algorithm to identify the attribute values which optimally explain the data. the inferred models allow us to 1) identify the feedback links between metabolites and regulators and their functions, 2) identify the active paths responsible for relaying perturbation effects, 3) computationally test the general hypotheses pertaining to the feedback control of enzyme expressions, 4) evaluate the advantage of an integrated model over separate {systems.conclusion}:the modeling results provide insight about the mechanisms of the coupling between the two systems and possible \"design rules\" pertaining to enzyme gene regulation. the model can be used to investigate the less well-probed systems and generate consistent hypotheses and predictions for further validation.this viewpoint comments on recent advances in understanding the design principles of biological networks. it highlights the surprising discovery of  ” good-engineering” principles in biochemical circuitry that evolved by random tinkering.the environment significantly influences the dynamic expression and assembly of all components encoded in the genome of an organism into functional biological networks. we have constructed a model for this process in halobacterium salinarum {nrc}-1 through the data-driven discovery of regulatory and functional interrelationships among ?80\\\\% of its genes and key abiotic factors in its hypersaline environment. using relative changes in 72 transcription factors and 9 environmental factors ({efs}) this model accurately predicts dynamic transcriptional responses of all these genes in 147 newly collected experiments representing completely novel genetic backgrounds and environments?suggesting a remarkable degree of network completeness. using this model we have constructed and tested hypotheses critical to this organism\\'s interaction with its changing hypersaline environment. this study supports the claim that the high degree of connectivity within biological and {ef} networks will enable the construction of similar models for any organism from relatively modest numbers of experiments.we have developed a general probabilistic system for query-based discovery of pathway-specific networks through integration of diverse genome-wide data. this framework was validated by accurately recovering known networks for 31 biological processes in saccharomyces cerevisiae and experimentally verifying predictions for the process of chromosomal segregation. our system, {biopixie}, a public, comprehensive system for integration, analysis, and visualization of biological network predictions for s. cerevisiae, is freely accessible over the worldwide web.a common task posed by microarray experiments is to infer the binding site preferences for a known transcription factor from a collection of genes that it regulates and to ascertain whether the factor acts alone or in a complex. the converse problem can also be posed: given a collection of binding sites, can the regulatory factor or complex of factors be inferred? both tasks are substantially facilitated by using relatively simple homology models for {protein–dna} interactions, as well as the rapidly expanding protein structure database. for budding yeast, we are able to construct reliable structural models for 67 transcription factors and with them redetermine factor binding sites by using a bayesian gibbs sampling algorithm and an extensive protein localization data set. for 49 factors in common with a prior analysis of this data set (based largely on phylogenetic conservation), we find that half of the previously predicted binding motifs are in need of some revision. we also solve the inverse problem of ascertaining the factors from the binding sites by assigning a correct protein fold to 25 of the 49 cases from a previous study. our approach is easily extended to other organisms, including higher eukaryotes. our study highlights the utility of enlarging current structural genomics projects that exhaustively sample fold structure space to include all factors with significantly different {dna}-binding specificities.motivation: with the exponential growth of expression and protein–protein interaction ({ppi}) data, the frontier of research in systems biology shifts more and more to the integrated analysis of these large datasets. of particular interest is the identification of functional modules in {ppi} networks, sharing common cellular function beyond the scope of classical pathways, by means of detecting differentially expressed regions in {ppi} networks. this requires on the one hand an adequate scoring of the nodes in the network to be identified and on the other hand the availability of an effective algorithm to find the maximally scoring network regions. various heuristic approaches have been proposed in the {literature.results}: here we present the first exact solution for this problem, which is based on integer-linear programming and its connection to the well-known prize-collecting steiner tree problem from operations research. despite the {np}-hardness of the underlying combinatorial problem, our method typically computes provably optimal subnetworks in large {ppi} networks in a few minutes. an essential ingredient of our approach is a scoring function defined on network nodes. we propose a new additive score with two desirable properties: (i) it is scalable by a statistically interpretable parameter and (ii) it allows a smooth integration of data from various {sources.we} apply our method to a well-established lymphoma microarray dataset in combination with associated survival data and the large interaction network of {hprd} to identify functional modules by computing optimal-scoring subnetworks. in particular, we find a functional interaction module associated with proliferation over-expressed in the aggressive {abc} subtype as well as modules derived from non-malignant by-stander {cells.availability}: our software is available freely for non-commercial purposes at {http://www.planet-lisa.net.contact}: tobias.mueller@biozentrum.uni-wuerzburg.dehistopathological classification of human tumors relies in part on the degree of differentiation of the tumor sample. to date, there is no objective systematic method to categorize tumor subtypes by maturation. in this paper, we introduce a novel computational algorithm to rank tumor subtypes according to the dissimilarity of their gene expression from that of stem cells and fully differentiated tissue, and thereby construct a phylogenetic tree of cancer. we validate our methodology with expression data of leukemia, breast cancer and liposarcoma subtypes and then apply it to a broader group of sarcomas. this ranking of tumor subtypes resulting from the application of our methodology allows the identification of genes correlated with differentiation and may help to identify novel therapeutic targets. our algorithm represents the first phylogeny-based tool to analyze the differentiation status of human tumors. gene expression profiling of malignancies is often held to demonstrate genes that are  ” up-regulated” or  ” down-regulated”, but the appropriate frame of reference against which observations should be compared has not been determined. fully differentiated somatic cells arise from stem cells, with changes in gene expression that can be experimentally determined. if cancers arise as the result of an abruption of the differentiation process, then poorly differentiated cancers would have a gene expression more similar to stem cells than to normal differentiated tissue, and well differentiated cancers would have a gene expression more similar to fully differentiated cells than to stem cells. in this paper, we describe a novel computational algorithm that allows orientation of cancer gene expression between the poles of the gene expression of stem cells and of fully differentiated tissue. our methodology allows the construction of a multi-branched phylogeny of human malignancies and can be used to identify genes related to differentiation as well as novel therapeutic targets.protein–protein interactions ({ppis}) and their networks play a central role in all biological processes. akin to the complete sequencing of genomes and their comparative analysis, complete descriptions of interactomes and their comparative analysis is fundamental to a deeper understanding of biological processes. a first step in such an analysis is to align two or more {ppi} networks. here, we introduce an algorithm, {isorank}, for global alignment of multiple {ppi} networks. the guiding intuition here is that a protein in one {ppi} network is a good match for a protein in another network if their respective sequences and neighborhood topologies are a good match. we encode this intuition as an eigenvalue problem in a manner analogous to google\\'s {pagerank} method. using {isorank}, we compute a global alignment of the saccharomyces cerevisiae, drosophila melanogaster, caenorhabditis elegans, mus musculus, and homo sapiens {ppi} networks. we demonstrate that incorporating {ppi} data in ortholog prediction results in improvements over existing sequence-only approaches and over predictions from local alignments of the yeast and fly networks. previous methods have been effective at identifying conserved, localized network patterns across pairs of networks. this work takes the further step of performing a global alignment of multiple {ppi} networks. it simultaneously uses sequence similarity and network data and, unlike previous approaches, explicitly models the tradeoff inherent in combining them. we expect {isorank}—with its simultaneous handling of node similarity and network similarity—to be applicable across many scientific domains.{background}: in the past years devising methods for discovering gene regulatory mechanisms at a genome-wide level has become a fundamental topic in the field of systems biology. the aim is to infer gene-gene interactions in an increasingly sophisticated and reliable way through the continuous improvement of reverse engineering algorithms exploiting microarray data. {motivation}: this work is inspired by the several studies suggesting that coexpression is mostly related to \\'static\\' stable binding relationships, like belonging to the same protein complex, rather than other types of interactions more of a \\'causal\\' and transient nature (e.g. transcription factor-binding site interactions). the aim of this work is to verify if direct or conditional network inference algorithms (e.g. pearson correlation for the former, partial pearson correlation for the latter) are indeed useful in discerning static from causal dependencies in artificial and real gene networks (derived from escherichia coli and saccharomyces cerevisiae). {results}: even in the regime of weak inference power we have to work in, our analysis confirms the differences in the performances of the algorithms: direct methods are more robust in detecting stable interactions, conditional ones are better for causal interactions especially in presence of combinatorial transcriptional regulation. {supplementary} {information}: supplementary data are available at bioinformatics online.{dna} evolution models made invaluable contributions to comparative genomics, although it seemed formidable to include non-genomic features into these models. in order to build an evolutionary model of transcription networks ({tns}), we had to forfeit the substitution model used in {dna} evolution and to start from modeling the evolution of the regulatory relationships. we present a quantitative evolutionary model of {tns}, subjecting the phylogenetic distance and the evolutionary changes of cis-regulatory sequence, gene expression and network structure to one probabilistic framework. using the genome sequences and gene expression data from multiple species, this model can predict regulatory relationships between a transcription factor ({tf}) and its target genes in all species, and thus identify {tn} re-wiring events. applying this model to analyze the pre-implantation development of three mammalian species, we identified the conserved and re-wired components of the {tns} downstream to a set of {tfs} including oct4, gata3/4/6, {cmyc} and {nmyc}. evolutionary events on the {dna} sequence that led to turnover of {tf} binding sites were identified, including a birth of an oct4 binding site by a 2nt deletion. in contrast to recent reports of large interspecies differences of {tf} binding sites and gene expression patterns, the interspecies difference in {tf}-target relationship is much smaller. the data showed increasing conservation levels from genomic sequences to {tf}-{dna} interaction, gene expression, {tn}, and finally to morphology, suggesting that evolutionary changes are larger at molecular levels and smaller at functional levels. the data also showed that evolutionarily older {tfs} are more likely to have conserved target genes, whereas younger {tfs} tend to have larger re-wiring rates. {dna} evolution models made invaluable contributions to comparative genomic studies. still lacking is an evolutionary model of transcription networks ({tns}). to develop such a model, we had to forfeit the substitution model used in {dna} evolution and to start from modeling the evolution of the regulatory relationships, and then subject the phylogenetic distance and the multi-species {dna} sequence and gene expression data to one probabilistic framework. this model enabled us to infer the evolutionary changes of transcriptional regulatory relationships. applying this model to analyze three yeast species, we found the anaerobic phenotype in two species was associated with the evolutionary loss of a larger cis-regulatory motif than previously thought. analyzing three mammalian species, we found increasing conservation levels from genomic sequences to transcription {factor-dna} interaction, gene expression, {tn}, and finally to morphology, suggesting that evolutionary changes are larger at molecular levels and smaller at functional levels. we also found that evolutionarily younger {tfs} are more likely to regulate different target genes in different species.we use in\\xa0situ {hi-c} to probe the {3d} architecture of genomes, constructing haploid and diploid maps of nine cell types. the densest, in human lymphoblastoid cells, contains 4.9 billion contacts, achieving 1 kb resolution. we find that genomes are partitioned into contact domains (median length, 185 kb), which are associated with distinct patterns of histone marks and segregate into six subcompartments. we identify ?10,000 loops. these loops frequently link promoters and enhancers, correlate with gene activation, and show conservation across cell types and species. loop anchors typically occur at domain boundaries and bind {ctcf}. {ctcf} sites at loop anchors occur predominantly (>90\\\\%) in a convergent orientation, with the asymmetric motifs ?facing? one another. the inactive x chromosome splits into two massive domains and contains large loops anchored at {ctcf}-binding repeats.we present a method (the inferelator) for deriving genome-wide transcriptional regulatory interactions, and apply the method to predict a large portion of the regulatory network of the archaeon halobacterium {nrc}-1. the inferelator uses regression and variable selection to identify transcriptional influences on genes based on the integration of genome annotation and expression data. the learned network successfully predicted halobacterium\\'s global expression under novel perturbations with predictive power similar to that seen over training data. several specific regulatory predictions were experimentally tested and verified.protein interactions play a vital part in the function of a cell. as experimental techniques for detection and validation of protein interactions are time consuming, there is a need for computational methods for this task. protein interactions appear to form a network with a relatively high degree of local clustering. in this paper we exploit this clustering by suggesting a score based on triplets of observed protein interactions. the score utilises both protein characteristics and network properties. our score based on triplets is shown to complement existing techniques for predicting protein interactions, outperforming them on data sets which display a high degree of clustering. the predicted interactions score highly against test measures for accuracy. compared to a similar score derived from pairwise interactions only, the triplet score displays higher sensitivity and specificity. by looking at specific examples, we show how an experimental set of interactions can be enriched and validated. as part of this work we also examine the effect of different prior databases upon the accuracy of prediction and find that the interactions from the same kingdom give better results than from across kingdoms, suggesting that there may be fundamental differences between the networks. these results all emphasize that network structure is important and helps in the accurate prediction of protein interactions. the protein interaction data set and the program used in our analysis, and a list of predictions and validations, are available at http://www.stats.ox.ac.uk/bioinfo/resour\\u200b{ces/predictinginteractions}. for understanding the complex activities within an organism, a complete and error-free network of protein interactions which occur in the organism would be a significant step forward. the large amount of experimentally derived data now available has provided us with a chance to study the complicated behaviour of protein interactions. the power of such studies, however, has been limited due to the high false positive and false negative rates in the datasets. we propose a network-based method, taking advantage of the tendency of clustering in protein interaction networks, to validate experimental data and to predict unknown interactions. the integration of multiple protein characteristics (i.e., structure, function, etc.) allows our predictive method to significantly outperform two other approaches based on homology and protein-domain relationships on datasets which contain a large amount of interactions, but not much detailed information on the proteins involved in the interactions. in addition, our predictive score based on triadic interaction patterns improves over a pair-wise approach, suggesting the importance of network structure. moreover, using pooled interactions as prior information, we find evidence for fundamental differences in protein interaction networks between eukaryotes and prokaryotes.gene regulation is mediated by site-specific {dna}-binding proteins or transcription factors ({tfs}), which form protein complexes at regulatory loci either to activate or repress the expression of a target gene. the study of the dynamic properties of these regulatory {dna}-binding complexes has so far been dominated by protein-centered methodologies, aiming to characterize the {dna}-binding behavior of one specific protein at a time. with the emerging evidence for a role of {dna} in allosterically influencing {dna}-binding protein complex formation, there is renewed interest in {dna}-centered approaches to capture protein complexes on defined regulatory loci and to correlate changes in their composition with alterations in target gene expression. in this review, we present the current state-of-the-art in such {dna}-centered approaches and evaluate recent technological improvements in the purification as well as in the identification of regulatory {dna}-binding protein complexes within or outside their biological context. finally, we suggest possible areas of improvement and assess the putative impact of {dna}-centered methodologies on the gene regulation field for the forthcoming years.perennial questions of evolutionary biology can be applied to gene regulatory systems using the abundance of experimental data addressing gene regulation in a comparative context. what is the tempo (frequency, rate) and mode (way, mechanism) of transcriptional regulatory evolution? here we synthesize the results of 230 experiments performed on insects and nematodes in which regulatory {dna} from one species was used to drive gene expression in another species. general principles of regulatory evolution emerge. gene regulatory evolution is widespread and accumulates with genetic divergence in both insects and nematodes. divergence in cis is more common than divergence in trans. coevolution between cis and trans shows a particular increase over greater evolutionary timespans, especially in sex-specific gene regulation. despite these generalities, the evolution of gene regulation is gene- and taxon-specific. the congruence of these conclusions with evidence from other types of experiments suggests that general principles are discoverable, and a unified view of the tempo and mode of regulatory evolution may be achievable.metazoan transcription is controlled through either coordinated recruitment of transcription machinery to the gene promoter or regulated pausing of {rna} polymerase {ii} (pol {ii}) in early elongation. we report that a striking difference between genes that use these distinct regulatory strategies lies in the \\x1cdefault\\x1d chromatin architecture specified by their {dna} sequences. pol {ii} pausing is prominent at highly\\xa0regulated genes whose sequences inherently disfavor nucleosome formation within the gene but favor occlusion of the promoter by nucleosomes. in contrast, housekeeping genes that lack pronounced pol {ii} pausing show higher nucleosome occupancy downstream, but their promoters are deprived of nucleosomes regardless of polymerase binding. our results indicate that a key role of paused pol {ii} is to compete with nucleosomes for occupancy of highly regulated promoters, thereby preventing the formation of repressive chromatin architecture to facilitate further or future gene activation. º transcribed genes globally exhibit pausing of polymerase during early elongation º the most paused genes have promoter {dna} sequences that favor nucleosome assembly º paused pol {ii} prevents nucleosome formation to maintain or potentiate gene activity º promoter-proximal nucleosomes are not required to establish paused pol {ii}{chip}-seq has become an important tool for identifying genome-wide {protein-dna} interactions, including transcription factor binding and histone modifications. in {chip}-seq experiments, {chip} samples are usually coupled with their matching control samples. proper normalization between the {chip} and control samples is an essential aspect of {chip}-seq data analysis.biological networks provide additional information for the analysis of human diseases, beyond the traditional analysis that focuses on single variables. gaussian graphical model ({ggm}), a probability model that characterizes the conditional dependence structure of a set of random variables by a graph, has wide applications in the analysis of biological networks, such as inferring interaction or comparing differential networks. however, existing approaches are either not statistically rigorous or are inefficient for high-dimensional data that include tens of thousands of variables for making inference. in this study, we propose an efficient algorithm to implement the estimation of {ggm} and obtain p-value and confidence interval for each edge in the graph, based on a recent proposal by ren et al., 2015. through simulation studies, we demonstrate that the algorithm is faster by several orders of magnitude than the current implemented algorithm for ren et al. without losing any accuracy. then, we apply our algorithm to two real data sets: transcriptomic data from a study of childhood asthma and proteomic data from a study of alzheimer\\'s disease. we estimate the global gene or protein interaction networks for the disease and healthy samples. the resulting networks reveal interesting interactions and the differential networks between cases and controls show functional relevance to the diseases. in conclusion, we provide a computationally fast algorithm to implement a statistically sound procedure for constructing gaussian graphical model and making inference with high-dimensional biological data. the algorithm has been implemented in an r package named  ” {fastggm}”. gaussian graphical model ({ggm}), a probability model for characterizing conditional dependence among a set of random variables, has been widely used in studying biological networks. it is important and practical to make inference with rigorous statistical properties and high efficiency under a high-dimensional setting, which is common in biological systems that usually contain tens of thousands of molecular elements, such as genes and proteins. this work proposes a novel efficient algorithm, {fastggm}, to implement asymptotically normal estimation of large {ggm} established by ren et al [1]. it quickly estimates the precision matrix, partial correlations, as well as p-values and confidence intervals for the graph. simulation studies demonstrate our algorithm outperforms the current algorithm for ren et al. and algorithms for some other estimation methods, and real data analyses further prove its efficiency in studying biological networks. in conclusion, {fastggm} is a statistically sound and computationally fast algorithm for constructing {ggm} with high-dimensional data. an r package for implementation can be downloaded from {http://www.pitt.edu/\\\\~{}wec47/fastggm}.html.new sequencing methods generate data that can allow the assembly of microbial genome sequences in days. with such revolutionary advances in technology come new challenges in methodologies and informatics. in this article, we review the capabilities of high-throughput sequencing technologies and discuss the many options for getting useful information from the data.distinguishing the somatic mutations responsible for cancer (driver mutations) from random, passenger mutations is a key challenge in cancer genomics. driver mutations generally target cellular signaling and regulatory pathways consisting of multiple genes. this heterogeneity complicates the identification of driver mutations by their recurrence across samples, as different combinations of mutations in driver pathways are observed in different samples. we introduce the {multi-dendrix} algorithm for the simultaneous identification of multiple driver pathways de novo in somatic mutation data from a cohort of cancer samples. the algorithm relies on two combinatorial properties of mutations in a driver pathway: high coverage and mutual exclusivity. we derive an integer linear program that finds set of mutations exhibiting these properties. we apply {multi-dendrix} to somatic mutations from glioblastoma, breast cancer, and lung cancer samples. {multi-dendrix} identifies sets of mutations in genes that overlap with known pathways – including rb, p53, {pi}({3)k}, and cell cycle pathways – and also novel sets of mutually exclusive mutations, including mutations in several transcription factors or other genes involved in transcriptional regulation. these sets are discovered directly from mutation data with no prior knowledge of pathways or gene interactions. we show that {multi-dendrix} outperforms other algorithms for identifying combinations of mutations and is also orders of magnitude faster on genome-scale data. software available at: http://compbio.cs.brown.edu/software. cancer is a disease driven largely by the accumulation of somatic mutations during the lifetime of an individual. the declining costs of genome sequencing now permit the measurement of somatic mutations in hundreds of cancer genomes. a key challenge is to distinguish driver mutations responsible for cancer from random passenger mutations. this challenge is compounded by the observation that different combinations of driver mutations are observed in different patients with the same cancer type. one reason for this heterogeneity is that driver mutations target signaling and regulatory pathways which have multiple points of failure. we introduce an algorithm, {multi-dendrix}, to find these pathways solely from patterns of mutual exclusivity between mutations across a cohort of patients. unlike earlier approaches, we simultaneously find multiple pathways, an essential feature for analyzing cancer genomes where multiple pathways are typically perturbed. we apply our algorithm to mutation data from hundreds of glioblastoma, breast cancer, and lung adenocarcinoma patients. we identify sets of interacting genes that overlap known pathways, and gene sets containing subtype-specific mutations. these results show that multiple cancer pathways can be identified directly from patterns in mutation data, and provide an approach to analyze the ever-growing cancer mutation datasets.several recent methods have addressed the problem of heterogeneous data integration and network prediction by modeling the noise inherent in high-throughput genomic datasets, which can dramatically improve specificity and sensitivity and allow the robust integration of datasets with heterogeneous properties. however, experimental technologies capture different biological processes with varying degrees of success, and thus, each source of genomic data can vary in relevance depending on the biological process one is interested in predicting. accounting for this variation can significantly improve network prediction, but to our knowledge, no previous approaches have explicitly leveraged this critical information about biological context. we confirm the presence of context-dependent variation in functional genomic data and propose a bayesian approach for context-sensitive integration and query-based recovery of biological process-specific networks. by applying this method to saccharomyces cerevisiae, we demonstrate that leveraging contextual information can significantly improve the precision of network predictions, including assignment for uncharacterized genes. we expect that this general context-sensitive approach can be applied to other organisms and prediction scenarios. a software implementation of our approach is available on request from the authors. supplementary data are available at {http://avis.princeton.edu/contextpixie}/post-genomic molecular biology has resulted in an explosion of data, providing measurements for large numbers of genes, proteins and metabolites. time series experiments have become increasingly common, necessitating the development of novel analysis tools that capture the resulting data structure. outlier measurements at one or more time points present a significant challenge, while potentially valuable replicate information is often ignored by existing techniques. we present a generative model-based bayesian hierarchical clustering algorithm for microarray time series that employs gaussian process regression to capture the structure of the data. by using a mixture model likelihood, our method permits a small proportion of the data to be modelled as outlier measurements, and adopts an empirical bayes approach which uses replicate observations to inform a prior distribution of the noise variance. the method automatically learns the optimum number of clusters and can incorporate non-uniformly sampled time points. using a wide variety of experimental data sets, we show that our algorithm consistently yields higher quality and more biologically meaningful clusters than current state-of-the-art methodologies. we highlight the importance of modelling outlier values by demonstrating that noisy genes can be grouped with other genes of similar biological function. we demonstrate the importance of including replicate information, which we find enables the discrimination of additional distinct expression profiles. by incorporating outlier measurements and replicate values, this clustering algorithm for time series microarray data provides a step towards a better treatment of the noise inherent in measurements from high-throughput genomic technologies. timeseries {bhc} is available as part of the r package \\'{bhc}\\' (version 1.5), which is available for download from bioconductor (version 2.9 and above) via {http://www.bioconductor.org/packages/release/bioc/html/bhc}.html?pagewanted=all.cancer cells acquire pathological phenotypes through accumulation of mutations that perturb signaling networks. however, global analysis of these events is currently limited. here, we identify six types of network-attacking mutations ({nams}), including changes in kinase and {sh2} modulation, network rewiring, and the genesis and extinction of phosphorylation sites. we developed a computational platform ({rekinect}) to identify {nams} and systematically interpreted the exomes and quantitative (phospho-)proteomes of five ovarian cancer cell lines and the global cancer genome repository. we identified and experimentally validated several {nams}, including {pkc}? {m501i} and {pkd1} {d665n}, which encode specificity switches analogous to the appearance of kinases de novo within the kinome. we discover mutant molecular logic gates, a drift toward phospho-threonine signaling, weakening of phosphorylation motifs, and kinase-inactivating hotspots in cancer. our method pinpoints functional {nams}, scales with the complexity of cancer genomes and cell signaling, and may enhance our capability to therapeutically target tumor-specific networks.{background}:recent technological advances have enabled high-throughput measurements of protein-protein interactions in the cell, producing large protein interaction networks for various species at an ever-growing pace. however, common technologies like yeast two-hybrid may experience high rates of false positive detection. to combat false positive discoveries, a number of different methods have been recently developed that associate confidence scores with protein interactions. here, we perform a rigorous comparative analysis and performance assessment among these different {methods.results}:we measure the extent to which each set of confidence scores correlates with similarity of the interacting proteins in terms of function, expression, pattern of sequence conservation, and homology to interacting proteins in other species. we also employ a new metric, the {signal-to-noise} ratio of protein complexes embedded in each network, to assess the power of the different methods. seven confidence assignment schemes, including those of bader et al., deane et al., deng et al., sharan et al., and qi et al., are compared in this {work.conclusion}:although the performance of each assignment scheme varies depending on the particular metric used for assessment, we observe that deng et al. yields the best performance overall (in three out of four viable measures). importantly, we also find that utilizing any of the probability assignment schemes is always more beneficial than assuming all observed interactions to be true or equally likely.it is demonstrated that, properly represented, the amino acid composition of protein sequences contains the information necessary to delineate the global properties of protein structure space. a numerical representation of amino acid sequence in terms of a set of property factors is used, and the values of those property factors are averaged over individual sequences and then over sets of sequences belonging to structurally defined groups. these sequence sets then can be viewed as points in a 10-dimensional space, and the organization of that space, determined only by sequence properties, is similar at both local and global scales to that of the space of protein structures determined previously.genome-wide mapping of three dimensional chromatin organization is an important yet technically challenging task. to aid experimental effort and to understand the determinants of long-range chromatin interactions, we have developed a computational model integrating {hi-c} and histone mark {chip}-seq data to predict two important features of chromatin organization: chromatin interaction hubs and topologically associated domain ({tad}) boundaries. our model accurately and robustly predicts these features across datasets and cell types. cell-type specific histone mark information is required for prediction of chromatin interaction hubs but not for {tad} boundaries. our predictions provide a useful guide for the exploration of chromatin organization.analyses of gene expression data sets for multiple individuals and species promise to shed light on the mode of evolution of gene expression. however, complementary complexities challenge this goal. characterization of the genetic variation underlying gene expression can easily be compromised by lack of environmental control. conversely, the breadth of conclusions from studies of environmental effects has been limited by the use of single strains. controlled studies have hinted at extensive gene\\xa0×\\xa0environment interaction. thus, both genetics and environment are key components in models of the evolution of gene expression. we review the literature on the evolution of gene expression in terms of genetics (g), environmental response (e) and g\\xa0×\\xa0e interactions to make this conceptual point.▪ abstract\\u2002the {myc/max}/mad network comprises a group of transcription factors whose distinct interactions result in gene-specific transcriptional activation or repression. a great deal of research indicates that the functions of the network play roles in cell proliferation, differentiation, and death. in this review we focus on the myc and mad protein families and attempt to relate their biological functions to their transcriptional activities and gene targets. both myc and mad, as well as the more recently described mnt and mga proteins, form heterodimers with max, permitting binding to specific {dna} sequences. these {dna}-bound heterodimers recruit coactivator or corepressor complexes that generate alterations in chromatin structure, which in turn modulate transcription. initial identification of target genes suggests that the network regulates genes involved in the cell cycle, growth, life span, and morphology. because myc and mad proteins are expressed in response to diverse signaling pathways, the network can be viewed as a functional module which acts to convert environmental signals into specific gene-regulatory programs.{background}:structural analysis of biochemical networks is a growing field in bioinformatics and systems biology. the availability of an increasing amount of biological data from molecular biological networks promises a deeper understanding but confronts researchers with the problem of combinatorial explosion. the amount of qualitative network data is growing much faster than the amount of quantitative data, such as enzyme kinetics. in many cases it is even impossible to measure quantitative data because of limitations of experimental methods, or for ethical reasons. thus, a huge amount of qualitative data, such as interaction data, is available, but it was not sufficiently used for modeling purposes, until now. new approaches have been developed, but the complexity of data often limits the application of many of the methods. biochemical petri nets make it possible to explore static and dynamic qualitative system properties. one petri net approach is model validation based on the computation of the system\\'s invariant properties, focusing on t-invariants. t-invariants correspond to subnetworks, which describe the basic system {behavior.with} increasing system complexity, the basic behavior can only be expressed by a huge number of t-invariants. according to our validation criteria for biochemical petri nets, the necessary verification of the biological meaning, by interpreting each subnetwork (t-invariant) manually, is not possible anymore. thus, an automated, biologically meaningful classification would be helpful in analyzing t-invariants, and supporting the understanding of the basic behavior of the considered biological {system.methods}:here, we introduce a new approach to automatically classify t-invariants to cope with network complexity. we apply clustering techniques such as {upgma}, complete linkage, single linkage, and neighbor joining in combination with different distance measures to get biologically meaningful clusters (t-clusters), which can be interpreted as modules. to find the optimal number of t-clusters to consider for interpretation, the cluster validity measure, silhouette width, is {applied.results}:we considered two different case studies as examples: a small signal transduction pathway (pheromone response pathway in saccharomyces cerevisiae) and a medium-sized gene regulatory network (gene regulation of duchenne muscular dystrophy). we automatically classified the t-invariants into functionally distinct t-clusters, which could be interpreted biologically as functional modules in the network. we found differences in the suitability of the various distance measures as well as the clustering methods. in terms of a biologically meaningful classification of t-invariants, the best results are obtained using the tanimoto distance measure. considering clustering methods, the obtained results suggest that {upgma} and complete linkage are suitable for clustering t-invariants with respect to the biological {interpretability.conclusion}:we propose a new approach for the biological classification of petri net t-invariants based on cluster analysis. due to the biologically meaningful data reduction and structuring of network processes, large sets of t-invariants can be evaluated, allowing for model validation of qualitative biochemical petri nets. this approach can also be applied to elementary mode analysis.motivation: time series expression experiments are an increasingly popular method for studying a wide range of biological systems. however, when analyzing these experiments researchers face many new computational challenges. algorithms that are specifically designed for time series experiments are required so that we can take advantage of their unique features (such as the ability to infer causality from the temporal response pattern) and address the unique problems they raise (e.g. handling the different non-uniform sampling {rates).results}: we present a comprehensive review of the current research in time series expression data analysis. we divide the computational challenges into four analysis levels: experimental design, data analysis, pattern recognition and networks. for each of these levels, we discuss computational and biological problems at that level and point out some of the methods that have been proposed to deal with these issues. many open problems in all these levels are discussed. this review is intended to serve as both, a point of reference for experimental biologists looking for practical solutions for analyzing their data, and a starting point for computer scientists interested in working on the computational problems related to time series expression analysis.genotypic differences greatly influence susceptibility and resistance to disease. understanding genotype-phenotype relationships requires that phenotypes be viewed as manifestations of network properties, rather than simply as the result of individual genomic variations. genome sequencing efforts have identified numerous germline mutations, and large numbers of somatic genomic alterations, associated with a predisposition to cancer. however, it remains difficult to distinguish background, or \\'passenger\\', cancer mutations from causal, or \\'driver\\', mutations in these data sets. human viruses intrinsically depend on their host cell during the course of infection and can elicit pathological phenotypes similar to those arising from mutations. here we test the hypothesis that genomic variations and tumour viruses may cause cancer through related mechanisms, by systematically examining host interactome and transcriptome network perturbations caused by {dna} tumour virus proteins. the resulting integrated viral perturbation data reflects rewiring of the host cell networks, and highlights pathways, such as notch signalling and apoptosis, that go awry in cancer. we show that systematic analyses of host targets of viral proteins can identify cancer genes with a success rate on a par with their identification through functional genomics and large-scale cataloguing of tumour mutations. together, these complementary approaches increase the specificity of cancer gene identification. combining systems-level studies of pathogen-encoded gene products with genomic approaches will facilitate the prioritization of cancer-causing driver genes to advance the understanding of the genetic basis of human cancer.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsunderstanding the regulation of gene expression, including transcription start site usage, alternative splicing, and polyadenylation, requires accurate quantification of expression levels down to the level of individual transcript isoforms. to comparatively evaluate the accuracy of the many methods that have been proposed for estimating transcript isoform abundance from {rna} sequencing data, we have used both synthetic data as well as an independent experimental method for quantifying the abundance of transcript ends at the genome-wide level.in multicellular organisms, transcription regulation is one of the central mechanisms modelling lineage differentiation and cell-fate determination. transcription requires dynamic chromatin configurations between promoters and their corresponding distal regulatory elements. it is believed that their communication occurs within large discrete foci of aggregated {rna} polymerases termed transcription factories in three-dimensional nuclear space. however, the dynamic nature of chromatin connectivity has not been characterized at the genome-wide level. here, through a chromatin interaction analysis with paired-end tagging approach using an antibody that primarily recognizes the pre-initiation complexes of {rna} polymerase {ii}, we explore the transcriptional interactomes of three mouse cells of progressive lineage commitment, including pluripotent embryonic stem cells, neural stem cells and neurosphere stem/progenitor cells. our global chromatin connectivity maps reveal approximately 40,000 long-range interactions, suggest precise enhancer-promoter associations and delineate cell-type-specific chromatin structures. analysis of the complex regulatory repertoire shows that there are extensive colocalizations among promoters and distal-acting enhancers. most of the enhancers associate with promoters located beyond their nearest active genes, indicating that the linear juxtaposition is not the only guiding principle driving enhancer target selection. although promoter-enhancer interactions exhibit high cell-type specificity, promoters involved in interactions are found to be generally common and mostly active among different cells. chromatin connectivity networks reveal that the pivotal genes of reprogramming functions are transcribed within physical proximity to each other in embryonic stem cells, linking chromatin architecture to coordinated gene expression. our study sets the stage for the full-scale dissection of spatial and temporal genome structures and their roles in orchestrating development.recent studies have emphasized the importance of pathway-specific interpretations for understanding the functional relevance of gene alterations in human cancers. although signaling activities are often conceptualized as linear events, in reality, they reflect the activity of complex functional networks assembled from modules that each respond to input signals. to acquire a deeper understanding of this network structure, we developed an approach to deconstruct pathways into modules represented by gene expression signatures. our studies confirm that they represent units of underlying biological activity linked to known biochemical pathway structures. importantly, we show that these signaling modules provide tools to dissect the complexity of oncogenic states that define disease outcomes as well as response to pathway-specific therapeutics. we propose that this model of pathway structure constitutes a framework to study the processes by\\xa0which information propogates through cellular networks and to elucidate the relationships of fundamental modules to cellular and clinical phenotypes.there have been various attempts to reconstruct gene regulatory networks from microarray expression data in the past. however, owing to the limited amount of independent experimental conditions and noise inherent in the measurements, the results have been rather modest so far. for this reason it seems advisable to include biological prior knowledge, related, for instance, to transcription factor binding locations in promoter regions or partially known signalling pathways from the literature. in the present paper, we consider a bayesian approach to systematically integrate expression data with multiple sources of prior knowledge. each source is encoded via a separate energy function, from which a prior distribution over network structures in the form of a gibbs distribution is constructed. the hyperparameters associated with the different sources of prior knowledge, which measure the influence of the respective prior relative to the data, are sampled from the posterior distribution with {mcmc}. we have evaluated the proposed scheme on the yeast cell cycle and the raf signalling pathway. our findings quantify to what extent the inclusion of independent prior knowledge improves the network reconstruction accuracy, and the values of the hyperparameters inferred with the proposed scheme were found to be close to optimal with respect to minimizing the reconstruction error.transcriptional regulation and posttranscriptional processing underlie many cellular and organismal phenotypes. we used {rna} sequence data generated by {genotype-tissue} expression ({gtex}) project to investigate the patterns of transcriptome variation across individuals and tissues. tissues exhibit characteristic transcriptional signatures that show stability in postmortem samples. these signatures are dominated by a relatively small number of genes—which is most clearly seen in blood—though few are exclusive to a particular tissue and vary more across tissues than individuals. genes exhibiting high interindividual expression variation include disease candidates associated with sex, ethnicity, and age. primary transcription is the major driver of cellular specificity, with splicing playing mostly a complementary role; except for the brain, which exhibits a more divergent splicing program. variation in splicing, despite its stochasticity, may play in contrast a comparatively greater role in defining individual phenotypes. copyright {\\\\copyright} 2015, american association for the advancement of science.the number of complete, publicly available genome sequences is now greater than 200, and this number is expected to rapidly grow in the near future as metagenomic and environmental sequencing efforts escalate and the cost of sequencing drops. in order to make use of this data for understanding particular organisms and for discerning general principles about how organisms function, it will be necessary to reconstruct their various biochemical reaction networks. principal among these will be transcriptional regulatory networks. given the physical and logical complexity of these networks, the various sources of (often noisy) data that can be utilized for their elucidation, the monetary costs involved, and the huge number of potential experiments (\\\\~{}1012) that can be performed, experiment design algorithms will be necessary for synthesizing the various computational and experimental data to maximize the efficiency of regulatory network reconstruction. this paper presents an algorithm for experimental design to systematically and efficiently reconstruct transcriptional regulatory networks. it is meant to be applied iteratively in conjunction with an experimental laboratory component. the algorithm is presented here in the context of reconstructing transcriptional regulation for metabolism in escherichia coli, and, through a retrospective analysis with previously performed experiments, we show that the produced experiment designs conform to how a human would design experiments. the algorithm is able to utilize probability estimates based on a wide range of computational and experimental sources to suggest experiments with the highest potential of discovering the greatest amount of new regulatory knowledge. in recent years, the exploration of life has been bolstered through the advent of whole genome sequencing. this new data source significantly enables the reconstruction of genome-scale metabolic networks. after a metabolic reconstruction, it will be necessary to discover the genetic control mechanisms that operate within an organism. transcriptional regulatory network ({trn}) reconstruction is costly both in terms of time and money, so it is critical that the reconstruction efforts be made as efficient as possible. experiments must be designed so that the most new regulatory knowledge is discovered in each experiment. the huge number of possible experiments (\\\\~{}1012) and the vast amount of heterogeneous data available for designing experiments overwhelms the human ability to assimilate. the authors have developed an algorithm that utilizes a mathematical model of a reconstructed metabolic network integrated with a partially reconstructed {trn} to identify the experiment designs with the highest potential of yielding the most new regulatory knowledge. the authors show that the produced experiment designs are similar to those a human expert would produce, and that the algorithm has a facility to incorporate any relevant data source to design such experiments.plasmodium falciparum is the pathogen responsible for over 90\\\\% of human deaths from malaria1. therefore, it has been the focus of a considerable research initiative, involving the complete {dna} sequencing of the genome2, large-scale expression analyses3, 4, and protein characterization of its life-cycle stages5. the plasmodium genome sequence is relatively distant from those of most other eukaryotes, with more than 60\\\\% of the 5,334 encoded proteins lacking any notable sequence similarity to other organisms2. to systematically elucidate functional relationships among these proteins, a large two-hybrid study has recently mapped a network of 2,846 interactions involving 1,312 proteins within plasmodium6. this network adds to a growing collection of available interaction maps for a number of different organisms, and raises questions about whether the divergence of plasmodium at the sequence level is reflected in the configuration of its protein network. here we examine the degree of conservation between the plasmodium protein network and those of model organisms. although we find 29 highly connected protein complexes specific to the network of the pathogen, we find very little conservation with complexes observed in other organisms (three in yeast, none in the others). overall, the patterns of protein interaction in plasmodium, like its genome sequence, set it apart from other species.responses to extracellular stress directly confer survival fitness by means of complex regulatory networks. despite their complexity, the networks must be evolvable because of changing ecological and environmental pressures. although the regulatory networks underlying stress responses are characterized extensively, their mechanism of evolution remains poorly understood. here, we examine the evolution of three candidate stress response networks (chemotaxis, competence for {dna} uptake, and endospore formation) by analyzing their phylogenetic distribution across several hundred diverse bacterial and archaeal lineages. we report that genes in the chemotaxis and sporulation networks group into well defined evolutionary modules with distinct functions, phenotypes, and substitution rates as compared with control sets of randomly chosen genes. the evolutionary modules vary in both number and cohesiveness among the three pathways. chemotaxis has five coherent modules whose distribution among species shows a clear pattern of interdependence and rewiring. sporulation, by contrast, is nearly monolithic and seems to be inherited vertically, with three weak modules constituting early and late stages of the pathway. competence does not seem to exhibit well defined modules either at or below the pathway level. many of the detected modules are better understood in engineering terms than in protein functional terms, as we demonstrate using a control-based ontology that classifies gene function according to roles such as  ” sensor,”  ” regulator,” and  ” actuator.” moreover, we show that combinations of the modules predict phenotype, yet surprisingly do not necessarily correlate with phylogenetic inheritance. the architectures of these three pathways are therefore emblematic of different modes and constraints on evolution.transcriptional positive-feedback loops are widely associated with bistability, characterized by two stable expression states that allow cells to respond to analog signals in a digital manner. using a synthetic system in budding yeast, we show that positive feedback involving a promoter with multiple transcription factor ({tf}) binding sites can induce a steady-state bimodal response without cooperative binding of the {tf}. deterministic models of this system do not predict bistability. rather, the bimodal response requires a short-lived {tf} and stochastic fluctuations in the {tf}\\'s expression. multiple binding sites provide these fluctuations. because many promoters possess multiple binding sites and many {tfs} are unstable, positive-feedback loops in gene regulatory networks may exhibit bimodal responses, but not necessarily because of deterministic bistability, as is commonly thought.this paper presents a computational paradigm called {data-driven} markov chain monte carlo ({ddmcmc}) for image segmentation in the bayesian statistical framework. the paper contributes to image segmentation in four aspects. first, it designs efficient and well-balanced markov chain dynamics to explore the complex solution space and, thus, achieves a nearly global optimal solution independent of initial segmentations. second, it presents a mathematical principle and a k-adventurers algorithm for computing multiple distinct solutions from the markov chain sequence and, thus, it incorporates intrinsic ambiguities in image segmentation. third, it utilizes data-driven (bottom-up) techniques, such as clustering and edge detection, to compute importance proposal probabilities, which drive the markov chain dynamics and achieve tremendous speedup in comparison to the traditional jump-diffusion methods. fourth, the {ddmcmc} paradigm provides a unifying framework in which the role of many existing segmentation algorithms, such as, edge detection, clustering, region growing, split-merge, snake/balloon, and region competition, are revealed as either realizing markov chain dynamics or computing importance proposal probabilities. thus, the {ddmcmc} paradigm combines and generalizes these segmentation methods in a principled way. the {ddmcmc} paradigm adopts seven parametric and nonparametric image models for intensity and color at various regions. we test the {ddmcmc} paradigm extensively on both color and gray-level images and some results are reported in this paper.an abstract is not available.the idea that most morphological adaptations can be attributed to changes in the cis-regulation of gene expression levels has been gaining increasing acceptance, despite the fact that only a handful of such cases have so far been demonstrated. moreover, because each of these cases involves only one gene, we lack any understanding of how natural selection may act on cis-regulation across entire pathways or networks. here we apply a genome-wide test for selection on cis-regulation to two subspecies of the mouse mus musculus. we find evidence for lineage-specific selection at over 100 genes involved in diverse processes such as growth, locomotion, and memory. these gene sets implicate candidate genes that are supported by both quantitative trait loci and a validated causality-testing framework, and they predict a number of phenotypic differences, which we confirm in all four cases tested. our results suggest that gene expression adaptation is widespread and that these adaptations can be highly polygenic, involving cis-regulatory changes at numerous functionally related genes. these coordinated adaptations may contribute to divergence in a wide range of morphological, physiological, and behavioral phenotypes. evolution can involve changes that are advantageous—known as adaptations—as well as changes that are neutral or slightly deleterious, which are established through a process of random drift. discerning what specific differences between any two lineages are adaptive is a major goal of evolutionary biology. for gene expression differences, this has traditionally proven to be a challenging question, and previous studies of gene expression adaptation in metazoans have been restricted to the single-gene level. here we present a genome-wide analysis of gene expression evolution in two subspecies of the mouse mus musculus. we find several groups of genes that have likely been subject to selection for up-regulation in a specific lineage. these groups include genes related to mitochondria, growth, locomotion, and memory. analysis of the phenotypes of these mice indicates that these adaptations may have had a significant impact on a wide range of phenotypes.coexpression of genes within a functional module can be conserved at great evolutionary distances, whereas the associated regulatory mechanisms can substantially diverge. for example, ribosomal protein ({rp}) genes are tightly coexpressed in saccharomyces cerevisiae, but the cis and trans factors associated with them are surprisingly diverged across ascomycota fungi. little is known, however, about the functional impact of such changes on actual expression levels or about the selective pressures that affect them. here, we address this question in the context of the evolution of the regulation of {rp} gene expression by using a comparative genomics approach together with cross-species functional assays. we show that an activator (ifh1) and a repressor (crf1) that control {rp} gene regulation in normal and stress conditions in s. cerevisiae are derived from the duplication and subsequent specialization of a single ancestral protein. we provide evidence that this regulatory innovation coincides with the duplication of {rp} genes in a whole-genome duplication ({wgd}) event and may have been important for tighter control of higher levels of {rp} transcripts. we find that subsequent loss of the derived repressor led to the loss of a stress-dependent repression of {rps} in the fungal pathogen candida glabrata. our comparative computational and experimental approach shows how gene duplication can constrain and drive regulatory evolution and provides a general strategy for reconstructing the evolutionary trajectory of gene regulation across species.we describe a systematic genome-wide approach for learning the complex combinatorial code underlying gene expression. our probabilistic approach identifies local {dna}-sequence elements and the positional and combinatorial constraints that determine their context-dependent role in transcriptional regulation. the inferred regulatory rules correctly predict expression patterns for 73\\\\% of genes in saccharomyces cerevisiae, utilizing microarray expression data and sequences in the 800 bp upstream of genes. application to caenorhabditis elegans identifies predictive regulatory elements and combinatorial rules that control the phased temporal expression of transcription factors, histones, and germline specific genes. successful prediction requires diverse and complex rules utilizing {and}, {or}, and {not} logic, with significant constraints on motif strength, orientation, and relative position. this system generates a large number of mechanistic hypotheses for focused experimental validation, and establishes a predictive dynamical framework for understanding cellular behavior from genomic sequence.a transcriptional regulatory network ({trn}) constitutes the collection of regulatory rules that link environmental cues to the transcription state of a cell\\'s genome. we recently proposed a matrix formalism that quantitatively represents a system of such rules (a transcriptional regulatory system [{trs}]) and allows systemic characterization of {trs} properties. the matrix formalism not only allows the computation of the transcription state of the genome but also the fundamental characterization of the input-output mapping that it represents. furthermore, a key advantage of this  ” pseudo-stoichiometric” matrix formalism is its ability to easily integrate with existing stoichiometric matrix representations of signaling and metabolic networks. here we demonstrate for the first time how this matrix formalism is extendable to large-scale systems by applying it to the genome-scale escherichia coli {trs}. we analyze the fundamental subspaces of the regulatory network matrix (r) to describe intrinsic properties of the {trs}. we further use monte carlo sampling to evaluate the e. coli transcription state across a subset of all possible environments, comparing our results to published gene expression data as validation. finally, we present novel in silico findings for the e. coli {trs}, including (1) a gene expression correlation matrix delineating functional motifs; (2) sets of gene ontologies for which regulatory rules governing gene transcription are poorly understood and which may direct further experimental characterization; and (3) the appearance of a distributed {trn} structure, which is in stark contrast to the more hierarchical organization of metabolic networks. cells are comprised of genomic information that encodes for proteins, the basic building blocks underlying all biological processes. a transcriptional regulatory system ({trs}) connects a cell\\'s environmental cues to its genome and in turn determines which genes are turned  ” on” in response to these cues. consequently, {trss} control which proteins of an intracellular biochemical reaction network are present. these systems have been mathematically described, often through boolean expressions that represent the activation or inhibition of gene transcription in response to various inputs. we recently developed a matrix formalism that extends these approaches and facilitates a quantitative representation of the boolean logic underlying a {trs}. we demonstrated on small-scale {trss} that this matrix representation is advantageous in that it facilitates the calculation of unique properties of a given {trs}. here we apply this matrix formalism to the genome-scale escherichia coli {trs}, demonstrating for the first time the predictive power of the approach at a large scale. we use the matrix-based model of e. coli transcriptional regulation to generate novel findings about the system, including new functional motifs; sets of genes whose regulation is poorly understood; and features of the {trs} structure.the gradual modification of transcription circuits over evolutionary time scales is an important source of the diversity of life. over the past decade, studies in animals have shown how seemingly small molecular changes in gene regulation can have large effects on morphology and physiology and how selective pressures can act on these changes. more recently, genome-wide studies, particularly those in single-cell yeasts, have uncovered evidence of extensive transcriptional rewiring, indicating that even closely related organisms regulate their genes using markedly different circuitries.both spatial characteristics and temporal features are often the subjects of concern in physical, social, and biological studies. this work tackles the clustering problems for time course data in which the cluster number and clustering structure change with respect to time, dubbed time-variant clustering. we developed a hierarchical model that simultaneously clusters the objects at every time point and describes the relationships of the clusters between time points. the hidden layer of this model is a generalized form of branching processes. a reversible-jump markov chain monte carlo method was implemented for model inference, and a feature selection procedure was developed. we applied this method to explore an open question in preimplantation embryonic development. our analyses using single-cell gene expression data suggested that the earliest cell fate decision could start at the 4-cell stage in mice, earlier than the commonly thought 8- to 16-cell stage. these results together with independent experimental data from single-cell {rna}-seq provided support against a prevailing hypothesis in mammalian development.we consider the problem of predicting cis-regulatory modules without knowledge of motifs. we formulate this problem in a pragmatic setting, and create over 30 new data sets, using drosophila modules, to use as a \\'benchmark\\'. we propose two new methods for the problem, and evaluate these, as well as two existing methods, on our benchmark. we find that the challenge of predicting cis-regulatory modules ab initio, without any input of relevant motifs, is a realizable goal.quantitative studies of protein abundance rarely span more than a small number of experimental conditions and replicates. in contrast, quantitative studies of transcript abundance often span hundreds of experimental conditions and replicates. this situation exists, in part, because extracting quantitative data from large proteomics datasets is significantly more difficult than reading quantitative data from a gene expression microarray. to address this problem, we introduce two algorithmic advances in the processing of quantitative proteomics data. first, we use space-partitioning data structures to handle the large size of these datasets. second, we introduce techniques that combine graph-theoretic algorithms with space-partitioning data structures to collect relative protein abundance data across hundreds of experimental conditions and replicates. we validate these algorithmic techniques by analyzing several datasets and computing both internal and external measures of quantification accuracy. we demonstrate the scalability of these techniques by applying them to a large dataset that comprises a total of 472 experimental conditions and replicates.10.1105/tpc.107.054700an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsa considerable fraction of gene promoters are bound by multiple transcription factors. it is therefore important to understand how such factors interact in regulating the genes. in this paper, we propose a computational method to identify groups of co-regulated genes and the corresponding regulatory programs of multiple transcription factors from protein- {dna} binding and gene expression data. the key concept is to characterize a regulatory program in terms of two properties of individual transcription factors: the function of a regulator as an activator or a repressor, and its direction of effectiveness as necessary or sufficient. we apply a greedy algorithm to find the regulatory models which best explain the available data. empirical analysis indicates that the inferred regulatory models agree with known combinatorial interactions between regulators and are robust against various parameter choices.many real-world domains are relational in nature, consisting of a set of objects related to each other in complex ways. this paper focuses on predicting the existence and the type of links between entities in such domains. we apply the relational markov network framework of taskar et al. to define a joint probabilistic model over the entire link graph --- entity attributes and links. the application of the {rmn} algorithm to this task requires the definition of probabilistic patterns over...we describe latent dirichlet allocation ({lda}), a generative probabilistic model for collections of  discrete data such as text corpora. {lda} is a three-level hierarchical bayesian model, in which each  item of a collection is modeled as a finite mixture over an underlying set of topics. each topic is, in  turn, modeled as an infinite mixture over an underlying set of topic probabilities. in the context of  text modeling, the topic probabilities provide an explicit representation of a document. we present  efficient approximate inference techniques based on variational methods and an {em} algorithm for  empirical bayes parameter estimation. we report results in document modeling, text classification,  and collaborative filtering, comparing to a mixture of unigrams model and the probabilistic {lsi}  model.transcription factor-binding sites and the cis-regulatory modules they compose are central determinants of gene expression. we previously showed that binding site motifs and modules in proximal promoters can be used to predict a significant portion of mammalian tissue-specific transcription. here, we report on a systematic analysis of promoters controlling tissue-specific expression in heart, kidney, liver, pancreas, skeletal muscle, testis and {cd4} t cells, for both human and mouse. we integrated multiple sources of expression data to compile sets of transcripts with strong evidence for tissue-specific regulation. the analysis of the promoters corresponding to these sets produced a catalog of predicted tissue-specific motifs and modules, and cis-regulatory elements. predicted regulatory interactions are supported by statistical evidence, and provide a foundation for targeted experiments that will improve our understanding of tissue-specific regulatory networks. in a broader context, methods used to construct the catalog provide a model for the analysis of genomic regions that regulate differentially expressed genes.the ongoing analyses of published genome-scale data sets is evidence that different approaches are required to completely mine this data. we report the use of novel tools for both visualization and data set comparison to analyze yeast gene-expression (cell cycle and exit from stationary {phase/g}(0)) and protein-interaction studies. this analysis led to new insights about each data set. for example, g(1)-regulated genes are not co-regulated during exit from stationary phase, indicating that the cells are not synchronized. the tight clustering of other genes during exit from stationary-phase data set further indicates the physiological responses during g(0) exit are separable from cell-cycle events. comparison of the two data sets showed that ribosomal-protein genes cluster tightly during exit from stationary phase, but are found in three significantly different clusters in the cell-cycle data set. two protein-interaction data sets were also compared with the gene-expression data. visual analysis of the complete data sets showed no clear correlation between co-expression of genes and protein interactions, in contrast to published reports examining subsets of the protein-interaction data. neither two-hybrid study identified a large number of interactions between ribosomal proteins, consistent with recent structural data, indicating that for both data sets, the identification of false-positive interactions may be lower than previously thought.spatiotemporal changes in gene expression underlie many evolutionary novelties in nature. however, the evolutionary origins of novel expression patterns, and the transcriptional control elements ( ” enhancers”) that govern them, remain unclear. here, we sought to explore the molecular genetic mechanisms by which new enhancers arise. we undertook a survey of closely related drosophila species to identify recently evolved novel gene expression patterns and traced their evolutionary history. analyses of gene expression in a variety of developing tissues of the drosophila melanogaster species subgroup revealed high rates of expression pattern divergence, including numerous evolutionary losses, heterochronic shifts, and expansions or contractions of expression domains. however, gains of novel expression patterns were much less frequent. one gain was observed for the neprilysin-1 (nep1) gene, which has evolved a unique expression pattern in optic lobe neuroblasts of drosophila santomea. dissection of the nep1 cis-regulatory region localized a newly derived optic lobe enhancer activity to a region of an intron that has accumulated a small number of mutations. the nep1 optic lobe enhancer overlaps with other enhancer activities, from which the novel activity was co-opted. we suggest that the novel optic lobe enhancer evolved by exploiting the cryptic activity of extant regulatory sequences, and this may reflect a general mechanism whereby new enhancers evolve.mammalian chromosomes fold into arrays of megabase-sized topologically associating domains ({tads}), which are arranged into compartments spanning multiple megabases of genomic {dna}. {tads} have internal substructures that are often cell type specific, but their higher-order organization remains elusive. here, we investigate {tad} higher-order interactions with {hi-c} through neuronal differentiation and show that they form a hierarchy of domains-within-domains ({metatads}) extending across genomic scales up to the range of entire chromosomes. we find that {tad} interactions are well captured by tree-like, hierarchical structures irrespective of cell type. {metatad} tree structures correlate with genetic, epigenomic and expression features, and structural tree rearrangements during differentiation are linked to transcriptional state changes. using polymer modelling, we demonstrate that hierarchical folding promotes efficient chromatin packaging without the loss of contact specificity, highlighting a role far beyond the simple need for packing efficiency. {\\\\copyright} 2015 the authors. published under the terms of the {cc} {by} 4.0 license.gain-of-function {idh} mutations are initiating events that define major clinical and prognostic classes of gliomas. mutant {idh} protein produces a new onco-metabolite, 2-hydroxyglutarate, which interferes with iron-dependent hydroxylases, including the {tet} family of 5\\'-methylcytosine hydroxylases. {tet} enzymes catalyse a key step in the removal of {dna} methylation. {idh} mutant gliomas thus manifest a {cpg} island methylator phenotype ({g-cimp}), although the functional importance of this altered epigenetic state remains unclear. here we show that human {idh} mutant gliomas exhibit hypermethylation at cohesin and {ccctc}-binding factor ({ctcf})-binding sites, compromising binding of this methylation-sensitive insulator protein. reduced {ctcf} binding is associated with loss of insulation between topological domains and aberrant gene activation. we specifically demonstrate that loss of {ctcf} at a domain boundary permits a constitutive enhancer to interact aberrantly with the receptor tyrosine kinase gene {pdgfra}, a prominent glioma oncogene. treatment of {idh} mutant gliomaspheres with a demethylating agent partially restores insulator function and downregulates {pdgfra}. conversely, {crispr}-mediated disruption of the {ctcf} motif in {idh} wild-type gliomaspheres upregulates {pdgfra} and increases proliferation. our study suggests that {idh} mutations promote gliomagenesis by disrupting chromosomal topology and allowing aberrant regulatory interactions that induce oncogene expression.{dnasei} footprinting is an established assay for identifying transcription factor ({tf})–{dna} interactions with single base pair resolution. high-throughput {dnase}-seq assays have recently been used to detect in vivo {dnase} footprints across the genome. multiple computational approaches have been developed to identify {dnase}-seq footprints as predictors of {tf} binding. however, recent studies have pointed to a substantial cleavage bias of {dnase} and its negative impact on predictive performance of footprinting. to assess the potential for using {dnase}-seq to identify individual binding sites, we performed {dnase}-seq on deproteinized genomic {dna} and determined sequence cleavage bias. this allowed us to build bias corrected and {tf}-specific footprint models. the predictive performance of these models demonstrated that predicted footprints corresponded to high-confidence {tf}–{dna} interactions. {dnase}-seq footprints were absent under a fraction of {chip}-seq peaks, which we show to be indicative of weaker binding, indirect {tf}–{dna} interactions or possible {chip} artifacts. the modeling approach was also able to detect variation in the consensus motifs that {tfs} bind to. finally, cell type specific footprints were detected within {dnase} hypersensitive sites that are present in multiple cell types, further supporting that footprints can identify changes in {tf} binding that are not detectable using other strategies.global network modeling of distal regulatory interactions is essential in understanding the overall architecture of gene expression programs. here, we developed a bayesian probabilistic model and computational method for global causal network construction with breast cancer as a model. whereas physical regulator binding was well supported by gene expression causality in general, distal elements in intragenic regions or loci distant from the target gene exhibited particularly strong functional effects. modeling the action of long-range enhancers was critical in recovering true biological interactions with increased coverage and specificity overall and unraveling regulatory complexity underlying tumor subclasses and drug responses in particular. transcriptional cancer drivers and risk genes were discovered based on the network analysis of somatic and genetic cancer-related {dna} variants. notably, we observed that the risk genes were functionally downstream of the cancer drivers and were selectively susceptible to network perturbation by tumorigenic changes in their upstream drivers. furthermore, cancer risk alleles tended to increase the susceptibility of the transcription of their associated genes. these findings suggest that transcriptional cancer drivers selectively induce a combinatorial misregulation of downstream risk genes, and that genetic risk factors, mostly residing in distal regulatory regions, increase transcriptional susceptibility to upstream cancer-driving somatic changes.motivation: modeling regulatory networks using expression data observed in a differentiation process may help identify context-specific interactions. the outcome of the current algorithms highly depends on the quality and quantity of a single time-course dataset, and the performance may be compromised for datasets with a limited number of {samples.results}: in this work, we report a multi-layer graphical model that is capable of leveraging many publicly available time-course datasets, as well as a cell lineage-specific data with small sample size, to model regulatory networks specific to a differentiation process. first, a collection of network inference methods are used to predict the regulatory relationships in individual public datasets. then, the inferred directional relationships are weighted and integrated together by evaluating against the cell lineage-specific dataset. to test the accuracy of this algorithm, we collected a time-course {rna}-seq dataset during human erythropoiesis to infer regulatory relationships specific to this differentiation process. the resulting erythroid-specific regulatory network reveals novel regulatory relationships activated in erythropoiesis, which were further validated by genome-wide {tr4} binding studies using {chip}-seq. these erythropoiesis-specific regulatory relationships were not identifiable by single dataset-based methods or context-independent integrations. analysis of the predicted targets reveals that they are all closely associated with hematopoietic lineage {differentiation.availability} and implementation: the predicted erythroid regulatory network is available at {http://guanlab.ccmb.med.umich.edu/data/inferencenetwork}/.contact: {gyuanfan@umich.edusupplementary} information: supplementary data are available at bioinformatics online.identifying control strategies for biological networks is paramount for practical applications that involve reprogramming a cell\\'s fate, such as disease therapeutics and stem cell reprogramming. here we develop a novel network control framework that integrates the structural and functional information available for intracellular networks to predict control targets. formulated in a logical dynamic scheme, our approach drives any initial state to the target state with 100\\\\% effectiveness and needs to be applied only transiently for the network to reach and stay in the desired state. we illustrate our method\\'s potential to find intervention targets for cancer treatment and cell differentiation by applying it to a leukemia signaling network and to the network controlling the differentiation of helper t cells. we find that the predicted control targets are effective in a broad dynamic framework. moreover, several of the predicted interventions are supported by experiments. practical applications in modern molecular and systems biology such as the search for new therapeutic targets for diseases and stem cell reprogramming have generated a great interest in controlling the internal dynamics of a cell. here we present a network control approach that integrates the structural and functional information of the network. we show that stabilizing the expression or activity of a few select components can drive the cell towards a desired fate or away from an undesired fate. we demonstrate our method\\'s effectiveness by applying it to a type of blood cell cancer and to the differentiation of a type of immune cell. overall, our approach provides new insights into how to control the dynamics of intracellular networks.current approaches for identification and detection of transcription factor binding sites rely on an extensive set of known target genes. here we describe a novel structure-based approach applicable to transcription factors with no prior binding data. our approach combines sequence data and structural information to infer context-specific amino acid–nucleotide recognition preferences. these are used to predict binding sites for novel transcription factors from the same structural family. we demonstrate our approach on the {cys2his2} zinc finger protein family, and show that the learned {dna}-recognition preferences are compatible with experimental results. we use these preferences to perform a genome-wide scan for direct targets of drosophila melanogaster {cys2his2} transcription factors. by analyzing the predicted targets along with gene annotation and expression data we infer the function and activity of these proteins. cells respond to dynamic changes in their environment by invoking various cellular processes, coordinated by a complex regulatory program. a main component of this program is the regulation of transcription, which is mainly accomplished by transcription factors that bind the {dna} in the vicinity of genes. to better understand transcriptional regulation, advanced computational approaches are needed for linking between transcription factors and their targets. the authors describe a novel approach by which the binding site of a given transcription factor can be characterized without previous experimental binding data. this approach involves learning a set of context-specific amino acid–nucleotide recognition preferences that, when combined with the sequence and structure of the protein, can predict its specific binding preferences. applying this approach to the {cys2his2} zinc finger protein family demonstrated its genome-wide potential by automatically predicting the direct targets of 29 regulators in the genome of the fruit fly drosophila melanogaster. at present, with the availability of many genome sequences, there are numerous proteins annotated as transcription factors based on their sequence alone. this approach offers a promising direction for revealing the targets of these factors and for understanding their roles in the cellular network.{abstractvarious} algorithms are available for predicting {mrna} expression and modeling gene regulatory processes. they differ in whether they rely on the existence of modules of coregulated genes or build a model that applies to all genes, whether they represent regulatory activities as hidden variables or as {mrna} levels, and whether they implicitly or explicitly model the complex cis-regulatory logic of multiple interacting transcription factors binding the same {dna}. the fact that functional genomics data of different types reflect the same molecular processes provides a natural strategy for integrative computational analysis. one promising avenue toward an accurate and comprehensive model of gene regulation combines biophysical modeling of the interactions among proteins, {dna}, and {rna} with the use of large-scale functional genomics data to estimate regulatory network connectivity and activity parameters. as the ability of these models to represent complex cis-regulatory logic increases, the need for approaches based on cross-species conservation may diminish.statistical applications in fields such as bioinformatics, information retrieval, speech processing, image processing and communications often involve large-scale models in which thousands or millions of random variables are linked in complex ways. graphical models provide a general methodology for approaching these problems, and indeed many of the models developed by researchers in these applied fields are instances of the general graphical model formalism. we review some of the basic ideas underlying graphical models, including the algorithmic ideas that allow graphical models to be deployed in large-scale data analysis problems. we also present examples of graphical models in bioinformatics, error-control coding and language processing. key words and phrases: probabilistic graphical models, junction tree algorithm, sum-product algorithm, markov chain monte carlo, variational inference, bioinformatics, error-control coding.the chromatin interaction assays {5c} and {hic} have advanced our understanding of genomic spatial organization, but analysis approaches for these data are limited by usability and flexibility. the {hifive} tool suite provides efficient data handling and a variety of normalization approaches for easy, fast analysis and method comparison. integration of {mpi}-based parallelization allows scalability and rapid processing time. in addition to single-command analysis of an entire experiment from mapped reads to interaction values, {hifive} has been integrated into the open-source, web-based platform galaxy to connect users with computational resources and a graphical interface. {hifive} is open-source software available from http://taylorlab.org/software/hifive/.identifying the nucleotides that cause gene expression variation is a critical step in dissecting the genetic basis of complex traits. here, we focus on polymorphisms that are predicted to alter transcription factor binding sites ({tfbss}) in the yeast, saccharomyces cerevisiae. we assembled a confident set of transcription factor motifs using recent protein binding microarray and {chip}-chip data and used our collection of motifs to predict a comprehensive set of {tfbss} across the s. cerevisiae genome. we used a population genomics analysis to show that our predictions are accurate and significantly improve on our previous annotation. although predicting gene expression from sequence is thought to be difficult in general, we identified a subset of genes for which changes in predicted {tfbss} correlate well with expression divergence between yeast strains. our analysis thus demonstrates both the accuracy of our new {tfbs} predictions and the feasibility of using simple models of gene regulation to causally link differences in gene expression to variation at individual nucleotides.regulatory divergence is likely a major driving force in evolution. comparative genomics is being increasingly used to infer the evolution of gene regulation. ascomycota fungi are uniquely suited among eukaryotes for regulatory evolution studies, due to broad phylogenetic scope, many sequenced genomes, and tractability of genomic analysis. here we review recent advances in the identification of the contribution of cis- and trans-factors to expression divergence. whereas current strategies have led to the discovery of surprising signatures and mechanisms, we still understand very little about the adaptive role of regulatory evolution. empirical studies including experimental evolution, comparative functional genomics and hybrid and engineered strains are showing early promise toward deciphering the contribution of regulatory divergence to adaptation.inference of gene regulatory network from expression data is a challenging task. many methods have been developed to this purpose but a comprehensive evaluation that covers unsupervised, semi-supervised and supervised methods, and provides guidelines for their practical application, is lacking.duplication of genes encoding transcription factors plays an essential role in driving phenotypic variation. because regulation can occur at multiple levels, it is often difficult to discern how each duplicated factor achieves its regulatory specificity. in these cases, a \"systems approach\" may distinguish the role of each factor by integrating complementary large-scale measurements of the regulatory network. to explore such an approach, we integrate growth phenotypes, promoter binding profiles, and gene expression patterns to model the {dna} damage response network controlled by the yeast-specific {ap}-1 ({yap}) family of transcription factors. this analysis reveals that {yap} regulatory specificity is achieved by at least three mechanisms: (i) divergence of {dna}-binding sequences into two subfamilies; (ii) condition-specific combinatorial regulation by multiple yap factors; and (iii) interactions of yap 1, 4, and 6 with chromatin remodeling proteins. additional microarray experiments establish that yap 4 and 6 regulate gene expression through interactions with the histone deacetylase, hda1. the data further highlight differences among yap paralogs in terms of their regulatory mode of action (activation vs. repression). this study suggests how other large {tf} families might be disentangled in the future.reduced activity of two genes in combination often has a more detrimental effect than expected. such epistatic interactions not only occur when genes are mutated but also due to variation in gene expression, including among isogenic individuals in a controlled environment. we hypothesized that these \\'epigenetic\\' epistatic interactions could place important constraints on the evolution of gene expression. consistent with this, we show here that yeast genes with many epistatic interaction partners typically show low expression variation among isogenic individuals and low variation across different conditions. in addition, their expression tends to remain stable in response to the accumulation of mutations and only diverges slowly between strains and species. yeast promoter architectures, the retention of gene duplicates, and the divergence of expression between humans and chimps are also consistent with selective pressure to reduce the likelihood of harmful epigenetic epistatic interactions. based on these and previous analyses, we propose that the tight regulation of epistatic interaction network hubs makes an important contribution to the maintenance of a robust, \\'canalized\\' phenotype. moreover, that epigenetic epistatic interactions may contribute substantially to fitness defects when single genes are deleted.this paper addresses three questions. is it useful to attempt to learn a bayesian network structure with hundreds of thousands of nodes? how should such structure search proceed practically? the third question arises out of our approach to the second: how can frequent sets (agrawal et al., 1993), which are extremely popular in the area of descriptive data mining, be turned into a probabilistic model? large sparse datasets with hundreds of thousands of records and attributes appear in...sequencing-based gene expression methods like {rna}-sequencing ({rna}-seq) have become increasingly common, but it is often claimed that results obtained in different studies are not comparable owing to the influence of laboratory batch effects, differences in {rna} extraction and sequencing library preparation methods and bioinformatics processing pipelines. it would be unfortunate if different experiments were in fact incomparable, as there is great promise in data fusion and meta-analysis applied to sequencing data sets. we therefore compared reported gene expression measurements for ostensibly similar samples (specifically, human brain, heart and kidney samples) in several different {rna}-seq studies to assess their overall consistency and to examine the factors contributing most to systematic differences. the same comparisons were also performed after preprocessing all data in a consistent way, eliminating potential bias from bioinformatics pipelines. we conclude that published human tissue {rna}-seq expression measurements appear relatively consistent in the sense that samples cluster by tissue rather than laboratory of origin given simple preprocessing transformations. the article is supplemented by a detailed walkthrough with embedded r code and figures.proteins are traditionally identified on the basis of their individual actions as catalysts, signalling molecules, or building blocks in cells and microorganisms. but our post-genomic view is expanding the protein\\'s role into an element in a network of protein–protein interactions as well, in which it has a contextual or cellular function within functional modules1, 2. here we provide quantitative support for this idea by demonstrating that the phenotypic consequence of a single gene deletion in the yeast saccharomyces cerevisiae is affected to a large extent by the topological position of its protein product in the complex hierarchical web of molecular interactions.chromatin immunoprecipitation experiments followed by sequencing ({chip}–seq) detect {protein–dna} binding events and chemical modifications of histone proteins. challenges in the standard {chip}–seq protocol have motivated recent enhancements in this approach, such as reducing the number of cells that are required and increasing the resolution. complementary experimental approaches — for example, {dnasei} hypersensitive site mapping and analysis of chromatin interactions that are mediated by particular proteins — provide additional information about {dna}-binding proteins and their function. these data are now being used to identify variability in the functions of {dna}-binding proteins across genomes and individuals. in this review, i describe the latest advances in methods to detect and functionally characterize {dna}-bound proteins.networks play a crucial role in computational biology, yet their analysis and representation is still an open problem. power graph analysis is a lossless transformation of biological networks into a compact, less redundant representation, exploiting the abundance of cliques and bicliques as elementary topological motifs. we demonstrate with five examples the advantages of power graph analysis. investigating protein-protein interaction networks, we show how the catalytic subunits of the casein kinase {ii} complex are distinguishable from the regulatory subunits, how interaction profiles and sequence phylogeny of {sh3} domains correlate, and how false positive interactions among high-throughput interactions are spotted. additionally, we demonstrate the generality of power graph analysis by applying it to two other types of networks. we show how power graphs induce a clustering of both transcription factors and target genes in bipartite transcription networks, and how the erosion of a phosphatase domain in type 22 non-receptor tyrosine phosphatases is detected. we apply power graph analysis to high-throughput protein interaction networks and show that up to 85\\\\% (56\\\\% on average) of the information is redundant. experimental networks are more compressible than rewired ones of same degree distribution, indicating that experimental networks are rich in cliques and bicliques. power graphs are a novel representation of networks, which reduces network complexity by explicitly representing re-occurring network motifs. power graphs compress up to 85\\\\% of the edges in protein interaction networks and are applicable to all types of networks such as protein interactions, regulatory networks, or homology networks.motivation: significant efforts have been made to acquire data under different conditions and to construct static networks that can explain various gene regulation mechanisms. however, gene regulatory networks are dynamic and condition-specific; under different conditions, networks exhibit different regulation patterns accompanied by different transcriptional network topologies. thus, an investigation on the topological changes in transcriptional networks can facilitate the understanding of cell development or provide novel insights into the pathophysiology of certain diseases, and help identify the key genetic players that could serve as biomarkers or drug {targets.results}: here, we report a differential dependency network ({ddn}) analysis to detect statistically significant topological changes in the transcriptional networks between two biological conditions. we propose a local dependency model to represent the local structures of a network by a set of conditional probabilities. we develop an efficient learning algorithm to learn the local dependency model using the lasso technique. a permutation test is subsequently performed to estimate the statistical significance of each learned local structure. in testing on a simulation dataset, the proposed algorithm accurately detected all the genes with network topological changes. the method was then applied to the estrogen-dependent {t-47d} estrogen receptor-positive ({er}+) breast cancer cell line datasets and human and mouse embryonic stem cell datasets. in both experiments using real microarray datasets, the proposed method produced biologically meaningful results. we expect {ddn} to emerge as an important bioinformatics tool in transcriptional network analyses. while we focus specifically on transcriptional networks, the {ddn} method we introduce here is generally applicable to other biological networks with similar {characteristics.availability}: the {ddn} {matlab} toolbox and experiment data are available at {http://www.cbil.ece.vt.edu/software.htm.contact}: {yuewang@vt.edusupplementary} information: supplementary data are available at bioinformatics online.genomic and proteomic approaches can provide hypotheses concerning function for the large number of genes predicted from genome sequences. because of the artificial nature of the assays, however, the information from these high-throughput approaches should be considered with caution. although it is possible that more meaningful hypotheses could be formulated by integrating the data from various functional genomic and proteomic projects, it has yet to be seen to what extent the data can be correlated and how such integration can be achieved. we developed a \\'transcriptome-interactome correlation mapping\\' strategy to compare the interactions between proteins encoded by genes that belong to common expression-profiling clusters with those between proteins encoded by genes that belong to different clusters. using this strategy with currently available data sets for saccharomyces cerevisiae, we provide the first global evidence that genes with similar expression profiles are more likely to encode interacting proteins. we show how this correlation between transcriptome and interactome data can be used to improve the quality of hypotheses based on the information from both approaches. the strategy described here may help to integrate other functional genomic and proteomic data, both in yeast and in higher organisms.the sequence {alignment/map} ({sam}) format is a generic alignment format for storing read alignments against reference sequences, supporting short and long reads (up to 128 mbp) produced by different sequencing platforms. it is flexible in style, compact in size, efficient in random access and is the format in which alignments from the 1000 genomes project are released. {samtools} implements various utilities for post-processing alignments in the {sam} format, such as indexing, variant caller and alignment viewer, and thus provides universal tools for processing read alignments. http://samtools.sourceforge.net.reconstructions of cellular metabolism are publicly available for a variety of different microorganisms and some mammalian genomes. to date, these reconstructions are \"genome-scale\" and strive to include all reactions implied by the genome annotation, as well as those with direct experimental evidence. clearly, many of the reactions in a genome-scale reconstruction will not be active under particular conditions or in a particular cell type. methods to tailor these comprehensive genome-scale reconstructions into context-specific networks will aid predictive in silico modeling for a particular situation. we present a method called gene inactivity moderated by metabolism and expression ({gimme}) to achieve this goal. the {gimme} algorithm uses quantitative gene expression data and one or more presupposed metabolic objectives to produce the context-specific reconstruction that is most consistent with the available data. furthermore, the algorithm provides a quantitative inconsistency score indicating how consistent a set of gene expression data is with a particular metabolic objective. we show that this algorithm produces results consistent with biological experiments and intuition for adaptive evolution of bacteria, rational design of metabolic engineering strains, and human skeletal muscle cells. this work represents progress towards producing constraint-based models of metabolism that are specific to the conditions where the expression profiling data is available.motivation: microarrays are widely used to measure gene expression differences between sets of biological samples. many of these differences will be due to differences in the activities of transcription factors. in principle, these differences can be detected by associating motifs in promoters with differences in gene expression levels between the groups. in practice, this is hard to do.biological networks have an inherent simplicity: they are modular with a design that can be separated into units that perform almost independently. furthermore, they show reuse of recurring patterns termed network motifs. little is known about the evolutionary origin of these properties. current models of biological evolution typically produce networks that are highly nonmodular and lack understandable motifs. here, we suggest a possible explanation for the origin of modularity and network motifs in biology. we use standard evolutionary algorithms to evolve networks. a key feature in this study is evolution under an environment (evolutionary goal) that changes in a modular fashion. that is, we repeatedly switch between several goals, each made of a different combination of subgoals. we find that such  ” modularly varying goals” lead to the spontaneous evolution of modular network structure and network motifs. the resulting networks rapidly evolve to satisfy each of the different goals. such switching between related goals may represent biological evolution in a changing environment that requires different combinations of a set of basic biological functions. the present study may shed light on the evolutionary forces that promote structural simplicity in biological networks and offers ways to improve the evolutionary design of engineered systems.evolutionary changes in gene expression are a main driver of phenotypic evolution. in yeast, genes that have rapidly diverged in expression are associated with particular promoter features, including the presence of a {tata} box, a nucleosome-covered promoter and unstable tracts of tandem repeats. here, we discuss how these promoter properties may confer an inherent capacity for flexibility of expression.cells have self-organizing properties that control their behavior in complex tissues. contact between cells expressing either b-type eph receptors or their transmembrane ephrin ligands initiates bidirectional signals that regulate cell positioning. however, simultaneously investigating how information is processed in two interacting cell types remains a challenge. we implemented a proteomic strategy to systematically determine cell-specific signaling networks underlying {ephb2}- and {ephrin-b1}-controlled cell sorting. quantitative mass spectrometric analysis of mixed populations of {ephb2}- and {ephrin-b1}-expressing cells that were labeled with different isotopes revealed cell-specific tyrosine phosphorylation events. functional associations between these phosphotyrosine signaling networks and cell sorting were established with small interfering {rna} screening. data-driven network modeling revealed that signaling between mixed {ephb2}- and {ephrin-b1}-expressing cells is asymmetric and that the distinct cell types use different tyrosine kinases and targets to process signals induced by cell-cell contact. we provide systems- and cell-specific network models of contact-initiated signaling between two distinct cell types. 10.1126/science.1176615biological networks change in response to genetic and environmental cues. changes are reflected in the abundances of biomolecules, the composition of protein complexes and other descriptors of the biological state. methods to infer the dynamic state of a cell would have great value for understanding how cells change over time to accomplish biological goals. a new method predicts the dynamic state of protein complexes in a cell, with protein expression inferred from transcription profile time courses and protein complexes inferred by joint analysis of protein co-expression and protein-protein interaction maps. two algorithmic advances are presented: a new method, {dhac} (dynamical hierarchical agglomerative clustering), for clustering time-evolving networks; and a companion method, {match}-{em}, for matching corresponding clusters across time points. with link prediction as an objective assessment metric, {dhac} provides a substantial advance over existing clustering methods. an application to the yeast metabolic cycle demonstrates how waves of gene expression correspond to individual protein complexes. our results suggest regulatory mechanisms for assembling the mitochondrial ribosome and illustrate dynamic changes in the components of the nuclear pore. all source code and data are available under the boost software license as supplementary material, at www.baderzone.org, and at sourceforge.net/projects/dhacdist.the comparison of related genomes has emerged as a powerful lens for genome interpretation. here we report the sequencing and comparative analysis of 29 eutherian genomes. we confirm that at least 5.5\\\\% of the human genome has undergone purifying selection, and locate constrained elements covering ∼4.2\\\\% of the genome. we use evolutionary signatures and comparisons with experimental data sets to suggest candidate functions for ∼60\\\\% of constrained bases. these elements reveal a small number of new coding exons, candidate stop codon readthrough events and over 10,000 regions of overlapping synonymous constraint within protein-coding exons. we find 220 candidate {rna} structural families, and nearly a million elements overlapping potential promoter, enhancer and insulator regions. we report specific amino acid residues that have undergone positive selection, 280,000 non-coding elements exapted from mobile elements and more than 1,000 primate- and human-accelerated elements. overlap with disease-associated variants indicates that our findings will be relevant for studies of human biology, health and disease.algorithms for learning bayesian networks from data have two components: a scoring metric and a search procedure. the scoring metric computes a score reflecting the goodness-of-fit of the structure to the data. the search procedure tries to identify network structures with high scores. heckerman et al. (1994) introduced a bayesian metric, called the {bde} metric, that computes the relative posterior probability of a network structure given data. they show that the metric has a property ...current yeast interactome network maps contain several hundred molecular complexes with limited and somewhat controversial representation of direct binary interactions. we carried out a comparative quality assessment of current yeast interactome data sets, demonstrating that high-throughput yeast two-hybrid ({y2h}) screening provides high-quality binary interaction information. because a large fraction of the yeast binary interactome remains to be mapped, we developed an empirically controlled mapping framework to produce a \"second-generation\" high-quality, high-throughput {y2h} data set covering approximately 20\\\\% of all yeast binary interactions. both {y2h} and affinity purification followed by mass spectrometry ({ap}/{ms}) data are of equally high quality but of a fundamentally different and complementary nature, resulting in networks with different topological and biological properties. compared to co-complex interactome models, this binary map is enriched for transient signaling interactions and intercomplex connections with a highly significant clustering between essential proteins. rather than correlating with essentiality, protein connectivity correlates with genetic pleiotropy.gene class, ontology, or pathway testing analysis has become increasingly popular in microarray data analysis. such approaches allow the integration of gene annotation databases, such as gene ontology and {kegg} pathway, to formally test for subtle but coordinated changes at a system level. higher power in gene class testing is gained by combining weak signals from a number of individual genes in each pathway. we propose an alternative approach for gene-class testing based on mixed models, a class of statistical models that:to understand biology at the system level, we must examine the structure and dynamics of cellular and organismal function, rather than the characteristics of isolated parts of a cell or organism. properties of systems, such as robustness, emerge as central issues, and understanding these properties may have an impact on the future of medicine. however, many breakthroughs in experimental devices, advanced software, and analytical methods are required before the achievements of systems biology can live up to their much-touted potential.the characterization of functional redundancy and divergence between duplicate genes is an important step in understanding the evolution of genetic systems. large-scale genetic network analysis in saccharomyces cerevisiae provides a powerful perspective for addressing these questions through quantitative measurements of genetic interactions between pairs of duplicated genes, and more generally, through the study of genome-wide genetic interaction profiles associated with duplicated genes. we show that duplicate genes exhibit fewer genetic interactions than other genes because they tend to buffer one another functionally, whereas observed interactions are non-overlapping and reflect their divergent roles. we also show that duplicate gene pairs are highly imbalanced in their number of genetic interactions with other genes, a pattern that appears to result from asymmetric evolution, such that one duplicate evolves or degrades faster than the other and often becomes functionally or conditionally specialized. the differences in genetic interactions are predictive of differences in several other evolutionary and physiological properties of duplicate pairs.the complexity of cellular gene, protein, and metabolite networks can hinder attempts to elucidate their structure and function. to address this problem, we used systematic transcriptional perturbations to construct a first-order model of regulatory interactions in a nine-gene subnetwork of the {sos} pathway in escherichia coli. the model correctly identified the major regulatory genes and the transcriptional targets of mitomycin c activity in the subnetwork. this approach, which is experimentally and computationally scalable, provides a framework for elucidating the functional properties of genetic networks and identifying molecular targets of pharmacological compounds.the solution of high-dimensional inference and prediction problems in computational biology is almost always a compromise between mathematical theory and practical constraints, such as limited computational resources. as time progresses, computational power increases but well-established inference methods often remain locked in their initial suboptimal solution. we revisit the approach of segal et al. to infer regulatory modules and their condition-specific regulators from gene expression data. in contrast to their direct optimization-based solution, we use a more representative centroid-like solution extracted from an ensemble of possible statistical models to explain the data. the ensemble method automatically selects a subset of most informative genes and builds a quantitatively better model for them. genes which cluster together in the majority of models produce functionally more coherent modules. regulators which are consistently assigned to a module are more often supported by literature, but a single model always contains many regulator assignments not supported by the ensemble. reliably detecting condition-specific or combinatorial regulation is particularly hard in a single optimum but can be achieved using ensemble averaging. all software developed for this study is available from http://bioinformatics.psb.ugent.be/software.we show that, in yeast, the divergence rate of gene expression is not correlated with that of its associated coding sequence. gene essentiality influences both modes of evolution, but other properties related to protein structure or promoter composition are only correlated with coding-sequence divergence or gene expression divergence, respectively. based on these findings, we discuss the possibilities of neutral evolution of gene expression and of different modes of evolution in unicellular versus multicellular organisms.the inference of transcriptional networks that regulate transitions into physiological or pathological cellular states remains a central challenge in systems biology. a mesenchymal phenotype is the hallmark of tumour aggressiveness in human malignant glioma, but the regulatory programs responsible for implementing the associated molecular signature are largely unknown. here we show that reverse-engineering and an unbiased interrogation of a glioma-specific regulatory network reveal the transcriptional module that activates expression of mesenchymal genes in malignant glioma. two transcription factors ({c/ebpβ} and {stat3}) emerge as synergistic initiators and master regulators of mesenchymal transformation. ectopic co-expression of {c/ebpβ} and {stat3} reprograms neural stem cells along the aberrant mesenchymal lineage, whereas elimination of the two factors in glioma cells leads to collapse of the mesenchymal signature and reduces tumour aggressiveness. in human glioma, expression of {c/ebpβ} and {stat3} correlates with mesenchymal differentiation and predicts poor clinical outcome. these results show that the activation of a small regulatory module is necessary and sufficient to initiate and maintain an aberrant phenotypic state in cancer cells.understanding how genes are expressed and regulated in different tissues is a fundamental and challenging question. however, most of currently available biological databases do not focus on tissue-specific gene regulation.{background}:evolutionary conservation has been used successfully to help identify cis-acting {dna} regions that are important in regulating tissue-specific gene expression. motivated by increasing evidence that some {dna} regulatory regions are not evolutionary conserved, we have developed an approach for cis-regulatory region identification that does not rely upon evolutionary sequence {conservation.results}:the conservation-independent approach is based on an empirical potential energy between interacting transcription factors ({tfs}). in this analysis, the potential energy is defined as a function of the number of {tf} interactions in a genomic region and the strength of the interactions. by identifying sets of interacting {tfs}, the analysis locates regions enriched with the binding sites of these interacting {tfs}. we applied this approach to 30 human tissues and identified 6232 putative cis-regulatory modules ({crms}) regulating 2130 tissue-specific genes. interestingly, some genes appear to be regulated by different {crms} in different tissues. known regulatory regions are highly enriched in our predicted {crms}. in addition, {dnase} i hypersensitive sites, which tend to be associated with active regulatory regions, significantly overlap with the predicted {crms}, but not with more conserved regions. we also find that conserved and non-conserved {crms} regulate distinct gene groups. conserved {crms} control more essential genes and genes involved in fundamental cellular activities such as transcription. in contrast, non-conserved {crms}, in general, regulate more non-essential genes, such as genes related to neural {activity.conclusion}:these results demonstrate that identifying relevant sets of binding motifs can help in the mapping of {dna} regulatory regions, and suggest that non-conserved {crms} play an important role in gene regulation.increasing knowledge about the organization of proteins into complexes, systems, and pathways has led to a flowering of theoretical approaches for exploiting this knowledge in order to better learn the functions of proteins and their roles underlying phenotypic traits and diseases. much of this body of theory has been developed and tested in model organisms, relying on their relative simplicity and genetic and biochemical tractability to accelerate the research. in this review, we discuss several of the major approaches for computationally integrating proteomics and genomics observations into integrated protein networks, then applying guilt-by-association in these networks in order to identify genes underlying traits. recent trends in this field include a rising appreciation of the modular network organization of proteins underlying traits or mutational phenotypes, and how to exploit such protein modularity using computational approaches related to the internet search algorithm {pagerank}. many protein network-based predictions have recently been experimentally confirmed in yeast, worms, plants, and mice, and several successful approaches in model organisms have been directly translated to analyze human disease, with notable recent applications to glioma and breast cancer prognosis.we explored genomic expression patterns in the yeast saccharomyces cerevisiae responding to diverse environmental transitions. {dna} microarrays were used to measure changes in transcript levels over time for almost every yeast gene, as cells responded to temperature shocks, hydrogen peroxide, the superoxide-generating drug menadione, the sulfhydryl-oxidizing agent diamide, the disulfide-reducing agent dithiothreitol, hyper- and hypo-osmotic shock, amino acid starvation, nitrogen source depletion, and progression into stationary phase. a large set of genes (\\\\~{} 900) showed a similar drastic response to almost all of these environmental changes. additional features of the genomic responses were specialized for specific conditions. promoter analysis and subsequent characterization of the responses of mutant strains implicated the transcription factors yap1p, as well as msn2p and msn4p, in mediating specific features of the transcriptional response, while the identification of novel sequence elements provided clues to novel regulators. physiological themes in the genomic responses to specific environmental stresses provided insights into the effects of those stresses on the cell.bacteria adapt to environmental stimuli by adjusting their transcriptomes in a complex manner, the full potential of which has yet to be established for any individual bacterial species. here, we report the transcriptomes of bacillus subtilis exposed to a wide range of environmental and nutritional conditions that the organism might encounter in nature. we comprehensively mapped transcription units ({tus}) and grouped 2935 promoters into regulons controlled by various {rna} polymerase sigma factors, accounting for \\\\~{}66\\\\% of the observed variance in transcriptional activity. this global classification of promoters and detailed description of {tus} revealed that a large proportion of the detected antisense {rnas} arose from potentially spurious transcription initiation by alternative sigma factors and from imperfect control of transcription termination.chromatin immunoprecipitation followed by sequencing ({chip}-seq) is a technique for genome-wide profiling of {dna}-binding proteins, histone modifications or nucleosomes. owing to the tremendous progress in next-generation sequencing technology, {chip}-seq offers higher resolution, less noise and greater coverage than its array-based predecessor {chip}-chip. with the decreasing cost of sequencing, {chip}-seq has become an indispensable tool for studying gene regulation and epigenetic mechanisms. in this review, i describe the benefits and challenges in harnessing this technique with an emphasis on issues related to experimental design and data analysis. {chip}-seq experiments generate large quantities of data, and effective computational analysis will be crucial for uncovering biological mechanisms.{jaspar} (http://jaspar.genereg.net) is the leading open-access database of matrix profiles describing the {dna}-binding patterns of transcription factors ({tfs}) and other proteins interacting with {dna} in a sequence-specific manner. its fourth major release is the largest expansion of the core database to date: the database now holds 457 non-redundant, curated profiles. the new entries include the first batch of profiles derived from {chip}-seq and {chip}-chip whole-genome binding experiments, and 177 yeast {tf} binding profiles. the introduction of a yeast division brings the convenience of {jaspar} to an active research community. as binding models are refined by newer data, the {jaspar} database now uses versioning of matrices: in this release, 12\\\\% of the older models were updated to improved versions. classification of {tf} families has been improved by adopting a new {dna}-binding domain nomenclature. a curated catalog of mammalian {tfs} is provided, extending the use of the {jaspar} profiles to additional {tfs} belonging to the same structural family. the changes in the database set the system ready for more rapid acquisition of new high-throughput data sources. additionally, three new special collections provide matrix profile data produced by recent alternative high-throughput approaches.in recent years, spectral clustering has become one of the most popular modern clustering algorithms. it is simple to implement, can be solved efficiently by standard linear algebra software, and very often outperforms traditional clustering algorithms such as the k-means algorithm. on the first glance spectral clustering appears slightly mysterious, and it is not obvious to see why it works at all and what it really does. the goal of this tutorial is to give some intuition on those questions. we describe different graph laplacians and their basic properties, present the most common spectral clustering algorithms, and derive those algorithms from scratch by several different approaches. advantages and disadvantages of the different spectral clustering algorithms are discussed.{ctcf}-binding locations represent regulatory sequences that are highly constrained over the course of evolution. to gain insight into how these {dna} elements are conserved and spread through the genome, we defined the full spectrum of {ctcf}-binding sites, including a 33/34-mer motif, and identified over five thousand highly conserved, robust, and tissue-independent {ctcf}-binding locations by comparing {chip}-seq data from six mammals. our data indicate that activation of retroelements has produced species-specific expansions of {ctcf} binding in rodents, dogs, and opossum, which often functionally serve as chromatin and transcriptional insulators. we discovered fossilized repeat elements flanking deeply conserved {ctcf}-binding regions, indicating that similar retrotransposon expansions occurred hundreds of millions of years ago. repeat-driven dispersal of {ctcf} binding is a fundamental, ancient, and still highly active mechanism of genome evolution in mammalian lineages. \\\\^{a}\\x96º {ctcf}-binding locations are highly conserved across mammals \\\\^{a}\\x96º new locations for {ctcf} binding are carried by {sine} repeats in many mammals \\\\^{a}\\x96º ancient and newly born {ctcf}-binding events similarly demarcate chromatin barriers \\\\^{a}\\x96º retroelements can reposition organizing elements throughout the genomethe spatial organization of the genome is intimately linked to its biological function, yet our understanding of higher order genomic structure is coarse, fragmented and incomplete. in the nucleus of eukaryotic cells, interphase chromosomes occupy distinct chromosome territories, and numerous models have been proposed for how chromosomes fold within chromosome territories. these models, however, provide only few mechanistic details about the relationship between higher order chromatin structure and genome function. recent advances in genomic technologies have led to rapid advances in the study of three-dimensional genome organization. in particular, {hi-c} has been introduced as a method for identifying higher order chromatin interactions genome wide. here we investigate the three-dimensional organization of the human and mouse genomes in embryonic stem cells and terminally differentiated cell types at unprecedented resolution. we identify large, megabase-sized local chromatin interaction domains, which we term \\'topological domains\\', as a pervasive structural feature of the genome organization. these domains correlate with regions of the genome that constrain the spread of heterochromatin. the domains are stable across different cell types and highly conserved across species, indicating that topological domains are an inherent property of mammalian genomes. finally, we find that the boundaries of topological domains are enriched for the insulator binding protein {ctcf}, housekeeping genes, transfer {rnas} and short interspersed element ({sine}) retrotransposons, indicating that these factors may have a role in establishing the topological domain structure of the genome.identification of active gene regulatory elements is a key to understanding transcriptional control governing biological processes such as cell-type specificity, differentiation, development, proliferation, and response to the environment. mapping {dnase} i hypersensitive ({hs}) sites has historically been a valuable tool for identifying all different types of regulatory elements, including promoters, enhancers, silencers, insulators, and locus control regions. this method utilizes {dnase} i to selectively digest nucleosome-depleted {dna} (presumably by transcription factors), whereas {dna} regions tightly wrapped in nucleosome and higher-order structures are more resistant. the traditional low-throughput method for identifying {dnase} i {hs} sites uses southern blots. here, we describe the complete and improved protocol for {dnase}-seq, a high-throughput method that identifies {dnase} i {hs} sites across the whole genome by capturing {dnase}-digested fragments and sequencing them by high-throughput, next-generation sequencing. in a single experiment, {dnase}-seq can identify most active regulatory regions from potentially any cell type, from any species with a sequenced genome.super-enhancers are large clusters of transcriptional enhancers that drive expression of genes that define cell identity. improved understanding of the roles that super-enhancers play in biology would be afforded by knowing the constellation of factors that constitute these domains and by identifying super-enhancers across the spectrum of human cell types. we describe here the population of transcription factors, cofactors, chromatin regulators, and transcription apparatus occupying super-enhancers in embryonic stem cells and evidence that super-enhancers are highly transcribed. we produce a catalog of super-enhancers in a broad range of human cell types and find that super-enhancers associate with genes that control and define the biology of these cells. interestingly, disease-associated variation is especially enriched in the super-enhancers of disease-relevant cell types. furthermore, we find that cancer cells generate super-enhancers at oncogenes and other genes important in tumor pathogenesis. thus, super-enhancers play key roles in human cell identity in health and in disease. copyright {\\\\copyright} 2013 elsevier inc. all rights reserved.variation in an individual\\'s response to environmental factors is likely to influence susceptibility to complex human diseases. the genetic basis of such variation is poorly understood. here, we identify natural genetic variants that underlie variation in the host innate immune response to infection and analyze the mechanisms by which such variants alter these responses.at different stages of any research project, molecular biologists need to choose - often somewhat arbitrarily, even after careful statistical data analysis - which genes or proteins to investigate further experimentally and which to leave out because of limited resources. computational methods that integrate complex, heterogeneous data sets - such as expression data, sequence information, functional annotation and the biomedical literature - allow prioritizing genes for future study in a more informed way. such methods can substantially increase the yield of downstream studies and are becoming invaluable to researchers.despite the explosive growth of genomic data, functional annotation of regulatory sequences remains difficult. here, we introduce  ” comparative epigenomics”—interspecies comparison of {dna} and histone modifications—as an approach for annotation of the regulatory genome. we measured in human, mouse, and pig pluripotent stem cells the genomic distributions of cytosine methylation, {h2a}.z, {h3k4me1}/2/3, {h3k9me3}, {h3k27me3}, {h3k27ac}, {h3k36me3}, transcribed {rnas}, and p300, {taf1}, {oct4}, and {nanog} binding. we observed that epigenomic conservation was strong in both rapidly evolving and slowly evolving {dna} sequences, but not in neutrally evolving sequences. in contrast, evolutionary changes of the epigenome and the transcriptome exhibited a linear correlation. we suggest that the conserved colocalization of different epigenomic marks can be used to discover regulatory sequences. indeed, seven pairs of epigenomic marks identified exhibited regulatory functions during differentiation of embryonic stem cells into mesendoderm cells. thus, comparative epigenomics reveals regulatory features of the genome that cannot be discerned from sequence comparisons alone. \\\\^{a}\\x96º epigenetic patterns of histone and {dna} modification are conserved across species \\\\^{a}\\x96º epigenomic conservation occurs in both fast- and slow-evolving {dna} sequences \\\\^{a}\\x96º changes in the epigenome, transcriptome, and {protein-dna} binding patterns are correlated \\\\^{a}\\x96º the conserved colocalization of different epigenetic marks defines regulatory {dna}overlaying differential changes in gene expression on protein interaction networks has proven to be a useful approach to interpreting the cell\\'s dynamic response to a changing environment. despite successes in finding active subnetworks in the context of a single species, the idea of overlaying lists of differentially expressed genes on networks has not yet been extended to support the analysis of multiple species\\' interaction networks. to address this problem, we designed a scalable, cross-species network search algorithm, {nexus} (network - {cross(x})-species - search), that discovers conserved, active subnetworks based on parallel differential expression studies in multiple species. our approach leverages functional linkage networks, which provide more comprehensive coverage of functional relationships than physical interaction networks by combining heterogeneous types of genomic data. we applied our cross-species approach to identify conserved modules that are differentially active in stem cells relative to differentiated cells based on parallel gene expression studies and functional linkage networks from mouse and human. we find hundreds of conserved active subnetworks enriched for stem cell-associated functions such as cell cycle, {dna} repair, and chromatin modification processes. using a variation of this approach, we also find a number of species-specific networks, which likely reflect mechanisms of stem cell function that have diverged between mouse and human. we assess the statistical significance of the subnetworks by comparing them with subnetworks discovered on random permutations of the differential expression data. we also describe several case examples that illustrate the utility of comparative analysis of active subnetworks. microarrays are a powerful tool for discovering genes whose expression is associated with a particular biological process or phenotype. differential expression analysis can often generate a list of several hundred or even thousands of significant genes. while these genes represent real expression differences, the large number of candidates can make the process of hypothesis generation for further experimental studies challenging. use of complementary datasets such as protein-protein interactions can help filter such candidate lists to genes involved with the most relevant pathways. this approach has been applied successfully by many groups, but to date, no one has developed an approach for discovering active pathways or subnetworks that are conserved across multiple species. we propose an algorithm, {nexus} (network – {cross(x})-species – search), for cross-species active subnetwork discovery given candidate gene lists from two species and weighted protein-protein interaction networks. we validate our approach on expression studies from human and mouse stem cells. we find many active subnetworks that are conserved across species relevant to stem cell biology as well as other subnetworks that show species-specific behavior. we show that these networks are not likely to have been discovered by chance and discuss several specific cases that reveal potentially novel stem cell biology.as genomic sequencing projects attempt ever more ambitious integration of genetic, molecular, and phenotypic information, a specialization of genomics has emerged, embodied in the subdiscipline of computational genomics. models inherited from population genetics, phylogenetics, and human disease genetics merge with those from graph theory, statistics, signal processing, and computer science to provide a rich quantitative foundation for genomics that can only be realized with the aid of a computer. unleashed on a rapidly increasing sample of the planet\\'s 1030 organisms, these analyses will have an impact on diverse fields of science while providing an extraordinary new window into the story of life.systems biology provides a framework for assembling models of biological systems from systematic measurements. since the field was first introduced a decade ago, considerable progress has been made in technologies for global cell measurement and in computational analyses of these data to map and model cell function. it has also greatly expanded into the translational sciences, with approaches pioneered in yeast now being applied to elucidate human development and disease. here, we review the state of the field with a focus on four emerging applications of systems biology that are likely to be of particular importance during the decade to follow: (a) pathway-based biomarkers, (b) global genetic interaction maps, (c) systems approaches to identify disease genes, and (d) stem cell systems biology. we also cover recent advances in software tools that allow biologists to explore system-wide models and to formulate new hypotheses. the applications and methods covered in this review provide a set of prime exemplars useful to cell and developmental biologists wishing to apply systems approaches to areas of interest.evolution maintains organismal fitness by preserving genomic information. this is widely assumed to involve conservation of specific genomic loci among species. many genomic encodings are now recognized to integrate small contributions from multiple genomic positions into quantitative dispersed codes, but the evolutionary dynamics of such codes are still poorly understood. here we show that in yeast, sequences that quantitatively affect nucleosome occupancy evolve under compensatory dynamics that maintain heterogeneous levels of {a+t} content through spatially coupled {a/t}-losing and {a/t}-gaining substitutions. evolutionary modeling combined with data on yeast polymorphisms supports the idea that these substitution dynamics are a consequence of weak selection. this shows that compensatory evolution, so far believed to affect specific groups of epistatically linked loci like paired {rna} bases, is a widespread phenomenon in the yeast genome, affecting the majority of intergenic sequences in it. the model thus derived suggests that compensation is inevitable when evolution conserves quantitative and dispersed genomic functions.targets for drugs have so far been predicted on the basis of molecular or cellular features, for example, by exploiting similarity in chemical structure or in activity across cell lines. we used phenotypic side-effect similarities to infer whether two drugs share a target. applied to 746 marketed drugs, a network of 1018 side effect-driven drug-drug relations became apparent, 261 of which are formed by chemically dissimilar drugs from different therapeutic indications. we experimentally tested 20 of these unexpected drug-drug relations and validated 13 implied drug-target relations by in vitro binding assays, of which 11 reveal inhibition constants equal to less than 10 micromolar. nine of these were tested and confirmed in cell assays, documenting the feasibility of using phenotypic information to infer molecular interactions and hinting at new uses of marketed drugs.we consider the problem of estimating sparse graphs by a lasso penalty applied to the inverse covariance matrix. using a coordinate descent procedure for the lasso, we develop a simple algorithm—the graphical lasso—that is remarkably fast: it solves a 1000-node problem (∼500000 parameters) in at most a minute and is 30–4000 times faster than competing methods. it also provides a conceptual link between the exact problem and the approximation suggested by meinshausen and b\\\\\"{u}hlmann (2006). we illustrate the method on some cell-signaling data from proteomics.individuals who belong to the same family or the same population are related because of their shared ancestry. population and quantitative genetics theory is built with parameters that describe relatedness, and the estimation of these parameters from genetic markers enables progress in fields as disparate as plant breeding, human disease gene mapping and forensic science. the large number of multiallelic microsatellite loci and biallelic {snps} that are now available have markedly increased the precision with which relationships can be estimated, although they have also revealed unexpected levels of genomic heterogeneity of relationship measures.sequencing of multiple related species followed by comparative genomics analysis constitutes a powerful approach for the systematic understanding of any genome. here, we use the genomes of 12 drosophila species for the de novo discovery of functional elements in the fly. each type of functional element shows characteristic patterns of change, or \\'evolutionary signatures\\', dictated by its precise selective constraints. such signatures enable recognition of new protein-coding genes and exons, spurious and incorrect gene annotations, and numerous unusual gene structures, including abundant stop-codon readthrough. similarly, we predict non-protein-coding {rna} genes and structures, and new {microrna} ({mirna}) genes. we provide evidence of {mirna} processing and functionality from both hairpin arms and both {dna} strands. we identify several classes of pre- and post-transcriptional regulatory motifs, and predict individual motif instances with high confidence. we also study how discovery power scales with the divergence and number of species compared, and we provide general guidelines for comparative studies.metabolite concentrations can regulate gene expression, which can in turn regulate metabolic activity. the extent to which functionally related transcripts and metabolites show similar patterns of concentration changes, however, remains unestablished. we measure and analyze the metabolomic and transcriptional responses of saccharomyces cerevisiae to carbon and nitrogen starvation. our analysis demonstrates that transcripts and metabolites show coordinated response dynamics. furthermore, metabolites and gene products whose concentration profiles are alike tend to participate in related biological processes. to identify specific, functionally related genes and metabolites, we develop an approach based on bayesian integration of the joint metabolomic and transcriptomic data. this algorithm finds interactions by evaluating transcript\\\\^{a}€ ” metabolite correlations in light of the experimental context in which they occur and the class of metabolite involved. it effectively predicts known enzymatic and regulatory relationships, including a gene\\\\^{a}€ ” metabolite interaction central to the glycolytic\\\\^{a}€ ” gluconeogenetic switch. this work provides quantitative evidence that functionally related metabolites and transcripts show coherent patterns of behavior on the genome scale and lays the groundwork for building gene\\\\^{a}€ ” metabolite interaction networks directly from systems-level data.motivation: many bioinformatics data resources not only hold data in the form of sequences, but also as annotation. in the majority of cases, annotation is written as scientific natural language: this is suitable for humans, but not particularly useful for machine processing. ontologies offer a mechanism by which knowledge can be represented in a form capable of such processing. in this paper we investigate the use of ontological annotation to measure the similarities in knowledge content or semantic similarity\\' between entries in a data resource. these allow a bioinformatician to perform a similarity measure over annotation in an analogous manner to those performed over sequences. a measure of semantic similarity for the knowledge component of bioinformatics resources should afford a biologist a new tool in their repetoire of analyses. results: we present the results from experiments that investigate the validity of using semantic similarity by comparison with sequence similarity. we show a simple extension that enables a semantic search of the knowledge held within sequence databases. availability: software available from http://www.russet.org.uk contact: p.lord@russet.org.uk 10.1093/bioinformatics/btg153pathway analysis has become the first choice for gaining insight into the underlying biology of differentially expressed genes and proteins, as it reduces complexity and has increased explanatory power. we discuss the evolution of knowledge base–driven pathway analysis over its first decade, distinctly divided into three generations. we also discuss the limitations that are specific to each generation, and how they are addressed by successive generations of methods. we identify a number of annotation challenges that must be addressed to enable development of the next generation of pathway analysis methods. furthermore, we identify a number of methodological challenges that the next generation of methods must tackle to take advantage of the technological advances in genomics and proteomics in order to improve specificity, sensitivity, and relevance of pathway analysis.to leverage the potential of multi-omics studies, exploratory data analysis methods that provide systematic integration and comparison of multiple layers of omics information are required. we describe multiple co-inertia analysis ({mcia}), an exploratory data analysis method that identifies co-relationships between multiple high dimensional datasets. based on a covariance optimization criterion, {mcia} simultaneously projects several datasets into the same dimensional space, transforming diverse sets of features onto the same scale, to extract the most variant from each dataset and facilitate biological interpretation and pathway analysis.while the vast majority of clustering algorithms are partitional, many real world datasets have inherently overlapping clusters. several approaches to finding overlapping clusters have come from work on analysis of biological datasets. in this paper, we interpret an overlapping clustering model proposed by segal et al. [23] as a generalization of gaussian mixture models, and we extend it to an overlapping clustering model based on mixtures of any regular exponential family distribution and the corresponding bregman divergence. we provide the necessary algorithm modifications for this extension, and present results on synthetic data as well as subsets of {20-newsgroups} and {eachmovie} datasets.defining the transcriptional dynamics of a temporal process such as cell differentiation is challenging owing to the high variability in gene expression between individual cells. time-series gene expression analyses of bulk cells have difficulty distinguishing early and late phases of a transcriptional cascade or identifying rare subpopulations of cells, and single-cell proteomic methods rely on a priori knowledge of key distinguishing markers. here we describe monocle, an unsupervised algorithm that increases the temporal resolution of transcriptome dynamics using single-cell {rna}-seq data collected at multiple time points. applied to the differentiation of primary human myoblasts, monocle revealed switch-like changes in expression of key regulatory factors, sequential waves of gene regulation, and expression of regulators that were not known to act in differentiation. we validated some of these predicted regulators in a loss-of function screen. monocle can in principle be used to recover single-cell gene expression kinetics from a wide array of cellular processes, including differentiation, proliferation and oncogenic transformation.the combinatorial cross-regulation of hundreds of sequence-specific transcription factors ({tfs}) defines a regulatory network that underlies cellular identity and function. here we use genome-wide maps of in vivo {dnasei} footprints to assemble an extensive core human regulatory network comprising connections among 475 sequence-specific {tfs} and to analyze the dynamics of these connections across 41 diverse cell and tissue types. we find that human {tf} networks are highly cell selective and are driven by cohorts of factors that include regulators with previously unrecognized roles in control of cellular identity. moreover, we identify many widely expressed factors that impact transcriptional regulatory networks in a cell-selective manner. strikingly, in spite of their inherent diversity, all cell-type regulatory networks independently converge on a common architecture that closely resembles the topology of living neuronal networks. together, our results provide an extensive description of the circuitry, dynamics, and organizing principles of the human {tf} regulatory network. copyright {\\\\copyright} 2012 elsevier inc. all rights reserved.biological processes are often dynamic, thus researchers must monitor their activity at multiple time points. the most abundant source of information regarding such dynamic activity is time-series gene expression data. these data are used to identify the complete set of activated genes in a biological process, to infer their rates of change, their order and their causal effects and to model dynamic systems in the cell. in this review we discuss the basic patterns that have been observed in time-series experiments, how these patterns are combined to form expression programs, and the computational analysis, visualization and integration of these data to infer models of dynamic biological systems.deciphering the modular organization of metabolic networks and understanding how modularity evolves have attracted tremendous interest in recent years. here, we present a comprehensive large scale characterization of modularity across the bacterial tree of life, systematically quantifying the modularity of the metabolic networks of >300 bacterial species. three main determinants of metabolic network modularity are identified. first, network size is an important topological determinant of network modularity. second, several environmental factors influence network modularity, with endosymbionts and mammal-specific pathogens having lower modularity scores than bacterial species that occupy a wider range of niches. moreover, even among the pathogens, those that alternate between two distinct niches, such as insect and mammal, tend to have relatively high metabolic network modularity. third, horizontal gene transfer is an important force that contributes significantly to metabolic modularity. we additionally reconstruct the metabolic network of ancestral bacterial species and examine the evolution of modularity across the tree of life. this reveals a trend of modularity decrease from ancestors to descendants that is likely the outcome of niche specialization and the incorporation of peripheral metabolic reactions.extracellular cues affect signaling, metabolic, and regulatory processes to elicit cellular responses. although intracellular signaling, metabolic, and regulatory networks are highly integrated, previous analyses have largely focused on independent processes (e.g., metabolism) without considering the interplay that exists among them. however, there is evidence that many diseases arise from multifunctional components with roles throughout signaling, metabolic, and regulatory networks. therefore, in this study, we propose a flux balance analysis ({fba})–based strategy, referred to as integrated dynamic {fba} ({idfba}), that dynamically simulates cellular phenotypes arising from integrated networks. the {idfba} framework requires an integrated stoichiometric reconstruction of signaling, metabolic, and regulatory processes. it assumes quasi-steady-state conditions for  ” fast” reactions and incorporates  ” slow” reactions into the stoichiometric formalism in a time-delayed manner. to assess the efficacy of {idfba}, we developed a prototypic integrated system comprising signaling, metabolic, and regulatory processes with network features characteristic of actual systems and incorporating kinetic parameters based on typical time scales observed in literature. {idfba} was applied to the prototypic system, which was evaluated for different environments and gene regulatory rules. in addition, we applied the {idfba} framework in a similar manner to a representative module of the single-cell eukaryotic organism saccharomyces cerevisiae. ultimately, {idfba} facilitated quantitative, dynamic analysis of systemic effects of extracellular cues on cellular phenotypes and generated comparable time-course predictions when contrasted with an equivalent kinetic model. since {idfba} solves a linear programming problem and does not require an exhaustive list of detailed kinetic parameters, it may be efficiently scaled to integrated intracellular systems that incorporate signaling, metabolic, and regulatory processes at the genome scale, such as the s. cerevisiae system presented here. cellular systems comprise many diverse components and component interactions spanning signal transduction, transcriptional regulation, and metabolism. although signaling, metabolic, and regulatory activities are often investigated independently of one another, there is growing evidence that considerable interplay occurs among them, and that the malfunctioning of this interplay is associated with disease. the computational analysis of integrated networks has been challenging because of the varying time scales involved as well as the sheer magnitude of such systems (e.g., the numbers of rate constants involved). to this end, we developed a novel computational framework called integrated dynamic flux balance analysis ({idfba}) that generates quantitative, dynamic predictions of species concentrations spanning signaling, regulatory, and metabolic processes. {idfba} extends an existing approach called flux balance analysis ({fba}) in that it couples  ” fast” and  ” slow” reactions, thereby facilitating the study of whole-cell phenotypes and not just sub-cellular network properties. we applied this framework to a prototypic integrated system derived from literature as well as a representative integrated yeast module (the high-osmolarity glycerol [{hog}] pathway) and generated time-course predictions that matched with available experimental data. by extending this framework to larger-scale systems, phenotypic profiles of whole-cell systems could be attained expeditiously.{background}:cell growth underlies many key cellular and developmental processes, yet a limited number of studies have been carried out on cell-growth regulation. comprehensive studies at the transcriptional, proteomic and metabolic levels under defined controlled conditions are currently {lacking.results}:metabolic control analysis is being exploited in a systems biology study of the eukaryotic cell. using chemostat culture, we have measured the impact of changes in flux (growth rate) on the transcriptome, proteome, endometabolome and exometabolome of the yeast saccharomyces cerevisiae. each functional genomic level shows clear growth-rate-associated trends and discriminates between carbon-sufficient and carbon-limited conditions. genes consistently and significantly upregulated with increasing growth rate are frequently essential and encode evolutionarily conserved proteins of known function that participate in many protein-protein interactions. in contrast, more unknown, and fewer essential, genes are downregulated with increasing growth rate; their protein products rarely interact with one another. a large proportion of yeast genes under positive growth-rate control share orthologs with other eukaryotes, including humans. significantly, transcription of genes encoding components of the {tor} complex (a major controller of eukaryotic cell growth) is not subject to growth-rate regulation. moreover, integrative studies reveal the extent and importance of post-transcriptional control, patterns of control of metabolic fluxes at the level of enzyme synthesis, and the relevance of specific enzymatic reactions in the control of metabolic fluxes during cell {growth.conclusion}:this work constitutes a first comprehensive systems biology study on growth-rate control in the eukaryotic cell. the results have direct implications for advanced studies on cell growth, in vivo regulation of metabolic fluxes for comprehensive metabolic engineering, and for the design of genome-scale systems biology models of the eukaryotic cell.the target of rapamycin ({tor}) is a conserved {ser/thr} kinase that regulates cell growth and metabolism in response to environmental cues. here, highlighting contributions from studies in model organisms, we review mammalian {tor} complexes and the signaling branches they mediate. {tor} is part of two distinct multiprotein complexes, {tor} complex 1 ({torc1}), which is sensitive to rapamycin, and {torc2}, which is not. the physiological consequences of mammalian {torc1} dysregulation suggest that inhibitors of mammalian {tor} may be useful in the treatment of cancer, cardiovascular disease, autoimmunity, and metabolic disorders.motivation: network component analysis ({nca}) is an efficient method of reconstructing the transcription factor activity ({tfa}), which makes use of the gene expression data and prior information available about transcription factor ({tf}) - gene regulations. most of the contemporary algorithms either exhibit the drawback of inconsistency and poor reliability, or suffer from prohibitive computational complexity. in addition, the existing algorithms do not possess the ability to counteract the presence of outliers in the microarray data. hence, robust and computationally efficient algorithms are needed to enable practical applications.recent years have seen much interest in the study of systems characterized by multiple interacting components. a class of statistical models called graphical models, in which graphs are used to represent probabilistic relationships between variables, provides a framework for formal inference regarding such systems. in many settings, the object of inference is the network structure itself. this problem of  ” network inference” is well known to be a challenging one. however, in scientific settings there is very often existing information regarding network connectivity. a natural idea then is to take account of such information during inference. this article addresses the question of incorporating prior information into network inference. we focus on directed models called bayesian networks, and use markov chain monte carlo to draw samples from posterior distributions over network structures. we introduce prior distributions on graphs capable of capturing information regarding network features including edges, classes of edges, degree distributions, and sparsity. we illustrate our approach in the context of systems biology, applying our methods to network inference in cancer signaling.the extent by which different cellular components generate phenotypic diversity is an ongoing debate in evolutionary biology that is yet to be addressed by quantitative comparative studies. we conducted an in vivo mass-spectrometry study of the phosphoproteomes of three yeast species (saccharomyces cerevisiae, candida albicans, and schizosaccharomyces pombe) in order to quantify the evolutionary rate of change of phosphorylation. we estimate that kinase–substrate interactions change, at most, two orders of magnitude more slowly than transcription factor ({tf})–promoter interactions. our computational analysis linking kinases to putative substrates recapitulates known phosphoregulation events and provides putative evolutionary histories for the kinase regulation of protein complexes across 11 yeast species. to validate these trends, we used the {e-map} approach to analyze over 2,000 quantitative genetic interactions in s. cerevisiae and sc. pombe, which demonstrated that protein kinases, and to a greater extent {tfs}, show lower than average conservation of genetic interactions. we propose therefore that protein kinases are an important source of phenotypic diversity. natural selection at a population level requires phenotypic diversity, which at the molecular level arises by mutation of the genome of each individual. what kinds of changes at the level of the {dna} are most important for the generation of phenotypic differences remains a fundamental question in evolutionary biology. one well-studied source of phenotypic diversity is mutation in gene regulatory regions that results in changes in gene expression, but what proportion of phenotypic diversity is due to such mutations is not entirely clear. we investigated the relative contribution to phenotypic diversity of mutations in protein-coding regions compared to mutations in gene regulatory sequences. given the important regulatory role played by phosphorylation across biological systems, we focused on mutations in protein-coding regions that alter protein–protein interactions involved in the binding of kinases to their substrate proteins. we studied the evolution of this  ” phosphoregulation” by analyzing the in vivo complement of phosphorylated proteins (the  ” phosphoproteome”) in three highly diverged yeast species—the budding yeast saccharomyces cerevisiae, the pathogenic yeast candida albicans, and the fission yeast schizosaccharomyces pombe—and integrating those data with existing data on thousands of known genetic interactions from s. cerevisiae and sc. pombe. we show that kinase–substrate interactions are altered at a rate that is at most two orders of magnitude slower than the alteration of transcription factor ({tf})–promoter interactions, whereas {tfs} and kinases both show a faster than average rate of functional divergence estimated by the cross-species analysis of genetic interactions. our data provide a quantitative estimate of the relative frequencies of different kinds of functionally relevant mutations and demonstrate that, like mutations in gene regulatory regions, mutations that result in changes in kinase–substrate interactions are an important source of phenotypic diversity.transposable element ({te})-derived sequence dominates the landscape of mammalian genomes and can modulate gene function by dysregulating transcription and translation. our current knowledge of {tes} in laboratory mouse strains is limited primarily to those present in the {c57bl}/{6j} reference genome, with most mouse {tes} being drawn from three distinct classes, namely short interspersed nuclear elements ({sines}), long interspersed nuclear elements ({lines}) and the endogenous retrovirus ({erv}) superfamily. despite their high prevalence, the different genomic and gene properties controlling whether {tes} are preferentially purged from, or are retained by, genetic drift or positive selection in mammalian genomes remain poorly defined. using whole genome sequencing data from 13 classical laboratory and 4 wild-derived mouse inbred strains, we developed a comprehensive catalogue of 103,798 polymorphic {te} variants. we employ this extensive data set to characterize {te} variants across the mus lineage, and to infer neutral and selective processes that have acted over 2 million years. our results indicate that the majority of {te} variants are introduced though the male germline and that only a minority of {te} variants exert detectable changes in gene expression. however, among genes with differential expression across the strains there are twice as many {te} variants identified as being putative causal variants as expected. most {te} variants that cause gene expression changes appear to be purged rapidly by purifying selection. our findings demonstrate that past {te} insertions have often been highly deleterious, and help to prioritize {te} variants according to their likely contribution to gene expression or phenotype variation.{background}:complex diseases are associated with altered interactions between thousands of genes. we developed a novel method to identify and prioritize disease genes, which was generally applicable to complex {diseases.results}:we identified modules of highly interconnected genes in disease-specific networks derived from integrating gene-expression and protein interaction data. we examined if those modules were enriched for disease-associated {snps}, and could be used to find novel genes for functional studies. first, we analyzed publicly available gene expression microarray and genome-wide association study ({gwas}) data from 13, highly diverse, complex diseases. in each disease, highly interconnected genes formed modules, which were significantly enriched for genes harboring disease-associated {snps}. to test if such modules could be used to find novel genes for functional studies, we repeated the analyses using our own gene expression microarray and {gwas} data from seasonal allergic rhinitis. we identified a novel gene, {fgf2}, whose relevance was supported by functional studies using combined small interfering {rna}-mediated knock-down and gene expression microarrays. the modules in the 13 complex diseases analyzed here tended to overlap and were enriched for pathways related to oncological, metabolic and inflammatory diseases. this suggested that this union of the modules would be associated with a general increase in susceptibility for complex diseases. indeed, we found that this union was enriched with {gwas} genes for 145 other complex {diseases.conclusions}:modules of highly interconnected complex disease genes were enriched for disease-associated {snps}, and could be used to find novel genes for functional studies.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsto complement the human encyclopedia of {dna} elements ({encode}) project and to enable a broad range of mouse genomics efforts, the mouse {encode} consortium is applying the same experimental pipelines developed for human {encode} to annotate the mouse genome.epistasis refers to the interaction between genes. although high-throughput epistasis data from model organisms are being generated and used to construct genetic networks, the extent to which genetic epistasis reflects biologically meaningful interactions remains unclear. we have addressed this question through in silico mapping of positive and negative epistatic interactions amongst biochemical reactions within the metabolic networks of escherichia coli and saccharomyces cerevisiae using flux balance analysis. we found that negative epistasis occurs mainly between nonessential reactions with overlapping functions, whereas positive epistasis usually involves essential reactions, is highly abundant and, unexpectedly, often occurs between reactions without overlapping functions. we offer mechanistic explanations of these findings and experimentally validate them for 61 s. cerevisiae gene pairs.chromatin organization plays a major role in gene regulation and can affect the function and evolution of new transcriptional programs. however, it can be difficult to decipher the basis of changes in chromatin organization and their functional effect on gene expression. here, we present a large-scale comparative genomic analysis of the relationship between chromatin organization and gene expression, by measuring {mrna} abundance and nucleosome positions genome-wide in 12 hemiascomycota yeast species. we found substantial conservation of global and functional chromatin organization in all species, including prominent nucleosome-free regions ({nfrs}) at gene promoters, and distinct chromatin architecture in growth and stress genes. chromatin organization has also substantially diverged in both global quantitative features, such as spacing between adjacent nucleosomes, and in functional groups of genes. expression levels, intrinsic anti-nucleosomal sequences, and trans-acting chromatin modifiers all play important, complementary, and evolvable roles in determining {nfrs}. we identify five mechanisms that couple chromatin organization to evolution of gene regulation and have contributed to the evolution of respiro-fermentation and other key systems, including (1) compensatory evolution of alternative modifiers associated with conserved chromatin organization, (2) a gradual transition from constitutive to trans-regulated {nfrs}, (3) a loss of intrinsic anti-nucleosomal sequences accompanying changes in chromatin organization and gene expression, (4) re-positioning of motifs from {nfrs} to nucleosome-occluded regions, and (5) the expanded use of {nfrs} by paralogous activator-repressor pairs. our study sheds light on the molecular basis of chromatin organization, and on the role of chromatin organization in the evolution of gene regulation. divergence in gene regulation plays a major role in organismal evolution. evidence suggests that changes in the packaging of eukaryotic genomes into chromatin can underlie the evolution of divergent gene expression patterns. here, we explore the role of chromatin structure in regulatory evolution by whole-genome measurements of nucleosome positions and {mrna} levels in 12 yeast species spanning \\\\~{}250 million years of evolution. we find several distinct ways in which changes in chromatin structure are associated with changes in gene expression. these include changes in promoter accessibility, changes in promoter chromatin architecture, and changes in the accessibility of specific transcription factor binding sites. in many cases, changes in chromatin architecture are coupled to physiological diversity, including the evolution of a respiration- or fermentation-based lifestyle, mating behavior, salt tolerance, and broad aspects of genomic structure. together, our data will provide a rich resource for future investigations into the interplay between chromatin structure, gene regulation, and evolution.motivation: network inference algorithms are powerful computational tools for identifying putative causal interactions among variables from observational data. bayesian network inference algorithms hold particular promise in that they can capture linear, non-linear, combinatorial, stochastic and other types of relationships among variables across multiple levels of biological organization. however, challenges remain when applying these algorithms to limited quantities of experimental data collected from biological systems. here, we use a simulation approach to make advances in our dynamic bayesian network ({dbn}) inference algorithm, especially in the context of limited quantities of biological data.  results: we test a range of scoring metrics and search heuristics to find an effective algorithm configuration for evaluating our methodological advances. we also identify sampling intervals and levels of data discretization that allow the best recovery of the simulated networks. we develop a novel influence score for {dbns} that attempts to estimate both the sign (activation or repression) and relative magnitude of interactions among variables. when faced with limited quantities of observational data, combining our influence score with moderate data interpolation reduces a significant portion of false positive interactions in the recovered networks. together, our advances allow {dbn} inference algorithms to be more effective in recovering biological networks from experimentally collected data.  availability: source code and simulated data are available upon request.  supplementary information: {http://www.jarvislab.net/bioinformatics}/{bnadvances}/ 10.1093/bioinformatics/bth448the rapid accumulation of biological networks poses new challenges and calls for powerful integrative analysis tools. most existing methods capable of simultaneously analyzing a large number of networks were primarily designed for unweighted networks, and cannot easily be extended to weighted networks. however, it is known that transforming weighted into unweighted networks by dichotomizing the edges of weighted networks with a threshold generally leads to information loss. we have developed a novel, tensor-based computational framework for mining recurrent heavy subgraphs in a large set of massive weighted networks. specifically, we formulate the recurrent heavy subgraph identification problem as a heavy {3d} subtensor discovery problem with sparse constraints. we describe an effective approach to solving this problem by designing a multi-stage, convex relaxation protocol, and a non-uniform edge sampling technique. we applied our method to 130 co-expression networks, and identified 11,394 recurrent heavy subgraphs, grouped into 2,810 families. we demonstrated that the identified subgraphs represent meaningful biological modules by validating against a large set of compiled biological knowledge bases. we also showed that the likelihood for a heavy subgraph to be meaningful increases significantly with its recurrence in multiple networks, highlighting the importance of the integrative approach to biological network analysis. moreover, our approach based on weighted graphs detects many patterns that would be overlooked using unweighted graphs. in addition, we identified a large number of modules that occur predominately under specific phenotypes. this analysis resulted in a genome-wide mapping of gene network modules onto the phenome. finally, by comparing module activities across many datasets, we discovered high-order dynamic cooperativeness in protein complex networks and transcriptional regulatory networks. to study complex cellular networks, we need to consider their dynamic topologies under many different experimental or physiological conditions. integrative analysis over large numbers of massive biological networks thus emerges as a new challenge in data mining. recently, we and others have proposed several algorithms for recurrent pattern mining across many () biological networks (with the main focus on unweighted networks). however, thus far no algorithms have been specifically designed to mine recurrent patterns across a large collection of weighted massive networks. in this paper, we propose a computational framework to identify recurrent heavy subgraphs from many weighted large networks. by applying our method to 130 co-expression networks, we identified an atlas of modules that are highly likely to represent functional modules, transcriptional modules, and protein complexes. many of these modules would be overlooked with unweighted networks analysis. furthermore, many of the identified modules constituted signatures of specific phenotypes. finally, we demonstrated that our results facilitate the study of high-order dynamic coordination in protein complex networks and transcriptional regulatory networks.summary: the great variety of human cell types in morphology and function is due to the diverse gene expression profiles that are governed by the distinctive regulatory networks in different cell types. it is still a challenging task to explain how the regulatory networks achieve the diversity of different cell types. here, we report on our studies of the design principles of the tissue regulatory system by constructing the regulatory networks of eight human tissues, which subsume the regulatory interactions between transcription factors ({tfs}), {micrornas} ({mirnas}) and {non-tf} target genes. the results show that there are in-/out-hubs of high in-/out-degrees in tissue networks. some hubs (strong hubs) maintain the hub status in all the tissues where they are expressed, whereas others (weak hubs), in spite of their ubiquitous expression, are hubs only in some tissues. the network motifs are mostly feed-forward loops. some of them having no {mirnas} are the common motifs shared by all tissues, whereas the others containing {mirnas} are the tissue-specific ones owned by one or several tissues, indicating that the transcriptional regulation is more conserved across tissues than the post-transcriptional regulation. in particular, a common bow-tie framework was found that underlies the motif instances and shows diverse patterns in different tissues. such bow-tie framework reflects the utilization efficiency of the regulatory system as well as its high variability in different tissues, and could serve as the model to further understand the structural adaptation of the regulatory system to the specific requirements of different cell functions.motivation: in model organisms such as yeast, large databases of protein–protein and {protein-dna} interactions have become an extremely important resource for the study of protein function, evolution, and gene regulatory dynamics. in this paper we demonstrate that by integrating these interactions with widely-available {mrna} expression data, it is possible to generate concrete hypotheses for the underlying mechanisms governing the observed changes in gene expression. to perform this integration systematically and at large scale, we introduce an approach for screening a molecular interaction network to identify active subnetworks, i.e., connected regions of the network that show significant changes in expression over particular subsets of conditions. the method we present here combines a rigorous statistical measure for scoring subnetworks with a search algorithm for identifying subnetworks with high score.we describe the construction of a digital brain atlas composed of data from manually delineated {mri} data. a total of 56 structures were labeled in {mri} of 40 healthy, normal volunteers. this labeling was performed according to a set of protocols developed for this project. pairs of raters were assigned to each structure and trained on the protocol for that structure. each rater pair was tested for concordance on 6 of the 40 brains; once they had achieved reliability standards, they divided the task of delineating the remaining 34 brains. the data were then spatially normalized to well-known templates using 3 popular algorithms: {air5}.2.5\\'s nonlinear warp (woods et al., 1998) paired with the {icbm452} warp 5 atlas (rex et al., 2003), {fsl}\\'s {flirt} (smith et al., 2004) was paired with its own template, a skull-stripped version of the {icbm152} t1 average; and {spm5}\\'s unified segmentation method (ashburner and friston, 2005) was paired with its canonical brain, the whole head {icbm152} t1 average. we thus produced 3 variants of our atlas, where each was constructed from 40 representative samples of a data processing stream that one might use for analysis. for each normalization algorithm, the individual structure delineations were then resampled according to the computed transformations. we next computed averages at each voxel location to estimate the probability of that voxel belonging to each of the 56 structures. each version of the atlas contains, for every voxel, probability densities for each region, thus providing a resource for automated probabilistic labeling of external data types registered into standard spaces; we also computed average intensity images and tissue density maps based on the three methods and target spaces. these atlases will serve as a resource for diverse applications including meta-analysis of functional and structural imaging data and other bioinformatics applications where display of arbitrary labels in probabilistically defined anatomic space will facilitate both knowledge-based development and visualization of findings from multiple disciplines.a common biological pathway reconstruction approach—as implemented by many automatic biological pathway services (such as the {kaas} and {rast} servers) and the functional annotation of metagenomic sequences—starts with the identification of protein functions or families (e.g., {ko} families for the {kegg} database and the {fig} families for the {seed} database) in the query sequences, followed by a direct mapping of the identified protein families onto pathways. given a predicted patchwork of individual biochemical steps, some metric must be applied in deciding what pathways actually exist in the genome or metagenome represented by the sequences. commonly, and straightforwardly, a complete biological pathway can be identified in a dataset if at least one of the steps associated with the pathway is found. we report, however, that this na\\\\\"{i}ve mapping approach leads to an inflated estimate of biological pathways, and thus overestimates the functional diversity of the sample from which the {dna} sequences are derived. we developed a parsimony approach, called {minpath} (minimal set of pathways), for biological pathway reconstructions using protein family predictions, which yields a more conservative, yet more faithful, estimation of the biological pathways for a query dataset. {minpath} identified far fewer pathways for the genomes collected in the {kegg} database—as compared to the na\\\\\"{i}ve mapping approach—eliminating some obviously spurious pathway annotations. results from applying {minpath} to several metagenomes indicate that the common methods used for metagenome annotation may significantly overestimate the biological pathways encoded by microbial communities. even though there is only a single large biological network within any cell and all pathways are to some extent connected, the partition of the entire cellular network into smaller units (e.g., {kegg} pathways) is extremely important for understanding biological processes. biological pathway reconstruction, therefore, is essential for understanding the biological functions that a newly sequenced genome encodes and recently for studying the functionality of a natural environment via metagenomics. the common practice of pathway reconstruction in metagenomics first identifies functions encoded by the metagenomic sequences and then reconstructs pathways from the annotated functions by mapping the functions to reference pathways. to address the issues of both incomplete data (e.g., metagenomes, unlike individual genomes, are most likely incomplete) and pathway redundancy (e.g., the same function is involved in multiple pathway units), we formulate a parsimony version of the pathway reconstruction/inference problem, called {minpath} (minimal set of pathways): given a set of reference pathways and a set of functions that can be mapped to one or more pathways, {minpath} aims at finding a minimum number of pathways that can explain all functions. {minpath} achieves a more conservative, yet more faithful, estimation of the biological pathways encoded by genomes and metagenomes.{rna} is not only a messenger operating between {dna} and protein. transcription of essentially the entire eukaryotic genome generates a myriad of non-protein-coding {rna} species that show complex overlapping patterns of expression and regulation. although long noncoding {rnas} ({lncrnas}) are among the least well-understood of these transcript species, they cannot all be dismissed as merely transcriptional \\x1cnoise.\\x1d here, we review the evolution of {lncrnas} and their roles in transcriptional regulation, epigenetic gene regulation, and disease.for many complex traits, genetic variants have been found associated. however, it is still mostly unclear through which downstream mechanism these variants cause these phenotypes. knowledge of these intermediate steps is crucial to understand pathogenesis, while also providing leads for potential pharmacological intervention. here we relied upon natural human genetic variation to identify effects of these variants on trans-gene expression (expression quantitative trait locus mapping, {eqtl}) in whole peripheral blood from 1,469 unrelated individuals. we looked at 1,167 published trait- or disease-associated {snps} and observed {trans-eqtl} effects on 113 different genes, of which we replicated 46 in monocytes of 1,490 different individuals and 18 in a smaller dataset that comprised subcutaneous adipose, visceral adipose, liver tissue, and muscle tissue. {hla} single-nucleotide polymorphisms ({snps}) were 10-fold enriched for {trans-eqtls}: 48\\\\% of the trans-acting {snps} map within the {hla}, including ulcerative colitis susceptibility variants that affect plausible candidate genes {aoah} and {trbv18} in trans. we identified 18 pairs of unlinked {snps} associated with the same phenotype and affecting expression of the same trans-gene (21 times more than expected, p<10−16). this was particularly pronounced for mean platelet volume ({mpv}): two independent {snps} significantly affect the well-known blood coagulation genes {gp9} and {f13a1} but also c19orf33, {samd14}, {vcl}, and {gng11}. several of these {snps} have a substantially higher effect on the downstream trans-genes than on the eventual phenotypes, supporting the concept that the effects of these {snps} on expression seems to be much less multifactorial. therefore, these {trans-eqtls} could well represent some of the intermediate genes that connect genetic variants with their eventual complex phenotypic outcomes. many genetic variants have been found associated with diseases. however, for many of these genetic variants, it remains unclear how they exert their effect on the eventual phenotype. we investigated genetic variants that are known to be associated with diseases and complex phenotypes and assessed whether these variants were also associated with gene expression levels in a set of 1,469 unrelated whole blood samples. for several diseases, such as type 1 diabetes and ulcerative colitis, we observed that genetic variants affect the expression of genes, not implicated before. for complex traits, such as mean platelet volume and mean corpuscular volume, we observed that independent genetic variants on different chromosomes influence the expression of exactly the same genes. for mean platelet volume, these genes include well-known blood coagulation genes but also genes with still unknown functions. these results indicate that, by systematically correlating genetic variation with gene expression levels, it is possible to identify downstream genes, which provide important avenues for further research.many studies have demonstrated that divergence levels generated by different mutation types vary and covary across the human genome. to improve our still-incomplete understanding of the mechanistic basis of this phenomenon, we analyze several mutation types simultaneously, anchoring their variation to specific regions of the genome. using hidden markov models on insertion, deletion, nucleotide substitution, and microsatellite divergence estimates inferred from human-orangutan alignments of neutrally evolving genomic sequences, we segment the human genome into regions corresponding to different divergence states--each uniquely characterized by specific combinations of divergence levels. we then parsed the mutagenic contributions of various biochemical processes associating divergence states with a broad range of genomic landscape features. we find that high divergence states inhabit guanine- and cytosine ({gc})-rich, highly recombining subtelomeric regions; low divergence states cover inner parts of autosomes; chromosome x forms its own state with lowest divergence; and a state of elevated microsatellite mutability is interspersed across the genome. these general trends are mirrored in human diversity data from the 1000 genomes project, and departures from them highlight the evolutionary history of primate chromosomes. we also find that genes and noncoding functional marks [annotations from the encyclopedia of {dna} elements ({encode})] are concentrated in high divergence states. our results provide a powerful tool for biomedical data analysis: segmentations can be used to screen personal genome variants--including those associated with cancer and other diseases--and to improve computational predictions of noncoding functional elements.enhancer function underlies regulatory processes by which cells establish patterns of gene expression. recent results suggest that many enhancers are specified by particular chromatin marks in pluripotent cells, which may be modified later in development to alter patterns of gene expression and cell differentiation choices. these marks may contribute to the repertoire of epigenetic mechanisms responsible for cellular memory and determine the timing of transcription factor accessibility to the enhancer. mechanistically, cohesin and non-coding {rnas} are emerging as crucial players responsible for facilitating enhancer–promoter interactions at some genes. surprisingly, these interactions may be required not only to facilitate initiation of transcription but also to activate the release of {rna} polymerase {ii} ({rnapii}) from promoter-proximal pausing.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsan important problem in systems biology is the inference of biochemical pathways and regulatory networks from postgenomic data. various reverse engineering methods have been proposed in the literature, and it is important to understand their relative merits and shortcomings. in the present paper, we compare the accuracy of reconstructing gene regulatory networks with three different modelling and inference paradigms: (1) relevance networks ({rns}): pairwise association scores independent of the remaining network; (2) graphical gaussian models ({ggms}): undirected graphical models with constraint-based inference, and (3) bayesian networks ({bns}): directed graphical models with score-based inference. the evaluation is carried out on the raf pathway, a cellular signalling network describing the interaction of 11 phosphorylated proteins and phospholipids in human immune system cells. we use both laboratory data from cytometry experiments as well as data simulated from the gold-standard network. we also compare passive observations with active interventions. on gaussian observational data, {bns} and {ggms} were found to outperform {rns}. the difference in performance was not significant for the non-linear simulated data and the cytoflow data, though. also, we did not observe a significant difference between {bns} and {ggms} on observational data in general. however, for interventional data, {bns} outperform {ggms} and {rns}, especially when taking the edge directions rather than just the skeletons of the graphs into account. this suggests that the higher computational costs of inference with {bns} over {ggms} and {rns} are not justified when using only passive observations, but that active interventions in the form of gene knockouts and over-expressions are required to exploit the full potential of {bns}. data, software and supplementary material are available from http://www.bioss.sari.ac.uk/staff/adriano/research.htmlgene expression is regulated by transcription factors ({tfs}), proteins that recognize short {dna} sequence motifs. such sequences are very common in the human genome, and an important determinant of the specificity of gene expression is the cooperative binding of multiple {tfs} to closely located motifs. however, interactions between {dna}-bound {tfs} have not been systematically characterized. to identify {tf} pairs that bind cooperatively to {dna}, and to characterize their spacing and orientation preferences, we have performed consecutive affinity-purification systematic evolution of ligands by exponential enrichment ({cap}-{selex}) analysis of 9,400 {tf}\\\\&\\\\#{19;tf}\\\\&\\\\#{19;dna} interactions. this analysis revealed 315 {tf}\\\\&\\\\#{19;tf} interactions recognizing 618 heterodimeric motifs, most of which have not been previously described. the observed cooperativity occurred promiscuously between {tfs} from diverse structural families. structural analysis of the {tf} pairs, including a novel crystal structure of {meis1} and {dlx3} bound to their identified recognition site, revealed that the interactions between the {tfs} were predominantly mediated by {dna}. most {tf} pair sites identified involved a large overlap between individual {tf} recognition motifs, and resulted in recognition of composite sites that were markedly different from the individual {tf}\\\\&\\\\#25;s motifs. together, our results indicate that the {dna} molecule commonly plays an active role in cooperative interactions that define the gene regulatory lexicon.regulatory elements recruit transcription factors that modulate gene expression distinctly across cell types, but the relationships among these remains elusive. to address this, we analyzed matched {dnase}-seq and gene expression data for 112 human samples representing 72 cell types. we first defined more than 1800 clusters of {dnase} i hypersensitive sites ({dhss}) with similar tissue specificity of {dnase}-seq signal patterns. we then used these to uncover distinct associations between {dhss} and promoters, {cpg} islands, conserved elements, and transcription factor motif enrichment. motif analysis within clusters identified known and novel motifs in cell-type-specific and ubiquitous regulatory elements and supports a role for {ap}-1 regulating open chromatin. we developed a classifier that accurately predicts cell-type lineage based on only 43 {dhss} and evaluated the tissue of origin for cancer cell types. a similar classifier identified three sex-specific loci on the x chromosome, including the {xist} {lincrna} locus. by correlating {dnase} i signal and gene expression, we predicted regulated genes for more than {500k} {dhss}. finally, we introduce a web resource to enable researchers to use these results to explore these regulatory patterns and better understand how expression is modulated within and across human cell types.as cells develop and differentiate into different types, the shape and accessibility of their {dna} can change. {lara-astiaso} et al. studied this phenomenon in blood. they developed a technique that examines a relatively small number of cells to identify the changes that affect {dna} during blood development. they found that the {dna} of noncoding regions, called enhancers, is set in an open position when cells are undifferentiated and able to take on a variety of roles and gradually closes as cells mature into their final forms.chromatin organization and gene-gene interactions are critical components of carrying out developmental programs. {phillips-cremins} et\\xa0al. identify a series of unexpected architectural proteins that work in a combinatorial manner to functionally organize chromatin in a cell-type-specific manner at the submegabase-length scale.finding functional {dna} binding sites of transcription factors ({tfs}) throughout the genome is a crucial step in understanding transcriptional regulation. unfortunately, these binding sites are typically short and degenerate, posing a significant statistical challenge: many more matches to known {tf} motifs occur in the genome than are actually functional. however, information about chromatin structure may help to identify the functional sites. in particular, it has been shown that active regulatory regions are usually depleted of nucleosomes, thereby enabling {tfs} to bind {dna} in those regions. here, we describe a novel motif discovery algorithm that employs an informative prior over {dna} sequence positions based on a discriminative view of nucleosome occupancy. when a gibbs sampling algorithm is applied to yeast sequence-sets identified by {chip}-chip, the correct motif is found in 52\\\\% more cases with our informative prior than with the commonly used uniform prior. this is the first demonstration that nucleosome occupancy information can be used to improve motif discovery. the improvement is dramatic, even though we are using only a statistical model to predict nucleosome occupancy; we expect our results to improve further as high-resolution genome-wide experimental nucleosome occupancy data becomes increasingly available. identifying transcription factor ({tf}) binding sites across the genome is an important problem in molecular biology. large-scale discovery of {tf} binding sites is usually carried out by searching for short {dna} patterns that appear often within promoter regions of genes that are known to be co-bound by a {tf}. in such problems, promoters have traditionally been treated as strings of nucleotide bases in which {tf} binding sites are assumed to be equally likely to occur at any position. in vivo, however, {tfs} localize to {dna} binding sites as part of a complicated thermodynamic process of cooperativity and competition, both with one another and, importantly, with {dna} packaging proteins called nucleosomes. in particular, {tfs} are more likely to bind {dna} at sites that are not occupied by nucleosomes. in this paper, we show that it is possible to incorporate knowledge of the nucleosome landscape across the genome to aid binding site discovery; indeed, our algorithm incorporating nucleosome occupancy information is significantly more accurate than conventional methods. we use our algorithm to generate a condition-dependent, nucleosome-guided map of binding sites for 55 {tfs} in yeast.to understand how transcription factors function, it is essential to determine the range of genes that they each bind and regulate in\\xa0vivo. here i review evidence that most animal transcription factors each bind to a majority of genes over a quantitative series of {dna} occupancy levels. these continua span functional, quasifunctional, and nonfunctional {dna} binding events. factor regulatory specificities are distinguished by quantitative differences in {dna} occupancy patterns. i contrast these results with models for transcription networks that define discrete sets of direct target and nontarget genes and consequently do not fully capture the complexity observed in\\xa0vivo.a method is presented to approximate optimally ann-dimensional discrete probability distribution by a product of second-order distributions, or the distribution of the first-order tree dependence. the problem is to find an optimum set ofn - 1first order dependence relationship among thenvariables. it is shown that the procedure derived in this paper yields an approximation of a minimum difference in information. it is further shown that when this procedure is applied to empirical observations from an unknown distribution of tree dependence, the procedure is the maximum-likelihood estimate of the distribution.we studied the relationship between growth rate and genome-wide gene expression, cell cycle progression, and glucose metabolism in 36 steady-state continuous cultures limited by one of six different nutrients (glucose, ammonium, sulfate, phosphate, uracil, or leucine). the expression of more than one quarter of all yeast genes is linearly correlated with growth rate, independent of the limiting nutrient. the subset of negatively growth-correlated genes is most enriched for peroxisomal functions, whereas positively correlated genes mainly encode ribosomal functions. many (not all) genes associated with stress response are strongly correlated with growth rate, as are genes that are periodically expressed under conditions of metabolic cycling. we confirmed a linear relationship between growth rate and the fraction of the cell population in the {g0/g1} cell cycle phase, independent of limiting nutrient. cultures limited by auxotrophic requirements wasted excess glucose, whereas those limited on phosphate, sulfate, or ammonia did not; this phenomenon (reminiscent of the  ” warburg effect” in cancer cells) was confirmed in batch cultures. using an aggregate of gene expression values, we predict (in both continuous and batch cultures) an  ” instantaneous growth rate.” this concept is useful in interpreting the system-level connections among growth rate, metabolism, stress, and the cell cycle.although much of the information regarding genes\\' expressions is encoded in the genome, deciphering such information has been very challenging. we reexamined beer and tavazoie\\'s ({bt}) approach to predict {mrna} expression patterns of 2,587 genes in saccharomyces cerevisiae from the information in their respective promoter sequences. instead of fitting complex bayesian network models, we trained na\\\\\"{i}ve bayes classifiers using only the sequence-motif matching scores provided by {bt}. our simple models correctly predict expression patterns for 79\\\\% of the genes, based on the same criterion and the same cross-validation ({cv}) procedure as {bt}, which compares favorably to the 73\\\\% accuracy of {bt}. the fact that our approach did not use position and orientation information of the predicted binding sites but achieved a higher prediction accuracy, motivated us to investigate a few biological predictions made by {bt}. we found that some of their predictions, especially those related to motif orientations and positions, are at best circumstantial. for example, the combinatorial rules suggested by {bt} for the {pac} and {rrpe} motifs are not unique to the cluster of genes from which the predictive model was inferred, and there are simpler rules that are statistically more significant than {bt}\\'s ones. we also show that {cv} procedure used by {bt} to estimate their method\\'s prediction accuracy is inappropriate and may have overestimated the prediction accuracy by about 10\\\\%. through binding to certain sequence-specific sites upstream of the target genes, a special class of proteins called transcription factors ({tfs}) control transcription activities, i.e., expression amounts, of the downstream genes. the {dna} sequence patterns bound by {tfs} are called motifs. it has been shown in an article by beer and tavazoie ({bt}) published in cell in 2004 that a gene\\'s expression pattern can be well-predicted based only on its upstream sequence information in the form of matching scores of a set of sequence motifs and the location and orientation of corresponding predicted binding sites. here we report a new na\\\\\"{i}ve bayes method for such a prediction task. compared to {bt}\\'s work, our model is simpler, more robust, and achieves a higher prediction accuracy using only the motif matching score. in our method, the location and orientation information do not further help the prediction in a global way. our result also casts doubt on several biological hypotheses generated by {bt} based on their model. finally, we show that the cross-validation procedure used by {bt} to estimate their method\\'s prediction accuracy is inappropriate and may have overestimated the accuracy by about 10\\\\%.10.1073/pnas.0805644105condition-specific networks capture system-wide behavior under varying conditions such as environmental stresses, cell types or tissues. these networks frequently comprise parts that are unique to each condition, and parts that are shared among related conditions. existing approaches for learning condition-specific networks typically identify either only differences or only similarities across conditions. most of these approaches first learn networks per condition independently, and then identify similarities and differences in a post-learning step. such approaches do not exploit the shared information across conditions during network learning. we describe an approach for learning condition-specific networks that identifies the shared and unique subgraphs during network learning simultaneously, rather than as a post-processing step. our approach learns networks across condition sets, shares data from different conditions and produces high-quality networks that capture biologically meaningful information. on simulated data, our approach outperformed an existing approach that learns networks independently for each condition, especially for small training datasets. on microarray data of hundreds of deletion mutants in two, yeast stationary-phase cell populations, the inferred network structure identified several common and population-specific effects of these deletion mutants and several high-confidence cases of double-deletion pairs, which can be experimentally tested. our results are consistent with and extend the existing knowledge base of differentiated cell populations in yeast stationary phase. c++ code can be accessed from http://www.broadinstitute.org/\\\\~{}sroy/condspec/ .an open question in human genetics is what underlies the tissue-specific manifestation of hereditary diseases, which are caused by genomic aberrations that are present in cells across the human body. here we analyzed this phenomenon for over 300 hereditary diseases by using comparative network analysis. we created an extensive resource of protein expression and interactions in 16 main human tissues, by integrating recent data of gene and protein expression across tissues with data of protein-protein interactions ({ppis}). the resulting tissue interaction networks (interactomes) shared a large fraction of their proteins and {ppis}, and only a small fraction of them were tissue-specific. applying this resource to hereditary diseases, we first show that most of the disease-causing genes are widely expressed across tissues, yet, enigmatically, cause disease phenotypes in few tissues only. upon testing for factors that could lead to tissue-specific vulnerability, we find that disease-causing genes tend to have elevated transcript levels and increased number of tissue-specific {ppis} in their disease tissues compared to unaffected tissues. we demonstrate through several examples that these tissue-specific {ppis} can highlight disease mechanisms, and thus, owing to their small number, provide a powerful filter for interrogating disease etiologies. as two thirds of the hereditary diseases are associated with these factors, comparative tissue analysis offers a meaningful and efficient framework for enhancing the understanding of the molecular basis of hereditary diseases. an open question in human genetics is what underlies the tissue-specific manifestation of hereditary diseases, which are caused by genomic aberrations that are present in cells across the entire human body. in order to answer this question, we created an extensive resource of protein expression and interactions across 16 main human tissues. using this resource, we first show that the genes underlying hundreds of hereditary diseases are widely expressed across tissues, yet, enigmatically, cause disease phenotypes in few tissues only. we then identify two distinct, statistically-significant factors that could lead to tissue-specific vulnerability in the face of this broad expression: (i) many disease-causing genes have elevated expression levels in their disease tissues, and (ii) disease-causing genes have a significantly higher tendency for tissue-specific interactions in their disease tissues. as we show for several disease-causing genes, these tissue-specific interactions highlight disease mechanisms and provide an efficient filter for interrogating the molecular basis of diseases. together the two factors we identified are relevant for as many as two thirds of the tissue-specific hereditary diseases. our comparative tissue analysis therefore provides a meaningful and efficient framework for enhancing the understanding of the molecular basis of hereditary diseases.unraveling of the unified networking characteristics of complex networking phenomena is of great interest yet a formidable task. there is currently no simple strategy with a rigorous framework. using an analogy to the exact algebraic property for a transition matrix of a master equation in statistical physics, we propose a method based on a laplacian matrix for the discovery and prediction of new classes in the unsupervised complex networking phenomena where the class of each sample is completely unknown. using this proposed laplacian approach, we can simultaneously discover different classes and determine the identity of each class. through an illustrative test of the laplacian approach applied to real datasets of gene expression profiles, leukemia data [golub {tr}, et al. (1999) science 286:531537], and lymphoma data [alizadeh {aa}, et al. (2000) nature 403:503511], we demonstrate that this approach is accurate and robust with a mathematical and physical realization. it offers a general framework for characterizing any kind of complex networking phenomenon in broad areas irrespective of whether they are supervised or unsupervised. 10.1073/pnas.0708598105the genetics of complex disease produce alterations in the molecular interactions of cellular pathways whose collective effect may become clear through the organized structure of molecular networks. to\\xa0characterize molecular systems associated with late-onset alzheimer?s disease ({load}), we constructed gene-regulatory networks in 1,647 postmortem brain tissues from {load} patients and nondemented subjects, and we demonstrate that {load} reconfigures specific portions of the molecular\\xa0interaction structure. through an integrative network-based approach, we rank-ordered these network structures for relevance to {load} pathology, highlighting an immune- and microglia-specific module that is dominated by genes involved in pathogen phagocytosis, contains {tyrobp} as a key regulator, and is upregulated in {load}. mouse microglia cells overexpressing intact or truncated {tyrobp} revealed expression changes that significantly overlapped the human brain {tyrobp} network. thus the causal network structure is a useful predictor of response to gene perturbations and presents a framework to test models of disease mechanisms underlying {load}.in this paper, we explore modeling overlapping biological processes. we discuss a probabilistic model of overlapping biological processes, gene membership in those processes, and an addition to that model that identifies regulatory mechanisms controlling process activation. a key feature of our approach is that we allow genes to participate in multiple processes, thus providing a more biologically plausible model for the process of gene regulation. we present algorithms to learn each model automatically from data, using only genomewide measurements of gene expression as input. we compare our results to those obtained by other approaches and show that significant benefits can be gained by modeling both the organization of genes into overlapping cellular processes and the regulatory programs of these processes. moreover, our method successfully grouped genes known to function together, recovered many regulatory relationships that are known in the literature, and suggested novel hypotheses regarding the regulatory role of previously uncharacterized proteins.motivation: the complex program of gene expression allows the cell to cope with changing genetic, developmental and environmental conditions. the accumulating large-scale measurements of gene knockout effects and molecular interactions allow us to begin to uncover regulatory and signaling pathways within the cell that connect causal to affected genes on a network of physical {interactions.results}: we present a novel framework, {spine}, for signaling-regulatory pathway {inference}. the framework aims at explaining gene expression experiments in which a gene is knocked out and as a result multiple genes change their expression levels. to this end, an integrated network of protein–protein and {protein-dna} interactions is constructed, and signaling pathways connecting the causal gene to the affected genes are searched for in this network. the reconstruction problem is translated into that of assigning an activation/repression attribute with each protein so as to explain (in expectation) a maximum number of the knockout effects observed. we provide an integer programming formulation for the latter problem and solve it using a commercial {solver.we} validate the method by applying it to a yeast subnetwork that is involved in mating. in cross-validation tests, {spine} obtains very high accuracy in predicting knockout effects (99\\\\%). next, we apply {spine} to the entire yeast network to predict protein effects and reconstruct signaling and regulatory pathways. overall, we are able to infer 861 paths with confidence and assign effects to 183 genes. the predicted effects are found to be in high agreement with current biological {knowledge.availability}: the algorithm and data are available at {http://cs.tau.ac.il/\\\\~{}roded/spine}.{htmlcontact}: roded@post.tau.ac.il10.1101/gr.087890.108 insulin resistance is one of the dominant symptoms of type 2 diabetes ({t2d}). although the molecular mechanisms leading to this resistance are largely unknown, experimental data support that the insulin signaling pathway is impaired in patients who are insulin resistant. to identify novel components/modulators of the insulin signaling pathway, we designed {sirnas} targeting over 300 genes and tested the effects of knocking down these genes in an insulin-dependent, anti-lipolysis assay in {3t3}-l1 adipocytes. for 126 genes, significant changes in free fatty acid release were observed. however, due to off-target effects (in addition to other limitations), high-throughput {rnai}-based screens in cell-based systems generate significant amounts of noise. therefore, to obtain a more reliable set of genes from the {sirna} hits in our screen, we developed and applied a novel network-based approach that elucidates the mechanisms of action for the true positive {sirna} hits. our analysis results in the identification of a core network underlying the insulin signaling pathway that is more significantly enriched for genes previously associated with insulin resistance than the set of genes annotated in the {kegg} database as belonging to the insulin signaling pathway. we experimentally validated one of the predictions, s1pr2, as a novel candidate gene for {t2d}.{background}: a widely-used approach for discovering functional and physical interactions among proteins involves phylogenetic profile comparisons ({ppcs}). here, proteins with similar profiles are inferred to be functionally related under the assumption that proteins involved in the same metabolic pathway or cellular system are likely to have been co-inherited during evolution. {results}: our experimentation with e. coli and yeast proteins with 16 different carefully composed reference sets of genomes revealed that the phyletic patterns of proteins in prokaryotes alone could be adequate enough to make reasonably accurate functional linkage predictions. a slight improvement in performance is observed on adding few eukaryotes into the reference set, but a noticeable drop-off in performance is observed with increased number of eukaryotes. inclusion of most parasitic, pathogenic or vertebrate genomes and multiple strains of the same species into the reference set do not necessarily contribute to an improved sensitivity or accuracy. interestingly, we also found that evolutionary histories of individual pathways have a significant affect on the performance of the {ppc} approach with respect to a particular reference set. for example, to accurately predict functional links in carbohydrate or lipid metabolism, a reference set solely composed of prokaryotic (or bacterial) genomes performed among the best compared to one composed of genomes from all three super-kingdoms; this is in contrast to predicting functional links in translation for which a reference set composed of prokaryotic (or bacterial) genomes performed the worst. we also demonstrate that the widely used random null model to quantify the statistical significance of profile similarity is incomplete, which could result in an increased number of false-positives. {conclusion}: contrary to previous proposals, it is not merely the number of genomes but a careful selection of informative genomes in the reference set that influences the prediction accuracy of the {ppc} approach. we note that the predictive power of the {ppc} approach, especially in eukaryotes, is heavily influenced by the primary endosymbiosis and subsequent bacterial contributions. the over-representation of parasitic unicellular eukaryotes and vertebrates additionally make eukaryotes less useful in the reference sets. reference sets composed of highly non-redundant set of genomes from all three super-kingdoms fare better with pathways showing considerable vertical inheritance and strong conservation (e.g. translation apparatus), while reference sets solely composed of prokaryotic genomes fare better for more variable pathways like carbohydrate metabolism. differential performance of the {ppc} approach on various pathways, and a weak positive correlation between functional and profile similarities suggest that caution should be exercised while interpreting functional linkages inferred from genome-wide large-scale profile comparisons using a single reference set.the analysis of network evolution has been hampered by limited availability of protein interaction data for different organisms. in this study, we investigate evolutionary mechanisms in src homology 3 ({sh3}) domain and kinase interaction networks using high-resolution specificity profiles. we constructed and examined networks for 23 fungal species ranging from saccharomyces cerevisiae to schizosaccharomyces pombe. we quantify rates of different rewiring mechanisms and show that interaction change through binding site evolution is faster than through gene gain or loss. we found that {sh3} interactions evolve swiftly, at rates similar to those found in phosphoregulation evolution. importantly, we show that interaction changes are sufficiently rapid to exhibit saturation phenomena at the observed timescales. finally, focusing on the {sh3} interaction network, we observe extensive clustering of binding sites on target proteins by {sh3} domains and a strong correlation between the number of domains that bind a target protein (target in-degree) and interaction conservation. the relationship between in-degree and interaction conservation is driven by two different effects, namely the number of clusters that correspond to interaction interfaces and the number of domains that bind to each cluster leads to sequence specific conservation, which in turn results in interaction conservation. in summary, we uncover several network evolution mechanisms likely to generalize across peptide recognition modules. protein interaction networks control virtually all cellular processes. the rules governing their evolution have remained elusive, as comprehensive protein interaction data is available for only a small number of very distant species, making evolutionary network studies difficult. here we attempt to overcome this limitation by computationally constructing protein interaction networks for 23 relatively tightly spaced yeast species. we focus on networks consisting of kinase and peptide binding domain interactions, which play central roles in signaling pathways. these networks enable us to investigate evolutionary network mechanisms. we are able, for the first time, to accurately quantify the contribution of different rewiring mechanisms. interaction change appears to be mainly accomplished through binding site evolution rather than through gene gain or loss. this is in contrast to other evolutionary processes, where gene duplication or deletion is a major driving factor. moreover, our analysis reveals that interaction changes are very fast – fast enough that the number of changes saturates, i.e., the actual rate of change has been strongly underestimated in previous studies. our analysis also reveals different mechanisms by which certain interactions are conserved throughout evolution. our results likely transfer to other species and networks, and will benefit future evolutionary studies of signaling pathways.immense populations of viruses are present in the human gut and other body sites. understanding the role of these populations (the human \"virome\") in health and disease requires a much deeper understanding of their composition and dynamics in the face of environmental perturbation. here, we investigate viromes from human subjects on a controlled feeding regimen. longitudinal fecal samples were analyzed by metagenomic sequencing of {dna} from virus-like particles ({vlp}) and total microbial communities. assembly of 336 mb of {vlp} sequence yielded 7175 contigs, many identifiable as complete or partial bacteriophage genomes. contigs were rich in viral functions required in lytic and lysogenic growth, as well as unexpected functions such as viral {crispr} arrays and genes for antibiotic resistance. the largest source of variance among virome samples was interpersonal variation. parallel deep-sequencing analysis of bacterial populations showed covaration of the virome with the larger microbiome. the dietary intervention was associated with a change in the virome community to a new state, in which individuals on the same diet converged. thus these data provide an overview of the composition of the human gut virome and associate virome structure with diet.we review statistical methods for high-dimensional data analysis and pay particular attention to recent developments for assessing uncertainties in terms of controlling false positive statements (type i error) and p-values. the main focus is on regression models, but we also discuss graphical modeling and causal inference based on observational data. we illustrate the concepts and methods with various packages from the statistical software using a high-throughput genomic data set about riboflavin production with bacillus subtilis, which we make publicly available for the first time.multiple abiotic stimuli are integrated in the regulation of rice gene expression under field conditions | plants rely on transcriptional dynamics to respond to multiple climatic fluctuations and contexts in nature. we analyzed genome-wide gene expression patterns of rice (oryza sativa) growing in rainfed and irrigated fields during two distinct tropical seasons and determined simple linear models that relate transcriptomic variation to climatic fluctuations. these models combine multiple environmental parameters to account for patterns of expression in the field of co-expressed gene clusters. we examined the correspondence of our environmental models between tropical and temperate field conditions, using previously published data. we found that field type and macroclimate had broad impacts on transcriptional responses to environmental fluctuations, especially for genes involved in photosynthesis and development. nevertheless, variation in solar radiation and temperature at the timescale of hours had reproducible effects across environmental contexts. these results provide a basis for broad-based predictive modeling of plant gene expression in the field.the knowledge of protein and domain interactions provide crucial insights into their function within a cell. several computational methods have been proposed to detect interactions between proteins and their constitutive domains. in this work, we focus on approaches based on correlated evolution (coevolution) of sequences of interacting proteins. in this type of approach, often referred to as the mirrortree method, a high correlation of evolutionary histories of two proteins is used as an indicator to predict protein interactions. recently, it has been observed that subtracting the underlying speciation process by separating coevolution due to common speciation divergence from that due to common function of interacting pairs greatly improves the predictive power of the mirrortree approach. in this article, we investigate possible improvements and limitations of this method. in particular, we demonstrate that the performance of the mirrortree method that can be further improved by restricting the coevolution analysis to the relatively conserved regions in the protein domain sequences (disregarding highly divergent regions). we provide a theoretical validation of our results leading to new insights into the interplay between coevolution and speciation of interacting proteins.little is known about broad patterns of variation and evolution of gene expression during any developmental process. here we investigate variation in genome-wide gene expression among drosophila simulans, drosophila yakuba and four strains of drosophila melanogaster during a major developmental transition—the start of metamorphosis. differences in gene activity between these lineages follow a phylogenetic pattern, and 27\\\\% of all of the genes in these genomes differ in their developmental gene expression between at least two strains or species. we identify, on a gene-by-gene basis, the evolutionary forces that shape this variation and show that, both within the transcriptional network that controls metamorphosis and across the whole genome, the expression changes of transcription factor genes are relatively stable, whereas those of their downstream targets are more likely to have evolved. our results demonstrate extensive evolution of developmental gene expression among closely related species.our current understanding of how {dna} is packed in the nucleus is most accurate at the fine scale of individual nucleosomes and at the large scale of chromosome territories. however, accurate modeling of {dna} architecture at the intermediate scale of ∼50 kb-10 mb is crucial for identifying functional interactions among regulatory elements and their target promoters. we describe a method, {fit-hi}-c, that assigns statistical confidence estimates to mid-range intra-chromosomal contacts by jointly modeling the random polymer looping effect and previously observed technical biases in {hi-c} data sets. we demonstrate that our proposed approach computes accurate empirical null models of contact probability without any distribution assumption, corrects for binning artifacts, and provides improved statistical power relative to a previously described method. high-confidence contacts identified by {fit-hi}-c preferentially link expressed gene promoters to active enhancers identified by chromatin signatures in human embryonic stem cells ({escs}), capture 77\\\\% of {rna} polymerase {ii}-mediated enhancer-promoter interactions identified using {chia}-{pet} in mouse {escs}, and confirm previously validated, cell line-specific interactions in mouse cortex cells. we observe that insulators and heterochromatin regions are hubs for high-confidence contacts, while promoters and strong enhancers are involved in fewer contacts. we also observe that binding peaks of master pluripotency factors such as {nanog} and {pou5f1} are highly enriched in high-confidence contacts for human {escs}. furthermore, we show that pairs of loci linked by high-confidence contacts exhibit similar replication timing in human and mouse {escs} and preferentially lie within the boundaries of topological domains for human and mouse cell lines. {\\\\copyright} 2014 ay et al.; published by cold spring harbor laboratory press.the saccharomyces genome database ({sgd}; http://www.yeastgenome.org/), a scientific database of the molecular biology and genetics of the yeast saccharomyces cerevisiae, has recently developed several new resources that allow the comparison and integration of information on a genome-wide scale, enabling the user not only to find detailed information about individual genes, but also to make connections across groups of genes with common features and across different species. the fungal alignment viewer displays alignments of sequences from multiple fungal genomes, while the sequence similarity query tool displays {psi}-{blast} alignments of each s.cerevisiae protein with similar proteins from any species whose sequences are contained in the non-redundant (nr) protein data set at {ncbi}. the yeast biochemical pathways tool integrates groups of genes by their common roles in metabolism and displays the metabolic pathways in a graphical form. finally, the find chromosomal features search interface provides a versatile tool for querying multiple types of information in {sgd}.defining protein complexes is critical to virtually all aspects of cell biology. two recent affinity purification/mass spectrometry studies in saccharomyces cerevisiae have vastly increased the available protein interaction data. the practical utility of such high throughput interaction sets, however, is substantially decreased by the presence of false positives. here we created a novel probabilistic metric that takes advantage of the high density of these data, including both the presence and absence of individual associations, to provide a measure of the relative confidence of each potential protein-protein interaction. this analysis largely overcomes the noise inherent in high throughput immunoprecipitation experiments. for example, of the 12,122 binary interactions in the general repository of interaction data ({biogrid}) derived from these two studies, we marked 7504 as being of substantially lower confidence. additionally, applying our metric and a stringent cutoff we identified a set of 9074 interactions (including 4456 that were not among the 12,122 interactions) with accuracy comparable to that of conventional small scale methodologies. finally we organized proteins into coherent multisubunit complexes using hierarchical clustering. this work thus provides a highly accurate physical interaction map of yeast in a format that is readily accessible to the biological community.cellular decision making is the process whereby cells assume different, functionally important and heritable fates without an associated genetic or environmental difference. such stochastic cell fate decisions generate nongenetic cellular diversity, which may be critical for metazoan development as well as optimized microbial resource utilization and survival in a fluctuating, frequently stressful environment. here, we review several examples of cellular decision making from viruses, bacteria, yeast, lower metazoans, and mammals, highlighting the role of regulatory network structure and molecular noise. we propose that cellular decision making is one of at least three key processes underlying development at various scales of biological organization. copyright {\\\\copyright} 2011 elsevier inc. all rights reserved.relatively little is known about the mechanisms through which gene expression regulation evolves. to investigate this, we systematically explored the conservation of regulatory networks in fungi by examining the cis-regulatory elements that govern the expression of coregulated genes. we first identified groups of coregulated saccharomyces cerevisiae genes enriched for genes with known upstream or downstream cis-regulatory sequences. reasoning that many of these gene groups are coregulated in related species as well, we performed similar analyses on orthologs of coregulated s. cerevisiae genes in 13 other ascomycete species. we find that many species-specific gene groups are enriched for the same flanking regulatory sequences as those found in the orthologous gene groups from s. cerevisiae, indicating that those regulatory systems have been conserved in multiple ascomycete species. in addition to these clear cases of regulatory conservation, we find examples of cis-element evolution that suggest multiple modes of regulatory diversification, including alterations in transcription factor-binding specificity, incorporation of new gene targets into an existing regulatory system, and cooption of regulatory systems to control a different set of genes. we investigated one example in greater detail by measuring the in vitro activity of the s. cerevisiae transcription factor rpn4p and its orthologs from candida albicans and neurospora crassa. our results suggest that the {dna} binding specificity of these proteins has coevolved with the sequences found upstream of the rpn4p target genes and suggest that rpn4p has a different function in n. crassa.comparative genomics provides a general methodology for discovering functional {dna} elements and understanding their evolution. the availability of many related genomes enables more powerful analyses, but requires rigorous phylogenetic methods to resolve orthologous genes and regions. here, we use 12 recently sequenced drosophila genomes and nine fungal genomes to address the problem of accurate gene-tree reconstruction across many complete genomes. we show that existing phylogenetic methods that treat each gene tree in isolation show large-scale inaccuracies, largely due to insufficient phylogenetic information in individual genes. however, we find that gene trees exhibit common properties that can be exploited for evolutionary studies and accurate phylogenetic reconstruction. evolutionary rates can be decoupled into gene-specific and species-specific components, which can be learned across complete genomes. we develop a phylogenetic reconstruction methodology that exploits these properties and achieves significantly higher accuracy, addressing the species-level heterotachy and enabling studies of gene evolution in the context of species evolution.modeling relational data is an important problem for modern data analysis and machine learning. in this paper we propose a bayesian model that uses a hierarchy of probabilistic assumptions about the way objects interact with one another in order to learn latent groups, their typical interaction patterns, and the degree of membership of objects to groups. our model explains the data using a small set of parameters that can be reliably estimated with an efficient inference algorithm. in our approach, the set of probabilistic assumptions may be tailored to a specific application domain in order to incorporate intuitions and/or semantics of interest. we demonstrate our methods on simulated data and we successfully apply our model to a data set of protein-to-protein interactions.protein interactions regulate the systems-level behavior of cells; thus, deciphering the structure and dynamics of protein interaction networks in their cellular context is a central goal in biology. we have performed a genome-wide in vivo screen for protein-protein interactions in saccharomyces cerevisiae by means of a protein-fragment complementation assay ({pca}). we identified 2770 interactions among 1124 endogenously expressed proteins. comparison with previous studies confirmed known interactions, but most were not known, revealing a previously unexplored subspace of the yeast protein interactome. the {pca} detected structural and topological relationships between proteins, providing an 8-nanometer-resolution map of dynamically interacting complexes in vivo and extended networks that provide insights into fundamental cellular processes, including cell polarization and autophagy, pathways that are evolutionarily conserved and central to both development and human health. 10.1126/science.1153878genome-scale screening studies are gradually accumulating a wealth of data on the putative involvement of hundreds of genes/proteins in various cellular responses or functions. a fundamental challenge is to chart out the protein pathways that underlie these systems. previous approaches to the problem have either employed a local optimization criterion, aiming to infer each pathway independently, or a global criterion, searching for the overall most parsimonious subnetwork. here, we study the trade-off between the two approaches and present a new intermediary scheme that provides explicit control over it. we demonstrate its utility in the analysis of the apoptosis network in humans, and the telomere length maintenance ({tlm}) system in yeast. our results show that in the majority of real-life cases, the intermediary approach provides the most plausible solutions. we use a new set of perturbation experiments measuring the role of essential genes in telomere length regulation to further study the {tlm} network. surprisingly, we find that the proteasome plays an important role in telomere length regulation through its associations with transcription and {dna} repair circuits.studies of gene regulation often utilize genome-wide predictions of transcription factor ({tf}) binding sites. most existing prediction methods are based on sequence information alone, ignoring biological contexts such as developmental stages and tissue types. experimental methods to study in vivo binding, including {chip}-chip and {chip}-seq, can only study one transcription factor in a single cell type and under a specific condition in each experiment, and therefore cannot scale to determine the full set of regulatory interactions in mammalian transcriptional regulatory networks. we developed a new computational approach, {pipes}, for predicting tissue-specific {tf} binding. {pipes} integrates in vitro protein binding microarrays ({pbms}), sequence conservation and tissue-specific epigenetic ({dnase} i hypersensitivity) information. we demonstrate that {pipes} improves over existing methods on distinguishing between in vivo bound and unbound sequences using {chip}-seq data for 11 mouse {tfs}. in addition, our predictions are in good agreement with current knowledge of tissue-specific {tf} regulation. we provide a systematic map of computationally predicted tissue-specific binding targets for 284 mouse {tfs} across 55 tissue/cell types. such comprehensive resource is useful for researchers studying gene regulation.genome-wide maps of transcription factor ({tf}) occupancy and regions of open chromatin implicitly contain {dna} sequence signals for multiple factors. we present {seqgl}, a novel de novo motif discovery algorithm to identify multiple {tf} sequence signals from {chip}-, {dnase}-, and {atac}-seq profiles. {seqgl} trains a discriminative model using a k-mer feature representation together with group lasso regularization to extract a collection of sequence signals that distinguish peak sequences from flanking regions. benchmarked on over 100 {chip}-seq experiments, {seqgl} outperformed traditional motif discovery tools in discriminative accuracy. furthermore, {seqgl} can be naturally used with multitask learning to identify genomic and cell-type context determinants of {tf} binding. {seqgl} successfully scales to the large multiplicity of sequence signals in {dnase}- or {atac}-seq maps. in particular, {seqgl} was able to identify a number of {chip}-seq validated sequence signals that were not found by traditional motif discovery algorithms. thus compared to widely used motif discovery algorithms, {seqgl} demonstrates both greater discriminative accuracy and higher sensitivity for detecting the {dna} sequence signals underlying regulatory element maps. {seqgl} is available at {http://cbio.mskcc.org/public/leslie}/{seqgl}/. transcriptional regulation is the cell\\'s primary mode of controlling gene expression. transcription factors ({tfs}) are proteins that recognize and bind specific {dna} sequence signals to regulate the expression of target genes. recent years have seen the rapid development of genome-wide assays to profile the binding locations of a single {tf} or, more generally, regions of open chromatin that are occupied by a complex repertoire of {dna} binding factors. new methods are therefore needed to detect and represent {dna} sequence signals in these genome-wide regulatory element maps. here we present a novel tool called {seqgl} to extract multiple {tf} binding signals from genome-wide maps. {seqgl} employs a machine learning framework to identify features that best discriminate the peaks, where we expect {dna} sequence signals to occur, from the flank regions that should not contain these signals. our tool performed significantly better than widely used motif discovery methods in discriminative accuracy and achieved higher sensitivity in detecting the numerous sequence signals underlying regulatory element maps.failure of cells to respond to {dna} damage is a primary event associated with mutagenesis and environmental toxicity. to map the transcriptional network controlling the damage response, we measured genomewide binding locations for 30 damage-related transcription factors ({tfs}) after exposure of yeast to methyl-methanesulfonate ({mms}). the resulting 5272 {tf}-target interactions revealed extensive changes in the pattern of promoter binding and identified damage-specific binding motifs. as systematic functional validation, we identified interactions for which the target changed expression in wild-type cells in response to {mms} but was nonresponsive in cells lacking the {tf}. validated interactions were assembled into causal pathway models that provide global hypotheses of how signaling, transcription, and phenotype are integrated after damage.the gene regulatory circuitry through which pluripotent embryonic stem ({es}) cells choose between self-renewal and differentiation appears vast and has yet to be distilled into an executive molecular program. we developed a data-constrained, computational approach to reduce complexity and to derive a set of functionally validated components and interaction combinations sufficient to explain observed {es} cell behavior. this minimal set, the simplest version of which comprises only 16 interactions, 12 components, and three inputs, satisfies all prior specifications for self-renewal and furthermore predicts unknown and nonintuitive responses to compound genetic perturbations with an overall accuracy of 70\\\\%. we propose that propagation of {es} cell identity is not determined by a vast interactome but rather can be explained by a relatively simple process of molecular computation. copyright {\\\\copyright} 2014, american association for the advancement of science.the causal discovery from data is important for various scientific investigations. because we cannot distinguish the different directed acyclic graphs ({dags}) in a markov equivalence class learned from observational data, we have to collect further information on causal structures from experiments with external interventions. in this paper, we propose an active learning approach for discovering causal structures in which we first find a markov equivalence class from observational data, and then we orient undirected edges in every chain component via intervention experiments separately. in the experiments, some variables are manipulated through external interventions. we discuss two kinds of intervention experiments, randomized experiment and quasi-experiment. furthermore, we give two optimal designs of experiments, a batch-intervention design and a sequential-intervention design, to minimize the number of manipulated variables and the set of candidate structures based on the minimax and the maximum entropy criteria. we show theoretically that structural learning can be done locally in subgraphs of chain components without need of checking illegal v-structures and cycles in the whole network and that a markov equivalence subclass obtained after each intervention can still be depicted as a chain graph.developmental progression is driven by specific spatiotemporal domains of gene expression, which give rise to stereotypically patterned embryos even in the presence of environmental and genetic variation. views of how transcription factors regulate gene expression are changing owing to recent genome-wide studies of transcription factor binding and {rna} expression. such studies reveal patterns that, at first glance, seem to contrast with the robustness of the developmental processes they encode. here, we review our current knowledge of transcription factor function from genomic and genetic studies and discuss how different strategies, including extensive cooperative regulation (both direct and indirect), progressive priming of regulatory elements, and the integration of activities from multiple enhancers, confer specificity and robustness to transcriptional regulation during development.motivation: network-based gene function inference methods have proliferated in recent years, but measurable progress remains elusive. we wished to better explore performance trends by controlling data and algorithm implementation, with a particular focus on the performance of aggregate predictions.we compared the response of saccharomyces cerevisiae to carbon (glucose) and nitrogen (ammonia) limitation in chemostat cultivation at the proteome level. protein levels were differentially quantified using unlabeled and {15n} metabolically labeled yeast cultures. a total of 928 proteins covering a wide range of isoelectric points, molecular weights and subcellular localizations were identified. stringent statistical analysis identified 51 proteins upregulated in response to glucose limitation and 51 upregulated in response to ammonia limitation. under glucose limitation, typical glucose-repressed genes encoding proteins involved in alternative carbon source utilization, fatty acids beta-oxidation and oxidative phosphorylation displayed an increased protein level. proteins upregulated in response to nitrogen limitation were mostly involved in scavenging of alternative nitrogen sources and protein degradation. comparison of transcript and protein levels clearly showed that upregulation in response to glucose limitation was mainly transcriptionally controlled, whereas upregulation in response to nitrogen limitation was essentially controlled at the post-transcriptional level by increased translational efficiency and/or decreased protein degradation. these observations underline the need for multilevel analysis in yeast systems biology.a complete description of protein metabolism requires knowledge of the rates of protein production and destruction within cells. using an epitope-tagged strain collection, we measured the half-life of >3,750 proteins in the yeast proteome after inhibition of translation. by integrating our data with previous measurements of protein and {mrna} abundance and translation rate, we provide evidence that many proteins partition into one of two regimes for protein metabolism: one optimized for efficient production or a second optimized for regulatory efficiency. incorporation of protein half-life information into a simple quantitative model for protein production improves our ability to predict steady-state protein abundance values. analysis of a simple dynamic protein production model reveals a remarkable correlation between transcriptional regulation and protein half-life within some groups of coregulated genes, suggesting that cells coordinate these two processes to achieve uniform effects on protein abundances. our experimental data and theoretical analysis underscore the importance of an integrative approach to the complex interplay between protein degradation, transcriptional regulation, and other determinants of protein metabolism.natural genetic variation can cause significant differences in gene expression, but little is known about the polymorphisms that affect gene regulation. we analyzed regulatory variation in a cross between laboratory and wild strains of saccharomyces cerevisiae. clustering and linkage analysis defined groups of coregulated genes and the loci involved in their regulation. most expression differences mapped to trans-acting loci. positional cloning and functional assays showed that polymorphisms in {gpa1} and {amn1} affect expression of genes involved in pheromone response and daughter cell separation, respectively. we also asked whether particular classes of genes were more likely to contain trans-regulatory polymorphisms. notably, transcription factors showed no enrichment, and trans-regulatory variation seems to be broadly dispersed across classes of genes with different molecular functions.transcription factor ({tf}) binding and histone modification ({hm}) are important for the precise control of gene expression. hence, we constructed statistical models to relate these to gene expression levels in mouse embryonic stem cells. while both {tf} binding and {hms} are highly \\'predictive\\' of gene expression levels (in a statistical, but perhaps not strictly mechanistic, sense), we find they show distinct differences in the spatial patterning of their predictive strength: {tf} binding achieved the highest predictive power in a small {dna} region centered at the transcription start sites of genes, while the {hms} exhibited high predictive powers across a wide region around genes. intriguingly, our results suggest that {tf} binding and {hms} are redundant in strict statistical sense for predicting gene expression. we also show that our {tf} and {hm} models are cell line specific; specifically, {tf} binding and {hm} are more predictive of gene expression in the same cell line, and the differential gene expression between cell lines is predictable by differential {hms}. finally, we found that the models trained solely on protein-coding genes are predictive of expression levels of {micrornas}, suggesting that their regulation by {tfs} and {hms} may share a similar mechanism to that for protein-coding genes.motivation: most existing methods for predicting causal disease genes rely on specific type of evidence, and are therefore limited in terms of applicability. more often than not, the type of evidence available for diseases varies—for example, we may know linked genes, keywords associated with the disease obtained by mining text, or co-occurrence of disease symptoms in patients. similarly, the type of evidence available for genes varies—for example, specific microarray probes convey information only for certain sets of genes. in this article, we apply a novel matrix-completion method called inductive matrix completion to the problem of predicting gene-disease associations; it combines multiple types of evidence (features) for diseases and genes to learn latent factors that explain the observed gene–disease associations. we construct features from different biological sources such as microarray expression data and disease-related textual data. a crucial advantage of the method is that it is inductive; it can be applied to diseases not seen at training time, unlike traditional matrix-completion approaches and network-based inference methods that are transductive.next-generation sequencing ({ngs}) has revolutionized the study of cancer genomes. however, the reads obtained from {ngs} of tumor samples often consist of a mixture of normal and tumor cells, which themselves can be of multiple clonal types. a prominent problem in the analysis of cancer genome sequencing data is deconvolving the mixture to identify the reads associated with tumor cells or a particular subclone of tumor cells. solving the problem is, however, challenging because of the so-called \\'identifiability problem\\', where different combinations of tumor purity and ploidy often explain the sequencing data equally well. we propose a new model to resolve the identifiability problem by integrating two types of sequencing information-somatic copy number alterations and loss of heterozygosity-within a unified probabilistic framework. we derive algorithms to solve our model, and implement them in a software package called {pyloh}. we benchmark the performance of {pyloh} using both simulated data and 12 breast cancer sequencing datasets and show that {pyloh} outperforms existing methods in disambiguating the identifiability problem and estimating tumor purity. the {pyloh} package is written in python and is publicly available at {https://github.com/uci-cbcl/pyloh}. xhx@ics.uci.edu supplementary data are available at bioinformatics online. {\\\\copyright} the author 2014. published by oxford university press. all rights reserved. for permissions, please e-mail: journals.permissions@oup.com.eusocial behavior has arisen in few animal groups, most notably in the aculeate hymenoptera, a clade comprising ants, bees, and stinging wasps [1\\x134]. phylogeny is crucial to understanding the evolution of the salient features of these insects, including eusociality [5]. yet the phylogenetic relationships among the major lineages of aculeate hymenoptera remain contentious [6\\x1312]. we address this problem here by generating and analyzing genomic data for a representative series of taxa. we obtain a single well-resolved and strongly supported tree, robust to multiple methods of phylogenetic inference. apoidea (spheciform wasps and bees) and ants are sister groups, a novel finding that contradicts earlier views that ants are closer to ectoparasitoid wasps. vespid wasps (paper wasps, yellow jackets, and relatives) are sister to all other aculeates except chrysidoids. thus, all eusocial species of hymenoptera are contained within two major groups, characterized by transport of larval provisions and nest construction, likely prerequisites for the evolution of eusociality. these two lineages are interpolated among three other clades of wasps whose species are predominantly ectoparasitoids on concealed hosts, the inferred ancestral condition for aculeates [2]. this phylogeny provides a new framework for exploring the evolution of nesting, feeding, and social behavior within the stinging hymenoptera. \"genomic data resolve the phylogenetic relationships of ants, bees, and stinging {wasps\"ants} are revealed to be the sister group of spheciform wasps and bees ({apoidea)\"typical} social wasps (e.g., yellow jackets) are not closely related to ants and {bees\"clades} containing eusocial taxa are interposed among solitary ectoparasitoid waspsevolutionary adaptation can be rapid and potentially help species counter stressful conditions or realize ecological opportunities arising from climate change. the challenges are to understand when evolution will occur and to identify potential evolutionary winners as well as losers, such as species lacking adaptive capacity living near physiological limits. evolutionary processes also need to be incorporated into management programmes designed to minimize biodiversity loss under rapid climate change. these challenges can be met through realistic models of evolutionary change linked to experimental data across a range of taxa.there is increasing, but largely indirect, evidence pointing to an effect of commensal gut microbiota on the central nervous system ({cns}). however, it is unknown whether lactic acid bacteria such as lactobacillus rhamnosus could have a direct effect on neurotransmitter receptors in the {cns} in normal, healthy animals. {gaba} is the main {cns} inhibitory neurotransmitter and is significantly involved in regulating many physiological and psychological processes. alterations in central {gaba} receptor expression are implicated in the pathogenesis of anxiety and depression, which are highly comorbid with functional bowel disorders. in this work, we show that chronic treatment with l. rhamnosus ({jb}-1) induced region-dependent alterations in {gabab1b} {mrna} in the brain with increases in cortical regions (cingulate and prelimbic) and concomitant reductions in expression in the hippocampus, amygdala, and locus coeruleus, in comparison with control-fed mice. in addition, l. rhamnosus ({jb}-1) reduced {gabaaα2} {mrna} expression in the prefrontal cortex and amygdala, but increased {gabaaα2} in the hippocampus. importantly, l. rhamnosus ({jb}-1) reduced stress-induced corticosterone and anxiety- and depression-related behavior. moreover, the neurochemical and behavioral effects were not found in vagotomized mice, identifying the vagus as a major modulatory constitutive communication pathway between the bacteria exposed to the gut and the brain. together, these findings highlight the important role of bacteria in the bidirectional communication of the gut–brain axis and suggest that certain organisms may prove to be useful therapeutic adjuncts in stress-related disorders such as anxiety and depression.mammalian genomes are organized into megabase-scale topologically associated domains ({tads}). we demonstrate that disruption of {tads} can rewire long-range regulatory architecture and result in pathogenic phenotypes. we show that distinct human limb malformations are caused by deletions, inversions, or duplications altering the structure of the {tad}-spanning {wnt6}/{ihh}/{epha4}/{pax3} locus. using {crispr}/cas genome editing, we generated mice with corresponding rearrangements. both in mouse limb tissue and patient-derived fibroblasts, disease-relevant structural changes cause ectopic interactions between promoters and non-coding {dna}, and a cluster of limb enhancers normally associated with epha4 is misplaced relative to {tad} boundaries and drives ectopic limb expression of another gene in the locus. this rewiring occurred only if the variant disrupted a {ctcf}-associated boundary domain. our results demonstrate the functional importance of {tads} for orchestrating gene expression via genome architecture and indicate criteria for predicting the pathogenicity of human structural variants, particularly in non-coding regions of the human genome.a fundamental goal in biology is to achieve a mechanistic understanding of how and to what extent ecological variation imposes selection for distinct traits and favors the fixation of specific genetic variants. key to such an understanding is the detailed mapping of the natural genomic and phenomic space and a bridging of the gap that separates these worlds. here we chart a high-resolution map of natural trait variation in one of the most important genetic model organisms, the budding yeast saccharomyces cerevisiae, and its closest wild relatives and trace the genetic basis and timing of major phenotype changing events in its recent history. we show that natural trait variation in s. cerevisiae exceeds that of its relatives, despite limited genetic variation, and follows the population history rather than the source environment. in particular, the west african population is phenotypically unique, with an extreme abundance of low-performance alleles, notably a premature translational termination signal in {gal3} that cause inability to utilize galactose. our observations suggest that many s. cerevisiae traits may be the consequence of genetic drift rather than selection, in line with the assumption that natural yeast lineages are remnants of recent population bottlenecks. disconcertingly, the universal type strain {s288c} was found to be highly atypical, highlighting the danger of extrapolating gene-trait connections obtained in mosaic, lab-domesticated lineages to the species as a whole. overall, this study represents a step towards an in-depth understanding of the causal relationship between co-variation in ecology, selection pressure, natural traits, molecular mechanism, and alleles in a key model organism. an overall aim in modern biology is to achieve an in-depth understanding of an organism\\'s physiology in the context of its ecology and historic selective pressures that have been acting on its genome. the baker\\'s yeast, saccharomyces cerevisiae, has a peculiar life history completely dominated by clonal reproduction and self-fertilization, prompting the suggestion that natural yeasts are remnants of repeated population bottlenecks in essentially clonal lineages. such a life history dominated by mitotic proliferation purports a strong evolutionary influence of genetic drift and predicts trait variation to be high and largely defined by the genetic history of each population. here we chart a highly resolved map of natural trait variation in s. cerevisiae and its closest non-domesticated relative, saccharomyces paradoxus, and confirm this prediction. we found that trait variation in budding yeast is indeed high and largely defined by population rather than source environment. in particular, the west african population was found to be phenotypically unique with an extreme abundance of low-performance alleles. our findings support the idea of population bottlenecks in the recent yeast evolutionary history and a large influence of genetic drift.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismscellular reprogramming highlights the epigenetic plasticity of the somatic cell state. long noncoding {rnas} ({lncrnas}) have emerging roles in epigenetic regulation, but their potential functions in reprogramming cell fate have been largely unexplored. we used single-cell {rna} sequencing to characterize the expression patterns of over 16,000 genes, including 437 {lncrnas}, during defined stages of reprogramming to pluripotency. self-organizing maps ({soms}) were used as an intuitive way to structure and interrogate transcriptome data at the single-cell level. early molecular events during reprogramming involved the activation of ras signaling pathways, along with hundreds of {lncrnas}. loss-of-function studies showed that activated {lncrnas} can repress lineage-specific genes, while {lncrnas} activated in multiple reprogramming cell types can regulate metabolic gene expression. our findings demonstrate that reprogramming cells activate defined sets of functionally relevant {lncrnas} and provide a resource to further investigate how dynamic changes in the transcriptome reprogram cell state.we used genome-wide expression analysis to explore how gene expression in saccharomyces cerevisiae is remodeled in response to various changes in extracellular environment, including changes in temperature, oxidation, nutrients, {ph}, and osmolarity. the results demonstrate that more than half of the genome is involved in various responses to environmental change and identify the global set of genes induced and repressed by each condition. these data implicate a substantial number of previously uncharacterized genes in these responses and reveal a signature common to environmental responses that involves approximately 10\\\\% of yeast genes. the results of expression analysis with {msn2}/{msn4} mutants support the model that the {msn2/msn4} activators induce the common response to environmental change. these results provide a global description of the transcriptional response to environmental change and extend our understanding of the role of activators in effecting this response.the understanding of selective constraints affecting genes is a major issue in biology. it is well established that gene expression level is a major determinant of the rate of protein evolution, but the reasons for this relationship remain highly debated. here we demonstrate that gene expression is also a major determinant of the evolution of gene dosage: the rate of gene losses after whole genome duplications in the paramecium lineage is negatively correlated to the level of gene expression, and this relationship is not a byproduct of other factors known to affect the fate of gene duplicates. this indicates that changes in gene dosage are generally more deleterious for highly expressed genes. this rule also holds for other taxa: in yeast, we find a clear relationship between gene expression level and the fitness impact of reduction in gene dosage. to explain these observations, we propose a model based on the fact that the optimal expression level of a gene corresponds to a trade-off between the benefit and cost of its expression. this {costex} model predicts that selective pressure against mutations changing gene expression level or affecting the encoded protein should on average be stronger in highly expressed genes and hence that both the frequency of gene loss and the rate of protein evolution should correlate negatively with gene expression. thus, the {costex} model provides a simple and common explanation for the general relationship observed between the level of gene expression and the different facets of gene evolution. the analysis of gene evolution is a powerful approach to recognize the genetic features that contribute to the fitness of organisms. it was shown previously that selective constraints on protein sequences increase with expression level. this observation was surprising because there is a priori no reason why lowly expressed genes should be less important than highly expressed genes for the proper function of an organism. here we show that selective pressure on the evolution of gene dosage, which is another important aspect of gene evolution, is also directly dependent on gene expression level. to explain these observations, we propose a model based on the fact that gene expression is a costly process (notably protein synthesis), so that there is an optimal expression level for each gene corresponding to a trade-off between the benefit and the cost of its expression. this model predicts that selective pressure on gene expression level or on the encoded protein should on average be stronger in highly expressed genes, providing a simple and common explanation for the general relationship observed between gene expression and the different facets of gene evolution.the analysis of large-scale genome-wide experiments carries the promise of dramatically broadening our understanding on biological networks. the challenge of systematic integration of experimental results with established biological knowledge on a pathway is still unanswered. here we present a methodology that attempts to answer this challenge when investigating signaling pathways. we formalize existing qualitative knowledge as a probabilistic model that depicts known interactions between molecules (genes, proteins, etc.) as a network and known regulatory relations as logics. we present algorithms that analyze experimental results (e.g., transcription profiles) vis-\\\\`{a}-vis the model and propose improvements to the model based on the fit to the experimental data. these algorithms refine the relations between model components, as well as expand the model to include new components that are regulated by components of the original network. using our methodology, we have modeled together the knowledge on four established signaling pathways related to osmotic shock response in saccharomyces cerevisiae. using over 100 published transcription profiles, our refinement methodology revealed three cross talks in the network. the expansion procedure identified with high confidence large groups of genes that are coregulated by transcription factors from the original network via a common logic. the results reveal a novel delicate repressive effect of the {hog} pathway on many transcriptional target genes and suggest an unexpected alternative functional mode of the {map} kinase hog1. these results demonstrate that, by integrated analysis of data and of well-defined knowledge, one can generate concrete biological hypotheses about signaling cascades and their downstream regulatory programs.we address the problem of comparing and characterizing the promoter regions of genes with similar expression patterns. this remains a challenging problem in sequence analysis, because often the promoter regions of co-expressed genes do not show discernible sequence conservation. in our approach, thus, we have not directly compared the nucleotide sequence of promoters. instead, we have obtained predictions of transcription factor binding sites, annotated the predicted sites with the labels of the corresponding binding factors, and aligned the resulting sequences of labels—to which we refer here as transcription factor maps ({tf}-maps). to obtain the global pairwise alignment of two {tf}-maps, we have adapted an algorithm initially developed to align restriction enzyme maps. we have optimized the parameters of the algorithm in a small, but well-curated, collection of human–mouse orthologous gene pairs. results in this dataset, as well as in an independent much larger dataset from the {cisred} database, indicate that {tf}-map alignments are able to uncover conserved regulatory elements, which cannot be detected by the typical sequence alignments. sequence comparisons and alignments are among the most powerful tools in research in biology. since similar sequences play, in general, similar functions, identification of sequence conservation between two or more nucleotide or amino acid sequences is often used to infer common biological functionality. sequence comparisons, however, have limitations; often similar functions are encoded by higher order elements which do not hold a univocal relationship to the underlying primary sequence. in consequence, similar functions are frequently encoded by diverse sequences. promoter regions are a case in point. often, promoter sequences of genes with similar expression patterns do not show conservation. this is because, even though their expression may be regulated by a similar arrangement of transcription factors, the binding sites for these factors may exhibit great sequence variability. to overcome this limitation, the authors obtain predictions of transcription factor binding sites on promoter sequences, and annotate the predicted sites with the labels of the corresponding transcription factors. they develop an algorithm—inspired in an early algorithm to align restriction enzyme maps—to align the resulting sequence of labels—the so-called {tf}-maps (transcription factor maps). they show that {tf}-map alignments are able to uncover conserved regulatory elements common to the promoter regions of co-regulated genes, but those regulatory elements cannot be detected by typical sequence alignments.the effective control of tuberculosis ({tb}) has been thwarted by the need for prolonged, complex and potentially toxic drug regimens, by reliance on an inefficient vaccine and by the absence of biomarkers of clinical status. the promise of the genomics era for {tb} control is substantial, but has been hindered by the lack of a central repository that collects and integrates genomic and experimental data about this organism in a way that can be readily accessed and analyzed. the tuberculosis database ({tbdb}) is an integrated database providing access to {tb} genomic data and resources, relevant to the discovery and development of {tb} drugs, vaccines and biomarkers. the current release of {tbdb} houses genome sequence data and annotations for 28 different mycobacterium tuberculosis strains and related bacteria. {tbdb} stores pre- and post-publication gene-expression data from m. tuberculosis and its close relatives. {tbdb} currently hosts data for nearly 1500 public tuberculosis microarrays and 260 arrays for streptomyces. in addition, {tbdb} provides access to a suite of comparative genomics and microarray analysis software. by bringing together m. tuberculosis genome annotation and gene-expression data with a suite of analysis tools, {tbdb} (http://www.tbdb.org/) provides a unique discovery platform for {tb} research.the packaging of {dna} into nucleosomes influences the accessibility of underlying regulatory information. nucleosome occupancy and positioning are best characterized in the budding yeast saccharomyces cerevisiae, albeit in asynchronous cell populations or on individual promoters such as {pho5} and {gal1\\\\^{a}}\\x80\\x9310. using {faire} (formaldehyde-assisted isolation of regulatory elements) and whole-genome microarrays, we examined changes in nucleosome occupancy throughout the mitotic cell cycle in synchronized populations of s. cerevisiae. perhaps surprisingly, nucleosome occupancy did not exhibit large, global variation between cell cycle phases. however, nucleosome occupancy at the promoters of cell cycle\\\\^{a}\\x80\\x93regulated genes was reduced specifically at the cell cycle phase in which that gene exhibited peak expression, with the notable exception of s-phase genes. we present data that establish {faire} as a high-throughput method for assaying nucleosome occupancy. for the first time in any system, nucleosome occupancy was mapped genome-wide throughout the cell cycle. fluctuation of nucleosome occupancy at promoters of most cell cycle\\\\^{a}\\x80\\x93regulated genes provides independent evidence that periodic expression of these genes is controlled mainly at the level of transcription. the promoters of {g2/m} genes are distinguished from other cell cycle promoters by an unusually low baseline nucleosome occupancy throughout the cell cycle. this observation, coupled with the maintenance throughout the cell cycle of the stereotypic nucleosome occupancy states between coding and non-coding loci, suggests that the largest component of variation in nucleosome occupancy is \\\\^{a}\\x80\\x9chard wired,\\\\^{a}\\x80\\x9d perhaps at the level of {dna} sequence. every cell contains a complete copy of the genome, which is comprised of {dna}. to fit the genome into the cell nucleus, {dna} strands are wound around spools made of proteins called histones. by packing {dna}, histones also control access to {dna}, which influences when genes can be turned on and off. the authors present two main findings. first, they demonstrate a new, simple method to determine how tightly genes are wrapped around histones. second, they use the new method to determine what happens to {dna} packaging at every gene through one cycle of cell growth and division. they find that as genes get turned on throughout the cell cycle, the packaging of {dna} that controls those genes becomes loose, and as genes get turned off, it becomes tight again. however, some cell cycle\\\\^{a}\\x80\\x93regulated genes do not fit this pattern, suggesting another means of regulation for these genes. interestingly, cell cycle\\\\^{a}\\x80\\x93related differences in packaging at single genes are smaller than differences in packaging that are always apparent between different groups of genes. this suggests that most differences in how {dna} is packaged may be controlled by the sequence of {dna} itself, with smaller contributions from other factors.summary: we have developed a new software system, {regulatory} network generator with {combinatorial} control ({renco}), for automatic generation of differential equations describing pre-transcriptional combinatorics in artificial regulatory networks. {renco} has the following benefits: (a) it explicitly models protein–protein interactions among transcription factors, (b) it captures combinatorial control of transcription factors on target genes and (c) it produces output in systems biology markup language ({sbml}) format, which allows these equations to be directly imported into existing simulators. explicit modeling of the protein interactions allows {renco} to incorporate greater mechanistic detail of the transcription machinery compared to existing models and can provide a better assessment of algorithms for regulatory network {inference.availability}: {renco} is a c++ command line program, available at {http://sourceforge.net/projects/renco/contact}: {terran@cs.unm.edusupplementary} information: supplementary data are available at bioinformatics online.motivation: recent technological advances allow the measurement, in a single {hi-c} experiment, of the frequencies of physical contacts among pairs of genomic loci at a genome-wide scale. the next challenge is to infer, from the resulting {dna}–{dna} contact maps, accurate {3d} models of how chromosomes fold and fit into the nucleus. many existing inference methods rely on multidimensional scaling ({mds}), in which the pairwise distances of the inferred model are optimized to resemble pairwise distances derived directly from the contact counts. these approaches, however, often optimize a heuristic objective function and require strong assumptions about the biophysics of {dna} to transform interaction frequencies to spatial distance, and thereby may lead to incorrect structure reconstruction.determining how the evolving genome-wide map of distal regulatory elements is connected with target genes has remained a significant challenge, despite progress in understanding chromatin architecture and regulation. a new study presents a computational approach for predicting distal element\\\\&\\\\#19;gene interactions.we present a generative model, lux, to quantify {dna} methylation modifications from any combination of bisulfite sequencing approaches, including reduced, oxidative, {tet}-assisted, chemical-modification assisted, and methylase-assisted bisulfite sequencing data. lux models all cytosine modifications (c, {5mc}, {5hmc}, {5fc}, and {5cac}) simultaneously together with experimental parameters, including bisulfite conversion and oxidation efficiencies, as well as various chemical labeling and protection steps. we show that lux improves the quantification and comparison of cytosine modification levels and that lux can process any oxidized methylcytosine sequencing data sets to quantify all cytosine modifications. analysis of targeted data from tet2-knockdown embryonic stem cells and t cells during development demonstrates {dna} modification quantification at unprecedented detail, quantifies active demethylation pathways and reveals {5hmc} localization in putative regulatory regions.many biological networks naturally form a hierarchy with a preponderance of downward information flow. in this study, we define a score to quantify the degree of hierarchy in a network and develop a simulated-annealing algorithm to maximize the hierarchical score globally over a network. we apply our algorithm to determine the hierarchical structure of the phosphorylome in detail and investigate the correlation between its hierarchy and kinase properties. we also compare it to the regulatory network, finding that the phosphorylome is more hierarchical than the regulome.improving the ability to reverse engineer biochemical networks is a major goal of systems biology. lesions in signaling networks lead to alterations in gene expression, which in principle should allow network reconstruction. however, the information about the activity levels of signaling proteins conveyed in overall gene expression is limited by the complexity of gene expression dynamics and of regulatory network topology. two observations provide the basis for overcoming this limitation: a. genes induced without de-novo protein synthesis (early genes) show a linear accumulation of product in the first hour after the change in the cell\\'s state; b. the signaling components in the network largely function in the linear range of their stimulus-response curves. therefore, unlike most genes or most time points, expression profiles of early genes at an early time point provide direct biochemical assays that represent the activity levels of upstream signaling components. such expression data provide the basis for an efficient algorithm (plato\\'s cave algorithm; {placa}) to reverse engineer functional signaling networks. unlike conventional reverse engineering algorithms that use steady state values, {placa} uses stimulated early gene expression measurements associated with systematic perturbations of signaling components, without measuring the signaling components themselves. besides the reverse engineered network, {placa} also identifies the genes detecting the functional interaction, thereby facilitating validation of the predicted functional network. using simulated datasets, the algorithm is shown to be robust to experimental noise. using experimental data obtained from gonadotropes, {placa} reverse engineered the interaction network of six perturbed signaling components. the network recapitulated many known interactions and identified novel functional interactions that were validated by further experiment. {placa} uses the results of experiments that are feasible for any signaling network to predict the functional topology of the network and to identify novel relationships.directed differentiation of human embryonic stem cells ({escs}) into cardiovascular cells provides a model for studying molecular mechanisms of human cardiovascular development. although it is known that chromatin modification patterns in {escs} differ markedly from those in lineage-committed progenitors and differentiated cells, the temporal dynamics of chromatin alterations during differentiation along a defined lineage have not been studied. we show that differentiation of human {escs} into cardiovascular cells is accompanied by programmed temporal alterations in chromatin structure that distinguish key regulators of cardiovascular development from other genes. we used this temporal chromatin signature to identify regulators of cardiac development, including the homeobox gene {meis2}. using the zebrafish model, we demonstrate that {meis2} is critical for proper heart tube formation and subsequent cardiac looping. temporal chromatin signatures should be broadly applicable to other models of stem cell differentiation to identify regulators and provide key insights into major developmental decisions. copyright {\\\\copyright} 2012 elsevier inc. all rights reserved.a nonparametric model is introduced that allows multiple related regression tasks to take inputs from a common data space. traditional transfer learning models can be inappropriate if the dependence among the outputs cannot be fully resolved by known input-specific and task-specific predictors. the proposed model treats such output responses as conditionally independent, given known predictors and appropriate unobserved random effects. the model is nonparametric in the sense that the dimensionality of random effects is not specified a priori but is instead determined from data. an approach to estimating the model is presented uses an {em} algorithm that is efficient on a very large scale collaborative prediction problem. the obtained prediction accuracy is competitive with state-of-the-art results.transcription factor ({tf}) networks determine cell-type identity by establishing and maintaining lineage-specific expression profiles, yet reconstruction of mammalian regulatory network models has been hampered by a lack of comprehensive functional validation of regulatory interactions. here, we report comprehensive {chip}-seq, transgenic and reporter gene experimental data that have allowed us to construct an experimentally validated regulatory network model for haematopoietic stem/progenitor cells ({hspcs}). model simulation coupled with subsequent experimental validation using single cell expression profiling revealed potential mechanisms for cell state stabilisation, and also how a leukaemogenic {tf} fusion protein perturbs key {hspc} regulators. the approach presented here should help to improve our understanding of both normal physiological and disease processes.the task of causal structure discovery from empirical data is a fundamental problem in many areas. experimental data is crucial for accomplishing this task. however, experiments are typically expensive, and must be selected with great care. this paper uses active learning to determine the experiments that are most informative towards uncovering the underlying structure. we formalize the causal learning task as that of learning the structure of a causal bayesian network. we consider an active learner that is allowed to conduct experiments, where it intervenes in the domain by setting the values of certain variables. we provide a theoretical framework for the active learning problem, and an algorithm that actively chooses the experiments to perform based on the model learned so far. experimental results show that active learning can substantially reduce the number of observations required to determine the structure of a domain. 1 introduction determining the causal struc...single-cell technologies have recently gained popularity in cellular differentiation studies regarding their ability to resolve potential heterogeneities in cell populations. analyzing such high-dimensional single-cell data has its own statistical and computational challenges. popular multivariate approaches are based on data normalization, followed by dimension reduction and clustering to identify subgroups. however, in the case of cellular differentiation, we would not expect clear clusters to be present but instead expect the cells to follow continuous branching lineages. here, we propose the use of diffusion maps to deal with the problem of defining differentiation trajectories. we adapt this method to single-cell data by adequate choice of kernel width and inclusion of uncertainties or missing measurement values, which enables the establishment of a pseudotemporal ordering of single cells in a high-dimensional gene expression space. we expect this output to reflect cell differentiation trajectories, where the data originates from intrinsic diffusion-like dynamics. starting from a pluripotent stage, cells move smoothly within the transcriptional landscape towards more differentiated states with some stochasticity along their path. we demonstrate the robustness of our method with respect to extrinsic noise (e.g. measurement noise) and sampling density heterogeneities on simulated toy data as well as two single-cell quantitative polymerase chain reaction datasets (i.e. mouse haematopoietic stem cells and mouse embryonic stem cells) and an {rna}-seq data of human pre-implantation embryos. we show that diffusion maps perform considerably better than principal component analysis and are advantageous over other techniques for non-linear dimension reduction such as t-distributed stochastic neighbour embedding for preserving the global structures and pseudotemporal ordering of cells. the matlab implementation of diffusion maps for single-cell data is available at https://www.helmholtz-muenchen.de/icb/single-cell-diffusion-map. fbuettner.phys@gmail.com, fabian.theis@helmholtz-muenchen.de supplementary data are available at bioinformatics online. {\\\\copyright} the author 2015. published by oxford university press. all rights reserved. for permissions, please e-mail: journals.permissions@oup.com.motivation: experiments in animal models are often conducted to infer how humans will respond to stimuli by assuming that the same biological pathways will be affected in both organisms. the limitations of this assumption were tested in the {improver} species translation challenge, where 52 stimuli were applied to both human and rat cells and perturbed pathways were identified. in the inter-species pathway perturbation prediction sub-challenge, multiple teams proposed methods to use rat transcription data from 26 stimuli to predict human gene set and pathway activity under the same perturbations. submissions were evaluated using three performance metrics on data from the remaining 26 stimuli.{dna} variants ({snps}) that predispose to common traits often localize within noncoding regulatory elements such as enhancers. moreover, loci identified by genome-wide association studies ({gwas}) often contain multiple {snps} in linkage disequilibrium ({ld}), any of which may be causal. thus, determining the effect of these multiple variant {snps} on target transcript levels has been a major challenge. here, we provide evidence that for six common autoimmune disorders (rheumatoid arthritis, crohn\\'s disease, celiac disease, multiple sclerosis, lupus, and ulcerative colitis), the {gwas} association arises from multiple polymorphisms in {ld} that map to clusters of enhancer elements active in the same cell type. this finding suggests a \"multiple enhancer variant\" hypothesis for common traits, where several variants in {ld} impact multiple enhancers and cooperatively affect gene expression. using a novel method to delineate enhancer-gene interactions, we show that multiple enhancer variants within a given locus typically target the same gene. using available data from {hapmap} and b lymphoblasts as a model system, we provide evidence at numerous loci that multiple enhancer variants cooperatively contribute to altered expression of their gene targets. the effects on target transcript levels tend to be modest and can be either gain- or loss-of-function. additionally, the genes associated with multiple enhancer variants encode proteins that are often functionally related and enriched in common pathways. overall, the multiple enhancer variant hypothesis offers a new paradigm by which noncoding variants can confer susceptibility to common traits.{background}:causal networks based on the vector autoregressive ({var}) process are a promising statistical tool for modeling regulatory interactions in a cell. however, learning these networks is challenging due to the low sample size and high dimensionality of genomic {data.results}:we present a novel and highly efficient approach to estimate a {var} network. this proceeds in two steps: (i) improved estimation of {var} regression coefficients using an analytic shrinkage approach, and (ii) subsequent model selection by testing the associated partial correlations. in simulations this approach outperformed for small sample size all other considered approaches in terms of true discovery rate (number of correctly identified edges relative to the significant edges). moreover, the analysis of expression time series data from arabidopsis thaliana resulted in a biologically sensible {network.conclusion}:statistical learning of large-scale {var} causal models can be done efficiently by the proposed procedure, even in the difficult data situations prevalent in genomics and {proteomics.availability}:the method is implemented in r code that is available from the authors on request.motivation: recent advances in {dna} microarray technologies have made it possible to measure the expression levels of thousands of genes simultaneously under different conditions. the data obtained by microarray analyses are called expression profile data. one type of important information underlying the expression profile data is the \\\\^{a}\\x80\\x98genetic network,\\\\^{a}\\x80\\x99 that is, the regulatory network among genes. graphical gaussian modeling ({ggm}) is a widely utilized method to infer or test relationships among a plural of {variables.results}: in this study, we developed a method combining the cluster analysis with {ggm} for the inference of the genetic network from the expression profile data. the expression profile data of 2467 saccharomyces cerevisiae genes measured under 79 different conditions (eisen et al. , proc. natl acad. sci. {usa} , 95, 14683\\\\^{a}\\x80\\x9314868, 1998) were used for this study. at first, the 2467 genes were classified into 34 clusters by a cluster analysis, as a preprocessing for {ggm}. then, the expression levels of the genes in each cluster were averaged for each condition. the averaged expression profile data of 34 clusters were subjected to {ggm}, and a partial correlation coefficient matrix was obtained as a model of the genetic network of s. cerevisiae. the accuracy of the inferred network was examined by the agreement of our results with the cumulative results of experimental {studies.availability}: a set of programs will be electronically sent upon {request.contact}: toh@beri.co.jp; horimoto@ged.saga-med.ac.jpthe human genome encodes the blueprint of life, but the function of the vast majority of its nearly three billion bases is unknown. the encyclopedia of {dna} elements ({encode}) project has systematically mapped regions of transcription, transcription factor association, chromatin structure and histone modification. these data enabled us to assign biochemical functions for 80\\\\% of the genome, in particular outside of the well-studied protein-coding regions. many discovered candidate regulatory elements are physically associated with one another and with expressed genes, providing new insights into the mechanisms of gene regulation. the newly identified elements also show a statistical correspondence to sequence variants linked to human disease, and can thereby guide interpretation of this variation. overall, the project provides new insights into the organization and regulation of our genes and genome, and is an expansive resource of functional annotations for biomedical research.we present model-based analysis of {chip}-seq data, {macs}, which analyzes data generated by short read sequencers such as solexa\\'s genome analyzer. {macs} empirically models the shift size of {chip}-seq tags, and uses it to improve the spatial resolution of predicted binding sites. {macs} also uses a dynamic poisson distribution to effectively capture local biases in the genome, allowing for more robust predictions. {macs} compares favorably to existing {chip}-seq peak-finding algorithms, and is freely available.differential gene expression is the fundamental mechanism underlying animal development and cell differentiation. however, it is a challenge to identify comprehensively and accurately the {dna} sequences that are required to regulate gene expression: namely, cis-regulatory modules ({crms}). three major features, either singly or in combination, are used to predict {crms}: clusters of transcription factor binding site motifs, non-coding {dna} that is under evolutionary constraint and biochemical marks associated with {crms}, such as histone modifications and protein occupancy. the validation rates for predictions indicate that identifying diagnostic biochemical marks is the most reliable method, and understanding is enhanced by the analysis of motifs and conservation patterns within those predicted {crms}.motivation: there is accumulating evidence that the chromatin environment of transcription factor ({tf}) binding sites in promoter regions has a critical influence on their regulatory potential. recent studies have mapped {tf} binding sites and nucleosome positions throughout the yeast genome; however, there is a lack of computation tools to integrate these data types.  results: we have developed the ceres software to facilitate the integrated analysis of {tf} binding sites and nucleosome positions in the model eukaryote s.cerevisiae. ceres enables users to dynamically display the spatial organization of {tf} binding sites and nucleosome positions of individual genes, or the average profiles for large gene sets. ceres provides novel statistical tools to test for the enrichment of {tf} binding sites and chromatin environments for user-selected gene sets. ceres also enables users to search the genome for combinations of {tf} binding sites that are associated with specific chromatin environments. preliminary analysis using the ceres software indicates that functional and conserved {tf} binding sites are often associated with specific chromatin environments.  availability: {http://bioinformatics1.smb.wsu.edu/ceres}  contact: jwyrick@wsu.edu  supplementary information: supplementary data are available at bioinformatics online. 10.1093/bioinformatics/btp657functional partnerships between proteins are at the core of complex cellular phenotypes, and the networks formed by interacting proteins provide researchers with crucial scaffolds for modeling, data reduction and annotation. {string} is a database and web resource dedicated to protein-protein interactions, including both physical and functional interactions. it weights and integrates information from numerous sources, including experimental repositories, computational prediction methods and public text collections, thus acting as a meta-database that maps all interaction evidence onto a common set of genomes and proteins. the most important new developments in {string} 8 over previous releases include a {url}-based programming interface, which can be used to query {string} from other resources, improved interaction prediction via genomic neighborhood in prokaryotes, and the inclusion of protein structures. version 8.0 of {string} covers about 2.5 million proteins from 630 organisms, providing the most comprehensive view on protein-protein interactions currently available. {string} can be reached at http://string-db.org/.proteins, such as many transcription factors, that bind to specific {dna} sequences are essential for the proper regulation of gene expression. identifying the specific sequences that each factor binds can help to elucidate regulatory networks within cells and how genetic variation can cause disruption of normal gene expression, which is often associated with disease. traditional methods for determining the specificity of {dna}-binding proteins are slow and laborious, but several new high-throughput methods can provide comprehensive binding information much more rapidly. combined with in vivo determinations of transcription factor binding locations, this information provides more detailed views of the regulatory circuitry of cells and the effects of variation on gene expression.motivation: when running experiments that involve multiple high density oligonucleotide arrays, it is important to remove sources of variation between arrays of non-biological origin. normalization is a process for reducing this variation. it is common to see non-linear relations between arrays and the standard normalization provided by affymetrix does not perform well in these situations.genome-wide association studies have identified many noncoding variants associated with common diseases and traits. we show that these variants are concentrated in regulatory {dna} marked by deoxyribonuclease i ({dnase} i) hypersensitive sites ({dhss}). eighty-eight percent of such {dhss} are active during fetal development and are enriched in variants associated with gestational exposure-related phenotypes. we identified distant gene targets for hundreds of variant-containing {dhss} that may explain phenotype associations. disease-associated variants systematically perturb transcription factor recognition sequences, frequently alter allelic chromatin states, and form regulatory networks. we also demonstrated tissue-selective enrichment of more weakly disease-associated variants within {dhss} and the de novo identification of pathogenic cell types for crohn\\'s disease, multiple sclerosis, and an electrocardiogram trait, without prior knowledge of physiological mechanisms. our results suggest pervasive involvement of regulatory {dna} variation in common human disease and provide pathogenic insights into diverse disorders.during a decade of proof-of-principle analysis in model organisms, protein networks have been used to further the study of molecular evolution, to gain insight into the robustness of cells to perturbation, and for assignment of new protein functions. following these analyses, and with the recent rise of protein interaction measurements in mammals, protein networks are increasingly serving as tools to unravel the molecular basis of disease. we review promising applications of protein networks to disease in four major areas: identifying new disease genes; the study of their network properties; identifying disease-related subnetworks; and network-based disease classification. applications in infectious disease, personalized medicine, and pharmacology are also forthcoming as the available protein network information improves in quality and coverage.{incorporating new and updated information, this second edition of the bestselling text in bayesian data analysis continues to emphasize practice over theory, describing how to conceptualize, perform, and critique statistical analyses from a bayesian perspective. its world-class authors provide guidance on all aspects of bayesian data analysis and include examples of real statistical analyses, based on their own research, that demonstrate how to solve complicated problems. changes in the new edition include: \\\\&\\\\#183;stronger focus on mcmc\\\\&\\\\#183;revision of the computational advice in part iii\\\\&\\\\#183;new chapters on nonlinear models and decision analysis\\\\&\\\\#183;several additional applied examples from the authors\\' recent research\\\\&\\\\#183;additional chapters on current models for bayesian data analysis such as nonlinear models, generalized linear mixed models, and more\\\\&\\\\#183;reorganization of chapters 6 and 7 on model checking and data collectionbayesian computation is currently at a stage where there are many reasonable ways to compute any given posterior distribution. however, the best approach is not always clear ahead of time. reflecting this, the new edition offers a more pluralistic presentation, giving advice on performing computations from many perspectives while making clear the importance of being aware that there are different ways to implement any given iterative simulation computation. the new approach, additional examples, and updated information make bayesian data analysis an excellent introductory text and a reference that working scientists will use throughout their professional life.}the lasso estimate for linear regression parameters can be interpreted as a bayesian posterior mode estimate when the regression parameters have independent laplace (i.e., double-exponential) priors. gibbs sampling from this posterior is possible using an expanded hierarchy with conjugate normal priors for the regression parameters and independent exponential priors on their variances. a connection with the {inverse-gaussian} distribution provides tractable full conditional distributions. the bayesian lasso provides interval estimates (bayesian credible intervals) that can guide variable selection. moreover, the structure of the hierarchical model provides both bayesian and likelihood methods for selecting the lasso parameter. slight modifications lead to bayesian versions of other lasso-related estimation methods, including bridge regression and a robust variant.eukaryotic gene transcription is accompanied by acetylation and methylation of nucleosomes near promoters, but the locations and roles of histone modifications elsewhere in the genome remain unclear. we determined the chromatin modification states in high resolution along 30 mb of the human genome and found that active promoters are marked by trimethylation of lys4 of histone h3 ({h3k4}), whereas enhancers are marked by monomethylation, but not trimethylation, of {h3k4}. we developed computational algorithms using these distinct chromatin signatures to identify new regulatory elements, predicting over 200 promoters and 400 enhancers within the {30-mb} region. this approach accurately predicted the location and function of independently identified regulatory elements with high sensitivity and specificity and uncovered a novel functional enhancer for the carnitine transporter {slc22a5} ({octn2}). our results give insight into the connections between chromatin modifications and transcriptional regulatory activity and provide a new tool for the functional annotation of the human genome.during cellular reprogramming, only a small fraction of cells become induced pluripotent stem cells ({ipscs}). previous analyses of gene expression during reprogramming were based on populations of cells, impeding single-cell level identification of reprogramming events. we utilized two gene expression technologies to profile 48 genes in single cells at various stages during the reprogramming process. analysis of early stages revealed considerable variation in gene expression between cells in contrast to late stages. expression of esrrb, utf1, lin28, and dppa2 is a better predictor for cells to progress into {ipscs} than expression of the previously suggested reprogramming markers fbxo15, fgf4, and oct4. stochastic gene expression early in reprogramming is followed by a late hierarchical phase with sox2 being the upstream factor in a gene expression hierarchy. finally, downstream factors derived from the late phase, which do not include oct4, sox2, klf4, {c-myc}, and nanog, can activate the pluripotency circuitry. copyright {\\\\copyright} 2012 elsevier inc. all rights reserved.we use the budding yeast, saccharomyces cerevisiae, to investigate one model for the initial emergence of multicellularity: the formation of multicellular aggregates as a result of incomplete cell separation. we combine simulations with experiments to show how the use of secreted public goods favors the formation of multicellular aggregates. yeast cells can cooperate by secreting invertase, an enzyme that digests sucrose into monosaccharides, and many wild isolates are multicellular because cell walls remain attached to each other after the cells divide. we manipulate invertase secretion and cell attachment, and show that multicellular clumps have two advantages over single cells: they grow under conditions where single cells cannot and they compete better against cheaters, cells that do not make invertase. we propose that the prior use of public goods led to selection for the incomplete cell separation that first produced multicellularity. the evolution of multicellularity is one of the major steps in the history of life and has occurred many times independently. despite this, we do not understand how and why single-celled organisms first joined together to form multicellular clumps of cells. here, we show that clumps of cells can cooperate, using secreted enzymes, to collect food from the environment. in nature, the budding yeast saccharomyces cerevisiae grows as multicellular clumps and secretes invertase, an enzyme that breaks down sucrose into smaller sugars (glucose and fructose) that cells can import. we genetically manipulate both clumping and secretion to show that multicellular clumps of cells can grow when sucrose is scarce, whereas single cells cannot. in addition, we find that clumps of cells have an advantage when competing against  ” cheating” cells that import sugars but do not make invertase. since the evolution of secreted enzymes predates the origin of multicellularity, we argue that the social benefits conferred by secreted enzymes were the driving force for the evolution of cell clumps that were the first, primitive form of multicellular life.the processes responsible for the evolution of key innovations, whereby lineages acquire qualitatively new functions that expand their ecological opportunities, remain poorly understood. we examined how a virus, bacteriophage λ, evolved to infect its host, escherichia coli, through a novel pathway. natural selection promoted the fixation of mutations in the virus\\'s host-recognition protein, j, that improved fitness on the original receptor, {lamb}, and set the stage for other mutations that allowed infection through a new receptor, {ompf}. these viral mutations arose after the host evolved reduced expression of {lamb}, whereas certain other host mutations prevented the phage from evolving the new function. this study shows the complex interplay between genomic processes and ecological conditions that favor the emergence of evolutionary innovations.although cellular behaviors are dynamic, the networks that govern these behaviors have been mapped primarily as static snapshots. using an approach called differential epistasis mapping, we have discovered widespread changes in genetic interaction among yeast kinases, phosphatases, and transcription factors as the cell responds to {dna} damage. differential interactions uncover many gene functions that go undetected in static conditions. they are very effective at identifying {dna} repair pathways, highlighting new damage-dependent roles for the slt2 kinase, pph3 phosphatase, and histone variant htz1. the data also reveal that protein complexes are generally stable in response to perturbation, but the functional relations between these complexes are substantially reorganized. differential networks chart a new type of genetic landscape that is invaluable for mapping cellular responses to stimuli.despite extensive research, our understanding of the rules according to which cis-regulatory sequences are converted into gene expression is limited. we devised a method for obtaining parallel, highly accurate gene expression measurements from thousands of designed promoters and applied it to measure the effect of systematic changes in the location, number, orientation, affinity and organization of transcription-factor binding sites and nucleosome-disfavoring sequences. our analyses reveal a clear relationship between expression and binding-site multiplicity, as well as dependencies of expression on the distance between transcription-factor binding sites and gene starts which are transcription-factor specific, including a striking ∼10-bp periodic relationship between gene expression and binding-site location. we show how this approach can measure transcription-factor sequence specificities and the sensitivity of transcription-factor sites to the surrounding sequence context, and compare the activity of 75 yeast transcription factors. our method can be used to study both cis and trans effects of genotype on transcriptional, post-transcriptional and translational control.even simple organisms have the ability to respond to internal and external stimuli. this response is carried out by a dynamic network of {protein–dna} interactions that allows the specific regulation of genes needed for the response. we have developed a novel computational method that uses an input–output hidden markov model to model these regulatory networks while taking into account their dynamic nature. our method works by identifying bifurcation points, places in the time series where the expression of a subset of genes diverges from the rest of the genes. these points are annotated with the transcription factors regulating these transitions resulting in a unified temporal map. applying our method to study yeast response to stress, we derive dynamic models that are able to recover many of the known aspects of these responses. predictions made by our method have been experimentally validated leading to new roles for ino4 and gcn4 in controlling yeast response to stress. the temporal cascade of factors reveals common pathways and highlights differences between master and secondary factors in the utilization of network motifs and in condition-specific regulation.positive natural selection is the force that drives the increase in prevalence of advantageous traits, and it has played a central role in our development as a species. until recently, the study of natural selection in humans has largely been restricted to comparing individual candidate genes to theoretical expectations. the advent of genome-wide sequence and polymorphism data brings fundamental new tools to the study of natural selection. it is now possible to identify new candidates for selection and to reevaluate previous claims by comparison with empirical distributions of {dna} sequence variation across the human genome and among populations. the flood of data and analytical methods, however, raises many new challenges. here, we review approaches to detect positive natural selection, describe results from recent analyses of genome-wide data, and discuss the prospects and challenges ahead as we expand our understanding of the role of natural selection in shaping the human genome.high-dimensional data sets generated by high-throughput technologies, such as {dna} microarray, are often the outputs of complex networked systems driven by hidden regulatory signals. traditional statistical methods for computing low-dimensional or hidden representations of these data sets, such as principal component analysis and independent component analysis, ignore the underlying network structures and provide decompositions based purely on a priori statistical constraints on the computed component signals. the resulting decomposition thus provides a phenomenological model for the observed data and does not necessarily contain physically or biologically meaningful signals. here, we develop a method, called network component analysis, for uncovering hidden regulatory signals from outputs of networked systems, when only a partial knowledge of the underlying network topology is available. the a priori network structure information is first tested for compliance with a set of identifiability criteria. for networks that satisfy the criteria, the signals from the regulatory nodes and their strengths of influence on each output node can be faithfully reconstructed. this method is first validated experimentally by using the absorbance spectra of a network of various hemoglobin species. the method is then applied to microarray data generated from yeast saccharamyces cerevisiae and the activities of various transcription factors during cell cycle are reconstructed by using recently discovered connectivity information for the underlying transcriptional regulatory networks.evolution of gene regulation is an important contributor to the variety of life. here, we analyse the evolution of a combinatorial transcriptional circuit composed of sequence-specific {dna}-binding proteins that are conserved among all eukaryotes. this circuit regulates mating in the ascomycete yeast lineage. we first identify a group of mating genes that was transcriptionally regulated by an activator in a fungal ancestor, but is now transcriptionally regulated by a repressor in modern bakers\\' yeast. despite this change in regulatory mechanism, the logical output of the overall circuit remains the same. by examining the regulation of mating in modern yeasts that are related to different extents, we deduce specific, sequential changes in both cis- and trans-regulatory elements that constitute the transition from positive to negative regulation. these changes indicate specific mechanisms by which fitness barriers were traversed during the transition.{background}:there has been a lot of interest in recent years focusing on the modeling and simulation of gene regulatory networks ({grns}). however, the evolutionary mechanisms that give rise to {grns} in the first place are still largely unknown. in an earlier work, we developed a framework to analyze the effect of objective functions, input types and starting populations on the evolution of {grns} with a specific emphasis on the robustness of evolved {grns}.{results}:in this work, we present a parallel software package, {genesis} for the modeling and simulation of the evolution of gene regulatory networks ({grns}). the software models the process of gene regulation through a combination of finite-state and stochastic models. the evolution of {grns} is then simulated by means of a genetic algorithm with the network connections represented as binary strings. the software allows users to simulate the evolution under varying selective pressures and starting conditions. we believe that the software provides a way for researchers to understand the evolutionary behavior of populations of {grns}.{conclusion}:we believe that {genesis} will serve as a useful tool for scientists interested in understanding the evolution of gene regulatory networks under a range of different conditions and selective pressures. such modeling efforts can lead to a greater understanding of the network characteristics of {grns}.the differentiation of both gene expression and protein function is thought to be important as a mechanism of the functionalization of duplicate genes. however, it has not been addressed whether expression or protein divergence of duplicate genes is greater in those genes that have undergone functionalization compared with those that have not. we examined a total of 492 paralogous gene pairs associated with morphological diversification in a plant model organism (arabidopsis thaliana). classifying these paralogous gene pairs into high, low, and no morphological diversification groups, based on knock-out data, we found that the divergence rate of both gene expression and protein sequences were significantly higher in either high or low morphological diversification groups compared with those in the no morphological diversification group. these results strongly suggest that the divergence of both expression and protein sequence are important sources for morphological diversification of duplicate genes. although both mechanisms are not mutually exclusive, our analysis suggested that changes of expression pattern play the minor role (33\\\\%–41\\\\%) and that changes of protein sequence play the major role (59\\\\%–67\\\\%) in morphological diversification. finally, we examined to what extent duplicate genes are associated with expression or protein divergence exerting morphological diversification at the whole-genome level. interestingly, duplicate genes randomly chosen from a. thaliana had not experienced expression or protein divergence that resulted in morphological diversification. these results indicate that most duplicate genes have experienced minor functionalization. the relationship between morphological and molecular evolution is a central issue to the understanding of eukaryote evolution. in particular, there is much interest in how duplicate genes have contributed to morphological diversification during evolution. as a mechanism of functionalization of duplicate genes, differentiation of both gene expression and protein function are believed to be important. although it has been reported that both expression and protein divergence tend to increase as a duplication ages, it is unclear whether expression or protein divergence in duplicate genes is greater in those genes that have undergone functionalization compared with those that have not. here, we studied 492 duplicate gene pairs associated with various degrees of morphological diversification in arabidopsis thaliana. using these data, we found that the divergence of both expression and protein sequence were important sources for morphological diversification of duplicate genes. although both mechanisms are not mutually exclusive, our analysis suggested that expression divergence is the minor contributor and protein divergence is the major contributor to morphological diversification. however, the expression or protein sequence of randomly chosen duplicate genes did not show significant divergence that resulted in morphological diversification. these results indicate that most duplicate genes experienced minor functionalization in the genome.combinatorial interaction of transcription factors ({tfs}) is important for gene regulation. although various genomic datasets are relevant to this issue, each dataset provides relatively weak evidence on its own. developing methods that can integrate different sequence, expression and localization data have become important.{dna} methylation is frequently described as a \\'silencing\\' epigenetic mark, and indeed this function of 5-methylcytosine was originally proposed in the 1970s. now, thanks to improved genome-scale mapping of methylation, we can evaluate {dna} methylation in different genomic contexts: transcriptional start sites with or without {cpg} islands, in gene bodies, at regulatory elements and at repeat sequences. the emerging picture is that the function of {dna} methylation seems to vary with context, and the relationship between {dna} methylation and transcription is more nuanced than we realized at first. improving our understanding of the functions of {dna} methylation is necessary for interpreting changes in this mark that are observed in diseases such as cancer.recent studies have characterized significant differences in the cis-regulatory sequences of related organisms, but the impact of these differences on gene expression remains largely unexplored. here, we show that most previously identified differences in transcription factor ({tf})-binding sequences of yeasts and mammals have no detectable effect on gene expression, suggesting that compensatory mechanisms allow promoters to rapidly evolve while maintaining a stabilized expression pattern. to examine the impact of changes in cis-regulatory elements in a more controlled setting, we compared the genes induced during mating of three yeast species. this response is governed by a single {tf} ({ste12}), and variations in its predicted binding sites can indeed account for about half of the observed expression differences. the remaining unexplained differences are correlated with the increased divergence of the sequences that flank the binding sites and an apparent modulation of chromatin structure. our analysis emphasizes the flexibility of promoter structure, and highlights the interplay between specific binding sites and general chromatin structure in the control of gene expression.there are clearly strong evolutionary constraints on tissue-specific gene expression. a major challenge will be to understand the precise mechanisms by which many gene expression patterns remain similar despite extensive cis-regulatory restructuring.motivation: many biological systems operate in a similar manner across a large number of species or conditions. cross-species analysis of sequence and interaction data is often applied to determine the function of new genes. in contrast to these static measurements, microarrays measure the dynamic, condition-specific response of complex biological systems. the recent exponential growth in microarray expression datasets allows researchers to combine expression experiments from multiple species to identify genes that are not only conserved in sequence but also operated in a similar way in the different species studied.microarrays can measure the expression of thousands of genes to identify changes in expression between different biological states. methods are needed to determine the significance of these changes while accounting for the enormous number of genes. we describe a method, significance analysis of microarrays ({sam}), that assigns a score to each gene on the basis of change in gene expression relative to the standard deviation of repeated measurements. for genes with scores greater than an adjustable threshold, {sam} uses permutations of the repeated measurements to estimate the percentage of genes identified by chance, the false discovery rate ({fdr}). when the transcriptional response of human cells to ionizing radiation was measured by microarrays, {sam} identified 34 genes that changed at least 1.5-fold with an estimated {fdr} of 12\\\\%, compared with {fdrs} of 60 and 84\\\\% by using conventional methods of analysis. of the 34 genes, 19 were involved in cell cycle regulation and 3 in apoptosis. surprisingly, four nucleotide excision repair genes were induced, suggesting that this repair pathway for {uv}-damaged {dna} might play a previously unrecognized role in repairing {dna} damaged by ionizing radiation.transcription factors are key cellular components that control gene expression: their activities determine how cells function and respond to the environment. currently, there is great interest in research into human transcriptional regulation. however, surprisingly little is known about these regulators themselves. for example, how many transcription factors does the human genome contain? how are they expressed in different tissues? are they evolutionarily conserved? here, we present an analysis of 1,391 manually curated sequence-specific {dna}-binding transcription factors, their functions, genomic organization and evolutionary conservation. much remains to be explored, but this study provides a solid foundation for future investigations to elucidate regulatory mechanisms underlying diverse mammalian biological processes.eukaryotic genomes are packaged into nucleosome particles that occlude the {dna} from interacting with most {dna} binding proteins. nucleosomes have higher affinity for particular {dna} sequences, reflecting the ability of the sequence to bend sharply, as required by the nucleosome structure. however, it is not known whether these sequence preferences have a significant influence on nucleosome position in vivo, and thus regulate the access of other proteins to {dna}. here we isolated nucleosome-bound sequences at high resolution from yeast and used these sequences in a new computational approach to construct and validate experimentally a {nucleosome-dna} interaction model, and to predict the genome-wide organization of nucleosomes. our results demonstrate that genomes encode an intrinsic nucleosome organization and that this intrinsic organization can explain approximately 50\\\\% of the in vivo nucleosome positions. this nucleosome positioning code may facilitate specific chromosome functions including transcription factor binding, transcription initiation, and even remodelling of the nucleosomes themselves.{rna} sequencing ({rna}-seq) has been rapidly adopted for the profiling of transcriptomes in many areas of biology, including studies into gene regulation, development and disease. of particular interest is the discovery of differentially expressed genes across different conditions (e.g., tissues, perturbations) while optionally adjusting for other systematic factors that affect the data-collection process. there are a number of subtle yet crucial aspects of these analyses, such as read counting, appropriate treatment of biological variability, quality control checks and appropriate setup of statistical modeling. several variations have been presented in the literature, and there is a need for guidance on current best practices. this protocol presents a state-of-the-art computational and statistical {rna}-seq differential expression analysis workflow largely based on the free open-source r language and bioconductor software and, in particular, on two widely used tools, {deseq} and {edger}. hands-on time for typical small experiments (e.g., 4-10 samples) can be <1 h, with computation time <1 d using a standard desktop {pc}.we introduce the {gibbsmodule} algorithm for de novo detection of cis-regulatory motifs and modules in eukaryote genomes. {gibbsmodule} models the coexpressed genes within one species as sharing a core cis-regulatory motif and each homologous gene group as sharing a homologous cis-regulatory module ({crm}), characterized by a similar composition of motifs. without using a predetermined alignment result, {gibbsmodule} iteratively updates the core motif shared by coexpressed genes and traces the homologous {crms} that contain the core motif. {gibbsmodule} achieved substantial improvements in both precision and recall as compared with peer algorithms on a number of synthetic and real data sets. applying {gibbsmodule} to analyze the binding regions of the kr\\\\\"{u}ppel-like factor ({klf}) transcription factor in embryonic stem cells ({escs}), we discovered a motif that differs from a previously published {klf} motif identified by a {selex} experiment, but the new motif is consistent with mutagenesis analysis. the {sox2} motif was found to be a collaborating motif to the {klf} motif in {escs}. we used quantitative chromatin immunoprecipitation ({chip}) analysis to test whether {gibbsmodule} could distinguish functional and nonfunctional binding sites. all seven tested binding sites in {gibbsmodule}-predicted {crms} had higher {chip} signals as compared with the other seven tested binding sites located outside of predicted {crms}. {gibbsmodule} is available at ({http://biocomp.bioen.uiuc.edu/gibbsmodule}).motivation: many programs for aligning short sequencing reads to a reference genome have been developed in the last 2 years. most of them are very efficient for short reads but inefficient or not applicable for reads >200 bp because the algorithms are heavily and specifically tuned for short queries with low sequencing error rate. however, some sequencing platforms already produce longer reads and others are expected to become available soon. for longer reads, hashing-based software such as {blat} and {ssaha2} remain the only choices. nonetheless, these methods are substantially slower than short-read aligners in terms of aligned bases per unit time.chromatin profiling has emerged as a powerful means of genome annotation and detection of regulatory activity. the approach is especially well suited to the characterization of non-coding portions of the genome, which critically contribute to cellular phenotypes yet remain largely uncharted. here we map nine chromatin marks across nine cell types to systematically characterize regulatory elements, their cell-type specificities and their functional interactions. focusing on cell-type-specific patterns of promoters and enhancers, we define multicell activity profiles for chromatin state, gene expression, regulatory motif enrichment and regulator expression. we use correlations between these profiles to link enhancers to putative target genes, and predict the cell-type-specific activators and repressors that modulate them. the resulting annotations and regulatory predictions have implications for the interpretation of genome-wide association studies. top-scoring disease single nucleotide polymorphisms are frequently positioned within enhancer elements specifically active in relevant cell types, and in some cases affect a motif instance for a predicted regulator, thus suggesting a mechanism for the association. our study presents a general framework for deciphering cis-regulatory connections and their roles in disease. {\\\\copyright}2011 macmillan publishers limited. all rights reserveda causal network is used in a number of areas as a depiction of patterns of `influence\\' among sets of variables. in expert systems it is common to perform `inference\\' by means of local computations on such large but sparse networks. in general, non-probabilistic methods are used to handle uncertainty when propagating the effects of evidence, and it has appeared that exact probabilistic methods are not computationally feasible. motivated by an application in electromyography, we counter this claim by exploiting a range of local representations for the joint probability distribution, combined with topological changes to the original network termed `marrying\\' and `filling-in\\'. the resulting structure allows efficient algorithms for transfer between representations, providing rapid absorption and propagation of evidence. the scheme is first illustrated on a small, fictitious but challenging example, and the underlying theory and computational aspects are then discussed.interpreting variants, especially noncoding ones, in the increasing number of personal genomes is challenging. we used patterns of polymorphisms in functionally annotated regions in 1092 humans to identify deleterious variants; then we experimentally validated candidates. we analyzed both coding and noncoding regions, with the former corroborating the latter. we found regions particularly sensitive to mutations (\"ultrasensitive\") and variants that are disruptive because of mechanistic effects on transcription-factor binding (that is, \"motif-breakers\"). we also found variants in regions with higher network centrality tend to be deleterious. insertions and deletions followed a similar pattern to single-nucleotide variants, with some notable exceptions (e.g., certain deletions and enhancers). on the basis of these patterns, we developed a computational tool ({funseq}), whose application to \\\\~{}90 cancer genomes reveals nearly a hundred candidate noncoding drivers.{dna} copy number aberrations ({cnas}) are a hallmark of cancer genomes. however, little is known about how such changes affect global gene expression. we develop a modeling framework, {epoc} (endogenous perturbation analysis of cancer), to (1) detect disease-driving {cnas} and their effect on target {mrna} expression, and to (2) stratify cancer patients into long- and short-term survivors. our method constructs causal network models of gene expression by combining genome-wide {dna}- and {rna}-level data. prognostic scores are obtained from a singular value decomposition of the networks. by applying {epoc} to glioblastoma data from the cancer genome atlas consortium, we demonstrate that the resulting network models contain known disease-relevant hub genes, reveal interesting candidate hubs, and uncover predictors of patient survival. targeted validations in four glioblastoma cell lines support selected predictions, and implicate the p53-interacting protein necdin in suppressing glioblastoma cell growth. we conclude that large-scale network modeling of the effects of {cnas} on gene expression may provide insights into the biology of human cancer. free software in {matlab} and r is provided.the orchestrated binding of transcriptional activators and repressors to specific {dna} sequences in the context of chromatin defines the regulatory program of eukaryotic genomes. we developed a digital approach to assay regulatory protein occupancy on genomic {dna} in vivo by dense mapping of individual {dnase} i cleavages from intact nuclei using massively parallel {dna} sequencing. analysis of >23 million cleavages across the saccharomyces cerevisiae genome revealed thousands of protected regulatory protein footprints, enabling de novo derivation of factor binding motifs and the identification of hundreds of new binding sites for major regulators. we observed striking correspondence between single-nucleotide resolution {dnase} i cleavage patterns and {protein-dna} interactions determined by crystallography. the data also yielded a detailed view of larger chromatin features including positioned nucleosomes flanking factor binding regions. digital genomic footprinting should be a powerful approach to delineate the cis-regulatory framework of any organism with an available genome sequence.machine learning was applied for the automated derivation of causal influences in cellular signaling networks. this derivation relied on the simultaneous measurement of multiple phosphorylated protein and phospholipid components in thousands of individual primary human immune system cells. perturbing these cells with molecular interventions drove the ordering of connections between pathway components, wherein bayesian network computational methods automatically elucidated most of the traditionally reported signaling relationships and predicted novel interpathway network causalities, which we verified experimentally. reconstruction of network models from physiologically relevant primary single cells might be applied to understanding native-state tissue signaling biology, complex drug actions, and dysfunctional signaling in diseased cells.motivation: cellular processes cause changes over time. observing and measuring those changes over time allows insights into the how and why of regulation. the experimental platform for doing the appropriate large-scale experiments to obtain time-courses of expression levels is provided by microarray technology. however, the proper way of analyzing the resulting time course data is still very much an issue under investigation. the inherent time dependencies in the data suggest that clustering techniques which reflect those dependencies yield improved performance.  results: we propose to use hidden markov models ({hmms}) to account for the horizontal dependencies along the time axis in time course data and to cope with the prevalent errors and missing values. the {hmms} are used within a model-based clustering framework.  we are given a number of clusters, each represented by one hidden markov model from a finite collection encompassing typical qualitative behavior. then, our method finds in an iterative procedure cluster models and an assignment of data points to these models that maximizes the joint likelihood of clustering and models.  partially supervised learning--adding groups of labeled data to the initial collection of clusters--is supported. a graphical user interface allows quering an expression profile dataset for time course similar to a prototype graphically defined as a sequence of levels and durations. we also propose a heuristic approach to automate determination of the number of clusters.  we evaluate the method on published yeast cell cycle and fibroblasts serum response datasets, and compare them, with favorable results, to the autoregressive curves method.  availability: the software is freely available at http://algorithmics.molgen.mpg.de/ghmm  contact: schliep@molgen.mpg.de  keywords: gene expression, time course, model-based clustering, hidden markov models 10.1093/bioinformatics/btg1036emergence of form and function during embryogenesis arises in large part through cell-type- and cell-state-specific variation in gene expression patterns, mediated by specialized cis-regulatory elements called enhancers. recent large-scale epigenomic mapping revealed unexpected complexity and dynamics of enhancer utilization patterns, with 400,000 putative human enhancers annotated by the {encode} project alone. these large-scale efforts were largely enabled through the understanding that enhancers share certain stereotypical chromatin features. however, an important question still lingers: what is the functional significance of enhancer chromatin modification? here we give an overview of enhancer-associated modifications of histones and {dna} and discuss enzymatic activities involved in their dynamic deposition and removal. we\\xa0describe potential downstream effectors of these marks and propose models for exploring functions of chromatin modification in regulating enhancer activity during development.physical and genetic mapping data have become as important to network biology as they once were to the human genome project. integrating physical and genetic networks currently faces several challenges: increasing the coverage of each type of network; establishing methods to assemble individual interaction measurements into contiguous pathway models; and annotating these pathways with detailed functional information. a particular challenge involves reconciling the wide variety of interaction types that are currently available. for this purpose, recent studies have sought to classify genetic and physical interactions along several complementary dimensions, such as ordered versus unordered, alleviating versus aggravating, and first versus second degree.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismswe describe the use of a higher-order singular value decomposition ({hosvd}) in transforming a data tensor of genes ×  ” x-settings,” that is, different settings of the experimental variable x ×  ” y-settings,” which tabulates {dna} microarray data from different studies, to a  ” core tensor” of  ” eigenarrays” ×  ” x-eigengenes” ×  ” y-eigengenes.” reformulating this multilinear {hosvd} such that it decomposes the data tensor into a linear superposition of all outer products of an eigenarray, an x- and a y-eigengene, that is, rank-1  ” subtensors,” we define the significance of each subtensor in terms of the fraction of the overall information in the data tensor that it captures. we illustrate this {hosvd} with an integration of genome-scale {mrna} expression data from three yeast cell cycle time courses, two of which are under exposure to either hydrogen peroxide or menadione. we find that significant subtensors represent independent biological programs or experimental phenomena. the picture that emerges suggests that the conserved genes {yku70}, {mre11}, {aif1}, and {zwf1}, and the processes of retrotransposition, apoptosis, and the oxidative pentose phosphate pathway that these genes are involved in, may play significant, yet previously unrecognized, roles in the differential effects of hydrogen peroxide and menadione on cell cycle progression. a genome-scale correlation between {dna} replication initiation and {rna} transcription, which is equivalent to a recently discovered correlation and might be due to a previously unknown mechanism of regulation, is independently uncovered.changes in gene expression and regulation--due in particular to the evolution of cis-regulatory {dna} sequences--may underlie many evolutionary changes in phenotypes, yet little is known about the distribution of such variation in populations. we present in this study the first survey of experimentally validated functional cis-regulatory polymorphism. these data are derived from more than 140 polymorphisms involved in the regulation of 107 genes in homo sapiens, the eukaryote species with the most available data. we find that functional cis-regulatory variation is widespread in the human genome and that the consequent variation in gene expression is twofold or greater for 63\\\\% of the genes surveyed. transcription {factor-dna} interactions are highly polymorphic, and regulatory interactions have been gained and lost within human populations. on average, humans are heterozygous at more functional cis-regulatory sites (>16,000) than at amino acid positions (<13,000), in part because of an overrepresentation among the former in multiallelic tandem repeat variation, especially ({ac})(n) dinucleotide microsatellites. the role of microsatellites in gene expression variation may provide a larger store of heritable phenotypic variation, and a more rapid mutational input of such variation, than has been realized. finally, we outline the distinctive consequences of cis-regulatory variation for the genotype-phenotype relationship, including ubiquitous epistasis and genotype-by-environment interactions, as well as underappreciated modes of pleiotropy and overdominance. ordinary small-scale mutations contribute to pervasive variation in transcription rates and consequently to patterns of human phenotypic variation.reactome (http://www.reactome.org) is an expert-authored, peer-reviewed knowledgebase of human reactions and pathways that functions as a data mining resource and electronic textbook. its current release includes 2975 human proteins, 2907 reactions and 4455 literature citations. a new entity-level pathway viewer and improved search and data mining tools facilitate searching and visualizing pathway data and the analysis of user-supplied high-throughput data sets. reactome has increased its utility to the model organism communities with improved orthology prediction methods allowing pathway inference for 22 species and through collaborations to create manually curated reactome pathway datasets for species including arabidopsis, oryza sativa (rice), drosophila and gallus gallus (chicken). reactome\\'s data content and software can all be freely used and redistributed under open source terms.motivation: identifying alterations in gene expression associated with different clinical states is important for the study of human biology. however, clinical samples used in gene expression studies are often derived from heterogeneous mixtures with variable cell-type composition, complicating statistical analysis. considerable effort has been devoted to modeling sample heterogeneity, and presently, there are many methods that can estimate cell proportions or pure cell-type expression from mixture data. however, there is no method that comprehensively addresses mixture analysis in the context of differential expression without relying on additional proportion information, which can be inaccurate and is frequently unavailable.transcription of long noncoding {rnas} ({lncrnas}) within gene regulatory elements can modulate gene activity in response to external stimuli, but the scope and functions of such activity are not known. here we use an ultrahigh-density array that tiles the promoters of 56 cell-cycle genes to interrogate 108 samples representing diverse perturbations. we identify 216 transcribed regions that encode putative {lncrnas}, many with {rt}-{pcr}-validated periodic expression during the cell cycle, show altered expression in human cancers and are regulated in expression by specific oncogenic stimuli, stem cell differentiation or {dna} damage. {dna} damage induces five {lncrnas} from the {cdkn1a} promoter, and one such {lncrna}, named {panda}, is induced in a p53-dependent manner. {panda} interacts with the transcription factor {nf}-{ya} to limit expression of pro-apoptotic genes; {panda} depletion markedly sensitized human fibroblasts to apoptosis by doxorubicin. these findings suggest potentially widespread roles for promoter {lncrnas} in cell-growth control.we describe an assay for transposase-accessible chromatin using sequencing ({atac}-seq), based on direct in vitro transposition of sequencing adaptors into native chromatin, as a rapid and sensitive method for integrative epigenomic analysis. {atac}-seq captures open chromatin sites using a simple two-step protocol with 500-50,000 cells and reveals the interplay between genomic locations of open chromatin, {dna}-binding proteins, individual nucleosomes and chromatin compaction at nucleotide resolution. we discovered classes of {dna}-binding factors that strictly avoided, could tolerate or tended to overlap with nucleosomes. using {atac}-seq maps of human {cd4}(+) t cells from a proband obtained on consecutive days, we demonstrated the feasibility of analyzing an individual\\'s epigenome on a timescale compatible with clinical decision-making.the cancer genome atlas ({tcga}) has used the latest sequencing and analysis methods to identify somatic variants across thousands of tumours. here we present data and analytical results for point mutations and small insertions/deletions from 3,281 tumours across 12 tumour types as part of the {tcga} {pan-cancer} effort. we illustrate the distributions of mutation frequencies, types and contexts across tumour types, and establish their links to tissues of origin, environmental/carcinogen influences, and {dna} repair defects. using the integrated data sets, we identified 127 significantly mutated genes from well-known (for example, mitogen-activated protein kinase, {phosphatidylinositol-3-oh} kinase, wnt/β-catenin and receptor tyrosine kinase signalling pathways, and cell cycle control) and emerging (for example, histone, histone modification, splicing, metabolism and proteolysis) cellular processes in cancer. the average number of mutations in these significantly mutated genes varies across tumour types; most tumours have two to six, indicating that the number of driver mutations required during oncogenesis is relatively small. mutations in transcriptional factors/regulators show tissue specificity, whereas histone modifiers are often mutated across several cancer types. clinical association analysis identifies genes having a significant effect on survival, and investigations of mutations with respect to clonal/subclonal architecture delineate their temporal orders during tumorigenesis. taken together, these results lay the groundwork for developing new diagnostics and individualizing cancer treatment.a central goal of systems biology is to elucidate the structural and functional architecture of the cell. to this end, large and complex networks of molecular interactions are being rapidly generated for humans and model organisms. a recent focus of bioinformatics research has been to integrate these networks with each other and with diverse molecular profiles to identify sets of molecules and interactions that participate in a common biological function - that is, \\'modules\\'. here, we classify such integrative approaches into four broad categories, describe their bioinformatic principles and review their applications.the last few years have seen extensive efforts to catalogue human genetic variation and correlate it with phenotypic differences. most common {snps} have now been assessed in genome-wide studies for statistical associations with many complex traits, including many important common diseases. although these studies have provided new biological insights, only a limited amount of the heritable component of any complex trait has been identified and it remains a challenge to elucidate the functional link between associated variants and phenotypic traits. technological advances, such as the ability to detect rare and structural variants, and a clear understanding of the challenges in linking different types of variation with phenotype, will be essential for future progress.this paper presents a number of new algorithras for discovering the markov blanket of a target variable t from training data. the markov blanket can be used for variable selection for classification, for causal discovery, and for bayesian network learning. we introduce a low-order polynomial algorithm and several variants that soundly induce the markov blanket under certain broad conditions in datasets with thousands of variables and compare them to other state-of-the-art local and global methods with excellent results.an important goal of whole-cell computational modeling is to integrate detailed biochemical information with biological intuition to produce testable predictions. based on the premise that prokaryotes such as escherichia coli have maximized their growth performance along evolution, flux balance analysis ({fba}) predicts metabolic flux distributions at steady state by using linear programming. corroborating earlier results, we show that recent intracellular flux data for wild-type e. coli {jm101} display excellent agreement with {fba} predictions. although the assumption of optimality for a wild-type bacterium is justifiable, the same argument may not be valid for genetically engineered knockouts or other bacterial strains that were not exposed to long-term evolutionary pressure. we address this point by introducing the method of minimization of metabolic adjustment ({moma}), whereby we test the hypothesis that knockout metabolic fluxes undergo a minimal redistribution with respect to the flux configuration of the wild type. {moma} employs quadratic programming to identify a point in flux space, which is closest to the wild-type point, compatibly with the gene deletion constraint. comparing {moma} and {fba} predictions to experimental flux data for e. coli pyruvate kinase mutant {pb25}, we find that {moma} displays a significantly higher correlation than {fba}. our method is further supported by experimental data for e. coli knockout growth rates. it can therefore be used for predicting the behavior of perturbed metabolic networks, whose growth performance is in general suboptimal. {moma} and its possible future extensions may be useful in understanding the evolutionary optimization of metabolism.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsmany genome-wide datasets are routinely generated to study different aspects of biological systems, but integrating them to obtain a coherent view of the underlying biology remains a challenge. we propose simultaneous clustering of multiple networks as a framework to integrate large-scale datasets on the interactions among and activities of cellular components. specifically, we develop an algorithm {jointcluster} that finds sets of genes that cluster well in multiple networks of interest, such as coexpression networks summarizing correlations among the expression profiles of genes and physical networks describing protein-protein and {protein-dna} interactions among genes or gene-products. our algorithm provides an efficient solution to a well-defined problem of jointly clustering networks, using techniques that permit certain theoretical guarantees on the quality of the detected clustering relative to the optimal clustering. these guarantees coupled with an effective scaling heuristic and the flexibility to handle multiple heterogeneous networks make our method {jointcluster} an advance over earlier approaches. simulation results showed {jointcluster} to be more robust than alternate methods in recovering clusters implanted in networks with high false positive rates. in systematic evaluation of {jointcluster} and some earlier approaches for combined analysis of the yeast physical network and two gene expression datasets under glucose and ethanol growth conditions, {jointcluster} discovers clusters that are more consistently enriched for various reference classes capturing different aspects of yeast biology or yield better coverage of the analysed genes. these robust clusters, which are supported across multiple genomic datasets and diverse reference classes, agree with known biology of yeast under these growth conditions, elucidate the genetic control of coordinated transcription, and enable functional predictions for a number of uncharacterized genes. the generation of high-dimensional datasets in the biological sciences has become routine (protein interaction, gene expression, and {dna}/{rna} sequence data, to name a few), stretching our ability to derive novel biological insights from them, with even less effort focused on integrating these disparate datasets available in the public domain. hence a most pressing problem in the life sciences today is the development of algorithms to combine large-scale data on different biological dimensions to maximize our understanding of living systems. we present an algorithm for simultaneously clustering multiple biological networks to identify coherent sets of genes (clusters) underlying cellular processes. the algorithm allows theoretical guarantees on the quality of the detected clusters relative to the optimal clusters that are computationally infeasible to find, and could be applied to coexpression, protein interaction, {protein-dna} networks, and other network types. when combining multiple physical and gene expression based networks in yeast, the clusters we identify are consistently enriched for reference classes capturing diverse aspects of biology, yield good coverage of the analysed genes, and highlight novel members in well-studied cellular processes.phylogenetic hidden markov models, or {phylo-hmms}, are probabilistic models that consider not only the way substitutions occur through evolutionary history at each site of a genome, but also the way this process changes from one site to the next. by treating molecular evolution as a combination of two markov processes\\\\^{a}\\x80\\x94one that operates in the dimension of space (along a genome) and one that operates in the dimension of time (along the branches of a phylogenetic tree)\\\\^{a}\\x80\\x94these models allow aspects of both sequence structure and sequence evolution to be captured. moreover, as we will discuss, they permit key computations to be performed exactly and efficiently. {phylo-hmms} allow evolutionary information to be brought to bear on a wide variety of problems of sequence \\\\^{a}\\x80\\x9csegmentation, \\\\^{a}\\x80\\x9d such as gene prediction and the identification of conserved elements. {phylo-hmms} were first proposed as a way of improving phylogenetic models that allow for variation among sites in the rate of substitution [8, 52]. soon afterward, they were adapted for the problem of secondary structureto what extent can variation in phenotypic traits such as disease risk be accurately predicted in individuals? in this review, i highlight recent studies in model organisms that are relevant both to the challenge of accurately predicting phenotypic variation from individual genome sequences (\\'whole-genome reverse genetics\\') and for understanding why, in many cases, this may be impossible. these studies argue that only by combining genetic knowledge with in vivo measurements of biological states will it be possible to make accurate genetic predictions for individual humans.candida albicans is a normal resident of the gastrointestinal tract and also the most prevalent fungal pathogen of humans. it last shared a common ancestor with the model yeast saccharomyces cerevisiae over 300 million years ago. we describe a collection of 143 genetically matched strains of c. albicans , each of which has been deleted for a specific transcriptional regulator. this collection represents a large fraction of the non-essential transcription circuitry. a phenotypic profile for each mutant was developed using a screen of 55 growth conditions. the results identify the biological roles of many individual transcriptional regulators; for many, this work represents the first description of their functions. for example, a quarter of the strains showed altered colony formation, a phenotype reflecting transitions among yeast, pseudohyphal, and hyphal cell forms. these transitions, which have been closely linked to pathogenesis, have been extensively studied, yet our work nearly doubles the number of transcriptional regulators known to influence them. as a second example, nearly a quarter of the knockout strains affected sensitivity to commonly used antifungal drugs; although a few transcriptional regulators have previously been implicated in susceptibility to these drugs, our work indicates many additional mechanisms of sensitivity and resistance. finally, our results inform how transcriptional networks evolve. comparison with the existing s. cerevisiae data (supplemented by additional s. cerevisiae experiments reported here) allows the first systematic analysis of phenotypic conservation by orthologous transcriptional regulators over a large evolutionary distance. we find that, despite the many specific wiring changes documented between these species, the general phenotypes of orthologous transcriptional regulator knockouts are largely conserved. these observations support the idea that many wiring changes affect the detailed architecture of the circuit, but not its overall output.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsaccurate models of the cross-talk between signaling pathways and transcriptional regulatory networks within cells are essential to understand complex response programs. we present a new computational method that combines condition-specific time-series expression data with general protein interaction data to reconstruct dynamic and causal stress response networks. these networks characterize the pathways involved in the response, their time of activation, and the affected genes. the signaling and regulatory components of our networks are linked via a set of common transcription factors that serve as targets in the signaling network and as regulators of the transcriptional response network. detailed case studies of stress responses in budding yeast demonstrate the predictive power of our method. our method correctly identifies the core signaling proteins and transcription factors of the response programs. it further predicts the involvement of additional transcription factors and other proteins not previously implicated in the response pathways. we experimentally verify several of these predictions for the osmotic stress response network. our approach requires little condition-specific data: only a partial set of upstream initiators and time-series gene expression data, which are readily available for many conditions and species. consequently, our method is widely applicable and can be used to derive accurate, dynamic response models in several species.cells respond to stimuli by changes in various processes, including signaling pathways and gene expression. efforts to identify components of these responses increasingly depend on {mrna} profiling and genetic library screens. by comparing the results of these two assays across various stimuli, we found that genetic screens tend to identify response regulators, whereas {mrna} profiling frequently detects metabolic responses. we developed an integrative approach that bridges the gap between these data using known molecular interactions, thus highlighting major response pathways. we used this approach to reveal cellular pathways responding to the toxicity of alpha-synuclein, a protein implicated in several neurodegenerative disorders including parkinson\\'s disease. for this we screened an established yeast model to identify genes that when overexpressed alter alpha-synuclein toxicity. bridging these data and data from {mrna} profiling provided functional explanations for many of these genes and identified previously unknown relations between alpha-synuclein toxicity and basic cellular pathways.the functions of many open reading frames ({orfs}) identified in genome-sequencing projects are unknown. new, whole-genome approaches are required to systematically determine their function. a total of 6925 saccharomyces cerevisiae strains were constructed, by a high-throughput strategy, each with a precise deletion of one of 2026 {orfs} (more than one-third of the {orfs} in the genome). of the deleted {orfs}, 17 percent were essential for viability in rich medium. the phenotypes of more than 500 deletion strains were assayed in parallel. of the deletion strains, 40 percent showed quantitative growth defects in either rich or minimal medium.learning regulatory networks from genomics data is an important problem with applications spanning all of biology and biomedicine. functional genomics projects offer a cost-effective means of greatly expanding the completeness of our regulatory models, and for some prokaryotic organisms they offer a means of learning accurate models that incorporate the majority of the genome. there are, however, several reasons to believe that regulatory network inference is beyond our current reach, such as (i) the combinatorics of the problem, (ii) factors we can\\'t (or don\\'t often) collect genome-wide measurements for and (iii) dynamics that elude cost-effective experimental designs. recent works have demonstrated the ability to reconstruct large fractions of prokaryotic regulatory networks from compendiums of genomics data; they have also demonstrated that these global regulatory models can be used to predict the dynamics of the transcriptome. we review an overall strategy for the reconstruction of global networks based on these results in microbial systems.the precise regulation of many alternative splicing ({as}) events by specific splicing factors is essential to determine tissue types and developmental stages. however, the molecular basis of tissue-specific {as} regulation and the properties of splicing regulatory networks ({srns}) are poorly understood. here we comprehensively predict the targets of the brain- and muscle-specific splicing factor fox-1 ({a2bp1}) and its paralog fox-2 ({rbm9}) and systematically define the corresponding {srns} genome-wide. fox-1/2 are conserved from worm to human, and specifically recognize the {rna} element {ugcaug}. we integrate fox-1/2-binding specificity with phylogenetic conservation, splicing microarray data, and additional computational and experimental characterization. we predict thousands of fox-1/2 targets with conserved binding sites, at a false discovery rate ({fdr}) of ∼24\\\\%, including many validated experimentally, suggesting a surprisingly extensive {srn}. the preferred position of the binding sites differs according to {as} pattern, and determines either activation or repression of exon recognition by fox-1/2. many predicted targets are important for neuromuscular functions, and have been implicated in several genetic diseases. we also identified instances of binding site creation or loss in different vertebrate lineages and human populations, which likely reflect fine-tuning of gene expression regulation during evolution.{chip}-seq is widely used to characterize genome-wide binding patterns of transcription factors and other chromatin-associated proteins. although comparison of {chip}-seq data sets is critical for understanding cell type-dependent and cell state-specific binding, and thus the study of cell-specific gene regulation, few quantitative approaches have been developed. here, we present a simple and effective method, {manorm}, for quantitative comparison of {chip}-seq data sets describing transcription factor binding sites and epigenetic modifications. the quantitative binding differences inferred by {manorm} showed strong correlation with both the changes in expression of target genes and the binding of cell type-specific regulators.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismstranscriptional enhancers play critical roles in regulation of gene expression, but their identification in the eukaryotic genome has been challenging. recently, it was shown that enhancers in the mammalian genome are associated with characteristic histone modification patterns, which have been increasingly exploited for enhancer identification. however, only a limited number of cell types or chromatin marks have previously been investigated for this purpose, leaving the question unanswered whether there exists an optimal set of histone modifications for enhancer prediction in different cell types. here, we address this issue by exploring genome-wide profiles of 24 histone modifications in two distinct human cell types, embryonic stem cells and lung fibroblasts. we developed a {random-forest} based algorithm, {rfecs} (random forest based enhancer identification from chromatin states) to integrate histone modification profiles for identification of enhancers, and used it to identify enhancers in a number of cell-types. we show that {rfecs} not only leads to more accurate and precise prediction of enhancers than previous methods, but also helps identify the most informative and robust set of three chromatin marks for enhancer prediction.motivation: estimating gene regulatory networks over biological lineages is central to a deeper understanding of how cells evolve during development and differentiation. however, one challenge in estimating such evolving networks is that their host cells not only contiguously evolve, but also branch over time. for example, a stem cell evolves into two more specialized daughter cells at each division, forming a tree of networks. another example is in a laboratory setting: a biologist may apply several different drugs individually to malignant cancer cells to analyze the effects of each drug on the cells; the cells treated by one drug may not be intrinsically similar to those treated by another, but rather to the malignant cancer cells they were derived from.we describe an algorithm, multi-species {cmonkey}, for the simultaneous biclustering of heterogeneous multiple-species data collections and apply the algorithm to a group of bacteria containing bacillus subtilis, bacillus anthracis, and listeria monocytogenes. the algorithm reveals evolutionary insights into the surprisingly high degree of conservation of regulatory modules across these three species and allows data and insights from well-studied organisms to complement the analysis of related but less well studied organisms.expression quantitative trait loci ({eqtl}) mapping studies have become a widely used tool for identifying genetic variants that affect gene regulation. in these studies, expression levels are viewed as quantitative traits, and gene expression phenotypes are mapped to particular genomic loci by combining studies of variation in gene expression patterns with genome-wide genotyping. results from recent {eqtl} mapping studies have revealed substantial heritable variation in gene expression within and between populations. in many cases, genetic factors that influence gene expression levels can be mapped to proximal (putatively cis) {eqtls} and, less often, to distal (putatively trans) {eqtls}. beyond providing great insight into the biology of gene regulation, a combination of {eqtl} studies with results from traditional linkage or association studies of human disease may help predict a specific regulatory role for polymorphic sites previously associated with disease.high-throughput genomics, transcriptomics, proteomics and metabolomics have the potential to identify the functional consequences of induced and natural genetic variation. surprisingly, the experiments of most genomics researchers still mainly involve perturbing a biological system of interest by modifying either one factor or one gene at a time. by contrast, this article argues that multifactorial experimentation would allow the study of many more biologically relevant questions in parallel at the same or lower cost.many studies have identified quantitative trait loci ({qtls}) that contribute to continuous variation in heritable traits of interest. however, general principles regarding the distribution of {qtl} numbers, effect sizes, and combined effects of multiple {qtls} remain to be elucidated. here, we characterize complex genetics underlying inheritance of thousands of transcript levels in a cross between two strains of saccharomyces cerevisiae. most detected {qtls} have weak effects, with a median variance explained of 27\\\\% for highly heritable transcripts. despite the high statistical power of the study, no {qtls} were detected for 40\\\\% of highly heritable transcripts, indicating extensive genetic complexity. modeling of {qtl} detection showed that only 3\\\\% of highly heritable transcripts are consistent with single-locus inheritance, 17–18\\\\% are consistent with control by one or two loci, and half require more than five loci under additive models. strikingly, analysis of parent and progeny trait distributions showed that a majority of transcripts exhibit transgressive segregation. sixteen percent of highly heritable transcripts exhibit evidence of interacting loci. our results will aid design of future {qtl} mapping studies and may shed light on the evolution of quantitative traits.motivation: the integration of multiple datasets remains a key challenge in systems biology and genomic medicine. modern high-throughput technologies generate a broad array of different data types, providing distinct—but often complementary—information. we present a bayesian method for the unsupervised integrative modelling of multiple datasets, which we refer to as {mdi} (multiple dataset integration). {mdi} can integrate information from a wide range of different datasets and data types simultaneously (including the ability to model time series data explicitly using gaussian processes). each dataset is modelled using a dirichlet-multinomial allocation ({dma}) mixture model, with dependencies between these models captured through parameters that describe the agreement among the datasets.motivation: genetic networks are often described statistically using graphical models (e.g. bayesian networks). however, inferring the network structure offers a serious challenge in microarray analysis where the sample size is small compared to the number of considered genes. this renders many standard algorithms for graphical models inapplicable, and inferring genetic networks an \\'ill-posed\\' inverse problem.to dissect common human diseases such as obesity and diabetes, a systematic approach is needed to study how genes interact with one another, and with genetic and environmental factors, to determine clinical end points or disease phenotypes. bayesian networks provide a convenient framework for extracting relationships from noisy data and are frequently applied to large-scale data to derive causal relationships among variables of interest. given the complexity of molecular networks underlying common human disease traits, and the fact that biological networks can change depending on environmental conditions and genetic factors, large datasets, generally involving multiple perturbations (experiments), are required to reconstruct and reliably extract information from these networks. with limited resources, the balance of coverage of multiple perturbations and multiple subjects in a single perturbation needs to be considered in the experimental design. increasing the number of experiments, or the number of subjects in an experiment, is an expensive and time-consuming way to improve network reconstruction. integrating multiple types of data from existing subjects might be more efficient. for example, it has recently been demonstrated that combining genotypic and gene expression data in a segregating population leads to improved network reconstruction, which in turn may lead to better predictions of the effects of experimental perturbations on any given gene. here we simulate data based on networks reconstructed from biological data collected in a segregating mouse population and quantify the improvement in network reconstruction achieved using genotypic and gene expression data, compared with reconstruction using gene expression data alone. we demonstrate that networks reconstructed using the combined genotypic and gene expression data achieve a level of reconstruction accuracy that exceeds networks reconstructed from expression data alone, and that fewer subjects may be required to achieve this superior reconstruction accuracy. we conclude that this integrative genomics approach to reconstructing networks not only leads to more predictive network models, but also may save time and money by decreasing the amount of data that must be generated under any given condition of interest to construct predictive network models.we present the {yeast} search for transcriptional regulators and consensus tracking ({yeastract}; www.yeastract.com) database, a tool for the analysis of transcription regulatory associations in saccharomyces cerevisiae. this database is a repository of 12 346 regulatory associations between transcription factors and target genes, based on experimental evidence which was spread throughout 861 bibliographic references. it also includes 257 specific {dna}-binding sites for more than a hundred characterized transcription factors. further information about each yeast gene included in the database was obtained from saccharomyces genome database ({sgd}), regulatory sequences analysis tools and gene ontology ({go}) consortium. computational tools are also provided to facilitate the exploitation of the gathered data when solving a number of biological questions as exemplified in the tutorial also available on the system. {yeastract} allows the identification of documented or potential transcription regulators of a given gene and of documented or potential regulons for each transcription factor. it also renders possible the comparison between {dna} motifs, such as those found to be over-represented in the promoter regions of co-regulated genes, and the transcription factor-binding sites described in the literature. the system also provides an useful mechanism for grouping a list of genes (for instance a set of genes with similar expression profiles as revealed by microarray analysis) based on their regulatory associations with known transcription factors.motivation: the regulatory machinery controlling gene expression is complex, frequently requiring multiple, simultaneous {dna}-protein interactions. the rate at which a gene is transcribed may depend upon the presence or absence of a collection of transcription factors bound to the {dna} near the gene. locating transcription factor binding sites in genomic {dna} is difficult because the individual sites are small and tend to occur frequently by chance. true binding sites may be identified by their tendency to occur in clusters, sometimes known as regulatory modules.{background}:gut microbiota and the host exist in a mutualistic relationship, with the functional composition of the microbiota strongly affecting the health and well-being of the host. thus, it is important to develop a synthetic approach to study the host transcriptome and the microbiome simultaneously. early microbial colonization in infants is critically important for directing neonatal intestinal and immune development, and is especially attractive for studying the development of human-commensal interactions. here we report the results from a simultaneous study of the gut microbiome and host epithelial transcriptome of three-month-old exclusively breast- and formula-fed {infants.results}:variation in both host {mrna} expression and the microbiome phylogenetic and functional profiles was observed between breast- and formula-fed infants. to examine the interdependent relationship between host epithelial cell gene expression and bacterial metagenomic-based profiles, the host transcriptome and functionally profiled microbiome data were subjected to novel multivariate statistical analyses. gut microbiota metagenome virulence characteristics concurrently varied with immunity-related gene expression in epithelial cells between the formula-fed and the breast-fed {infants.conclusions}:our data provide insight into the integrated responses of the host transcriptome and microbiome to dietary substrates in the early neonatal period. we demonstrate that differences in diet can affect, via gut colonization, host expression of genes associated with the innate immune system. furthermore, the methodology presented in this study can be adapted to assess other host-commensal and host-pathogen interactions using genomic and transcriptomic data, providing a synthetic genomics-based picture of host-commensal relationships.10.1073/pnas.0806007105 the modular components, or subcircuits, of developmental gene regulatory networks ({grns}) execute specific developmental functions, such as the specification of cell identity. we survey examples of such subcircuits and relate their structures to corresponding developmental functions. these relations transcend organisms and genes, as illustrated by the similar structures of the subcircuits controlling the specification of the mesectoderm in the  embryo and the endomesoderm in the sea urchin, even though the respective subcircuits are composed of nonorthologous regulatory genes.motivation: as biological inquiry produces ever more network data, such as protein–protein interaction networks, gene regulatory networks and metabolic networks, many algorithms have been proposed for the purpose of pairwise network alignment—finding a mapping from the nodes of one network to the nodes of another in such a way that the mapped nodes can be considered to correspond with respect to both their place in the network topology and their biological attributes. this technique is helpful in identifying previously undiscovered homologies between proteins of different species and revealing functionally similar subnetworks. in the past few years, a wealth of different aligners has been published, but few of them have been compared with one another, and no comprehensive review of these algorithms has yet appeared.here we introduce the causal structure identification ({csi}) package, a gaussian process based approach to inferring gene regulatory networks ({grns}) from multiple time series data. the standard {csi} approach infers a single {grn} via joint learning from multiple time series datasets; the hierarchical approach ({hcsi}) infers a separate {grn} for each dataset, albeit with the networks constrained to favor similar structures, allowing for the identification of context specific networks. the software is implemented in {matlab} and includes a graphical user interface ({gui}) for user friendly inference. finally the {gui} can be connected to high performance computer clusters to facilitate analysis of large genomic datasets.motivation: we addressed the problem of inferring gene regulatory network ({grn}) from gene expression data of knockout ({ko}) experiments. this inference is known to be underdetermined and the {grn} is not identifiable from data. past studies have shown that suboptimal design of experiments ({doe}) contributes significantly to the identifiability issue of biological networks, including {grns}. however, optimizing {doe} has received much less attention than developing methods for {grn} inference.{flyatlas}, a new online resource, provides the most comprehensive view yet of expression in multiple tissues of drosophila melanogaster. meta-analysis of the data shows that a significant fraction of the genome is expressed with great tissue specificity in the adult, demonstrating the need for the functional genomic community to embrace a wide range of functional phenotypes. well-known developmental genes are often reused in surprising tissues in the adult, suggesting new functions. the homologs of many human genetic disease loci show selective expression in the drosophila tissues analogous to the affected human tissues, providing a useful filter for potential candidate genes. additionally, the contributions of each tissue to the whole-fly array signal can be calculated, demonstrating the limitations of whole-organism approaches to functional genomics and allowing modeling of a simple tissue fractionation procedure that should improve detection of weak or tissue-specific signals.transposable elements have been shaping the genome throughout evolution, contributing to the creation of new genes and sophisticated regulatory network systems. today, most of genomes (animals and plants) allow the expression and accommodate transposition of a few transposon families. the potential genetic impact of this small fraction of mobile elements should not be underestimated. although new insertions that happen in germ cells are likely to be passed to the next generation, mobilization in pluripotent embryonic stem cells or in somatic cells may contribute to the differences observed in genetic makeup and epigenetic gene regulation during development at the cellular level. the fact that these elements are still active, generating innovative ways to alter gene expression and genomic structure, suggests that the cellular genome is not static or deterministic but rather dynamic. in this short review, we collect a set of recent observations that point to a new appreciation of transposable elements as a source of genetic variation.genetics aims to understand the relation between genotype and phenotype. however, because complete deletion of most yeast genes (∼80\\\\%) has no obvious phenotypic consequence in rich medium, it is difficult to study their functions. to uncover phenotypes for this nonessential fraction of the genome, we performed 1144 chemical genomic assays on the yeast whole-genome heterozygous and homozygous deletion collections and quantified the growth fitness of each deletion strain in the presence of chemical or environmental stress conditions. we found that 97\\\\% of gene deletions exhibited a measurable growth phenotype, suggesting that nearly all genes are essential for optimal growth in at least one condition.maximum likelihood estimates ({mles}) in autologistic models and other exponential family models for dependent data can be calculated with markov chain monte carlo methods (the metropolis algorithm or the gibbs sampler), which simulate ergodic markov chains having equilibrium distributions in the model. from one realization of such a markov chain, a monte carlo approximant to the whole likelihood function can be constructed. the parameter value (if any) maximizing this function approximates the {mle}. when no parameter point in the model maximizes the likelihood, the {mle} in the closure of the exponential family may exist and can be calculated by a two-phase algorithm, first finding the support of the {mle} by linear programming and then finding the distribution within the family conditioned on the support by maximizing the likelihood for that family. these methods are illustrated by a constrained autologistic model for {dna} fingerprint data. {mles} are compared with maximum pseudolikelihood estimates ({mples}) and with maximum conditional likelihood estimates ({mcles}), neither of which produce acceptable estimates, the {mple} because it overestimates dependence, and the {mcle} because conditioning removes the constraints.interaction (nonadditive effects) between genetic variants has been highlighted as an important mechanism underlying phenotypic variation, but the discovery of genetic interactions in humans has proved difficult. in this study, we show that the spectrum of variation in the human genome has been shaped by modifier effects of cis-regulatory variation on the functional impact of putatively deleterious protein-coding variants. we analyzed 1000 genomes population-scale resequencing data from europe ({ceu} [utah residents with northern and western european ancestry from the {ceph} collection]) and africa ({yri} [yoruba in ibadan, nigeria]) together with gene expression data from arrays and {rna} sequencing for the same samples. we observed an underrepresentation of derived putatively functional coding variation on the more highly expressed regulatory haplotype, which suggests stronger purifying selection against deleterious coding variants that have increased penetrance because of their regulatory background. furthermore, the frequency spectrum and impact size distribution of common regulatory polymorphisms ({eqtls}) appear to be shaped in order to minimize the selective disadvantage of having deleterious coding mutations on the more highly expressed haplotype. interestingly, {eqtls} explaining common disease {gwas} signals showed an enrichment of putative epistatic effects, suggesting that some disease associations might arise from interactions increasing the penetrance of rare coding variants. in conclusion, our results indicate that regulatory and coding variants often modify the functional impact of each other. this specific type of genetic interaction is detectable from sequencing data in a genome-wide manner, and characterizing these joint effects might help us understand functional mechanisms behind genetic associations to human phenotypes\\x14including both mendelian and common disease.predicting organismal phenotypes from genotype data is important for plant and animal breeding, medicine, and evolutionary biology. genomic-based phenotype prediction has been applied for single-nucleotide polymorphism ({snp}) genotyping platforms, but not using complete genome sequences. here, we report genomic prediction for starvation stress resistance and startle response in drosophila melanogaster, using \\\\~{}2.5 million {snps} determined by sequencing the drosophila genetic reference panel population of inbred lines. we constructed a genomic relationship matrix from the {snp} data and used it in a genomic best linear unbiased prediction ({gblup}) model. we assessed predictive ability as the correlation between predicted genetic values and observed phenotypes by cross-validation, and found a predictive ability of 0.239±0.008 (0.230±0.012) for starvation resistance (startle response). the predictive ability of {bayesb}, a bayesian method with internal {snp} selection, was not greater than {gblup}. selection of the 5\\\\% {snps} with either the highest absolute effect or variance explained did not improve predictive ability. predictive ability decreased only when fewer than 150,000 {snps} were used to construct the genomic relationship matrix. we hypothesize that predictive power in this population stems from the {snp}–based modeling of the subtle relationship structure caused by long-range linkage disequilibrium and not from population structure or {snps} in linkage disequilibrium with causal variants. we discuss the implications of these results for genomic prediction in other organisms. the ability to accurately predict values of complex phenotypes from genotype data will revolutionize plant and animal breeding, personalized medicine, and evolutionary biology. to date, genomic prediction has utilized high-density single-nucleotide polymorphism ({snp}) genotyping arrays, but the availability of sequence data opens new frontiers for genomic prediction methods. this article is the first application of genomic phenotype prediction using whole-genome sequence data in a substantial sample of a higher eukaryote. we use \\\\~{}2.5 million {snps} with minor allele frequency greater than 2.5\\\\% derived from genomic sequences of the  ” drosophila genetic reference panel” to predict phenotypes for two traits, starvation resistance and startle-induced locomotor behavior. we systematically address prediction within versus across sexes, genomic best linear unbiased prediction ({gblup}) versus a bayesian approach, and the effect of {snp} density. we find that (i) genomic prediction can be efficiently implemented using sequence data via {gblup}, (ii) there is little gain in predictive ability if the number of {snps} is increased above 150,000, and (iii) neither implicit nor explicit marker selection substantially improves the predictive ability. although the findings must be seen against the background of small sample sizes, the results illustrate both the potential of the approach and the challenges ahead.a major goal of system biology is the characterization of transcription factors and {micrornas} ({mirnas}) and the transcriptional programs they regulate. we present allegro, a method for de-novo discovery of cis-regulatory transcriptional programs through joint analysis of genome-wide expression data and promoter or 3′ {utr} sequences. the algorithm uses a novel log-likelihood-based, non-parametric model to describe the expression pattern shared by a group of co-regulated genes. we show that allegro is more accurate and sensitive than existing techniques, and can simultaneously analyze multiple expression datasets with more than 100 conditions. we apply allegro on datasets from several species and report on the transcriptional modules it uncovers. our analysis reveals a novel motif over-represented in the promoters of genes highly expressed in murine oocytes, and several new motifs related to fly development. finally, using stem-cell expression profiles, we identify three {mirna} families with pivotal roles in human embryogenesis.an international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismsan international, peer-reviewed genome sciences journal featuring outstanding original research that offers novel insights into the biology of all organismscombined analyses of molecular data, such as {dna} copy-number alteration, {mrna} and protein expression, point to biological functions and molecular pathways being deregulated in multiple cancers. genomic, metabolomic and clinical data from various solid cancers and model systems are emerging and can be used to{background}: the introduction of measures such as evolutionary rate and propensity for gene loss have significantly advanced our knowledge of the evolutionary history and selection forces acting upon individual genes and cellular processes. {results}: we present two new measures, the \\'relative evolutionary rate pattern\\' ({rerp}), which records the relative evolutionary rates of conserved genes across the different branches of a species\\' phylogenetic tree, and the \\'copy number pattern\\' ({cnp}), which quantifies the rate of gene loss of less conserved genes. together, these measures yield a high-resolution study of the co-evolution of genes in 9 fungal species, spanning 3,540 sets of orthologs. we find that the evolutionary tempo of conserved genes varies in different evolutionary periods. the co-evolution of genes\\' gene ontology categories exhibits a significant correlation with their functional distance in the gene ontology hierarchy, but not with their location on chromosomes, showing that cellular functions are a more important driving force in gene co-evolution than their chromosomal proximity. two fundamental patterns of co-evolution of conserved genes, cooperative and reciprocal, are identified; only genes co-evolving cooperatively functionally back each other up. the co-evolution of conserved and less conserved genes exhibits both commonalities and differences; {dna} metabolism is positively correlated with nuclear traffic, transcription processes and vacuolar biology in both analyses. {conclusions}: overall, this study charts the first global network view of gene co-evolution in fungi. the future application of the approach presented here to other phylogenetic trees holds much promise in characterizing the forces that shape cellular co-evolution.large-scale protein-protein interaction networks provide new opportunities for understanding cellular organization and functioning. we introduce network schemas to elucidate shared mechanisms within interactomes. network schemas specify descriptions of proteins and the topology of interactions among them. we develop algorithms for systematically uncovering recurring, over-represented schemas in physical interaction networks. we apply our methods to the s. cerevisiae interactome, focusing on schemas consisting of proteins described via sequence motifs and molecular function annotations and interacting with one another in one of four basic network topologies. we identify hundreds of recurring and over-represented network schemas of various complexity, and demonstrate via graph-theoretic representations how more complex schemas are organized in terms of their lower-order constituents. the uncovered schemas span a wide range of cellular activities, with many signaling and transport related higher-order schemas. we establish the functional importance of the schemas by showing that they correspond to functionally cohesive sets of proteins, are enriched in the frequency with which they have instances in the h. sapiens interactome, and are useful for predicting protein function. our findings suggest that network schemas are a powerful paradigm for organizing, interrogating, and annotating cellular networks. large-scale networks of protein-protein interactions provide a view into the workings of the cell. however, these interaction maps do not come with a key for interpreting them, so it is necessary to develop methods that shed light on their functioning and organization. we propose the language of network schemas for describing recurring patterns of specific types of proteins and their interactions. that is, network schemas describe proteins and specify the topology of interactions among them. a single network schema can describe, for example, a common template that underlies several distinct cellular pathways, such as signaling pathways. we develop a computational methodology for identifying network schemas that are recurrent and over-represented in the network, even given the distributions of their constituent components. we apply this methodology to the physical interaction network in s. cerevisiae and begin to build a hierarchy of schemas starting with the four simplest topologies. we validate the biological relevance of the schemas that we find, discuss the insights our findings lend into the organization of interactomes, touch upon cross-genomic aspects of schema analysis, and show how to use schemas to annotate uncharacterized protein families.local adaptations within species are often governed by several interacting genes scattered throughout the genome. single-locus models of selection cannot explain the maintenance of such complex variation because recombination separates co-adapted alleles. here we report a previously unrecognized type of intraspecific multi-locus genetic variation that has been maintained over a vast period. the galactose ({gal}) utilization gene network of saccharomyces kudriavzevii, a relative of brewer\\'s yeast, exists in two distinct states: a functional gene network in portuguese strains and, in japanese strains, a non-functional gene network of allelic pseudogenes. genome sequencing of all available s. kudriavzevii strains revealed that none of the functional {gal} genes were acquired from other species. rather, these polymorphisms have been maintained for nearly the entire history of the species, despite more recent gene flow genome-wide. experimental evidence suggests that inactivation of the {gal3} and {gal80} regulatory genes facilitated the origin and long-term maintenance of the two gene network states. this striking example of a balanced unlinked gene network polymorphism introduces a remarkable type of intraspecific variation that may be widespread.whereas genomic data are universally machine-readable, data from imaging, multiplex biochemistry, flow cytometry and other cell- and tissue-based assays usually reside in loosely organized files of poorly documented provenance. this arises because the relational databases used in genomic research are difficult to adapt to rapidly evolving experimental designs, data formats and analytic algorithms. here we describe an adaptive approach to managing experimental data based on semantically typed data hypercubes ({sdcubes}) that combine hierarchical data format 5 ({hdf5}) and extensible markup language ({xml}) file types. we demonstrate the application of {sdcube}-based storage using {imagerail}, a software package for high-throughput microscopy. experimental design and its day-to-day evolution, not rigid standards, determine how {imagerail} data are organized in {sdcubes}. we applied {imagerail} to collect and analyze drug dose-response landscapes in human cell lines at single-cell resolution.a database of coexpressed gene sets can provide valuable information for a wide variety of experimental designs, such as targeting of genes for functional identification, gene regulation and/or protein-protein interactions. coexpressed gene databases derived from publicly available {genechip} data are widely used in arabidopsis research, but platforms that examine coexpression for higher mammals are rather limited. therefore, we have constructed a new database, {coxpresdb} (coexpressed gene database) (http://coxpresdb.hgc.jp), for coexpressed gene lists and networks in human and mouse. coexpression data could be calculated for 19 777 and 21 036 genes in human and mouse, respectively, by using the {genechip} data in {ncbi} {geo}. {coxpresdb} enables analysis of the four types of coexpression networks: (i) highly coexpressed genes for every gene, (ii) genes with the same {go} annotation, (iii) genes expressed in the same tissue and (iv) user-defined gene sets. when the networks became too big for the static picture on the web in {go} networks or in tissue networks, we used google maps {api} to visualize them interactively. {coxpresdb} also provides a view to compare the human and mouse coexpression patterns to estimate the conservation between the two species. 10.1093/nar/gkm840cis-regulatory sequences are not always conserved across species. divergence within cis-regulatory sequences may result from the evolution of species-specific patterns of gene expression or the flexible nature of the cis-regulatory code. the identification of functional divergence in cis-regulatory sequences is therefore important for both understanding the role of gene regulation in evolution and annotating regulatory elements. we have developed an evolutionary model to detect the loss of constraint on individual transcription factor binding sites ({tfbss}). we find that a significant fraction of functionally constrained binding sites have been lost in a lineage-specific manner among three closely related yeast species. binding site loss has previously been explained by turnover, where the concurrent gain and loss of a binding site maintains gene regulation. we estimate that nearly half of all loss events cannot be explained by binding site turnover. recreating the mutations that led to binding site loss confirms that these sequence changes affect gene expression in some cases. we also estimate that there is a high rate of binding site gain, as more than half of experimentally identified s. cerevisiae binding sites are not conserved across species. the frequent gain and loss of {tfbss} implies that cis-regulatory sequences are labile and, in the absence of turnover, may contribute to species-specific patterns of gene expression.defining the functional relationships between proteins is critical for understanding virtually all aspects of cell biology. large-scale identification of protein complexes has provided one important step towards this goal; however, even knowledge of the stoichiometry, affinity and lifetime of every protein-protein interaction would not reveal the functional relationships between and within such complexes. genetic interactions can provide functional information that is largely invisible to protein-protein interaction data sets. here we present an epistatic miniarray profile ({e-map}) consisting of quantitative pairwise measurements of the genetic interactions between 743 saccharomyces cerevisiae genes involved in various aspects of chromosome biology (including {dna} replication/repair, chromatid segregation and transcriptional regulation). this {e-map} reveals that physical interactions fall into two well-represented classes distinguished by whether or not the individual proteins act coherently to carry out a common function. thus, genetic interaction data make it possible to dissect functionally multi-protein complexes, including mediator, and to organize distinct protein complexes into pathways. in one pathway defined here, we show that rtt109 is the founding member of a novel class of histone acetyltransferases responsible for asf1-dependent acetylation of histone h3 on lysine 56. this modification, in turn, enables a ubiquitin ligase complex containing the cullin rtt101 to ensure genomic integrity during {dna} replication.sea stars and sea urchins evolved from a last common ancestor that lived at the end of the cambrian, approximately half a billion years ago. in a previous comparative study of the gene regulatory networks ({grns}) that embody the genomic program for embryogenesis in these animals, we discovered an almost perfectly conserved five-gene network subcircuit required for endoderm specification. we show here that the {grn} structure upstream and downstream of the conserved network kernel has, by contrast, diverged extensively. mesoderm specification is accomplished quite differently; the {delta-notch} signaling system is used in radically distinct ways; and various regulatory genes have been coopted to different functions. the conservation of the conserved kernel is thus the more remarkable. the results indicate types of network linkage subject to evolutionary change. an emergent theme is that subcircuit design may be preserved even while the identity of genes performing given roles changes because of alteration in their cis-regulatory control systems.the rapidly increasing quantity of genome-wide chromosome conformation capture data presents great opportunities and challenges in the computational modeling and interpretation of the three-dimensional genome. in particular, with recent trends towards higher-resolution high-throughput chromosome conformation capture ({hi-c}) data, the diversity and complexity of biological hypotheses that can be tested necessitates rigorous computational and statistical methods as well as scalable pipelines to interpret these datasets. here we review computational tools to interpret {hi-c} data, including pipelines for mapping, filtering, and normalization, and methods for confidence estimation, domain calling, visualization, and three-dimensional modeling.data may now be recorded concurrently from {eeg} and functional {mri}, using the simultaneous imaging for tomographic electrophysiology ({site}) method. as yet, there is no established means to integrate the analysis of the combined data set. recognizing that the hemodynamically convolved time-varying {eeg} spectrum, s , is intrinsically multidimensional in space, frequency, and time motivated us to use multiway partial {least-squares} ({n-pls}) analysis to decompose {eeg} (independent variable) and {fmri} (dependent variable) data uniquely as a sum of  ” atoms”. each {eeg} atom is the outer product of spatial, spectral, and temporal signatures and each {fmri} atom the product of spatial and temporal signatures. the decomposition was constrained to maximize the covariance between corresponding temporal signatures of the {eeg} and {fmri}. on all data sets, three components whose spectral peaks were in the theta, alpha, and gamma bands appeared; only the alpha atom had a significant temporal correlation with the {fmri} signal. the spatial distribution of the alpha-band atom included parieto-occipital cortex, thalamus, and insula, and corresponded closely to that reported by goldman et al. [{neuroreport} 13(18) (2002) 2487] using a more conventional analysis. the source reconstruction from {eeg} spatial signature showed only the parieto-occipital sources. we interpret these results to indicate that some electrical sources may be intrinsically invisible to scalp {eeg}, yet may be revealed through conjoint analysis of {eeg} and {fmri} data. these results may also expose brain regions that participate in the control of brain rhythms but may not themselves be generators. as of yet, no single neuroimaging method offers the optimal combination of spatial and temporal resolution; fusing {fmri} and {eeg} meaningfully extends the spatio-temporal resolution and sensitivity of each method.blood production in humans culminates in the daily release of around 1011 cells into the circulation, mainly platelets and red blood cells. all blood cells originate from a minute population of hematopoietic stem cells ({hscs}) that expands and differentiates into progenitor cells with increasingly restricted lineage choice. characterizing alternative splicing events involved in hematopoiesis is critical for interpreting the effects of mutations leading to inherited disorders and blood cancers and for the rational design of strategies to advance transplantation and regenerative medicine.although the human genome has been sequenced, progress in understanding gene regulation in humans has been particularly slow. many computational approaches developed for lower eukaryotes to identify cis-regulatory elements and their associated target genes often do not generalize to mammals, largely due to the degenerate and interactive nature of such elements. motivated by the switch-like behavior of transcriptional responses, we present a systematic approach that allows adaptive determination of active transcriptional subnetworks (cis-motif combinations, the direct target genes and physiological processes regulated by the corresponding transcription factors) from microarray data in mammals, with accuracy similar to that achieved in lower eukaryotes. our analysis uncovered several new subnetworks active in human liver and in cell-cycle regulation, with similar functional characteristics as the known ones. we present biochemical evidence for our predictions, and show that the recently discovered {g2/m}-specific {e2f} pathway is wider than previously thought; in particular, {e2f} directly activates certain mitotic genes involved in hepatocellular carcinomas. additionally, we demonstrate that this method can predict subnetworks in a condition-specific manner, as well as regulatory crosstalk across multiple tissues. our approach allows systematic understanding of how phenotypic complexity is regulated at the transcription level in mammals and offers marked advantage in systems where little or no prior knowledge of transcriptional regulation is available.motivation: one of the present challenges in biological research is the organization of the data originating from high-throughput technologies. one way in which this information can be organized is in the form of networks of influences, physical or statistical, between cellular components. we propose an experimental method for probing biological networks, analyzing the resulting data and reconstructing the network architecture.  methods: we use networks of known topology consisting of nodes (genes), directed edges (gene-gene interactions) and a dynamics for the genes\\' {mrna} concentrations in terms of the gene-gene interactions. we proposed a network reconstruction algorithm based on the conditional correlation of the {mrna} equilibrium concentration between two genes given that one of them was knocked down. using simulated gene expression data on networks of known connectivity, we investigated how the reconstruction error is affected by noise, network topology, size, sparseness and dynamic parameters.  results: errors arise from correlation between nodes connected through intermediate nodes (false positives) and when the correlation between two directly connected nodes is obscured by noise, non-linearity or multiple inputs to the target node (false negatives). two critical components of the method are as follows: (1) the choice of an optimal correlation threshold for predicting connections and (2) the reduction of errors arising from indirect connections (for which a novel algorithm is proposed). with these improvements, we can reconstruct networks with the topology of the transcriptional regulatory network in escherichia coli with a reasonably low error rate.  contact: gustavo@us.ibm.com  supplementary information: available from our website: {www.research.ibm.com/fungen} 10.1093/bioinformatics/bti064nucleosomes in all eukaryotes examined to date adopt a characteristic architecture within genes and play fundamental roles in regulating transcription, yet the identity and precise roles of many of the trans-acting factors responsible for the establishment and maintenance of this organization remain to be identified. we profiled a compendium of 50 yeast strains carrying conditional alleles or complete deletions of genes involved in transcriptional regulation, histone biology, and chromatin remodeling, as well as compounds that target transcription and histone deacetylases, to assess their respective roles in nucleosome positioning and transcription. we find that nucleosome patterning in genes is affected by many factors, including the {caf}-1 complex, spt10, and spt21, in addition to previously reported remodeler {atpases} and histone chaperones. disruption of these factors or reductions in histone levels led genic nucleosomes to assume positions more consistent with their intrinsic sequence preferences, with pronounced and specific shifts of the +1 nucleosome relative to the transcription start site. these shifts of +1 nucleosomes appear to have functional consequences, as several affected genes in ino80 mutants exhibited altered expression responses. our parallel expression profiling compendium revealed extensive transcription changes in intergenic and antisense regions, most of which occur in regions with altered nucleosome occupancy and positioning. we show that the nucleosome-excluding transcription factors reb1, abf1, tbf1, and rsc3 suppress cryptic transcripts at their target promoters, while a combined analysis of nucleosome and expression profiles identified 36 novel transcripts that are normally repressed by {tup1/cyc8}. our data confirm and extend the roles of chromatin remodelers and chaperones as major determinants of genic nucleosome positioning, and these data provide a valuable resource for future studies. the genome in eukaryotic cells is packaged into nucleosomes, which play critical roles in regulating where and when different genes are expressed. for example, nucleosomes can physically block access of transcription factor to sites on {dna} or direct regulatory proteins to {dna}. consistent with these roles, nucleosomes assume a stereotypical pattern around genes: they are depleted at the promoter region that marks the start of genes and assume a regularly spaced array within genes. to identify factors involved in this organization, we generated high-resolution nucleosome and transcriptome maps for 50 loss-of-function mutants with known or suspected roles in nucleosome biology in budding yeast. we show that nucleosome organization is determined by the combined effects of many factors that often exert opposing forces on nucleosomes. we further demonstrate that specific nucleosomes can be positioned independently within genes and that repositioning of nucleosomes at the start of genes may affect expression of these genes in response to environmental stimuli. data mining of this extensive resource allowed us to show that general transcription factors act as insulators at diverging promoters to prevent the formation of cryptic transcripts, and also revealed 36 novel transcripts regulated by the {tup1/cyc8} complex.gottgens and colleagues have analysed the expression of 18 haematopoietic factors in single primary blood and progenitor cells from mouse bone marrow. they delineate distinct states of expression for these transcription factors and identify regulatory relationships between the key factors gata2, gfi1 and gfi2.we have analyzed publicly available k562 {hi-c} data, which enable genome-wide unbiased capturing of chromatin interactions, using a mixture poisson regression model and a power-law decay background to define a highly specific set of interacting genomic regions. we integrated multiple {encode} consortium resources with the {hi-c} data, using {dnase}-seq data and {chip}-seq data for 45 transcription factors and 9 histone modifications. we classified 12 different sets (clusters) of interacting loci that can be distinguished by their chromatin modifications and which can be categorized into two types of chromatin linkages. the different clusters of loci display very different relationships with transcription factor-binding sites. as expected, many of the transcription factors show binding patterns specific to clusters composed of interacting loci that encompass promoters or enhancers. however, cluster 9, which is distinguished by marks of open chromatin but not by active enhancer or promoter marks, was not bound by most transcription factors but was highly enriched for three transcription factors ({gata1}, {gata2} and {c-jun}) and three chromatin modifiers ({brg1}, {ini1} and {sirt6}). to investigate the impact of chromatin organization on gene regulation, we performed ribonucleicacid-seq analyses before and after knockdown of {gata1} or {gata2}. we found that knockdown of the {gata} factors not only alters the expression of genes having a nearby bound {gata} but also affects expression of genes in interacting loci. our work, in combination with previous studies linking regulation by {gata} factors with {c-jun} and {brg1}, provides genome-wide evidence that {hi-c} data identify sets of biologically relevant interacting loci.motivation: time-course gene expression data are often measured to study dynamic biological systems and gene regulatory networks. to account for time dependency of the gene expression measurements over time and the noisy nature of the microarray data, the mixed-effects model using b-splines was introduced. this paper further explores such mixed-effects model in analyzing the time-course gene expression data and in performing clustering of genes in a mixture model framework.we present {distiller}, a data integration framework for the inference of transcriptional module networks. experimental validation of predicted targets for the well-studied fumarate nitrate reductase regulator showed the effectiveness of our approach in escherichia coli. in addition, the condition dependency and modularity of the inferred transcriptional network was studied. surprisingly, the level of regulatory complexity seemed lower than that which would be expected from {regulondb}, indicating that complex regulatory programs tend to decrease the degree of modularity.motivation: knowledge of the activation patterns of transcription factors ({tfs}) is fundamental to elucidate the dynamics of gene regulation in response to environmental conditions. direct experimental measurement of {tfs}\\' activities is, however, challenging, resulting in a need to develop statistical tools to infer {tf} activities from {mrna} expression levels of target genes. current models, however, neglect important features of transcriptional regulation; in particular, the combinatorial nature of regulation, which is fundamental for signal integration, is not accounted {for.results}: we present a novel method to infer combinatorial regulation of gene expression by multiple transcription factors in large-scale transcriptional regulatory networks. the method implements a factorial hidden markov model with a non-linear likelihood to represent the interactions between the hidden transcription factors. we explore our model\\'s performance on artificial datasets and demonstrate the applicability of our method on genome-wide scale for three expression datasets. the results obtained using our model are biologically coherent and provide a tool to explore the concealed nature of combinatorial transcriptional {regulation.availability}: {http://homepages.inf.ed.ac.uk/gsanguin/software.html.contact}: {g.sanguinetti@ed.ac.uksupplementary} information: supplementary data are available at bioinformatics online.regulatory divergence is likely a major driving force in evolution. comparative transcriptomics provides a new glimpse into the evolution of gene regulation. ascomycota fungi are uniquely suited among eukaryotes for studies of regulatory evolution, because of broad phylogenetic scope, many sequenced genomes, and facility of genomic analysis. here we review the substantial divergence in gene expression in ascomycota and how this is reconciled with the modular organization of transcriptional networks. we show that flexibility and redundancy in both cis-regulation and trans-regulation can lead to changes from altered expression of single genes to wholesale rewiring of regulatory modules. redundancy thus emerges as a major driving force facilitating expression divergence while preserving the coherent functional organization of a transcriptional response.transcription factor binding site(s) ({tfbs}) gain and loss (i.e., turnover) is a well-documented feature of cis-regulatory module ({crm}) evolution, yet little attention has been paid to the evolutionary force(s) driving this turnover process. the predominant view, motivated by its widespread occurrence, emphasizes the importance of compensatory mutation and genetic drift. positive selection, in contrast, although it has been invoked in specific instances of adaptive gene expression evolution, has not been considered as a general alternative to neutral compensatory evolution. in this study we evaluate the two hypotheses by analyzing patterns of single nucleotide polymorphism in the {tfbs} of well-characterized {crm} in two closely related drosophila species, drosophila melanogaster and drosophila simulans. an important feature of the analysis is classification of {tfbs} mutations according to the direction of their predicted effect on binding affinity, which allows gains and losses to be evaluated independently along the two phylogenetic lineages. the observed patterns of polymorphism and divergence are not compatible with neutral evolution for either class of mutations. instead, multiple lines of evidence are consistent with contributions of positive selection to {tfbs} gain and loss as well as purifying selection in its maintenance. in discussion, we propose a model to reconcile the finding of selection driving {tfbs} turnover with constrained {crm} function over long evolutionary time. transcription factor binding sites ({tfbs}) turnover (i.e. lineage-specific gain and loss) is a well-documented phenomenon in eukaryote cis-regulatory modules ({crm}). the wide spread of the phenomenon and the appearance of conserved expression patterns for diverged orthologous {crm} led to the standing view that the observed gain and loss of {tfbs} were functionally and selectively neutral. to the contrary, genome-wide population genetics analyses have unequivocally identified signatures of positive selection acting in noncoding regions in general, and particularly in 5′ and 3′ untranscribed regions of genes. to specifically test the neutral versus selection hypotheses for the {tfbs} turnover process, we analyzed natural variation patterns within and between two closely related drosophila species. we found the patterns of divergence and polymorphism for two types of mutations—those inferred to increase or decrease the binding affinity respectively—are not compatible with a neutral hypothesis. instead, multiple lines of evidence suggested that positive selection has contributed to gain as well as loss of {tfbs} in the two lineages, with purifying selection maintaining existing {tfbs} in the population. spacer sequences also showed signatures of negative and positive selection. we proposed a model of {crm} evolution to reconcile the finding of frequent adaptive changes with constraints on long-term evolution.recent advances in next-generation {dna} sequencing and proteomics provide an unprecedented ability to survey {mrna} and protein abundances. such proteome-wide surveys are illuminating the extent to which different aspects of gene expression help to regulate cellular protein abundances. current data demonstrate a substantial role for regulatory processes occurring after {mrna} is made - that is, post-transcriptional, translational and protein degradation regulation - in controlling steady-state protein abundances. intriguing observations are also emerging in relation to cells following perturbation, single-cell studies and the apparent evolutionary conservation of protein and {mrna} abundances. here, we summarize current understanding of the major factors regulating protein expression.decoding {dna} symbols using next-generation sequencers was a major breakthrough in genomic research. despite the many advantages of next-generation sequencers, e.g., the high-throughput sequencing rate and relatively low cost of sequencing, the assembly of the reads produced by these sequencers still remains a major challenge. in this review, we address the basic framework of next-generation genome sequence assemblers, which comprises four basic stages: preprocessing filtering, a graph construction process, a graph simplification process, and postprocessing filtering. here we discuss them as a framework of four stages for data analysis and processing and survey variety of techniques, algorithms, and software tools used during each stage. we also discuss the challenges that face current assemblers in the next-generation environment to determine the current state-of-the-art. we recommend a layered architecture approach for constructing a general assembler that can handle the sequences generated by different sequencing platforms.variation among individuals is a prerequisite of evolution by natural selection. as such, identifying the origins of variation is a fundamental goal of biology. we investigated the link between gene interactions and variation in gene expression among individuals and species using the mammalian limb as a model system. we first built interaction networks for key genes regulating early (outgrowth; e9.5–11) and late (expansion and elongation; e11-13) limb development in mouse. this resulted in an early ({esn}) and late ({lsn}) stage network. computational perturbations of these networks suggest that the {esn} is more robust. we then quantified levels of the same key genes among mouse individuals and found that they vary less at earlier limb stages and that variation in gene expression is heritable. finally, we quantified variation in gene expression levels among four mammals with divergent limbs (bat, opossum, mouse and pig) and found that levels vary less among species at earlier limb stages. we also found that variation in gene expression levels among individuals and species are correlated for earlier and later limb development. in conclusion, results are consistent with the robustness of the {esn} buffering among-individual variation in gene expression levels early in mammalian limb development, and constraining the evolution of early limb development among mammalian species. the variation generating mechanisms of development interact with the variation sorting mechanism of natural selection to produce organismal diversity. while the impacts of natural selection on existing variation have received much study, those of development on the generation of this variation remain less understood. this fundamental gap in our knowledge restricts our understanding of the key processes shaping evolution. in this study, we combine mathematical modeling, and population-level and cross-species assays of gene expression to investigate the relationship between the structure of the gene interactions regulating limb development and variation in the expression of limb genes among individuals and species. results suggest that the way in which genes interact (i.e., development) biases the distribution of variation in gene expression among individuals, and that this in turn biases the distribution of variation among species.a major goal of systems biology is the development of models that accurately predict responses to perturbation. constructing such models requires the collection of dense measurements of system states, yet transformation of data into predictive constructs remains a challenge. to begin to model human immunity, we analyzed immune parameters in depth both at baseline and in response to influenza vaccination. peripheral blood mononuclear cell transcriptomes, serum titers, cell subpopulation frequencies, and b cell responses were assessed in 63 individuals before and after vaccination and were used to develop a systematic framework to dissect inter- and intra-individual variation and build predictive models of postvaccination antibody responses. strikingly, independent of age and pre-existing antibody titers, accurate models could be\\xa0constructed using pre-perturbation cell populations alone, which were validated using independent baseline time points. most of the parameters contributing to prediction delineated temporally stable baseline differences across individuals, raising the prospect of immune monitoring before intervention.epigenetic changes are being increasingly recognized as a prominent feature of cancer. this occurs not only at individual genes, but also over larger chromosomal domains. to investigate this, we set out to identify large chromosomal domains of epigenetic dysregulation in breast cancers.high-throughput, \\'omic\\' methods provide sensitive measures of biological responses to perturbations. however, inherent biases in high-throughput assays make it difficult to interpret experiments in which more than one type of data is collected. in this work, we introduce omics integrator, a software package that takes a variety of \\'omic\\' data as input and identifies putative underlying molecular pathways. the approach applies advanced network optimization algorithms to a network of thousands of molecular interactions to find high-confidence, interpretable subnetworks that best explain the data. these subnetworks connect changes observed in gene expression, protein abundance or other global assays to proteins that may not have been measured in the screens due to inherent bias or noise in measurement. this approach reveals unannotated molecular pathways that would not be detectable by searching pathway databases. omics integrator also provides an elegant framework to incorporate not only positive data, but also negative evidence. incorporating negative evidence allows omics integrator to avoid unexpressed genes and avoid being biased toward highly-studied hub proteins, except when they are strongly implicated by the data. the software is comprised of two individual tools, garnet and forest, that can be run together or independently to allow a user to perform advanced integration of multiple types of high-throughput data as well as create condition-specific subnetworks of protein interactions that best connect the observed changes in various datasets. it is available at http://fraenkel.mit.edu/omicsintegrator and on {github} at {https://github.com/fraenkel-lab/omicsintegrator}.morphogenesis of epithelial tissues relies on the precise developmental control of cell polarity and architecture. in the early drosophila embryo, the primary epithelium forms during cellularisation, following a tightly controlled genetic programme where specific sets of genes are upregulated. some of them, for example, control membrane invagination between the nuclei anchored at the apical surface of the syncytium. we used microarrays to describe the global programme of gene expression underlying cellularisation and identified distinct classes of upregulated genes during this process. fifty-seven genes were then tested functionally by {rnai}. we found six genes affecting various aspects of cellular architecture: membrane growth, organelle transport or organisation and junction assembly. we focus here on charleston (char), a new regulator of nuclear morphogenesis and of apical nuclear anchoring. in char-depleted embryos, the nuclei fail to maintain their elongated shape and, instead, become rounded. in addition, together with a disruption of the centrosome-nuclear envelope interaction, the nuclei lose their regular apical anchoring. these nuclear defects perturb the regular columnar organisation of epithelial cells in the embryo. although microtubules are required for both nuclear morphogenesis and anchoring, char does not control microtubule organisation and association to the nuclear envelope. we show that char is lipid anchored at the nuclear envelope by a farnesylation group, and localises at the inner nuclear membrane together with lamin. our data suggest that char forms a scaffold that regulates nuclear architecture to constrain nuclei in tight columnar epithelial cells. the upregulation of char during cellularisation and gastrulation reveals the existence of an as yet unknown developmental control of nuclear morphology and anchoring in embryonic epithelia.elucidating how complex regulatory networks have assembled during evolution requires a detailed understanding of the evolutionary dynamics that follow gene duplication events, including changes in post-translational modifications. we compared the phosphorylation profiles of paralogous proteins in the budding yeast saccharomyces cerevisiae to that of a species that diverged from the budding yeast before the duplication of those genes. we found that 100 million years of post-duplication divergence are sufficient for the majority of phosphorylation sites to be lost or gained in one paralog or the other, with a strong bias toward losses. however, some losses may be partly compensated for by the evolution of other phosphosites, as paralogous proteins tend to preserve similar numbers of phosphosites over time. we also found that up to 50\\\\% of kinase–substrate relationships may have been rewired during this period. our results suggest that after gene duplication, proteins tend to subfunctionalize at the level of post-translational regulation and that even when phosphosites are preserved, there is a turnover of the kinases that phosphorylate them.studies on various model systems have shown that a relatively small number of transcription factors can set up strikingly complex spatial and temporal patterns of gene expression. this is achieved mainly by means of combinatorial or differential gene regulation, i.e. regulation of a gene by two or more transcription factors simultaneously or under different conditions. while a number of specific molecular details of the mechanisms of combinatorial regulation have emerged, our understanding of the general principles of combinatorial regulation on a genomic scale is still limited. in this work, we approach this problem by using the largest assembled transcriptional regulatory network for yeast. a specific network transformation procedure was used to obtain the co-regulatory network describing the set of all significant associations among transcription factors in regulating common target genes. analysis of the global properties of the co-regulatory network suggested the presence of two classes of regulatory hubs: (i) those that make many co-regulatory associations, thus serving as integrators of disparate cellular processes; and (ii) those that make few co-regulatory associations, and thereby specifically regulate one or a few major cellular processes. investigation of the local structure of the co-regulatory network revealed a significantly higher than expected modular organization, which might have emerged as a result of selection by functional constraints. these constraints probably emerge from the need for extensive modular backup and the requirement to integrate transcriptional inputs of multiple distinct functional systems. we then explored the transcriptional control of three major regulatory systems (ubiquitin signaling, protein kinase and transcriptional regulation systems) to understand specific aspects of their upstream control. as a result, we observed that ubiquitin e3 ligases are regulated primarily by unique transcription factors, whereas e1 and e2 enzymes share common transcription factors to a much greater extent. this suggested that the deployment of e3s unique to specific functional contexts may be mediated significantly at the transcriptional level. likewise, we were able to uncover evidence for much higher upstream transcription control of transcription factors themselves, in comparison to components of other regulatory systems. we believe that the results presented here might provide a framework for testing the role of co-regulatory associations in eukaryotic transcriptional control.one of the outstanding challenges in comparative genomics is to interpret the evolutionary importance of regulatory variation between species. rigorous molecular evolution-based methods to infer evidence for natural selection from expression data are at a premium in the field, and to date, phylogenetic approaches have not been well-suited to address the question in the small sets of taxa profiled in standard surveys of gene expression. we have developed a strategy to infer evolutionary histories from expression profiles by analyzing suites of genes of common function. in a manner conceptually similar to molecular evolution models in which the evolutionary rates of {dna} sequence at multiple loci follow a gamma distribution, we modeled expression of the genes of an a priori-defined pathway with rates drawn from an inverse gamma distribution. we then developed a fitting strategy to infer the parameters of this distribution from expression measurements, and to identify gene groups whose expression patterns were consistent with evolutionary constraint or rapid evolution in particular species. simulations confirmed the power and accuracy of our inference method. as an experimental testbed for our approach, we generated and analyzed transcriptional profiles of four saccharomyces yeasts. the results revealed pathways with signatures of constrained and accelerated regulatory evolution in individual yeasts and across the phylogeny, highlighting the prevalence of pathway-level expression change during the divergence of yeast species. we anticipate that our pathway-based phylogenetic approach will be of broad utility in the search to understand the evolutionary relevance of regulatory change. comparative transcriptomic studies routinely identify thousands of genes differentially expressed between species. the central question in the field is whether and how such regulatory changes have been the product of natural selection. can the signal of evolutionarily relevant expression divergence be detected amid the noise of changes resulting from genetic drift? our work develops a theory of gene expression variation among a suite of genes that function together. we derive a formalism that relates empirical observations of expression of pathway genes in divergent species to the underlying strength of natural selection on expression output. we show that fitting this type of model to simulated data accurately recapitulates the parameters used to generate the simulation. we then make experimental measurements of gene expression in a panel of single-celled eukaryotic yeast species. to these data we apply our inference method, and identify pathways with striking evidence for accelerated or constrained regulatory evolution, in particular species and across the phylogeny. our method provides a key advance over previous approaches in that it maximizes the power of rigorous molecular-evolution analysis of regulatory variation even when data are relatively sparse. as such, the theory and tools we have developed will likely find broad application in the field of comparative genomics.cis-regulatory modules that drive precise spatial-temporal patterns of gene expression are central to the process of metazoan development. we describe a new computational strategy to annotate genomic sequences based on their  ” pattern generating potential” and to produce quantitative descriptions of transcriptional regulatory networks at the level of individual protein-module interactions. we use this approach to convert the qualitative understanding of interactions that regulate drosophila segmentation into a network model in which a confidence value is associated with each transcription factor-module interaction. sequence information from multiple drosophila species is integrated with transcription factor binding specificities to determine conserved binding site frequencies across the genome. these binding site profiles are combined with transcription factor expression information to create a model to predict module activity patterns. this model is used to scan genomic sequences for the potential to generate all or part of the expression pattern of a nearby gene, obtained from available gene expression databases. interactions between individual transcription factors and modules are inferred by a statistical method to quantify a factor\\'s contribution to the module\\'s pattern generating potential. we use these pattern generating potentials to systematically describe the location and function of known and novel cis-regulatory modules in the segmentation network, identifying many examples of modules predicted to have overlapping expression activities. surprisingly, conserved transcription factor binding site frequencies were as effective as experimental measurements of occupancy in predicting module expression patterns or factor-module interactions. thus, unlike previous module prediction methods, this method predicts not only the location of modules but also their spatial activity pattern and the factors that directly determine this pattern. as databases of transcription factor specificities and in vivo gene expression patterns grow, analysis of pattern generating potentials provides a general method to decode transcriptional regulatory sequences and networks. the developmental program specifying segmentation along the anterior-posterior axis of the drosophila embryo is one of the best studied examples of transcriptional regulatory networks. previous work has identified the location and function of dozens of {dna} segments called cis-regulatory  ” modules” that regulate several genes in precise spatial patterns in the early embryo. in many cases, transcription factors that interact with such modules have also been identified. we present a novel computational framework that turns a qualitative and fragmented understanding of modules and factor-module interactions into a quantitative, systems-level view. the formalism utilizes experimentally characterized binding specificities of transcription factors and gene expression patterns to describe how multiple transcription factors (working as activators or repressors) act together in a module to determine its regulatory activity. this formalism can explain the expression patterns of known modules, infer factor-module interactions and quantify the potential of an arbitrary {dna} segment to drive a gene\\'s expression. we have also employed databases of gene expression patterns to find novel modules of the regulatory network. as databases of binding motifs and gene expression patterns grow, this new approach provides a general method to decode transcriptional regulatory sequences and networks.background: a key goal of systems biology is to understand how genomewide {mrna} expression levels are controlled by transcription factors ({tfs}) in a condition-specific fashion. {tf} activity is frequently modulated at the post-translational level through ligand binding, covalent modification, or changes in sub-cellular localization. in this paper, we demonstrate how prior information about regulatory network connectivity can be exploited to infer condition-specific {tf} activity as a hidden variable from the genomewide {mrna} expression pattern in the yeast saccharomyces cerevisiae. {methodology/principal} findings: we first validate experimentally that by scoring differential expression at the level of gene sets or  ” regulons” comprised of the putative targets of a {tf}, we can accurately predict modulation of {tf} activity at the post-translational level. next, we create an interactive database of inferred activities for a large number of {tfs} across a large number of experimental conditions in s. cerevisiae. this allows us to perform {tf}-centric analysis of the yeast regulatory network. {conclusions/significance}: we analyze the degree to which the {mrna} expression level of each {tf} is predictive of its regulatory activity. we also organize {tfs} into  ” co-modulation networks” based on their inferred activity profile across conditions, and find that this reveals functional and mechanistic relationships. finally, we present evidence that the {pac} and {rrpe} motifs antagonize {tbp}-dependent regulation, and function as core promoter elements governed by the transcription regulator {nc2}. regulon-based monitoring of {tf} activity modulation is a powerful tool for analyzing regulatory network function that should be applicable in other organisms. tools and results are available online at {http://bussemakerlab.org/regulonprofiler}/.domestication of plants and animals promoted humanity\\'s transition from nomadic to sedentary lifestyles, demographic expansion, and the emergence of civilizations. in contrast to the well-documented successes of crop and livestock breeding, processes of microbe domestication remain obscure, despite the importance of microbes to the production of food, beverages, and biofuels. lager-beer, first brewed in the 15th century, employs an allotetraploid hybrid yeast, saccharomyces pastorianus (syn. saccharomyces carlsbergensis), a domesticated species created by the fusion of a saccharomyces cerevisiae ale-yeast with an unknown cryotolerant saccharomyces species. we report the isolation of that species and designate it saccharomyces eubayanus sp. nov. because of its resemblance to saccharomyces bayanus (a complex hybrid of s. eubayanus, saccharomyces uvarum, and s. cerevisiae found only in the brewing environment). individuals from populations of s. eubayanus and its sister species, s. uvarum, exist in apparent sympatry in nothofagus (southern beech) forests in patagonia, but are isolated genetically through intrinsic postzygotic barriers, and ecologically through host-preference. the draft genome sequence of s. eubayanus is 99.5\\\\% identical to the {non-s}. cerevisiae portion of the s. pastorianus genome sequence and suggests specific changes in sugar and sulfite metabolism that were crucial for domestication in the lager-brewing environment. this study shows that combining microbial ecology with comparative genomics facilitates the discovery and preservation of wild genetic stocks of domesticated microbes to trace their history, identify genetic changes, and suggest paths to further industrial improvement.heart development is exquisitely sensitive to the precise temporal regulation of thousands of genes that govern developmental decisions during differentiation. however, we currently lack a detailed understanding of how chromatin and gene expression patterns are coordinated during developmental transitions in the cardiac lineage. here, we interrogated the transcriptome and several histone modifications across the genome during defined stages of cardiac differentiation. we find distinct chromatin patterns that are coordinated with stage-specific expression of functionally related genes, including many human disease-associated genes. moreover, we discover a novel preactivation chromatin pattern at the promoters of genes associated with heart development and cardiac function. we further identify stage-specific distal enhancer elements and find enriched {dna} binding motifs within these regions that predict sets of transcription factors that orchestrate cardiac differentiation. together, these findings form a basis for understanding developmentally regulated chromatin transitions during lineage commitment and the molecular etiology of congenital heart disease. º chromatin signatures at promoters predict coregulated genes º identify novel preactivation chromatin pattern at cardiac genes º predict transcription factors that bind and activate cardiac specific enhancers º dynamic chromatin and expression landscape reveals new clues into heart development massively parallel {rna}-seq and {chip}-seq identify distinct chromatin patterns that coordinate gene expression during the differentiation of mouse embryonic stem cell into cardiomyocytes. stage-specific distal enhancer elements predict transcription factors orchestrating cardiac differentiation, while a new preactivation chromatin pattern marks promoters associated with heart development.motivation: current methods for reconstructing dynamic regulatory networks are focused on modeling a single response network using model organisms or cell lines. unlike these models or cell lines, humans differ in their background expression profiles due to age, genetics and life factors. in addition, there are often differences in start and end times for time series human data and in the rate of progress based on the specific individual. thus, new methods are required to integrate time series data from multiple individuals when modeling and constructing disease response networks.we develop fast algorithms for estimation of generalized linear models with convex penalties. the models include linear regression, two-class logistic regression, and multi- nomial regression problems while the penalties include ℓ<sub>1</sub> (the lasso), ℓ<sub>2</sub> (ridge regression) and mixtures of the two (the elastic net). the algorithms use cyclical coordinate descent, computed along a regularization path. the methods can handle large problems and can also deal efficiently with sparse features. in comparative timings we find that the new algorithms are considerably faster than competing methods.the genomic neighborhood of a gene influences its activity, a behavior that is attributable in part to domain-scale regulation, in which regions of hundreds or thousands of kilobases known as domains are regulated as a unit. previous studies using genomics assays such as chromatin immunoprecipitation ({chip})-seq and chromatin conformation capture ({3c})-based assays have identified many types of regulatory domains. however, due to the difficulty of integrating genomics data sets, the relationships among these domain types are poorly understood. semi-automated genome annotation ({saga}) algorithms facilitate human interpretation of heterogeneous collections of genomics data by simultaneously partitioning the human genome and assigning labels to the resulting genomic segments. however, existing {saga} methods can incorporate only data sets that can be expressed as a one-dimensional vector over the genome and therefore cannot integrate inherently pairwise chromatin conformation data. we developed a new computational method, called graph-based regularization ({gbr}), for expressing a pairwise prior that encourages certain pairs of genomic loci to receive the same label in a genome annotation. we used {gbr} to exploit chromatin conformation information during genome annotation by encouraging positions that are close in {3d} to occupy the same type of domain. using this approach, we produced a comprehensive model of chromatin domains in eight human cell types, thereby revealing the relationships among known domain types. through this model, we identified clusters of tightly-regulated genes expressed in only a small number of cell types, which we term \"specific expression domains.\" we additionally found that a subset of domain boundaries marked by promoters and {ctcf} motifs are consistent between cell types even when domain activity changes. finally, we showed that {gbr} can be used for the seemingly unrelated task of transferring information from well-studied cell types to less well characterized cell types during genome annotation, making it possible to produce high-quality annotations of the hundreds of cell types with limited available data.functional genomics has enormous potential to facilitate our understanding of normal and disease-specific physiology. in the past decade, intensive research efforts have been focused on modeling functional relationship networks, which summarize the probability of gene co-functionality relationships. such modeling can be based on either expression data only or heterogeneous data integration. numerous methods have been deployed to infer the functional relationship networks, while most of them target the global (non-context-specific) functional relationship networks. however, it is expected that functional relationships consistently reprogram under different tissues or biological processes. thus, advanced methods have been developed targeting tissue-specific or developmental stage-specific networks. this article brings together the state-of-the-art functional relationship network modeling methods, emphasizes the need for heterogeneous genomic data integration and context-specific network modeling and outlines future directions for functional relationship networks.many sensory systems (e.g., vision and hearing) show a response that is proportional to the fold-change in the stimulus relative to the background, a feature related to weber\\'s law. recent experiments suggest such a fold-change detection feature in signaling systems in cells: a response that depends on the fold-change in the input signal, and not on its absolute level. it is therefore of interest to find molecular mechanisms of gene regulation that can provide such fold-change detection. here, we demonstrate theoretically that fold-change detection can be generated by one of the most common network motifs in transcription networks, the incoherent feedforward loop ({i1-ffl}), in which an activator regulates both a gene and a repressor of the gene. the fold-change detection feature of the {i1-ffl} applies to the entire shape of the response, including its amplitude and duration, and is valid for a wide range of biochemical parameters.an abstract is not available.how does a cell respond to numerous external stresses with a limited number of internal molecular components? it has been observed that there are some common responses of yeast to various stresses, but most observations were based on gene-expression profiles and only some part of the common responses were intensively investigated. so far there has been no system-level analysis to identify commonly responsive or regulated genes against various stresses. in this study, we identified a core regulation module ({crm}), a commonly involved regulation structure in the regulatory networks of yeast, which cells reuse in response to an array of environmental stresses. we found that regulators in the {crm} constitute a hierarchical backbone of the yeast regulatory network and that the {crm} is evolutionarily well conserved, stable against genetic variations and crucial for cell growth. all these findings were consistently held up to considerable noise levels that we introduced to address experimental noise and the resulting false positives of regulatory interactions. we conclude that the {crm} of yeast might be an evolutionarily conserved information processing unit that endows a cell with enhanced robustness and efficiency in dealing with numerous environmental stresses with a limited number of internal elements.long range regulatory interactions among distal enhancers and target genes are important for tissue-specific gene expression. genome-scale identification of these interactions in a cell line-specific manner, especially using the fewest possible datasets, is a significant challenge. we develop a novel computational approach, regulatory interaction prediction for promoters and long-range enhancers ({ripple}), that integrates published chromosome conformation capture ({3c}) data sets with a minimal set of regulatory genomic data sets to predict enhancer-promoter interactions in a cell line-specific manner. our results suggest that {ctcf}, {rad21}, a general transcription factor ({tbp}) and activating chromatin marks are important determinants of enhancer-promoter interactions. to predict interactions in a new cell line and to generate genome-wide interaction maps, we develop an ensemble version of {ripple} and apply it to generate interactions in five human cell lines. computational validation of these predictions using existing {chia}-{pet} and {hi-c} data sets showed that {ripple} accurately predicts interactions among enhancers and promoters. enhancer-promoter interactions tend to be organized into subnetworks representing coordinately regulated sets of genes that are enriched for specific biological processes and cis-regulatory elements. overall, our work provides a systematic approach to predict and interpret enhancer-promoter interactions in a genome-wide cell-type specific manner using a few experimentally tractable measurements.motivation: recent large-scale chromatin states mapping efforts have revealed characteristic chromatin modification signatures for various types of functional {dna} elements. given the important influence of chromatin states on gene regulation and the rapid accumulation of genome-wide chromatin modification data, there is a pressing need for computational methods to analyze these data in order to identify functional {dna} elements. however, existing computational tools do not exploit data transformation and feature extraction as a means to achieve a more accurate {prediction.results}: we introduce a new computational framework for identifying functional {dna} elements using chromatin signatures. the framework consists of a data transformation and a feature extraction step followed by a classification step using time-delay neural network. we implemented our framework in a software tool {csi}-{ann} (chromatin signature identification by artificial neural network). when applied to predict transcriptional enhancers in the {encode} region, {csi}-{ann} achieved a 65.5\\\\% sensitivity and 66.3\\\\% positive predictive value, a 5.9\\\\% and 11.6\\\\% improvement, respectively, over the previously best {approach.availability} and implementation: {csi}-{ann} is implemented in matlab. the source code is freely available at {http://www.medicine.uiowa.edu/labs}/{tan/csiannsoft}.{zipcontact}: {kai-tan@uiowa.edusupplementary} information: supplementary materials are available at bioinformatics online.transcription factors and their binding sites have been proposed as primary targets of evolutionary adaptation because changes to single transcription factors can lead to far-reaching changes in gene expression patterns. nevertheless, there is very little concrete evidence for such evolutionary changes. industrial wine yeast strains, of the species saccharomyces cerevisiae, are a geno- and phenotypically diverse group of organisms that have adapted to the ecological niches of industrial winemaking environments and have been selected to produce specific styles of wine. variation in transcriptional regulation among wine yeast strains may be responsible for many of the observed differences and specific adaptations to different fermentative conditions in the context of commercial winemaking. we analyzed gene expression profiles of wine yeast strains to assess the impact of transcription factor expression on metabolic networks. the data provide new insights into the molecular basis of variations in gene expression in industrial strains and their consequent effects on metabolic networks important to wine fermentation. we show that the metabolic phenotype of a strain can be shifted in a relatively predictable manner by changing expression levels of individual transcription factors, opening opportunities to modify transcription networks to achieve desirable outcomes.{background}:the relationship between genetic variation in gene expression and phenotypic variation observable in nature is not well understood. identifying how many phenotypes are associated with differences in gene expression and how many gene-expression differences are associated with a phenotype is important to understanding the molecular basis and evolution of complex {traits.results}:we compared levels of gene expression among nine natural isolates of saccharomyces cerevisiae grown either in the presence or absence of copper sulfate. of the nine strains, two show a reduced growth rate and two others are rust colored in the presence of copper sulfate. we identified 633 genes that show significant differences in expression among strains. of these genes, 20 were correlated with resistance to copper sulfate and 24 were correlated with rust coloration. the function of these genes in combination with their expression pattern suggests the presence of both correlative and causative expression differences. but the majority of differentially expressed genes were not correlated with either phenotype and showed the same expression pattern both in the presence and absence of copper sulfate. to determine whether these expression differences may contribute to phenotypic variation under other environmental conditions, we examined one phenotype, freeze tolerance, predicted by the differential expression of the aquaporin gene {aqy2}. we found freeze tolerance is associated with the expression of {aqy2}.{conclusions}:gene expression differences provide substantial insight into the molecular basis of naturally occurring traits and can be used to predict environment dependent phenotypic variation.gene set analysis is the analysis of a set of genes that collectively contribute to a biological process. most popular gene set analysis methods are based on empirical p-value that requires large number of permutations. despite numerous gene set analysis methods developed in the past decade, the most popular methods still suffer from serious limitations. we present a gene set analysis method ({mgsz}) based on gene set z-scoring function ({gsz}) and asymptotic p-values. asymptotic p-value calculation requires fewer permutations, and thus speeds up the gene set analysis process. we compare the {gsz}-scoring function with seven popular gene set scoring functions and show that {gsz} stands out as the best scoring function. in addition, we show improved performance of the {gsa} method when the max-mean statistics is replaced by the {gsz} scoring function. we demonstrate the importance of both gene and sample permutations by showing the consequences in the absence of one or the other. a comparison of asymptotic and empirical methods of p-value estimation demonstrates a clear advantage of asymptotic p-value over empirical p-value. we show that {mgsz} outperforms the state-of-the-art methods based on two different evaluations. we compared {mgsz} results with permutation and rotation tests and show that rotation does not improve our asymptotic p-values. we also propose well-known asymptotic distribution models for three of the compared methods. {mgsz} is available as r package from cran.r-project.org. {\\\\copyright} the author 2014. published by oxford university press. all rights reserved. for permissions, please e-mail: journals.permissions@oup.com.summary: increasing number of {chip}-seq experiments are investigating transcription factor binding under multiple experimental conditions, for example, various treatment conditions, several distinct time points and different treatment dosage levels. hence, identifying differential binding sites across multiple conditions is of practical importance in biological and medical research. to this end, we have developed a powerful and flexible program, called {dbchip}, to detect differentially bound sharp binding sites across multiple conditions, with or without matching control samples. by assigning uncertainty measure to the putative differential binding sites, {dbchip} facilitates downstream analysis. {dbchip} is implemented in r programming language and can work with a wide range of sequencing file {formats.availability}: r package {dbchip} is available at {http://pages.cs.wisc.edu/\\\\~{}kliang/dbchip}/contact: {kliang@stat.wisc.edusupplementary} information: supplementary data are available at bioinformatics online.')]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Reduced by key before counting words for each user.\n",
    "grouped = user_abs.reduceByKey(lambda x,y: x + y)\n",
    "grouped.take(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('8123c9baf50aa777ed3211b5b6306bd0',\n",
       "  'meme is a tool for discovering motifs in sets of protein or dna sequences this paper describes several extensions to meme which increase its ability to find motifs in a totally unsupervised fashion but which also allow it to benefit when prior knowledge is available when no background knowledge is asserted meme obtains increased robustness from a method for determining motif widths automatically and from probabilistic models that allow motifs to be absent in some input sequences on the other hand meme can exploit prior knowledge about a motif being present in all input sequences about the length of a motif and whether it is a palindrome and using dirichlet mixtures about expected patterns in individual motif positions extensive experiments are reported which support the claim that meme benefits from but does not require background knowledge the experiments use seven previously studied dna and protein sequence families and 75 of the protein families documented in the prosite database of sites and patterns release 111systematic mapping of protein–protein interactions or interactome mapping was initiated in model organisms starting with defined biological processes1 2 and then expanding to the scale of the proteome3 4 5 6 7 although far from complete such maps have revealed global topological and dynamic features of interactome networks that relate to known biological properties8 9 suggesting that a human interactome map will provide insight into development and disease mechanisms at a systems level here we describe an initial version of a proteomescale map of human binary protein–protein interactions using a stringent highthroughput yeast twohybrid system we tested pairwise interactions among the products of 8100 currently available gatewaycloned open reading frames and detected 2800 interactions this data set called ccsbhi1 has a verification rate of 78 as revealed by an independent coaffinity purification assay and correlates significantly with other biological attributes the ccsbhi1 data set increases by 70 the set of available binary interactions within the tested space and reveals more than 300 new connections to over 100 diseaseassociated proteins this work represents an important step towards a systematic and comprehensive human interactome projectoligonucleotide and complementary dna microarrays are being used to subclassify histologically similar tumours monitor disease progress and individualize treatment regimens1 2 3 4 5 however extracting new biological insight from highthroughput genomic studies of human diseases is a challenge limited by difficulties in recognizing and evaluating relevant biological processes from huge quantities of experimental data here we present a structured network knowledgebase approach to analyse genomewide transcriptional responses in the context of known functional interrelationships among proteins small molecules and phenotypes this approach was used to analyse changes in blood leukocyte gene expression patterns in human subjects receiving an inflammatory stimulus bacterial endotoxin we explore the known genomewide interaction network to identify significant functional modules perturbed in response to this stimulus our analysis reveals that the human blood leukocyte response to acute systemic inflammation includes the transient dysregulation of leukocyte bioenergetics and modulation of translational machinery these findings provide insight into the regulation of global leukocyte activities as they relate to innate immune system tolerance and increased susceptibility to infection in humansprotein complexes are key molecular entities that integrate multiple gene products to perform cellular functions here we report the first genomewide screen for complexes in an organism budding yeast using affinity purification and mass spectrometry through systematic tagging of open reading frames orfs the majority of complexes were purified several times suggesting screen saturation the richness of the data set enabled a de novo characterization of the composition and organization of the cellular machinery the ensemble of cellular proteins partitions into 491 complexes of which 257 are novel that differentially combine with additional attachment proteins or protein modules to enable a diversification of potential functions support for this modular organization of the proteome comes from integration with available data on expression localization function evolutionary conservation protein structure and binary interactions this study provides the largest collection of physically determined eukaryotic cellular machines so far and a platform for biological data integration and modellingthe role of small rnas as key regulators of mrna turnover and translation has been well established recent advances indicate that the small rnas termed micrornas play important roles in animal development and physiology cellular activities such as proliferation morphogenesis apoptosis and differentiation are regulated by micrornas the expression of various genes are regulated by micrornas and several micrornas act in reciprocal negative feedback loops with protein factors to control cell fate decisions that are triggered by signal transduction activity these observations implicate small rnas as important mediators of gene regulation in response to cellcell signaling the mechanism by which micrornas silence gene expression is posttranscriptional possibly influencing the stability compartmentalization and translation of mrnas this mechanism is an efficient means to regulate production of a diverse range of proteinswe describe a systematic genomewide approach for learning the complex combinatorial code underlying gene expression our probabilistic approach identifies local dnasequence elements and the positional and combinatorial constraints that determine their contextdependent role in transcriptional regulation the inferred regulatory rules correctly predict expression patterns for 73 of genes in saccharomyces cerevisiae utilizing microarray expression data and sequences in the 800 bp upstream of genes application to caenorhabditis elegans identifies predictive regulatory elements and combinatorial rules that control the phased temporal expression of transcription factors histones and germline specific genes successful prediction requires diverse and complex rules utilizing and or and not logic with significant constraints on motif strength orientation and relative position this system generates a large number of mechanistic hypotheses for focused experimental validation and establishes a predictive dynamical framework for understanding cellular behavior from genomic sequencethe exploration of quantitative variation in human populations has become one of the major priorities for medical genetics the successful identification of variants that contribute to complex traits is highly dependent on reliable assays and genetic maps we have performed a genomewide quantitative trait analysis of 630 genes in 60 unrelated utah residents with ancestry from northern and western europe using the publicly available phase i data of the international hapmap project the genes are located in regions of the human genome with elevated functional annotation and disease interest including the encode regions spanning 1 of the genome chromosome 21 and chromosome 20q12–132 we apply three different methods of multiple test correction including bonferroni false discovery rate and permutations for the 374 expressed genes we find many regions with statistically significant association of single nucleotide polymorphisms snps with expression variation in lymphoblastoid cell lines after correcting for multiple tests based on our analyses the signal proximal cis to the genes of interest is more abundant and more stable than distal and trans across statistical methodologies our results suggest that regulatory polymorphism is widespread in the human genome and show that the 5kb phase i hapmap has sufficient density to enable linkage disequilibrium mapping in humans such studies will significantly enhance our ability to annotate the noncoding part of the genome and interpret functional variation in addition we demonstrate that the hapmap cell lines themselves may serve as a useful resource for quantitative measurements at the cellular level with the finished reference sequence of the human genome now available focus has shifted towards trying to identify all of the functional elements within the sequence although quite a lot of progress has been made towards identifying some classes of genomic elements in particular proteincoding sequences the characterization of regulatory elements remains a challenge the authors describe the genetic mapping of regions of the genome that have functional effects on quantitative levels of gene expression gene expression of 630 genes was measured in cell lines derived from 60 unrelated human individuals the same utah residents of northern and western european ancestry that have been genetically wellcharacterized by the international hapmap project this paper reports significant variation among individuals with respect to levels of gene expression and demonstrates that this quantitative trait has a genetic basis for some genes the genetic signal was localized to specific locations in the human genome sequence in most cases the genomic region associated with expression variation was physically close to the gene whose expression it regulated the authors demonstrate the feasibility of performing wholegenome association scans to map quantitative traits and highlight statistical issues that are increasingly important for wholegenome disease mapping studiesthe synthesis of gene expression data and cis regulatory analysis permits the elucidation of genomic regulatory networks these networks provide a direct visualization of the functional interconnections among the regulatory genes and signaling components leading to cellspecific patterns of gene activity complex developmental processes are thereby illuminated in ways not revealed by the conventional analysis of individual genes in this review we describe emerging networks in several different model systems and compare them with the gene regulatory network that controls dorsoventral patterning of the drosophila embryowe propose a modeldriven approach for analyzing genomic expression data that permits genetic regulatory networks to be represented in a biologically interpretable computational form our models permit latent variables capturing unobserved factors describe arbitrarily complex more than pairwise relationships at varying levels of refinement and can be scored rigorously against observational data the models that we use are based on bayesian networks and their extensions as a demonstration of this approach we utilize 52 genomes worth of affymetrix genechip expression data to correctly differentiate between alternative hypotheses of the galactose regulatory network in s cerevisiae when we extend the graph semantics to permit annotated edges we are able to score models describing relationships at a finer degree of specificationwe develop a new framework for inferring models of transcriptional regulation the models which we call physical network models are annotated molecular interaction graphs the attributes in the model correspond to verifiable properties of the underlying biological system such as the existence of proteinprotein and proteindna interactions the directionality of signal transduction in proteinprotein interactions as well as signs of the immediate effects of these interactions possible configurations of these variables are constrained by the available data sources some of the data sources such as factorbinding data involve measurements that are directly tied to the variables in the model other sources such as gene knockouts are functional in nature and provide only indirect evidence about the variables we associate each observed knockout effect in the deletion mutant data with a set of causal paths molecular cascades that could in principle explain the effect resulting in aggregate constraints about the physical variables in the model the most likely settings of all the variables specifying the most likely graph annotations are found by a recursive application of the maxproduct algorithm by testing our approach on datasets related to the pheromone response pathway in s cerevisiae we demonstrate that the resulting model is consistent with previous studies about the pathway moreover we successfully predict gene knockout effects with a high degree of accuracy in a crossvalidation setting when applying this approach genomewide we extract submodels consistent with previous studies the approach can be readily extended to other data sources or to facilitate automated experimental designas genomescale measurements lead to increasingly complex models of gene regulation systematic approaches are needed to validate and refine these models towards this goal we describe an automated procedure for prioritizing genetic perturbations in order to discriminate optimally between alternative models of a generegulatory network using this procedure we evaluate 38 candidate regulatory networks in yeast and perform four highpriority gene knockout experiments the refined networks support previously unknown regulatory mechanisms downstream of sok2 and swi4background the study of complex biological networks and prediction of gene function has been enabled by highthroughput htp methods for detection of genetic and protein interactions sparse coverage in htp datasets may however distort network properties and confound predictions although a vast number of well substantiated interactions are recorded in the scientific literature these data have not yet been distilled into networks that enable systemlevel inference results we describe here a comprehensive database of genetic and protein interactions and associated experimental evidence for the budding yeast saccharomyces cerevisiae as manually curated from over 31793 abstracts and online publications this literaturecurated lc dataset contains 33311 interactions on the order of all extant htp datasets combined surprisingly htp proteininteraction datasets currently achieve only around 14 coverage of the interactions in the literature the lc network nevertheless shares attributes with htp networks including scalefree connectivity and correlations between interactions abundance localization and expression we find that essential genes or proteins are enriched for interactions with other essential genes or proteins suggesting that the global network may be functionally unified this interconnectivity is supported by a substantial overlap of protein and genetic interactions in the lc dataset we show that the lc dataset considerably improves the predictive power of networkanalysis approaches the full lc dataset is available at the biogrid httpwwwthebiogridorg and sgd httpwwwyeastgenomeorg databases conclusion comprehensive datasets of biological interactions derived from the primary literature provide critical benchmarks for htp methods augment functional prediction and reveal systemlevel attributes of biological networksin this paper we describe an approach for identifying pathways from gene expression and protein interaction data our approach is based on the assumption that many pathways exhibit two properties their genes exhibit a similar gene expression profile and the protein products of the genes often interact  our approach is based on a unified probabilistic model which is learned from the data using the em algorithm we present results on two saccharomyces cerevisiae gene expression data sets combined with a binary protein interaction data set our results show that our approach is much more successful than other approaches at discovering both coherent functional groups and entire protein complexescontact erancsstanfordedukeywords probabilistic models protein interaction gene expressionwe propose a statistical method to estimate gene networks from dna microarray data and proteinprotein interactions because physical interactions between proteins or multiprotein complexes are likely to regulate biological processes using only mrna expression data is not sufficient for estimating a gene network accurately our method adds knowledge about proteinprotein interactions to the estimation method of gene networks under a bayesian statistical framework in the estimated gene network a protein complex is modeled as a virtual node based on principal component analysis we show the effectiveness of the proposed method through the analysis of saccharomyces cerevisiae cell cycle data the proposed method improves the accuracy of the estimated gene networks and successfully identifies some biological factsthe prediction of regulatory elements is a problem where computational methods offer great hope over the past few years numerous tools have become available for this task the purpose of the current assessment is twofold to provide some guidance to users regarding the accuracy of currently available tools in various settings and to provide a benchmark of data sets for assessing future toolswe develop principled methods for the automatic induction discovery of genetic regulatory network models from multiple data sources and data modalities models of regulatory networks are represented as bayesian networks allowing the models to compactly and robustly capture probabilistic multivariate statistical dependencies between the various cellular factors in these networks we build on previous bayesian network validation results by extending the validation framework to the context of model induction leveraging heuristic simulated annealing search algorithms and posterior model averaging using expression data in isolation yields results inconsistent with location data so we incorporate genomic location data to guide the model induction process we combine these two data modalities by allowing location data to influence the model prior and expression data to influence the model likelihood we demonstrate the utility of this approach by discovering genetic regulatory models of thirtythree variables involved in s cerevisiae pheromone response the models we automatically generate are consistent with the current understanding regarding this regulatory network but also suggest new directions for future experimental investigationciona is an emerging model system for elucidating gene networks in development comprehensive in situ hybridization assays have identified 76 regulatory genes with localized expression patterns in the early embryo at the time when naive blastomeres are determined to follow specific cell fates systematic gene disruption assays provided more than 3000 combinations of gene expression profiles in mutant backgrounds deduced gene circuit diagrams describing the formation of larval tissues were computationally visualized these diagrams constitute a blueprint for the ciona embryo and provide a foundation for understanding the evolutionary origins of the chordate body plan 101126science1123404genomewide transcript profiling was used to monitor signal transduction during yeast pheromone response genetic manipulations allowed analysis of changes in gene expression underlying pheromone signaling cell cycle control and polarized morphogenesis a twodimensional hierarchical clustered matrix covering 383 of the most highly regulated genes was constructed from 46 diverse experimental conditions diagnostic subsets of coexpressed genes reflected signaling activity cross talk and overlap of multiple mitogenactivated protein kinase mapk pathways analysis of the profiles specified by two different mapksfus3p and kss1prevealed functional overlap of the filamentous growth and mating responses global transcript analysis reflects biological responses associated with the activation and perturbation of signal transduction pathwayswe present a probabilistic framework that models the process by which transcriptional binding explains the mrna expression of different genes our joint probabilistic model unifies the two key components of this process the prediction of gene regulation events from sequence motifs in the genes promoter region and the prediction of mrna expression from combinations of gene regulation events in different settings our approach has several advantages by learning promoter sequence motifs thatwe propose a probabilistic model for cellular processes and an algorithm for discovering them from gene expression data a process is associated with a set of genes that participate in it unlike clustering techniques our model allows genes to participate in multiple processes each process may be active to a different degree in each experiment the expression measurement for gene g in array a is a sum over all processes in which g participates of the activity levels of these processes in array a we describe an iterative procedure based on the em algorithm for decomposing the expression matrix into a given number of processes we present results on yeast gene expression data which indicate that our approach identifies real biological processesmotivation in model organisms such as yeast large databases of protein–protein and proteindna interactions have become an extremely important resource for the study of protein function evolution and gene regulatory dynamics in this paper we demonstrate that by integrating these interactions with widelyavailable mrna expression data it is possible to generate concrete hypotheses for the underlying mechanisms governing the observed changes in gene expression to perform this integration systematically and at large scale we introduce an approach for screening a molecular interaction network to identify active subnetworks ie connected regions of the network that show significant changes in expression over particular subsets of conditions the method we present here combines a rigorous statistical measure for scoring subnetworks with a search algorithm for identifying subnetworks with high scorethe modeling of cellular signaling pathways is an emerging field sachs et al illustrate the application of bayesian networks to an example cellular pathway involving the activation of focal adhesion kinase fak and extracellular signalregulated kinase erk in response to fibronectin binding to an integrin they describe how to use the analysis to select from among proposed models formulate hypotheses regarding component interactions and uncover potential dynamic changes in the interactions between these components although the data sets currently available for this example problem are too small to definitively point to a particular model the approach and results provide a glimpse into the power that these methods will achieve once the technology for obtaining the necessary data becomes readily availablemotivation biological processes in cells are properly performed by gene regulations signal transductions and interactions between proteins to understand such molecular networks we propose a statistical method to estimate gene regulatory networks and proteinprotein interaction networks simultaneously from dna microarray data proteinprotein interaction data and other genomewide data results we unify bayesian networks and markov networks for estimating gene regulatory networks and proteinprotein interaction networks according to the reliability of each biological information source through the simultaneous construction of gene regulatory networks and proteinprotein interaction networks of saccharomyces cerevisiae cell cycle we predict the role of several genes whose functions are currently unknown by using our probabilistic model we can detect false positives of highthroughput data such as yeast twohybrid data in a genomewide experiment we find possible gene regulatory relationships and proteinprotein interactions between large protein complexes that underlie complex regulatory mechanisms of biological processes contact nariaiimsutokyoacjpgenetic loci that regulate inherited traits are routinely identified using quantitative trait locus qtl mapping methods however the genotypephenotype associations do not provide information on the gene expression program through which the genetic loci regulate the traits transcription modules are selfconsistent regulatory units and are closely related to the modular components of gene regulatory network ihmels j friedlander g bergmann s sarig o ziv y and barkai n 2002 revealing modular organization in the yeast transcriptional network nat genet 31 370377 segal e shapira m regev a peer d botstein d koller d and friedman n 2003 module networks identifying regulatory modules and their conditionspecific regulators from gene expression data nat genet 34 166176 we used genomewide genotype and gene expression data of a genetic reference population that consists of mice of 32 recombinant inbred strains to identify the transcription modules and the genetic loci regulating them twentynine transcription modules defined by genetic variations were identified statistically significant associations between the transcription modules and 18 classical physiological and behavioral traits were found genomewide interval mapping showed that major qtls regulating the transcription modules are often colocalized with the qtls regulating the associated classical traits the association and the possible coregulation of the classical trait and transcription module indicate that the transcription module may be involved in the gene pathways connecting the qtl and the classical trait our results show that a transcription module may associate with multiple seemingly unrelated classical traits and a classical trait may associate with different modules literature mining results provided strong independent evidences for the relations among genes of the transcription modules genes in the regions of the qtls regulating the transcription modules and the keywords representing the classical traitsa central problem in the bioinformatics of gene regulation is to find the binding sites for regulatory proteins one of the most promising approaches toward identifying these short and fuzzy sequence patterns is the comparative analysis of orthologous intergenic regions of related species this analysis is complicated by various factors first one needs to take the phylogenetic relationship between the species into account in order to distinguish conservation that is due to the occurrence of functional sites from spurious conservation that is due to evolutionary proximity second one has to deal with the complexities of multiple alignments of orthologous intergenic regions and one has to consider the possibility that functional sites may occur outside of conserved segments here we present a new motif sampling algorithm phylogibbs that runs on arbitrary collections of multiple local sequence alignments of orthologous sequences the algorithm searches over all ways in which an arbitrary number of binding sites for an arbitrary number of transcription factors tfs can be assigned to the multiple sequence alignments these binding site configurations are scored by a bayesian probabilistic model that treats aligned sequences by a model for the evolution of binding sites and  ” background” intergenic dna this model takes the phylogenetic relationship between the species in the alignment explicitly into account the algorithm uses simulated annealing and monte carlo markovchain sampling to rigorously assign posterior probabilities to all the binding sites that it reports in tests on synthetic data and real data from five saccharomyces species our algorithm performs significantly better than four other motiffinding algorithms including algorithms that also take phylogeny into account our results also show that in contrast to the other algorithms phylogibbs can make realistic estimates of the reliability of its predictions our tests suggest that running on the fivespecies multiple alignment of a single genes upstream region phylogibbs on average recovers over 50 of all binding sites in s cerevisiae at a specificity of about 50 and 33 of all binding sites at a specificity of about 85 we also tested phylogibbs on collections of multiple alignments of intergenic regions that were recently annotated based on chiponchip data to contain binding sites for the same tf we compared phylogibbss results with the previous analysis of these data using six other motiffinding algorithms for 16 of 21 tfs for which all other motiffinding methods failed to find a significant motif phylogibbs did recover a motif that matches the literature consensus in 11 cases where there was disagreement in the results we compiled lists of known target genes from the literature and found that running phylogibbs on their regulatory regions yielded a binding motif matching the literature consensus in all but one of the cases interestingly these literature gene lists had little overlap with the targets annotated based on the chiponchip data the phylogibbs code can be downloaded from httpwwwbiozentrumunibaschnimwege\\u200bncgibinphylogibbscgi or httpwwwimscresinrsiddphylogibbs the full set of predicted sites from our tests on yeast are available at httpwwwswissregulonunibasch computational discovery of regulatory sites in intergenic dna is one of the central problems in bioinformatics up until recently motif finders would typically take one of the following two general approaches given a known set of coregulated genes one searches their promoter regions for significantly overrepresented sequence motifs alternatively in a  ” phylogenetic footprinting” approach one searches multiple alignments of orthologous intergenic regions for short segments that are significantly more conserved than expected based on the phylogeny of the species in this work the authors present an algorithm phylogibbs that combines these two approaches into one integrated bayesian framework the algorithm searches over all ways in which an arbitrary number of binding sites for an arbitrary number of transcription factors can be assigned to arbitrary collections of multiple sequence alignments while taking into account the phylogenetic relations between the sequences the authors perform a number of tests on synthetic data and real data from saccharomyces genomes in which phylogibbs significantly outperforms other existing methods finally a novel annealandtrack strategy allows phylogibbs to make accurate estimates of the reliability of its predictionsmachine learning was applied for the automated derivation of causal influences in cellular signaling networks this derivation relied on the simultaneous measurement of multiple phosphorylated protein and phospholipid components in thousands of individual primary human immune system cells perturbing these cells with molecular interventions drove the ordering of connections between pathway components wherein bayesian network computational methods automatically elucidated most of the traditionally reported signaling relationships and predicted novel interpathway network causalities which we verified experimentally reconstruction of network models from physiologically relevant primary single cells might be applied to understanding nativestate tissue signaling biology complex drug actions and dysfunctional signaling in diseased cellsmotivation micrornas mirnas are small endogenous rnas that can play important regulatory roles via the rnainterference pathway by targeting mrnas for cleavage or translational repression we propose a computational method to predict mirna regulatory modules mrms or groups of mirnas and target genes that are believed to participate cooperatively in posttranscriptional gene regulation results we tested our method with the human genes and mirnas predicting 431 mrms we analyze a module with genes btg2 wt1 ppm1d pak7 and rab9b and mirnas mir15a and mir16 review of the literature and annotation with gene ontology terms reveal that the roles of these genes can indeed be closely related in specific biological processes such as gene regulation involved in breast renal and prostate cancers furthermore it has been reported that mir15a and mir16 are deleted together in certain types of cancer suggesting a possible connection between these mirnas and cancers given that most known functionalities of mirnas are related to negative gene regulation extending our approach and exploiting the insight thus obtained may provide clues to achieving practical accuracy in the reverseengineering of gene regulatory networks availability a list of predicted modules is available from the authors upon requestthousands of mammalian messenger rnas are under selective pressure to maintain 7nucleotide sites matching micrornas mirnas we found that these conserved targets are often highly expressed at developmental stages before mirna expression and that their levels tend to fall as the mirna that targets them begins to accumulate nonconserved sites which outnumber the conserved sites 10 to 1 also mediate repression as a consequence genes preferentially expressed at the same time and place as a mirna have evolved to selectively avoid sites matching the mirna this phenomenon of selective avoidance extends to thousands of genes and enables spatial and temporal specificities of mirnas to be revealed by finding tissues and developmental stages in which messages with corresponding sites are expressed at lower levelsgene expression variation has been the focus of many studies in the past few years the relevance of gene regulation and gene expression to disease and the development of the technologies used to screen large numbers of genes simultaneously have allowed this rapid development in this review we discuss issues relating to the biological information one obtains from such studies and the biological significance and use of signals from mapping of gene expression variation')]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Remove punctuations to make the text clear and can be split on ' '\n",
    "def lower_clean_str(x):\n",
    "  punc='!\"#$%&\\'()*+,./:;<=>?@[\\\\]^_`{|}~-'\n",
    "  lowercased_str = x[1].lower()\n",
    "  for ch in punc:\n",
    "    lowercased_str = lowercased_str.replace(ch, '')\n",
    "  return x[0], lowercased_str\n",
    "\n",
    "punc = grouped.map(lower_clean_str)\n",
    "\n",
    "punc.take(1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Creating Combiner, to use it as combineByKey\n",
    "createCombiner = (lambda el: [el]) \n",
    "mergeVal = (lambda aggregated, el: aggregated + [el]) # append to aggregated\n",
    "mergeComb = (lambda agg1,agg2: agg1 + agg2 )  # append agg1 with agg2\n",
    "\n",
    "# Counting words and Combining then using key.\n",
    "new = punc.map(lambda x: [((x[0], a), 1) for a in x[1].split(' ')]).flatMap(lambda l : l).reduceByKey(add)\\\n",
    "        .filter(lambda x: x[0][1] not in broadcastVar.value).sortBy(lambda x: x[1], False)\\\n",
    "        .map(lambda x: (x[0][0], x[0][1])).combineByKey(cComb,mVal,mComb)\n",
    "rdd_new = new.map(lambda x: (x[0], x[1][0:10]))\n",
    "rdd_new.take(2)\n",
    "print(\"Time: \", end-start )\n",
    "\n",
    "# 2750.50794 secs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Writing RDD to file\n",
    "rdd_new.coalesce(1).saveAsTextFile(\"Result\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exercise 1.4: Basic Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28416\n",
      "172079\n",
      "828481\n"
     ]
    }
   ],
   "source": [
    "# No. of Distinct Users\n",
    "d_users = userRatingsRDD.countByKey()\n",
    "total_users = sum(y for x, y in d_users.items())\n",
    "print(total_users)\n",
    "\n",
    "#28416\n",
    "\n",
    "# No. of Distinct Items\n",
    "d_items = paperTermsRDD.countByKey()\n",
    "total_items = sum(y for x, y in d_items.items())\n",
    "print(total_items)\n",
    "\n",
    "#172079\n",
    "\n",
    "# No. of Ratings\n",
    "ratings = exploded.countByValue()\n",
    "total_ratings = sum(y for x,y in ratings.items())\n",
    "print(total_ratings)\n",
    "\n",
    "#828481"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimum:  1\n",
      "Maximum:  1922\n"
     ]
    }
   ],
   "source": [
    "# Minimum Rating Given by a user: userhash ratings.\n",
    "ratings = exploded.countByKey()\n",
    "ratings = ratings.values()\n",
    "\n",
    "mini = min(ratings)\n",
    "print(\"Minimum: \", mini)\n",
    "\n",
    "#Minimum: 1\n",
    "\n",
    "maxx = max(ratings)\n",
    "print(\"Maximum: \", maxx)\n",
    "#Maximum: 1922"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Average:  29.155440596846848\n"
     ]
    }
   ],
   "source": [
    "# Average number ratings of users: For it we will use total ratings/distinct_users we calculated earlier.\n",
    "avg_rating = total_ratings/total_users\n",
    "print(\"Average: \", avg_rating)\n",
    "#Average:  29.155440596846848\n",
    "\n",
    "#Standard Deviation:\n",
    "#stdev"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimum:  3\n",
      "Maximum:  924\n",
      "Average:  4.81453867119172\n"
     ]
    }
   ],
   "source": [
    "# Minimum number of Ratings an item has received.\n",
    "# CountByValue giving IOPub data rate exceeded Error.\n",
    "# Therefore using reversed_ rdd and countByKey.\n",
    "ratings_ = reversed_.countByKey()\n",
    "ratings_ = ratings_.values()\n",
    "\n",
    "min_rat = min(ratings_)\n",
    "print(\"Minimum: \", min_rat)\n",
    "\n",
    "#Minimum: 3\n",
    "max_rat = max(ratings_)\n",
    "print(\"Maximum: \", max_rat)\n",
    "\n",
    "#Maximum: 924\n",
    "\n",
    "#Average no. of ratings of Items, for This we will use total_ratings/total_items\n",
    "avg_rat_item = total_ratings/total_items\n",
    "print(\"Average: \", avg_rat_item)\n",
    "\n",
    "#Average: 4.81453867119172"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exercise 1.5 DataFrames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "root\n",
      " |-- User: string (nullable = true)\n",
      " |-- Papers_ID: string (nullable = true)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Using two columns as StringType (Later will assign second column (Papers_ID) to array.)\n",
    "schema = StructType([\n",
    "    StructField(\"User\", StringType(), False),\n",
    "    StructField(\"Papers_ID\", StringType(), True)\n",
    "])\n",
    "\n",
    "# Creating DataFrame df for users_libraries\n",
    "userDF = sqlC.read.schema(schema)\\\n",
    "        .option(\"header\", 'False').option(\"delimiter\", \";\").csv(\"users_libraries.txt\")\n",
    "\n",
    "\n",
    "#userDF.show(10, False)\n",
    "userDF.printSchema()\n",
    "#userDF.show(20, False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------+-------+--------------------+----------+------+---------+-----+------+------+----+-----+-------------------+-------+--------------------+--------------------+\n",
      "|paper_id|   type|             journal|bookـtitle|series|publisher|pages|volume|number|year|month|           postedat|address|               title|            abstract|\n",
      "+--------+-------+--------------------+----------+------+---------+-----+------+------+----+-----+-------------------+-------+--------------------+--------------------+\n",
      "|   80546|article|biology and philo...|      null|  null|     null|   17|    19|     2|2004|  mar|2005-01-26 21:35:21|   null|the arbitrariness...|the genetic code ...|\n",
      "| 5842862|article|      molecular cell|      null|  null| elsevier|    2|    35|     6|2009|  sep|2009-09-30 17:11:23|   null|how to choose a g...|choosing good pro...|\n",
      "+--------+-------+--------------------+----------+------+---------+-----+------+------+----+-----+-------------------+-------+--------------------+--------------------+\n",
      "only showing top 2 rows\n",
      "\n",
      "root\n",
      " |-- paper_id: string (nullable = true)\n",
      " |-- type: string (nullable = true)\n",
      " |-- journal: string (nullable = true)\n",
      " |-- bookـtitle: string (nullable = true)\n",
      " |-- series: string (nullable = true)\n",
      " |-- publisher: string (nullable = true)\n",
      " |-- pages: string (nullable = true)\n",
      " |-- volume: string (nullable = true)\n",
      " |-- number: string (nullable = true)\n",
      " |-- year: string (nullable = true)\n",
      " |-- month: string (nullable = true)\n",
      " |-- postedat: string (nullable = true)\n",
      " |-- address: string (nullable = true)\n",
      " |-- title: string (nullable = true)\n",
      " |-- abstract: string (nullable = true)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Creating Papers.csv DataFrame\n",
    "# I didnt create schema for this DF, because at the end i needed two columns and wanted to change\n",
    "# type of only paper_id column to IntegerType()\n",
    "paperDF = sqlC.read.csv(\"papers.csv\")\n",
    "columns= [\"paper_id\", \"type\", \"journal\", \"bookـtitle\", \"series\", \"publisher\", \"pages\", \"volume\", \"number\", \"year\", \"month\", \"postedat\",\"address\",\"title\", \"abstract\"]\n",
    "oldCols = paperDF.schema.names\n",
    "for i, x in enumerate(oldCols):\n",
    "    paperDF = paperDF.withColumnRenamed(x, columns[i])\n",
    "paperDF.show(2)\n",
    "paperDF.printSchema()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exercise 1.3 with DataFrames ( Joining Collections )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+\n",
      "|User                            |PapersID                                                                                                                                                                                                                                                                                                                                                                                                       |\n",
      "+--------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+\n",
      "|28d3f81251d94b09735497477a5e4e02|[3929762, 503574, 5819422, 4238883, 5788061, 462949, 635215, 635216, 4810441, 3481823, 4165233, 3366480, 5984302, 4238942, 5490453, 4636156, 5996865, 4194836, 5828780, 4450195]                                                                                                                                                                                                                               |\n",
      "|d0c9aaa788153daeaf1f1538b3d46bbb|[2080631, 6343346, 5184704, 7756088, 2653863, 6607628, 4236212, 1277953, 226864, 3140015, 8806369, 311570, 5687747, 767516, 4781370, 2841637, 2445106, 1959511, 2688186, 2363430, 6614346, 853030, 5336762, 4226226, 239571, 4089758, 4140337, 913868, 7562861, 3190274, 2782576, 12571584, 2049617, 5761055, 5441098, 3466838, 2080691, 1805577, 7570111, 5760287, 2855355, 3281547, 1012525, 3512183, 678653]|\n",
      "+--------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+\n",
      "only showing top 2 rows\n",
      "\n",
      "root\n",
      " |-- User: string (nullable = true)\n",
      " |-- PapersID: array (nullable = true)\n",
      " |    |-- element: string (containsNull = true)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Type casting from str to array so i can apply explode function on papers_id column.\n",
    "def str_to_arr(my_list):\n",
    "    my_list = my_list.split(\",\")\n",
    "    return [x for x in my_list]\n",
    "\n",
    "str_to_arr_udf = udf(str_to_arr,ArrayType(StringType()))\n",
    "\n",
    "userDF = userDF.withColumn('PapersID',str_to_arr_udf(userDF[\"Papers_ID\"]))\n",
    "userDF = userDF.drop(\"Papers_ID\")\n",
    "userDF.show(2, False)\n",
    "userDF.printSchema()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------------------------+--------+\n",
      "|User                            |Paper_ID|\n",
      "+--------------------------------+--------+\n",
      "|28d3f81251d94b09735497477a5e4e02|3929762 |\n",
      "|28d3f81251d94b09735497477a5e4e02|503574  |\n",
      "+--------------------------------+--------+\n",
      "only showing top 2 rows\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Checking if explode function is working properly\n",
    "userDF2 = userDF.select(userDF.User, sf.explode(userDF.PapersID))\n",
    "userDF2 = userDF2.withColumn(\"Paper_ID\", userDF2[\"col\"])\n",
    "userDF2 = userDF2.drop(\"col\")\n",
    "userDF2.show(2, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "root\n",
      " |-- paper_id: integer (nullable = true)\n",
      " |-- type: string (nullable = true)\n",
      " |-- journal: string (nullable = true)\n",
      " |-- bookـtitle: string (nullable = true)\n",
      " |-- series: string (nullable = true)\n",
      " |-- publisher: string (nullable = true)\n",
      " |-- pages: string (nullable = true)\n",
      " |-- volume: string (nullable = true)\n",
      " |-- number: string (nullable = true)\n",
      " |-- year: string (nullable = true)\n",
      " |-- month: string (nullable = true)\n",
      " |-- postedat: string (nullable = true)\n",
      " |-- address: string (nullable = true)\n",
      " |-- title: string (nullable = true)\n",
      " |-- abstract: string (nullable = true)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Changing datatype of paper_id to IntegerType()\n",
    "paperDF = paperDF.withColumn(\"paper_id\", col(\"paper_id\").cast(IntegerType()))\n",
    "paperDF.printSchema()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------+--------------------+\n",
      "|paper_id|            abstract|\n",
      "+--------+--------------------+\n",
      "|   80546|the genetic code ...|\n",
      "| 5842862|choosing good pro...|\n",
      "+--------+--------------------+\n",
      "only showing top 2 rows\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Taking subset of Dataset before joining with userDF.\n",
    "papersDF = paperDF[\"paper_id\", \"abstract\"]\n",
    "papersDF.show(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------------------------+--------+--------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+\n",
      "|User                            |Paper_ID|paper_id|abstract                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |\n",
      "+--------------------------------+--------+--------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+\n",
      "|54daa5d96cafa229a801ba8dd7a24e57|80546   |80546   |the genetic code has been regarded as arbitrary in the sense that the codon-amino acid assignments could be different than they actually are. this general idea has been spelled out differently by previous, often rather implicit accounts of arbitrariness. they have drawn on the frozen accident theory, on evolutionary contingency, on alternative causal pathways, and on the absence of direct stereochemical interactions between codons and amino acids. it has also been suggested that the arbitrariness of the genetic code justifies attributing semantic information to macromolecules, notably to {dna}. i argue that these accounts of arbitrariness are unsatisfactory. i propose that the code is arbitrary in the sense of jacques monod's concept of chemical arbitrariness: the genetic code is arbitrary in that any codon requires certain chemical and structural properties to specify a particular amino acid, but these properties are not required in virtue of a principle of chemistry. this notion of arbitrariness is compatible with several recent hypotheses about code evolution. i maintain that the code's chemical arbitrariness is neither sufficient nor necessary for attributing semantic information to nucleic acids.|\n",
      "|14b532ec70ec6d5ad633792b9c800919|80546   |80546   |the genetic code has been regarded as arbitrary in the sense that the codon-amino acid assignments could be different than they actually are. this general idea has been spelled out differently by previous, often rather implicit accounts of arbitrariness. they have drawn on the frozen accident theory, on evolutionary contingency, on alternative causal pathways, and on the absence of direct stereochemical interactions between codons and amino acids. it has also been suggested that the arbitrariness of the genetic code justifies attributing semantic information to macromolecules, notably to {dna}. i argue that these accounts of arbitrariness are unsatisfactory. i propose that the code is arbitrary in the sense of jacques monod's concept of chemical arbitrariness: the genetic code is arbitrary in that any codon requires certain chemical and structural properties to specify a particular amino acid, but these properties are not required in virtue of a principle of chemistry. this notion of arbitrariness is compatible with several recent hypotheses about code evolution. i maintain that the code's chemical arbitrariness is neither sufficient nor necessary for attributing semantic information to nucleic acids.|\n",
      "+--------------------------------+--------+--------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+\n",
      "only showing top 2 rows\n",
      "\n"
     ]
    }
   ],
   "source": [
    "joined = userDF2.join(papersDF, [papersDF.paper_id == userDF2.Paper_ID], \"inner\")\n",
    "joined.show(2, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+\n",
      "|User                            |abstract                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |\n",
      "+--------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+\n",
      "|54daa5d96cafa229a801ba8dd7a24e57|the genetic code has been regarded as arbitrary in the sense that the codon-amino acid assignments could be different than they actually are. this general idea has been spelled out differently by previous, often rather implicit accounts of arbitrariness. they have drawn on the frozen accident theory, on evolutionary contingency, on alternative causal pathways, and on the absence of direct stereochemical interactions between codons and amino acids. it has also been suggested that the arbitrariness of the genetic code justifies attributing semantic information to macromolecules, notably to {dna}. i argue that these accounts of arbitrariness are unsatisfactory. i propose that the code is arbitrary in the sense of jacques monod's concept of chemical arbitrariness: the genetic code is arbitrary in that any codon requires certain chemical and structural properties to specify a particular amino acid, but these properties are not required in virtue of a principle of chemistry. this notion of arbitrariness is compatible with several recent hypotheses about code evolution. i maintain that the code's chemical arbitrariness is neither sufficient nor necessary for attributing semantic information to nucleic acids.|\n",
      "+--------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+\n",
      "only showing top 1 row\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Dropping PaperID columns as we just need to count words of a user.\n",
    "user_abs = joined.drop(\"paper_id\", \"Paper_ID\")\n",
    "user_abs.show(1, False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exercise 1.4 with DataFrames ( Basic Analysis )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Distinct Users: 28416\n",
      "Distinct Items:  172079\n",
      "Total Ratings:  828481\n"
     ]
    }
   ],
   "source": [
    "#Number of Distinct Users.\n",
    "dist_Users = user_abs.select(user_abs.User).distinct().count()\n",
    "print(\"Distinct Users:\",dist_Users)\n",
    "\n",
    "#Distinct Users: 28416\n",
    "\n",
    "dist_items = papersDF.select(papersDF.paper_id).distinct().count()\n",
    "print(\"Distinct Items: \",dist_items)\n",
    "\n",
    "# Distinct Items: 172079\n",
    "\n",
    "#Total Ratings:\n",
    "total_ratings = user_abs.select(user_abs.abstract).count()\n",
    "print(\"Total Ratings: \", total_ratings)\n",
    "\n",
    "#Total Ratings: 828481"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+----------+\n",
      "|min(Count)|\n",
      "+----------+\n",
      "|         1|\n",
      "+----------+\n",
      "\n",
      "+----------+\n",
      "|max(Count)|\n",
      "+----------+\n",
      "|      1922|\n",
      "+----------+\n",
      "\n",
      "+------------------+\n",
      "|        avg(Count)|\n",
      "+------------------+\n",
      "|29.155440596846848|\n",
      "+------------------+\n",
      "\n",
      "+-----------------+\n",
      "|    stddev(Count)|\n",
      "+-----------------+\n",
      "|81.17660451011595|\n",
      "+-----------------+\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Minimum number of Ratings a user has given.\n",
    "rat_user = userDF.select(userDF.PapersID).withColumn(\"Count\", size(userDF.PapersID))\n",
    "\n",
    "#Minimum\n",
    "rat_user.agg({\"Count\": 'min'}).show()\n",
    "\n",
    "#Maximum\n",
    "rat_user.agg({\"Count\": 'max'}).show()\n",
    "\n",
    "#Average\n",
    "rat_user.agg({\"Count\": 'avg'}).show()\n",
    "\n",
    "#Std\n",
    "rat_user.agg({\"Count\": 'std'}).show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+----------+\n",
      "|min(Count)|\n",
      "+----------+\n",
      "|         3|\n",
      "+----------+\n",
      "\n",
      "+----------+\n",
      "|max(Count)|\n",
      "+----------+\n",
      "|       924|\n",
      "+----------+\n",
      "\n",
      "+----------------+\n",
      "|      avg(Count)|\n",
      "+----------------+\n",
      "|4.81453867119172|\n",
      "+----------------+\n",
      "\n",
      "+-----------------+\n",
      "|    stddev(Count)|\n",
      "+-----------------+\n",
      "|5.477818208917285|\n",
      "+-----------------+\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Minimum rating an Item has received\n",
    "#min_rat_item = userDF2.select(userDF2.Paper_ID).withColumn(\"Count\", )\n",
    "\n",
    "min_rat_items = userDF2.groupBy(\"Paper_ID\").count()\n",
    "\n",
    "#Minimum\n",
    "min_rat_items.agg({\"Count\": 'min'}).show()\n",
    "\n",
    "#Maximum\n",
    "min_rat_items.agg({\"Count\": 'max'}).show()\n",
    "\n",
    "#Average\n",
    "min_rat_items.agg({\"Count\": 'avg'}).show()\n",
    "\n",
    "# Standard Deviation\n",
    "min_rat_items.agg({\"Count\": \"std\"}).show()\n",
    "#5.477818208917285"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "tryy = userRatingsRDD.take(10) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('f05bcffe7951de9e5a32fff4a42eb088',\n",
       "  ['1158654',\n",
       "   '478707',\n",
       "   '12054725',\n",
       "   '6670515',\n",
       "   '781057',\n",
       "   '13329754',\n",
       "   '781055',\n",
       "   '2825',\n",
       "   '553840',\n",
       "   '988447',\n",
       "   '12938139',\n",
       "   '12037219',\n",
       "   '12788583',\n",
       "   '6595566',\n",
       "   '4027225',\n",
       "   '920055',\n",
       "   '3129258',\n",
       "   '2242776',\n",
       "   '3112352',\n",
       "   '144287',\n",
       "   '706033',\n",
       "   '525396',\n",
       "   '622633',\n",
       "   '9172127',\n",
       "   '7357993',\n",
       "   '230211',\n",
       "   '12790816',\n",
       "   '503161',\n",
       "   '12937120',\n",
       "   '942241',\n",
       "   '166220',\n",
       "   '8493138',\n",
       "   '2945819',\n",
       "   '227173',\n",
       "   '11191048',\n",
       "   '949352',\n",
       "   '227096',\n",
       "   '11597926',\n",
       "   '921623',\n",
       "   '833638',\n",
       "   '882809',\n",
       "   '12738996',\n",
       "   '99',\n",
       "   '7154210',\n",
       "   '11852474',\n",
       "   '72879',\n",
       "   '10723139',\n",
       "   '11923443',\n",
       "   '2862386',\n",
       "   '802634',\n",
       "   '1959297',\n",
       "   '4488840',\n",
       "   '12735311',\n",
       "   '5102465',\n",
       "   '5177766',\n",
       "   '129',\n",
       "   '1467354',\n",
       "   '9396734',\n",
       "   '494336',\n",
       "   '1218954',\n",
       "   '2783692',\n",
       "   '3733678',\n",
       "   '8423325',\n",
       "   '1160828',\n",
       "   '507529',\n",
       "   '7804574',\n",
       "   '5201306',\n",
       "   '407124',\n",
       "   '504896',\n",
       "   '3281478',\n",
       "   '13687046',\n",
       "   '6486855',\n",
       "   '13207613',\n",
       "   '11847579',\n",
       "   '556147',\n",
       "   '10362464',\n",
       "   '6531882',\n",
       "   '714977',\n",
       "   '9946377',\n",
       "   '742892',\n",
       "   '3044720',\n",
       "   '1119267',\n",
       "   '7538620',\n",
       "   '1426283',\n",
       "   '2705980',\n",
       "   '9001253',\n",
       "   '2121165',\n",
       "   '12901647',\n",
       "   '10462916',\n",
       "   '261290',\n",
       "   '921399',\n",
       "   '6196237',\n",
       "   '5664256',\n",
       "   '213068',\n",
       "   '1769635',\n",
       "   '1055600',\n",
       "   '876703',\n",
       "   '7623038',\n",
       "   '1405472',\n",
       "   '1320137',\n",
       "   '4492925',\n",
       "   '1825619',\n",
       "   '13408130',\n",
       "   '1320157',\n",
       "   '682116',\n",
       "   '405672',\n",
       "   '1814546',\n",
       "   '12732269',\n",
       "   '12394780',\n",
       "   '1835552',\n",
       "   '9445526',\n",
       "   '80543',\n",
       "   '5413529',\n",
       "   '1153421',\n",
       "   '12925513',\n",
       "   '1784935',\n",
       "   '10802130',\n",
       "   '6305290',\n",
       "   '3856997',\n",
       "   '2097471',\n",
       "   '3748163',\n",
       "   '12796779',\n",
       "   '878326',\n",
       "   '10098153',\n",
       "   '3459771',\n",
       "   '827938',\n",
       "   '4448813',\n",
       "   '239',\n",
       "   '9478785',\n",
       "   '579614',\n",
       "   '7355647',\n",
       "   '423698',\n",
       "   '556149',\n",
       "   '13329593',\n",
       "   '9081047',\n",
       "   '151946',\n",
       "   '1453872',\n",
       "   '671699',\n",
       "   '3578102',\n",
       "   '137486',\n",
       "   '703464',\n",
       "   '9069826',\n",
       "   '622635',\n",
       "   '6792520',\n",
       "   '247',\n",
       "   '12716731',\n",
       "   '10045309',\n",
       "   '255030',\n",
       "   '968854',\n",
       "   '3726797',\n",
       "   '8217163',\n",
       "   '2945839',\n",
       "   '11839662',\n",
       "   '10685085',\n",
       "   '2308256',\n",
       "   '9983047',\n",
       "   '10138849',\n",
       "   '11536389',\n",
       "   '945310',\n",
       "   '1036492',\n",
       "   '6982661',\n",
       "   '1649749',\n",
       "   '1325105',\n",
       "   '6475030',\n",
       "   '488726',\n",
       "   '100186',\n",
       "   '3317637',\n",
       "   '336911',\n",
       "   '1321106',\n",
       "   '3956160']),\n",
       " ('28d3f81251d94b09735497477a5e4e02',\n",
       "  ['3929762',\n",
       "   '503574',\n",
       "   '5819422',\n",
       "   '4238883',\n",
       "   '5788061',\n",
       "   '462949',\n",
       "   '635215',\n",
       "   '635216',\n",
       "   '4810441',\n",
       "   '3481823',\n",
       "   '4165233',\n",
       "   '3366480',\n",
       "   '5984302',\n",
       "   '4238942',\n",
       "   '5490453',\n",
       "   '4636156',\n",
       "   '5996865',\n",
       "   '4194836',\n",
       "   '5828780',\n",
       "   '4450195']),\n",
       " ('d0c9aaa788153daeaf1f1538b3d46bbb',\n",
       "  ['2080631',\n",
       "   '6343346',\n",
       "   '5184704',\n",
       "   '7756088',\n",
       "   '2653863',\n",
       "   '6607628',\n",
       "   '4236212',\n",
       "   '1277953',\n",
       "   '226864',\n",
       "   '3140015',\n",
       "   '8806369',\n",
       "   '311570',\n",
       "   '5687747',\n",
       "   '767516',\n",
       "   '4781370',\n",
       "   '2841637',\n",
       "   '2445106',\n",
       "   '1959511',\n",
       "   '2688186',\n",
       "   '2363430',\n",
       "   '6614346',\n",
       "   '853030',\n",
       "   '5336762',\n",
       "   '4226226',\n",
       "   '239571',\n",
       "   '4089758',\n",
       "   '4140337',\n",
       "   '913868',\n",
       "   '7562861',\n",
       "   '3190274',\n",
       "   '2782576',\n",
       "   '12571584',\n",
       "   '2049617',\n",
       "   '5761055',\n",
       "   '5441098',\n",
       "   '3466838',\n",
       "   '2080691',\n",
       "   '1805577',\n",
       "   '7570111',\n",
       "   '5760287',\n",
       "   '2855355',\n",
       "   '3281547',\n",
       "   '1012525',\n",
       "   '3512183',\n",
       "   '678653']),\n",
       " ('ca4f1ba4094011d9a8757b1bfcadae5b', ['278019']),\n",
       " ('d1d41a152019155035ceb2db7d331c44',\n",
       "  ['6610569',\n",
       "   '6493797',\n",
       "   '6609079',\n",
       "   '7469737',\n",
       "   '7469738',\n",
       "   '6609102',\n",
       "   '6610585',\n",
       "   '6609240',\n",
       "   '7499869',\n",
       "   '7364976',\n",
       "   '7427963',\n",
       "   '7364991',\n",
       "   '7499833',\n",
       "   '7363519',\n",
       "   '7364990',\n",
       "   '6609245',\n",
       "   '7499801',\n",
       "   '7364971',\n",
       "   '7363487',\n",
       "   '7363498',\n",
       "   '7364986',\n",
       "   '7363513',\n",
       "   '7363526',\n",
       "   '6609222',\n",
       "   '7241881',\n",
       "   '6609169',\n",
       "   '7364933',\n",
       "   '6609099',\n",
       "   '1153734',\n",
       "   '7499794',\n",
       "   '3105791',\n",
       "   '6743243',\n",
       "   '6609238',\n",
       "   '7469728',\n",
       "   '7465494',\n",
       "   '7329626',\n",
       "   '1391614',\n",
       "   '1397150',\n",
       "   '7364987',\n",
       "   '7339380',\n",
       "   '7364973',\n",
       "   '7364823']),\n",
       " ('f2f77383828ea6d39438e525e40d54ba', ['943458', '238121', '763429']),\n",
       " ('9c883d02115400f7bab2c3f27d121d7c',\n",
       "  ['3509971',\n",
       "   '3509965',\n",
       "   '2336685',\n",
       "   '2081051',\n",
       "   '3365220',\n",
       "   '1595391',\n",
       "   '4174633',\n",
       "   '1734381',\n",
       "   '3878624']),\n",
       " ('b656009a6efdc8b1a43f5e25350392cf',\n",
       "  ['771870',\n",
       "   '181369',\n",
       "   '1216141',\n",
       "   '695889',\n",
       "   '2686454',\n",
       "   '2862856',\n",
       "   '2811674',\n",
       "   '2620767',\n",
       "   '1349894',\n",
       "   '1550295',\n",
       "   '1779811',\n",
       "   '901222',\n",
       "   '333403',\n",
       "   '1670361',\n",
       "   '740681',\n",
       "   '1216168',\n",
       "   '1345170',\n",
       "   '500841',\n",
       "   '1247499',\n",
       "   '622844',\n",
       "   '1141826',\n",
       "   '2939259',\n",
       "   '2625223',\n",
       "   '1220110',\n",
       "   '1871533',\n",
       "   '1299464',\n",
       "   '1343286',\n",
       "   '12254734',\n",
       "   '965334',\n",
       "   '1645745',\n",
       "   '2752879',\n",
       "   '816066',\n",
       "   '2240718',\n",
       "   '2752937',\n",
       "   '919232',\n",
       "   '2200230',\n",
       "   '300569',\n",
       "   '1074519',\n",
       "   '664041',\n",
       "   '1227212',\n",
       "   '1286088',\n",
       "   '2686494',\n",
       "   '1264228',\n",
       "   '898387',\n",
       "   '2610707',\n",
       "   '972902',\n",
       "   '2915212',\n",
       "   '2156871',\n",
       "   '1356989',\n",
       "   '822253',\n",
       "   '2388549',\n",
       "   '658201',\n",
       "   '2365209',\n",
       "   '913645',\n",
       "   '1746151',\n",
       "   '2249820',\n",
       "   '384364',\n",
       "   '1624397',\n",
       "   '1290791',\n",
       "   '1890690',\n",
       "   '3407406',\n",
       "   '600598',\n",
       "   '1938132',\n",
       "   '221390',\n",
       "   '2761647',\n",
       "   '7018499',\n",
       "   '1870439',\n",
       "   '117727',\n",
       "   '1217723',\n",
       "   '688160',\n",
       "   '298775',\n",
       "   '2719627',\n",
       "   '1450033',\n",
       "   '2045589',\n",
       "   '3032525',\n",
       "   '80546',\n",
       "   '305755',\n",
       "   '2752921',\n",
       "   '1356997',\n",
       "   '539299',\n",
       "   '2200182',\n",
       "   '90413',\n",
       "   '1453145',\n",
       "   '883747',\n",
       "   '1129585',\n",
       "   '404679',\n",
       "   '2045619',\n",
       "   '1279898',\n",
       "   '608358',\n",
       "   '761569',\n",
       "   '4395159',\n",
       "   '1597404',\n",
       "   '2150799',\n",
       "   '699815',\n",
       "   '941108',\n",
       "   '2811677',\n",
       "   '484851',\n",
       "   '1326856',\n",
       "   '2752882',\n",
       "   '2188733',\n",
       "   '1130952',\n",
       "   '1326268',\n",
       "   '1511759',\n",
       "   '1815955']),\n",
       " ('cf9c7f356092c34be14d1d35407b26f2', ['90558']),\n",
       " ('0f5cbb39410a9278f535dd027a511fa3', ['9344598'])]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tryy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "t = sqlC.createDataFrame(tryy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------------+--------------------+\n",
      "|                  _1|                  _2|\n",
      "+--------------------+--------------------+\n",
      "|f05bcffe7951de9e5...|[1158654, 478707,...|\n",
      "|28d3f81251d94b097...|[3929762, 503574,...|\n",
      "|d0c9aaa788153daea...|[2080631, 6343346...|\n",
      "|ca4f1ba4094011d9a...|            [278019]|\n",
      "|d1d41a15201915503...|[6610569, 6493797...|\n",
      "|f2f77383828ea6d39...|[943458, 238121, ...|\n",
      "|9c883d02115400f7b...|[3509971, 3509965...|\n",
      "|b656009a6efdc8b1a...|[771870, 181369, ...|\n",
      "|cf9c7f356092c34be...|             [90558]|\n",
      "|0f5cbb39410a9278f...|           [9344598]|\n",
      "+--------------------+--------------------+\n",
      "\n"
     ]
    }
   ],
   "source": [
    "t.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
